---
title: 'Trisomy 12 Gene Dosage'
author: "Holly Giles"
date: "`r doc_date()`"
output:
  BiocStyle::html_document
---

To investigate the role of Trisomy 12 as an enhancer of response to ME stimulation: 

Aim: 
1. Find which components of key ME signalling pathways can be found on Chromosome 12     
2. Compare RNA and Protein abundances for these components    

# Set up

## Libraries
```{r setup, include=FALSE}

#Load libraries
library(msigdbr)
library(ggbeeswarm)
library(ggpubr)
library(MultiAssayExperiment)
library(dplyr)
library(ggplot2)
library(tidyr)
library(DESeq2)
library(biobroom)
library(biomartr)
library(biomaRt)
library(patchwork)

```

## Themes 
```{r}
### ggplot themes
fontsize = 11

## theme for ggplots
t1 <- 
  theme(                              
  plot.background = element_blank(), 
  panel.grid.major = element_line(),
  panel.grid.major.x = element_line(linetype = "dotted", colour = "grey"),
  panel.grid.minor = element_blank(), 
  panel.border = element_blank(), 
  panel.background = element_blank(),
  axis.line = element_line(size=.4),
  axis.line.x = element_line(),
  axis.line.y = element_line(),
  axis.text.x  = element_text(angle=90, size=11, hjust = 1, vjust = 0.4),
  axis.text.y = element_text(size = 11),
  axis.ticks.length = unit(0.3,"cm"),
  axis.title.x = element_text(face="bold", size=13), 
  axis.title.y = element_text(face="bold", size=13),
  plot.title = element_text(face="bold", size=14, hjust = 0.5),
  strip.text = element_text(size = fontsize)
)

t2 <- t1+
  theme( axis.text.x  = element_text(angle=0, size=11, hjust = 0.5, vjust = 1))

## theme for legends
t.leg <-  theme(legend.title = element_text(face='bold', 
                                            hjust = 1, size=11),
                legend.key = element_blank(),
                legend.text = element_text(size=11),
                legend.background = element_rect(color = "black"))



### Set colour palettes

#For Categorical: 
colors <- c("#A1BE1F", #green
            "#F4C61F", #yellow
            "#734595", #purple
            "#D41645", #red
            "#3B6FB6", #blue
            "#B65417", #orange
            "#E2E868", #light green
            "#CBA3D8", #light purple
            "#E58F9E", #light purple
            "#8BB8E8", #light blue
            "#F49E17", #light orange
            "#303030", #black
            "#A8A99E", #grey
            "#007B53") #dark green



#For Divergent: 
Divergent <- c("#003DA5", "#2055B0", "#406EBC", "#6086C7", "#809ED2", "#9FB6DD", "#BFCFE9", "#DFE7F4", "white", "white", "white","#F4E0E7", "#E9C2CF", "#DEA3B6", "#D3849E", "#C76586", "#BC476E", "#B12855", "#A6093D")

#for negatives only: 
palblues <- c("#003DA5", "#2055B0", "#406EBC", "#6086C7", "#809ED2", "#9FB6DD", "#BFCFE9", "#DFE7F4")

#for positives only:
palreds <- c("#F4E0E7", "#E9C2CF", "#DEA3B6", "#D3849E", "#C76586", "#BC476E", "#B12855", "#A6093D")

#For mutations: 
Mutant <- c("#b5b5b5","#373A36")
Sex <- c("#707372","#D0D0CE")
IGHV <- c("#373A36","#D0D0CE")
Methylation_cluster <- c("#373A36","#A8A99E","#D0D0CE")

#For drugs: 
drugpal <- c("#734595", "#CBA3D8") #purples

#For cytokines: 
cytpal <- c("#F49E17", "#EFC06E") #yellows


#neutral
offwhite <- "#f8f8ff"
lightergrey <- "#D0D0CE"
darkergrey <- "#707372"


na_color="#f0f0f0"



```

## Data
```{r warning=FALSE, message = FALSE}

##########LOAD ALL FILES################

#Data
#df: tibble containing all screening viability data
load("../../data/df.RData")
df_complete <- df

#patMeta: tibble containing all patient genetic data
load("../../data/patMeta.RData")

#RNA
load("../../data/dds_smp.RData")

#Load proteomics data
load("../../data/multiomics_MAE.RData")

#get table to convert between IDS
ensembl = useEnsembl(biomart="ensembl", dataset="hsapiens_gene_ensembl", version="90")
transAnno <- getBM(attributes=c('ensembl_gene_id','hgnc_symbol','chromosome_name', 'transcript_biotype'), mart = ensembl)
transAnno <- dplyr::filter(transAnno, hgnc_symbol != "")
id2gene <- dplyr::select(transAnno, ensembl_gene_id, hgnc_symbol)

```

# Extract genesets
Get TGF beta, IL4 and TLR genes on chr12
```{R}

tgf_beta_genes <- 
  msigdbr(species = "Homo sapiens", category = "C2", subcategory = "CP:KEGG") %>% 
  #select pathway name and entrez id 
  dplyr::select(gs_name, entrez_gene) %>%
  filter(gs_name == "KEGG_TGF_BETA_SIGNALING_PATHWAY")

#get ens ids
tgf_beta_genes <- getBM(
  attributes = c('ensembl_gene_id','entrezgene', 'hgnc_symbol', "chromosome_name"), 
  filters = 'entrezgene',
  values = tgf_beta_genes$entrez_gene,
  mart = ensembl)


#get chromosome info on tgfbeta genes
tgf_beta_12 <- getBM(
  attributes = c('ensembl_gene_id','entrezgene', 'hgnc_symbol', "chromosome_name"), 
  filters = 'entrezgene',
  values = tgf_beta_genes$entrezgene,
  mart = ensembl
) %>% filter(chromosome_name == 12)

il4_genes <- 
  msigdbr(species = "Homo sapiens", category = "C2", subcategory = "CP:KEGG") %>% 
  #select pathway name and entrez id 
  dplyr::select(gs_name, entrez_gene) %>%
  filter(gs_name == "KEGG_JAK_STAT_SIGNALING_PATHWAY")

#get ens ids
il4_genes <- getBM(
  attributes = c('ensembl_gene_id','entrezgene', 'hgnc_symbol', "chromosome_name"), 
  filters = 'entrezgene',
  values = il4_genes$entrez_gene,
  mart = ensembl)

#get chromosome info on tgfbeta genes
il4_12 <- getBM(
  attributes = c('ensembl_gene_id','entrezgene', 'hgnc_symbol', "chromosome_name"), 
  filters = 'entrezgene',
  values = il4_genes$entrezgene,
  mart = ensembl
) %>% filter(chromosome_name == 12)

tlr_genes <- 
  msigdbr(species = "Homo sapiens", category = "C2", subcategory = "CP:KEGG") %>% 
  #select pathway name and entrez id 
  dplyr::select(gs_name, entrez_gene) %>%
  filter(gs_name == "KEGG_TOLL_LIKE_RECEPTOR_SIGNALING_PATHWAY")

#get ens ids
tlr_genes <- getBM(
  attributes = c('ensembl_gene_id','entrezgene', 'hgnc_symbol', "chromosome_name"), 
  filters = 'entrezgene',
  values = tlr_genes$entrez_gene,
  mart = ensembl)

#get chromosome info on tgfbeta genes
tlr_12 <- getBM(
  attributes = c('ensembl_gene_id','entrezgene', 'hgnc_symbol', "chromosome_name"), 
  filters = 'entrezgene',
  values = tlr_genes$entrezgene,
  mart = ensembl
) %>% filter(chromosome_name == 12)

```

# Run Deseq 
Compare tri 12 versus non-tri12
```{r rnaPrep}
cytSeq <- dds_smp


#split into 3,4
cytSeq <- cytSeq[,colData(cytSeq)[,"trisomy12"] %in% c(0,1)]

#set order of factors
cytSeq$Cluster <- factor(cytSeq$trisomy12, 
                            levels = c("0","1"))


design(cytSeq) <- ~ trisomy12

```

```{r RunDeseq}


cytSeq <- DESeq(cytSeq)

```


```{r  deseqResults}

res.ds <- 
  results(cytSeq, contrast = c("trisomy12", 0, 1), tidy = TRUE) %>%
  dplyr::rename(Symbol = "row") %>% 
  dplyr::arrange(pvalue) 

```

# Inspect Deseq results
See which genes from each pathway are differentiall expressed  
```{r}
#TGF beta

tgfall <- filter(res.ds, Symbol %in% tgf_beta_genes$ensembl_gene_id, padj <0.1) %>% left_join(tgf_beta_genes, by = c("Symbol" = "ensembl_gene_id"))

tgf12 <- filter(res.ds, Symbol %in% tgf_beta_12$ensembl_gene_id, padj <0.1) %>% left_join(tgf_beta_genes, by = c("Symbol" = "ensembl_gene_id"))

DT::datatable(tgfall)
DT::datatable(tgf12)

```

```{r}

#IL4
il4all <- filter(res.ds, Symbol %in% il4_genes$ensembl_gene_id, padj <0.1) %>% left_join(il4_genes, by = c("Symbol" = "ensembl_gene_id"))

il412 <- filter(res.ds, Symbol %in% il4_12$ensembl_gene_id, padj <0.1) %>% left_join(il4_genes, by = c("Symbol" = "ensembl_gene_id"))


DT::datatable(il4all)
DT::datatable(il412)
```

```{r}
#TLR 
tlrall <- filter(res.ds, Symbol %in% tlr_genes$ensembl_gene_id, padj <0.1) %>% left_join(tlr_genes, by = c("Symbol" = "ensembl_gene_id"))

#TLR 
tlr12 <- filter(res.ds, Symbol %in% tlr_12$ensembl_gene_id, padj <0.1) %>% left_join(tlr_genes, by = c("Symbol" = "ensembl_gene_id"))

DT::datatable(tlrall)
DT::datatable(tlr12)
```


# Plot interesting genes
## TGF beta
```{r fig.height = 5}
# ifn gamma and SP1, faceted by RNA and proteomics
ifn_spi1_rna <- 
lapply(tgf12$Symbol, function(x){
    #Prep data for plot
    data <- plotCounts(dds_smp, gene=x, intgroup=c("trisomy12"), returnData=TRUE) %>%
      filter(trisomy12 %in% c("0", "1"))
    
    #get hgnc symbol from ENS id
    gene_name <- dplyr::select(filter(tgf_beta_genes, ensembl_gene_id == x), hgnc_symbol)
    
    #Make plot
    ggplot(data, aes(x = trisomy12, y = count, color = trisomy12)) +
      geom_boxplot() +
      geom_beeswarm() + 
      ggtitle(gene_name) +
      ylab("Counts") +
      xlab("Trisomy 12 Status") +
      
      t1 +
      scale_color_manual(values = c(colors[1], colors[2])) +
      theme(aspect.ratio=1,  text = element_text(size=15), legend.position = "none") +
      
      stat_compare_means(method = "t.test", comparisons = list(c("1", "0")), size=5)
  
})

#get names of proteins and cross reference in proteomics
allproteins <- rownames(MultiAssayExperiment::assays(multiomics_MAE)[["proteomics"]])
proteins <- tgf12$hgnc_symbol[tgf12$hgnc_symbol%in% allproteins]

ifn_spi1_pro = lapply(proteins, function(x){
     #Prepare data for plot
     df_tbl_prot <- wideFormat(multiomics_MAE[x,,"proteomics"] ) %>% as_tibble()
     colnames(df_tbl_prot)[2] <- "protein"
     
     mae_mini <- multiomics_MAE[,,c("SNPs", "chrom_abber", "health_record_bin")]
     df_tbl_SNP <- wideFormat(mae_mini["trisomy12",,]) %>% as_tibble()
     colnames(df_tbl_SNP)[2] <- "alteration"
     df_tbl_SNP <- df_tbl_SNP %>% mutate(alteration=factor(alteration))
     
     df_tbl <- left_join(df_tbl_prot, df_tbl_SNP, by="primary" )
     
     #Make plot
      gg = df_tbl %>% filter(!is.na(alteration), !is.na(protein)) %>% ggplot(aes(alteration, protein, colour = alteration, group=alteration)) +
       scale_y_log10() +
       geom_boxplot() + 
       geom_point(position=position_jitter(width=.1,height=0), size=3)+
        
       ggtitle(x)+
       ylab("Protein abundance") +
       xlab("Trisomy 12 Status") + 
        
       theme_bw()+  
       scale_colour_manual(values = c(colors[1],colors[2])) + 
       t2 +  
       theme(aspect.ratio=1,  text = element_text(size=15), legend.position = "none") +
       
       stat_compare_means(method = "t.test", comparisons = list(c("1", "0")), size=5)
  
})

wrap_elements(ifn_spi1_rna[[1]]) + wrap_elements(ifn_spi1_rna[[2]]) + wrap_elements(ifn_spi1_pro[[1]])

# SMAD 3 RNA and proteomics 
    #Prep data for plot
    data <- plotCounts(dds_smp, gene="ENSG00000166949", intgroup=c("trisomy12"), returnData=TRUE) %>%
      filter(trisomy12 %in% c("0", "1"))
    
    
    #Make plot
    SMAD3rna <- 
      ggplot(data, aes(x = trisomy12, y = count, color=trisomy12)) +
      geom_boxplot() +
      geom_beeswarm() + 
      ggtitle("SMAD3") +
      ylab("Counts") +
      xlab("Trisomy 12 Status") +
      
      t2 +
      scale_color_manual(values = c(colors[1], colors[2])) +
      theme(aspect.ratio=1,  text = element_text(size=15), legend.position = "none") +
      
      stat_compare_means(method = "t.test", comparisons = list(c("1", "0")), size=5)
 
    
     
 df_tbl_prot <- wideFormat(multiomics_MAE["SMAD3",,"proteomics"] ) %>% as_tibble()
     colnames(df_tbl_prot)[2] <- "protein"
     
     mae_mini <- multiomics_MAE[,,c("SNPs", "chrom_abber", "health_record_bin")]
     df_tbl_SNP <- wideFormat(mae_mini["trisomy12",,]) %>% as_tibble()
     colnames(df_tbl_SNP)[2] <- "alteration"
     df_tbl_SNP <- df_tbl_SNP %>% mutate(alteration=factor(alteration))
     
     df_tbl <- left_join(df_tbl_prot, df_tbl_SNP, by="primary" )
     
     #Make plot
     SMAD3pro = df_tbl %>% filter(!is.na(alteration), !is.na(protein)) %>% ggplot(aes(alteration, protein, colour = alteration, group=alteration)) +
       scale_y_log10() +
       geom_boxplot() + 
       geom_point() +
        
       ggtitle("SMAD3")+
       ylab("Protein abundance") +
       xlab("Trisomy 12 Status") + 
        
       theme_bw()+  
       scale_colour_manual(values = c(colors[1],colors[2])) + 
       t2 +  
       theme(aspect.ratio=1,  text = element_text(size=15), legend.position = "none") +
       stat_compare_means(method = "t.test", comparisons = list(c("1", "0")), size=5)
  
     
     SMAD3rna + SMAD3pro
  
```


## IL4
```{R}

# STAT2 and STAT6

stat_rna <- 
lapply(c("ENSG00000166888", "ENSG00000170581"), function(x){
    #Prep data for plot
    data <- plotCounts(dds_smp, gene=x, intgroup=c("trisomy12"), returnData=TRUE) %>%
      filter(trisomy12 %in% c("0", "1"))
    
    #get hgnc symbol from ENS id
    gene_name <- dplyr::select(filter(il4_genes, ensembl_gene_id == x), hgnc_symbol)
    
    #Make plot
    ggplot(data, aes(x = trisomy12, y = count, color=trisomy12)) +
      geom_boxplot() +
      geom_beeswarm() + 
      ggtitle(gene_name) +
      ylab("Counts") +
      xlab("") +
      
      t2 +
      scale_color_manual(values = c(colors[1], colors[2])) +
      theme(aspect.ratio=1,  text = element_text(size=15), legend.position = "none") +
      
      stat_compare_means(method = "t.test", comparisons = list(c("1", "0")), size=5)
  
})


stat_pro = lapply(c("STAT6", "STAT2"), function(x){
     #Prepare data for plot
     df_tbl_prot <- wideFormat(multiomics_MAE[x,,"proteomics"] ) %>% as_tibble()
     colnames(df_tbl_prot)[2] <- "protein"
     
     mae_mini <- multiomics_MAE[,,c("SNPs", "chrom_abber", "health_record_bin")]
     df_tbl_SNP <- wideFormat(mae_mini["trisomy12",,]) %>% as_tibble()
     colnames(df_tbl_SNP)[2] <- "alteration"
     df_tbl_SNP <- df_tbl_SNP %>% mutate(alteration=factor(alteration))
     
     df_tbl <- left_join(df_tbl_prot, df_tbl_SNP, by="primary" )
     
     #Make plot
      gg = df_tbl %>% filter(!is.na(alteration), !is.na(protein)) %>% ggplot(aes(alteration, protein, colour = alteration, group=alteration)) +
       geom_boxplot() + 
       geom_point() +
        
       ggtitle("")+
       ylab("Protein abundance") +
       xlab("Trisomy 12 Status") + 
        
       theme_bw()+  
       scale_colour_manual(values = c(colors[1],colors[2])) + 
       t2 +  
       theme(aspect.ratio=1,  text = element_text(size=15), legend.position = "none") +
       stat_compare_means(method = "t.test", comparisons = list(c("1", "0")), size=5)
  
})

(stat_rna[[1]] + stat_rna[[2]])/
  (stat_pro[[1]] + stat_pro[[2]])



```

## TLR 
```{r}


#IRAK4

irak4_rna <- 
lapply(c("ENSG00000198001","ENSG00000172936"),  function(x){
    #Prep data for plot
    data <- plotCounts(dds_smp, gene=x, intgroup=c("trisomy12"), returnData=TRUE) %>%
      filter(trisomy12 %in% c("0", "1"))
    
    #get hgnc symbol from ENS id
    gene_name <- dplyr::select(filter(tlr_genes, ensembl_gene_id == x), hgnc_symbol)
    
    #Make plot
    ggplot(data, aes(x = trisomy12, y = count, color=trisomy12)) +
      geom_boxplot() +
      geom_beeswarm() + 
      ggtitle(gene_name) +
      ylab("Counts") +
      xlab("") +
      
      t2 +
      scale_color_manual(values = c(colors[1], colors[2])) +
      theme(aspect.ratio=1,  text = element_text(size=15), legend.position = "none") +
      
      stat_compare_means(method = "t.test", comparisons = list(c("1", "0")), size=5)
  
})


irak4_pro = lapply(c("IRAK4", "MYD88"), function(x){
     #Prepare data for plot
     df_tbl_prot <- wideFormat(multiomics_MAE[x,,"proteomics"] ) %>% as_tibble()
     colnames(df_tbl_prot)[2] <- "protein"
     
     mae_mini <- multiomics_MAE[,,c("SNPs", "chrom_abber", "health_record_bin")]
     df_tbl_SNP <- wideFormat(mae_mini["trisomy12",,]) %>% as_tibble()
     colnames(df_tbl_SNP)[2] <- "alteration"
     df_tbl_SNP <- df_tbl_SNP %>% mutate(alteration=factor(alteration))
     
     df_tbl <- left_join(df_tbl_prot, df_tbl_SNP, by="primary" )
     
     #Make plot
      gg = df_tbl %>% filter(!is.na(alteration), !is.na(protein)) %>% ggplot(aes(alteration, protein, colour = alteration, group=alteration)) +
       geom_boxplot() + 
       geom_point() +
        
       ggtitle("")+
       ylab("Protein abundance") +
       xlab("Trisomy 12 Status") + 
        
       theme_bw()+  
       scale_colour_manual(values = c(colors[1],colors[2])) + 
       t2 +  
       theme(aspect.ratio=1,  text = element_text(size=15), legend.position = "none") +
       stat_compare_means(method = "t.test", comparisons = list(c("1", "0")), size=5) 
  
})

(irak4_rna[[1]] + irak4_rna[[2]])/
  (irak4_pro[[1]] + irak4_pro[[2]])

```



