\documentclass[11pt, a4paper, twosided]{book}
\usepackage[hmarginratio=1:1]{geometry} % equal right and left raitios
\usepackage{amsmath,amssymb,amsfonts,amsthm}    % Typical maths resource packages
\usepackage{graphicx}                           % Packages to allow inclusion of graphics
\usepackage[authoryear]{natbib}                 % literature reference style
\usepackage[bf]{caption}
\usepackage{textcomp}                           % For single quotes
\usepackage{floatrow}                           % For image and table position
\usepackage{booktabs}                           % For tables
% \usepackage[colorlinks=true]{hyperref}
% \usepackage[bottom]{footmisc}
\usepackage[bottom, flushmargin]{footmisc}                   % For footnotes
\usepackage[citebordercolor={0 1 0}]{hyperref}  % For creating hyperlinks in cross references
\usepackage{footnotebackref}
\usepackage[utf8]{inputenc}
\usepackage[T1]{fontenc}
% i have added the below, so that kable works
\usepackage{colortbl}
\usepackage{longtable}
\usepackage{array}
\usepackage{multirow}
\usepackage{wrapfig}
\usepackage{float}
\usepackage{pdflscape}
\usepackage{tabu}
\usepackage{threeparttable}
\usepackage{threeparttablex}
\usepackage[normalem]{ulem}
\usepackage{makecell}
\usepackage{xcolor}
\usepackage[gen]{eurosym}
\usepackage{breqn} %to break up equations accross multiple lines, where neccesary 
\usepackage{helvet} %to set font as helvetica 

\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.77,0.63,0.00}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
       % For being able to set `echo = TRUE` in code chunk options, and thus in-line representation of code

%--------------------------------------------
% define line spacing = 1.5
\renewcommand{\baselinestretch}{1.2}

%define font
\renewcommand{\familydefault}{\sfdefault}

% define position of graphics
\floatsetup[figure]{capposition=bottom}
\floatsetup[table]{capposition=top}
\floatplacement{figure}{ht}
\floatplacement{table}{ht}

% define CSL references (required)
\newlength{\cslhangindent}
\setlength{\cslhangindent}{1.5em}
\newenvironment{CSLReferences}%
  {}%
  {\par}

% save thesis parameters for later
\newcommand{\thesistype}{Dissertation}
\newcommand{\thesisauthor}{Holly Amelia Rebecca Giles}
\newcommand{\thesisdate}{Dec 05, 2021}

% Added by CII (Thanks, Hadley!)
% Use ref for internal links
\renewcommand{\hyperref}[2][???]{\autoref{#1}}
\def\chapterautorefname{Chapter}
\def\sectionautorefname{Section}
\def\subsectionautorefname{Subsection}
% End of CII addition

% Added by CII
\usepackage{caption}
\captionsetup{width=5in,font={small}}
% End of CII addition

% Added by CII
%%% Copied from knitr
%% maxwidth is the original width if it's less than linewidth
%% otherwise use linewidth (to make sure the graphics do not exceed the margin)
\makeatletter
\def\maxwidth{ %
  \ifdim\Gin@nat@width>\linewidth
    \linewidth
  \else
    \Gin@nat@width
  \fi
}
\makeatother

\renewcommand{\contentsname}{Table of Contents}
% End of CII addition

\setlength{\parskip}{0pt}

% Added by CII
  %\setlength{\parskip}{\baselineskip}
  \usepackage[parfill]{parskip}

\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}

	\usepackage{float}
% End of CII addition
%%
%% End Preamble
%%
%
\begin{document}

% ------------------------------------
% ---- frontmatter: First Page -------
% ------------------------------------

\thispagestyle{empty}
  \begin{center}
  {\Large{Dissertation\\
    submitted to the\\
    Combined Faculty of Natural Sciences and Mathematics\\
    of the Ruperto Carola University Heidelberg, Germany\\
    for the degree of\\
    Doctor of Natural Sciences\\}} \vspace*{\fill}

  \Large{Presented by} \\\vspace{0.5cm}
  \Large{M.A. Holly Amelia Rebecca Giles} \\
  \Large{born in Portsmouth} \\
  \Large{Oral examination: Feb 20, 2021}

\end{center}
\newpage\null\thispagestyle{empty}\newpage

% ------------------------------------
% --- frontmatter: Secound Page ------
% ------------------------------------

\pagebreak
\thispagestyle{empty}
\null
\begin{center}
\Large{Drug - microenvironment - gene interplay in Chronic Lymphocytic Leukemia} \\
\vspace*{\fill}
\begin{tabular}{ c c }
Referees: & Dr.~Judith Zaugg \\
          & Prof.~Dr.~Michael Boutros
\end{tabular}
\end{center}
\newpage\null\thispagestyle{empty}\newpage

% --------------------------------------------
% -------- Declaration of Authorship ---------
% --------------------------------------------

\newpage
\thispagestyle{empty}
% \hypertarget{declaration-of-authorship}{%
% \section*{Declaration of Authorship}\label{declaration-of-authorship}}
%
% I hereby confirm that I have authored this Dissertation independently and
% without the use of others sources than the ones indicated. I have not yet
% presented this thesis or parts thereof to a university as part of an
% examination or degree. This work was carried out in the Genome Biology at
% the European Molecular Biology Laboratory (EMBL) in the group of Dr.~Wolfgang Huber and Prof.~Dr.~Sascha Dietrich.
% \vspace{1cm}
%
% \noindent Heidelberg, \thesisdate{}
% \vspace{3cm}
%
% \noindent . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
% \vspace{0.1cm}
%
% \noindent \thesisauthor{}
% \newpage
% \thispagestyle{empty}
% \vspace*{\fill}
% Holly Amelia Rebecca Giles, \textit{Drug - microenvironment - gene interplay in Chronic Lymphocytic Leukemia}, \textcopyright\, \thesisdate{}
% \newpage

%\frontmatter % this stuff will be roman-numbered
\pagestyle{empty} % this removes page numbers from the frontmatter



% edit style below this point? 


% --------------------------------------------
% --------------- Dedication -----------------
% --------------------------------------------
\vspace*{\fill}
\begin{center}
\textit{To Mum, Dad and Milly x}
\end{center}
\vspace*{\fill}
\newpage\null\thispagestyle{empty}\newpage

% ------------------------------------
% --- frontmatter: Acknowledgement ---
% ------------------------------------
\pagestyle{plain}
\pagenumbering{roman}   %  define page number in roman style
\setcounter{page}{1}    % start page numbering anew
\newpage
\hypertarget{acknowledgements}{%
\chapter*{Acknowledgements}\label{acknowledgements}}
\addcontentsline{toc}{chapter}{Acknowledgements}

I want to thank a few people.
\newpage\null\newpage


% --------------------------------------------
% ---------- List of publications ------------
% --------------------------------------------
    \hypertarget{list-of-publications}{%
    \chapter*{List of publications}\label{list-of-publications}}
    \addcontentsline{toc}{chapter}{List of publications}

    \hypertarget{thesis-related}{%
    \subsubsection*{Thesis related}\label{thesis-related}}
    \addcontentsline{toc}{subsubsection}{Thesis related}
    \begin{itemize}
        \item  Peter-Martin Bruch\ast,  \textbf{\underline{Holly A. R. Giles}}\ast, Carolin Kolb, Sophie Herbst, Tina Becirovic, Tobias Roider, Junyan Lu, Sebastian Scheinost, Lena Wagner, Jennifer Huellein, Ivan Berest, Mark Kriegsmann, Katharina Kriegsmann, Christiane Zgorzelski, Peter Dreger, Judith Zaugg, Carsten Mueller-Tidow, Thorsten Zenz, Wolfgang Huber, Sascha Dietrich et al. \textit{in preparation.}  "Mapping drug-microenvironment-genetic interplay in CLL reveals trisomy 12 as a modulator of microenvironment."\textbf{\textit{bioRxiv}}. [doi: 10.1101/2021.07.23.453514](https://doi.org/10.1101/2021.07.23.453514)
    \end{itemize}
    \hypertarget{other-contributions}{%
    \subsubsection*{Other contributions}\label{other-contributions}}
    \addcontentsline{toc}{subsubsection}{Other contributions}

    The author of this thesis also contributed to a number of other projects throughout the PhD. The following have been published:
    \begin{itemize}
        \item Berest, Ivan\ast, Christian Arnold\ast, Armando Reyes-Palomares, Giovanni Palla, Kasper Dindler Rasmussen, \textbf{\underline{Holly Giles}}, and Peter-Martin Bruch et al. 2019. "Quantification Of Differential Transcription Factor Activity And Multiomics-Based Classification Into Activators And Repressors: DiffTF". \textit{Cell Reports} 29 (10): 3147-3159.e12. [doi: 10.1016/j.celrep.2019.10.106](https://doi.org/10.1016/j.celrep.2019.10.106)
        
      \item Lu, Junyan\ast, Ester Cannizzaro\ast, Fabienne Meier-Abt, Sebastian Scheinost, Peter-Martin Bruch, \textbf{\underline{Holly A. R. Giles}}, and Almut Lütge et al. 2021. "Multi-Omics Reveals Clinically Relevant Proliferative Drive Associated With Mtor-MYC-OXPHOS Activity In Chronic Lymphocytic Leukemia". \textbf{\textit{Nature Cancer}}. [doi: 10.1038/s43018-021-00216-6](https://doi.org/10.1038/s43018-021-00216-6)
    \end{itemize}
\newpage\null\newpage

% -------------------------------------
% --- frontmatter: Zusammendfassung ---
% -------------------------------------
    \hypertarget{zusammenfassung}{%
    \chapter*{Zusammenfassung}\label{zusammenfassung}}
    \addcontentsline{toc}{chapter}{Zusammenfassung}

    Abstract in German
    Aktuelle Entwicklungen im Bereich der ``omik''-Technologien tragen wesentlich zurBeschleunigung des Fortschritts in der Krebsforschung bei etc
\newpage\null\newpage

% -----------------------------
% --- frontmatter: Abstract ---
% -----------------------------

    \hypertarget{abstract}{%
    \chapter*{Abstract}\label{abstract}}
    \addcontentsline{toc}{chapter}{Abstract}

    The tumour microenvironment and genetic alterations collectively influence disease biology and drug resistance in Chronic Lymphocytic Leukaemia (CLL). To establish an integrative understanding of these factors in CLL biology, we performed a combinatorial assay using 12 drugs individually co-applied with each of 17 microenvironmental stimuli on 192 samples of CLL peripheral blood mononucleated cells. We examined microenvironment response across a heterogeneous patient cohort and identified four distinct CLL subgroups that differed in their response landscapes and in patient outcomes. By combining our data with whole-exome sequencing, DNA-methylation, RNA-sequencing and copy number variant data of the same tumours, we systematically searched for molecular determinants of stimulus response and found trisomy 12 as a key modulator. Our data suggest that the amplifying effect of trisomy 12 on the response to environmental signals is mediated by the transcription factors Spi-B and PU.1. We generated a comprehensive map of drug-microenvironment interactions in CLL, and profiled the modulating impact of genetic features on these antagonistic and synergistic effects. Interleukin (IL) 4 and Toll-Like Receptor (TLR) 7/8/9 stimuli showed the most interactions. Both pathways were more active in CLL-infiltrated lymph nodes than in healthy samples (p\textless0.001), and high IL4 activity in lymph nodes correlated with shorter survival (p=0.038). We provide a multi-layered resource to investigate microenvironmental and drug interplay in CLL (\href{https://git.embl.de/giles/cllcytokinescreen2021}{Repository} \& \href{https://www.google.com/url?q=https://www.imbi.uni-heidelberg.de/dietrichlab/CLL_Microenvironment/\&sa=D\&source=editors\&ust=1625764640786000\&usg=AOvVaw01HOFP7k2BJRLTteCXa8qy}{Shiny}). Our results highlight the importance of cell-extrinsic influences on drug response and disease progression, and how these further depend on molecular features.
\newpage\null\newpage

% -----------------------------
% --- frontmatter: Contents ---
% -----------------------------
\newpage
\tableofcontents
\clearpage

% ----------------------------------------------------------
% --- frontmatter: List of Abbreviations (not mandatory) ---
% ----------------------------------------------------------
\newpage
\hypertarget{list-of-abbreviations}{%
\chapter*{List of Abbreviations}\label{list-of-abbreviations}}
\addcontentsline{toc}{chapter}{List of Abbreviations}
\begin{tabular}{rp{0.2cm}lp{1cm}rp{0.2cm}l}
    AML & & Acute myeloid leukaemia \\
    AKT & & Protein kinase B \\
    ATACseq & & Assay for Transposase-Accessible Chromatin using sequencing \\
    ATP & & Adenosine triphosphate \\
    AUC & & Area under curve \\
    BAFF & & B cell-activating factor \\
    BCR & & B-cell receptor \\
    BH & & Benjamini-Hochberg \\
    BMSC & & Mesenchymal bone marrow stromal cell \\
    BTK & & Bruton’s tyrosine kinase \\
    CD40L & & CD40 Ligand \\
    CDF & & Cumulative Distribution Function \\
    ChIPseq & & Chromatin immunoprecipitation sequencing \\
    CLL & & Chronic lymphocytic leukemia \\
    CNV & & Copy number variation \\
    CpG ODN & & CpG oligodeoxynucleotides \\
    DMSO & & Dimethyl sulfoxide  \\
    DNA & & Deoxyribonucleic acid  \\
    DNAseq & & DNA sequencing \\
    ERK & & Ras-dependent extracellular signal-regulated kinase \\
    FBS & & Fetal bovine serum \\
    FDC & & Follicular dendritic cell \\
    FDR & & False discovery rate \\
    GSEA & & Gene set enrichment analysis  \\
    IFN$\gamma$ & & Interferon$\gamma$   \\ 
    
\end{tabular}
\begin{tabular}{rp{0.2cm}lp{1cm}rp{0.2cm}l}
    
    
  Ig & & Immunoglobulin \\
    IGHV & & Immunoglobulin heavy chain variable region \\
    IL & & Interleukin \\
    JAK & & Janus kinase \\
    M-CLL & & CLL with somatic hypermutations in IGHV loci \\
    U-CLL & & CLL without somatic hypermutations in IGHV loci \\
    MAPK & & Mitogen-activated protein kinase  \\
    MRD & & Minimal residual disease  \\
    MSCs & & Mesenchymal stromal cells  \\
    NF$\kappa$B & & Nuclear factor kappa-light-chain-enhancer of activated B cells \\
    NLC & & nurselike cell  \\
    NOTCH & & Neurogenic locus notch  \\
    OS & & Overall survival \\
    PBMC & & Peripheral blood mononuclear cells \\
    PCA & & Principal component analysis \\
    PKC & & Protein kinase C \\
    PLC-$\gamma$2 & & phospholipase C $\gamma$ 2 \\
    PI3K & & Phosphoinositide 3-kinase \\
    RNA & & Ribonucleic acid  \\
    RNAseq & & RNA sequencing \\
    SDF1$\alpha$ & & Stromal Cell-Derived Factor-1 alpha  \\
    SF3B1 & & Splicing factor 3B subunit 1 \\
    STAT & & Signal transducer and activator of transcription \\
    
\end{tabular}
\begin{tabular}{rp{0.2cm}lp{1cm}rp{0.2cm}l}

  SYK & & Spleen tyrosine kinase \\
    TF & & Transcription factor \\
    TGF$\beta$ & & Transforming growth factor $\beta$ \\
    Th cell & & T-helper cell \\
    TLR & & Toll-like receptor \\ 
    TSS & & Transcription start site \\
    TTFT & & Time to first treatment \\
    TTT & & Time to next treatment \\
    U-CLL & & CLL without somatic hypermutations in IGHV loci \\
    VST & & Variance Stabilising Transformation \\
    WES & & Whole exome sequencing \\
    WGS & & Whole genome sequencing \\
    WT & & Wild type \\
    ZAP70 & & Tyrosine-protein kinase ZAP-70 \\

    
   
\end{tabular}
% ----------------------------------------------------
% --- frontmatter: List of Figures (not mandatory) ---
% ----------------------------------------------------
\newpage
\listoffigures
\addcontentsline{toc}{section}{List of Figures}

% ---------------------------------------------------
% --- frontmatter: List of Tables (not mandatory) ---
% ---------------------------------------------------
\newpage
\listoftables
\addcontentsline{toc}{section}{List of Tables}

% edit here to affect style of main document 
%\mainmatter % here the regular arabic numbering starts
\newpage
\pagestyle{plain} % use this to set the heading style 
\setcounter{page}{1}    % start page numbering anew
\pagenumbering{arabic}  % page numbers in arabic style

\hypertarget{introduction}{%
\chapter{Introduction}\label{introduction}}

\hypertarget{cll}{%
\section{CLL}\label{cll}}

\hypertarget{intro-disease-characteristics}{%
\subsection{Disease characteristics}\label{intro-disease-characteristics}}

Chronic lymphocytic leukaemia (CLL) is a malignancy of mature B cells, characterised by the progressive accumulation of malignant lymphocytes in the blood, bone marrow and lymph nodes (\protect\hyperlink{ref-Kipps2017}{Thomas J. Kipps et al. 2017}). The malignant cells can be distinguished by expression of CD5, CD19, and CD23, and by lower levels of membrane IgM, IgD, and CD79B (\protect\hyperlink{ref-Matutes1994}{Matutes et al. 1994}; \protect\hyperlink{ref-Moreau1997}{Moreau et al. 1997}; \protect\hyperlink{ref-Chiorazzi2005}{Chiorazzi, Rai, and Ferrarini 2005}) (Figure \ref{fig:CLLmorphology}).

The disease pathogenesis is driven by multiple factors, including molecular features, signalling via the B cell receptor (BCR) and interactions with non-neoplastic cells within the lymphoid tissues (known as the tumour microenvironment) (\protect\hyperlink{ref-Kipps2017}{Thomas J. Kipps et al. 2017}). Whilst the majority of CLL cells are in a resting state (\protect\hyperlink{ref-Messmer2005}{B. T. Messmer et al. 2005}; \protect\hyperlink{ref-Defoiche2008}{Defoiche et al. 2008}), evidence suggests that CLL cells migrate towards lymph nodes where they form proliferation centres (\protect\hyperlink{ref-Granziero2001}{Granziero et al. 2001}; \protect\hyperlink{ref-Messmer2005}{B. T. Messmer et al. 2005}), similar to germinal centres in healthy lymph nodes, which can show a daily birth rate of up to 3.3\% of the tumour (\protect\hyperlink{ref-Herndon2017}{Herndon et al. 2017}).


\begin{figure}

{\centering \includegraphics[width=0.4\linewidth]{figures/CLL_morphology} 

}

\caption{Wright--Giemsa-stained blood smear of a CLL malignant B cell, depicting typical morphology. \emph{Figure from \protect\hyperlink{ref-Kipps2017}{Thomas J. Kipps et al.} (\protect\hyperlink{ref-Kipps2017}{2017}) }.}\label{fig:CLLmorphology}
\end{figure}
The disease is the most common leukaemia in the West, accounting for 37\% of leukaemia cases and \textasciitilde19,000 newly detected cancers in the US in 2016 (\protect\hyperlink{ref-Kipps2017}{Thomas J. Kipps et al. 2017}; \protect\hyperlink{ref-Dubois2020}{Dubois et al. 2020}). The risk of developing CLL is twice as high for men than women, and more likely with increasing age (\protect\hyperlink{ref-Siegel2012}{Siegel et al. 2012}; \protect\hyperlink{ref-Nabhan2014}{Nabhan et al. 2014}; \protect\hyperlink{ref-Li2015}{Li et al. 2015}; \protect\hyperlink{ref-Pulte2015}{Pulte et al. 2015}). Chemotherapy (\protect\hyperlink{ref-Robak2005}{Robak 2005}; \protect\hyperlink{ref-Chang2012}{Chang and Kahl 2012}; \protect\hyperlink{ref-Lukenbill2013}{Lukenbill and Kalaycio 2013}) and chemoimmunotherapy (\protect\hyperlink{ref-Hallek2010}{M. Hallek et al. 2010}; \protect\hyperlink{ref-Goede2014}{Goede et al. 2014}; \protect\hyperlink{ref-Hillmen2015}{Hillmen et al. 2015}) has been the mainstay of therapy for many years. More recently, the central role of BCR signalling in disease pathogenesis has been appreciated and new drugs targeting this pathway have improved patient outcomes. The overall 5-years relative survival for CLL is 84\% (\protect\hyperlink{ref-Miller2019}{Miller et al. 2019}).

In spite of this, CLL is widely considered to be incurable (\protect\hyperlink{ref-Bosch2019}{Bosch and Dalla-Favera 2019}), and treatment regimens remain harsh. Moreover, the disease is characterised by its clinical heterogeneity (\protect\hyperlink{ref-Miller2019}{Miller et al. 2019}) and there is significant variation in survival amongst patients. Some patients harbour indolent disease, and can be followed by a watch-and-wait approach, sometimes going decades without requiring treatment. Others may need immediate treatment and survival can be short (\protect\hyperlink{ref-Miller2019}{Miller et al. 2019}). There is a clinical need to understand the underlying biology of the disease, with a view to understanding the causes of this clinical heterogeneity and to improving patient outcomes and experiences.

\hypertarget{intro-cell-of-origin}{%
\subsection{Cell of Origin}\label{intro-cell-of-origin}}

Identifying the cell of origin of a cancer can help explain disease pathogenesis, and understand the different subtypes, and their associated prognosis. Various studies have established mature CD5+ B cells as the cell of origin in CLL (\protect\hyperlink{ref-Campo2017}{\textbf{Campo2017?}}; \protect\hyperlink{ref-Bosch2019}{Bosch and Dalla-Favera 2019}).

In healthy tissues, these mature B cells are derived from haematopoietic stem cells in the bone marrow (\protect\hyperlink{ref-Kondo2010}{Kondo 2010}; \protect\hyperlink{ref-Fischer2020}{Fischer et al. 2020}). These stem cells develop in multiple stages, where each stage is defined by rearrangements within the immunoglobulin heavy chain and light chain loci (\protect\hyperlink{ref-Pelanda2012}{Pelanda and Torres 2012}) that encode components of the BCR. The aim of this process is to generate a large BCR repertoire, capable of recognising a range of foreign antigens.

Developing B cells in the bone marrow undergo positive and negative selection, ensuring that each BCR binds effectively to foreign antigen, whilst eliminating those that bind strongly to self-antigen (\protect\hyperlink{ref-Lebien2008}{Lebien and Tedder 2008}; \protect\hyperlink{ref-Martensson2010}{Mårtensson et al. 2010}). These immature B cells then migrate to the lymph nodes and spleen where they differentiate into mature cells that are considered to be antigen ``naive''(\protect\hyperlink{ref-Chung2003}{Chung, Silverman, and Monroe 2003}).

B cells are later activated when they encounter their respective antigen. When activated, the B cells form germinal centres, which are specialised microenvironments within the lymph node that facilitate extensive proliferation. Here they undergo a process called affinity maturation, in which the loci encoding components of the BCR undergo somatic hypermutation to optimise BCR antigen specificity (\protect\hyperlink{ref-Shlomchik2012}{Shlomchik and Weisel 2012}). This process generates short-lived plasmablasts which provide immediate protection, along with plasma cells and memory B cells procuring longer-term immunity (\protect\hyperlink{ref-Nutt2015}{Nutt et al. 2015}).

\hypertarget{intro-ighv-status}{%
\subsection{IGHV status}\label{intro-ighv-status}}

The two major subtypes of CLL can be defined by whether the B cell of origin has undergone this process of somatic hypermutation within a germinal centre. This is reflected in the degree of mutation of the immunoglobulin heavy-chain variable region (IGHV) genes. Evidence suggests that IGHV-mutated (IGHV-M) CLLs (60\% of cases) are derived from antigen-experienced B cells. On the other hand, there is continued debate as to whether IGHV-unmutated (IGHV-U) CLLs derivate from naive B cells that have yet to transit through the germinal centre, or germinal-centre, antigen-experienced B cells (\protect\hyperlink{ref-Klein2001}{Klein et al. 2001}).

The two subtypes show distinct clinical and molecular properties. In contrast to IGHV-M CLLs, IGHV-U CLLs have a higher proportion of high-risk genetic lesions, more commonly undergo clonal evolution and consequently have a shorter time to first treatment (TTFT) and less favourable overall survival (OS) (\protect\hyperlink{ref-Hamblin1999}{Hamblin et al. 1999}; \protect\hyperlink{ref-Damle1999}{Damle et al. 1999}; \protect\hyperlink{ref-Landau2013}{Landau et al. 2013}; \protect\hyperlink{ref-Puente2015}{Xose S. Puente et al. 2015}; \protect\hyperlink{ref-Bosch2019}{Bosch and Dalla-Favera 2019}). Their differential antigen experience also affects how they respond to signals from the microenvironment, including stimulation of the BCR.

\hypertarget{intro-bcr-signalling}{%
\subsection{The central role of BCR signalling}\label{intro-bcr-signalling}}

The BCR pathway is central to the process of selection, development, proliferation and survival of CLL clones (\protect\hyperlink{ref-Chiorazzi2003}{Chiorazzi and Ferrarini 2003}; \protect\hyperlink{ref-Stevenson2004}{Stevenson and Caligaris-Cappio 2004}; \protect\hyperlink{ref-Agathangelidis2012}{Agathangelidis et al. 2012}; \protect\hyperlink{ref-Burger2013}{Jan A. Burger and Chiorazzi 2013}; \protect\hyperlink{ref-Iacovelli2015}{Iacovelli et al. 2015}; \protect\hyperlink{ref-Dubois2020}{Dubois et al. 2020}). Evidence of the importance of this pathway comes from the observation that BCRs of CLL cells show a biased selection of IGHV and IGLV\(\kappa\)/\(\lambda\) genes. This generates BCRs that are remarkably similar across patients (\protect\hyperlink{ref-Kipps1989}{T. J. Kipps et al. 1989}; \protect\hyperlink{ref-Fais1998}{Fais et al. 1998}; \protect\hyperlink{ref-Widhopf2004}{Widhopf et al. 2004}; \protect\hyperlink{ref-Messmer2004}{B. T. Messmer et al. 2004}; \protect\hyperlink{ref-Stamatopoulos2007}{K. Stamatopoulos et al. 2007}), suggesting that BCR binding to certain antigens may drive selection and proliferation of CLL clones. Other compelling studies have shown that BCR signalling genes are upregulated in CLL cells taken from lymph nodes (\protect\hyperlink{ref-Herishanu2011}{\textbf{Herishanu2011?}}) and that cells from IGHV-U CLL patients (with poorer outcomes), also show activation of BCR-related genes (\protect\hyperlink{ref-Rosenwald2001}{Rosenwald et al. 2001}). The success of BCR inhibitors in the clinic (\protect\hyperlink{ref-Byrd2013}{John C. Byrd et al. 2013}), also underlines the critical importance of this pathway in CLL.

Figure \ref{fig:BCRsignalling} shows a schematic of the BCR pathway itself. The BCR is a multimeric complex consisting of surface immunoglobulin (Ig), which recognises antigen, plus the Ig-\(\alpha\)/Ig-\(\beta\) hetero-dimers, known as CD79A and CD79B. The BCR may bind to external antigens within the microenvironment (\protect\hyperlink{ref-Binder2010}{Binder et al. 2010}) or intra-BCR self-antigens (\protect\hyperlink{ref-Minden}{\textbf{Minden?}}), which in turn recruits SYK and the Src kinase LYN. These kinases phosphorylate motifs located on the cytoplasmic tails of CD79A and CD79B, initiating a signalling cascade involving a number of proteins and pathways. These include the proteins BTK (\protect\hyperlink{ref-Herman2011}{Herman et al. 2011}) and PI3K (\protect\hyperlink{ref-Longo2006}{\textbf{Longo2006?}}) which activate a number of downstream pathways and players including PLC\(\gamma\) 2, calcium signalling, PKC, NF\(\kappa\)B signalling, ERK and MAPKs, and nuclear transcription (\protect\hyperlink{ref-Burger2013}{Jan A. Burger and Chiorazzi 2013}; \protect\hyperlink{ref-Dubois2020}{Dubois et al. 2020}).

There are two types of BCR signalling in healthy and CLL B cells: ligand-dependent ``active'' signalling that relies on antigen-binding, and ligand-independent ``tonic'' signalling (\protect\hyperlink{ref-Lam1997}{\textbf{Lam1997?}}; \protect\hyperlink{ref-Kraus2004}{Kraus et al. 2004}). Whilst active signalling engages the entire signalling cascade described above, tonic signalling activates only a subset of these. \protect\hyperlink{ref-Kraus2004}{Kraus et al.} (\protect\hyperlink{ref-Kraus2004}{2004}) showed that tonic signalling is important for prolonged B cell survival whereby PI3K signalling is thought to play a key role in delivering survival signals (\protect\hyperlink{ref-Srinivasan2009}{\textbf{Srinivasan2009?}}; \protect\hyperlink{ref-Pula2019}{\textbf{Pula2019?}}). Both these modes of BCR signalling are believed to impact of the survival and growth of the tumour, although the dominant mode remains a matter of debate (\protect\hyperlink{ref-Burger2013}{Jan A. Burger and Chiorazzi 2013}).

The different clinical properties of IGHV-M and IGHV-U cell is believed to be determined by in part by their differential response to BCR stimulation, which is influenced by their cell of origin (\protect\hyperlink{ref-Burger2013}{Jan A. Burger and Chiorazzi 2013}). IGHV-U CLLs have not undergone somatic hypermutation and consequently express low-affinity BCRs that are frequently activated by numerous antigens and auto-antigens in the microenvironment (\protect\hyperlink{ref-Borche1990}{Borche et al. 1990}; \protect\hyperlink{ref-Broker1988}{Bröker et al. 1988}; \protect\hyperlink{ref-Sthoeger1989}{Sthoeger et al. 1989}; \protect\hyperlink{ref-Herve2005}{Hervé et al. 2005}; \protect\hyperlink{ref-Lanemo2008}{Myhrinder et al. 2008}; \protect\hyperlink{ref-Chu2008}{Chu et al. 2008}; \protect\hyperlink{ref-Binder2010}{Binder et al. 2010}; \protect\hyperlink{ref-Krysov2010}{Krysov et al. 2010}; \protect\hyperlink{ref-Kostareli2012}{Kostareli et al. 2012}; \protect\hyperlink{ref-Hoogeboom2012}{\textbf{Hoogeboom2012?}}). In contrast, IGHV-M BCRs only recognise highly specific antigens, which either occur infrequently or induce anergy due to high-affinity binding (\protect\hyperlink{ref-Chiorazzi2003}{Chiorazzi and Ferrarini 2003}; \protect\hyperlink{ref-Stevenson2004}{Stevenson and Caligaris-Cappio 2004}; \protect\hyperlink{ref-Chiorazzi2005}{Chiorazzi, Rai, and Ferrarini 2005}; \protect\hyperlink{ref-Burger2013}{Jan A. Burger and Chiorazzi 2013}). IGHV-M CLL clones therefore are more stable and expand at a slower rate.


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{figures/BCR_signalling} 

}

\caption{Graphical depiction of the BCR pathway. \emph{Figure originally published in \protect\hyperlink{ref-Kipps2017}{Thomas J. Kipps et al.} (\protect\hyperlink{ref-Kipps2017}{2017}) }}\label{fig:BCRsignalling}
\end{figure}
\hypertarget{intro-genetic-features}{%
\subsection{Recurrent genetic features in CLL}\label{intro-genetic-features}}

The genomic landscape of CLL has been thoroughly characterised in a number of studies, including two seminal papers involving \textgreater500 CLL samples (\protect\hyperlink{ref-Landau2015}{Landau et al. 2015}; \protect\hyperlink{ref-Puente2015}{Xose S. Puente et al. 2015}). Such studies have indicated that CLL shows a lower mutational load than other lymphoid neoplasms (\protect\hyperlink{ref-Vogelstein2013}{Vogelstein et al. 2013}; \protect\hyperlink{ref-Alexandrov2013}{Alexandrov et al. 2013}; \protect\hyperlink{ref-Bosch2019}{Bosch and Dalla-Favera 2019}), in which a relatively large number of genes are rarely mutated (\protect\hyperlink{ref-Fabbri2016}{Fabbri and Dalla-Favera 2016}). A small number of driver genes are mutated in a significant proportion of cases, though a common genetic event which can accounts for most cases of CLL has not been identified (\protect\hyperlink{ref-Fabbri2016}{Fabbri and Dalla-Favera 2016}). These genetic alterations encompass chromosomal alterations, mutations, alterations in miRNA expression and epigenetic modifications (\protect\hyperlink{ref-Kipps2017}{Thomas J. Kipps et al. 2017}).

\hypertarget{intro-mutations}{%
\subsubsection{Somatic Mutations}\label{intro-mutations}}

There are many recurrent somatic mutations in CLL, and these centre on several major pathways and functions that are frequently altered in patients (Figure \ref{fig:CLLmutations}). (\protect\hyperlink{ref-Puente2015}{Xose S. Puente et al. 2015}; \protect\hyperlink{ref-Fabbri2016}{Fabbri and Dalla-Favera 2016}; \protect\hyperlink{ref-Kipps2017}{Thomas J. Kipps et al. 2017}). These pathways include Notch signalling, DNA damage response, RNA processing, NF\(\kappa\)B signalling, BCR signalling, WNT signalling and chromatin modification (\protect\hyperlink{ref-Kipps2017}{Thomas J. Kipps et al. 2017}; \protect\hyperlink{ref-Bosch2019}{Bosch and Dalla-Favera 2019}).

Within these pathways, most mutations are rare and only a few occur at a frequency \textgreater5\% (\protect\hyperlink{ref-Bosch2019}{Bosch and Dalla-Favera 2019}). One landmark study indicated the most frequent mutations occur in \emph{NOTCH1} (12.6\% of patients), \emph{ATM} (11\%), \emph{BIRC3} (8.8\%) and \emph{SF3B1} (8.6\%), although these frequencies are dependent disease stage and treatment status (\protect\hyperlink{ref-Puente2015}{Xose S. Puente et al. 2015}). The functional role and prognostic importance of a number of these putative driver mutations has been established, and are discussed below.

\textbf{Notch signalling}
The Notch pathway activates genes required for proliferation, metabolism and survival, including \emph{MYC}, via the activation of the NOTCH1 transmembrane receptor (\protect\hyperlink{ref-Guruharsha2012}{Guruharsha, Kankel, and Artavanis-Tsakonas 2012}). Mutations in \emph{NOTCH1} are very common in CLL (\textasciitilde4-20\%) (\protect\hyperlink{ref-Fabbri2011}{Fabbri et al. 2011}; \protect\hyperlink{ref-Puente2011}{Xose S. Puente et al. 2011}; \protect\hyperlink{ref-Landau2013}{Landau et al. 2013}, \protect\hyperlink{ref-Landau2015}{2015}; \protect\hyperlink{ref-Puente2015}{Xose S. Puente et al. 2015}). A number of other recurrent mutations also centre on Notch deregulation, indicating that this pathway is disrupted in many CLL cases (\protect\hyperlink{ref-Fabbri2016}{Fabbri and Dalla-Favera 2016}). \emph{NOTCH1} mutations occur more often in IGVH-U CLLs and are associated with a less favourable OS (\protect\hyperlink{ref-Fabbri2011}{Fabbri et al. 2011}). \emph{NOTCH1} mutants also respond less-well to anti-CD20-based therapies, due to decreased surface expression of CD20 (\protect\hyperlink{ref-Fabbri2001}{\textbf{Fabbri2001?}}; \protect\hyperlink{ref-Rossi2012}{Rossi et al. 2012}).

\textbf{DNA Damage Response}
A number of frequently occurring mutations disrupt the DNA damage response, the most clinically important of which is \emph{TP53}. \emph{TP53} is known as the ``guardian of the genome,'' owing to its role as a tumour suppressor gene protecting genome integrity by preventing mutation. Despite being a tumour suppressor, \emph{TP53} mutations can have a dominant negative effect on function, such that loss of genomic stability occurs even when a single allele is mutated (\protect\hyperlink{ref-Zenz2008}{Zenz et al. 2008}). \emph{TP53} is an important prognostic marker, as dysregulation of its function is associated with resistance to DNA-damaging agents (chemotherapy and radiotherapy) (\protect\hyperlink{ref-Zenz2008}{Zenz et al. 2008}; \protect\hyperlink{ref-Dicker2009}{\textbf{Dicker2009?}}; \protect\hyperlink{ref-Rossi2009}{Rossi et al. 2009}), and patients who present with mutations at diagnosis often show shorter TTFT and a less favourable OS (\protect\hyperlink{ref-Dohner2000}{H. Döhner et al. 2000}; \protect\hyperlink{ref-Zenz2008}{Zenz et al. 2008}; \protect\hyperlink{ref-Rossi2013}{Rossi et al. 2013}; \protect\hyperlink{ref-Puente2015}{Xose S. Puente et al. 2015}).

Upstream of \emph{TP53}, the tumour suppressor \emph{ATM} is also frequently mutated in CLL (\protect\hyperlink{ref-Bosch2019}{Bosch and Dalla-Favera 2019}). \emph{ATM} activates the DNA damage response upon recognition of DNA double strand breaks (\protect\hyperlink{ref-Austen2005}{Austen et al. 2005}; \protect\hyperlink{ref-Shiloh2013}{Shiloh and Ziv 2013}). Similarly to \emph{TP53} mutants, \emph{ATM}-disrupted CLLs also show genomic instability and these mutations are associated with shorter TTT and OS, and chemoresistance (\protect\hyperlink{ref-Austen2005}{Austen et al. 2005}; \protect\hyperlink{ref-Stankovic2014}{Stankovic and Skowronska 2014}).

\emph{POT1} is also involved in genomic stability, and is mutated in around 3 -- 7\% of cases (\protect\hyperlink{ref-Bosch2019}{Bosch and Dalla-Favera 2019}; \protect\hyperlink{ref-Ramsay2012}{\textbf{Ramsay2012?}}). \emph{POT1} mutations disrupt telomere protection, leading to a increased in structural aberrations and chromosomal breaks (\protect\hyperlink{ref-Ramsay2012}{\textbf{Ramsay2012?}}). Mutations in \emph{POT1} are associated with IGHV-U CLL and advanced clinical stage (\protect\hyperlink{ref-Ramsay2012}{\textbf{Ramsay2012?}}).

\textbf{RNA processing}
30\% of CLL patients harbour mutations disrupting RNA processing and the spliceosome machinery (\protect\hyperlink{ref-Puente2011}{Xose S. Puente et al. 2011}; \protect\hyperlink{ref-Wang2011}{Wang et al. 2011}; \protect\hyperlink{ref-Quesada2012}{Quesada et al. 2012}; \protect\hyperlink{ref-Puente2015}{Xose S. Puente et al. 2015}; \protect\hyperlink{ref-Fabbri2016}{Fabbri and Dalla-Favera 2016}).
\emph{SF3B1} mutations are the most frequent to occur, and are found in ∼10\% of cases, usually in IGHV-U patients (\protect\hyperlink{ref-Puente2011}{Xose S. Puente et al. 2011}; \protect\hyperlink{ref-Wang2011}{Wang et al. 2011}; \protect\hyperlink{ref-Quesada2012}{Quesada et al. 2012}; \protect\hyperlink{ref-Puente2015}{Xose S. Puente et al. 2015}; \protect\hyperlink{ref-Fabbri2016}{Fabbri and Dalla-Favera 2016}). The \emph{SF3B1} gene encodes part of the U2 snRNP complex which is involved in RNA splicing (\protect\hyperlink{ref-Shin2004}{Shin and Manley 2004}), though the functional implication of \emph{SF3B1} mutations are yet to be established, and many transcripts show abnormal splicing in\emph{SF3B1} cases (\protect\hyperlink{ref-Quesada2012}{Quesada et al. 2012}). The presence of \emph{SF3B1} mutations is associated with a decreased TTFT and unfavourable OS (\protect\hyperlink{ref-Bosch2019}{Bosch and Dalla-Favera 2019}).

\textbf{NF\(\kappa\)B signalling}
A wide range of mutations across various pathways converge on the activation of NF\(\kappa\)B (\protect\hyperlink{ref-Fabbri2011}{Fabbri et al. 2011}; \protect\hyperlink{ref-Puente2011}{Xose S. Puente et al. 2011}; \protect\hyperlink{ref-Wang2011}{Wang et al. 2011}; \protect\hyperlink{ref-Quesada2012}{Quesada et al. 2012}; \protect\hyperlink{ref-Puente2015}{Xose S. Puente et al. 2015}; \protect\hyperlink{ref-Landau2015}{Landau et al. 2015}), including \emph{BIRC3} and \emph{MYD88}. For example, certain mutations in \emph{MYD88} result in increased binding to IRAK1 and higher activation of NF\(\kappa\)B (\protect\hyperlink{ref-Puente2011}{Xose S. Puente et al. 2011}). However, the role and prognostic importance of NF\(\kappa\)B activation in CLL is still unclear (\protect\hyperlink{ref-Bosch2019}{Bosch and Dalla-Favera 2019}).


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{figures/genetic_mutations} 

}

\caption{Graphical depiction of commonly mutated genes in CLL, grouped into cellular pathways (blue boxes). Minus sign indicates negative regulation. \_Figure originally published in (\protect\hyperlink{ref-Kipps2017_}{\textbf{Kipps2017\_?}})}\label{fig:CLLmutations}
\end{figure}
\hypertarget{intro-structural-abberations}{%
\subsubsection{Structural Aberrations}\label{intro-structural-abberations}}

In addition to the aforementioned mutations, a number of common structural aberrations confer similar disruption to normal B cell function. The key structural aberrations and their prognostic value were set out in a landmark paper by \protect\hyperlink{ref-Dohner2000}{H. Döhner et al.} (\protect\hyperlink{ref-Dohner2000}{2000}), as follows.

\textbf{del(11q)}
Deletions of chromosome 11q (del(11q)) are fairly common (\textasciitilde10\%) in CLL, and are believed to target the \emph{ATM} gene in the 11q22-23 region (\protect\hyperlink{ref-Dohner2000}{H. Döhner et al. 2000}; \protect\hyperlink{ref-Austen2005}{Austen et al. 2005}; \protect\hyperlink{ref-Bosch2019}{Bosch and Dalla-Favera 2019}). del(11q) is usually monoallelic, but can also be associated with mutations in the remaining ATM allele (\textasciitilde30\% of cases) (\protect\hyperlink{ref-Austen2005}{Austen et al. 2005}). In certain instances, the deleted region does not include \emph{ATM} but rather \emph{BIRC3}, a negative regulator of the NF\(\kappa\)B pathway (\protect\hyperlink{ref-Rossi2012}{Rossi et al. 2012}). Patients with del(11q) or \emph{ATM} lesions have a shorter TTFT and OS, especially if the lesion is biallelic (\protect\hyperlink{ref-Austen2005}{Austen et al. 2005}; \protect\hyperlink{ref-Skowronska2012}{Skowronska et al. 2012}; \protect\hyperlink{ref-Stankovic2014}{Stankovic and Skowronska 2014}; \protect\hyperlink{ref-Nadeu2016}{Nadeu et al. 2016}).

\textbf{del(17p)}
Deletion of chromosomal region 17p13 (del(17p)) is found in 1 -- 20\% of cases, depending on the stage of the disease and most common in chemo-refractory cases (\protect\hyperlink{ref-Dohner2000}{H. Döhner et al. 2000}; \protect\hyperlink{ref-Zenz2008}{Zenz et al. 2008}, \protect\hyperlink{ref-Zenz2010}{2010}; \protect\hyperlink{ref-Stilgenbauer2014}{Stilgenbauer et al. 2014}). The target of this lesion is thought to be \emph{TP53}; the deleted region consistently includes the \emph{TP53} locus (\protect\hyperlink{ref-Dohner1995}{Hartmut Döhner et al. 1995}), and around 80\% del(17p) cases also have missense mutations in the second \emph{TP53} allele (\protect\hyperlink{ref-Zenz2008}{Zenz et al. 2008}; \protect\hyperlink{ref-Gonzalez2011}{Gonzalez et al. 2011}; \protect\hyperlink{ref-Trbusek2011}{Trbusek et al. 2011}; \protect\hyperlink{ref-Rossi2014}{\textbf{Rossi2014?}}). Del(17p) CLLs show increased genomic instability (\protect\hyperlink{ref-Yu2017}{L. Yu et al. 2017}), resistance to chemotherapy and, correspondingly, a shorter TTFT and a less favourable OS (\protect\hyperlink{ref-Dohner2000}{H. Döhner et al. 2000}; \protect\hyperlink{ref-Zenz2008}{Zenz et al. 2008}; \protect\hyperlink{ref-Rossi2013}{Rossi et al. 2013}; \protect\hyperlink{ref-Puente2015}{Xose S. Puente et al. 2015}).

\textbf{del(13q)}
Deletion in the 13q14 region (del(13q)) is the most common genetic lesion in CLL (\textasciitilde50--60\% of patients) (\protect\hyperlink{ref-Dohner2000}{H. Döhner et al. 2000}). Experiments to determine the minimal deleted region identified that this invariably contains \emph{DLEU1} and \emph{DLEU2}, two long non-coding RNA genes, and the microRNA gene cluster \emph{MIR15A--MIR16-1} (\protect\hyperlink{ref-Kalachikov1997}{Kalachikov et al. 1997}; \protect\hyperlink{ref-Migliazza2001}{Migliazza et al. 2001}; \protect\hyperlink{ref-Calin2002}{Calin et al. 2002}; \protect\hyperlink{ref-Hammarsund2004}{\textbf{Hammarsund2004?}}; \protect\hyperlink{ref-Palamarchuk2010}{Palamarchuk et al. 2010}). \emph{in vitro} studies have demonstrated the role of these genes in regulation of the cell cycle and apoptosis (\protect\hyperlink{ref-Cimmino2005}{Cimmino et al. 2005}; \protect\hyperlink{ref-Klein2010}{\textbf{Klein2010?}}; \protect\hyperlink{ref-Bosch2019}{Bosch and Dalla-Favera 2019}). In some CLL cases, del(13q) is the sole genetic abnormality, indicating that this lesion may be involved in early CLL development (\protect\hyperlink{ref-Dohner2000}{H. Döhner et al. 2000}; \protect\hyperlink{ref-Klein2010}{\textbf{Klein2010?}}; \protect\hyperlink{ref-Landau2015}{Landau et al. 2015}). Moreover, conditional deletion of the equivalent minimal deleted region in mice recapitulated CLL initiation and progression, and these mice developed clonal lymphoproliferations (\protect\hyperlink{ref-Migliazza2001}{Migliazza et al. 2001}; \protect\hyperlink{ref-Klein2010}{\textbf{Klein2010?}}). del(13q) patients have the best prognosis, with prolonged TTFT, and OS compared to patients with other lesions (\protect\hyperlink{ref-Dohner2000}{H. Döhner et al. 2000}; \protect\hyperlink{ref-Rossi2013}{Rossi et al. 2013}).

\hypertarget{intro-trisomy12}{%
\subsubsection{The incompletely understood role of trisomy12}\label{intro-trisomy12}}

Complete duplication of chromosome 12 (trisomy 12) is observed in \textasciitilde15\% of CLL patients at diagnosis (\protect\hyperlink{ref-Dohner2000}{H. Döhner et al. 2000}). Despite its recurrence, there is currently no functional explanation for this lesion (\protect\hyperlink{ref-Bosch2019}{Bosch and Dalla-Favera 2019}), although a number of features have been observed. Trisomy 12 is more common in IGHV-M patients than IGHV-U (\protect\hyperlink{ref-Hamblin1999}{Hamblin et al. 1999}), and is thought to confer an abnormal cellular morphology (\protect\hyperlink{ref-Bosch2019}{Bosch and Dalla-Favera 2019}). Previous work in our lab has also demonstrated that trisomy 12 CLLs show a specific signalling signature and distinct transcriptomic (\protect\hyperlink{ref-JCIpaper}{Dietrich et al. 2017}) and proteomic profiles (\protect\hyperlink{ref-HerbstThesis}{Herbst 2020}), including differential expression of genes within the BCR, PI3K, AKT, and mTOR signaling and chemokine signaling pathways. Moreover, trisomy 12 CLLs show higher sensitivity to BCR inhibitors, indicating that BCR signalling may be amplified in these cases (\protect\hyperlink{ref-JCIpaper}{Dietrich et al. 2017}).

Traditionally trisomy 12 has been classified as an intermediate-risk lesion (\protect\hyperlink{ref-Dohner2000}{H. Döhner et al. 2000}): these cases have a higher proliferative capacity, but are more treatable with chemotherapeutics and BCR inhibitors. However, \emph{NOTCH1} mutations are frequently observed in trisomy 12 cases, and this is associated with poorer outcomes (\protect\hyperlink{ref-Balatti2012}{Balatti et al. 2012}; \protect\hyperlink{ref-Giudice2012}{Del Giudice et al. 2012}).

\hypertarget{intro-epigenetic-alterations}{%
\subsubsection{Epigenetic alterations}\label{intro-epigenetic-alterations}}

In addition to genetic lesions, the epigenome is also modified in CLL and samples typically show global hypomethylation combined with local hypermethylation (\protect\hyperlink{ref-Wahlfors1992}{Wahlfors et al. 1992}; \protect\hyperlink{ref-Cahill2013}{Cahill et al. 2013}; \protect\hyperlink{ref-Ziller2013}{Ziller et al. 2013}; \protect\hyperlink{ref-Kipps2017}{Thomas J. Kipps et al. 2017}; \protect\hyperlink{ref-Bosch2019}{Bosch and Dalla-Favera 2019}). Studies investigating the epigenome in CLL have proven revealing, in particular, higher levels of intra-sample methylation heterogeneity have been shown to be associated with high-risk genetic lesions and poorer prognosis (\protect\hyperlink{ref-Landau2014}{Landau et al. 2014}).

Moreover, methylation signatures have been used to classify distinct clinical CLL subgroups (\protect\hyperlink{ref-Kulis2012}{Kulis et al. 2012}; \protect\hyperlink{ref-Bhoi2016}{Bhoi et al. 2016}), as they are useful to trace the cell of origin. For example, CLL cells from distinct patients originate from many different B cell maturation states, possibly reflecting the biological and phenotypic heterogeneity of CLL (\protect\hyperlink{ref-Oakes2016}{Oakes et al. 2016}). Accordingly, IGHV-U CLLs have a distinct methylation signature to IGHV-M CLLs, and these patterns correspond approximately to those of pre-germinal centre or post-germinal centre memory B cells, respectively (\protect\hyperlink{ref-Kulis2012}{Kulis et al. 2012}; \protect\hyperlink{ref-Oakes2016}{Oakes et al. 2016}). Epigenetic studies have also revealed the relationship between certain genetic lesions and specific epigenetic signatures, for example \emph{MYD88} mutations and trisomy 12 (\protect\hyperlink{ref-Beekman2018}{Beekman et al. 2018}). The CLL epigenome can also be modulated by drugs and thus is of increasing clinical interest (\protect\hyperlink{ref-Timp2013}{Timp and Feinberg 2013}; \protect\hyperlink{ref-Beekman2018}{Beekman et al. 2018}; \protect\hyperlink{ref-Gaiti2019}{Gaiti et al. 2019}).

\hypertarget{intro-therapies}{%
\section{Therapies in CLL}\label{intro-therapies}}

Extensive work to uncover the molecular drivers of CLL has led to the development of several therapeutic strategies. CLL represents a successful example of how developing a complex understanding of the biological characteristics of a disease can lead to significantly improved patient outcomes (\protect\hyperlink{ref-Yosifov2019}{Yosifov et al. 2019}).
Treatment of CLL patients can be via chemotherapy, chemoimmunotherapy or targeted therapies that inhibit specific pathways (\protect\hyperlink{ref-Kipps2017}{Thomas J. Kipps et al. 2017}; \protect\hyperlink{ref-Burger2020}{Jan A. Burger 2020}). Additionally, allogeneic stem cell transplantation is increasingly considered as an alternative option in relapsed or refractory patients (\protect\hyperlink{ref-Kipps2017}{Thomas J. Kipps et al. 2017}).

\hypertarget{intro-chemotherapy}{%
\subsection{Chemotherapy and chemoimmunotherapy}\label{intro-chemotherapy}}

Chemotherapy has been the standard of care for CLL for many decades, either with purine analogues such as fludarabine or alkylating agents such as chlorambucil (\protect\hyperlink{ref-Robak2005}{Robak 2005}; \protect\hyperlink{ref-Lukenbill2013}{Lukenbill and Kalaycio 2013}). Chemoimmunotherapy has also benefited many patients owing to the rapid improvements to monoclonal antibody technology and the development of anti-CD20 treatments, such as rituximab (\protect\hyperlink{ref-Yosifov2019}{Yosifov et al. 2019}; \protect\hyperlink{ref-Robak2010}{Robak et al. 2010}). However, patients with higher risk lesions such as \emph{TP53} and del(17p) do not respond well to chemoimmunotherapy, and require alternative therapeutic options (\protect\hyperlink{ref-Zenz2010}{Zenz et al. 2010}).

\hypertarget{intro-targeted-therapies}{%
\subsection{Targeted therapies}\label{intro-targeted-therapies}}

More recently, the importance of BCR signalling and upregulation of anti-apoptotic proteins in CLL expansion has been increasingly realised and led to the development of therapies targeting these pathways. Drugs targeting BCR signalling and BCL-2 have changed the treatment landscape dramatically (\protect\hyperlink{ref-Scheffold2020}{Scheffold and Stilgenbauer 2020}). Three main drug classes that target BCR signalling have been developed for CLL: BTK inhibitors, PI3K inhibitors and SYK inhibitors (\protect\hyperlink{ref-Rooij2012}{De Rooij et al. 2012}; \protect\hyperlink{ref-Kipps2017}{Thomas J. Kipps et al. 2017}).

Ibrutinib is a BTK inhibitor approved for use as an initial therapy and for patients who are refractory to chemoimmunotherapy (\protect\hyperlink{ref-Byrd2013}{John C. Byrd et al. 2013}; \protect\hyperlink{ref-Byrd2014}{John C. Byrd et al. 2014}). Despite such success, complete remission is rare and many patients continue to harbour minimal residual disease within the bone marrow (\protect\hyperlink{ref-Byrd2013}{John C. Byrd et al. 2013}), requiring continued therapy for years (\protect\hyperlink{ref-Byrd2013}{John C. Byrd et al. 2013}; \protect\hyperlink{ref-Woyach2015}{Woyach and Johnson 2015}). Resistance can also occur via the acquisition of mutations in BTK or PLC\(\gamma\) 2 (\protect\hyperlink{ref-Woyach2015}{Woyach and Johnson 2015}) genes. Treatment initiation with ibrutinib is associated with a concomitant increase in the absolute lymphocyte count in the blood (\protect\hyperlink{ref-Woyach2014}{Woyach et al. 2014}), thought to caused by the inhibition of chemokine receptor signalling leading to the release of malignant B cells from the lymph nodes into the peripheral blood.

Idelalisib also acts to inhibits BCR signalling and chemokine signalling (\protect\hyperlink{ref-Hoellenriegel2011}{Hoellenriegel et al. 2011}), via inhibition of PI3K. Whilst the drug is highly efficacious in CLL (\protect\hyperlink{ref-Furman2014}{Furman et al. 2014}; \protect\hyperlink{ref-Brown2014}{Brown et al. 2014}), idelalisib demonstrates more toxicities and lower efficacy than BTK inhibitors and thus is generally used as an alternative therapy in patients for whom BTK inhibitors are unsuitable (\protect\hyperlink{ref-Ghia2020}{Ghia et al. 2020}; \protect\hyperlink{ref-Burger2020}{Jan A. Burger 2020}).

Other drugs targeting SYK, downstream of BTK, have also shown promise in Phase I/II clinical trials (\protect\hyperlink{ref-Friedberg2010}{Friedberg et al. 2010}), though none are licenced as yet.

Aside from BCR inhibitors, BCL-2 inhibitors such as venetoclax, have also shown good efficacy in CLL. Venetoclax is thought to induce apoptosis in CLL cells, by acting as a BH3 mimetic interfering with the ability of BCL-2 to sequester BIM (\protect\hyperlink{ref-Moore2007}{Moore et al. 2007}). This makes venetoclax an attractive alternative in patients with relapsed or refractory disease (\protect\hyperlink{ref-Roberts2016}{Roberts et al. 2016}), or who have del(17p) i.e.~loss of \emph{TP53 }(\protect\hyperlink{ref-Stilgenbauer2016}{Stilgenbauer et al. 2016}). Complete remissions are observed in \textasciitilde20\% patients (\protect\hyperlink{ref-Roberts2016}{Roberts et al. 2016}) which is higher than with other targeted therapies, and minimal residual disease in the bone marrow is less common (\protect\hyperlink{ref-Roberts2016}{Roberts et al. 2016}).

\hypertarget{intro-management-algorithm}{%
\subsection{Management algorithm}\label{intro-management-algorithm}}

The decision to initiate therapy is guided by the stage of disease, evidence for rapid disease progression or disease-related symptoms (\protect\hyperlink{ref-Hallek2008}{Michael Hallek et al. 2008}; \protect\hyperlink{ref-Kipps2017}{Thomas J. Kipps et al. 2017}). Disease stage is determined by an index such as the Rai staging system, which categorises patients based on disease severity (\protect\hyperlink{ref-Kipps2017}{Thomas J. Kipps et al. 2017}). The choice of therapy accounts for certain mutations, in particular the presence of \emph{TP53} or del(17p) mutations, the age of the patient and the objective of therapy. IGHV status is also increasingly used for patient stratification (Figure \ref{fig:CLLstratification}, (\protect\hyperlink{ref-Kipps2017}{Thomas J. Kipps et al. 2017})).

CLL is a manageable disease, with a well-established arsenal of treatments and associated management algorithm. However, treatments are harsh, and the disease is still considered incurable (\protect\hyperlink{ref-Bosch2019}{Bosch and Dalla-Favera 2019}). Many cases develop resistance to therapy, both via acquired mutations and through survival signals provided by the microenvironment (see section \ref{intro-microenvironment-drug-response}) and minimal residual disease within the bone marrow is common. Collectively, these lead to relapse or required prolonged therapy and its associated toxicities. There remains a clinical need to improve patient outcomes and experience.


\begin{figure}

{\centering \includegraphics[width=0.8\linewidth]{figures/disease_stratification} 

}

\caption{Management algorithm for patients with CLL. CIT (chemoimmunotherapy), LDT (lymphocyte doubling time). \emph{Figure originally published in \protect\hyperlink{ref-Kipps2017}{Thomas J. Kipps et al.} (\protect\hyperlink{ref-Kipps2017}{2017}) }.}\label{fig:CLLstratification}
\end{figure}
\hypertarget{intro-tumour-microenvironment}{%
\section{The tumour microenvironment in CLL}\label{intro-tumour-microenvironment}}

\hypertarget{intro-tumour-microenvironment-role}{%
\subsection{The role of the tumour microenvironment in CLL}\label{intro-tumour-microenvironment-role}}

In addition to genetic aberrations, the tumour microenvironment is an important driver of disease pathogenesis in CLL (\protect\hyperlink{ref-tenHacken2016}{\textbf{tenHacken2016?}}). The term microenvironment encompasses the set of non-neoplastic cells within the lymphoid tissues, including the bone marrow and lymph nodes, that provide survival signals to the tumour, leading to clonal expansion and drug resistance (\protect\hyperlink{ref-Burger2014}{Jan A. Burger and Gribben 2014}). Malignant B cells engage in a dialogue with the non-neoplastic cells, via cell-cell contacts and soluble factors, including chemokines, integrins, cytokines and survival factors, centring on a number of important pathways including BCR signalling and tissue homing chemokine receptors (\protect\hyperlink{ref-tenHacken2016}{\textbf{tenHacken2016?}}).

The importance of the microenvironment in CLL pathogenesis was first recognised in studies that showed CLL cells rapidly undergo apoptosis \emph{in vitro}, whilst their survival can be extended by stimulation or by co-culture with nurselike cells (NLCs) or mesenchymal bone marrow stromal cells (BMSCs) (\protect\hyperlink{ref-Collins1989}{Collins et al. 1989}; \protect\hyperlink{ref-Burger2000}{Jan A. Burger et al. 2000}; \protect\hyperlink{ref-Kurtova2009}{Kurtova et al. 2009}; \protect\hyperlink{ref-DeaglioMalavasi2009}{\textbf{DeaglioMalavasi2009?}}; \protect\hyperlink{ref-Purroy2015}{Purroy et al. 2015}). This observation indicated that the ability of CLL to progressively accumulate \emph{in vivo} may be highly dependent on external stimulation, rather than some cell-intrinsic feature of the tumour. Building on these observations, further \emph{in vitro} studies have shown a number of cell types and soluble factors belonging to the microenvironment are also capable of protecting the tumour cells from drugs and chemotherapeutic agents. Many CLL patients continue to harbour minimal residual disease (MRD),in which a fraction of the malignant cells remain whilst the patient is in remission and eventually lead to relapse (\protect\hyperlink{ref-Hayden2012}{Hayden et al. 2012}). It is believed that the tumour microenvironment provides a sanctuary for the malignant B cells to shield from the effects of therapy (\protect\hyperlink{ref-Dubois2020}{Dubois et al. 2020}).

\hypertarget{components-of-the-tumour-microenvironment}{%
\subsection{Components of the tumour microenvironment}\label{components-of-the-tumour-microenvironment}}

\textbf{Lymph Nodes}
Over the last two decades, significant progress has been made in unravelling this complex cross-talk and many of the important cellular and molecular components have been defined and studied. Malignant B cells circulate through the blood in a resting state, and follow chemokine gradients towards the lymph nodes to form ``profileration centres'' (Figure \ref{fig:proliferationCentres}), similar to germinal centres (\protect\hyperlink{ref-Herishanu2011}{\textbf{Herishanu2011?}}). Studies using detuerated water labelling have shown that up to 3\% of the clone is actively proliferating within the lymph node (\protect\hyperlink{ref-Messmer2005}{B. T. Messmer et al. 2005}; \protect\hyperlink{ref-Herndon2017}{Herndon et al. 2017}). Cross-talk with the non-neoplastic cells in the lymph node shapes the transcriptomic profile of the malignant B cells, and leads to upregulation of the BCR pathway \protect\hyperlink{ref-Mittal2014}{Mittal et al.} (\protect\hyperlink{ref-Mittal2014}{2014}), a central driver of CLL pathogenesis.


\begin{figure}

{\centering \includegraphics[width=0.5\linewidth]{figures/proliferation_centres} 

}

\caption{Haemotoxylin and eosin stain of CLL-infiltrated lymph node tissue section, showing pale-staining profileration centres (circled). \_Figure originally published in (\protect\hyperlink{ref-Kipps2017_}{\textbf{Kipps2017\_?}}).}\label{fig:proliferationCentres}
\end{figure}
\textbf{Bone marrow}
The bone marrow is also known to be important, and several studies have shown \emph{ex vivo} BMSCs to protect CLL cells against the drug toxicity (\protect\hyperlink{ref-Kay2007}{Kay et al. 2007}; \protect\hyperlink{ref-Kurtova2009}{Kurtova et al. 2009}). However, gene expression changes are less pronounced within the bone marrow compared to the lymph node (\protect\hyperlink{ref-Herishanu2011}{\textbf{Herishanu2011?}}).

\textbf{Cellular components}
Within these compartments, the CLL cells engage in a dialogue with mesenchymal stromal cells (MSCs), NLCs and follicular dendritic cells (FDCs), in concert with T cells, natural killer cells (NK cells) and components of the extracellular matrix (\protect\hyperlink{ref-tenHacken2016}{\textbf{tenHacken2016?}}). NLCs are of monocytic origin: their critical role in CLL was first demonstrated by the observation that peripheral blood-derived monocytes differentiate into NLCs, and that these cells prolong CLL cell survival \emph{ex vivo} (\protect\hyperlink{ref-Burger2000}{Jan A. Burger et al. 2000}). They are also found within the lymphoid tissues of CLL patients (\protect\hyperlink{ref-Tsukada2002}{Tsukada et al. 2002}; \protect\hyperlink{ref-Burkle2007}{Bürkle et al. 2007}). MSCs, which include BMSCs, are frequently observed within the secondary lymphatic tissues of CLL patients. Many studies have demonstrated the ability of these cells to inhibit spontaneous and drug-induced apoptosis in \emph{in vitro} CLL co-cultures \protect\hyperlink{ref-Kurtova2009}{Kurtova et al.} (\protect\hyperlink{ref-Kurtova2009}{2009}).

FDCs are important for tissue homing and retention of CLL cells within tissues. In healthy tissues, they are usually found within germinal centres (\protect\hyperlink{ref-Allen2008}{Allen and Cyster 2008}), and present unprocessed antigen to B cells. In CLL they play an important role within the secondary lymphoid organs, having a protective effect on CLL via cytokine secreton, adhesion molecules and the activation of BCR signalling (\protect\hyperlink{ref-Dubois2020}{Dubois et al. 2020}). CLL co-culture with FDCs leads to inhibition of spontaneous apoptosis and upregulation of anti-apoptotic MCL (\protect\hyperlink{ref-Pedersen2002}{Pedersen et al. 2002}).

The T cell compartment is also altered in CLL, first described by \protect\hyperlink{ref-Scrivener2003}{Scrivener et al.} (\protect\hyperlink{ref-Scrivener2003}{2003}). T cells have been observed to have pro-tumour and anti-tumour behaviour. On the one hand, higher numbers of CD4+ T-helper (Th) cells are seen in CLL patient blood samples (\protect\hyperlink{ref-Palma2017}{Palma et al. 2017}; \protect\hyperlink{ref-Catakovic2017}{\textbf{Catakovic2017?}}; \protect\hyperlink{ref-Elston2020}{Elston et al. 2020}), and in line with this, Th cell cytokines have been shown to provide pro-survival signals \emph{in vitro}, for example IL4 from Th2 cells (\protect\hyperlink{ref-Dancescu1992}{Dancescu et al. 1992}; \protect\hyperlink{ref-Bhattacharya2015}{Bhattacharya et al. 2015}; \protect\hyperlink{ref-AguilarHernandez2016}{Aguilar-Hernandez et al., n.d.}). Activated CD4+ T-cells in murine xenograft models (\protect\hyperlink{ref-Bagnara2011}{Bagnara et al. 2011}; \protect\hyperlink{ref-Os2013}{Os et al. 2013}) of CLL have also been shown to increase survival and growth of the tumour. On the other hand, there is evidence of increased antigen-experienced CD8+ T cells in CLL, which control tumour growth in a CLL mouse model (\protect\hyperlink{ref-Roessner2020}{Roessner and Seiffert 2020}; \protect\hyperlink{ref-Grioni2021}{Grioni et al. 2021}).

\hypertarget{intro-microenvironmental-pathways}{%
\subsection{Microenvironmental pathways}\label{intro-microenvironmental-pathways}}

Cross-talk between these non-neoplastic cells and the malignant B cells can occur directly, via cell-cell contacts and adhesion molecules, indirectly, via soluble factors that bind to receptors on the CLL cells, or through the exchange of material held in extracellular vesicles (\protect\hyperlink{ref-Guarini2008}{Guarini et al. 2008}; \protect\hyperlink{ref-Oppezzo2013}{Oppezzo and Dighiero 2013}; \protect\hyperlink{ref-Crompot2017}{Crompot et al. 2017}). Collectively these induce pathway activation (most importantly BCR and NF\(\kappa\)B (\protect\hyperlink{ref-Herishanu2011}{\textbf{Herishanu2011?}})) and gene expression modifications with the CLL cells, leading to chemotaxis, homing to lymphoid tissues and survival of the tumour cells (\protect\hyperlink{ref-Dubois2020}{Dubois et al. 2020}). Figure \ref{fig:microenvironmentSignalling} depicts an overview of this cross-talk.


\begin{figure}

{\centering \includegraphics[width=0.8\linewidth]{figures/microenivironmental_signalling} 

}

\caption{Graphic summarising soluble factors and cell-cell contacts involved in cross-talk between CLL cells and non-neoplastic cells of the tumour microenvironment \emph{Figure originally published in \protect\hyperlink{ref-Kipps2017}{Thomas J. Kipps et al.} (\protect\hyperlink{ref-Kipps2017}{2017}) .}}\label{fig:microenvironmentSignalling}
\end{figure}
\textbf{Cell-cell contacts}
The importance of direct contact between cells became clear with observations that the ability of MSCs to provide efficient rescue from spontaneous and drug-induced apoptosis is dependent on direct contact and can be blocked by separation through a filter (\protect\hyperlink{ref-Lagneaux1998}{Lagneaux et al. 1998}; \protect\hyperlink{ref-Burger2000}{Jan A. Burger et al. 2000}; \protect\hyperlink{ref-Kay2007}{Kay et al. 2007}; \protect\hyperlink{ref-Kurtova2009}{Kurtova et al. 2009}; \protect\hyperlink{ref-Ding2009}{Ding et al. 2009}). Likewise, FDCs also operate through direct contact, as evidenced by the observation that contact with HK cells (an FDC cell line) protects against CLL cells from apoptosis (\protect\hyperlink{ref-Pedersen2002}{Pedersen et al. 2002}).

This direct contact operates through a number of receptor-receptor interactions. For example, CLL-stromal cell binding involves \(\beta\) 1 integrin (ITGB1, or CD29) and \(\beta\) 2 integrin (ITGB2, or CD18) (\protect\hyperlink{ref-Lagneaux1999}{\textbf{Lagneaux1999?}}; \protect\hyperlink{ref-Lee2001}{\textbf{Lee2001?}}). VLA-4 is also an important integrin for retention of CLL cells within the lymph nodes and bone marrow, by interacting with its ligand VCAM-1 (or CD106) on stromal cells (\protect\hyperlink{ref-Burger2001}{J. A. Burger et al. 2001}).

These cell-cell interactions then lead to pathway activation (including BCR (\protect\hyperlink{ref-Stamatopoulos2015}{Basile Stamatopoulos et al. 2015}) and TLR (\protect\hyperlink{ref-Schulz2011}{Schulz et al. 2011})), gene expression changes and epigenetic changes (\protect\hyperlink{ref-Vangapandu2017}{Vangapandu et al. 2017}; \protect\hyperlink{ref-Xu2018}{Xu et al. 2018}) within the tumour cells. For example, contact between CLL B cells and MSCs alters the transcriptomic profile of the cells (\protect\hyperlink{ref-Schulz2011}{Schulz et al. 2011}; \protect\hyperlink{ref-Mangolini2018}{Mangolini et al. 2018}), leading to increased expression of anti-apoptotic proteins such as BCL2 (\protect\hyperlink{ref-Nwabo2012}{\textbf{Nwabo2012?}}; \protect\hyperlink{ref-Patel2013}{\textbf{Patel2013?}}), BCL-XL (\protect\hyperlink{ref-Patel2013}{\textbf{Patel2013?}}; \protect\hyperlink{ref-AmigoJimenez2015}{Amigo-Jiménez et al. 2015}), MCL1(\protect\hyperlink{ref-Kurtova2009}{Kurtova et al. 2009}; \protect\hyperlink{ref-AmigoJimenez2015}{Amigo-Jiménez et al. 2015}), and \(\beta\)-catenin (\protect\hyperlink{ref-Mangolini2018}{Mangolini et al. 2018}).

\textbf{Soluble Factors}
NLCs, MSCs, FDCs and T cells also secrete soluble factors that have a protective effect on the tumour. For example, MSCs secrete a number of cytokines. One of the most widely studied is SDF1-\(\alpha\) (or CXCL12), which interacts with CXCR4 on CLL cells (\protect\hyperlink{ref-Burger1999}{Jan A. Burger, Burger, and Kipps 1999}; \protect\hyperlink{ref-Burger2001}{J. A. Burger et al. 2001}; \protect\hyperlink{ref-Kay2007}{Kay et al. 2007}), stimulating the PI3K (\protect\hyperlink{ref-Burger2005}{M. Burger et al. 2005}), STAT3 (\protect\hyperlink{ref-Burger1999}{Jan A. Burger, Burger, and Kipps 1999}), and p44/42 MAPK (\protect\hyperlink{ref-Burger2000}{Jan A. Burger et al. 2000}) pathways which activates BTK (\protect\hyperlink{ref-Montresor2011}{\textbf{Montresor2011?}}), ERK (\protect\hyperlink{ref-Messmer2011}{D. Messmer et al. 2011}), and AKT (\protect\hyperlink{ref-OHayre2010}{O'Hayre et al. 2010}). FDCs on the other hand secrete B cell-activating factor (BAFF), which has been shown to increase survival of CLL cells through the activation of canonical NF\(\kappa\)B signalling (\protect\hyperlink{ref-Nishio2005}{Nishio et al. 2005}). A number of Th T-cell-derived cytokines have also been shown to increase CLL viability \emph{in vitro}, including IFN\(\gamma\) (\protect\hyperlink{ref-Buschle2003}{\textbf{Buschle2003?}}), IL15 (\protect\hyperlink{ref-Trentin1996}{Trentin et al. 1996}), IL21 (\protect\hyperlink{ref-deTotero2006}{Totero et al. 2006}; \protect\hyperlink{ref-Pascutti2013}{Pascutti et al. 2013}), IL4 (\protect\hyperlink{ref-Dancescu1992}{Dancescu et al. 1992}; \protect\hyperlink{ref-Bhattacharya2015}{Bhattacharya et al. 2015}; \protect\hyperlink{ref-AguilarHernandez2016}{Aguilar-Hernandez et al., n.d.}), IL2 (\protect\hyperlink{ref-Decker2010}{Decker et al. 2010}) and CD40L (\protect\hyperlink{ref-Kitada1999}{Kitada et al. 1999}; \protect\hyperlink{ref-Pascutti2013}{Pascutti et al. 2013}; \protect\hyperlink{ref-Bhattacharya2015}{Bhattacharya et al. 2015}). Certain soluble factors can also increase apoptosis and act against the tumour, including TGF\(\beta\) (\protect\hyperlink{ref-Lotz1994}{Lotz, Ranheim, and Kipps 1994}).

\hypertarget{intro-microenvironment-drug-response}{%
\subsection{The influence of the microenvironment on drug response}\label{intro-microenvironment-drug-response}}

In addition to their effect of spontaneous apoptosis, the cell types and soluble factors outlined above have also been shown to impact on drug-induced apoptosis \emph{in vitro}. For example, NLCs and stromal cells have been shown to meditate ibrutinib resistance (\protect\hyperlink{ref-Cheng2014}{Cheng et al. 2014}; \protect\hyperlink{ref-Boissard2015}{Boissard et al. 2015}; \protect\hyperlink{ref-Guo2017}{Guo et al. 2017}), and the chemotherapeutics fludarabine, oxaliplatin, chlorambucil, cyclophosphamide and doxorubicine show reduced efficacy in stromal cell co-cultures (\protect\hyperlink{ref-Kay2007}{Kay et al. 2007}; \protect\hyperlink{ref-Kurtova2009}{Kurtova et al. 2009}; \protect\hyperlink{ref-Mraz2011}{Mraz et al. 2011}; \protect\hyperlink{ref-Zhang2012}{Zhang et al. 2012}). A number of soluble factors also induce resistance to drugs \emph{in vitro}. These include decreased efficacy of fludarabine and venetoclax in the presence of TLR stimulation (\protect\hyperlink{ref-Fonte2013}{\textbf{Fonte2013?}}; \protect\hyperlink{ref-Jayappa2017}{Kallesh D. Jayappa et al. 2017}) and reduced sensitivity to ibrutinib in the presence of IL4 (\protect\hyperlink{ref-AguilarHernandez2016}{Aguilar-Hernandez et al., n.d.}) and BAFF (\protect\hyperlink{ref-McWilliams2019}{McWilliams et al. 2019}).

Evidence of microenvironmentally-induced drug resistance \emph{in vivo} is less prevalent, although there is widespread consensus that the microenvironment, in particular the lymph node, plays an important role in patient outcomes. Low rates of complete response and the inevitability of relapse in CLL have implicated the protective niche in enabling MRD (\protect\hyperlink{ref-OBrien2011}{O'Brien and Kay 2011}; \protect\hyperlink{ref-Hayden2012}{Hayden et al. 2012}). A number of studies have shown enlarged lymph nodes are associated with MRD (\protect\hyperlink{ref-Moreton2005}{Moreton et al. 2005}), in particular, incomplete response to ibrutinib is associated with persistently enlarged lymph nodes (\protect\hyperlink{ref-Ahn2018}{Ahn et al. 2018}). MSCs have also been shown to protect CLL cells taken from patients before and after \emph{in vivo} fludarabine therapy (\protect\hyperlink{ref-Trimarco2015}{Trimarco et al. 2015}).

In light of this, an important goal in CLL research is to develop strategies to overcome microenvironmentally-induced drug resistance. Targeting microenvironmental signalling in the lymph node tissue could be key to achieving long term remission and cure (\protect\hyperlink{ref-Hayden2012}{Hayden et al. 2012}) and thus there is a need for combinatorial therapies that aim to eliminate CLL cells in the lymph node and reduce CLL load in the peripheral blood. For example, \protect\hyperlink{ref-Guo2017}{Guo et al.} (\protect\hyperlink{ref-Guo2017}{2017}) have proposed cerdulatinib as a potential CLL therapy. Cerdulatinib is a dual inhibitor of the BCR pathway and the JAK-STAT pathway, capable of inducing cell death whilst also inhibiting the protective effects from the microenvironment.

The development of rational strategies to target the microenvironment requires a more comprehensive understanding of drug -- microenvironment interactions, and how these interplay with molecular features. Some studies have worked in this direction, for example \protect\hyperlink{ref-Jayappa2018}{Kallesh D. Jayappa et al.} (\protect\hyperlink{ref-Jayappa2018}{2018}) tested the impact of several agonists on ibrutinib and venetoclax. In a larger scale approach, (\protect\hyperlink{ref-Giminez2020}{\textbf{Giminez2020?}}) applied machine learning to identify drugs targeting proteins involved in microenvironmental signalling, and later screened these drugs in combination with venetoclax and ibrutinib, for activity against CLL in the presence of stromal cells.

Previous work in our lab probed drug activity in CLL peripheral blood mononuclear cell (PBMC) samples in the context of BMSC co-culture in a large-scale screen of 81 CLL patients (\protect\hyperlink{ref-HerbstThesis}{Herbst 2020}). Similar larger scale systematic studies of drug -- microenvironment interactions, particularly in the context of molecular features, are required.

\hypertarget{modelling-microenvironment-intro}{%
\subsection{Modelling the tumour microenvironment}\label{modelling-microenvironment-intro}}

A major goal of current research in CLL is to unravel the complexity of CLL-microenvironment cross-talk and its role in drug response. Many studies have applied a range of strategies to mimic the pro-survival effect of the microenvironment, each with their own advantages (\protect\hyperlink{ref-Crassini2017}{Crassini et al. 2017}; \protect\hyperlink{ref-Scielzo2020}{Scielzo and Ghia 2020}).

\textbf{Stimulation with soluble factors}
One such strategy is to stimulate individual pathways in CLL samples \emph{ex vivo} in order to elucidate their impact on CLL survival and drug response. For example, studies of BCR, TLR, CD40L and interleukin stimulation (\protect\hyperlink{ref-Muzio2009}{Muzio et al. 2009}; \protect\hyperlink{ref-Crassini2017}{Crassini et al. 2017}; \protect\hyperlink{ref-Scielzo2020}{Scielzo and Ghia 2020}) have proven critical in demonstrating the marked effect each of these have on CLL survival and the key downstream pathways involved (in particular NF\(\kappa\)B and MAPK) (\protect\hyperlink{ref-Crassini2017}{Crassini et al. 2017}). This strategy allows a direct understanding of cause and effect, with the caveat that accurately mimicking cytokine concentrations \emph{in vitro} is challenging, and that the activity of certain stimuli may be altered in the absence of other signals.

\textbf{Co-culture}
Stimulation studies omit the impact of cell-cell contacts and thus co-culturing CLL cells with cell lines can provide a more complete picture. Various co-culture systems have been developed in order to mimic different components of the microenvironment, including stromal cells, T cells, endothelial cells, NLCs and FDCs (\protect\hyperlink{ref-Panayiotidis1996}{Panayiotidis et al. 1996}; \protect\hyperlink{ref-Lagneaux1998}{Lagneaux et al. 1998}; \protect\hyperlink{ref-Pedersen2002}{Pedersen et al. 2002}; \protect\hyperlink{ref-Kurtova2009}{Kurtova et al. 2009}; \protect\hyperlink{ref-Stamatopoulos2010}{B. Stamatopoulos et al. 2010}; \protect\hyperlink{ref-Stamatopoulos2012}{Basile Stamatopoulos et al. 2012}; \protect\hyperlink{ref-Asslaber2013}{Asslaber et al. 2013}; \protect\hyperlink{ref-Hamilton2012}{Hamilton et al. 2012}; \protect\hyperlink{ref-Crassini2017}{Crassini et al. 2017}).

\textbf{3D Models}
In recent years, interest has developed in the use of 3D culture systems, to create yet more \emph{in vivo}-like models of the microenvironment (\protect\hyperlink{ref-Jensen2020}{Jensen and Teng 2020}; \protect\hyperlink{ref-Scielzo2020}{Scielzo and Ghia 2020}). Static 3D approaches involve the use of scaffolds or the generation of spheroids, recapitulating the complexity of the protective niche to a greater degree (\protect\hyperlink{ref-Farinello2018}{Farinello et al. 2018}; \protect\hyperlink{ref-Scielzo2020}{Scielzo and Ghia 2020}). More ambitious still is the development of dynamic 3D cultures, through the use of bioreactors and microfluidics. These systems attempt to capture the influences of gravity, flow and mechanical stresses, to study the phenotypic changes that occur as CLL cells traffic through and communicate with the non-neoplastic tissues (\protect\hyperlink{ref-Walsby2014}{Walsby et al. 2014}; \protect\hyperlink{ref-Scielzo2020}{Scielzo and Ghia 2020}).

\textbf{in vivo murine models}
Murine models to investigate CLL-microenvironment interactions are also possible (\protect\hyperlink{ref-Lu2004}{D. Lu et al. 2004}; \protect\hyperlink{ref-Enzler2009}{Enzler et al. 2009}; \protect\hyperlink{ref-Herishanu2011}{\textbf{Herishanu2011?}}; \protect\hyperlink{ref-Fedorchenko2013}{Fedorchenko et al. 2013}; \protect\hyperlink{ref-Simonetti2014}{Simonetti et al. 2014}; \protect\hyperlink{ref-Crassini2017}{Crassini et al. 2017}), though the value of these models can be hampered by species-specific biological differences (\protect\hyperlink{ref-Simonetti2014}{Simonetti et al. 2014}) and \emph{in vitro} modelling is often a more accurate approach (\protect\hyperlink{ref-Crassini2017}{Crassini et al. 2017}).

Of the many strategies to model the microenvironment, the reductionist approach of stimulating individual pathways is a useful tool to demonstrate direct causal relationships between signal and response. So far, most of these studies have investigated individual stimuli. Larger scale systematic studies of stimuli in other lymphomas have proven successful, such as work by \protect\hyperlink{ref-Carey2017}{Carey et al.} (\protect\hyperlink{ref-Carey2017}{2017}) to functionally screen many immune stimuli in Acute Myeloid Leukaemia (AML), suggesting that similar approaches could be valuable in CLL.

Moreover, most of these studies have been performed in smaller patient cohorts, omitting the influence of the molecular heterogeneity of CLL. Indeed, integrative studies of the interplay between external stimuli and cell-intrinsic features in CLL are lacking. A few studies have identified interactions between genetic features and the microenvironment, for example, \protect\hyperlink{ref-Martinez2016}{Martínez-Trillos et al.} (\protect\hyperlink{ref-Martinez2016}{2016}) have established a link between MYD88 mutations and TLR response, and \protect\hyperlink{ref-Chatzouli2014}{Chatzouli et al.} (\protect\hyperlink{ref-Chatzouli2014}{2014}) demonstrated a link between IGHV status and the response to TLR activation. In addition, \protect\hyperlink{ref-Mansouri2016}{Mansouri et al.} (\protect\hyperlink{ref-Mansouri2016}{2016}) have discussed the convergence of mutations and external signals on the NF\(\kappa\)B pathway.

The importance of interplay between microenvironment and molecular features in CLL survival and drug response is abundantly clear. However, a systematic study of the integrative influence of mutations and signals, particularly in the context of drug response, is missing in CLL.

\hypertarget{background-approaches-intro}{%
\section{Background to the approaches used in this thesis}\label{background-approaches-intro}}

This thesis explores drug - microenvironment - gene interplay in CLL through the analysis of \emph{ex vivo} perturbation assays combined with multi-omic profiling of patient samples. Background information on these experimental approaches, and associated data analysis, is outlined below.

\hypertarget{drug-perturbation-screens-intro}{%
\subsection{\texorpdfstring{\emph{Ex-vivo} drug pertubation screens}{Ex-vivo drug pertubation screens}}\label{drug-perturbation-screens-intro}}

Drug perturbation screens have been invaluable in identifying pathway dependencies, biomarkers and potential therapies in CLL (\protect\hyperlink{ref-Bosch2019}{Bosch and Dalla-Favera 2019}). Drug perturbation screens are usually performed in microtiter plates that contain a grid of wells suitable for performing an array of pharmacological or genetic experiments (\protect\hyperlink{ref-Letai2017}{Letai 2017}). Tumour cells, either cell lines or primary samples, can be deposited in each well to test their sensitivity to a set of compounds of interest. Tumour cells are incubated with each of the compounds, commonly dissolved in an aqueous solution of dimethyl sulfoxide (DMSO). After a set amount of time has passed to allow the cells to respond to the compound, the effect of each compound is measured. This ``read-out'' can take a number of forms. For example the morphology of the cells can imaged (\protect\hyperlink{ref-Snijder2017}{Snijder et al. 2017}; \protect\hyperlink{ref-HerbstThesis}{Herbst 2020}), or the cell viability can be measured via the number of cells or the level of adenosine triphosphate (ATP) in the well (\protect\hyperlink{ref-JCIpaper}{Dietrich et al. 2017}).

High-throughput drug screens of cancer cell lines have been widely used to link drug responses to molecular features (\protect\hyperlink{ref-Barretina2012}{Barretina et al. 2012}; \protect\hyperlink{ref-Basu2013}{Basu et al. 2013}; \protect\hyperlink{ref-Garnett2012}{Garnett et al. 2012}; \protect\hyperlink{ref-Iorio2016}{Iorio et al. 2016}). However, cell lines do not capture the genetic heterogeneity of a cancer (\protect\hyperlink{ref-Goodspeed2016}{Goodspeed et al. 2016}), and thus drug screening of primary samples can be more valuable (\protect\hyperlink{ref-JCIpaper}{Dietrich et al. 2017}; \protect\hyperlink{ref-Tyner2013}{Tyner et al. 2013}; \protect\hyperlink{ref-Pemovska2013}{Pemovska et al. 2013}; \protect\hyperlink{ref-Snijder2017}{Snijder et al. 2017}). In the case of CLL, screening primary samples has the caveat that CLL cells do not proliferate \emph{ex vivo} (\protect\hyperlink{ref-Collins1987}{\textbf{Collins1987?}}) and thus read-outs need to be taken on the basis of the rate of apoptosis relative to controls, rather than proliferation rate.

A few \emph{ex vivo} perturbation screens have also investigated the impact of microenvironmental stimulation on cancer biology, for example \protect\hyperlink{ref-Carey2017}{Carey et al.} (\protect\hyperlink{ref-Carey2017}{2017})'s functional screen of 94 cytokines in primary AML samples. Studies of stimuli are much rarer despite the well-recognised role of the microenvironment across haematological malignancies and other cancers.

Drug perturbation assays are also suited to combinatorial approaches, most commonly to test the efficacy of pairs of drugs in order to identify synergistic combinations (\protect\hyperlink{ref-Axelrod2014}{Axelrod et al. 2014}; \protect\hyperlink{ref-Lukas2020}{Lukas et al. 2020}). Combinatorial screening to test drug efficacy in the context of microenvironmental stimulation is also possible, though rare.

\hypertarget{multi-omics-cll-intro}{%
\subsection{Multiomics datasets to study CLL}\label{multi-omics-cll-intro}}

Multi-omics profiling of samples, in combination with \emph{ex vivo} perturbation screening, is a powerful approach to link cell phenotypes with molecular features in cancer:

\textbf{The concept of systems biology }
Systems biology considers biological entities as a set of complex molecular and environmental components that each interact to shape the functional phenotype of the system as a whole (\protect\hyperlink{ref-Anda2020}{Anda-Jáuregui and Hernández-Lemus 2020}). In cancer, disease pathogenesis and drug response is determined by complex interactions between mutations, epigenetic alterations, gene expression, metabolic abnormalities, and aberrant signalling functions (\protect\hyperlink{ref-Anda2020}{Anda-Jáuregui and Hernández-Lemus 2020}). Thus, the study of tumour biology, and indeed CLL biology, requires integrative methodologies and analyses to decipher this complex network (\protect\hyperlink{ref-Du2015}{Du and Elemento 2015}).

Enter multi-omics, an approach to studying biological systems that utilises multiple ``omic'' layers to study biological entities (\protect\hyperlink{ref-Anda2020}{Anda-Jáuregui and Hernández-Lemus 2020}; \protect\hyperlink{ref-Menyhart2021}{Menyhárt and Győrffy 2021}). These ``omic'' layers can encompass next-generation sequencing techniques, including DNA sequencing (DNAseq) and RNA sequencing (RNAseq) and high-throughput proteomics and metabolomics, along with newer single cell technologies and other sequence-based approaches such as ChIPseq (chromatin immunoprecipitation sequencing) and ATACseq (Assay for Transposase-Accessible Chromatin using sequencing) (\protect\hyperlink{ref-Anda2020}{Anda-Jáuregui and Hernández-Lemus 2020}). A number of methods have been built to integrate these diverse data types, calling on tools from statistics, probability, machine learning and network analysis (\protect\hyperlink{ref-Hernandez2013}{\textbf{Hernandez2013?}}; \protect\hyperlink{ref-Hernandez2014}{Hernández-Lemus 2014}; \protect\hyperlink{ref-Argelaguet2018}{Argelaguet et al. 2018}; \protect\hyperlink{ref-Anda2020}{Anda-Jáuregui and Hernández-Lemus 2020}).

In CLL, studies have profiled each of these layers independently, including the genomic (\protect\hyperlink{ref-Landau2015}{Landau et al. 2015}; \protect\hyperlink{ref-Puente2015}{Xose S. Puente et al. 2015}), transcriptomic (\protect\hyperlink{ref-Ferreira2014}{Ferreira et al. 2014}; \protect\hyperlink{ref-Zenz2019}{Zenz et al. 2019}), epigenomic (\protect\hyperlink{ref-Rendeiro2016}{Rendeiro et al. 2016}; \protect\hyperlink{ref-Beekman2018}{Beekman et al. 2018}; \protect\hyperlink{ref-Mallm2019}{Mallm et al. 2019}; \protect\hyperlink{ref-Rendeiro2020}{\textbf{Rendeiro2020?}}) and proteomic landscapes (\protect\hyperlink{ref-HerbstThesis}{Herbst 2020}; \protect\hyperlink{ref-MeierAbt2021}{Meier-Abt et al. 2021}).

Building on these studies, multi-omics approaches have the power to identify causal relationships between phenotypic layers of CLL, and thus have led to important biological insights and clinical perspectives (\protect\hyperlink{ref-JCIpaper}{Dietrich et al. 2017}; \protect\hyperlink{ref-Berest2019}{Berest et al. 2019}; \protect\hyperlink{ref-Lipsky2020}{Lipsky et al. 2020}; \protect\hyperlink{ref-HerbstThesis}{Herbst 2020}; \protect\hyperlink{ref-Lu2021}{J. Lu et al. 2021}). For example, a recent study in our lab integrated multiple data types to identify a novel biological axis, termed CLL proliferative drive, which is strongly associated with disease outcome (\protect\hyperlink{ref-Lu2021}{J. Lu et al. 2021}). Other multi-omics studies have to identified markers of drug response (\protect\hyperlink{ref-JCIpaper}{Dietrich et al. 2017}). Integration of ATACseq and RNAseq has also been used to determine differences in transcription factor (TF) activity in CLL between the two major subtypes (IGHV-M and IGHV-U) (\protect\hyperlink{ref-Berest2019}{Berest et al. 2019}). These studies highlight the importance of integrative approaches to gain further insights into CLL pathogenesis, especially with a view towards more personalised treatment strategies for patients.

\hypertarget{intro-multivariate-modelling}{%
\subsection{Mathematical modelling}\label{intro-multivariate-modelling}}

A number of mathematical tools are valuable in analysing complex multi-omics datasets. In this thesis, linear regression with and without lasso penalisation is used extensively to model the impact of the microenvironment and molecular features on drug response and a basic background to these approaches is outlined below.

** Basic linear regression **
Linear regression involves fitting a linear model to a dataset, to model the process that generated the data. For example, equation \eqref{eq:basiclinearmodelsymbols} describes a basic model to map the values of \(X\), to the values of \(Y\):
\begin{equation}
            Y = \beta_{0} + \beta_{1}X  + \epsilon
                                       \label{eq:basiclinearmodelsymbols}
    \end{equation}
This model specifies two components: the linear predictor \(\beta_{0} + \beta_{1}X\) and the error \(\epsilon\). The linear predictor can be compared to the equation for a straight line (\(Y = mX + c\)), where \(\beta_{0}\) represents the intercept (\(c\)), and \(\beta_{1}\), the gradient (\(m\)). The error \(\epsilon\) can be modelled by sampling from a normal distribution e.g.~\(N(0,\sigma^2)\), a normal distribution with mean zero and variance \(\sigma^2\).

Thus, we can model a variable \(Y\) via an expected value derived from the \(X\) independent variable(s), plus a random value derived from normal distribution with specified variance (\protect\hyperlink{ref-Walker2018}{Walker 2018}; \protect\hyperlink{ref-Huber2019}{Huber and Holmes 2019}).

\textbf{Multiple Linear Regression and Interaction Effects}
In certain cases, multiple independent variables may be predictors of the value of \(Y\). In these cases, multiple linear regression is required, involving more than one explanatory variable. The basic model, for two independent variables \(X_{1}\) and \(X_{2}\), is denoted as follows:
\begin{equation}
            Y = \beta_{0} + \beta_{1}X_{1} + \beta_{2}X_{2}   + \epsilon
                                       \label{eq:multiplelinearmodel}
    \end{equation}
In some cases, the effect of an independent variable on \(Y\) may depend on the on the value of another independent variable. For example, the effect of a drug on the viability of a CLL cell may depend on whether \emph{TP53} is mutated. Such ``interactions'' between independent variables can also be accounted for within linear models. These interactions are denoted as the product of two or more independent variables:
\begin{equation}
            Y = \beta_{0} + \beta_{1}X_{1} + \beta_{2}X_{2}  +\beta_{3} X_{1}X_{2}   + \epsilon
                                       \label{eq:interactionlinearmodel}
    \end{equation}
Here \(X_{1}X_{2}\) represents the interaction, and \(\beta_{3}\) is the associated regression coefficient. Higher-order interactions with more terms are also possible.

\textbf{Generalised Linear Models}
Not all dependent variables (\(Y\)) can be assumed to derive from sampling a normal distribution, as in equation \eqref{eq:basiclinearmodelsymbols}. For example, if \(Y\) is binary and takes only the values 0 or 1, a Bernoulli distribution is more appropriate. Generalised linear modelling builds on linear regression such that the response variable can have an error distribution other than the normal distribution (\protect\hyperlink{ref-Nelder1972}{Nelder and Wedderburn 1972}). A number of distributions are possible, including the binomial, Poisson and gamma distributions. Where \(Y\) is categorical, binomial or multinomial distributions are valuable; for count data, the Poisson distribution is often used.

\textbf{Lasso Regularisation}
When fitting such models to a particular dataset, it is important to avoid overfitting the data. Overfitting occurs when a model conforms too precisely to the dataset in hand. The model may not be representative of other datasets, and thus any predictions made using model are not reliable. Regularisation is an important tool to minimise such issues (\protect\hyperlink{ref-Kumar}{Kumar n.d.}).

Regularisation adds a penalty term to the best fit model. This reduces the influences of dependent variables on the value of \(Y\), by compressing the coefficients. This often acts to reduce the number of predictors and to generate a lesser variance with the test dataset. There are two main methods for this, named L1 Lasso Regression and L2 Ridge Regression. The models described in this thesis use Lasso Regression, which is more interpretable than Ridge Regression.

Lasso regularisation shrinks coefficients towards a central point, by adding a penalty to each coefficient, equal to the absolute value of its magnitude. This shrinkage approach means than some coefficients are reduced to 0 and are eliminated from the model, generating models that are sparse and have fewer parameters (\protect\hyperlink{ref-Tibshirani1996}{Tibshirani 1996}; \protect\hyperlink{ref-Kumar}{Kumar n.d.}).

This can make models simpler to interpret and can be useful in cases where the dependent variables are highly correlated, as in these cases only one of the correlated variables will usually be assigned a coefficient. This also has the caveat that the correlated features may each independently effect \(Y\), but only one of these variables will be deemed important by the model. It is also often impossible to determine whether one or all the correlated variables are truly influencing \(Y\). Thus, it is important to bear in mind that whilst modelling approaches have proven invaluable in advancing our understanding of biological processes, careful interpretation is required.

\hypertarget{CLL-model-intro}{%
\subsection{CLL as a model for studying tumour biology}\label{CLL-model-intro}}

CLL represents a valuable model system in cancer and studies of CLL can offer proof-of-principle for the application of new approaches in other entities. Primary PMBC samples are relatively simple to obtain, as CLL is the most common leukaemia and biopsies are performed by taking blood samples rather than intrusive operations. In addition, multiple biopsies cab be taken over the course of a patient's monitoring and therapy.

For example, the work by \protect\hyperlink{ref-Lu2021}{J. Lu et al.} (\protect\hyperlink{ref-Lu2021}{2021}) to decipher a new multi-omic marker of disease aggression not only represents an important advance in our understanding of CLL drive, it also demonstrates an integrative approach to the study of cancer which could be useful in other cancers, particular where the heterogeneity of outcome remains unexplained.

Moreover, CLL is widely viewed as the prototypic disease for studying the integrative role of cell-intrinsic and cell-extrinsic features in disease initiation, expansion and drug response (\protect\hyperlink{ref-Mansouri2016}{Mansouri et al. 2016}; \protect\hyperlink{ref-Srinivasan2020}{Srinivasan et al. 2020}; \protect\hyperlink{ref-Opezzo2021}{\textbf{Opezzo2021?}}). Thus, studies seeking to integrate the impact of cell-intrinsic and cell-extrinsic features on CLL survival and drug response are important in the study of CLL and beyond.

\hypertarget{methods}{%
\chapter{Methods}\label{methods}}

Methods in quotation marks are taken from Bruch \& Giles et al.~2021 and I have authored the original text, unless stated otherwise. Paragraphs without quotation marks were rewritten for the thesis.

Routine:
all headings should be here
at the end of each chapter, make sure all methods are included
Copy things in and put in quotation marks if from Bruch \& Giles et al.~2021, mark if they were not written by me, or if they were written jointly
Add REFERENCE and TABLE for now
Return to this later and neaten / expand on methods, and provide all referencing and tables
-fine to use quotations (although check plaigerism email), suggested rewriting as i / he / pmb throughout

\hypertarget{experimental-methods}{%
\section{Experimental methods:}\label{experimental-methods}}

\hypertarget{drug---stimulus-combinatorial-pertubation-assay}{%
\subsection{Drug - Stimulus Combinatorial Pertubation Assay}\label{drug---stimulus-combinatorial-pertubation-assay}}

\hypertarget{patient-sample-preparation}{%
\subsubsection{Patient Sample Preparation}\label{patient-sample-preparation}}

\emph{with Peter-Martin Bruch} Original
Peripheral Blood was taken from 192 patients for the initial drug-stimulation assay. Blood was separated by Ficoll gradient (GE Healthcare) and mononuclear cells were cryopreserved.

\hypertarget{preparation-of-screening-plates}{%
\subsection{Preparation of screening plates}\label{preparation-of-screening-plates}}

\emph{with Peter-Martin Bruch} Original
``Sample preparation, cell-culture, drug-stimulation profiling and genomic annotation was performed on 192 CLL patient samples as previously described{[}REFERENCE{]} with the following adjustments. Stimuli and drugs were mixed and preplated in the culture plates directly before adding the cell suspensions. RPMI-1640 and supplements were acquired from Gibco by Life Technologies, human serum was acquired from PAN Biotech (Cat.No. P40-2701, Lot.No:P-020317). Luminescence was measured after 48h on a Perkin Elmer EnVision.''

\emph{with Peter-Martin Bruch} Original
Compounds were obtained from Selleckchem, MedChemExpress and Sigma-Aldrich, dissolved in DMSO and stored at -\(20^\circ\)C. 12 drugs were used in two concentrations {[}TABLE{]}. Final DMSO concentration did not exceed 0.3\% in all experiments. Carfilzomib, Panobinostat and Venetoclax were removed from the analysis as they showed inconsistent toxicity depending on used media.
Insert table

Recombinant cytokines and stimulatory agents were dissolved according to manufacturer's protocol. 17 stimuli were selected {[}TABLE{]}. HS-5 conditioned medium was produced by incubating HS-5 stroma cell line (gifted by Martina Seiffert, DKFZ, Heidelberg) for 4 days at 37\(^\circ\)C and 5\% CO2, after which the supernatant was centrifuged and stored at -20\(^\circ\)C, the final concentration of HS-5 CM was 20\%. Bead immobilised anti-IgM was removed from the analysis due to storage instability. ''

Paper:
``Compounds and stimulatory agents were dissolved, stored and diluted according to manufacturer's protocol. HS-5 conditioned medium was produced by incubating HS-5 stromal cell line to \textgreater80\% confluency and cell removal by centrifugation. For a detailed list of stimuli and drugs and associated concentrations see Supp. Tables 1 and 2. Final DMSO concentration did not exceed 0.3\%.''

Original
Compounds were preplated in 384-well polypropylene storage plates (Greiner Bio-One Cat.No.:781271), which were stored at -20\(^\circ\)C. For each batch of samples tested on the same day a new storage plate was thawed and diluted in serum free RPMI, with or without stimuli. 5\(\mu\)L of drug-stimulation dilution were added into each well of the assay plates, 20 \(\mu\)L of cell suspension were added. The final cell concentration was 8*10\^{}5 cells/ml. Cells were thawed as previously described{[}REFERENCE{]}.

\hypertarget{drug-stimulation-assay}{%
\section{Drug-Stimulation assay}\label{drug-stimulation-assay}}

\emph{with Peter-Martin Bruch} Original
Drug-stimulation assays were performed with RPMI-1640 (Gibco by Life Technologies) supplemented with Penicillin Streptomycin (Gibco) final concentration of 100 Units/ml and 100 \(\mu\)g/ml respectively, L-Glutamine (Gibco) final concentration 2mM, and 10\% pooled, heat-inactivated and sterile filtered human type AB male off the clot serum (PAN Biotech, Cat.No. P40-2701, Lot.No:P-020317).

Drugs and stimuli were added to the cells simultaneously. Culture was performed in 384-well plates (Greiner Bio-One Cat.No.: 781904) and cells were incubated at 37\(^\circ\)C and 5\% CO2 for 48h.

Cell Viability was determined using the ATP-based CellTiter-Glo assay (Promega, Cat.No.:G7573). Luminescence was measured for the drug-stimulation assays using a Perkin Elmer EnVision and for the drug-drug interaction experiments using a Perkin Elmer EnSight, with a measurement time of 100ms per well.

\hypertarget{wes-rna-sequencing-targeted-sequencing-and-dna-copy-number-variants}{%
\section{WES, RNA Sequencing, Targeted Sequencing and DNA Copy number variants}\label{wes-rna-sequencing-targeted-sequencing-and-dna-copy-number-variants}}

\emph{Myself} Original
Sequencing data generated by Dietrich et al{[}REFERENCE{]}, and all sequencing and data processing were performed as described there. For

\hypertarget{follow---up-investigations}{%
\subsection{Follow - up investigations}\label{follow---up-investigations}}

\hypertarget{lymphocyte-doubling-time}{%
\subsubsection{Lymphocyte doubling time}\label{lymphocyte-doubling-time}}

\hypertarget{survival-data}{%
\subsubsection{Survival data}\label{survival-data}}

\hypertarget{atacsequencing-of-4-cll-samples}{%
\subsubsection{ATACsequencing of 4 CLL samples}\label{atacsequencing-of-4-cll-samples}}

\emph{with Peter-Martin Bruch (mostly edited by me)} Paper
``Peripheral blood was taken from 4 CLL patients and separated by Ficoll gradient (GE Healthcare), mononuclear cells were cryopreserved on liquid nitrogen. Samples were later thawed from frozen as previously described{[}REFERENCE{]} and MACS sorted for CD19 positive cells (Milteny autoMACS). The cells were resuspended in RPMI (GIBCO, Cat.No. 21875-034), with the addition of 2mM glutamine (GIBCO, Cat.No. 25030-24), 1\% Pen/Strep (GIBCO, Cat.No. 15140-122) and 10\% pooled, heat-inactivated and sterile filtered human type AB male off the clot serum (PAN Biotech, Cat.No. P40-2701, Lot.No:P-020317). 5ml of cell suspension was cultured in 6-well plates (Greiner Bio-One Cat.No. 657160). After thawing, cells were incubated at 37\(^\circ\)C and 5\% CO2 for 6 hours in 0.2\% DMSO. The final cell concentration was 2x10ˆ6 cells/ml. Cell viability and purity was assessed using FACS. All samples had a viability over 90\% and over 95\% of CD19+/CD5+/CD3- cells.''

\emph{with Nayara} Paper
``ATAC-seq libraries were generated as described previously{[}REFERENCE{]}. Cell preparation and transposition was performed according to the protocol, starting with 5x10ˆ4 cells per sample. Purified DNA was stored at −20\(^\circ\)C until library preparation was performed. To generate multiplexed libraries, the transposed DNA was initially amplified for 5x PCR cycles using 2.5 \(\mu\)L each of 25 μM PCR Primer 1 and 2.5 \(\mu\)L of 25 \(\mu\)M Barcoded PCR Primer 2 (included in the Nextera index kit, Illumina, San Diego, CA, USA), 25 \(\mu\)L of NEBNext High-Fidelity 2x PCR Master Mix (New England Biolabs, Boston, Massachusetts) in a total volume of 50 \(\mu\)L. 5 \(\mu\)L of the amplified DNA was used to determine the appropriate number of additional PCR cycles using qPCR. Additional number of cycles was calculated through the plotting of the linear Rn versus cycle, and corresponds to one-third of the maximum fluorescent intensity. Finally, amplification was performed on the remaining 45 \(\mu\)L of the PCR reaction using the optimal number of cycles determined for each library by qPCR (max. 13 cycles in total). The amplified fragments were purified with two rounds of SPRI bead clean-up (1.4x). The size distribution of the libraries was assessed on Bioanalyzer with a DNA High Sensitivity kit (Agilent Technologies, Santa Clara, CA), concentration was measured with Qubit DNA High Sensitivity kit in Qubit 2.0 Flurometer (Life Technologies, Carlsbad, CA). Sequencing was performed on NextSeq 500 (Illumina, San Diego, CA, USA) using 75bp paired-end sequencing, generating ∼450 million paired-reads per run, with an average of 55 million reads per sample.''

\hypertarget{spi-b-and-pu.1-shrna-knockdowns}{%
\subsubsection{Spi-B and PU.1 shRNA Knockdowns}\label{spi-b-and-pu.1-shrna-knockdowns}}

\hypertarget{immunohistochemistry-of-patient-lymph-nodes}{%
\subsubsection{Immunohistochemistry of patient lymph nodes}\label{immunohistochemistry-of-patient-lymph-nodes}}

\emph{Peter-Martin Bruch} Paper
``LN biopsies of CLL-infiltrated and non-neoplastic lymph nodes were Paraffin fixed and arranged in Tissue Microarrays. Consecutively they were stained for PAX-5 (790-4420, Roche), CD3 (790-4341, Roche), pSTAT6 (ab28829, abcam), STAT6 (519-4290, Zytomed Systems) and pIRAK4 (ab216513, abcam). The slides were analysed using Qupath 50 and the recommended protocol. Cell based data on mean staining intensity was exported and further analysed using R.''

\hypertarget{ibrutinib---il4---stat6i-interaction-assay}{%
\subsubsection{Ibrutinib - IL4 - STAT6i interaction assay}\label{ibrutinib---il4---stat6i-interaction-assay}}

\hypertarget{investigation-of-ibrutinib-il4-ibet-762}{%
\subsection{Investigation of Ibrutinib + IL4 + IBET-762}\label{investigation-of-ibrutinib-il4-ibet-762}}

\hypertarget{ibrutinib---il4---ibet-762-interaction-assay}{%
\subsubsection{Ibrutinib - IL4 - IBET-762 interaction assay}\label{ibrutinib---il4---ibet-762-interaction-assay}}

\hypertarget{ibrutinib---il4---ibet--762-treatment}{%
\subsubsection{Ibrutinib - IL4 - IBET -762 treatment}\label{ibrutinib---il4---ibet--762-treatment}}

\hypertarget{atacsequencing}{%
\subsubsection{ATACsequencing}\label{atacsequencing}}

\emph{with Peter Martin Bruch} get from diffTF
Peripheral blood was taken from 4 CLL patients and separated by Ficoll gradient (GE Healthcare), mononuclear cells were cryopreserved on liquid nitrogen. Samples were later thawed from frozen as previously described{[}REFERENCE{]} and MACS sorted for CD19 positive cells (Milteny autoMACS). The cells were resuspended in RPMI (GIBCO, Cat.No. 21875-034), with the addition of 2mM glutamine (GIBCO, Cat.No. 25030-24), 1\% Pen/Strep (GIBCO, Cat.No. 15140-122) and 10\% pooled, heat-inactivated and sterile filtered human type AB male off the clot serum (PAN Biotech, Cat.No. P40-2701, Lot.No:P-020317). 5ml of cell suspension was cultured in 6-well plates (Greiner Bio-One Cat.No. 657160). After thawing, cells were incubated at 37\(^\circ\)C and 5\% CO2 for 6 hours in 0.2\% DMSO. The final cell concentration was 2x10ˆ6 cells/ml. Cell viability and purity was assessed using FACS. All samples had a viability over 90\% and over 95\% of CD19+/CD5+/CD3- cells.

\emph{with Nayara} get from diffTF
ATAC-seq libraries were generated as described previously{[}REFERENCE{]}. Cell preparation and transposition was performed according to the protocol, starting with 5x10ˆ4 cells per sample. Purified DNA was stored at −20\(^\circ\)C until library preparation was performed. To generate multiplexed libraries, the transposed DNA was initially amplified for 5x PCR cycles using 2.5 \(\mu\)L each of 25 \(\mu\)M PCR Primer 1 and 2.5 \(\mu\)L of 25 \(\mu\)M Barcoded PCR Primer 2 (included in the Nextera index kit, Illumina, San Diego, CA, USA), 25 \(\mu\)L of NEBNext High-Fidelity 2x PCR Master Mix (New England Biolabs, Boston, Massachusetts) in a total volume of 50\(\mu\)L. 5 \(\mu\) L of the amplified DNA was used to determine the appropriate number of additional PCR cycles using qPCR. Additional number of cycles was calculated through the plotting of the linear Rn versus cycle, and corresponds to one-third of the maximum fluorescent intensity. Finally, amplification was performed on the remaining 45 \(\mu\)L of the PCR reaction using the optimal number of cycles determined for each library by qPCR (max. 13 cycles in total). The amplified fragments were purified with two rounds of SPRI bead clean-up (1.4x). The size distribution of the libraries was assessed on Bioanalyzer with a DNA High Sensitivity kit (Agilent Technologies, Santa Clara, CA), concentration was measured with Qubit DNA High Sensitivity kit in Qubit 2.0 Flurometer (Life Technologies, Carlsbad, CA). Sequencing was performed on NextSeq 500 (Illumina, San Diego, CA, USA) using 75bp paired-end sequencing, generating ∼450 million paired-reads per run, with an average of 55 million reads per sample.

\hypertarget{rnasequencing}{%
\subsubsection{RNAsequencing}\label{rnasequencing}}

\hypertarget{proteomics}{%
\subsubsection{Proteomics}\label{proteomics}}

\hypertarget{data-availability}{%
\section{Data availability}\label{data-availability}}

European Genome-Phenome Archive (EGA) accession \ldots. The data for \ldots. and computational analysis code used in this study are available from \ldots.

Additionally, we obtained CLL ATACseq data{[}REFERENCE{]} from the European Genome-phenome Archive (EGA: EGAD00001002110) and from (Bruch \& Giles et al.~2021)
For the ChIPseq analysis of SPIB binding sites, we accessed the data 37 from the NCBI GEO database 51, accession GEO: GSE56857. We made use of the data for SPIB in the OCILY3 DLBCL cell line (GSM1370276).

\hypertarget{statistical-analysis}{%
\section{Statistical Analysis}\label{statistical-analysis}}

Mixutre of paper and original
Integrative data analysis of screening data, DNA and RNA sequencing, CNV and methylation profiles and follow up experiments was performed using R version 4 (REFERENCE R Core Team, 2018) with the RStudio interface (REFERENCE RStudio Team, 2016) and using packages that included (\protect\hyperlink{ref-R-DESeq2}{Love, Anders, and Huber 2021}), (\protect\hyperlink{ref-R-survival}{Therneau 2021}), (\protect\hyperlink{ref-R-glmnet}{Friedman et al. 2021}), (\protect\hyperlink{ref-R-ConsensusClusterPlus}{Wilkerson and Waltman 2021}), (\protect\hyperlink{ref-R-clusterProfiler}{G. Yu 2021b}), (\protect\hyperlink{ref-R-ChIPseeker}{G. Yu 2021a}), (\protect\hyperlink{ref-R-genomation}{Akalin et al. 2021}) and (\protect\hyperlink{ref-R-BloodCancerMultiOmics2017}{Oles et al. 2021}) to perform univariate association tests, multivariate regression with and without lasso penalization, Cox regression, generalised linear modelling and clustering.

\hypertarget{data-processing}{%
\subsection{Data processing}\label{data-processing}}

\hypertarget{processing-of-screening-data}{%
\subsubsection{Processing of screening data}\label{processing-of-screening-data}}

\emph{with Peter-Martin Bruch} Paper
To quantify the response of a patient sample to each treatment condition, we used viability relative to the control, i.e., the CellTiter Glo luminescence readout of the respective well divided by the median of luminescence readouts of the DMSO control wells on the same plate.

\hypertarget{quality-assessment-and-control}{%
\subsubsection{Quality Assessment and Control}\label{quality-assessment-and-control}}

\hypertarget{rna-data}{%
\subsubsection{RNA data}\label{rna-data}}

RNA data was taken from the PACE repository (\protect\hyperlink{ref-PACE}{\textbf{PACE?}}). For detailed explanaition of processing see (\protect\hyperlink{ref-PACE}{\textbf{PACE?}}).

\hypertarget{atacsequencing-1}{%
\subsubsection{ATACsequencing}\label{atacsequencing-1}}

\hypertarget{data-from-pace}{%
\subsubsection{Data from PACE}\label{data-from-pace}}

\hypertarget{analysis-of-screening-data}{%
\subsection{Analysis of Screening Data}\label{analysis-of-screening-data}}

\hypertarget{correlations}{%
\subsubsection{Drug-drug and stimulus - stimulus correlations}\label{correlations}}

\emph{myself} original complete
Section \ref{cytokine-profiling}. Pearson correlation coefficients were calculated for each drug - drug and stimulus - stimulus pair, using the \texttt{cor} functions of the (\protect\hyperlink{ref-R-base}{R Core Team 2021}) package with log transformed viability values which were normalised to untreated controls.

\hypertarget{stimulus-heatmap-method}{%
\subsubsection{Characterising stimulus responses across all samples}\label{stimulus-heatmap-method}}

\emph{myself} original complete (just check)
Section \ref{clusters}. For the heatmap in figure \ref{fig:stimuliHeatmap}, the viability data is represented as z scores. Log transformed viability values, normalised to DMSO controls, are row-scaled according to the Median Absolute Deviance. Limits were then applied to this row scaling factor for optimal visualisation. The matrix of z scores was plotted using the (\protect\hyperlink{ref-R-pheatmap}{Kolde 2019}) package. The ordering of the columns (patient samples) was obtained from the dendrogram that resulted from running \texttt{ConsensusClusterPlus}, from the (\protect\hyperlink{ref-R-ConsensusClusterPlus}{Wilkerson and Waltman 2021}) package. The rows were ordered using the dendrogram order produced by \texttt{hclust} with default branch arrangement.

ADD detial on clustering

\hypertarget{univariate-gene-stimulus-associations-method}{%
\subsubsection{Univariate analysis of gene - stimulus and gene - drug response assosciations}\label{univariate-gene-stimulus-associations-method}}

\emph{myself} writtten for thesis complete
Section \ref{univariate-gene-stimulus-associations} \& \ref{univariate-gene-drug-associations}. Two-sided Student's t-tests, with equal variance were performed for IGHV status and somatic mutations and copy number aberrations with ≥3 patient samples in each group (n = 63/54). A 10\% FDR cut off was used to determine significance.

\hypertarget{multivariate-gene-stimulus-assosciations-method}{%
\subsubsection{Penalised multivariate regression of gene - stimulus and gene - drug assosciations}\label{multivariate-gene-stimulus-assosciations-method}}

\emph{myself} written for thesis (but some overlap with JCI)
Section \ref{multivariate-gene-stimulus-assosciations} \& \ref{multivariate-gene-drug-assosciations}.\\
ADAPT FOR DRUGS ALSO A Gaussian linear model with L1-penalty implemented in the \texttt{R} package \texttt{glmnet} (\protect\hyperlink{ref-R-glmnet}{Friedman et al. 2021}), with mixing parameter alpha = 1,was used to identify gene-stimulus associations. Matrix of genetic features (p=39), IGHV status (encoded as M = 1 and U = 0), and Methylation Cluster (encoded as 0, 0.5, 1) were used to identify multivariate predictors. All features were thus encoded on a similar scale to ensure equal treatment by lasso constraint in model fitting. Genetic features with more than 20\% missing values were excluded from the analysis, and samples without complete annotation for remaining features were removed, resulting in n = 129 samples. The matrix of control-normalised log-transformed viability values for these 129 samples was provided as the dependent variable. Using 3-fold cross-validation, the optimal penalty parameter \(\lambda\) was selected so as to minimise the cross-validated R2. The reduction in cross-validated mean squared error compared to the null model was used as loss. The model was fitted for 30 bootstrapped repeats, and the resulting coefficients are the mean of those coefficients that were selected in \textgreater75\% model fits.

\hypertarget{correlation-of-rna-receptor-expression-with-viability}{%
\subsubsection{Correlation of RNA-Receptor Expression with viability}\label{correlation-of-rna-receptor-expression-with-viability}}

\emph{myself} original complete
RNA count data for matched samples was available for 49 patients and was transformed using the variance stabilising transformation. Stimulus - receptor pairs were defined using the available literature, see table \ref{tab:receptorPairs}. For each stimulus, pearson correlation coefficient were calculated between the control-normalised log-transformed viability value of a sample and its expression of the corresponding stimulus receptor.

\hypertarget{drug-stimulus-linear-model-method}{%
\subsubsection{Linear modelling of drug - stimulus interactions}\label{drug-stimulus-linear-model-method}}

\emph{myself} original
Section \ref{drug-stimulus-linear-model} Linear models were fitted for each drug - stimulus combination, to extract a \(\beta{int}\) term and assossciated p value for each combinatorial treatment. Linear model was fitted using equation \eqref{eq:drugCytInt}, using the \texttt{lm} function of the \texttt{R} package stats (\protect\hyperlink{ref-R-base}{R Core Team 2021}):
\begin{equation}
            log(V) = \beta_{drug}X_{drug} + \beta_{stimulus}X_{stimulus} + \beta_{int}X_{drug}X_{stimulus} + \epsilon
                                       \label{eq:drugCytInt}
    \end{equation}
\emph{where \(V\) is the predicted viability with a given treatment,} \(\beta_{drug}\), \(\beta_{stimulus}\) \emph{and} \(\beta_{int}\) \emph{are coefficients for the drug, stimulus and combinatorial terms and} \(X_{drug}\) \emph{and} \(X_{stimulus}\) \emph{are indicator variables (0 or 1) for the presence or absence of a drug/stimulus.} \(\epsilon\) \emph{is the vector of model residuals. Equation from Bruch \& Giles et al.~2021.}

To fit model, the matrix of log-transformed viability values, for control, single and combinatorial treatments was used. Interactions were defined as significant if p value for \(\beta{int}\) \textless{} 0.05. The matrix of significant \(\beta{int}\) was then plotted as a heatmap, with the package \texttt{pheatmap}(\protect\hyperlink{ref-R-pheatmap}{Kolde 2019}) where the rows (stimuli) and columns (drugs) were ordered according to the dendrogram order produced by \texttt{hclust} (\protect\hyperlink{ref-R-base}{R Core Team 2021}) using default branch arrangement.

To define the four interaction categories, drug - stimulus combinations were first divided with respect to the sign of \(\beta{int}\), whereby a positive \(\beta{int}\) indicates that the viability with combinatorial treatment is higher than would be expected based on additive effects alone and vice versa. The groups were further divided into synergies and antagonisms according to the values of the model coefficients (\(\beta{drug}\), \(\beta{stimulus}\) and \(\beta{int}\)). Synergisms were assigned when coefficients for single treatments (\(\beta{drug}\) and \(\beta{stimulus}\)) were both greater than, or both less than, the observed coefficient for the combinatorial treatment (i.e.~\(\beta{drug}\) + \(\beta{stimulus}\) + \(\beta{int}\)). For positive antagonisms, \(\beta{drug}\) + \(\beta{stimulus}\) + \(\beta{int}\) was less than either \(\beta{drug}\) or \(\beta{stimulus}\). For negative antagonisms, \(\beta{drug}\) + \(\beta{stimulus}\) + \(\beta{int}\) was greater than either \(\beta{drug}\) or \(\beta{stimulus}\). All drug - stimulus interactions, for which p value for \(\beta{int}\) \textless0.05 fit into one of these groups.

\hypertarget{drug-stimulus-gene-interactions-method}{%
\subsubsection{Modelling of drug - stimulus - gene interactions}\label{drug-stimulus-gene-interactions-method}}

\emph{myself} original
Section \ref{drug-stimulus-gene-interactions}. To identify drug - stimulus interactions that were dependent on genetic features, we first fitted linear models to the viability matrix, for each drug - stimulus combination, using the formula in Eqn2. We extracted drug - stimulus interaction coefficients for each patient to generate a response matrix for n = 137 patients. Next, we performed multivariate regression using a Gaussian linear model with L1-penalty (i.e., lasso regression) as implemented in the R package \texttt{glmnet}(\protect\hyperlink{ref-R-glmnet}{Friedman et al. 2021}). As the dependent variable, the matrix of drug - stimulus interaction coefficients for each patient sample was used. As input to the model, genetic features with more than 20\% missing values were excluded, and only patients with complete annotation were included in the model ( n= 137 ). As predictors, the genetic mutations and CNVs (p= 39), IGHV status (coded as 0-1) and Methylation CLuster (coded as 0, 0.5, 1) were used, using 3-fold cross-validation. Misclassification error was used as loss for cross- validation.

Equation \eqref{eq:drugCytGeneInt} quantifies how the viability with any combination can be predicted:
\begin{equation}
            log(V) = \beta_{drug}X_{drug} + \beta_{stimulus}X_{stimulus} + \beta_{patient}X_{patient} + \beta_{drug-stimulus}X_{drug}X_{stimulus} + \beta_{drug-patient}X_{drug}X_{patient} + \beta_{stimulus-patient}X_{stimulus}X_{patient} +\beta_{int}X_{drug}X_{stimulus}X_{patient} + \epsilon
                                       \label{eq:drugCytGeneInt}
    \end{equation}
\emph{where \(V\) is the predicted viability of a patient sample with a given treatment,} \(\beta_{drug}\), \(\beta_{stimulus}\),\(\beta_{int}\), \(\beta_{drug-stimulus}\),\(\beta_{drug-patient}\),\(\beta_{drug-stimulus}\) \emph{and} \(\beta_{int}X_{drug}\) \emph{are regression coefficients for the drug, stimulus, patient sample and combinatorial terms and} \(X_{drug}\), \(X_{stimulus}\) \emph{and} \(X_{patient}\) \emph{are indicator variables (0 or 1) for the presence or absence of a drug/stimulus/patient sample.} \(\epsilon\) \emph{is the vector of model residuals. See also Methods section \ref{drug-stimulus-gene-interactions-method}. Equation from Bruch \& Giles et al.~2021.}

This generates a higher order interaction term \(\beta_{int}X_{drug}X_{stimulus}X_{patient}\). This term represents a \emph{patient sample-specific} \(\beta{int}\) for each drug - stimulus combination, quantifying the size of an interaction between a drug and stimulus in each patient genetic background.

With these patient sample-specific \(\beta{int}\) terms, it was possible to search for associations between the size of \(\beta_{int}\)) and genetic features. The aim was to screen for molecular features that increased or decreased the size of a drug - stimulus interactions, using multivariate regression with L1 (lasso) regularisation.

As input to the model, the response matrix was composed of the sample - specific \(\beta_{int}\) values for each drug-stimulus combination. To generate the feature matrix (137 samples versus 40 features), I excluded genetic features for which \textgreater20\% of the values were missing, and patient samples with incomplete annotation. As predictors, I included genetic mutations and CNVs (p=39) and IGHV status (coded as 0-1). I ran lasso regression, as implemented in the \texttt{R} package \texttt{glmnet}(\protect\hyperlink{ref-R-glmnet}{Friedman et al. 2021}), using three-fold cross-validation with misclassification error as loss. The resulting predictors are the mean of those coefficients that were selected in at least 90\% of 30 bootstrapped repeats.

\^{}elaborate?
\#\#\# Follow up
\#\#\#\# Lympocyte doubling times \{\#LDT-method\}
\emph{Herbst et al., 2020c} - update
Section \ref{cluster-survival}. ``Patients which had lymphocyte counts available for less than 4 time points between the sample
collection date and the time of the next treatment and patients currently in treatment were ex-cluded. Thus, \ldots{} patients with enough data remained. Lymphocyte growth rates were calculated by fitting a linear model to the log10 transformed lymphocyte counts of all timepoints between the sample collection date and the time of the next treatment versus the period of time.''

\hypertarget{survival-method}{%
\subsubsection{Survival analyses}\label{survival-method}}

\emph{myself} original
Section \ref{cluster-survival}. Survival analyses were performed using TTT, TFT and OS as metrics. Follow-up information to calculate OS was available for all 192 CLL patients. For 188 of 192 CLL patients treatment information after sample collection was available.

\emph{Herbst et al}
''Time to next treatment (TTT) was calculated from the date of sample collection to subsequent
treatment initiation. Patients without treatment initiation during the observation time and
patients who died before treatment initiation were censored at the latest follow-up contact.''

\emph{myself} original
For for the purpose of visualising Kaplan-Meier plots , optimal cut-points of staining levels were calculated using maximally selected rank statistics as computed by the \texttt{R} package \texttt{maxstat} (\protect\hyperlink{ref-R-maxstat}{\textbf{R-maxstat?}}). Based on these cut points, patients were split into two groups, and their survival data were plotted using the Kaplan-Meier method, using the R package \texttt{survminer} (\protect\hyperlink{ref-R-survminer}{Kassambara, Kosinski, and Biecek 2021}).

For Cox proportional hazards regression, the \texttt{coxph} function of the \texttt{R} package \texttt{survival} (\protect\hyperlink{ref-R-survival}{Therneau 2021}) was used. Expand

\hypertarget{cluster-genetics-method}{%
\subsubsection{Penalised multivariate regression genetic predictors of cluster membership}\label{cluster-genetics-method}}

\emph{Myself} original (edit so not similar to JCI)
Section \ref{genetics-method}. I used a multinomial linear model with L1-penalty, implemented in the \texttt{glmnet} (\protect\hyperlink{ref-R-glmnet}{Friedman et al. 2021}) package. As the dependent variable, the cluster assignment for each patient was used. As input to the model ,genetic features with more than 20\% missing values were excluded, and only patients with complete annotation were included in the model ( n= 137 ). As predictors, the genetic mutations and CNVs (p= 39) and IGHV status (coded as 0-1) were used, using 3-fold cross-validation. Misclassification error was used as loss for cross- validation. The resulting coefficients indicated associations between genetic features and each cluster.

\hypertarget{cluster-rna-method}{%
\subsubsection{Gene expression and gene set enrichment analysis between clusters}\label{cluster-rna-method}}

\emph{myself} original
Section \ref{cluster-rna}. For the n=49 patient samples for which viability data and RNA--Seq data for matching samples was available, the \texttt{R} package \texttt{DESeq2} (\protect\hyperlink{ref-R-DESeq2}{Love, Anders, and Huber 2021}) was used to search for associations of these two data types.RNA-Seq read count data was regressed on to the patient clusters C3, C4 (design formula \textasciitilde{} IGHV.status + Cluster). Genes were ranked by their test statistics and Gene Set Enrichment Analysis (GSEA) (implementing the fgsea algorithm with the \texttt{clusterProfiler} (\protect\hyperlink{ref-R-clusterProfiler}{G. Yu 2021b}) package) was applied to the ranked lists with the KEGG pathway gene set selections from the MSigDBdatabase{[}REFERENCE{]}.

\hypertarget{associations-of-ex-vivo-stimulus-responses-with-genomic-features}{%
\subsubsection{Associations of ex-vivo stimulus responses with genomic features}\label{associations-of-ex-vivo-stimulus-responses-with-genomic-features}}

We tested for associations between stimulus viability assay results and genomic features by Student's t-tests (two-sided, with equal variance). We tested somatic mutations (aggregated at the gene level), copy number aberrations and IGHV status. We restricted the analysis to features that were present in at least 3 patient samples (63 features). p-values were adjusted for multiple testing by applying the Benjamini-Hochberg procedure.

\hypertarget{gene-dosage-effects-method}{%
\subsubsection{Analysis of differential gene dosage in trisomy 12 CLL}\label{gene-dosage-effects-method}}

\emph{myself} written for thesis
Section \ref{gene-dosage-effects}. For all RNA samples available in PACE (i.e.~not just those that matched the samples in the screen), differential expression was called using the \texttt{DESeq2} package (\protect\hyperlink{ref-R-DESeq2}{Love, Anders, and Huber 2021}), with the design formula \textasciitilde trisomy12. Raw RNA counts were visualised if the gene had BH-adjusted p \textless{} 0.1 and belonged to TGF\(\beta\), JAK-STAT or TLR pathways genesets, as defined in the KEGG database (\protect\hyperlink{ref-Kegg}{\textbf{Kegg?}}) downloaded using the \texttt{msigdbr} package (\protect\hyperlink{ref-R-msigdbr}{Dolgalev 2021}). Proteomic abundance data was also plotted, provided by Sophie Herbst (\protect\hyperlink{ref-HerbstThesis}{Herbst 2020}). Proteomics data partially overlaps with RNAseq data. Protein quantification was performed as described in \protect\hyperlink{ref-HerbstThesis}{Herbst} (\protect\hyperlink{ref-HerbstThesis}{2020}).

\hypertarget{trisomy12-classifier-method}{%
\subsubsection{Generation of a Trisomy 12 classifier}\label{trisomy12-classifier-method}}

\emph{edit and finish}
Section \ref{trisomy12-classifier}. To build the classifier, a binomial linear model with L1-penalty implemented in the \texttt{R} package \texttt{glmnet} (\protect\hyperlink{ref-R-glmnet}{Friedman et al. 2021}), with mixing parameter alpha = 1, was used. To identify coefficients that predict trisomy 12 status, the viability matrix for all stimuli, as described in section \ref{stimulus-heatmap-method} was used, i.e.~z scores of the control-normalised log-transformed viability values.

Matrix of genetic features (p=39), IGHV status (encoded as M = 1 and U = 0), and Methylation Cluster (encoded as 0, 0.5, 1) were used to identify multivariate predictors. All features were thus encoded on a similar scale to ensure equal treatment by lasso constraint in model fitting. Genetic features with more than 20\% missing values were excluded from the analysis, and samples without complete annotation for remaining features were removed, resulting in n = 129 samples. The matrix of control-normalised log-transformed viability values for these 129 samples was provided as the dependent variable. Using 3-fold cross-validation, the optimal penalty parameter \(\lambda\) was selected so as to minimise the cross-validated R2. The reduction in cross-validated mean squared error compared to the null model was used as loss. The model was fitted for 30 bootstrapped repeats, and the resulting coefficients are the mean of those coefficients that were selected in \textgreater75\% model fits.

The classifier was built using binomial regression, with lasso penalisation, as implemented in the \texttt{R} package \texttt{glmnet} (\protect\hyperlink{ref-R-glmnet}{Friedman et al. 2021}). The feature matrix consisted of z scores of the viability values after treatment with each stimulus, and was used to predict the response (trisomy 12 status). I ran the model for 50 bootstrapped repeats, using three-fold cross-validation and mean absolute error as loss. Resiquimod, sCD40L+IL4 and TGF\(\beta\) were selected as coefficients that predict trisomy 12 status (Figure \ref{fig:tri12Classfier}, as would be expected bed on the observations in section \ref{trisomy12-modulator}.

\hypertarget{atacseq-processing}{%
\subsubsection{ATACseq processing}\label{atacseq-processing}}

\emph{with Ivan Berest} original (update to paper)
We downloaded CLL data (\protect\hyperlink{ref-Rendeiro2016}{Rendeiro et al. 2016}) from the European Genome-phenome Archive (EGA: EGAD00001002110). Original dataset had 88 ATAC-seq samples from 55 patients, however we used for the analysis only one sample per patient passing quality checks, resulting in 52 samples. We used an in-house constructed ATAC-seq processing pipeline (\protect\hyperlink{ref-Berest2019}{Berest et al. 2019}) to obtain final bam files mapped to the hg19 annotation genome that were corrected for CG bias.

\hypertarget{annotation-of-trisomy-12-status}{%
\subsubsection{Annotation of trisomy 12 status}\label{annotation-of-trisomy-12-status}}

\emph{with Ivan Berest} original (update to paper)
As trisomy 12 status was not included in the original metadata, we used a mean amount of reads in the chromatin accessible peaks for each sample to distinguish trisomy 12 patients. All samples containing 1.4 times more reads in the peaks located on chromosome 12, compared to the peaks on all other chromosomes, were classified as trisomy 12 patients.

\hypertarget{difftf-analysis-of-tf-activity-in-trisomy-12-cll}{%
\subsubsection{diffTF analysis of TF activity in trisomy 12 CLL}\label{difftf-analysis-of-tf-activity-in-trisomy-12-cll}}

\emph{with Ivan Berest} original (update to paper)
FOr the larger ATACseq dataset, we ran analytical mode of diffTF (\protect\hyperlink{ref-R-diffTF}{\textbf{R-diffTF?}}) pipeline to investigate differential TF activity between trisomy 12 and non-trisomy 12 patients using the HOCOMOCO v10 database (\protect\hyperlink{ref-HOCOMOCO}{Kulakovskiy et al. 2016}) with the following design formula: ``\textasciitilde{} Batch + Gender + IGHV status + trisomy12 status.''

for tirsomy 12 software uses the ATACseq counts at each of the binding sites across the genome to generate a distribution of fold changes between trisomy 12 and WT samples. This fold change distribution is compared to a background distribution of fold changes, calculated similarly using ATACseq counts for a GC-matched motif, that does not contain the TF binding motif. The TF is classified as differentially active between trisomy 12 and WT samples where these two distributions are significantly different. Each TF is assigned a weighted mean difference value, which quantifies the change in activity, and a p value.

For the smaller ATACseq dataset, ATAC-seq data generated from our CLL samples were processed similarly with the in-house ATAC-seq pipeline, as described previously (\protect\hyperlink{ref-Berest2019}{Berest et al. 2019}), with the only exception that we didn't use CG bias correction step. We used a similarly analytical mode of diffTF with HOCOMOCO v10 database (\protect\hyperlink{ref-HOCOMOCO}{Kulakovskiy et al. 2016}) using the following parameters: minOverlap = 1; design formula = ``\textasciitilde{} Patient + Trisomy 12 status.''

\hypertarget{functional-enrichment-analysis-of-spib-chipseq-data}{%
\subsubsection{Functional enrichment analysis of SPIB ChIPseq data}\label{functional-enrichment-analysis-of-spib-chipseq-data}}

\emph{myeslf} original (update to paper)
We downloaded SPIB ChIPseq data {[}REFERENCE{]} from the NCBI GEO database{[}REFERENCE{]}, accession \url{GEO:GSE56857}. We made use of the data for SPIB in the OCILY3 DLBCL cell line (GSM1370276). SPIB ChIP peaks were filtered for significance (q value\textless0.05). We used the annotatePeaks function from the package clusterProfiler{[}REFERENCE{]} to annotate the nearest gene for each ChIPpeak. We filtered peaks within ±1kb of a TSS of a gene and performed over-representation of KEGG and Reactome pathways (using the clusterProfiler package{[}REFERENCE{]}) amongst the resulting list of genes.

\hypertarget{drug-drug-interaction-assay}{%
\subsubsection{Drug-Drug interaction assay}\label{drug-drug-interaction-assay}}

An independent patient cohort of 16 patients was used in the drug-drug interaction experiments.

\hypertarget{survival-analysis-of-immunohistory-chemistry-data}{%
\subsubsection{Survival analysis of immunohistory chemistry data}\label{survival-analysis-of-immunohistory-chemistry-data}}

\hypertarget{ibrutinib---ibet-762---il-4-analysis}{%
\subsection{Ibrutinib - IBET-762 - IL-4 analysis}\label{ibrutinib---ibet-762---il-4-analysis}}

\hypertarget{rnaseq-processing}{%
\subsubsection{RNAseq processing}\label{rnaseq-processing}}

\hypertarget{deseq2-analysis}{%
\subsubsection{Deseq2 analysis}\label{deseq2-analysis}}

\hypertarget{atacseq-processing-1}{%
\subsubsection{ATACseq processing}\label{atacseq-processing-1}}

\hypertarget{difftf-analysis}{%
\subsubsection{diffTF analysis}\label{difftf-analysis}}

\hypertarget{generation-of-gene-regulatory-network}{%
\subsubsection{Generation of Gene Regulatory Network}\label{generation-of-gene-regulatory-network}}

\hypertarget{proteomics-processing}{%
\subsubsection{Proteomics Processing}\label{proteomics-processing}}

\hypertarget{proteomics-analysis}{%
\subsubsection{Proteomics Analysis}\label{proteomics-analysis}}

\hypertarget{data}{%
\chapter{Data}\label{data}}

Plan:
-headings {[}x{]}
-add all figures \textless-
-all text

proof text
proof figures
proof code
proof numbers

Done - send

\hypertarget{aims-of-the-project}{%
\section{Aims of the project}\label{aims-of-the-project}}
\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\#add label column to df}
\NormalTok{df }\OtherTok{\textless{}{-}}\NormalTok{ df }\SpecialCharTok{\%\textgreater{}\%}
\NormalTok{  dplyr}\SpecialCharTok{::}\FunctionTok{mutate}\NormalTok{(}\AttributeTok{Cytokine =} \FunctionTok{as.character}\NormalTok{(Cytokine)) }\SpecialCharTok{\%\textgreater{}\%}
\NormalTok{  dplyr}\SpecialCharTok{::}\FunctionTok{mutate}\NormalTok{(}\AttributeTok{Cytlabel =} \FunctionTok{ifelse}\NormalTok{(Cytokine }\SpecialCharTok{==} \StringTok{"IL{-}2"}\NormalTok{, }\StringTok{"IL2"}\NormalTok{,}
                            \FunctionTok{ifelse}\NormalTok{(Cytokine }\SpecialCharTok{==} \StringTok{"IL{-}4"}\NormalTok{, }\StringTok{"IL4"}\NormalTok{,}
                                   \FunctionTok{ifelse}\NormalTok{(Cytokine }\SpecialCharTok{==} \StringTok{"IL{-}6"}\NormalTok{, }\StringTok{"IL6"}\NormalTok{,}
                                          \FunctionTok{ifelse}\NormalTok{(Cytokine }\SpecialCharTok{==} \StringTok{"IL{-}10"}\NormalTok{, }\StringTok{"IL10"}\NormalTok{,}

                                                        \FunctionTok{ifelse}\NormalTok{(Cytokine }\SpecialCharTok{==} \StringTok{"IL{-}21"}\NormalTok{, }\StringTok{"IL21"}\NormalTok{,}
                                                               \FunctionTok{ifelse}\NormalTok{(Cytokine }\SpecialCharTok{==} \StringTok{"sCD40L+IL{-}4"}\NormalTok{,}\StringTok{"sCD40L + IL4"}\NormalTok{,}
                                                                      \FunctionTok{ifelse}\NormalTok{(Cytokine }\SpecialCharTok{==} \StringTok{"IL{-}15"}\NormalTok{, }\StringTok{"IL15"}\NormalTok{,}
                                                                             \FunctionTok{ifelse}\NormalTok{(Cytokine }\SpecialCharTok{==} \StringTok{"Interferon gamma"}\NormalTok{,}\StringTok{"Interferon \textbackslash{}u03B3"}\NormalTok{,}
                                                                                    \FunctionTok{ifelse}\NormalTok{(Cytokine }\SpecialCharTok{==} \StringTok{"SDF{-}1a"}\NormalTok{,}\StringTok{"SDF{-}1\textbackslash{}u03B1"}\NormalTok{,}
                                                                                           \FunctionTok{ifelse}\NormalTok{(Cytokine }\SpecialCharTok{==} \StringTok{"IL{-}1b"}\NormalTok{,}\StringTok{"IL1\textbackslash{}u03B2"}\NormalTok{,}
                                                                                                  \FunctionTok{ifelse}\NormalTok{(Cytokine }\SpecialCharTok{==} \StringTok{"TGF{-}b1"}\NormalTok{,}\StringTok{"TGF\textbackslash{}u03B21"}\NormalTok{, Cytokine)))))))))))) }\SpecialCharTok{\%\textgreater{}\%}

\NormalTok{  dplyr}\SpecialCharTok{::}\FunctionTok{mutate}\NormalTok{(}\AttributeTok{Cytokine =} \FunctionTok{factor}\NormalTok{(Cytokine))}

\CommentTok{\#generate table of Cytokines and assosciated labels}
\NormalTok{Cytokine\_labels }\OtherTok{\textless{}{-}}\NormalTok{ df }\SpecialCharTok{\%\textgreater{}\%} 
  \FunctionTok{filter}\NormalTok{(PatientID }\SpecialCharTok{==} \StringTok{"Pat\_001"}\NormalTok{, Drug }\SpecialCharTok{==} \StringTok{"DMSO"}\NormalTok{) }\SpecialCharTok{\%\textgreater{}\%} 
  \FunctionTok{select}\NormalTok{( Cytokine, Cytlabel)}

\FunctionTok{colnames}\NormalTok{(Cytokine\_labels) }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(}\StringTok{"name"}\NormalTok{, }\StringTok{"Cytlabel"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}
Characterisation of primary CLL samples by high-throughput combinatorial screening and multi-omic profiling. An introduction to the dataset that my PhD is based on.

\hypertarget{experimental-overview}{%
\section{Experimental overview}\label{experimental-overview}}

\hypertarget{high-throughput-combinatorial-pertubation-assay}{%
\subsection{High-throughput combinatorial pertubation assay}\label{high-throughput-combinatorial-pertubation-assay}}

``We measured the effects of 17 cytokines and microenvironmental stimuli on cell viability in 192 primary CLL samples and combined each one with 12 drugs to investigate the influence on spontaneous and drug-induced apoptosis (Figure \ref{fig:studyOverview}). Viability was assessed by ATP measurement via CellTiterGlo after 48 h of culture and normalised to untreated controls.''


\begin{figure}

{\centering \includegraphics[width=0.7\linewidth]{figures/studyOverview} 

}

\caption{Schematic of experimental protocol. By combining 12 drugs and 17 stimuli, we systematically queried the effects of simultaneous stimulation and inhibition of critical pathways in CLL (n = 192). Integrating functional drug-stimulus response profiling with four additional omics layers, we identified pro-survival pathways, underlying molecular modulators of drug and microenvironment reponses, and drug-stimulus interactions in CLL. \emph{Figure and caption from Bruch \& Giles et al.~2021.}}\label{fig:studyOverview}
\end{figure}
See whether to include this here, or jsut with otehr stuff
\#\#\# Validation experiments
In addition to the screen, we acquired the following validatory data (Figure \ref{fig:additionalData}).

Ibrutinib + IL4 - IBET figure

\begin{figure}

{\centering \includegraphics[width=1\linewidth]{figures/ibribetil4} 

}

\caption{Overview of the approach for the validation experiments. Four CLL PBMC samples were incubated with ibrutinib, IL4 and IBET-762, both individually and in all combinations. Samples were taken for ATAC sequencing and RNA sequencing after 6 hours and for mass spectrometry after 24 hours. Sequencing data was processed for downstream analysis, outline in Chapter \ref{chapter7}.}\label{fig:ibribetil4}
\end{figure}
Lymph nodes

Cell line experiments

\hypertarget{additional-eternal-datasets}{%
\subsection{Additional eternal datasets}\label{additional-eternal-datasets}}

Additional datasets (just describe)
-CLL ATACseq data32 from the EGA (EGAD00001002110)
ChIPseq data for Spi-B binding35 from the NCBI GEO database36 (GSE56857, GSM1370276)
Proteomics data from Sophie
Clinical data:
LDT
TTT
TFS
OS

\hypertarget{processing-of-raw-values-obtained-from-cell-viability-assay}{%
\section{Processing of raw values obtained from cell viability assay}\label{processing-of-raw-values-obtained-from-cell-viability-assay}}

\hypertarget{raw-values-and-quality-control}{%
\subsection{Raw values and Quality control}\label{raw-values-and-quality-control}}

\textbf{raw viabilites}

(Figure \ref{fig:rawViabilities}).


\begin{figure}

{\centering \includegraphics[width=0.7\linewidth]{thesis_files/figure-latex/rawViabilities-1} 

}

\caption{Boxplots of raw viability count data prior to normalisation and log transformation. For each of the 192 patient samples, there were 50 DMSO-treated wells. Dashed horizontal lines indicate potential cut-offs for samples that showed low viability with DMSO control treatment.}\label{fig:rawViabilities}
\end{figure}
Rest might not be necessary?
\textbf{any QC plots}

(Figure \ref{fig:screenQC}).



\hypertarget{data-normalisation-and-log-transformation}{%
\subsection{Data normalisation and log transformation}\label{data-normalisation-and-log-transformation}}

\textbf{demo plot}

(Figure \ref{fig:screenNormalisation}).



\hypertarget{characteristics-of-drugs-used-in-the-screen}{%
\section{Characteristics of drugs used in the screen}\label{characteristics-of-drugs-used-in-the-screen}}

\hypertarget{drug-pathways}{%
\subsection{Drug pathways}\label{drug-pathways}}

We screened 17 different drugs (Figure \ref{fig:drugCategories}).
\textbf{Add table with details of drugs}


\begin{figure}

{\centering \includegraphics[width=1px]{thesis_files/figure-latex/drugCategories-1} 

}

\caption{Bar plot of the drugs used in screen.}\label{fig:drugCategories}
\end{figure}
(Table \ref{tab:drugTable}).



add licencing status , otherwise this is good on the drugs
\begin{table}
\centering
\begin{tabular}{l|l|l|l|l|l|l|l|l|l}
\hline
\textbf{ } & \textbf{Drug} & \textbf{Main targets} & \textbf{Target category} & \textbf{Drug Group} & \textbf{Pathway} & \textbf{Distributor} & \textbf{Catalogue number} & \textbf{Conc. 1} & \textbf{Conc. 2}\\
\hline
\cellcolor[HTML]{E2E868}{D\_001} & \cellcolor[HTML]{E2E868}{Ibrutinib} & \cellcolor[HTML]{E2E868}{BTK} & \cellcolor[HTML]{E2E868}{BCR} & \cellcolor[HTML]{E2E868}{FDA approved} & \cellcolor[HTML]{E2E868}{B-cell receptor} & \cellcolor[HTML]{E2E868}{Selleck Chemicals} & \cellcolor[HTML]{E2E868}{S2680} & \cellcolor[HTML]{E2E868}{500nM} & \cellcolor[HTML]{E2E868}{50nM}\\
\hline
D\_002 & Idelalisib & PI3K delta & BCR & FDA approved & B-cell receptor & Selleck Chemicals & S2226 & 500nM & 50nM\\
\hline
\cellcolor[HTML]{E2E868}{D\_003} & \cellcolor[HTML]{E2E868}{Fludarabine} & \cellcolor[HTML]{E2E868}{Purine analogue} & \cellcolor[HTML]{E2E868}{DDR} & \cellcolor[HTML]{E2E868}{FDA approved} & \cellcolor[HTML]{E2E868}{Other} & \cellcolor[HTML]{E2E868}{Selleck Chemicals} & \cellcolor[HTML]{E2E868}{S1491} & \cellcolor[HTML]{E2E868}{2000nM} & \cellcolor[HTML]{E2E868}{200nM}\\
\hline
D\_004 & Nutlin-3a & MDM2 & DDR & Clinical development/
tool compound & Other & Selleck Chemicals & S8059 & 10000nM & 1000nM\\
\hline
D\_005 & Selumetinib & MEK1/2 & MAPK & Clinical development/
\cellcolor[HTML]{E2E868}{tool compound} & \cellcolor[HTML]{E2E868}{Other} & \cellcolor[HTML]{E2E868}{Selleck Chemicals} & \cellcolor[HTML]{E2E868}{S1008} & \cellcolor[HTML]{E2E868}{1000nM} & \cellcolor[HTML]{E2E868}{100nM}\\
\hline
D\_006 & BAY-11-7085 & NFkB & NFkB & Clinical development/
tool compound & Other & Selleck Chemicals & S7352 & 2000nM & 200nM\\
\hline
\cellcolor[HTML]{E2E868}{D\_007} & \cellcolor[HTML]{E2E868}{Everolimus} & \cellcolor[HTML]{E2E868}{mTOR} & \cellcolor[HTML]{E2E868}{mTOR} & \cellcolor[HTML]{E2E868}{FDA approved} & \cellcolor[HTML]{E2E868}{Other} & \cellcolor[HTML]{E2E868}{Selleck Chemicals} & \cellcolor[HTML]{E2E868}{S1120} & \cellcolor[HTML]{E2E868}{500nM} & \cellcolor[HTML]{E2E868}{50nM}\\
\hline
D\_008 & PRT062607 & SYK & BCR & Clinical development/
tool compound & B-cell receptor & Selleck Chemicals & S8032 & 500nM & 50nM\\
\hline
D\_009 & Pyridone-6 & JAK1/2/3 & JAK/STAT & Clinical development/
\cellcolor[HTML]{E2E868}{tool compound} & \cellcolor[HTML]{E2E868}{Other} & \cellcolor[HTML]{E2E868}{MedChemExpress} & \cellcolor[HTML]{E2E868}{457021-03-7} & \cellcolor[HTML]{E2E868}{500nM} & \cellcolor[HTML]{E2E868}{50nM}\\
\hline
D\_010 & Ralimetinib & p38 MAPK & MAPK & Clinical development/
tool compound & Other & Selleck Chemicals & S1494 & 1500nM & 150nM\\
\hline
D\_011 & Luminespib & HSP90 & HSP90 & Clinical development/
\cellcolor[HTML]{E2E868}{tool compound} & \cellcolor[HTML]{E2E868}{Other} & \cellcolor[HTML]{E2E868}{Selleck Chemicals} & \cellcolor[HTML]{E2E868}{S1069} & \cellcolor[HTML]{E2E868}{200nM} & \cellcolor[HTML]{E2E868}{20nM}\\
\hline
D\_012 & I-BET 762 & BRD2/3/4 & Epigenome & Clinical development/
tool compound & Other & Selleck Chemicals & S7189 & 1000nM & 100nM\\
\hline
\end{tabular}
\end{table}
\hypertarget{drug---drug-correlations}{%
\subsection{Drug - Drug Correlations}\label{drug---drug-correlations}}

(sense check)
\textbf{corr plot}

(Figure \ref{fig:drugCorrelations}).


\begin{figure}

{\centering \includegraphics[width=0.7\linewidth]{thesis_files/figure-latex/drugCorrelations-1} 

}

\caption{Raw viabilities}\label{fig:drugCorrelations}
\end{figure}
\hypertarget{drug-responses}{%
\subsection{Drug responses}\label{drug-responses}}

The drug responses were as follows (Figure \ref{fig:drugResponses}).


\begin{figure}

{\centering \includegraphics[width=0.7\linewidth]{thesis_files/figure-latex/drugResponses-1} 

}

\caption{Log transformed viability values for all drugs that were included in the screen after qulaity control. p values from student's t test.}\label{fig:drugResponses}
\end{figure}
\hypertarget{characteristics-of-stimuli-used-in-the-screen}{%
\section{Characteristics of stimuli used in the screen}\label{characteristics-of-stimuli-used-in-the-screen}}

\hypertarget{stimuli-info}{%
\subsection{The panel of stimuli}\label{stimuli-info}}

\textbf{table of stimuli}
-details of how were slected

(Table \ref{tab:cytTable}).



more on pathways? or is the diagram good neough? more on how they were choseN? not really my PhD?
\begin{table}
\centering
\begin{tabular}{l|l|l|l|l|l|l}
\hline
\textbf{Stimulus} & \textbf{Name} & \textbf{Supplier} & \textbf{Concentration} & \textbf{Catalogue Number} & \textbf{Lot Number} & \textbf{Pathway}\\
\hline
\cellcolor[HTML]{E2E868}{IL4 human recombinant animal component free} & \cellcolor[HTML]{E2E868}{IL4} & \cellcolor[HTML]{E2E868}{Sigma-Aldrich} & \cellcolor[HTML]{E2E868}{10 ng/ml} & \cellcolor[HTML]{E2E868}{SRP3093} & \cellcolor[HTML]{E2E868}{0712AFC14} & \cellcolor[HTML]{E2E868}{JAK/STAT}\\
\hline
IL10 human Animal component free & IL10 & Sigma-Aldrich & 10 ng/ml & SRP3312 & 1012AFC21 & JAK/STAT\\
\hline
\cellcolor[HTML]{E2E868}{IL2 human recombinant animal component free} & \cellcolor[HTML]{E2E868}{IL2} & \cellcolor[HTML]{E2E868}{Sigma-Aldrich} & \cellcolor[HTML]{E2E868}{10 ng/ml} & \cellcolor[HTML]{E2E868}{SRP3085} & \cellcolor[HTML]{E2E868}{0416AFC12} & \cellcolor[HTML]{E2E868}{JAK/STAT}\\
\hline
R-848 & Resiquimod & Enzo Life Siences & 1000 ng/ml & ALX-420-038-M025 & 10211615 & TLR 7/8\\
\hline
\cellcolor[HTML]{E2E868}{Human IL-21} & \cellcolor[HTML]{E2E868}{IL21} & \cellcolor[HTML]{E2E868}{Peprotech} & \cellcolor[HTML]{E2E868}{10 ng/ml} & \cellcolor[HTML]{E2E868}{200-21} & \cellcolor[HTML]{E2E868}{414226} & \cellcolor[HTML]{E2E868}{JAK/STAT}\\
\hline
Human BAFF & BAFF & Peprotech & 250 ng/ml & 310-13 & 0706CY194 & NFkB\\
\hline
\cellcolor[HTML]{E2E868}{Human IL-1 beta} & \cellcolor[HTML]{E2E868}{IL1β} & \cellcolor[HTML]{E2E868}{Peprotech} & \cellcolor[HTML]{E2E868}{10 ng/ml} & \cellcolor[HTML]{E2E868}{200-01} & \cellcolor[HTML]{E2E868}{0606B95} & \cellcolor[HTML]{E2E868}{NFkB}\\
\hline
Human sCD40 Ligand & sCD40L & Peprotech & 1000 ng/ml & 310-02 & 1214145 & NFkB\\
\hline
\cellcolor[HTML]{E2E868}{Goat F(AB')2 Fragment to human IgM} & \cellcolor[HTML]{E2E868}{soluble anti-IgM} & \cellcolor[HTML]{E2E868}{MP Biomedicals} & \cellcolor[HTML]{E2E868}{20000 ng/ml} & \cellcolor[HTML]{E2E868}{55055} & \cellcolor[HTML]{E2E868}{7227} & \cellcolor[HTML]{E2E868}{BCR}\\
\hline
Human TGFβ1 & TGFβ1 & Peprotech & 10 ng/ml & 100-21 & 1117209 & MAPK\\
\hline
\cellcolor[HTML]{E2E868}{Human IL15} & \cellcolor[HTML]{E2E868}{IL15} & \cellcolor[HTML]{E2E868}{Peprotech} & \cellcolor[HTML]{E2E868}{10 ng/ml} & \cellcolor[HTML]{E2E868}{200-15} & \cellcolor[HTML]{E2E868}{91624} & \cellcolor[HTML]{E2E868}{JAK/STAT}\\
\hline
Human IL6 & IL6 & Peprotech & 10 ng/ml & 200-06 & 031316-2 & JAK/STAT\\
\hline
\cellcolor[HTML]{E2E868}{ODN 2006 (ODN 7909)} & \cellcolor[HTML]{E2E868}{CpG ODN} & \cellcolor[HTML]{E2E868}{Invivogen} & \cellcolor[HTML]{E2E868}{1000 ng/ml} & \cellcolor[HTML]{E2E868}{tlrl-2006-1} & \cellcolor[HTML]{E2E868}{3901-09T} & \cellcolor[HTML]{E2E868}{TLR 9}\\
\hline
Human SDF1 alpha (CXCL12) & SDF-1α & Peprotech & 200 ng/ml & 300-28A & 101492 & JAK/STAT\\
\hline
\cellcolor[HTML]{E2E868}{Human Interferon gamma} & \cellcolor[HTML]{E2E868}{Interferon γ} & \cellcolor[HTML]{E2E868}{Peprotech} & \cellcolor[HTML]{E2E868}{5 ng/ml} & \cellcolor[HTML]{E2E868}{300-02} & \cellcolor[HTML]{E2E868}{121527} & \cellcolor[HTML]{E2E868}{NFkB}\\
\hline
HS-5 conditioned medium & HS-5 CM & self produced & 20 \% & NA & NA & NA\\
\hline
\end{tabular}
\end{table}
\hypertarget{stimulus-pathways}{%
\subsection{Stimulus pathways}\label{stimulus-pathways}}

\textbf{details of which stimuli activates with pathways}

\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{figures/microenvironmentCrosstalk} 

}

\caption{(ref:microenvironmentOverview)}\label{fig:microenvironmentCrosstalk}
\end{figure}
\hypertarget{heterogeneity-of-response-is-not-caused-by-differences-in-receptor-expression}{%
\subsection{Heterogeneity of response is not caused by differences in receptor expression}\label{heterogeneity-of-response-is-not-caused-by-differences-in-receptor-expression}}

Next, we tested whether the observed heterogeneity of response was caused by differences in receptor expression. We calculated Pearson correlation coefficients comparing control - normalised log viability values for each stimulus with transformed RNA counts of the corresponding stimulus receptor(s). No stimulus-receptor pair showed a Pearson coefficient greater than 0.4, confirming that the heterogeneity of response was not caused by differential receptor expression.



\hypertarget{characteristics-of-patient-samples-used-in-the-screen}{%
\section{Characteristics of patient samples used in the screen}\label{characteristics-of-patient-samples-used-in-the-screen}}

\hypertarget{genetic-data-available-for-each-patient}{%
\subsection{Genetic Data available for each patient}\label{genetic-data-available-for-each-patient}}

\textbf{table}
WES, CNVs, Methylation, Transcriptomic, ATACseq, LDT, survT, IHC
PLus Ibrutinib + IBET + IL4 treated samples - ATACseq, RNAseq, proteomics

(Figure \ref{fig:geneticData}).



\hypertarget{things-form-other-script}{%
\subsection{Things form other script}\label{things-form-other-script}}

\textbf{any otehrs?}

(Figure \ref{fig:dataCharacteristics}).



\hypertarget{generation-of-public-resource}{%
\section{Generation of public resource}\label{generation-of-public-resource}}

\hypertarget{shiny-app}{%
\subsection{Shiny app}\label{shiny-app}}

\href{https://www.imbi.uni-heidelberg.de/dietrichlab/CLL_Microenvironment/}{Shiny app}

(Figure \ref{fig:shinyApp1}).


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{figures/shiny1} 

}

\caption{Home page of the shiny app accompanying this project. The app was published along side \protect\hyperlink{ref-Giles2021}{Bruch et al.} (\protect\hyperlink{ref-Giles2021}{2021}) .}\label{fig:shinyApp1}
\end{figure}
(Figure \ref{fig:shinyApp3}).


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{figures/shiny3} 

}

\caption{Image of one of the tabs, in which the user can explore log-transformed viability values for different drug and stimulus treatments, stratified by genetic features.}\label{fig:shinyApp3}
\end{figure}
\hypertarget{package}{%
\subsection{Package}\label{package}}

\href{https://github.com/Huber-group-EMBL/CLLCytokineScreen2021}{package}

(Figure \ref{fig:package1}).

(ref:package1)
\begin{figure}

{\centering \includegraphics[width=0.7\linewidth]{figures/package} 

}

\caption{(ref:package1)}\label{fig:package1}
\end{figure}
\hypertarget{chapter4}{%
\chapter{\texorpdfstring{\emph{Ex-vivo} sensitivity to microenvironmental stimulation in primary CLL cells}{Ex-vivo sensitivity to microenvironmental stimulation in primary CLL cells}}\label{chapter4}}

Our assay quantified the effects of 17 cytokines and microenvironmental stimuli on cell viability in 192 primary CLL samples. The 17 stimuli were selected based on evidence in the literature that each stimulus had been shown to impact on CLL viability \emph{in vitro}, aiming to minimise redundancy among the compounds (see also Section \ref{stimuli-info}) (\protect\hyperlink{ref-Giles2021}{Bruch et al. 2021}). Many studies have applied various methods to model the impact of microenvironmental signalling on CLL, and each method has its strengths (see also Section \ref{modelling-microenvironment-intro}). Our assay represents a reductionist model of microenvironmental signalling, making it possible to dissect the effect of individual soluble factors within the protective niche on baseline viability.

This chapter details my analysis of the phenotypic effects of stimuli on CLL viability, leading to the identification of four patient subgroups that differ in their stimulus response profiles. This chapter also covers the clinical and molecular characterisation of these four subgroups.

The results presented in this chapter centre on viability values of CLL PMBCs samples treated with our panel of stimuli. CLL cells do not proliferate \emph{in vitro}, but rather undergo spontaneous apoptosis (\protect\hyperlink{ref-Collins1989}{Collins et al. 1989}). Treatment with various stimuli, or co-culture with NLCs or BMSCs can extend survival of CLL PBMCs \emph{ex vivo} (\protect\hyperlink{ref-Collins1989}{Collins et al. 1989}; \protect\hyperlink{ref-Burger2000}{Jan A. Burger et al. 2000}; \protect\hyperlink{ref-Kurtova2009}{Kurtova et al. 2009}; \protect\hyperlink{ref-Deaglio2009}{Deaglio and Malavasi 2009}; \protect\hyperlink{ref-Purroy2015}{Purroy et al. 2015}). To measure the individual phenotypes generated by each of our stimuli, we thus quantified viability by comparing ATP counts in treated samples, with those in DMSO wells, after 48 hours. A positive viability value indicates that the sample viability was increased relative to control. Values shown have additionally been log-transformed for the purposes of statistical analysis.

\hypertarget{prolifing-responses-to-the-panel-of-stimuli}{%
\section{Prolifing responses to the panel of stimuli}\label{prolifing-responses-to-the-panel-of-stimuli}}

\hypertarget{cytokine-profiling}{%
\subsection{\texorpdfstring{\emph{ex vivo} assay demonstrates functional diversity of cytokines and microenvironmental stimuli}{ex vivo assay demonstrates functional diversity of cytokines and microenvironmental stimuli}}\label{cytokine-profiling}}

To begin the analysis, I started by investigating heterogeneity amongst responses to the stimuli. I calculated Pearson correlation coefficients for each stimulus pair, using the log-transformed normalised viabilities (\protect\hyperlink{ref-Giles2021}{Bruch et al. 2021}). The resulting coefficients were ordered using hierarchical clustering and visualised in a symmetrical heatmap (Figure \ref{fig:stimuliCorrelations}).

In the resulting heatmap, several clusters of stimuli could be identified, including a larger group corresponding to agonists of TLR and Nf\(\kappa\)B pathways and a smaller group encompassing IL4 and TLR stimuli.


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/stimuliCorrelations-1} 

}

\caption{Heatmap of Pearson correlation coefficients of each pair of stimuli, based on log transformed viability values. See Methods section \ref{correlations}. \emph{Figure adapted from \protect\hyperlink{ref-Giles2021}{Bruch et al.} (\protect\hyperlink{ref-Giles2021}{2021}) .}}\label{fig:stimuliCorrelations}
\end{figure}
However, whilst certain stimuli clustered into groups, very few stimulus pairs showed any significant correlation. Almost all stimulus pairs showed little correlation (R \textless{} 0.6), including those that targeted similar downstream pathways, indicating a high degree of functional diversity amongst soluble factors in the CLL microenvironment. For example, JAK-STAT agonists such as IL4 and IL6 showed little correlation (Figure \ref{fig:correlationsScatter}A).

Only two stimulus pairs showed correlations where R \textgreater{} 0.6, and in both cases these targeted near identical receptors or downstream pathways. These included CpG ODN (TLR 9) and Resiquimod (TLR 7 and 8) (Figure \ref{fig:correlationsScatter}B), and IL4 and IL4 + CD40L which primarily target JAK3 - STAT6.

In contrast, correlation of drug - drug pairs demonstrated that drugs targeting components of the same pathway were highly correlated. For quality control purposes, this indicated that our data sensitively and specifically reflect inter-individual differences in pathway dependencies (\protect\hyperlink{ref-JCIpaper}{Dietrich et al. 2017}).


\begin{figure}

{\centering \includegraphics{thesis_files/figure-latex/correlationsScatter-1} 

}

\caption{Scatter plot of log-transformed viability values, normalised to DMSO controls, for (A) treatment with JAK-STAT agonists IL4 and IL6 and (B) treatment with TLR agonists CpG ODN and Resiquimod.}\label{fig:correlationsScatter}
\end{figure}
Having observed that microenvironmental stimulation induced diverse phenotypes between patient samples, I next aimed to visualise a global overview of these phenotypes. I plotted log-transformed viability values normalised to DMSO controls for all patient samples and all stimuli (Figure \ref{fig:stimuliBeeswarms}, (\protect\hyperlink{ref-Giles2021}{Bruch et al. 2021})).


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/stimuliBeeswarms-1} 

}

\caption{Log transformed viabilities after treatment with each stimulus. Stimuli are grouped by corresponding pathway, and responses are stratified by IGHV status. \emph{Figure adapted from \protect\hyperlink{ref-Giles2021}{Bruch et al.} (\protect\hyperlink{ref-Giles2021}{2021}) .}}\label{fig:stimuliBeeswarms}
\end{figure}
Figure \ref{fig:stimuliBeeswarms} shows that the majority of the stimuli increased viability, underlining the supportive nature of the microenvironment in CLL. However, four out of 17 reduced CLL viability relative to control, namely IL6, TGF\(\beta\) and TLR 7/8/9 agonists CpG ODN and Resiquimod in IGHV-mutated (IGHV-M) samples.

IL4 and TLR7/8/9 agonists Resiquimod and CpG ODN induced the strongest responses, an indication of their potency in modulating CLL cell survival. Notably, TLR agonists increased viability in certain samples, in most cases IGHV-U, and decreased viability in others, mostly IGHV-M. Our assay identified IL4 and TLR7/8/9 as key players in CLL-microenvironment cross-talk, and thus remain central throughout the results of this thesis.

\hypertarget{clusters}{%
\subsection{Microenvironmental response profiling identifies discrete patient subgroups}\label{clusters}}

To further investigate the variability in responses across the cohort, we next generated a heatmap of all stimuli responses across all samples, using z-scores for optimal visualisation. We performed consensus clustering on the resulting heatmap to group patients according to their response profiles (Figure \ref{fig:stimuliHeatmap}).


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/stimuliHeatmap-1} 

}

\caption{The heatmap matrix shows viability measurements for 192 samples (columns) and 17 stimuli (rows). The data are shown z-scores of log-transformed, control-normalised viability values. The colour bars to the right show sample annotations. Consensus clustering was used to define column tree layout, using hierarchical clustering with the Euclidean metric. See Methods section \ref{stimulus-heatmap-method}. \emph{Figure from \protect\hyperlink{ref-Giles2021}{Bruch et al.} (\protect\hyperlink{ref-Giles2021}{2021}) }}\label{fig:stimuliHeatmap}
\end{figure}
The consensus clustering method generates robust hierarchical clustering by subsampling from the matrix of values (in this case, the viability z-scores). Using subsampling, it is possible to calculate a ``consensus matrix'' which indicates the proportion of times each pair of values occupy the same cluster when subsampled together. This is repeated for different numbers of clusters, denoted by k, allowing the user to select the optimal number of clusters for a given dataset.

I ran consensus clustering on the matrix of z-scores for different values of k, and arranged the corresponding heatmaps accordingly. Figure @ref(fig:stimuliHeatmap shows the arrangement for k = 4. Comparing the heatmaps, I concluded on the existence of four robust clusters within the cohort (add these to appendix?). Each cluster shows a unique response profile to the panel of stimuli.

To validate the choice of four clusters, I additionally visualised summaries of the consensus matrices for each value of k, to quantify the degree of confidence in the clusters for each k (Figure \ref{fig:consensusClusteringA}).


\begin{figure}

{\centering \includegraphics[width=0.45\linewidth]{figures/consensus008} \includegraphics[width=0.45\linewidth]{figures/consensus009} 

}

\caption{(left) CDFs of the consensus matrices for k = 2 - 7, as indicated in the legend, estimated using 100 bin histogram. (right) Relative change in area under the CDF curve, for k = 2 - 7, to compare k with k - 1. In the case of k = 2, there is no k - 1, so the total area is plotted. Line shows relative increase in consensus between each value of k. See Methods section \ref{stimulus-heatmap-method}.}\label{fig:consensusClusteringA}
\end{figure}

\begin{figure}

{\centering \includegraphics[width=0.45\linewidth]{figures/consensus010} 

}

\caption{Assignment of patient samples (columns), to each cluster, for k = 1 - 7 (rows) to demonstrate stability of cluster membership. Cluster colour for k = 4 match those in heatmap in \ref{fig:stimuliHeatmap}. See Methods section \ref{stimulus-heatmap-method}.}\label{fig:consensusClusteringB}
\end{figure}
The graph of the CDFs of the consensus matrix for each k indicated that the CDF reaches a maximum and cluster confidence is maximised at k = 7, though above k = 4 there is little appreciable increase (Figure \ref{fig:consensusClusteringA}). This is confirmed in the graph showing relative change in the area under the CDF curve, showing there is only a small increase in consensus between k = 4 and k = 5, supporting the choice of k = 4. The cluster tracking plot depicts how each patient sample is assigned for each value of k. For k = 4, the plot indicates that C3 and 4 in particular are highly stable (Figure \ref{fig:consensusClusteringB}).

\hypertarget{functional-characterisation-of-patient-clusters}{%
\section{Functional characterisation of patient clusters}\label{functional-characterisation-of-patient-clusters}}

\hypertarget{c1---c4-show-distinct-response-profiles-with-the-panel-of-stimuli}{%
\subsection{C1 - C4 show distinct response profiles with the panel of stimuli}\label{c1---c4-show-distinct-response-profiles-with-the-panel-of-stimuli}}

Consensus clustering identified four patient subgroups, which each responded differently to the panel of stimuli. This raised the possibility that responses to microenvironmental stimulation are linked to cell-intrinsic features, and could be prognostic. Thus, I next aimed to characterise the phenotypic and molecular differences between these four clusters.

I began by investigating the differences between the response profiles of each cluster. I refer to the four clusters as C1 to C4: C1 and C2 were enriched for IGHV-U whilst the samples in C3 and C4 were mostly IGHV-M. Amongst the IGHV-U enriched C1 and C2, both showed strong, positive responses to IL4 and TLR7/8/9 stimulation. C2 could be distinguished by stronger responses to the stimuli overall, in particular to NF\(\kappa\)B agonists IL1\(\beta\), anti-IgM, BAFF and sCD40L. Amongst the IGHV-M enriched clusters, C3 showed weaker responses to the majority of stimuli, and C4 was defined by a negative response to TLR7/8/9 stimulation (Bruch \& Giles et al.~2021). Figure \ref{fig:clusterStimuliResponse} summarises these findings in more detail, showing responses stratified by cluster for a subset of the stimuli.


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/clusterStimuliResponse-1} 

}

\caption{Log-transformed normalised viability values stratified by cluster, for each stimulus. Stimuli activating the same pathway are grouped together. P-values from Student's t-test. \emph{Figure adapted from \protect\hyperlink{ref-Giles2021}{Bruch et al.} (\protect\hyperlink{ref-Giles2021}{2021}). }}\label{fig:clusterStimuliResponse}
\end{figure}
\hypertarget{cluster-survival}{%
\subsection{The clusters show differences in disease dynamics}\label{cluster-survival}}

To validate the potential biological significance of these four clusters, we investigated whether the groups showed differential \emph{in vivo} disease progression (Bruch \& Giles et al.~2021). The study design was such that not all patients in the cohort were treatment - free, which confounded the analysis. For that reason, lymphocyte doubling time (LDT)\footnote{LDT indicates the rate a which malignant B cells accumulate, and is an independent biomarker that correlates with overall survival (OS) (\protect\hyperlink{ref-Baumann2021}{Baumann et al. 2021}).} and time to next treatment (TTT)\footnote{TTT reflects the duration of time between the initiation of one therapy and the start of the next treatment (\protect\hyperlink{ref-Campbell2020}{Campbell et al. 2020}; \protect\hyperlink{ref-Delgado2021}{Delgado and Guddati 2021}). This measurement accounts for the period of treatment, along with the period in which the disease and/or symptoms are controlled.} were used to quantify CLL proliferative capacity, independently of treatment.

C1 and C2 showed a shorter LDT than C3 and C4, which is expected due to the differential proportions of IGHV-U and M patient samples in these groups (Figure \ref{fig:clusterLDTTTT}A). Notably, within the IGHV-M enriched clusters C3 and C4, samples in C3 showed a significantly shorter LDT (Student's t-test, p-value = 0.025).

To further validate this, we observed that TTT in the IGHV-M enriched C3 was significantly shorter than C4 (p-value = 0.005) and comparable to the progression dynamics of IGHV-U enriched C1 and 2 (Figure \ref{fig:clusterLDTTTT}B).


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/clusterLDTTTT-1} 

}

\caption{(A) LDT stratified by cluster, p-values from Student's t-test. (B) Kaplan-Meier curves to show TTT for each cluster. p-values from univariate Cox proportional hazard models comparing IGHV-U enriched C1 with C2, and IGHV-M enriched C3 with C4. See Methods sections \ref{LDT-method} and \ref{survival-method}. \emph{Figure from \protect\hyperlink{ref-Giles2021}{Bruch et al.} (\protect\hyperlink{ref-Giles2021}{2021}) .}}\label{fig:clusterLDTTTT}
\end{figure}
The difference in disease progression between the clusters indicated that microenvironmental response represents an additional biological layer, holding information relevant to disease dynamics. To validate that these clusters were not simply an indication of any underlying genetic features, we checked whether the observed differences in progression dynamics could be explained by other prognostic markers (Bruch \& Giles et al.~2021).

A multivariate Cox proportional hazard model accounting for IGHV status, trisomy 12 and TP53 in addition to the cluster assignment indicated an independent prognostic value for cluster assignment between C3 and C4 (p = 0.039, Table \ref{tab:clusterCox}).


\begin{table}

\caption{\label{tab:clusterCox}Table depicting results of Multivariate Cox Proportional Hazard Model to test prognostic value of key genetic features and clusters using Time to Next Treatment and C3 as reference. }
\centering
\begin{tabular}[t]{l|r|r|r|r|l}
\hline
Factor & coef & exp(coef) & se(coef) & z & p.value\\
\hline
\cellcolor[HTML]{E2E868}{Cluster 3 vs Cluster 1} & \cellcolor[HTML]{E2E868}{-0.03979} & \cellcolor[HTML]{E2E868}{0.96099} & \cellcolor[HTML]{E2E868}{0.29813} & \cellcolor[HTML]{E2E868}{-0.13347} & \cellcolor[HTML]{E2E868}{0.89382}\\
\hline
Cluster 3 vs Cluster 2 & 0.51595 & 1.67522 & 0.37741 & 1.36708 & 0.1716\\
\hline
\cellcolor[HTML]{E2E868}{Cluster 3 vs Cluster 4} & \cellcolor[HTML]{E2E868}{-0.82011} & \cellcolor[HTML]{E2E868}{0.44038} & \cellcolor[HTML]{E2E868}{0.39760} & \cellcolor[HTML]{E2E868}{-2.06267} & \cellcolor[HTML]{E2E868}{0.03914}\\
\hline
IGHV.status & 0.55192 & 1.73658 & 0.27253 & 2.02513 & 0.04285\\
\hline
\cellcolor[HTML]{E2E868}{trisomy 12} & \cellcolor[HTML]{E2E868}{-0.13357} & \cellcolor[HTML]{E2E868}{0.87496} & \cellcolor[HTML]{E2E868}{0.35617} & \cellcolor[HTML]{E2E868}{-0.37503} & \cellcolor[HTML]{E2E868}{0.70764}\\
\hline
TP53 & 1.38977 & 4.01395 & 0.26072 & 5.33058 & <0.0001\\
\hline
\end{tabular}
\end{table}
\hypertarget{the-clusters-show-differential-responses-to-drugs-in-vitro}{%
\subsection{\texorpdfstring{The clusters show differential responses to drugs \emph{in vitro}}{The clusters show differential responses to drugs in vitro}}\label{the-clusters-show-differential-responses-to-drugs-in-vitro}}

The potential clinical relevance of the clusters was underlined by my observation that the samples within each group showed differential responses to drugs \emph{in vitro} (Figure \ref{fig:clusterDrugResponse}).

As expected, the IGHV-U enriched clusters C1 and 2 were more sensitive to BCR inhibition by ibrutinib, idelalisib and PRT062607, than C3 and 4. Between C1 and C2, C2 was more sensitive to a number of the drugs, including idelalisib (SYK) (p = 0.012), everolimus (mTOR) (p = 0.02) and the chemotherapeutics fludarabine (p = 0.031) and nutlin-3a (p = 0.042). Amongst C3 and C4, C3 showed lower sensitivity to everolimus (p = 0.051) and to fludarabine (p \textless{} 0.001) and nutlin-3a (p = 0.01). This aligns with the observation that patients in C3 have a poorer prognosis, despite the fact most of these samples were annotated as IGHV-M. C4 also showed a positive response to Nf\(\kappa\)B inhibition by BAY-11-7085, and p38 MAPK inhibition by ralimetinib.

Such differential drug response patterns suggests that microenvironmental response may reflect disease-specific CLL biology.


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/clusterDrugResponse-1} 

}

\caption{Log-transformed normalised viability values, stratified by cluster, for each drug. Drugs targeting the same pathway are grouped together. P-values from Student's t-test.}\label{fig:clusterDrugResponse}
\end{figure}
\hypertarget{cluster-genetics}{%
\subsection{The clusters are enriched for different genetic features}\label{cluster-genetics}}

Next I assessed differences in the molecular profiles of samples within each cluster. Visually, it appeared that certain clusters were enriched or depleted for various genetic features recurrent in CLL (Figure \ref{fig:clusterMatrix}).


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/clusterMatrix-1} 

}

\caption{Distribution of selected genetic features (rows) within each cluster for all patient samples (columns). White indicates patient sample is not annotated for that feature.}\label{fig:clusterMatrix}
\end{figure}
To quantify whether each cluster was enriched or depleted for certain genetic features, I used a multinomial modelling approach, with lasso penalisation (for more background on generalised linear models, see section \ref{intro-multivariate-modelling} (\protect\hyperlink{ref-Giles2021}{Bruch et al. 2021}).
I ran the multinomial model to assign coefficients to genetic features that were associated with each cluster, C1 to C4. Running the regression with lasso regularisation meant that only genetic features that were enriched or depleted in a cluster were selected by the model, and coefficients for all other genetic features were shrunk towards 0.

To run the model, I generated a feature matrix, consisting of the genetic data for patient samples and a discrete response matrix, which included the cluster assigned to each patient. Genetic features with \textgreater20\% missing values were excluded (n = 39), and only patients with complete annotation were included in the feature matrix (n = 137). I used the function \texttt{cv.glmnet} from the \texttt{R} package \texttt{glmnet} (\protect\hyperlink{ref-R-glmnet}{Friedman et al. 2021}) to generate the model, using 3-fold cross validation, and selecting the optimal model using \(\lambda{min}\).

Each cluster was thus assigned a set of genetic coefficients. To ensure that these were robust, I applied certain cut-offs. I ran 50 bootstrapped repeats, and removed coefficients if they were not selected in at least 60\% of cases. Additionally, I applied a minimum value of 0.35.

Figure \ref{fig:clusterEnrichment} shows the resulting mean coefficients. A positive coefficient indicated that this feature was enriched in the cluster, and a negative coefficient indicated it was depleted. The associated standard deviations are also indicated.


\begin{figure}

{\centering \includegraphics[width=0.8\linewidth]{thesis_files/figure-latex/clusterEnrichment-1} 

}

\caption{Enrichment and depletion of genetic features within each cluster, quantified using lasso-penalised regression. x axis shows genetic features, y axis indicates value and sign of coefficient assigned to feature, for each cluster (positive coefficients are enriched in the cluster, negative coefficients are depleted). Coefficients shown are mean coefficients from 50 bootstrapped repeats and error bars represent the mean plus and minus standard deviation. See Methods section \ref{cluster-genetics-method}. \emph{Figure from Bruch \& Giles et al 2021. }}\label{fig:clusterEnrichment}
\end{figure}
As expected, IGHV status was the main feature that predicted cluster membership. Beyond IGHV status, trisomy 12 and \emph{SF3B1} mutations were enriched in C2, which showed enhanced responses to many stimuli. C4, which was associated with slow in-vivo progression, showed depletion of \emph{TP53}, \emph{ATM}, mutations in \emph{RAS} and \emph{RAF} and gain(8q).

\hypertarget{cluster-rna}{%
\subsection{GSEA of DE genes between subgroups}\label{cluster-rna}}

In addition to genetic features, I investigated differential expression of genes within each cluster. For n = 49 samples, RNAseq data was available for matched PBMC samples. I focused on the difference between C3 and C4, for which 21 RNAseq samples were available (Bruch \& Giles et al.~2021).

To quantify differential gene expression, I began by filtering out immunoglobulin genes, including genes at the heavy, light and kappa loci that encode the antigen receptor of B cells. The clusters each show differential enrichment of IGHV-M and U samples, and thus the differential expression analysis would otherwise be dominated by immunoglobulin genes that are affected by this biomarker.

I followed the \texttt{DESeq2} protocol (\protect\hyperlink{ref-R-DESeq2}{Love, Anders, and Huber 2021}) using a design formula to quantify the difference between clusters, and accounting for the confounding effect of IGHV status. 87 genes were differentially expressed (adjusted p \textless{} 0.05) between C3 and 4 (Figure \ref{fig:clusterDE}).


\begin{figure}

{\centering \includegraphics[width=0.7\linewidth]{thesis_files/figure-latex/clusterDE-1} 

}

\caption{Volcano plot of differentially expressed genes between C3 and C4. x axis indicates log2 fold change values, calculated using the \texttt{DESeq2} package (\protect\hyperlink{ref-DESeq2}{\textbf{DESeq2?}}), y axis gives corresponding -log10(BH-adjusted p value). Darker grey points are labelled where adjusted p \textless{} 0.01. See Methods section \ref{cluster-rna-method}. \emph{Figure from Bruch \& Giles et al.~2021.}}\label{fig:clusterDE}
\end{figure}
To assign biological meaning to the differentially expressed genes, I quantified the enrichment of Hallmark pathways amongst the genes. I ranked the genes based on the Wald statistic, and then ran GSEA using the fgsea algorithm (Figure \ref{fig:clusterGSEA}) (Bruch \& Giles et al.~2021).

Several pathways were upregulated amongst samples in C3, compared to C4, indicating that these pathways may relate in some way to the shorter TTT and LDT of patients within this cluster. Pathways associated with higher disease aggression were regulated in C3 including genesets relating to stress response (Unfolded Protein Response, UV Response Up, P53 Pathway), metabolism (Oxidative Phosphorylation) and proliferation (G2M Checkpoint, MYC Targets V1, MTORC1 Signaling, E2F Targets) (Figure \ref{fig:GSEApathways}).


\begin{figure}

{\centering \includegraphics[width=0.7\linewidth]{thesis_files/figure-latex/clusterGSEA-1} 

}

\caption{Bar plot showing gene set enrichment to compare gene expression of samples in C3 and C4. Normalised enrichment scores (NES) are shown for top ten pathways with smallest adjusted p-value. Gene ranking performed based on Wald statistic, calculated using the \texttt{DESeq2} package. See Methods section \ref{cluster-rna-method}. \emph{Figure from Bruch \& Giles et al.~2021. }}\label{fig:clusterGSEA}
\end{figure}
In addition, C3 showed upregulation of microenvironmental signalling pathways relative to C4, including TNFa Signalling via NFKB and Interferon Gamma Response (Figure \ref{fig:GSEApathways}) (Bruch \& Giles et al.~2021). This finding underlines our hypothesis that differential activity of microenvironmental signalling, both \emph{in vivo} and \emph{ex vivo} may be relevant to disease prognosis.


\begin{figure}

{\centering \includegraphics{thesis_files/figure-latex/GSEApathways-1} 

}

\caption{Enrichment plot of selected pathways. Gene set enrichment analysis (GSEA) was performed with the Hallmark gene sets from the GSEA Molecular Signatures Database. Wald statistic was used to rank the genes. The green curve corresponds to the Enrichment Score curve, which is the running sum of the weighted enrichment score obtained from GSEA software. See Methods section \ref{cluster-rna-method}. \emph{Figure and caption from Bruch \& Giles et al.~2021.}}\label{fig:GSEApathways}
\end{figure}
\hypertarget{summary}{%
\section{Summary}\label{summary}}

The screen enabled a systematic study of the impact of individual microenvironmental pathways on CLL viability. The assay highlighted potent pro-survival signals that are active across heterogeneous genetic backgrounds, such as IL4, and others that operate in subsets of patients, such as TLR. IL4 and TLR represent the key pathways that we focus on throughout the rest of this thesis.

In addition, this screen demonstrates the ability of microenvironmental response profiling, to distinguish disease subgroups. The assay revealed four subgroups with distinct response profiles and molecular properties and clinical outcomes, suggesting that microenvironmental response holds biologically significant information that may be relevant to prognosis and treatment decision making.

\hypertarget{chapter5}{%
\chapter{Genetic modulators of responses to microenvironmental stimulation}\label{chapter5}}

Profiling the effects of the panel of stimuli revealed the heterogenous nature of responses to stimulation. The analysis revealed four patient sample groups, with distinct responses to the stimuli, unique molecular profiles, and differential disease progression. Two pathways emerged as the most potent modulators of CLL viability, namely IL4 and TLR. These too induced heterogeneous responses across the cohort, in particular stimulation of TLR 7/8/9 increased viability in some samples, whilst it reduced viability in others.

Next we asked to what extent this heterogeneity of response relates to the molecular profiles of the tumours. We sought to understand which underlying genetic features might modulate responses to external signals, and how these interactions may occur. We combined the screening dataset with multi-omics profiles of the patient samples taken from the PACE repository (\protect\hyperlink{ref-JCIpaper}{Dietrich et al. 2017}; \protect\hyperlink{ref-R-BloodCancerMultiOmics2017}{Oles et al. 2021}) and performed a systematic survey of molecular determinants of stimulus response, using whole-exome sequencing, DNA-methylation, RNA-sequencing and copy number variant data. In addition, for a small subset of patients we also generated ATAC sequencing and Mass Spectrometry data.

Collectively, these data enabled us to probe genetic and epigenetic modulators of microenvironmental signalling, in a heterogeneous cohort that encompasses the clinical and molecular diversity of CLL. In the first part of this chapter, I apply a broad systematic approach to identify important genetic modulators, and in the second, I outline my follow-up investigations into the impact of trisomy 12 on stimulus response.

\hypertarget{systematic-analysis-of-the-effect-of-genetic-features-on-responses-to-stimuli}{%
\section{Systematic analysis of the effect of genetic features on responses to stimuli}\label{systematic-analysis-of-the-effect-of-genetic-features-on-responses-to-stimuli}}

\hypertarget{univariate-gene-stimulus-assosciations}{%
\subsection{Univariate analysis identifies IGHV status and trisomy 12 as key modulators of microenvironmental response}\label{univariate-gene-stimulus-assosciations}}

To begin, I ran a univariate analysis to compare viability values post-stimulation for patient samples with and without each genetic feature. In total 63 genetic features were surveyed, including IGHV status, somatic gene mutations and structural variants, where there were at least three patient samples in each group (Figure \ref{fig:stimuliGeneAssosciations}).


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/stimuliGeneAssosciations-1} 

}

\caption{Plot showing BH-adjusted p values from Student's t-tests (two-sided, with equal variance), for all tested gene-stimulus associations. Tests performed for IGHV status and somatic mutations and copy number aberrations with ≥3 patient samples in each group (n = 63 or is it 54??). Each circle represents a gene-stimulus association. Associations that meet 10\% FDR cut off are indicated in colour, where the colour denotes the genetic feature. \emph{Figure from Bruch \& Giles et al.~2021.} See Methods section \ref{univariate-gene-stimulus-associations-method}.}\label{fig:stimuliGeneAssosciations}
\end{figure}
This analysis revealed the extent to which genetic features modulate microenvironmental response: for ten out of 17 stimuli, at least one genetic feature determined response and for six out of 17 stimuli, two or more genetic features significantly altered the response (Student's t-tests, FDR = 10\%, Bruch \& Giles et al.~2021). The most common features were IGHV status and trisomy 12. Del(11q) also affected response to several stimuli. Notably, del(13q) and del(17p), which like trisomy 12 and del(11q) are the most common aberrations in CLL and act as prognostic markers (\protect\hyperlink{ref-Dohner2000}{H. Döhner et al. 2000}), had no impact on the responses to the panel of stimuli.

\hypertarget{multivariate-gene-stimulus-assosciations}{%
\subsection{Multivariate analysis of gene - stimulus assosciations confirms IGHV status and trisomy 12 as key modulators of stimulus response}\label{multivariate-gene-stimulus-assosciations}}

It was possible that there may be interplay between genetic factors in determining responses to external signals. To address this, I applied multivariate modelling to integrate the influence of genetic features, IGHV status and DNA methylation on the size of response (Bruch \& Giles et al.~2021). I used a Gaussian linear model with L1-penalty (i.e., lasso regression), to derive a predictor for each stimulus, comprised of these covariates. For background on this approach, see section \ref{intro-multivariate-modelling}.

As input to the model, the response matrix was composed of the log transformed viability values for each stimulus. To generate the feature matrix (137 samples versus 41 features), I excluded genetic features for which \textgreater20\% of the values were missing, and patient samples with incomplete annotation. As predictors, I included genetic mutations and CNVs (p= 39), IGHV status (coded as 0-1) and Methylation Cluster (coded as 0, 0.5, 1). I ran lasso regression, as implemented in the \texttt{R} package \texttt{glmnet}(\protect\hyperlink{ref-R-glmnet}{Friedman et al. 2021}), using three-fold cross-validation with misclassification error as loss. The resulting predictors are the mean of those coefficients that were selected in at least 75\% of 30 bootstrapped repeats.

Using the output of the regression, I generated predictor profiles for each stimulus. For 5 / 17 stimuli, there was at least one genetic predictor that met the cut-offs (a selection are shown in figure \ref{fig:stimuliGeneAssosciationsMulti}).


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/stimuliGeneAssosciationsMulti-1} 

}

\caption{Predictor profiles for IL4 and CpG ODN depicting gene - stimulus associations. Bar plots (left) show size and sign of assigned coefficients from Gaussian linear modelling. A positive coefficient indicates that stimulated increase in viability is larger when feature is present. Scatter plots (bottom) and corresponding heatmaps above show how presence of selected genetic feature relates to sample viabilities. Scatter plots show ranked log(viability) values for each sample and heatmaps show mutation status for each predictor, for corresponding sample in scatter plot. See method \ref{multivariate-gene-stimulus-assosciations-method}. \emph{Figure adapted from Bruch \& Giles et al.~2021}.}\label{fig:stimuliGeneAssosciationsMulti}
\end{figure}
The multivariate analysis demonstrated that responses to IL4 and sCD40L + IL4, TLR and TGF\(\beta\) were all affected by multiple genetic features (Bruch \& Giles et al.~2021). For example, higher viability in response to IL4 stimulation was associated with trisomy 12 and unmutated IGHV. In contrast, IL4 generated little or no increase in viability in samples with a mutation in KRAS, NRAS or BRAF. These tumours benefited less from the anti-apoptotic effects of IL4, indicating that the ability of IL4 to increase viability could be dependent on signalling via Ras-Raf-MEK-ERK, which is already active in mutant samples.

Discussion that Inhibition of ERK (p38) also increased IL4 protective effect

\hypertarget{response-to-tlr-stimulation-is-dependent-on-ighv-status-trisomy-12-and-mutations-in-dna-damage-response-genes}{%
\subsection{Response to TLR stimulation is dependent on IGHV status, trisomy 12 and mutations in DNA Damage Response genes}\label{response-to-tlr-stimulation-is-dependent-on-ighv-status-trisomy-12-and-mutations-in-dna-damage-response-genes}}

The multivariate analysis highlighted that TLR stimulation by CpG ODN and Resiquimod both showed the largest number of predictors, reflecting the multiple layers of biology involved here. These included del(11q) and \emph{ATM} , del(17p) and \emph{TP53}, IGHV status, trisomy 12 and \emph{SF3B1} (Bruch \& Giles et al.~2021). Amongst the screening cohort, TLR stimulation increased viability in certain samples and decreased in others. The clustering groups identified in section \ref{clustering} also showed highly divergent responses to TLR, and this related to disease progression, thus warranting a more detailed look at the underlying features modulating TLR response (Figure \ref{fig:TLRGenes}).


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/TLRGenes-1} 

}

\caption{Control-normalised log transformed viability values after treatment with Resiquimod (TLR 7/8), stratified by named genetic features.}\label{fig:TLRGenes}
\end{figure}
IGHV status (along with Methylation Cluster, which is related to IGHV) and trisomy 12 had the strongest effect on TLR response. \protect\hyperlink{ref-Chatzouli2014}{Chatzouli et al.} (\protect\hyperlink{ref-Chatzouli2014}{2014}) have previously shown that TLR responses is dependent on IGHV status (see also, section \ref{intro-ME-pathways}. We noted trisomy 12 as an additional, novel determinant of TLR response (Bruch \& Giles et al.~2021). In samples that do not have trisomy 12, TLR stimulation increases the viability of IGHV-U samples, whilst it decreases viability in IGHV-M samples, as expected. In contrast, in samples with trisomy 12, TLR stimulation increases viability regardless of IGHV status (Student's t-test, p\textless0.001 and p=0.018, Figure \ref{fig:TLRIHGVtri12}).

\protect\hyperlink{ref-Chatzouli2014}{Chatzouli et al.} (\protect\hyperlink{ref-Chatzouli2014}{2014})'s observations point to the existence of specific types of BCR/TLR collaboration in CLL, leading to activation of pro-survival pathways, or apoptosis depending on the IGHV status of the tumour. Our results suggest that synergy between BCR and TLR may also be dependent on trisomy 12 status.


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/TLRIHGVtri12-1} 

}

\caption{Beeswarm-boxplot showing control-normalised log transformed viability values, after treatment with Resiquimod, stratified by trisomy 12 and IGHV status. p-values from Student's t-tests. \emph{Figure from (Bruch \& Giles et al.~2021)}}\label{fig:TLRIHGVtri12}
\end{figure}
In addition to trisomy 12, I noted that TLR stimulation also increased viability in samples with mutations in the DNA damage response pathway, namely del(11q), del(17p), and \emph{ATM} and \emph{TP53}. Del(17p) occurs in 7\% of CLL cases and is associated with loss of TP53, a tumour suppressor gene involved in DNA damage and repair (\protect\hyperlink{ref-Zenz2010}{Zenz et al. 2010}). Del(11q) is more common (18\% of cases) and is associated with loss of ATM, another protein involved in DNA repair(\protect\hyperlink{ref-Kipps2017}{Thomas J. Kipps et al. 2017}). This suggests that there may also be cross-talk between the TLR and DNA damage response pathway, and that this affects the outcome of TLR stimulation.

The effect of TLR stimulation on CLL viability is dependent on the molecular make-up of the tumour. The nature of TLR response may also be important in prognosis, underlined by our the observation in section \ref{clustering} that a subgroup of patients (C3) shows slower disease progression, and is the only group in which TLR induces apoptosis. The role of TLR signalling in pathogenesis and prognosis may thus so far be under appreciated.

Collectively, this work represents the first large-scale attempt to profile the integrative effects of cell-extrinsic signals and cell-intrinsic features in lymphoma on a large scale. This systematic approach highlighted trisomy 12 as the most common feature to modulate responses. The rest of this chapter outlines my work to investigate the role of trisomy 12 in microenvironmental response.

\hypertarget{investigating-trisomy-12-as-a-modulator-of-microenvironmental-response}{%
\section{Investigating Trisomy 12 as a modulator of microenvironmental response}\label{investigating-trisomy-12-as-a-modulator-of-microenvironmental-response}}

\hypertarget{trisomy12-modulator}{%
\subsection{Trisomy 12 is a modulator of microenvironmental response}\label{trisomy12-modulator}}

Trisomy 12 modulated responses to IL4, TGF\(\beta\), soluble CD40L + IL4 and TLR stimuli (Bruch \& Giles et al.~2021, Figure \ref{fig:tri12cytResponse}). For example, the increase in viability induced by IL4 was enhanced in trisomy 12 samples, as was the decrease in viability in response to TGF\(\beta\) stimulation.

Trisomy 12 is commonly mutated in CLL (15\% of patients)(\protect\hyperlink{ref-Dohner2000}{H. Döhner et al. 2000}). Until recently, this genetic lesion was regarded as conferring intermediate risk, though novel therapies have improved outcomes for trisomy 12 patients (\protect\hyperlink{ref-Bosch2019}{Bosch and Dalla-Favera 2019}). However, the functional explanation for its recurrence is incompletely understood. Previous work has shown a role for gene dosage effects in the pathogenic mechanism: \protect\hyperlink{ref-Kienle2005}{Kienle et al.} (\protect\hyperlink{ref-Kienle2005}{2005}) show that overexpression of genes including CDK4 and E2F1 leads to increased cell cycling and higher proliferative capacity. Likewise, \protect\hyperlink{ref-HerbstThesis}{Herbst} (\protect\hyperlink{ref-HerbstThesis}{2020}) showed that BCR signalling proteins are also upregulated, which may also contribute to the increased proliferative capacity. Despite this, outcomes in trisomy 12 CLL are more favourable, and these cases show increased susceptibility to BCR inhibition (\protect\hyperlink{ref-JCIpaper}{Dietrich et al. 2017}) (see also, section \ref{intro-trisomy12}).

In the following, I outline my work to investigate the incompletely understood role of trisomy 12 in CLL.


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/tri12cytResponse-1} 

}

\caption{Control-normalised log transformed viability values after treatment with IL4, sCD40L + IL4 and TGF\(\beta\), stratified by trisomy 12.}\label{fig:tri12cytResponse}
\end{figure}
\hypertarget{gene-dosage-effects}{%
\subsection{STAT6, IRAK4 and SMAD3 are more highly expressed in trisomy 12 CLL}\label{gene-dosage-effects}}

Trisomy 12 samples contain a third copy of chromosome 12: transcriptomic and proteomic profiling of CLL samples with this lesion have demonstrated that this has a major impact on gene expression and protein abundances (\protect\hyperlink{ref-Abruzzo2018}{Abruzzo et al. 2018}; \protect\hyperlink{ref-HerbstThesis}{Herbst 2020}; \protect\hyperlink{ref-MeierAbt2021}{Meier-Abt et al. 2021}). It has also been shown that BCR signalling proteins are amongst those with increased expression in trisomy 12 (\protect\hyperlink{ref-HerbstThesis}{Herbst 2020}), and this is thought to contribute to increased susceptibility to BCR inhibition in trisomy 12 CLL (\protect\hyperlink{ref-JCIpaper}{Dietrich et al. 2017}).

Guided by this observation, I began by investigating RNA and protein expression levels amongst genes involved in the TGF\(\beta\), IL4 and TLR pathways, to which trisomy 12 samples respond more strongly. The aim was to determine whether proteins in these pathways are more abundant in trisomy 12 CLL, thus contributing to the enhanced response.

I ran differential expression analysis to compare CLL samples from the screen with and without trisomy 12. Next, I filtered the differentially expressed genes (adjusted p \textless{} 0.1) for those belonging to the TGF\(\beta\), JAK-STAT and TLR pathways genesets, from the KEGG database. I visualised the RNA counts and protein abundances (\protect\hyperlink{ref-HerbstThesis}{Herbst 2020}) for those genes.

Only a small proportion of the differentially expressed genes belonged to the TGF\(\beta\), JAK-STAT and TLR pathways genesets. However of those that were, several genes were key downstream mediators of these pathways (Figure \ref{fig:geneDosage}. Amongst TGF\(\beta\) signalling genes, 7 / 95 were upregulated in trisomy 12, including SMAD3. 12 / 160 IL4 signalling genes were differentially expressed, including STAT2. Only 2 / 116 genes in the TLR geneset were differentially expressed, though this included IRAK4 on chromosome 12. Amongst these key mediators SMAD3, STAT2, and IRAK4, all showed higher protein abundance in addition to increased RNA expression. Notably, STAT6, the key downstream mediator of IL4 signalling, was not differentially expressed, but also showed differential protein abundance. Interestingly, many of the over-expressed genes are not located on chromosome 12, indicating the extent to which the differential dosage of this chromosome has on the expression of the entire genome.


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/geneDosage-1} 

}

\caption{Beeswarm-boxplots showing RNA counts and protein abundances stratified by trisomy 12 status. P values from Student's t-test. \_Proteomics dataset from (\protect\hyperlink{ref-HerbstThesis_}{\textbf{HerbstThesis\_?}}) Method: \ref{gene-dosage-effects-method}.}\label{fig:geneDosage}
\end{figure}
Next I looked for further evidence to directly link the higher gene dosage of cytokine signalling genes, to enhanced responses to cytokine signals.

\hypertarget{trisomy12-classifier}{%
\subsection{Classification analysis identifies trisomy 12 phenocopies that show increased expression of IRAK4 and SMAD3}\label{trisomy12-classifier}}

I next investigated whether any non-trisomy 12 samples might display a trisomy 12-like phenotype (referred to as a phenocopy), in that they respond in a similar way to the panel of stimuli. The aim was to identify trisomy 12 phenocopies and to isolate the feature of these samples that might explain the underlying cause of enhanced response to external signals in trisomy 12 CLL. In particular, I was interested to see if these phenocopies showed higher expression, or even gene amplification, of the signalling genes identified in section \ref{gene-dosage-effects}.

To identify trisomy 12 phenocopies, I began by generating a classifier to predict the trisomy 12 status of a sample from its stimulus response matrix. I aimed to find non-trisomy 12 samples that were consistently misclassified as trisomy 12.

The classifier was built using binomial regression, with lasso penalisation, as implemented in the \texttt{R} package \texttt{glmnet} (\protect\hyperlink{ref-R-glmnet}{Friedman et al. 2021}). The feature matrix consisted of z scores of the viability values after treatment with each stimulus, and was used to predict the response (trisomy 12 status). I ran the model for 50 bootstrapped repeats, using three-fold cross-validation and mean absolute error as loss. Resiquimod, sCD40L+IL4 and TGF\(\beta\) were selected as coefficients that predict trisomy 12 status (Figure \ref{fig:tri12Classfier}, as would be expected bed on the observations in section \ref{trisomy12-modulator}.


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/tri12Classifier-1} 

}

\caption{Predictor profile showing coefficients selected as predictors of trisomy 12 status. Here, binomial modelling with L1-penalty was used to identify associations between responses to stimuli and trisomy 12 status. Bar plots (left) show size and sign of assigned coefficients named on the right. A positive coefficient indicates that higher viability upon treatment named stimulus is associated with trisomy 12. Facet labels (top) and corresponding heatmap below show how the viability with each named stimulus relates to trisomy 12 status. The model shown was selected from 50 bootstrapped repeats, based on maximal AUC. Method \ref{trisomy12-classifier-method}.}\label{fig:tri12Classifier}
\end{figure}
I then predicted trisomy 12 status with the same viability matrix, using each of the 50 bootstrapped. I compared the results of the classification with the true trisomy 12 status. Two patient samples were misclassified as trisomy 12 in more than 50\% of repeats; I refer to these as patient sample A and B in the below. Patient sample A showed the lowest viability with TGF\(\beta\) (Figure \ref{fig:tri12PhenocopiesCytResponse}, and Patient sample B showed the highest viability with IL4, and the second highest with Resiquimod.


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/tri12PhenocopiesCytResponse-1} 

}

\caption{Control-normalised log-transformed viability values, for all samples after treatment with Resiquimod, sCD40L + IL4 and TGF\(\beta\). Patient sample A and B are indicated in blue and red, respectively.}\label{fig:tri12PhenocopiesCytResponse}
\end{figure}
I next investigated whether either of these non-trisomy 12 samples contained any regional amplifications on chromosome 12, which may help to isolate the gene(s) causing the tumour to respond more strongly to these signals. The whole exome sequencing for both patient samples indicated that Patient sample A had several amplified regions at 12p13.31 (42 copies), 12q24.13 (10 copies) and 12q24.33 (21 copies). An examination of the genes in these regions indicated that none of the signalling genes identified in \ref{gene-dosage-effects} could be found in these regions, and no clear candidate gene(s) emerged that may be involved in responses to external signals.

In the absence of gene amplification, I next checked patient samples A and B might show higher expression of the signalling genes identified in \ref{gene-dosage-effects} by visualising the RNA expression levels for SMAD3, IRAK4 and STAT6 for these patients (Figure \ref{fig:tri12PhenocopiesRNA}. Patient sample A, which responded most strongly to Resiquimod, showed high levels of IRAK4 compared to the other non-triomsy 12 samples. Patient sample B, which responded strongly to TGF\(\beta\) showed the highest level of SMAD3 expression amongst the non-trisomy 12 samples. Thus, whilst neither patient sample contained an amplicon of SMAD3 or IRAK4, its possible that increased expression of these proteins enables a stronger response to the corresponding pathway.


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/tri12PhenocopiesRNA-1} 

}

\caption{Raw RNA counts for SMAD3, IRAK4 and STAT6, stratified by trisomy 12 status. Patient sample A and B are indicated in blue and red, respectively. P values from Student's t-test.}\label{fig:tri12PhenocopiesRNA}
\end{figure}
These results collectively suggest that increased gene dosage of key genes in the IL4, TLR and TGF\(\beta\) pathways may underlie the increased response of trisomy 12 samples to these pathways. This is also reflected in the patient samples A and B, which responded more strongly to the TGF\(\beta\) and TLR pathways, respectively, and correspondingly higher levels of SMAD3 and STAT6 proteins. However, computational techniques can only go to far in providing biological proof, and more work is needed here to confirm this finding. In addition to gene dosage effects, I also investigated differences in transcription factor activity in trisomy 12. The results of this warranted further follow up, and this forms the focus of rest of this chapter.

\hypertarget{spi-b-and-pu.1-tfs-show-higher-activity-in-trisomy-12-cll}{%
\subsection{Spi-B and PU.1 TFs show higher activity in trisomy 12 CLL}\label{spi-b-and-pu.1-tfs-show-higher-activity-in-trisomy-12-cll}}

Trisomy 12 has been well-studied at both the transcriptomic and proteomic level, and yet the cause of its recurrence in CLL is not fully understood. Thus, I next decided to investigate the impact of trisomy 12 on the epigenetic landscape of CLL, which is less well studied. In particular, I wanted to investigate differential transcription factor activity in CLL, which would give an indicator of which pathways are differentially active.

We acquired two independent ATAC sequencing datasets. The first, generated and processed in our lab, consisted of two WT and two trisomy 12 samples (Bruch \& Giles et al.~2021). The second, taken from (\protect\hyperlink{ref-Rendeiro2017}{\textbf{Rendeiro2017?}}) and processed in our lab, comprised 43 WT and nine trisomy 12 samples.

In the external dataset, trisomy 12 status was not annotated. To do this, we used the ATACseq reads to call trisomy 12 in samples that contained \textgreater{} 1.4 times more reads per peak (i.e.~genomic region) on average in chromosome 12, compared to peaks on other chromosomes.

Next, we used the \texttt{R} package \texttt{diffTF} (\protect\hyperlink{ref-R-diffTF}{\textbf{R-diffTF?}}) to identify TFs that showed differential binding site accessibility between the WT and trisomy 12 samples (Figure \ref{fig:diffTFexplainer}. \texttt{diffTF} enables comparison of TF activity between two conditions, using chromatin accessibility data, see section \ref{intro-ATACseq}. We provided a list of 636 TFs, from the HOCOMOCO v10 database (\protect\hyperlink{ref-HOCOMOCO}{Kulakovskiy et al. 2016}) to measure their change in actiitvy between trisomy 12 and WT samples (Bruch \& Giles et al.~2021).


\begin{figure}

{\centering \includegraphics[width=0.5\linewidth]{figures/diffTF_method} 

}

\caption{diffTF workflow from \protect\hyperlink{ref-Berest2019}{Berest et al.} (\protect\hyperlink{ref-Berest2019}{2019}). diffTF accepts a list of TFs along with the genomic locations of their bindings sites. For each TF, the software computes the distribution of fold changes between the trisomy 12 and WT samples, using the ATACseq peaks at each TF binding site in each condition. The software compares this distribution to a background set of fold changes generated using GC content-matched loci that do not contain the same TF binding site motif. Each TF is thus assigned assigned a weighted mean difference value, which quantifies the change in activity, and a p value.}\label{fig:diffTFexplainer}
\end{figure}
Running this analysis on both ATACseq datasets indicated that the binding sites of nine TFs were more accessible (p\textless0.05) in the trisomy 12 samples of the larger, external dataset (Figure \ref{fig:tri12diffTF}. In the smaller in-house dataset, the binding sites of 92 TFs were likewise more accessible (Supp. Figure \ref{fig:diffTFsmallvolPlot}). RNA and proteomics data show different abundances of transcripts and proteins in trisomy 12 CLL: ATACseq data here shows that this also corresponds to a specific signalling signature in trisomy 12 CLL.

We confirmed that the results were not affected by the additional copy of chromosome 12. Rerunning diffTF analysis without the ATACseq reads from chromosome 12 had minor impact on the significant TFs.

In both datasets, the TFs with the largest increase in activity in trisomy 12 were Spi-B and/or PU.1 (Bruch \& Giles et al.~2021). Both TFs share similar binding motifs and exhibit functional redundancy (\protect\hyperlink{ref-GarrettSinha2001}{\textbf{GarrettSinha2001?}}), which make it difficult to distinguish from ATACseq data alone whether either or both are more active.

Spi-B and PU.1 are haematopoetic regulators that are known to be key regulators of healthy B-cell function (\protect\hyperlink{ref-Turkistany2011}{Turkistany and Dekoter 2011}){[}READ MORE{]}, controlling B-cell responses to environmental cues including CD40L, TLR ligands and IL4 (\protect\hyperlink{ref-Willis2017}{Willis et al. 2017}).


\begin{figure}

{\centering \includegraphics[width=0.8\linewidth]{thesis_files/figure-latex/tri12diffTF-1} 

}

\caption{Bar plot showing the results of the diffTF analysis for the \protect\hyperlink{ref-Rendeiro2016}{Rendeiro et al.} (\protect\hyperlink{ref-Rendeiro2016}{2016}) dataset. y axis shows change in TF activity (weighted mean difference) between trisomy 12 (n = 9) and non-trisomy 12 samples (n = 43),x axis indicates TF names. 17 / 636 TFs, with BH adjusted p \textless0.05 are shown. p values generated by diffTF in permutation mode. TF binding sites defined in HOCOMOCO v10 (\protect\hyperlink{ref-HOCOMOCO}{Kulakovskiy et al. 2016}).}\label{fig:tri12diffTF}
\end{figure}
Spi-B and PU.1 appeared to be upregulated in trisomy 12 CLL, and evidence in the literature indicated that these may regulate environmental sensing genes, providing a link between trisomy 12, and enhanced responses to external signals.

To provide further evidence of this, I next aimed to profile the downstream effects of Spi-B and PU.1 in lymphoma.

\hypertarget{spi-b-and-pu.1-targets-are-enriched-for-immune-signalling-pathways}{%
\subsection{Spi-B and PU.1 targets are enriched for immune signalling pathways}\label{spi-b-and-pu.1-targets-are-enriched-for-immune-signalling-pathways}}

I hypothesised that Spi-B and PU.1 might coordinate transcriptional response to external signals, thus modulating CLL proliferation in response to the microenvironment. To identify Spi-B and PU.1 target genes specifically in lymphoma, I acquired a ChIPseq dataset (\protect\hyperlink{ref-Care2014}{Care et al. 2014}) containing data on Spi-B and PU.1 binding in lymphoma cell lines. I used this dataset to test for functional enrichment of immune signalling pathways amongst the TF targets.

To define TF targets, I took the closest gene to each significant ChIP peak (q value\textless0.05), and within \(\pm\)kb of TSS. I then tested for over-representation of these TF targets amongst selected KEGG(\protect\hyperlink{ref-KEGG}{\textbf{KEGG?}}) and Reactome (\protect\hyperlink{ref-Reactome}{\textbf{Reactome?}}) genesets, using the `R' package \texttt{clusterProfiler} (\protect\hyperlink{ref-R-clusterProfiler}{G. Yu 2021b}). This method corresponded to a one-sided version of Fisher's exact test. This analysis showed TLR, BCR and TGF\(\beta\) signalling genes to be enriched (p\textless0.01) amongst Spi-B and PU.1 targets (Figure \ref{fig:SpiBChipSeq}).


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/SpiBChipSeq-1} 

}

\caption{Table shows results from over-representation tests of selected KEGG and Reactome pathways amongst Spi-B and PU.1 targets. Columns show geneset pathways, corresponding database, the number of genes within geneset, the number of Spi-B and PU.1 target genes within geneset (total number of target genes defined also shown), and p-value from over-representation test. TF targets defined as closest gene to each significant ChIP peak (q value\textless0.05), and within \$\pm\$1kb of TSS. ChIPseq data from \protect\hyperlink{ref-Care2014}{Care et al.} (\protect\hyperlink{ref-Care2014}{2014}). Over-representation tests run using (\protect\hyperlink{ref-clusterProfiler}{\textbf{clusterProfiler?}}) package, method corresponds to one-sided version of Fisher's exact test. \emph{Figure and caption from Bruch \& Giles et al 2021.}}\label{fig:SpiBChipSeq}
\end{figure}
\hypertarget{double-knockdown-of-spi-b-and-pu.1-reduces-profileration-of-trisomy-12-cell-lines}{%
\subsection{Double knockdown of Spi-B and PU.1 reduces profileration of trisomy 12+ cell lines}\label{double-knockdown-of-spi-b-and-pu.1-reduces-profileration-of-trisomy-12-cell-lines}}

To establish the functional impact of Spi-B and PU.1 inhibition in trisomy 12+ lymphoma, we tested th impact of inhibiting these TFs on proliferation of lymphoma cell lines (Figure \ref{fig:SpiBshRNAKD}. We generated single and double shRNA knockdowns in lymphoma cell lines, namely SU-DHL4 and SU-DHL5 (trisomy 12) and SU-DHL2 (no trisomy 12) and then measured cell counts at 24 hour intervals (Bruch \& Giles et al.~2021).


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/SpiBshRNAKD-1} 

}

\caption{Knockdown of Spi-B and PU.1 led to growth restriction and cell death. Data shown is the mean of three technical replicates. While SU-DHL 5 was already growth impaired after Spi-B KD, SU-DHL 2 and SU-DHL 4 were only impaired after double knock-down. Double knockdown lead to rapid cell death in SU-DHL 5 (data not shown). Cell count normalised to seeded cell number of diffuse large b cell lymphoma cell lines after knockdown of Spi-B or double knockdown of Spi-B and PU.1 or shRNA as control. \emph{Figure and caption from Bruch \& Giles et al.~2021.}}\label{fig:SpiBshRNAKD}
\end{figure}
The single knockdowns had a small impact on proliferation: Spi-B inhibition reduced proliferation in SU-DHL5, as did PU.1 to a lesser extent. Double knockdown of both TFs markedly reduced proliferation in SU-DHL2 and SU-DHL4, and was lethal in SU-DHL5. This result suggested that both these TFs play an important role in the proliferative capacity of the tumour cells, and that there is functional redundancy between Spi-B and PU.1 in this context.

Collectively, these results demonstrate that trisomy 12 modulates responses to microenvironmental signals. Trisomy 12 appears to increase Spi-B and PU.1 activity, which regulate genes relating to environmental sensing, and reduce proliferation of cell lines when inhibited.

\hypertarget{summary-1}{%
\section{Summary}\label{summary-1}}

In this chapter, we perform a systematic survey of genetic determinants of microenvironmental response, leading to two key findings. Firstly, TLR signalling has a range of effects on CLL viability, and this is determined by many genetic features, including IGHV status, trisomy 12 and mutation in the DNA Damage Response pathway. Secondly, we identify trisomy 12 as a modulator of microenvironmental response, and show that higher activity of Spi-B and PU.1 may mediate this effect.

\hypertarget{discussion}{%
\section{Discussion}\label{discussion}}

Some do affect, some dont at all
Notably, del(13q) and del(17p), which like trisomy 12 and del(11q) are the most common aberrations in CLL and act as prognostic markers (\protect\hyperlink{ref-Dohner2000}{H. Döhner et al. 2000}), had no impact on the responses to the panel of stimuli.
ie some are really affecting signalling, others are not

TLR and BCR
\protect\hyperlink{ref-Chatzouli2014}{Chatzouli et al.} (\protect\hyperlink{ref-Chatzouli2014}{2014}) show that stimulation of TLR in IGHV-U samples leads to pronounced p-ERK induction, whereas in M-CLL, stimulation of TLR concomitantly with BCR stimulation is required to induce a smaller increase in p-ERK. TLR stimulation without BCR stimulation in IGHV-M samples leads to an induction of caspase-8 and apoptosis, much more so than in IGHV-U samples(\protect\hyperlink{ref-Chatzouli2014}{Chatzouli et al. 2014}).

TLR as a prognostic tool?\textless- does this integrate multiple risk markers?
Reelvance in lymhp node??

Papeer whosing TLR response and survival that i found ages ago

REad up on trisomy 12
\protect\hyperlink{ref-JCIpaper}{Dietrich et al.} (\protect\hyperlink{ref-JCIpaper}{2017}).

Whats knwon about pathogenetic mechanims:
DOI: 10.1200/JCO.2005.02.568 Journal of Clinical Oncology 23, no. 16 (June 01, 2005) 3780-3792.

whole chromososm - is it lots of things - IRAK4, STAT6, gen dosge, SPIB and PU1 activity, BCR signalling?

gene dosage effects are obviosuly quite important here

the power of bioiformatics techqnieus , as well as traditioanl techinuqes, in bioogical discover, to save time and resouce

Read up on SpiB and PU1
SPib an dpU 1 regulate BTK: \url{https://pubmed.ncbi.nlm.nih.gov/8934542/}
SPIB and PU1 activity in B cells: \url{https://pubmed.ncbi.nlm.nih.gov/15936902/}
(hese data indicate that Spi-1 and Spi-B directly regulate the expression of Grap2 and that Grap2 functions to modulate BCR signaling, but that reduced Grap2 expression is unlikely to account for the BCR signaling defects observed in Spi1+/- SpiB-/- B cells.)
PU1 and SPIb and higher in Bc cells- this paper indicates this is to do with SP1 binding (which is on chromosome 12 and is hgiher ) \url{https://pubmed.ncbi.nlm.nih.gov/7566969/}
PU1 and PSIb bdinign in mouse lympohma cell ine \url{https://pubmed.ncbi.nlm.nih.gov/25765478/}
Spib and PU1 in B and T cell development \url{https://pubmed.ncbi.nlm.nih.gov/8691135/}

TLR signalling:
REad (\protect\hyperlink{ref-Chatzouli2014}{Chatzouli et al. 2014})
TLR stimulation can lead to both acitvation of MAPK and increase in viability or activation of apoptosis. If mutatiosn can affect which pathway is active, this poitns to aditoanl mechsnsim by which these mutations contribtue to CLL drive. It also points to TLR response as a useful prognostic marker.

POwer of multivariate modellings;
we need mor ltieracy about these things
discuovery of mulitple layers of biology rather thatn jsut using t tests

IL4 and KRAS - IL4 and ERK apthway reading

BCR singalling - which is important iwtih IHGV and tirsomy 12, is evleated in teh LN accoridng to gene sigantures:
\url{https://ashpublications.org/blood/article/117/2/563/28102/The-lymph-node-microenvironment-promotes-B-cell}

\hypertarget{contributions-statement}{%
\section{Contributions statement}\label{contributions-statement}}

The author of this thesis performed statistical inference and regression modelling outlined in this chapter. In essence, she performed the univariate and multivariate survey of genetic determinants of stimulus response. She identified trisomy 12 as a modulator of microenvironmental response, and conceptualised and performed the computational follow-up work on this finding, with support from Peter-Martin Bruch, Ivan Berest and Tina Becirovic.

This chapter integrates a number of datasets from the following sources. The screening data was generated by Peter-Martin Bruch. The associated patient genetic data, including WES, WGS, CNV profiles, DNA methylation data and RNA was generated and processed by \protect\hyperlink{ref-JCIpaper}{Dietrich et al.} (\protect\hyperlink{ref-JCIpaper}{2017}). The proteomics dataset was generated by \protect\hyperlink{ref-HerbstThesis}{Herbst} (\protect\hyperlink{ref-HerbstThesis}{2020}), and kindly shared for the purpose of this thesis. The larger ATACseq dataset was generated by (\protect\hyperlink{ref-Rendeiro2017}{\textbf{Rendeiro2017?}}), and processed by Ivan Berest. The smaller ATACseq dataset was planned by the author and Peter-Martin Bruch, and generated by Peter-Martin Bruch. The data was processed by the author, with support from Ivan Berest. The Spi-B and PU.1 knockdown data was generated by Tina Bericoviv and published in Bruch \& Giles et al.~2021.

In addition, the author received support on the following analysis work. Junyan Lu provided advice and support with multivariate modelling approaches. Ivan Berest provided support with running diffTF.

used I versus we and credited datasets, and my paper for figures and factrs - main analysis nad key finding, this hsould be routine
Then outline in detail at the bottom

\hypertarget{chapter6}{%
\chapter{Molecular and microenvironmental modulators of drug response in CLL}\label{chapter6}}

Chapters \ref{chapter4} and \ref{chapter5} provide a systematic exploration of the effect of microenvironmental stimulation on \emph{ex vivo} CLL biology, and how these pathways can be modulated by genetic features. Collectively, these results underline the fact that cancer biology is determined by an integrated network of cell-intrinsic molecular aberrations and cell-intrinsic signals generated by cell-cell contacts and soluble factors. The comprehensive nature of this dataset enables the study of these multiple axis.

In this chapter, we examine the third axis of the dataset: drug response. As well as providing protective signals that increase CLL viability and support tumour growth, the microenvironment is also believed to interfere with drug action. Several studies have demonstrated the impact of individual microenvironmental signals on drug response \emph{ex vivo}, and is also thought to occur within the protective niche (\protect\hyperlink{ref-Ahn2018}{Ahn et al. 2018}). Guided by this observation, we next wanted to investigate the impact of each of the stimuli on responses to the panel of drugs, and how this can be further modulated by genetic features.

\^{}sharpen this once more

In section \ref{mapping-interactions}, the impact of the stimuli on drug response is quantified by identifying drug - stimulus interactions. In section \ref{mapping-genetic-modulators}, the impact of gneetic features on drug response is explored. Finally, in section \ref{drug-stimulus-gene-interactions}, multivariate modelling is apploed to integrate all three axes of the dataset, to investigate how drug - microenvironment interactions can be further modulated by genetic features. These analyses benefit from linear regression and generalised linear modelling approaches, described in more detail in section \ref{intro-multivariate-modelling}.

\hypertarget{mapping-interactions}{%
\section{Mapping drug - stimulus interactions}\label{mapping-interactions}}

\hypertarget{drug-stimulus-linear-model}{%
\subsection{Linear modelling quantifies interactions between drugs and stimuli}\label{drug-stimulus-linear-model}}

To begin, I evaluated computational methods for quantifying the impact of the stimuli on drug action. The aim was to screen for cases where the stimulus pathway specifically interacted with the target pathway of the drug.

When a stimulus is individually applied to a CLL sample, the stimulus activates signalling cascades that modulate CLl viability and impact the rate of spontaneous apoptosis. When a stimulus is co-applied with a drug, the stimulus will continue to impact upon baseline viability, at the same time as the drug inhibits baseline viability. Assuming the drug and stimulus do not interact, the viability of the cells with the combinatorial treatment will equate to the additive impact of both these compounds. In the case that there is some interaction between the stimulus pathway and drug target pathway, the resulting viability will be more complex. The difference between the additive, or expected viability, and the true measured viability, can be quantified by an interaction factor.

I fitted a linear model (Equation \eqref{eq:drugCytInt}) to quantify how each stimulus modulates drug efficacy beyond its individual effect (\protect\hyperlink{ref-Giles2021}{Bruch et al. 2021}). I used the \texttt{lm} function implemented in the \texttt{R} package \texttt{stats}(\protect\hyperlink{ref-R-base}{R Core Team 2021}). The model accounts for the individual effects of the drug and stimulus treatment on spontaneous apoptosis and baseline viability, in addition to an interaction term, named \(\beta{int}\). \(\beta{int}\), the interaction factor, quantifies how the combined treatment effect differs from the sum of the individual treatments.

Equation \eqref{eq:drugCytInt} quantifies how the viability with any combination can be predicted:
\begin{equation}
            log(V) = \beta_{drug}X_{drug} + \beta_{stimulus}X_{stimulus} + \beta_{int}X_{drug}X_{stimulus} + \epsilon
                                       \label{eq:drugCytInt}
    \end{equation}
\emph{where \(V\) is the predicted viability with a given treatment,} \(\beta_{drug}\), \(\beta_{stimulus}\) \emph{and} \(\beta_{int}\) \emph{are coefficients for the drug, stimulus and combinatorial terms and} \(X_{drug}\) \emph{and} \(X_{stimulus}\) \emph{are indicator variables (0 or 1) for the presence or absence of a drug/stimulus.} \(\epsilon\) \_is the vector of model residuals. See also Methods section \ref{drug-stimulus-linear-model-method}. Equation from (\protect\hyperlink{ref-Giles2021._}{\textbf{Giles2021.\_?}})

This approach identified 45 drug-stimulus combinations (out of 204), where \(\beta{int}\) had p\textless0.05, i.e.~the stimulus had a specific effect on the efficacy of the drug, highlighting the extent to which drug action can be modulated by cell-external signals \emph{ex vivo}.

These interactions are of potential clinical interest, to support our understanding of \emph{in vivo} drug response, and the development of strategies to target drug resistance pathways. They are also relevant to laboratory studies of drug action, which are very often performed in the absence of such signals.

To establish which interactions may be important in these contexts, we began by classifying the drug - stimulus interactions into four categories (\protect\hyperlink{ref-Giles2021}{Bruch et al. 2021}). The categories were firstly defined based on whether the interaction was antagonistic or synergistic, i.e.~the stimulus acted to oppose or reinforce the activity of the drug. These were further divided based on the sign of \(\beta{int}\) i.e.~whether the combinatorial viability was higher (postive) or lower (negative) than would be expected based on additive effects (Figure \ref{fig:drugStimulusDummyPLots}A).


\begin{figure}[H]

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/drugStimulusDummyPLots-1} 

}

\caption{(A) Graphical line plots representing typical response patterns for each of the four drug - stimulus interaction categories (types I - IV). x-axis shows treatment, y-axis shows log transformed viability values, facet labels indicate interaction type. Blue and black horizontal lines demonstrate effect of interaction on viability with combinatorial treatment. Blue line shows predicted viability based on additive effects alone and black line shows viability accounting for additive effects and interaction. (B) Histogram showing number of drug-stimulus interactions within each category, where p value for \(\beta_{int}\) is \textless0.05. See Methods section \ref{drug-stimulus-linear-model-method}. \_Figure and caption from (\protect\hyperlink{ref-Giles2021._}{\textbf{Giles2021.\_?}})}\label{fig:drugStimulusDummyPLots}
\end{figure}
These categories encompassed the following interaction types. Type I defined positive antagonisms in which stimuli reversed drug action, leading to decreased drug efficacy and increased CLL viability. This group highlights drug - stimulus interactions that could be relevant to treatment resistance pathways \emph{in vivo}. Type II negative antagonisms highlighted cases where the drug inhibited the stimulus, reducing the effect of the stimulus on baseline viability, and decreasing CLL viability. These interactions include drugs that provide potential therapies to target treatment - resistance pathways \emph{in vivo}. Type III and type IV interactions cover synergies, in which the drug and stimulus increase the effect of the other. Type III positive synergies lead to increased CLL viability with the combinatorial treatment, whilst type IV negative synergies lead to increased drug efficacy and lower viability. In the latter case, these may highlight cases where drug efficacy is dependent on external pathways, and may be useful for understanding \emph{ex vivo} drug action. Figure \ref{fig:drugStimulusDummyPLots}B quantifies the number of interactions in each category.

\hypertarget{drug-stimulus-interaction-examples}{%
\subsection{\texorpdfstring{INF\(\gamma\) induces resistance to ibrutinib \emph{in vitro}}{INF\textbackslash gamma induces resistance to ibrutinib in vitro}}\label{drug-stimulus-interaction-examples}}

Using the linear modelling approach in combination with our classification system, we attempted to generate the first comprehensive map of drug - stimulus interactions in CLL (\protect\hyperlink{ref-Giles2021}{Bruch et al. 2021}). This map provides a resource to explore the impact of the microenvironment on the majority of drugs that are clinically-licensed in CLL (Figure \ref{fig:drugStimulusIntMapt}). We use this resource to investigate a number of drug - stimulus interactions here, and in section \ref{chapter7}.


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/drugStimulusIntMapt-1} 

}

\caption{Heatmap of all \(\beta_{int}\) values for all drug - stimulus combinations where p for \(\beta_{int}\) \textless0.05, annotated with interaction type. Scale indicates size and sign of \(\beta_{int}\). Rows and columns clustered according to hierarchical clustering. See Methods section \ref{drug-stimulus-linear-model-method}. \_Figure and caption from (\protect\hyperlink{ref-Giles2021._}{\textbf{Giles2021.\_?}})}\label{fig:drugStimulusIntMapt}
\end{figure}
To complement this broader approach, I next examined each interaction individually, to select the most clinically interesting examples (Figure \ref{fig:drugStimulusIntEx}).

Positive antagonistic interactions were the most common, in which stimuli reversed drug action. This group included resistance mechanisms such as the inactivation of ibrutinib by IL4 stimulation (Figure \ref{fig:drugStimulusIntEx}A)(\protect\hyperlink{ref-Aguilar-Hernandez2016}{\textbf{Aguilar-Hernandez2016?}}), indicating that the modelling and classification approach successfully recapitulated known drug - stimulus interactions. Our approach demonstrated that IL4 induced resistance to a range of drugs, beyond ibrutinib, and showed the highest number of positive antagonistic interactions of all of the stimuli.

Amongst the positive antagonistic interactions, we noted reduced efficacy of ibrutinib in the context of IFN\(\gamma\) stimulation (Figure \ref{fig:drugStimulusIntEx}B, (\protect\hyperlink{ref-Giles2021}{Bruch et al. 2021})). {[}ADD FOLLOW UP / STAT1 experiment{]}.

There were six cases of negative antagonisms, whereby drug action inhibited the pro-survival effect of the stimulus. As IL4 appeared to be the most potent conferrer of drug resistance, it was of greatest interest to identify drugs that may inhibit this pathway. For example, the Pan-JAK inhibitor pyridone-6 inhibited the increase in viability with sCD40L + IL4 stimulation (Figure \ref{fig:drugStimulusIntEx}C, (\protect\hyperlink{ref-Giles2021}{Bruch et al. 2021})).

The model identified a single positive synergism, in which IFN\(\gamma\) treatment in combination with ralimetinib, a p38 MAPK inhibitor, stimulated a large increase in viability that was not observed in each of the single treatments (Figure \ref{fig:drugStimulusIntEx}D, (\protect\hyperlink{ref-Giles2021}{Bruch et al. 2021})). This finding suggests a potential inhibitory effect of p38 MAPK activity on signalling via IFN\(\gamma\). {[}ADD TO THIS NOS THEORY{]}

16 drug - stimulus combinations demonstrated negative synergistic interactions. This implied that the drug efficacy was somehow dependent upon, or increased by, activation of the stimulus pathway. For example, luminespib, a HSP90 inhibitor, showed higher efficacy with a number of stimuli, including soluble anti-IgM, indicating that \emph{ex vivo} measurements of luminespib action are likely to be affected by the presence or absence of microenvironmental signals (Figure \ref{fig:drugStimulusIntEx}F, (\protect\hyperlink{ref-Giles2021}{Bruch et al. 2021})).

Collectively, these results represent a comprehensive and systematic study of the influence of the microenvironment on drug efficacy in CLL. This work highlights key resistance pathways, strategies for targeting microenvironmental resistance, and underlines important pathways in \emph{ex vivo} studies of drug efficacy. In this thesis, I have selected some of the more interesting examples, such as resistance to Ibrutinib induced by IL4 and IFN\(\gamma\), to investigate in more detail here and in Chapter \ref{chapter7}. This work also aims to serve as a resource for others, to provide a basis for further follow-up studies beyond the ones in this thesis. It also aims to provide a proof-of-principle highlighting the value of combinatorial screening, and linear modelling approaches, to study the impact of the microenvironment in cancers in general. All drug - stimulus combinations and interactions can be explored on the \href{https://www.imbi.uni-heidelberg.de/dietrichlab/CLL_Microenvironment/}{online shiny app}, and the whole analysis can be replicated, or adapted, from the \href{https://github.com/Huber-group-EMBL/CLLCytokineScreen2021}{online code repository}, published alongside (\protect\hyperlink{ref-Giles2021}{Bruch et al. 2021}).


\begin{figure}[H]

{\centering \includegraphics[width=0.8\linewidth]{thesis_files/figure-latex/drugStimulusIntEx-1} 

}

\caption{Line plots showing examples of drug-stimulus interactions. x axis indicates treatment, y axis shows log transformed viability values, with matching samples linked across treatments. Black horizontal lines represent predicted viability for each treatment, using coefficients from linear model fit. For combinatorial treatment, blue line indicates predicted viability based on additive effect alone, and black line indicates predicted viability accounting for additive effects plus interaction. See Methods section \ref{drug-stimulus-linear-model-method}. \emph{Figure and caption from (\protect\hyperlink{ref-Giles2021}{Bruch et al. 2021}).}}\label{fig:drugStimulusIntEx}
\end{figure}
\hypertarget{mapping-genetic-modulators}{%
\section{Genetic modulators of drug response}\label{mapping-genetic-modulators}}

\hypertarget{univariate-gene-drug-associations}{%
\subsection{Univariate analysis identifies genetic modulators of drug response}\label{univariate-gene-drug-associations}}

Most mutations in cancer have not been linked to drug response (\protect\hyperlink{ref-JCIpaper}{Dietrich et al. 2017}). CLL has provided an powerful model system in which to study the link between recurrent genetic features and drug response: liquid biopsies are easy to obtain and sequence, and screen \emph{ex vivo} and it is much easier than in many cancers to take samples over a time course. Previous work in our lab (\protect\hyperlink{ref-JCIpaper}{Dietrich et al. 2017}) has taken advantage of this, generating the largest survey of molecular determinants of response in CLL to date.

Before investigating the impact of mutations on drug - stimulus interactions, I performed a similar analysis of genetic determinants of drug response, to first establish the effects of mutations independently of stimuli on drug response. I used the genetic data, including all mutations, copy number variants and IGHV status for which there were greater than 3 cases in the cohort (n = 54), to perform t-tests to identify molecular features that impact on drug response (Figure \ref{fig:drugGeneIntMap}).

This analysis recapitulated the findings of previous work (\protect\hyperlink{ref-JCIpaper}{Dietrich et al. 2017}), including the impact of IGHV status on responses to BCR inhibitors ibrutinib (BTK) and idelalisib (Pi3K), and the impact of TP53 and del(17p) mutations on Nutlin-3a (MDM2) and Fludarabine (Purine analogue).


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/drugGeneIntMap-1} 

}

\caption{Plot showing BH-adjusted p values from Student's t-tests (two-sided, with equal variance), for all tested drug-gene associations. Tests performed for IGHV status and somatic mutations and copy number aberrations (n=54). Each coloured circle represents a gene-stimulus association meeting 10\% FDR cut off. Method \ref{univariate-gene-stimulus-associations-method} \emph{Figure from (\protect\hyperlink{ref-Giles2021}{Bruch et al. 2021}).}}\label{fig:drugGeneIntMap}
\end{figure}
In addition, it also highlighted the broad impact of trisomy 12 status on drug response \emph{ex vivo} (Figure \ref{fig:drugGeneIntEx}). While trisomy 12 is known to accelerate disease progression, tumours with trisomy 12 are more treatable due to higher sensitivity to chemotherapeutics and ibrutinib. In my results, trisomy 12 increased sensitivity to all drugs except ------fludarabine, BAY-11-7085 and Everolimus. These included increased sensitivity to p38 MAPK inhibition by ralimetinib, agreeing with previous work demonstrating higher trisomy 12 sensitivity with a range of MEK and ERK inhibitors, suggesting an essential role for MEK/ERK signalling in trisomy 12 CLL (\protect\hyperlink{ref-JCIpaper}{Dietrich et al. 2017}). Trisomy 12 samples also showed higher sensitivity to Nutlin-3a, an MDM2 inhibitor. Interesting, the MDM2 gene is located on chromosome 12, and is more abundant in trisomy 12 samples {[}ADD SUPP FIG{]}, which may explain this. I also noted increased trisomy 12 sensitivity to IBET-762, a BET inhibitor that has been shown to suppress transcriptional responses to cytokine signalling via JAK-STAT (\protect\hyperlink{ref-Chan2015}{Chan et al. 2015}). If trisomy 12 samples are more dependent on cytokine signals, as the results in section \ref{trisomy12-modulator} suggest, this makes sense.


\begin{figure}[H]

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/drugGeneIntEx-1} 

}

\caption{Beeswarm-boxplots showing control-normalised log transformed viability values after treatment with (A) Ralimetinib, (B) Nutlin-3a and (C) IBET-762, stratified by trisomy 12.}\label{fig:drugGeneIntEx}
\end{figure}
\hypertarget{drug-stimulus-gene-interactions}{%
\section{The modulatory effect of mutations on drug - stimuli interactions}\label{drug-stimulus-gene-interactions}}

\hypertarget{patient-specific-linear-modelling-identifies-drug---stimulus---gene-interactions}{%
\subsection{Patient-specific linear modelling identifies drug - stimulus - gene interactions}\label{patient-specific-linear-modelling-identifies-drug---stimulus---gene-interactions}}

Sections \ref{mapping-interactions} \& \ref{mapping-genetic-modulators}, explore the effects of microenvironmental signals and molecular features on drug response independently. I next aimed to investigate their collective effect on \emph{in vitro} drug efficacy in CLL, by quantifying the extent to which genetic driver mutations modulated the interactions between stimuli and drugs (\protect\hyperlink{ref-Giles2021}{Bruch et al. 2021}).

I began by conceptualising methods to quantify the collective effect of the drugs, stimuli and genetic features on CLL viability and biology. I decided to adapt the linear model in Equation \eqref{eq:drugCytInt}, by fitting this model in a patient sample - specific manner (\protect\hyperlink{ref-Giles2021}{Bruch et al. 2021}):

Equation \eqref{eq:drugCytGeneInt} quantifies how the viability with any combination can be predicted:
\begin{equation}
            log(V) = \beta_{drug}X_{drug} + \beta_{stimulus}X_{stimulus} + \beta_{patient}X_{patient} + \beta_{drug-stimulus}X_{drug}X_{stimulus} + \beta_{drug-patient}X_{drug}X_{patient} + \beta_{stimulus-patient}X_{stimulus}X_{patient} +\beta_{int}X_{drug}X_{stimulus}X_{patient} + \epsilon
                                       \label{eq:drugCytGeneInt}
    \end{equation}
\emph{where \(V\) is the predicted viability of a patient sample with a given treatment,} \(\beta_{drug}\), \(\beta_{stimulus}\), \(\beta_{int}\), \(\beta_{drug-stimulus}\), \(\beta_{drug-patient}\), \(\beta_{drug-stimulus}\) \emph{and} \(\beta_{int}X_{drug}\) \emph{are regression coefficients for the drug, stimulus, patient sample and combinatorial terms and} \(X_{drug}\), \(X_{stimulus}\) \emph{and} \(X_{patient}\) \emph{are indicator variables (0 or 1) for the presence or absence of a drug/stimulus/patient sample.} \(\epsilon\) \emph{is the vector of model residuals. See also Methods section \ref{drug-stimulus-gene-interactions-method}. Equation from (\protect\hyperlink{ref-Giles2021}{Bruch et al. 2021}).}

This generates a higher order interaction term \(\beta_{int}X_{drug}X_{stimulus}X_{patient}\). This term represents a \emph{patient sample-specific} \(\beta{int}\) for each drug - stimulus combination, quantifying the size of an interaction between a drug and stimulus in each patient genetic background.

With these patient sample-specific \(\beta{int}\) terms, it was possible to search for associations between the size of \(\beta_{int}\)) and genetic features. The aim was to screen for molecular features that increased or decreased the size of a drug - stimulus interactions, using multivariate regression with L1 (lasso) regularisation.

As input to the model, the response matrix was composed of the sample - specific \(\beta_{int}\) values for each drug-stimulus combination. To generate the feature matrix (137 samples versus 40 features), I excluded genetic features for which \textgreater20\% of the values were missing, and patient samples with incomplete annotation. As predictors, I included genetic mutations and CNVs (p=39) and IGHV status (coded as 0-1). I ran lasso regression, as implemented in the \texttt{R} package \texttt{glmnet}(\protect\hyperlink{ref-R-glmnet}{Friedman et al. 2021}), using three-fold cross-validation with misclassification error as loss. The resulting predictors are the mean of those coefficients that were selected in at least 90\% of 30 bootstrapped repeats.

This analysis revealed that 60/204 drug - stimulus interactions were modulated by at least one genetic feature (Figure \ref{fig:drugCytGeneIntMap}, Appendix Figure \ref{fig:drugcytGeneIntAll}). A positive coefficient here indicates that the presence of the genetic feature is associated with more positive \(\beta{int}\), in other words, the viability with the drug and stimulus is higher than expected in the presence of the genetic feature.

\emph{Have to make sure this figure takes a whole page but doesn't float to the bottom}


\begin{figure}

{\centering \includegraphics[width=0.5\linewidth]{thesis_files/figure-latex/drugCytGeneIntMap-1} 

}

\caption{Heatmap summarising genetic predictors of drug - stimulus interactions. Each row depicts a single drug - stimulus combination, and each tile indicates that \(\beta{int}\) for given drug and stimulus combination is modulated by corresponding genetic feature. Thus each row represents the output of a single model fit (as in Figure \ref{fig:FluCpGExA}). Colour of tile indicates size and sign of coefficient assigned to genetic feature, where a positive coefficient corresponds to a more positive \(\beta{int}\) if the feature is present. Only top 8 most commonly selected genetic features are shown. Drug - stimulus combinations with no genetic predictors of \(\beta{int}\) amongst top 8 shown are omitted for clarity. See Methods section \ref{drug-stimulus-gene-interactions-method}. \emph{Figure from (\protect\hyperlink{ref-Giles2021}{Bruch et al. 2021}).}}\label{fig:drugCytGeneIntMap}
\end{figure}
Applying this broad scale screening approach established some wider trends. Firstly, Trisomy 12 and IGHV status impacted the largest number of drug - stimulus interactions,indicating that their impact on stimulus and drug response individually also extends to drug-stimulus interactions. Secondly, out of all the stimuli, IL4 and sCD40L + IL4-based interactions were modulated by the largest number of features. IL4 alone increased viability uniformly accross genetic backgrounds; its interaction with drugs may be more influenced by genetic features. Notably, the presence of trisomy 12 acted to increase IL4-induced drug resistance, in almost all drugs for which \(\beta{int}\)\textgreater0.05. Interestingly, the presence of TP53 and del(17q) reduced the degree to which IL4 induced resistance to the chemotherapeutics Nutlin-3a and Fludarabine, most likely because both these mutations already induce resistance alone.

\hypertarget{patient---specific-drug---stimulus-interactions-may-have-clinical-significance}{%
\subsection{Patient - specific drug - stimulus interactions may have clinical significance}\label{patient---specific-drug---stimulus-interactions-may-have-clinical-significance}}

This broad scale screening approach was not perfect in itself. Modelling can only go so far, and it is likely that some cases identified here are artefacts of the modelling approach. Therefore, I decided to investigate each individual case one - by - one to establish which drug - stimulus - gene interactions are the most biologically interesting. This analysis highlighted several drug - stimulus - gene interactions that may have clinical importance and warrant further investigation (\protect\hyperlink{ref-Giles2021}{Bruch et al. 2021}).

The first of these concerned the interaction between fludarabine and CpG ODN. The value of \(\beta_{int}\) for fludarabine and CpG ODN was modulated by six genetic factors; most strongly by IGHV status, del(11q) and trisomy 12 (Figure \ref{fig:FluCpGExA}, (\protect\hyperlink{ref-Giles2021}{Bruch et al. 2021})).


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/FluCpGExA-1} 

}

\caption{Genetic features that modulate the size of \(\beta{int}\) between fludarabine and CpG ODN. Scatter plot, heatmap and bar plots indicate inputs and outputs of equation \eqref{eq:drugCytGeneInt}. Bar plots indicate size and sign of coefficients assigned to genetic features named on right. Scatter plot depicts \(\beta{int}\) values for each patient sample i.e.response matrix. Heatmap tiles indicate mutation status for the selected genetic features (i.e.~feature matrix) corresponding to same sample in scatter plot below, to show how size of \(\beta{int}\) varies with each feature. See Methods section \ref{drug-stimulus-gene-interactions-method}. \_Figure and caption from (\protect\hyperlink{ref-Giles2021._}{\textbf{Giles2021.\_?}})}\label{fig:FluCpGExA}
\end{figure}
To look in more detail at what this means, I visualised sample responses to CpG ODN and fludarabine, stratified by IGHV status and trisomy 12 (Figure \ref{fig:FluCpGExB}, (\protect\hyperlink{ref-Giles2021}{Bruch et al. 2021})). In IGHV-M non-trisomy 12 samples, TLR stimulation increased fludarabine efficacy. In samples that were either IGHV-U or trisomy 12+, TLR stimulation induced resistance to fludarabine. TLR stimulation \emph{ex vivo} can increase or decrease viability, depending on the genetic background and this also impacts on drug response. It would be useful to establish whether these dynamics are replicated within the \emph{in vivo} niche.


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/FluCpGExB-1} 

}

\caption{Beeswarm-boxplots of log(viability) values for fludarabine and CpG ODN single and combinatorial treatments, faceted by IGHV status and trisomy 12 status. P-values from paired Student's t-tests. \_Figure and caption from (\protect\hyperlink{ref-Giles2021._}{\textbf{Giles2021.\_?}})}\label{fig:FluCpGExB}
\end{figure}
TLR-induced drug resistance in certain genetic backgrounds extended to other chemotherapeutic drugs. Nutlin-3a is an MDM2 inhibitor, which in our dataset shows higher efficacy in certain backgrounds including IGHV-U, trisomy12, and del(11q) (Figure \ref{fig:drugGeneIntMap}, Figure \ref{fig:drugGeneIntEx}, Appendix Figure \ref{fig:NutlinPredictors}). Despite this, TLR stimulation reduces Nutlin-3a toxicity in these same genetic backgrounds, such that treatment with Nutlin-3a and CpG ODN is more toxic to IGHV-M, non-trisomy 12 and non-del(11q) samples i.e.~the opposite(Figure \ref{fig:nutlinCpgEx}, Figure \ref{fig:TLRCpGIGHVdel11q}). This observation underlines the need to observe drug activity \emph{in vitro} in the context of microenvironmental signals, in CLL and in other cancers.


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/nutlinCpgEx-1} 

}

\caption{Predictor profile depicting genetic features that modulate the size of \(\beta{int}\) between Nutlin-3a (MDM2) and CpG ODN. Scatter plot (bottom) and heatmap (right) and bar plots (left) indicate inputs and outputs of fitting model in Equation \eqref{eq:drugCytGeneInt}. Bar plots indicate size and sign of coefficients assigned to genetic features named on the right. Scatter plot depicts ranked patient-specific \(\beta{int}\) values i.e.~the response matrix. Heatmap above indicates patient mutation status for the selected genetic features i.e.~feature matrix. Heatmap tiles correspond to points in scatter plot (i.e.~patient data is aligned), to indicate how the size of \(\beta{int}\) varies with selected genetic feature. See Methods section \ref{drug-stimulus-gene-interactions-method}. \_Figure and caption from (\protect\hyperlink{ref-Giles2021._}{\textbf{Giles2021.\_?}})}\label{fig:nutlinCpgEx}
\end{figure}

\begin{figure}

{\centering \includegraphics[width=0.8\linewidth]{thesis_files/figure-latex/TLRCpGIGHVdel11q-1} 

}

\caption{Beeswarm-boxplot showing control-normalised log transformed viability values, after treatment with CpG ODN + Nutlin-3a, stratified by del(11q) and IGHV status. p-values from Student's t-tests. \emph{Figure from (\protect\hyperlink{ref-Giles2021}{Bruch et al. 2021})}}\label{fig:TLRCpGIGHVdel11q}
\end{figure}
The modelling approach also identified interactions, where the differential size of \(\beta{int}\) was driven by the single treatment effect, rather than the combination. For example, if a drug is more efficacious in a certain genetic background, but a stimulus nonetheless eradicates drug toxicity, the size of the interaction will be larger in this genetic background. This is because the increase in expected viability in these samples (quantified by \(\beta{int}\)) would be even higher than in other genetic backgrounds. This is the case with Ibrutinib + IL4 for instance (\protect\hyperlink{ref-Giles2021}{Bruch et al. 2021}). Ibrutinib showed higher efficacy in trisomy 12 and IGHV-U samples and IL4 induced complete resistance to ibrutinib independently of genetic background. Thus in trisomy 12 and IGHV-U samples treated with ibrutinib, the increase in viability in the context of IL4 stimulation is larger than in non-trisomy 12 and IGHV-M samples, and thus these features are assigned positive coefficients. This highlights the breadth of IL4-induced resistance to ibrutinib in CLL across genetic backgrounds, even where genetic features would indicate that ibrutinib efficacy should be high, and underlines the importance of understanding the activity of this resistance pathway \emph{in vivo} (Figure \ref{fig:ibrIl4ExA}, Figure \ref{fig:ibrIl4ExB}, (\protect\hyperlink{ref-Giles2021}{Bruch et al. 2021})).




\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/ibrIl4ExA-1} 

}

\caption{Predictor profile depicting genetic features that modulate the size of \(\beta{int}\) between ibrutinib (BTK) and IL4. Plot generated as in Figure \ref{fig:FluCpGExA}. See also Methods section \ref{drug-stimulus-gene-interactions-method}. \_Figure from (\protect\hyperlink{ref-Giles2021._}{\textbf{Giles2021.\_?}})}\label{fig:ibrIl4ExA}
\end{figure}
\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/ibrIl4ExB-1} 

}

\caption{Beeswarm boxplots of log(viability) values for ibrutinib and IL4 single and combinatorial treatments, faceted by IGHV status and trisomy 12 status. P-values from paired Student's t-tests. \_Figure and caption from (\protect\hyperlink{ref-Giles2021._}{\textbf{Giles2021.\_?}})}\label{fig:ibrIl4ExB}
\end{figure}
this is a resource, all models , and drug - stimulu resposnes stratifed by genetic features can be explored:

\hypertarget{summary-2}{%
\section{Summary}\label{summary-2}}

This work represents the first large - scale attempt to integrate the effects of mutations and the microenvironment in drug response in lymphoma, and to the best of my knowledge, in other cancers. The results provide a proof-of-principle of using combinatorial screening, molecular profiling and statistical modelling can generate more complex biological insights to guide further clinical studies.

More specifically, these results identify a number of key drug-resistance pathways, including the impact of IFN\(\gamma\) on ibrutinib. I highlight the importance of the NOS pathway in IL4 and IFN\(\gamma\) inhibitor of apoptosis. I also show how some of these resistance pathways are context-dependent. In particular, the ability of TLR stimulation to induce resistance to chemotherapeutics is modulated by many genetic feautres. THis warrants further follow-up.

Overall these findings illustrate how the presence of known genetic alterations determines how drugs and external stimuli interact with each other.

\hypertarget{discussion-1}{%
\section{Discussion}\label{discussion-1}}

IL4and ibrutinib- what's knwon
implications of ifn gamma and also p38 and ralitmetinib and ifn gamma

strategies for targeting these - pyridone 6, but not licensned? BET inihibition - paper showing that bet specfically inhibits ifn and il4 targets - so maybe somehting more braod specturm is useful

implciatiosnfor studying drug action ex vivo
In addition, we aimed to demonstrate that combinatorial screening in conjunction with linear modelling is a powerful tool to study the impact of the microenvironment on drug response in cancer, which is often under-appreciated relative to the impact of molecular features.

In addition, we aimed to demonstrate that combinatorial screening in combination with linear modelling is a powerful approach to study the impact of the microenvironment on drug response in cancer, which is often under-appreciated relative to the impact of molecular features.

How each category might be useful

the importanceof the resource here

OUr approach distinguishedbetween baseline and interaction s- this is a powerful proof of rpinciple
cancer biology is integratively influenced by cell-intrinsic and cell-extrinsic features \textless- what attempts hhave their been, persuade whythis needs to continue, how canthis behepful in the clinic

MUST READ: BET inhibitors and supression of cytokine signalling : BET Bromodomain Inhibition Suppresses Transcriptional Responses to Cytokine-Jak-STAT Signaling in a Gene-specific Manner in Human Monocytes

Nutline and IBET
nutlin-3ain trisomy 12 - MDM2 is higher

also IBET -
these also correlate - its possible that they work via a simliar mechanism -ibet inhibits brd4 whcih is inhibtiing p53 targets, so p53 is more active in thesecels

\url{https://www.nature.com/articles/s41467-020-20378-8}

NOS pathway

TLR - wahts going on in vivo, is it reallyhaving thhis bipolar effect in vivo
Flduarabine response - does this help to explain - more work needed

In Vitro Sensitivity of CLL Cells to Fludarabine May Be Modulated by the Stimulation of Toll-like Receptors
Eleonora Fonte

IL4-induced resistance- this is clearly important accross a rnage of drufs adn genetic background, more needed in vivo, possible combintatorial therapies

in vivo IL4 evidence:
from T-cells in chronic lymphocytic leukemia: Guardians or drivers of disease?
Philipp M. Roessner \& Martina Seiffert

Leukemia volume 34, pages 2012--2024 (2020)
In 2005, Görgün and colleagues investigated T-cell function in CLL. Gene expression profiling (GEP) of sorted T-cells from peripheral blood mononuclear cells (PBMCs) of CLL patients suggested that CD4+ T-cells are skewed toward a Th2 phenotype due to decreased expression of c-Jun NH2-terminal kinase (JNK) and p38 mitogen-activated protein kinase (MAPK) pathway-related genes {[}12{]}. This is supported by an increased number of interleukin (IL)-4 secreting CD4+ T-cells in PBMCs of CLL patients in comparison to cells of healthy controls (HC). Studies using the immunocompetent Eµ-TCL1 mouse model of CLL and after adoptive transfer of TCL1 leukemia (TCL1 AT) into BL/6 wild-type (WT) mice revealed more IL-4 secretion of CD4+ T-cells in comparison to control mice {[}13, 14{]}. Similarly, conditioned medium of CLL cells was able to induce secretion of IL-4 in allogeneic mixed lymphocyte reactions (MLR) of healthy individuals, suggesting that soluble factors secreted by CLL cells trigger Th2 T-cell polarization in vitro {[}15{]}.

in other cancers we shoudl integrate drugs - stimul and genetics

flurdarabine and Cpg, different outcomes in IGHV M and U patients

the imrpotatnce of IL4 in all patietn backgrounds

TLR Signaling Is Activated in Lymph Node--Resident CLL Cells and Is Only Partially Inhibited by Ibrutinib
Eman L. Dadashian
\url{https://cancerres.aacrjournals.org/content/79/2/360.short}

There are models to investigate ibrutinib resistance in vivo, but they are lacking in for ME:
Development and characterization of prototypes for in vitro and in vivo mouse models of ibrutinib-resistant CLL
Burcu Aslan

also check out discussion from :\url{https://www.sciencedirect.com/science/article/pii/S0006497119402206}

\hypertarget{contributions-statement-1}{%
\section{Contributions Statement}\label{contributions-statement-1}}

The author of this thesis performed the statistical inference and regression modelling outlined in this chapter. In essence, she conceptualised and performed the linear modelling and multivariate modelling in Equations \ldots. In addition, she performed univariate analysis of genetic determinants of drug response and data visualisation and investigation of hits.

Peter-Martin Bruch generated the viability screening data used in the model. The genetic annotation data was taken from (\protect\hyperlink{ref-JCIpaper}{Dietrich et al. 2017}). The categorisation of drug - stimulus interactions, and investigation of examples of drug - stimulus interactions was performed in collaboration with Peter-Martin Bruch for the paper Bruch \& Giles et al 2021. The author also received support from Frederich Zwiebell and Junyan Lu, who provided advice with regards to modelling approaches.

-check pdf and tidy
referencing
linear model
figures tidy
extra text

-check of text
= 6 done!

\hypertarget{chapter7}{%
\chapter{Results}\label{chapter7}}

\hypertarget{IL4resistance}{%
\subsection{IL4 induces resistance to BCR inhibitors and chemotherapeutics}\label{IL4resistance}}

(Figure \ref{fig:IL4Res}).


\begin{figure}

{\centering \includegraphics[width=0.8\linewidth]{thesis_files/figure-latex/IL4Res-1} 

}

\caption{Examples of IL4-induced resistance. Plots show log transformed viability values with each treatment, for all samples. Matching samples are linked across treatments. Black and blue horizontal lines indicate the predicted viability from the linear model for each single and combinatorial treatment. In combinatorial treatment, both the expected viability based on the additive effect of drug and stimulus (blue), and the viability with interaction (black) are shown, to indicate the impact of the interaction. Plots show IL4 + (A) Idelalisib (Pi3K), (B) PRT062607 (SYK) (C) Everolimus (mTOR), and (D) Selumetinib (p38 MAPK). See Methods section \ref{drug-stimulus-linear-model-method}. \emph{Figure from Bruch \& Giles et al.~2021.}}\label{fig:IL4Res}
\end{figure}
\begin{itemize}
\item
  Organize material and present results.
\item
  Use tables, figures (but prefer visual presentation):
  \begin{itemize}
  \item
    Tables and figures should supplement (and not duplicate) the text.
  \item
    Tables and figures should be provided with legends.
  \item
    \emph{Figure \ref{fig:graph} shows how to include and reference graphics.
    The graphic must be labelled before. Files must be in \textbf{.eps} format. You
    can do this really easily in R Markdown with \texttt{knitr::include\_graphics()}}!
  \item
    Figures can be referenced with \texttt{\textbackslash{}@ref(fig:\textless{}name\textgreater{})}, where \texttt{\textless{}name\textgreater{}} is the
    name of the code chunk.
  \end{itemize}
\end{itemize}
\begin{figure}

{\centering \includegraphics[width=0.5\linewidth]{figures/graph} 

}

\caption{Estimated residuals from model XXX4. ...}\label{fig:graph4}
\end{figure}
\begin{itemize}
\item
  Tables and graphics may appear in the text or in the appendix, especially if
  there are many simulation results tabulated, but is also depends on the study
  and number of tables resp. figures. The key graphs and tables must appear in
  the text!
\item
  R Markdown can also supports math equations just like \emph{LaTeX}!
  \begin{itemize}
  \item
    \emph{Equation \eqref{eq:SpecDens} represents the ACs of a stationary
    stochastic process:}
    \begin{equation}
            f_y(\lambda) = (2\pi)^{-1} \sum_{j=-\infty}^{\infty}
                           \gamma_j e^{-i\lambda j}
                         =(2\pi)^{-1}\left(\gamma_0 + 2 \sum_{j=1}^{\infty}
        \gamma_j \cos(\lambda j)\right)
                                       \label{eq:SpecDens}
    \end{equation}
    \emph{where \(i=\sqrt{-1}\) is the imaginary unit, \(\lambda \in [-\pi, \pi]\) is the
    frequency and the \(\gamma_j\) are the autocovariances of \(y_t\).}
  \item
    Equations can be referenced with \texttt{\textbackslash{}@ref(eq:\textless{}name\textgreater{})}, where name is defined
    by adding \texttt{(\textbackslash{}\#eq:\textless{}name\textgreater{})} in the line immediately before \texttt{\textbackslash{}end\{equation\}}.
  \end{itemize}
\end{itemize}
\hypertarget{review-of-results}{%
\section{Review of Results}\label{review-of-results}}
\begin{itemize}
\item
  Do the results support or do they contradict economic theory ?
\item
  What does the reader learn from the results?
\item
  Try to give an intuition for your results.
\item
  Provide robustness checks.
\item
  Compare to previous research.
\end{itemize}
\hypertarget{discussion-2}{%
\chapter{Discussion}\label{discussion-2}}

\newpage

\hypertarget{references}{%
\chapter*{References}\label{references}}
\addcontentsline{toc}{chapter}{References}

\noindent

\setlength{\parindent}{-0.5cm}
\setlength{\leftskip}{0.5cm}
\setlength{\parskip}{8pt}

\hypertarget{refs}{}
\begin{CSLReferences}{1}{0}
\leavevmode\vadjust pre{\hypertarget{ref-Abruzzo2018}{}}%
Abruzzo, Lynne V., Carmen D. Herling, George A. Calin, Christopher Oakes, Lynn L. Barron, Haley E. Banks, Vikram Katju, Michael J. Keating, and Kevin R. Coombes. 2018. {``{Trisomy 12 chronic lymphocytic leukemia expresses a unique set of activated and targetable pathways}.''} \emph{Haematologica} 103 (12): 2069--78. \url{https://doi.org/10.3324/haematol.2018.190132}.

\leavevmode\vadjust pre{\hypertarget{ref-Agathangelidis2012}{}}%
Agathangelidis, Andreas, Nikos Darzentas, Anastasia Hadzidimitriou, Xavier Brochet, Fiona Murray, Xiao Jie Yan, Zadie Davis, et al. 2012. {``{Stereotyped B-cell receptors in one-third of chronic lymphocytic leukemia: A molecular classification with implications for targeted therapies}.''} \emph{Blood} 119 (19): 4467--75. \url{https://doi.org/10.1182/blood-2011-11-393694}.

\leavevmode\vadjust pre{\hypertarget{ref-AguilarHernandez2016}{}}%
Aguilar-Hernandez, Maria M, Matthew D Blunt, Rachel Dobson, Alison Yeomans, Stephen Thirdborough, Marta Larrayoz, Lindsay D Smith, et al. n.d. {``{IL-4 enhances expression and function of surface IgM in CLL cells}.''} \url{https://doi.org/10.1182/blood-2015-11}.

\leavevmode\vadjust pre{\hypertarget{ref-Ahn2018}{}}%
Ahn, Inhye E, Mohammed Z H Farooqui, Xin Tian, Janet Valdez, Clare Sun, Susan Soto, Jennifer Lotter, et al. 2018. {``{Depth and durability of response to ibrutinib in CLL: 5-year follow-up of a phase 2 study.}''} \emph{Blood} 131 (21): 2357--66. \url{https://doi.org/10.1182/blood-2017-12-820910}.

\leavevmode\vadjust pre{\hypertarget{ref-R-genomation}{}}%
Akalin, Altuna, Vedran Franke, Katarzyna Wreczycka, Alexander Gosdschan, Liz Ing-Simmons, and Bozena Mika-Gospodorz. 2021. \emph{Genomation: Summary, Annotation and Visualization of Genomic Data}. \url{http://bioinformatics.mdc-berlin.de/genomation/}.

\leavevmode\vadjust pre{\hypertarget{ref-Alexandrov2013}{}}%
Alexandrov, Ludmil B., Serena Nik-Zainal, David C. Wedge, Samuel A. J. R. Aparicio, Sam Behjati, Andrew V. Biankin, Graham R. Bignell, et al. 2013. {``{Signatures of mutational processes in human cancer}.''} \emph{Nature} 500 (7463): 415--21. \url{https://doi.org/10.1038/nature12477}.

\leavevmode\vadjust pre{\hypertarget{ref-Allen2008}{}}%
Allen, Christopher D. C., and Jason G. Cyster. 2008. {``{Follicular dendritic cell networks of primary follicles and germinal centers: Phenotype and function}.''} \emph{Seminars in Immunology} 20 (1): 14--25. \url{https://doi.org/10.1016/J.SMIM.2007.12.001}.

\leavevmode\vadjust pre{\hypertarget{ref-AmigoJimenez2015}{}}%
Amigo-Jiménez, Irene, Elvira Bailón, Noemí Aguilera-Montilla, María José Terol, José A García-Marco, and Angeles García-Pardo. 2015. {``{Bone marrow stroma-induced resistance of chronic lymphocytic leukemia cells to arsenic trioxide involves Mcl-1 upregulation and is overcome by inhibiting the PI3K\(\delta\) or PKC\(\beta\) signaling pathways}.''} \emph{Oncotarget} 6 (42): 44832--48. \url{https://doi.org/10.18632/oncotarget.6265}.

\leavevmode\vadjust pre{\hypertarget{ref-Anda2020}{}}%
Anda-Jáuregui, Guillermo de, and Enrique Hernández-Lemus. 2020. {``{Computational Oncology in the Multi-Omics Era: State of the Art}.''} Frontiers. \url{https://doi.org/10.3389/fonc.2020.00423}.

\leavevmode\vadjust pre{\hypertarget{ref-Argelaguet2018}{}}%
Argelaguet, Ricard, Britta Velten, Damien Arnol, Sascha Dietrich, Thorsten Zenz, John C Marioni, Florian Buettner, Wolfgang Huber, and Oliver Stegle. 2018. {``{Multi‐Omics Factor Analysis---a framework for unsupervised integration of multi‐omics data sets}.''} \emph{Molecular Systems Biology} 14 (6): e8124. \url{https://doi.org/10.15252/msb.20178124}.

\leavevmode\vadjust pre{\hypertarget{ref-Asslaber2013}{}}%
Asslaber, Daniela, Eva M Grössinger, Tamara Girbl, Sebastian W Hofbauer, Alexander Egle, Lukas Weiss, Richard Greil, and Tanja N Hartmann. 2013. {``{Mimicking the microenvironment in chronic lymphocytic leukaemia - where does the journey go?}''} Br J Haematol. \url{https://doi.org/10.1111/bjh.12151}.

\leavevmode\vadjust pre{\hypertarget{ref-Austen2005}{}}%
Austen, Belinda, Judith E Powell, Azra Alvi, Ian Edwards, Laura Hooper, Jane Starczynski, A Malcolm R Taylor, Christopher Fegan, Paul Moss, and Tatjana Stankovic. 2005. {``{Mutations in the ATM gene lead to impaired overall and treatment-free survival that is independent of IGVH mutation status in patients with B-CLL}.''} \emph{Blood} 106 (9): 3175--82. \url{https://doi.org/10.1182/blood-2004-11-4516}.

\leavevmode\vadjust pre{\hypertarget{ref-Axelrod2014}{}}%
Axelrod, M, Z Ou, L K Brett, L Zhang, E R Lopez, A T Tamayo, V Gordon, et al. 2014. {``{Combinatorial drug screening identifies synergistic co-targeting of Bruton's tyrosine kinase and the proteasome in mantle cell lymphoma}.''} Nature Publishing Group. \url{https://doi.org/10.1038/leu.2013.249}.

\leavevmode\vadjust pre{\hypertarget{ref-Bagnara2011}{}}%
Bagnara, Davide, Matthew S. Kaufman, Carlo Calissano, Sonia Marsilio, Piers E. M. Patten, Rita Simone, Philip Chum, et al. 2011. {``{A novel adoptive transfer model of chronic lymphocytic leukemia suggests a key role for T lymphocytes in the disease}.''} \emph{Blood} 117 (20): 5463--72. \url{https://doi.org/10.1182/blood-2010-12-324210}.

\leavevmode\vadjust pre{\hypertarget{ref-Balatti2012}{}}%
Balatti, Veronica, Arianna Bottoni, Alexey Palamarchuk, Hansjuerg Alder, Laura Z. Rassenti, Thomas J. Kipps, Yuri Pekarsky, and Carlo M. Croce. 2012. {``{NOTCH1 mutations in CLL associated with trisomy 12}.''} \emph{Blood} 119 (2): 329--31. \url{https://doi.org/10.1182/blood-2011-10-386144}.

\leavevmode\vadjust pre{\hypertarget{ref-Barretina2012}{}}%
Barretina, Jordi, Giordano Caponigro, Nicolas Stransky, Kavitha Venkatesan, Adam A. Margolin, Sungjoon Kim, Christopher J. Wilson, et al. 2012. {``{The Cancer Cell Line Encyclopedia enables predictive modelling of anticancer drug sensitivity}.''} \emph{Nature} 483 (7391): 603--7. \url{https://doi.org/10.1038/nature11003}.

\leavevmode\vadjust pre{\hypertarget{ref-Basu2013}{}}%
Basu, Amrita, Nicole E Bodycombe, Jaime H Cheah, Edmund V Price, Ke Liu, Giannina I Schaefer, Richard Y Ebright, et al. 2013. {``{XAn interactive resource to identify cancer genetic and lineage dependencies targeted by small molecules}.''} \emph{Cell} 154 (5): 1151--61. \url{https://doi.org/10.1016/j.cell.2013.08.003}.

\leavevmode\vadjust pre{\hypertarget{ref-Baumann2021}{}}%
Baumann, Tycho, Riccardo Moia, Gianluca Gaidano, Julio Delgado, Adalgisa Condoluci, Neus Villamor, Anil Babu Payedimarri, et al. 2021. {``{Lymphocyte doubling time in chronic lymphocytic leukemia modern era: a real-life study in 848 unselected patients}.''} \emph{Leukemia} 35 (8): 2325--31. \url{https://doi.org/10.1038/s41375-021-01149-w}.

\leavevmode\vadjust pre{\hypertarget{ref-Beekman2018}{}}%
Beekman, Renée, Vicente Chapaprieta, Núria Russiñol, Roser Vilarrasa-Blasi, Núria Verdaguer-Dot, Joost H. A. Martens, Martí Duran-Ferrer, et al. 2018. {``{The reference epigenome and regulatory chromatin landscape of chronic lymphocytic leukemia}.''} \emph{Nature Medicine} 24 (6): 868--80. \url{https://doi.org/10.1038/s41591-018-0028-4}.

\leavevmode\vadjust pre{\hypertarget{ref-Berest2019}{}}%
Berest, Ivan, Christian Arnold, Armando Reyes-Palomares, Giovanni Palla, Kasper Dindler Rasmussen, Holly Amelia Rebecca Giles, Peter-Martin Bruch, et al. 2019. {``{Quantification of Differential Transcription Factor Activity and Multiomics-Based Classification into Activators and Repressors: diffTF}.''} \emph{Cell Reports} 29 (10): 3147--3159.e12. \url{https://doi.org/10.1016/J.CELREP.2019.10.106}.

\leavevmode\vadjust pre{\hypertarget{ref-Bhattacharya2015}{}}%
Bhattacharya, Nupur, Michaela Reichenzeller, Maiwen Caudron-Herger, Sarah Haebe, Nathan Brady, Susanne Diener, Maria Nothing, et al. 2015. {``{Loss of cooperativity of secreted CD40L and increased dose-response to IL4 on CLL cell viability correlates with enhanced activation of NF-kB and STAT6}.''} \emph{International Journal of Cancer} 136 (1): 65--73. \url{https://doi.org/10.1002/ijc.28974}.

\leavevmode\vadjust pre{\hypertarget{ref-Bhoi2016}{}}%
Bhoi, Sujata, Viktor Ljungström, Panagiotis Baliakas, Mattias Mattsson, Karin E Smedby, Gunnar Juliusson, Richard Rosenquist, and Larry Mansouri. 2016. \emph{Epigenetics} 11 (6): 449--55. \url{https://doi.org/10.1080/15592294.2016.1178432}.

\leavevmode\vadjust pre{\hypertarget{ref-Binder2010}{}}%
Binder, Mascha, Barbara Léchenne, Ramesh Ummanni, Christan Scharf, Stefan Balabanov, Maria Trusch, Hartmut Schlüter, Ingke Braren, Edzard Spillner, and Martin Trepel. 2010. {``{Stereotypical Chronic Lymphocytic Leukemia B-Cell Receptors Recognize Survival Promoting Antigens on Stromal Cells}.''} \emph{PLOS ONE} 5 (12): e15992. \url{https://doi.org/10.1371/JOURNAL.PONE.0015992}.

\leavevmode\vadjust pre{\hypertarget{ref-Boissard2015}{}}%
Boissard, F, J. J. Fournié, A Quillet-Mary, L Ysebaert, and M Poupot. 2015. {``{Nurse-like cells mediate ibrutinib resistance in chronic lymphocytic leukemia patients}.''} \emph{Blood Cancer Journal} 5 (10): e355. \url{https://doi.org/10.1038/bcj.2015.74}.

\leavevmode\vadjust pre{\hypertarget{ref-Borche1990}{}}%
Borche, L, A Lim, JL Binet, and G Dighiero. 1990. {``{Evidence that chronic lymphocytic leukemia B lymphocytes are frequently committed to production of natural autoantibodies}.''} \emph{Blood} 76 (3): 562--69. \url{https://doi.org/10.1182/blood.v76.3.562.562}.

\leavevmode\vadjust pre{\hypertarget{ref-Bosch2019}{}}%
Bosch, Francesc, and Riccardo Dalla-Favera. 2019. {``{Chronic lymphocytic leukaemia: from genetics to treatment}.''} \emph{Nature Reviews Clinical Oncology} 16 (11): 684--701. \url{https://doi.org/10.1038/s41571-019-0239-8}.

\leavevmode\vadjust pre{\hypertarget{ref-Broker1988}{}}%
Bröker, Barbara M., Adek Klajman, Pierre Youinou, Jean Jouquan, Colin P. Worman, John Murphy, Lorna Mackenzie, et al. 1988. {``{Chronic lymphocytic leukemic (CLL) cells secrete multispecific autoantibodies}.''} \emph{Journal of Autoimmunity} 1 (5): 469--81. \url{https://doi.org/10.1016/0896-8411(88)90068-6}.

\leavevmode\vadjust pre{\hypertarget{ref-Brown2014}{}}%
Brown, Jennifer R, John C Byrd, Steven E Coutre, Don M Benson, Ian W Flinn, Nina D Wagner-Johnston, Stephen E Spurgeon, et al. 2014. {``{Idelalisib, an inhibitor of phosphatidylinositol 3-kinase p110\(\delta\), for relapsed/refractory chronic lymphocytic leukemia}.''} \emph{Blood} 123 (22): 3390--97. \url{https://doi.org/10.1182/blood-2013-11-535047}.

\leavevmode\vadjust pre{\hypertarget{ref-Giles2021}{}}%
Bruch, Peter-Martin, Holly A. R. Giles, Carolin Kolb, Sophie A. Herbst, Tina Becirovic, Tobias Roider, Junyan Lu, et al. 2021. {``{Mapping drug-microenvironment-genetic interplay in CLL reveals trisomy 12 as a modulator of microenvironmental signals}.''} \emph{bioRxiv Cancer Biology}, July, 2021.07.23.453514. \url{https://doi.org/10.1101/2021.07.23.453514}.

\leavevmode\vadjust pre{\hypertarget{ref-Burger2001}{}}%
Burger, J A, N J Zvaifler, N Tsukada, G S Firestein, and T J Kipps. 2001. {``{Fibroblast-like synoviocytes support B-cell pseudoemperipolesis via a stromal cell-derived factor-1- and CD106 (VCAM-1)-dependent mechanism.}''} \emph{The Journal of Clinical Investigation} 107 (3): 305--15. \url{https://doi.org/10.1172/JCI11092}.

\leavevmode\vadjust pre{\hypertarget{ref-Burger2020}{}}%
Burger, Jan A. 2020. {``{Treatment of Chronic Lymphocytic Leukemia}.''} Edited by Dan L. Longo. \emph{New England Journal of Medicine} 383 (5): 460--73. \url{https://doi.org/10.1056/NEJMra1908213}.

\leavevmode\vadjust pre{\hypertarget{ref-Burger1999}{}}%
Burger, Jan A., Meike Burger, and Thomas J. Kipps. 1999. {``{Chronic Lymphocytic Leukemia B Cells Express Functional Cxcr4 Chemokine Receptors That Mediate Spontaneous Migration Beneath Bone Marrow Stromal Cells}.''} \emph{Blood} 94 (11): 3658--67. \url{https://doi.org/10.1182/BLOOD.V94.11.3658}.

\leavevmode\vadjust pre{\hypertarget{ref-Burger2013}{}}%
Burger, Jan A, and Nicholas Chiorazzi. 2013. {``{B cell receptor signaling in chronic lymphocytic leukemia}.''} Trends Immunol. \url{https://doi.org/10.1016/j.it.2013.07.002}.

\leavevmode\vadjust pre{\hypertarget{ref-Burger2014}{}}%
Burger, Jan A, and John G Gribben. 2014. {``{The microenvironment in chronic lymphocytic leukemia (CLL) and other B cell malignancies: insight into disease biology and new targeted therapies.}''} \emph{Seminars in Cancer Biology} 24 (February): 71--81. \url{https://doi.org/10.1016/j.semcancer.2013.08.011}.

\leavevmode\vadjust pre{\hypertarget{ref-Burger2000}{}}%
Burger, Jan A., Nobuhiro Tsukada, Meike Burger, Nathan J. Zvaifler, Marie Dell'Aquila, and Thomas J. Kipps. 2000. {``{Blood-derived nurse-like cells protect chronic lymphocytic leukemia B cells from spontaneous apoptosis through stromal cell--derived factor-1}.''} \emph{Blood} 96 (8): 2655--63. \url{https://doi.org/10.1182/BLOOD.V96.8.2655}.

\leavevmode\vadjust pre{\hypertarget{ref-Burger2005}{}}%
Burger, Meike, Tanja Hartmann, Myriam Krome, Justyna Rawluk, Hirokazu Tamamura, Nobutaka Fujii, Thomas J. Kipps, and Jan A. Burger. 2005. {``{Small peptide inhibitors of the CXCR4 chemokine receptor (CD184) antagonize the activation, migration, and antiapoptotic responses of CXCL12 in chronic lymphocytic leukemia B cells}.''} \emph{Blood} 106 (5): 1824--30. \url{https://doi.org/10.1182/blood-2004-12-4918}.

\leavevmode\vadjust pre{\hypertarget{ref-Burkle2007}{}}%
Bürkle, Andrea, Matthias Niedermeier, Annette Schmitt-Gräff, William G. Wierda, Michael J. Keating, and Jan A. Burger. 2007. {``{Overexpression of the CXCR5 chemokine receptor, and its ligand, CXCL13 in B-cell chronic lymphocytic leukemia}.''} \emph{Blood} 110 (9): 3316--25. \url{https://doi.org/10.1182/blood-2007-05-089409}.

\leavevmode\vadjust pre{\hypertarget{ref-Byrd2014}{}}%
Byrd, John C., Jennifer R. Brown, Susan O'Brien, Jacqueline C. Barrientos, Neil E. Kay, Nishitha M. Reddy, Steven Coutre, et al. 2014. {``{Ibrutinib versus Ofatumumab in Previously Treated Chronic Lymphoid Leukemia}.''} \emph{New England Journal of Medicine} 371 (3): 213--23. \url{https://doi.org/10.1056/nejmoa1400376}.

\leavevmode\vadjust pre{\hypertarget{ref-Byrd2013}{}}%
Byrd, John C, Richard R Furman, Steven E Coutre, Ian W Flinn, Jan A Burger, Kristie A Blum, Barbara Grant, et al. 2013. {``{Targeting BTK with ibrutinib in relapsed chronic lymphocytic leukemia.}''} \emph{The New England Journal of Medicine} 369 (1): 32--42. \url{https://doi.org/10.1056/NEJMoa1215637}.

\leavevmode\vadjust pre{\hypertarget{ref-Cahill2013}{}}%
Cahill, N, A. C. Bergh, M Kanduri, H Göransson-Kultima, L Mansouri, A Isaksson, F Ryan, et al. 2013. {``{450K-array analysis of chronic lymphocytic leukemia cells reveals global DNA methylation to be relatively stable over time and similar in resting and proliferative compartments}.''} \emph{Leukemia} 27 (1): 150--58. \url{https://doi.org/10.1038/leu.2012.245}.

\leavevmode\vadjust pre{\hypertarget{ref-Calin2002}{}}%
Calin, George Adrian, Calin Dan Dumitru, Masayoshi Shimizu, Roberta Bichi, Simona Zupo, Evan Noch, Hansjuerg Aldler, et al. 2002. {``{Frequent deletions and down-regulation of micro-RNA genes miR15 and miR16 at 13q14 in chronic lymphocytic leukemia}.''} \emph{Proceedings of the National Academy of Sciences of the United States of America} 99 (24): 15524--29. \url{https://doi.org/10.1073/pnas.242606799}.

\leavevmode\vadjust pre{\hypertarget{ref-Campbell2020}{}}%
Campbell, Belinda A, Julia J Scarisbrick, Youn H Kim, Ryan A Wilcox, Christopher McCormack, and H Miles Prince. 2020. {``{Time to next treatment as a meaningful endpoint for trials of primary cutaneous lymphoma}.''} \emph{Cancers} 12 (8): 1--11. \url{https://doi.org/10.3390/cancers12082311}.

\leavevmode\vadjust pre{\hypertarget{ref-Care2014}{}}%
Care, Matthew A., Mario Cocco, Jon P. Laye, Nicholas Barnes, Yuanxue Huang, Ming Wang, Sharon Barrans, et al. 2014. {``{SPIB and BATF provide alternate determinants of IRF4 occupancy in diffuse large B-cell lymphoma linked to disease heterogeneity}.''} \emph{Nucleic Acids Research} 42 (12): 7591--7610. \url{https://doi.org/10.1093/nar/gku451}.

\leavevmode\vadjust pre{\hypertarget{ref-Carey2017}{}}%
Carey, Alyssa, David K Edwards, Christopher A Eide, Laura Newell, Elie Traer, Bruno C Medeiros, Daniel A Pollyea, et al. 2017. {``{Identification of Interleukin-1 by Functional Screening as a Key Mediator of Cellular Expansion and Disease Progression in Acute Myeloid Leukemia}.''} \emph{Cell Reports} 18 (13): 3204--18. \url{https://doi.org/10.1016/j.celrep.2017.03.018}.

\leavevmode\vadjust pre{\hypertarget{ref-Chan2015}{}}%
Chan, Chun Hin, Celestia Fang, Anna Yarilina, Rab K Prinjha, Yu Qiao, and Lionel B Ivashkiv. 2015. {``{BET bromodomain inhibition suppresses transcriptional responses to cytokine-Jak-STAT signaling in a gene-specific manner in human monocytes.}''} \emph{European Journal of Immunology} 45 (1): 287--97. \url{https://doi.org/10.1002/eji.201444862}.

\leavevmode\vadjust pre{\hypertarget{ref-Chang2012}{}}%
Chang, Julie Elizabeth, and Brad Steven Kahl. 2012. {``{Bendamustine for treatment of chronic lymphocytic leukemia}.''} \emph{Expert Opinion on Pharmacotherapy} 13 (10): 1495--505. \url{https://doi.org/10.1517/14656566.2012.693163}.

\leavevmode\vadjust pre{\hypertarget{ref-Chatzouli2014}{}}%
Chatzouli, Maria, Stavroula Ntoufa, Nikos Papakonstantinou, Elisavet Chartomatsidou, Achilles Anagnostopoulos, Panagoula Kollia, Paolo Ghia, Marta Muzio, Kostas Stamatopoulos, and Chrysoula Belessi. 2014. {``{Heterogeneous Functional Effects of Concomitant B Cell Receptor and TLR Stimulation in Chronic Lymphocytic Leukemia with Mutated versus Unmutated Ig Genes}.''} \emph{The Journal of Immunology} 192 (10): 4518--24. \url{https://doi.org/10.4049/jimmunol.1302102}.

\leavevmode\vadjust pre{\hypertarget{ref-Cheng2014}{}}%
Cheng, S, J Ma, A Guo, P Lu, J P Leonard, M Coleman, M Liu, J J Buggy, R R Furman, and Y L Wang. 2014. {``{BTK inhibition targets in vivo CLL proliferation through its effects on B-cell receptor signaling activity}.''} \emph{Leukemia} 28 (3): 649--57. \url{https://doi.org/10.1038/leu.2013.358}.

\leavevmode\vadjust pre{\hypertarget{ref-Chiorazzi2003}{}}%
Chiorazzi, Nicholas, and Manlio Ferrarini. 2003. {``{B cell chronic lymphocytic leukemia: Lessons learned from studies of the B cell antigen receptor}.''} Annual Reviews 4139 El Camino Way, P.O. Box 10139, Palo Alto, CA 94303-0139, USA. \url{https://doi.org/10.1146/annurev.immunol.21.120601.141018}.

\leavevmode\vadjust pre{\hypertarget{ref-Chiorazzi2005}{}}%
Chiorazzi, Nicholas, Kanti R Rai, and Manlio Ferrarini. 2005. {``{Chronic lymphocytic leukemia.}''} \emph{The New England Journal of Medicine} 352 (8): 804--15. \url{https://doi.org/10.1056/NEJMra041720}.

\leavevmode\vadjust pre{\hypertarget{ref-Chu2008}{}}%
Chu, Charles C., Rosa Catera, Katerina Hatzi, Xiao Jie Yan, Lu Zhang, Xiao Bo Wang, Henry M. Fales, et al. 2008. {``{Chronic lymphocytic leukemia antibodies with a common stereotypic rearrangement recognize nonmuscle myosin heavy chain IIA}.''} \emph{Blood} 112 (13): 5122--29. \url{https://doi.org/10.1182/blood-2008-06-162024}.

\leavevmode\vadjust pre{\hypertarget{ref-Chung2003}{}}%
Chung, James B., Michael Silverman, and John G. Monroe. 2003. {``{Transitional B cells: step by step towards immune competence}.''} \emph{Trends in Immunology} 24 (6): 342--48. \url{https://doi.org/10.1016/S1471-4906(03)00119-4}.

\leavevmode\vadjust pre{\hypertarget{ref-Cimmino2005}{}}%
Cimmino, Amelia, George Adrian Calin, Muller Fabbri, Marilena V Iorio, Manuela Ferracin, Masayoshi Shimizu, Sylwia E Wojcik, et al. 2005. {``{miR-15 and miR-16 induce apoptosis by targeting BCL2}.''} \emph{Proceedings of the National Academy of Sciences of the United States of America} 102 (39): 13944--49. \url{https://doi.org/10.1073/pnas.0506654102}.

\leavevmode\vadjust pre{\hypertarget{ref-Collins1989}{}}%
Collins, Russell J., Louise A. Verschuer, Brian V. Harmon, Roger L. Prentice, John H. Pope, and John F. R. Kerr. 1989. {``{Spontaneous programmed death (apoptosis) of B‐chronic lymphocytic leukaemia cells following their culture in vitro}.''} \emph{British Journal of Haematology} 71 (3): 343--50. \url{https://doi.org/10.1111/j.1365-2141.1989.tb04290.x}.

\leavevmode\vadjust pre{\hypertarget{ref-Crassini2017}{}}%
Crassini, Kyle, Yandong Shen, Stephen Mulligan, and O. Giles Best. 2017. {``{Modeling the chronic lymphocytic leukemia microenvironment in vitro}.''} \emph{Leukemia and Lymphoma} 58 (2): 266--79. \url{https://doi.org/10.1080/10428194.2016.1204654}.

\leavevmode\vadjust pre{\hypertarget{ref-Crompot2017}{}}%
Crompot, Emerence, Michael Van Damme, Karlien Pieters, Marjorie Vermeersch, David Perez-Morga, Philippe Mineur, Marie Maerevoet, et al. 2017. {``{Extracellular vesicles of bone marrow stromal cells rescue chronic lymphocytic leukemia B cells from apoptosis, enhance their migration and induce gene expression modifications}.''} \emph{Haematologica} 102 (9): 1594--604. \url{https://doi.org/10.3324/haematol.2016.163337}.

\leavevmode\vadjust pre{\hypertarget{ref-Damle1999}{}}%
Damle, Rajendra N., Tarun Wasil, Franco Fais, Fabio Ghiotto, Angelo Valetto, Steven L. Allen, Aby Buchbinder, et al. 1999. {``{Ig V Gene Mutation Status and CD38 Expression As Novel Prognostic Indicators in Chronic Lymphocytic Leukemia}.''} \emph{Blood} 94 (6): 1840--47. \url{https://doi.org/10.1182/BLOOD.V94.6.1840}.

\leavevmode\vadjust pre{\hypertarget{ref-Dancescu1992}{}}%
Dancescu, M, M Rubio-Trujillo, G Biron, D Bron, G Delespesse, and M Sarfati. 1992. {``{Interleukin 4 protects chronic lymphocytic leukemic B cells from death by apoptosis and upregulates Bcl-2 expression.}''} \emph{The Journal of Experimental Medicine} 176 (5): 1319--26. \url{https://doi.org/10.1084/jem.176.5.1319}.

\leavevmode\vadjust pre{\hypertarget{ref-Rooij2012}{}}%
De Rooij, Martin F. M., Annemieke Kuil, Christian R Geest, Eric Eldering, Betty Y Chang, Joseph J Buggy, Steven T Pals, and Marcel Spaargaren. 2012. {``{The clinically active BTK inhibitor PCI-32765 targets B-cell receptor- and chemokine-controlled adhesion and migration in chronic lymphocytic leukemia}.''} \emph{Blood} 119 (11): 2590--94. \url{https://doi.org/10.1182/blood-2011-11-390989}.

\leavevmode\vadjust pre{\hypertarget{ref-Deaglio2009}{}}%
Deaglio, Silvia, and Fabio Malavasi. 2009. {``{Chronic lymphocytic leukemia microenvironment: Shifting the balance from apoptosis to proliferation}.''} \url{https://doi.org/10.3324/haematol.2009.006676}.

\leavevmode\vadjust pre{\hypertarget{ref-Decker2010}{}}%
Decker, Thomas, Christian Bogner, Madlen Oelsner, Christian Peschel, and Ingo Ringshausen. 2010. {``{Antiapoptotic effect of interleukin-2 (IL-2) in B-CLL cells with low and high affinity IL-2 receptors}.''} \emph{Annals of Hematology} 89 (11): 1125--32. \url{https://doi.org/10.1007/s00277-010-0994-1}.

\leavevmode\vadjust pre{\hypertarget{ref-Defoiche2008}{}}%
Defoiche, Julien, Christophe Debacq, Becca Asquith, Yan Zhang, Arsène Burny, Dominique Bron, Laurence Lagneaux, Derek Macallan, and Luc Willems. 2008. {``{Reduction of B cell turnover in chronic lymphocytic leukaemia}.''} \emph{British Journal of Haematology} 143 (2): 240--47. \url{https://doi.org/10.1111/j.1365-2141.2008.07348.x}.

\leavevmode\vadjust pre{\hypertarget{ref-Giudice2012}{}}%
Del Giudice, Ilaria, Davide Rossi, Sabina Chiaretti, Marilisa Marinelli, Simona Tavolaro, Sara Gabrielli, Luca Laurenti, et al. 2012. {``{NOTCH1 mutations in +12 chronic lymphocytic leukemia (CLL) confer an unfavorable prognosis, induce a distinctive transcriptional profiling and refine the intermediate prognosis of +12 CLL}.''} \emph{Haematologica} 97 (3): 437--41. \url{https://doi.org/10.3324/haematol.2011.060129}.

\leavevmode\vadjust pre{\hypertarget{ref-Delgado2021}{}}%
Delgado, Amanda, and Achuta Kumar Guddati. 2021. {``{Clinical endpoints in oncology - a primer.}''} \emph{American Journal of Cancer Research} 11 (4): 1121--31. \url{http://www.ncbi.nlm.nih.gov/pubmed/33948349\%20http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=PMC8085844}.

\leavevmode\vadjust pre{\hypertarget{ref-JCIpaper}{}}%
Dietrich, Sascha, Małgorzata Oleś, Junyan Lu, Leopold Sellner, Simon Anders, Britta Velten, Bian Wu, et al. 2017. {``{Drug-perturbation-based stratification of blood cancer}.''} \emph{Journal of Clinical Investigation} 128 (1): 427--45. \url{https://doi.org/10.1172/JCI93801}.

\leavevmode\vadjust pre{\hypertarget{ref-Ding2009}{}}%
Ding, Wei, Grzegorz S. Nowakowski, Traci R. Knox, Justin C. Boysen, Mary L. Maas, Susan M. Schwager, Wenting Wu, et al. 2009. {``{Bi-directional activation between mesenchymal stem cells and CLL B-cells: implication for CLL disease progression}.''} \emph{British Journal of Haematology} 147 (4): 471--83. \url{https://doi.org/10.1111/j.1365-2141.2009.07868.x}.

\leavevmode\vadjust pre{\hypertarget{ref-Dohner1995}{}}%
Döhner, Hartmut, Konstanze Fischer, Martin Bentz, Katrin Hansen, Axel Benner, Georges Cabot, Daniela Diehl, et al. 1995. {``{p53 Gene Deletion Predicts for Poor Survival and Non-Response to Therapy With Purine Analogs in Chronic B-Cell Leukemias}.''} \emph{Blood} 85 (6): 1580--89. \url{https://doi.org/10.1182/BLOOD.V85.6.1580.BLOODJOURNAL8561580}.

\leavevmode\vadjust pre{\hypertarget{ref-Dohner2000}{}}%
Döhner, H, Stephan Stilgenbauer, Axel Benner, Elke Leupolt, Alexander Kröber, Lars Bullinger, Konstanze Döhner, Martin Bentz, and Peter Lichter. 2000. {``{Genomic aberrations and survival in chronic lymphocytic leukemia.}''} \emph{The New England Journal of Medicine} 343 (26): 1910--16. \url{https://doi.org/10.1056/NEJM200012283432602}.

\leavevmode\vadjust pre{\hypertarget{ref-R-msigdbr}{}}%
Dolgalev, Igor. 2021. \emph{Msigdbr: MSigDB Gene Sets for Multiple Organisms in a Tidy Data Format}. \url{https://igordot.github.io/msigdbr/}.

\leavevmode\vadjust pre{\hypertarget{ref-Du2015}{}}%
Du, W, and O Elemento. 2015. {``{Cancer systems biology: Embracing complexity to develop better anticancer therapeutic strategies}.''} \emph{Oncogene} 34 (25): 3215--25. \url{https://doi.org/10.1038/onc.2014.291}.

\leavevmode\vadjust pre{\hypertarget{ref-Dubois2020}{}}%
Dubois, Nathan, Emerence Crompot, Nathalie Meuleman, Dominique Bron, Laurence Lagneaux, and Basile Stamatopoulos. 2020. {``{Importance of Crosstalk Between Chronic Lymphocytic Leukemia Cells and the Stromal Microenvironment: Direct Contact, Soluble Factors, and Extracellular Vesicles}.''} Frontiers Media SA. \url{https://doi.org/10.3389/fonc.2020.01422}.

\leavevmode\vadjust pre{\hypertarget{ref-Elston2020}{}}%
Elston, Lauren, Chris Fegan, Robert Hills, Shaikh S Hashimdeen, Elisabeth Walsby, Peter Henley, Chris Pepper, and Stephen Man. 2020. {``{Increased frequency of CD4+ PD-1+ HLA-DR+ T cells is associated with disease progression in CLL.}''} \emph{British Journal of Haematology} 188 (6): 872--80. \url{https://doi.org/10.1111/bjh.16260}.

\leavevmode\vadjust pre{\hypertarget{ref-Enzler2009}{}}%
Enzler, Thomas, Arnon P Kater, Weizhou Zhang, George F Widhopf, Han Yu Chuang, Jason Lee, Esther Avery, Carlo M Croce, Michael Karin, and Thomas J Kipps. 2009. {``{Chronic lymphocytic leukemia of E\(\mu\)-TCL1 transgenic mice undergoes rapid cell turnover that can be offset by extrinsic CD257 to accelerate disease progression}.''} \emph{Blood} 114 (20): 4469--76. \url{https://doi.org/10.1182/blood-2009-06-230169}.

\leavevmode\vadjust pre{\hypertarget{ref-Fabbri2016}{}}%
Fabbri, Giulia, and Riccardo Dalla-Favera. 2016. {``{The molecular pathogenesis of chronic lymphocytic leukaemia}.''} Nat Rev Cancer. \url{https://doi.org/10.1038/nrc.2016.8}.

\leavevmode\vadjust pre{\hypertarget{ref-Fabbri2011}{}}%
Fabbri, Giulia, Silvia Rasi, Davide Rossi, Vladimir Trifonov, Hossein Khiabanian, Jing Ma, Adina Grunn, et al. 2011. {``{Analysis of the chronic lymphocytic leukemia coding genome: Role of NOTCH1 mutational activation}.''} \emph{Journal of Experimental Medicine} 208 (7): 1389--1401. \url{https://doi.org/10.1084/jem.20110921}.

\leavevmode\vadjust pre{\hypertarget{ref-Fais1998}{}}%
Fais, Franco, Fabio Ghiotto, Shiori Hashimoto, Brian Sellars, Angelo Valetto, Steven L Allen, Philip Schulman, et al. 1998. {``{Chronic lymphocytic leukemia B cells express restricted sets of mutated and unmutated antigen receptors}.''} \emph{Journal of Clinical Investigation} 102 (8): 1515--25. \url{https://doi.org/10.1172/JCI3009}.

\leavevmode\vadjust pre{\hypertarget{ref-Farinello2018}{}}%
Farinello, Diego, Monika Wozińska, Elisa Lenti, Luca Genovese, Silvia Bianchessi, Edoardo Migliori, Nicolò Sacchetti, et al. 2018. {``{A retinoic acid-dependent stroma-leukemia crosstalk promotes chronic lymphocytic leukemia progression}.''} \emph{Nature Communications} 9 (1): 1787. \url{https://doi.org/10.1038/s41467-018-04150-7}.

\leavevmode\vadjust pre{\hypertarget{ref-Fedorchenko2013}{}}%
Fedorchenko, Oleg, Marius Stiefelhagen, Abdul A. Peer-Zada, Romy Barthel, Petra Mayer, Laura Eckei, Alexandra Breuer, et al. 2013. {``{CD44 regulates the apoptotic response and promotes disease development in chronic lymphocytic leukemia}.''} \emph{Blood} 121 (20): 4126--36. \url{https://doi.org/10.1182/blood-2012-11-466250}.

\leavevmode\vadjust pre{\hypertarget{ref-Ferreira2014}{}}%
Ferreira, Pedro G, Pedro Jares, Daniel Rico, Gonzalo Gómez-López, Alejandra Martínez-Trillos, Neus Villamor, Simone Ecker, et al. 2014. {``{Transcriptome characterization by RNA sequencing identifies a major molecular and clinical subdivision in chronic lymphocytic leukemia}.''} \emph{Genome Research} 24 (2): 212--26. \url{https://doi.org/10.1101/gr.152132.112}.

\leavevmode\vadjust pre{\hypertarget{ref-Fischer2020}{}}%
Fischer, Ute, Jun J. Yang, Tomokatsu Ikawa, Daniel Hein, Carolina Vicente-Dueñas, Arndt Borkhardt, and Isidro Sánchez-García. 2020. {``{Cell Fate Decisions: The Role of Transcription Factors in Early B-cell Development and Leukemia}.''} \emph{Blood Cancer Discovery} 1 (3): 224--33. \url{https://doi.org/10.1158/2643-3230.bcd-20-0011}.

\leavevmode\vadjust pre{\hypertarget{ref-Friedberg2010}{}}%
Friedberg, Jonathan W, Jeff Sharman, John Sweetenham, Patrick B Johnston, Julie M Vose, Ann LaCasce, Julia Schaefer-Cutillo, et al. 2010. {``{Inhibition of Syk with fostamatinib disodium has significant clinical activity in non-Hodgkin lymphoma and chronic lymphocytic leukemia}.''} \emph{Blood} 115 (13): 2578--85. \url{https://doi.org/10.1182/blood-2009-08-236471}.

\leavevmode\vadjust pre{\hypertarget{ref-R-glmnet}{}}%
Friedman, Jerome, Trevor Hastie, Rob Tibshirani, Balasubramanian Narasimhan, Kenneth Tay, and Noah Simon. 2021. \emph{Glmnet: Lasso and Elastic-Net Regularized Generalized Linear Models}. \url{https://CRAN.R-project.org/package=glmnet}.

\leavevmode\vadjust pre{\hypertarget{ref-Furman2014}{}}%
Furman, Richard R, Jeff P Sharman, Steven E Coutre, Bruce D Cheson, John M Pagel, Peter Hillmen, Jacqueline C Barrientos, et al. 2014. {``{Idelalisib and Rituximab in Relapsed Chronic Lymphocytic Leukemia}.''} \emph{New England Journal of Medicine} 370 (11): 997--1007. \url{https://doi.org/10.1056/nejmoa1315226}.

\leavevmode\vadjust pre{\hypertarget{ref-Gaiti2019}{}}%
Gaiti, Federico, Ronan Chaligne, Hongcang Gu, Ryan M Brand, Steven Kothen-Hill, Rafael C Schulman, Kirill Grigorev, et al. 2019. {``{Epigenetic evolution and lineage histories of chronic lymphocytic leukaemia}.''} \emph{Nature} 569 (7757): 576--80. \url{https://doi.org/10.1038/s41586-019-1198-z}.

\leavevmode\vadjust pre{\hypertarget{ref-Garnett2012}{}}%
Garnett, Mathew J., Elena J. Edelman, Sonja J. Heidorn, Chris D. Greenman, Anahita Dastur, King Wai Lau, Patricia Greninger, et al. 2012. {``{Systematic identification of genomic markers of drug sensitivity in cancer cells}.''} \emph{Nature} 483 (7391): 570--75. \url{https://doi.org/10.1038/nature11005}.

\leavevmode\vadjust pre{\hypertarget{ref-Ghia2020}{}}%
Ghia, Paolo, Andrzej Pluta, Malgorzata Wach, Daniel Lysak, Tomas Kozak, Martin Simkovic, Polina Kaplan, et al. 2020. {``{Ascend: Phase III, randomized trial of acalabrutinib versus idelalisib plus rituximab or bendamustine plus rituximab in relapsed or refractory chronic lymphocytic leukemia}.''} In \emph{Journal of Clinical Oncology}, 38:2849--61. 25. J Clin Oncol. \url{https://doi.org/10.1200/JCO.19.03355}.

\leavevmode\vadjust pre{\hypertarget{ref-Goede2014}{}}%
Goede, Valentin, Kirsten Fischer, Raymonde Busch, Anja Engelke, Barbara Eichhorst, Clemens M. Wendtner, Tatiana Chagorova, et al. 2014. {``{Obinutuzumab plus Chlorambucil in Patients with CLL and Coexisting Conditions}.''} \emph{New England Journal of Medicine} 370 (12): 1101--10. \url{https://doi.org/10.1056/nejmoa1313984}.

\leavevmode\vadjust pre{\hypertarget{ref-Gonzalez2011}{}}%
Gonzalez, David, Pilar Martinez, Rachel Wade, Sarah Hockley, David Oscier, Estella Matutes, Claire E Dearden, Sue M Richards, Daniel Catovsky, and Gareth J Morgan. 2011. {``{Mutational status of the TP53 gene as a predictor of response and survival in patients with chronic lymphocytic leukemia: Results from the LRF CLL4 trial}.''} \emph{Journal of Clinical Oncology} 29 (16): 2223--29. \url{https://doi.org/10.1200/JCO.2010.32.0838}.

\leavevmode\vadjust pre{\hypertarget{ref-Goodspeed2016}{}}%
Goodspeed, Andrew, Laura M Heiser, Joe W Gray, and James C Costello. 2016. {``{Tumor-Derived Cell Lines as Molecular Models of Cancer Pharmacogenomics.}''} \emph{Molecular Cancer Research : MCR} 14 (1): 3--13. \url{https://doi.org/10.1158/1541-7786.MCR-15-0189}.

\leavevmode\vadjust pre{\hypertarget{ref-Granziero2001}{}}%
Granziero, Luisa, Paolo Ghia, Paola Circosta, Daniela Gottardi, Giuliana Strola, Massimo Geuna, Licia Montagna, Paola Piccoli, Marco Chilosi, and Federico Caligaris-Cappio. 2001. {``{Survivin is expressed on CD40 stimulation and interfaces proliferation and apoptosis in B-cell chronic lymphocytic leukemia}.''} \emph{Blood} 97 (9): 2777--83. \url{https://doi.org/10.1182/blood.V97.9.2777}.

\leavevmode\vadjust pre{\hypertarget{ref-Grioni2021}{}}%
Grioni, Matteo, Arianna Brevi, Elena Cattaneo, Alessandra Rovida, Jessica Bordini, Maria Teresa Sabrina Bertilaccio, Maurilio Ponzoni, et al. 2021. {``{CD4+ T cells sustain aggressive chronic lymphocytic leukemia in E\(\mu\)-TCL1 mice through a CD40L-independent mechanism}.''} \emph{Blood Advances} 5 (14): 2817--28. \url{https://doi.org/10.1182/bloodadvances.2020003795}.

\leavevmode\vadjust pre{\hypertarget{ref-Guarini2008}{}}%
Guarini, Anna, Sabina Chiaretti, Simona Tavolaro, Roberta Maggio, Nadia Peragine, Franca Citarella, Maria Rosaria Ricciardi, et al. 2008. {``{BCR ligation induced by IgM stimulation results in gene expression and functional changes only in IgVH unmutated chronic lymphocytic leukemia (CLL) cells}.''} \emph{Blood} 112 (3): 782--92. \url{https://doi.org/10.1182/blood-2007-12-127688}.

\leavevmode\vadjust pre{\hypertarget{ref-Guo2017}{}}%
Guo, Ailin, Pin Lu, Greg Coffey, Pamela Conley, Anjali Pandey, and Y Lynn Wang. 2017. {``{Dual SYK/JAK inhibition overcomes ibrutinib resistance in chronic lymphocytic leukemia: Cerdulatinib, but not ibrutinib, induces apoptosis of tumor cells protected by the microenvironment.}''} \emph{Oncotarget} 8 (8): 12953--67. \url{https://doi.org/10.18632/oncotarget.14588}.

\leavevmode\vadjust pre{\hypertarget{ref-Guruharsha2012}{}}%
Guruharsha, K. G., Mark W. Kankel, and Spyros Artavanis-Tsakonas. 2012. {``{The Notch signalling system: Recent insights into the complexity of a conserved pathway}.''} Nature Publishing Group. \url{https://doi.org/10.1038/nrg3272}.

\leavevmode\vadjust pre{\hypertarget{ref-Hallek2010}{}}%
Hallek, M, K Fischer, G Fingerle-Rowson, A M Fink, R Busch, J Mayer, M Hensel, et al. 2010. {``{Addition of rituximab to fludarabine and cyclophosphamide in patients with chronic lymphocytic leukaemia: A randomised, open-label, phase 3 trial}.''} \emph{The Lancet} 376 (9747): 1164--74. \url{https://doi.org/10.1016/S0140-6736(10)61381-5}.

\leavevmode\vadjust pre{\hypertarget{ref-Hallek2008}{}}%
Hallek, Michael, Bruce D Cheson, Daniel Catovsky, Federico Caligaris-Cappio, Guillaume Dighiero, Hartmut Döhner, Peter Hillmen, et al. 2008. {``{Guidelines for the diagnosis and treatment of chronic lymphocytic leukemia: a report from the International Workshop on Chronic Lymphocytic Leukemia updating the National Cancer Institute-Working Group 1996 guidelines.}''} \emph{Blood} 111 (12): 5446--56. \url{https://doi.org/10.1182/blood-2007-06-093906}.

\leavevmode\vadjust pre{\hypertarget{ref-Hamblin1999}{}}%
Hamblin, Terry J., Zadie Davis, Anne Gardiner, David G. Oscier, and Freda K. Stevenson. 1999. {``{Unmutated Ig VH Genes Are Associated With a More Aggressive Form of Chronic Lymphocytic Leukemia}.''} \emph{Blood} 94 (6): 1848--54. \url{https://doi.org/10.1182/BLOOD.V94.6.1848}.

\leavevmode\vadjust pre{\hypertarget{ref-Hamilton2012}{}}%
Hamilton, Emma, Laurence Pearce, Liam Morgan, Sophie Robinson, Vicki Ware, Paul Brennan, N Shaun B Thomas, et al. 2012. {``{Mimicking the tumour microenvironment: Three different co-culture systems induce a similar phenotype but distinct proliferative signals in primary chronic lymphocytic leukaemia cells}.''} \emph{British Journal of Haematology} 158 (5): 589--99. \url{https://doi.org/10.1111/j.1365-2141.2012.09191.x}.

\leavevmode\vadjust pre{\hypertarget{ref-Hayden2012}{}}%
Hayden, Rachel E, Guy Pratt, Claudia Roberts, Mark T Drayson, and Chris M Bunce. 2012. {``{Treatment of chronic lymphocytic leukemia requires targeting of the protective lymph node environment with novel therapeutic approaches}.''} Leuk Lymphoma. \url{https://doi.org/10.3109/10428194.2011.610014}.

\leavevmode\vadjust pre{\hypertarget{ref-HerbstThesis}{}}%
Herbst, Sophie. 2020. {``{Systematic analysis of cell-intrinsic and extrinsic factors in chronic lymphocytic leukemia to understand functional consequences for drug response and clinical outcome}.''} PhD diss., University of Heidelberg. \url{https://archiv.ub.uni-heidelberg.de/volltextserver/27438/1/Doktorarbeit.pdf}.

\leavevmode\vadjust pre{\hypertarget{ref-Herman2011}{}}%
Herman, Sarah E. M., Amber L. Gordon, Erin Hertlein, Asha Ramanunni, Xiaoli Zhang, Samantha Jaglowski, Joseph Flynn, et al. 2011. {``{Bruton tyrosine kinase represents a promising therapeutic target for treatment of chronic lymphocytic leukemia and is effectively targeted by PCI-32765}.''} \emph{Blood} 117 (23): 6287--96. \url{https://doi.org/10.1182/blood-2011-01-328484}.

\leavevmode\vadjust pre{\hypertarget{ref-Hernandez2014}{}}%
Hernández-Lemus, Enrique. 2014. {``{Systems Biology and Integrative Omics in Breast Cancer},''} 333--52. \url{https://doi.org/10.1007/978-81-322-0843-3_17}.

\leavevmode\vadjust pre{\hypertarget{ref-Herndon2017}{}}%
Herndon, T M, S-S Chen, N S Saba, J Valdez, C Emson, M Gatmaitan, X Tian, et al. 2017. {``{Direct in vivo evidence for increased proliferation of CLL cells in lymph nodes compared to bone marrow and peripheral blood}.''} \emph{Leukemia} 31 (6): 1340--47. \url{https://doi.org/10.1038/leu.2017.11}.

\leavevmode\vadjust pre{\hypertarget{ref-Herve2005}{}}%
Hervé, Maxime, Kai Xu, Yen Shing Ng, Hedda Wardemann, Emilia Albesiano, Bradley T Messmer, Nicholas Chiorazzi, and Eric Meffre. 2005. {``{Unmutated and mutated chronic lymphocytic leukemias derive from self-reactive B cell precursors despite expressing different antibody reactivity}.''} \emph{Journal of Clinical Investigation} 115 (6): 1636--43. \url{https://doi.org/10.1172/JCI24387}.

\leavevmode\vadjust pre{\hypertarget{ref-Hillmen2015}{}}%
Hillmen, Peter, Tadeusz Robak, Ann Janssens, K Govind Babu, Janusz Kloczko, Sebastian Grosicki, Michael Doubek, et al. 2015. {``{Chlorambucil plus ofatumumab versus chlorambucil alone in previously untreated patients with chronic lymphocytic leukaemia (COMPLEMENT 1): a randomised, multicentre, open-label phase 3 trial.}''} \emph{Lancet (London, England)} 385 (9980): 1873--83. \url{https://doi.org/10.1016/S0140-6736(15)60027-7}.

\leavevmode\vadjust pre{\hypertarget{ref-Hoellenriegel2011}{}}%
Hoellenriegel, Julia, Sarah A Meadows, Mariela Sivina, William G Wierda, Hagop Kantarjian, Michael J Keating, Neill Giese, et al. 2011. {``{The phosphoinositide 3′-kinase delta inhibitor, CAL-101, inhibits B-cell receptor signaling and chemokine networks in chronic lymphocytic leukemia}.''} \emph{Blood} 118 (13): 3603--12. \url{https://doi.org/10.1182/blood-2011-05-352492}.

\leavevmode\vadjust pre{\hypertarget{ref-Huber2019}{}}%
Huber, Wolfgang, and Susan Holmes. 2019. {``{Modern Statistics for Modern Biology}.''} \url{https://www.huber.embl.de/msmb/index.html}.

\leavevmode\vadjust pre{\hypertarget{ref-Iacovelli2015}{}}%
Iacovelli, Stefano, Eva Hug, Sara Bennardo, Marcus Duehren-Von Minden, Stefania Gobessi, Andrea Rinaldi, Mirza Suljagic, et al. 2015. {``{Two types of BCR interactions are positively selected during leukemia development in the E\(\mu\)-TCL1 transgenic mouse model of CLL}.''} \emph{Blood} 125 (10): 1578--88. \url{https://doi.org/10.1182/blood-2014-07-587790}.

\leavevmode\vadjust pre{\hypertarget{ref-Iorio2016}{}}%
Iorio, Francesco, Theo A Knijnenburg, Daniel J Vis, Graham R Bignell, Michael P Menden, Michael Schubert, Nanne Aben, et al. 2016. {``{A Landscape of Pharmacogenomic Interactions in Cancer}.''} \emph{Cell} 166 (3): 740--54. \url{https://doi.org/10.1016/j.cell.2016.06.017}.

\leavevmode\vadjust pre{\hypertarget{ref-Jayappa2018}{}}%
Jayappa, Kallesh D., Konrad J. Cios, Vicki L. Gordon, Puja C. Arora, Timothy P. Bender, Michael E. Williams, Craig A. Portell, and Michael J. Weber. 2018. {``{A Multi-Drug Tolerant Phenotype Is Induced By Diverse Microenvironmental Agonists in Chronic Lymphocytic Leukemia and Mantle Cell Lymphoma}.''} \emph{Blood} 132 (Supplement 1): 4120--20. \url{https://doi.org/10.1182/blood-2018-99-119521}.

\leavevmode\vadjust pre{\hypertarget{ref-Jayappa2017}{}}%
Jayappa, Kallesh D, Craig A Portell, Vicki L Gordon, Brian J Capaldo, Stefan Bekiranov, Mark J Axelrod, L Kyle Brett, et al. 2017. {``{Microenvironmental agonists generate de novo phenotypic resistance to combined ibrutinib plus venetoclax in CLL and MCL}.''} \emph{Blood Advances} 1 (14): 933--46. \url{https://doi.org/10.1182/bloodadvances.2016004176}.

\leavevmode\vadjust pre{\hypertarget{ref-Jensen2020}{}}%
Jensen, Caleb, and Yong Teng. 2020. {``{Is It Time to Start Transitioning From 2D to 3D Cell Culture?}''} Frontiers. \url{https://doi.org/10.3389/fmolb.2020.00033}.

\leavevmode\vadjust pre{\hypertarget{ref-Kalachikov1997}{}}%
Kalachikov, S, A Migliazza, E Cayanis, N S Fracchiolla, M F Bonaldo, L Lawton, P Jelenc, et al. 1997. {``{Cloning and gene mapping of the chromosome 13q14 region deleted in chronic lymphocytic leukemia}.''} \emph{Genomics} 42 (3): 369--77. \url{https://doi.org/10.1006/geno.1997.4747}.

\leavevmode\vadjust pre{\hypertarget{ref-R-survminer}{}}%
Kassambara, Alboukadel, Marcin Kosinski, and Przemyslaw Biecek. 2021. \emph{Survminer: Drawing Survival Curves Using Ggplot2}. \url{https://rpkgs.datanovia.com/survminer/index.html}.

\leavevmode\vadjust pre{\hypertarget{ref-Kay2007}{}}%
Kay, Neil E, Tait D Shanafelt, Ann K Strege, Yean K Lee, Nancy D Bone, and Azra Raza. 2007. {``{Bone biopsy derived marrow stromal elements rescue chronic lymphocytic leukemia B-cells from spontaneous and drug induced cell death and facilitates an "angiogenic switch"}.''} \emph{Leukemia Research} 31 (7): 899--906. \url{https://doi.org/10.1016/j.leukres.2006.11.024}.

\leavevmode\vadjust pre{\hypertarget{ref-Kienle2005}{}}%
Kienle, Dirk L, Christian Korz, Beate Hosch, Axel Benner, Daniel Mertens, Annett Habermann, Alexander Kröber, et al. 2005. {``{Evidence for distinct pathomechanisms in genetic subgroups of chronic lymphocytic leukemia revealed by quantitative expression analysis of cell cycle, activation, and apoptosis-associated genes.}''} \emph{Journal of Clinical Oncology : Official Journal of the American Society of Clinical Oncology} 23 (16): 3780--92. \url{https://doi.org/10.1200/JCO.2005.02.568}.

\leavevmode\vadjust pre{\hypertarget{ref-Kipps1989}{}}%
Kipps, T. J., E. Tomhave, L. F. Pratt, S. Duffy, P. P. Chen, and D. A. Carson. 1989. {``{Developmentally restricted immunoglobulin heavy chain variable region gene expressed at high frequency in chronic lymphocytic leukemia}.''} \emph{Proceedings of the National Academy of Sciences of the United States of America} 86 (15): 5913--17. \url{https://doi.org/10.1073/pnas.86.15.5913}.

\leavevmode\vadjust pre{\hypertarget{ref-Kipps2017}{}}%
Kipps, Thomas J, Freda K Stevenson, Catherine J Wu, Carlo M Croce, Graham Packham, William G Wierda, Susan O'Brien, John Gribben, and Kanti Rai. 2017. {``{Chronic lymphocytic leukaemia.}''} \emph{Nature Reviews. Disease Primers} 3: 16096. \url{https://doi.org/10.1038/nrdp.2016.96}.

\leavevmode\vadjust pre{\hypertarget{ref-Kitada1999}{}}%
Kitada, S, J M Zapata, M Andreeff, and J C Reed. 1999. {``{Bryostatin and CD40-ligand enhance apoptosis resistance and induce expression of cell survival genes in B-cell chronic lymphocytic leukaemia.}''} \emph{British Journal of Haematology} 106 (4): 995--1004. \url{https://doi.org/10.1046/j.1365-2141.1999.01642.x}.

\leavevmode\vadjust pre{\hypertarget{ref-Klein2001}{}}%
Klein, Ulf, Yuhai Tu, Gustavo A Stolovitzky, Michela Mattioli, Giorgio Cattoretti, Hervé Husson, Arnold Freedman, et al. 2001. {``{Gene expression profiling of B cell chronic lymphocytic leukemia reveals a homogeneous phenotype related to memory B cells}.''} \emph{Journal of Experimental Medicine} 194 (11): 1625--38. \url{https://doi.org/10.1084/jem.194.11.1625}.

\leavevmode\vadjust pre{\hypertarget{ref-R-pheatmap}{}}%
Kolde, Raivo. 2019. \emph{Pheatmap: Pretty Heatmaps}. \url{https://CRAN.R-project.org/package=pheatmap}.

\leavevmode\vadjust pre{\hypertarget{ref-Kondo2010}{}}%
Kondo, Motonari. 2010. {``{Lymphoid and myeloid lineage commitment in multipotent hematopoietic progenitors}.''} \emph{Immunological Reviews} 238 (1): 37--46. \url{https://doi.org/10.1111/j.1600-065X.2010.00963.x}.

\leavevmode\vadjust pre{\hypertarget{ref-Kostareli2012}{}}%
Kostareli, E., M. Gounari, A. Janus, F. Murray, X. Brochet, V. Giudicelli, S. Pospisilova, et al. 2012. {``{Antigen receptor stereotypy across B-cell lymphoproliferations: The case of IGHV4-59/IGKV3-20 receptors with rheumatoid factor activity}.''} Leukemia. \url{https://doi.org/10.1038/leu.2011.311}.

\leavevmode\vadjust pre{\hypertarget{ref-Kraus2004}{}}%
Kraus, Manfred, Marat B. Alimzhanov, Nikolaus Rajewsky, and Klaus Rajewsky. 2004. {``{Survival of resting mature B lymphocytes depends on BCR signaling via the Ig\(\alpha\)/\(\beta\) heterodimer}.''} \emph{Cell} 117 (6): 787--800. \url{https://doi.org/10.1016/j.cell.2004.05.014}.

\leavevmode\vadjust pre{\hypertarget{ref-Krysov2010}{}}%
Krysov, Sergey, Kathleen N. Potter, C. Ian Mockridge, Vania Coelho, Isla Wheatley, Graham Packham, and Freda K. Stevenson. 2010. {``{Surface IgM of CLL cells displays unusual glycans indicative of engagement of antigen in vivo}.''} \emph{Blood} 115 (21): 4198--4205. \url{https://doi.org/10.1182/blood-2009-12-254847}.

\leavevmode\vadjust pre{\hypertarget{ref-HOCOMOCO}{}}%
Kulakovskiy, Ivan V, Ilya E Vorontsov, Ivan S Yevshin, Anastasiia V Soboleva, Artem S Kasianov, Haitham Ashoor, Wail Ba-Alawi, et al. 2016. {``{HOCOMOCO: expansion and enhancement of the collection of transcription factor binding sites models.}''} \emph{Nucleic Acids Research} 44 (D1): D116--25. \url{https://doi.org/10.1093/nar/gkv1249}.

\leavevmode\vadjust pre{\hypertarget{ref-Kulis2012}{}}%
Kulis, Marta, Simon Heath, Marina Bibikova, Ana C Queirós, Alba Navarro, Guillem Clot, Alejandra Martínez-Trillos, et al. 2012. {``{Epigenomic analysis detects widespread gene-body DNA hypomethylation in chronic lymphocytic leukemia}.''} \emph{Nature Genetics} 44 (11): 1236--42. \url{https://doi.org/10.1038/ng.2443}.

\leavevmode\vadjust pre{\hypertarget{ref-Kumar}{}}%
Kumar, Dinesh. n.d. {``{A Complete understanding of LASSO Regression}.''} Accessed October 3, 2021. \url{https://www.mygreatlearning.com/blog/understanding-of-lasso-regression/\#introduction}.

\leavevmode\vadjust pre{\hypertarget{ref-Kurtova2009}{}}%
Kurtova, Antonina V., Kumudha Balakrishnan, Rong Chen, Wei Ding, Susanne Schnabl, Maite P. Quiroga, Mariela Sivina, et al. 2009. {``{Diverse marrow stromal cells protect CLL cells from spontaneous and drug-induced apoptosis: development of a reliable and reproducible system to assess stromal cell adhesion-mediated drug resistance}.''} \emph{Blood} 114 (20): 4441--50. \url{https://doi.org/10.1182/blood-2009-07-233718}.

\leavevmode\vadjust pre{\hypertarget{ref-Lagneaux1998}{}}%
Lagneaux, L., A. Delforge, D. Bron, C. De Bruyn, and P. Stryckmans. 1998. {``{Chronic Lymphocytic Leukemic B Cells But Not Normal B Cells Are Rescued From Apoptosis by Contact With Normal Bone Marrow Stromal Cells}.''} \emph{Blood} 91 (7): 2387--96. \url{https://doi.org/10.1182/BLOOD.V91.7.2387}.

\leavevmode\vadjust pre{\hypertarget{ref-Landau2013}{}}%
Landau, Dan A, Scott L Carter, Petar Stojanov, Aaron McKenna, Kristen Stevenson, Michael S Lawrence, Carrie Sougnez, et al. 2013. {``{Evolution and impact of subclonal mutations in chronic lymphocytic leukemia.}''} \emph{Cell} 152 (4): 714--26. \url{https://doi.org/10.1016/j.cell.2013.01.019}.

\leavevmode\vadjust pre{\hypertarget{ref-Landau2014}{}}%
Landau, Dan A, Kendell Clement, Michael J Ziller, Patrick Boyle, Jean Fan, Hongcang Gu, Kristen Stevenson, et al. 2014. {``{Locally Disordered Methylation Forms the Basis of Intratumor Methylome Variation in Chronic Lymphocytic Leukemia}.''} \emph{Cancer Cell} 26 (6): 813--25. \url{https://doi.org/10.1016/j.ccell.2014.10.012}.

\leavevmode\vadjust pre{\hypertarget{ref-Landau2015}{}}%
Landau, Dan A, Eugen Tausch, Amaro N Taylor-Weiner, Chip Stewart, Johannes G Reiter, Jasmin Bahlo, Sandra Kluth, et al. 2015. {``{Mutations driving CLL and their evolution in progression and relapse}.''} \emph{Nature} 526 (7574): 525--30. \url{https://doi.org/10.1038/nature15395}.

\leavevmode\vadjust pre{\hypertarget{ref-Lebien2008}{}}%
Lebien, Tucker W., and Thomas F Tedder. 2008. {``{B lymphocytes: How they develop and function}.''} \emph{Blood} 112 (5): 1570--80. \url{https://doi.org/10.1182/blood-2008-02-078071}.

\leavevmode\vadjust pre{\hypertarget{ref-Letai2017}{}}%
Letai, Anthony. 2017. {``{Functional precision cancer medicine-moving beyond pure genomics.}''} \emph{Nature Medicine} 23 (9): 1028--35. \url{https://doi.org/10.1038/nm.4389}.

\leavevmode\vadjust pre{\hypertarget{ref-Li2015}{}}%
Li, Yang, Yu Wang, Zhuoyue Wang, Danhui Yi, and Shuangge Ma. 2015. {``{Racial differences in three major NHL subtypes: Descriptive epidemiology}.''} \emph{Cancer Epidemiology} 39 (1): 8--13. \url{https://doi.org/10.1016/j.canep.2014.12.001}.

\leavevmode\vadjust pre{\hypertarget{ref-Lipsky2020}{}}%
Lipsky, Andrew, Danny Luan, Shirley Chen, Ronan Chaligne, Neville Dusaj, Erica B Bhavsar, Chelston Ang, et al. 2020. {``{Single-Cell Multi-Omics Reveals Distinct Paths to Survival of Admixed BTKC481 Mutant Vs. Wild-Type Cells in Clinically Progressing Chronic Lymphocytic Leukemia}.''} \emph{Blood} 136 (Supplement 1): 40--42. \url{https://doi.org/10.1182/blood-2020-138374}.

\leavevmode\vadjust pre{\hypertarget{ref-Lotz1994}{}}%
Lotz, M, E Ranheim, and T J Kipps. 1994. {``{Transforming growth factor beta as endogenous growth inhibitor of chronic lymphocytic leukemia B cells.}''} \emph{Journal of Experimental Medicine} 179 (3): 999--1004. \url{https://doi.org/10.1084/jem.179.3.999}.

\leavevmode\vadjust pre{\hypertarget{ref-R-DESeq2}{}}%
Love, Michael, Simon Anders, and Wolfgang Huber. 2021. \emph{DESeq2: Differential Gene Expression Analysis Based on the Negative Binomial Distribution}. \url{https://github.com/mikelove/DESeq2}.

\leavevmode\vadjust pre{\hypertarget{ref-Lu2004}{}}%
Lu, Desheng, Yandong Zhao, Rommel Tawatao, Howard B Cottam, Malini Sen, Lorenzo M Leoni, Thomas J Kipps, Maripat Corr, and Dennis A Carson. 2004. {``{Activation of the Wnt signaling pathway in chronic lymphocytic leukemia}.''} \emph{Proceedings of the National Academy of Sciences of the United States of America} 101 (9): 3118--23. \url{https://doi.org/10.1073/pnas.0308648100}.

\leavevmode\vadjust pre{\hypertarget{ref-Lu2021}{}}%
Lu, Junyan, Ester Cannizzaro, Fabienne Meier-Abt, Sebastian Scheinost, Peter-Martin Bruch, Holly Ar Giles, Almut Lütge, et al. 2021. {``{Multi-omics reveals clinically relevant proliferative drive associated with mTOR-MYC-OXPHOS activity in chronic lymphocytic leukemia.}''} \emph{Nature Cancer} 2 (8): 853--64. \url{https://doi.org/10.1038/s43018-021-00216-6}.

\leavevmode\vadjust pre{\hypertarget{ref-Lukas2020}{}}%
Lukas, Marina, Britta Velten, Leopold Sellner, Katarzyna Tomska, Jennifer Hüellein, Tatjana Walther, Lena Wagner, et al. 2020. {``{Survey of ex vivo drug combination effects in chronic lymphocytic leukemia reveals synergistic drug effects and genetic dependencies}.''} \emph{Leukemia} 34 (11): 2934--50. \url{https://doi.org/10.1038/s41375-020-0846-5}.

\leavevmode\vadjust pre{\hypertarget{ref-Lukenbill2013}{}}%
Lukenbill, Joshua, and Matt Kalaycio. 2013. {``{Fludarabine: A review of the clear benefits and potential harms}.''} Leuk Res. \url{https://doi.org/10.1016/j.leukres.2013.05.004}.

\leavevmode\vadjust pre{\hypertarget{ref-Mallm2019}{}}%
Mallm, Jan-Philipp, Murat Iskar, Naveed Ishaque, Lara C Klett, Sabrina J Kugler, Jose M Muino, Vladimir B Teif, et al. 2019. {``{Linking aberrant chromatin features in chronic lymphocytic leukemia to transcription factor networks.}''} \emph{Molecular Systems Biology} 15 (5): e8339. \url{https://doi.org/10.15252/msb.20188339}.

\leavevmode\vadjust pre{\hypertarget{ref-Mangolini2018}{}}%
Mangolini, Maurizio, Frederik Götte, Andrew Moore, Tim Ammon, Madlen Oelsner, Gloria Lutzny-Geier, Ludger Klein-Hitpass, et al. 2018. {``{Notch2 controls non-autonomous Wnt-signalling in chronic lymphocytic leukaemia}.''} \emph{Nature Communications} 9 (1): 3839. \url{https://doi.org/10.1038/s41467-018-06069-5}.

\leavevmode\vadjust pre{\hypertarget{ref-Mansouri2016}{}}%
Mansouri, Larry, Nikos Papakonstantinou, Stavroula Ntoufa, Kostas Stamatopoulos, and Richard Rosenquist. 2016. {``{NF-\(\kappa\)B activation in chronic lymphocytic leukemia: A point of convergence of external triggers and intrinsic lesions}.''} Semin Cancer Biol. \url{https://doi.org/10.1016/j.semcancer.2016.07.005}.

\leavevmode\vadjust pre{\hypertarget{ref-Martensson2010}{}}%
Mårtensson, Inga Lill, Nina Almqvist, Ola Grimsholm, and Angelina I. Bernardi. 2010. {``{The pre-B cell receptor checkpoint}.''} FEBS Lett. \url{https://doi.org/10.1016/j.febslet.2010.04.057}.

\leavevmode\vadjust pre{\hypertarget{ref-Martinez2016}{}}%
Martínez-Trillos, Alejandra, Alba Navarro, Marta Aymerich, Julio Delgado, Armando López-Guillermo, Elias Campo, and Neus Villamor. 2016. {``{Clinical impact of MYD88 mutations in chronic lymphocytic leukemia}.''} Blood. \url{https://doi.org/10.1182/blood-2015-10-678490}.

\leavevmode\vadjust pre{\hypertarget{ref-Matutes1994}{}}%
Matutes, E., K. Owusu-Ankomah, R. Morilla, J. G. García Marco, A. Houlihan, T. Que, and D. Catovsky. 1994. {``{The immunological profile of B-cell disorders and proposal of a scoring system for the diagnosis of CLL.}''} \emph{Undefined}. \url{https://www.semanticscholar.org/paper/The-immunological-profile-of-B-cell-disorders-and-a-Matutes-Owusu-Ankomah/41ab989ce5c271aa59bd588e88919f4bea23dce6}.

\leavevmode\vadjust pre{\hypertarget{ref-McWilliams2019}{}}%
McWilliams, Emily M, Christopher R Lucas, Timothy Chen, Bonnie K Harrington, Ronni Wasmuth, Amanda Campbell, Kerry A Rogers, et al. 2019. {``{Anti--BAFF-R antibody VAY-736 demonstrates promising preclinical activity in CLL and enhances effectiveness of ibrutinib}.''} \emph{Blood Advances} 3 (3): 447--60. \url{https://doi.org/10.1182/bloodadvances.2018025684}.

\leavevmode\vadjust pre{\hypertarget{ref-MeierAbt2021}{}}%
Meier-Abt, Fabienne, Junyan Lu, Ester Cannizzaro, Marcel F Pohly, Sandra Kummer, Sibylle Pfammatter, Laura Kunz, et al. 2021. {``{The Protein Landscape of Chronic Lymphocytic Leukemia (CLL).}''} \emph{Blood}, June. \url{https://doi.org/10.1182/blood.2020009741}.

\leavevmode\vadjust pre{\hypertarget{ref-Menyhart2021}{}}%
Menyhárt, Otília, and Balázs Győrffy. 2021. {``{Multi-omics approaches in cancer research with applications in tumor subtyping, prognosis, and diagnosis}.''} Comput Struct Biotechnol J. \url{https://doi.org/10.1016/j.csbj.2021.01.009}.

\leavevmode\vadjust pre{\hypertarget{ref-Messmer2004}{}}%
Messmer, Bradley T, Emilia Albesiano, Dimitar G Efremov, Fabio Ghiotto, Steven L Allen, Jonathan Kolitz, Robin Foa, et al. 2004. {``{Multiple distinct sets of stereotyped antigen receptors indicate a role for antigen in promoting chronic lymphocytic leukemia.}''} \emph{The Journal of Experimental Medicine} 200 (4): 519--25. \url{https://doi.org/10.1084/jem.20040544}.

\leavevmode\vadjust pre{\hypertarget{ref-Messmer2005}{}}%
Messmer, Bradley T, Davorka Messmer, Steven L Allen, Jonathan E Kolitz, Prasad Kudalkar, Denise Cesar, Elizabeth J Murphy, et al. 2005. {``{In vivo measurements document the dynamic cellular kinetics of chronic lymphocytic leukemia B cells.}''} \emph{The Journal of Clinical Investigation} 115 (3): 755--64. \url{https://doi.org/10.1172/JCI23409}.

\leavevmode\vadjust pre{\hypertarget{ref-Messmer2011}{}}%
Messmer, Davorka, Jessie-F Fecteau, Morgan O'hayre, Ila S Bharati, Tracy M Handel, and Thomas J Kipps. 2011. {``{Chronic lymphocytic leukemia cells receive RAF-dependent survival signals in response to CXCL12 that are sensitive to inhibition by sorafenib}.''} \url{https://doi.org/10.1182/blood-2010}.

\leavevmode\vadjust pre{\hypertarget{ref-Migliazza2001}{}}%
Migliazza, Anna, Francesc Bosch, Hirokazu Komatsu, Eftihia Cayanis, Stefano Martinotti, Elena Toniato, Ernesto Guccione, et al. 2001. {``{Nucleotide sequence, transcription map, and mutation analysis of the 13q14 chromosomal region deleted in B-cell chronic lymphocytic leukemia}.''} \emph{Blood} 97 (7): 2098--2104. \url{https://doi.org/10.1182/blood.V97.7.2098}.

\leavevmode\vadjust pre{\hypertarget{ref-Miller2019}{}}%
Miller, Kimberly D., Leticia Nogueira, Angela B. Mariotto, Julia H. Rowland, K. Robin Yabroff, Catherine M. Alfano, Ahmedin Jemal, Joan L. Kramer, and Rebecca L. Siegel. 2019. {``{Cancer treatment and survivorship statistics, 2019}.''} \emph{CA: A Cancer Journal for Clinicians} 69 (5): 363--85. \url{https://doi.org/10.3322/caac.21565}.

\leavevmode\vadjust pre{\hypertarget{ref-Mittal2014}{}}%
Mittal, Amit K., Nagendra K. Chaturvedi, Karan J. Rai, Christine E. Gilling-Cutucache, Tara M. Nordgren, Margaret Moragues, Runqing Lu, et al. 2014. {``{Chronic Lymphocytic Leukemia Cells in a Lymph Node Microenvironment Depict Molecular Signature Associated with an Aggressive Disease}.''} \emph{Molecular Medicine} 20 (1): 290--301. \url{https://doi.org/10.2119/molmed.2012.00303}.

\leavevmode\vadjust pre{\hypertarget{ref-Moore2007}{}}%
Moore, Victoria Del Gaizo, Jennifer R Brown, Michael Certo, Tara M Love, Carl D Novina, and Anthony Letai. 2007. {``{Chronic lymphocytic leukemia requires BCL2 to sequester prodeath BIM, explaining sensitivity to BCL2 antagonist ABT-737}.''} \emph{Journal of Clinical Investigation} 117 (1): 112--21. \url{https://doi.org/10.1172/JCI28281}.

\leavevmode\vadjust pre{\hypertarget{ref-Moreau1997}{}}%
Moreau, Elisabeth J, Estella Matutes, Roger P A'Hern, Alison M Morilla, Ricardo M Morilla, Kwasi A Owusu-Ankomah, Ben K Seon, and Daniel Catovsky. 1997. {``{Improvement of the chronic lymphocytic leukemia scoring system with the monoclonal antibody SN8 (CD79b)}.''} \emph{American Journal of Clinical Pathology} 108 (4): 378--82. \url{https://doi.org/10.1093/ajcp/108.4.378}.

\leavevmode\vadjust pre{\hypertarget{ref-Moreton2005}{}}%
Moreton, Paul, Ben Kennedy, Guy Lucas, Michael Leach, Saad M. B. Rassam, Andrew Haynes, Jane Tighe, et al. 2005. {``{Eradication of minimal residual disease in B-cell chronic lymphocytic leukemia after alemtuzumab therapy is associated with prolonged survival}.''} \emph{Journal of Clinical Oncology} 23 (13): 2971--79. \url{https://doi.org/10.1200/JCO.2005.04.021}.

\leavevmode\vadjust pre{\hypertarget{ref-Mraz2011}{}}%
Mraz, Marek, Clive S Zent, Amy K Church, Diane F Jelinek, Xiaosheng Wu, Sarka Pospisilova, Stephen M Ansell, et al. 2011. {``{Bone marrow stromal cells protect lymphoma B-cells from rituximab-induced apoptosis and targeting integrin \(\alpha\)-4-\(\beta\)-1 (VLA-4) with natalizumab can overcome this resistance}.''} \emph{British Journal of Haematology} 155 (1): 53--64. \url{https://doi.org/10.1111/j.1365-2141.2011.08794.x}.

\leavevmode\vadjust pre{\hypertarget{ref-Muzio2009}{}}%
Muzio, Marta, Cristina Scielzo, Maria T. S. Bertilaccio, Michela Frenquelli, Paolo Ghia, and Federico Caligaris-Cappio. 2009. {``{Expression and function of toll like receptors in chronic lymphocytic leukaemia cells}.''} \emph{British Journal of Haematology} 144 (4): 507--16. \url{https://doi.org/10.1111/j.1365-2141.2008.07475.x}.

\leavevmode\vadjust pre{\hypertarget{ref-Lanemo2008}{}}%
Myhrinder, Anna Lanemo, Eva Hellqvist, Ekaterina Sidorova, Anita Söderberg, Helen Baxendale, Charlotte Dahle, Kerstin Willander, et al. 2008. {``{A new perspective: Molecular motifs on oxidized LDL, apoptotic cells, and bacteria are targets for chronic lymphocytic leukemia antibodies}.''} \emph{Blood} 111 (7): 3838--48. \url{https://doi.org/10.1182/blood-2007-11-125450}.

\leavevmode\vadjust pre{\hypertarget{ref-Nabhan2014}{}}%
Nabhan, Chadi, Briseis Aschebrook-Kilfoy, Brian C-H Chiu, Sonali M Smith, Tait D Shanafelt, Andrew M Evens, and Neil E Kay. 2014. {``{The impact of race, ethnicity, age and sex on clinical outcome in chronic lymphocytic leukemia: A comprehensive Surveillance, Epidemiology, and End Results analysis in the modern era}.''} \emph{Leukemia and Lymphoma} 55 (12): 2778--84. \url{https://doi.org/10.3109/10428194.2014.898758}.

\leavevmode\vadjust pre{\hypertarget{ref-Nadeu2016}{}}%
Nadeu, Ferran, Julio Delgado, Cristina Royo, Tycho Baumann, Tatjana Stankovic, Magda Pinyol, Pedro Jares, et al. 2016. {``{Clinical impact of clonal and subclonal TP53, SF3B1, BIRC3, NOTCH1, and ATM mutations in chronic lymphocytic leukemia.}''} \emph{Blood} 127 (17): 2122--30. \url{https://doi.org/10.1182/blood-2015-07-659144}.

\leavevmode\vadjust pre{\hypertarget{ref-Nelder1972}{}}%
Nelder, J. A., and R. W. M. Wedderburn. 1972. {``{Generalized Linear Models}.''} \emph{Journal of the Royal Statistical Society. Series A (General)} 135 (3): 370. \url{https://doi.org/10.2307/2344614}.

\leavevmode\vadjust pre{\hypertarget{ref-Nishio2005}{}}%
Nishio, Mitsufumi, Tomoyuki Endo, Nobuhiro Tsukada, Junko Ohata, Shinichi Kitada, John C Reed, Nathan J Zvaifler, and Thomas J Kipps. 2005. {``{Nurselike cells express BAFF and APRIL, which can promote survival of chronic lymphocytic leukemia cells via a paracrine pathway distinct from that of SDF-1alpha.}''} \emph{Blood} 106 (3): 1012--20. \url{https://doi.org/10.1182/blood-2004-03-0889}.

\leavevmode\vadjust pre{\hypertarget{ref-Nutt2015}{}}%
Nutt, Stephen L., Philip D. Hodgkin, David M. Tarlinton, and Lynn M. Corcoran. 2015. {``{The generation of antibody-secreting plasma cells}.''} \emph{Nature Reviews Immunology 2015 15:3} 15 (3): 160--71. \url{https://doi.org/10.1038/nri3795}.

\leavevmode\vadjust pre{\hypertarget{ref-OBrien2011}{}}%
O'Brien, Susan, and Neil E Kay. 2011. {``{Maintenance therapy for B-Chronic lymphocytic leukemia}.''} \emph{Clinical Advances in Hematology and Oncology} 9 (1): 22--31. \url{http://www.hematologyandoncology.net/files/2013/07/ho0111_obrien1.pdf}.

\leavevmode\vadjust pre{\hypertarget{ref-OHayre2010}{}}%
O'Hayre, Morgan, Catherina L. Salanga, Thomas J. Kipps, Davorka Messmer, Pieter C. Dorrestein, and Tracy M. Handel. 2010. {``{Elucidating the CXCL12/CXCR4 signaling network in chronic lymphocytic leukemia through phosphoproteomics analysis}.''} Edited by Syed A. Aziz. \emph{PLoS ONE} 5 (7): e11716. \url{https://doi.org/10.1371/journal.pone.0011716}.

\leavevmode\vadjust pre{\hypertarget{ref-Oakes2016}{}}%
Oakes, Christopher C, Marc Seifert, Yassen Assenov, Lei Gu, Martina Przekopowitz, Amy S Ruppert, Qi Wang, et al. 2016. {``{DNA methylation dynamics during B cell maturation underlie a continuum of disease phenotypes in chronic lymphocytic leukemia}.''} \emph{Nature Genetics} 48 (3): 253--64. \url{https://doi.org/10.1038/ng.3488}.

\leavevmode\vadjust pre{\hypertarget{ref-R-BloodCancerMultiOmics2017}{}}%
Oles, Malgorzata, Sascha Dietrich, Junyan Lu, Britta Velten, Andreas Mock, Vladislav Kim, and Wolfgang Huber. 2021. \emph{BloodCancerMultiOmics2017: "Drug-Perturbation-Based Stratification of Blood Cancer" by Dietrich s, Oles m, Lu j Et Al. - Experimental Data and Complete Analysis}.

\leavevmode\vadjust pre{\hypertarget{ref-Oppezzo2013}{}}%
Oppezzo, P, and G Dighiero. 2013. {``{``Role of the B-cell receptor and the microenvironment in chronic lymphocytic leukemia''}.''} \emph{Blood Cancer Journal} 3 (9): e149--49. \url{https://doi.org/10.1038/bcj.2013.45}.

\leavevmode\vadjust pre{\hypertarget{ref-Os2013}{}}%
Os, Audun, Simone Bürgler, Anna Parente Ribes, Ane Funderud, Dong Wang, Keith M. Thompson, Geir E. Tjønnfjord, Bjarne Bogen, and Ludvig A. Munthe. 2013. {``{Chronic Lymphocytic Leukemia Cells Are Activated and Proliferate in Response to Specific T Helper Cells}.''} \emph{Cell Reports} 4 (3): 566--77. \url{https://doi.org/10.1016/J.CELREP.2013.07.011}.

\leavevmode\vadjust pre{\hypertarget{ref-Palamarchuk2010}{}}%
Palamarchuk, Alexey, Alexey Efanov, Natalya Nazaryan, Urmila Santanam, Hansjuerg Alder, Laura Rassenti, Thomas Kipps, Carlo M Croce, and Yuri Pekarsky. 2010. {``{13q14 deletions in CLL involve cooperating tumor suppressors}.''} \emph{Blood} 115 (19): 3916--22. \url{https://doi.org/10.1182/blood-2009-10-249367}.

\leavevmode\vadjust pre{\hypertarget{ref-Palma2017}{}}%
Palma, Marzia, Giusy Gentilcore, Kia Heimersson, Fariba Mozaffari, Barbro Näsman-Glaser, Emma Young, Richard Rosenquist, Lotta Hansson, Anders Österborg, and Håkan Mellstedt. 2017. {``{T cells in chronic lymphocytic leukemia display dysregulated expression of immune checkpoints and activation markers}.''} \emph{Haematologica} 102 (3): 562--72. \url{https://doi.org/10.3324/haematol.2016.151100}.

\leavevmode\vadjust pre{\hypertarget{ref-Panayiotidis1996}{}}%
Panayiotidis, P., D. Jones, K. Ganeshaguru, L. Foroni, and A. V. Hoffbrand. 1996. {``{Human bone marrow stromal cells prevent apoptosis and support the survival of chronic lymphocytic leukaemia cells in vitro}.''} \emph{British Journal of Haematology} 92 (1): 97--103. \url{https://doi.org/10.1046/j.1365-2141.1996.00305.x}.

\leavevmode\vadjust pre{\hypertarget{ref-Pascutti2013}{}}%
Pascutti, Maria Fernanda, Margot Jak, Jacqueline M. Tromp, Ingrid A. M. Derks, Ester B. M. Remmerswaal, Rachel Thijssen, Martijn H. A. van Attekum, et al. 2013. {``{IL-21 and CD40L signals from autologous T cells can induce antigen-independent proliferation of CLL cells}.''} \emph{Blood} 122 (17): 3010--19. \url{https://doi.org/10.1182/blood-2012-11-467670}.

\leavevmode\vadjust pre{\hypertarget{ref-Pedersen2002}{}}%
Pedersen, Irene M., Shinichi Kitada, Lorenzo M. Leoni, Juan M. Zapata, James G. Karras, Nobuhiro Tsukada, Thomas J. Kipps, Yong Sung Choi, Frank Bennett, and John C. Reed. 2002. {``{Protection of CLL B cells by a follicular dendritic cell line is dependent on induction of Mcl-1}.''} \emph{Blood} 100 (5): 1795--1801. \url{https://doi.org/10.1182/BLOOD.V100.5.1795.H81702001795_1795_1801}.

\leavevmode\vadjust pre{\hypertarget{ref-Pelanda2012}{}}%
Pelanda, Roberta, and Raul M. Torres. 2012. {``{Central B-Cell tolerance: Where selection begins}.''} \emph{Cold Spring Harbor Perspectives in Biology} 4 (4). \url{https://doi.org/10.1101/cshperspect.a007146}.

\leavevmode\vadjust pre{\hypertarget{ref-Pemovska2013}{}}%
Pemovska, Tea, Mika Kontro, Bhagwan Yadav, Henrik Edgren, Samuli Eldfors, Agnieszka Szwajda, Henrikki Almusa, et al. 2013. {``{Individualized systems medicine strategy to tailor treatments for patients with chemorefractory acute myeloid leukemia}.''} \emph{Cancer Discovery} 3 (12): 1416--29. \url{https://doi.org/10.1158/2159-8290.CD-13-0350}.

\leavevmode\vadjust pre{\hypertarget{ref-Puente2015}{}}%
Puente, Xose S, Silvia Beà, Rafael Valdés-Mas, Neus Villamor, Jesús Gutiérrez-Abril, José I Martín-Subero, Marta Munar, et al. 2015. {``{Non-coding recurrent mutations in chronic lymphocytic leukaemia.}''} \emph{Nature} 526 (7574): 519--24. \url{https://doi.org/10.1038/nature14666}.

\leavevmode\vadjust pre{\hypertarget{ref-Puente2011}{}}%
Puente, Xose S., Magda Pinyol, Victor Quesada, Laura Conde, Gonzalo R. Ordóñez, Neus Villamor, Georgia Escaramis, et al. 2011. {``{Whole-genome sequencing identifies recurrent mutations in chronic lymphocytic leukaemia}.''} \emph{Nature} 475 (7354): 101--5. \url{https://doi.org/10.1038/nature10113}.

\leavevmode\vadjust pre{\hypertarget{ref-Pulte2015}{}}%
Pulte, Dianne, Maria Theresa Redaniel, Jenny Bird, and Mona Jeffreys. 2015. {``{Survival for patients with chronic leukemias in the US and Britain: Age-related disparities and changes in the early 21st century.}''} \emph{European Journal of Haematology} 94 (6): 540--45. \url{https://doi.org/10.1111/ejh.12468}.

\leavevmode\vadjust pre{\hypertarget{ref-Purroy2015}{}}%
Purroy, Noelia, Pau Abrisqueta, Júlia Carabia, Cecilia Carpio, Carles Palacio, Francesc Bosch, and Marta Crespo. 2015. {``{Co-culture of primary CLL cells with bone marrow mesenchymal cells, CD40 ligand and CpG ODN promotes proliferation of chemoresistant CLL cells phenotypically comparable to those proliferating in vivo}.''} \emph{Oncotarget} 6 (10): 7632--43. \url{https://doi.org/10.18632/oncotarget.2939}.

\leavevmode\vadjust pre{\hypertarget{ref-Quesada2012}{}}%
Quesada, Víctor, Laura Conde, Neus Villamor, Gonzalo R. Ordóñez, Pedro Jares, Laia Bassaganyas, Andrew J. Ramsay, et al. 2012. {``{Exome sequencing identifies recurrent mutations of the splicing factor SF3B1 gene in chronic lymphocytic leukemia}.''} \emph{Nature Genetics} 44 (1): 47--52. \url{https://doi.org/10.1038/ng.1032}.

\leavevmode\vadjust pre{\hypertarget{ref-R-base}{}}%
R Core Team. 2021. \emph{R: A Language and Environment for Statistical Computing}. Vienna, Austria: R Foundation for Statistical Computing. \url{https://www.R-project.org/}.

\leavevmode\vadjust pre{\hypertarget{ref-Rendeiro2016}{}}%
Rendeiro, André F., Christian Schmidl, Jonathan C. Strefford, Renata Walewska, Zadie Davis, Matthias Farlik, David Oscier, and Christoph Bock. 2016. {``{Chromatin accessibility maps of chronic lymphocytic leukaemia identify subtype-specific epigenome signatures and transcription regulatory networks}.''} \emph{Nature Communications} 7 (1): 11938. \url{https://doi.org/10.1038/ncomms11938}.

\leavevmode\vadjust pre{\hypertarget{ref-Robak2005}{}}%
Robak, Tadeusz. 2005. {``{Therapy of chronic lymphocytic leukaemia with purine nucleoside analogues: Facts and controversies}.''} Drugs Aging. \url{https://doi.org/10.2165/00002512-200522120-00002}.

\leavevmode\vadjust pre{\hypertarget{ref-Robak2010}{}}%
Robak, Tadeusz, Anna Dmoszynska, Philippe Solal-Céligny, Krzysztof Warzocha, Javier Loscertales, John Catalano, Boris V Afanasiev, et al. 2010. {``{Rituximab plus fludarabine and cyclophosphamide prolongs progression-free survival compared with fludarabine and cyclophosphamide alone in previously treated chronic lymphocytic leukemia}.''} \emph{Journal of Clinical Oncology} 28 (10): 1756--65. \url{https://doi.org/10.1200/JCO.2009.26.4556}.

\leavevmode\vadjust pre{\hypertarget{ref-Roberts2016}{}}%
Roberts, Andrew W., Matthew S. Davids, John M. Pagel, Brad S. Kahl, Soham D. Puvvada, John F. Gerecitano, Thomas J. Kipps, et al. 2016. {``{Targeting BCL2 with Venetoclax in Relapsed Chronic Lymphocytic Leukemia}.''} \emph{New England Journal of Medicine} 374 (4): 311--22. \url{https://doi.org/10.1056/nejmoa1513257}.

\leavevmode\vadjust pre{\hypertarget{ref-Roessner2020}{}}%
Roessner, Philipp M, and Martina Seiffert. 2020. {``{T-cells in chronic lymphocytic leukemia: Guardians or drivers of disease?}''} \emph{Leukemia} 34 (8): 2012--24. \url{https://doi.org/10.1038/s41375-020-0873-2}.

\leavevmode\vadjust pre{\hypertarget{ref-Rosenwald2001}{}}%
Rosenwald, Andreas, Ash A Alizadeh, George Widhopf, Richard Simon, R Eric Davis, Xin Yu, Liming Yang, et al. 2001. {``{Relation of gene expression phenotype to immunoglobulin mutation genotype in B cell chronic lymphocytic leukemia}.''} \emph{Journal of Experimental Medicine} 194 (11): 1639--47. \url{https://doi.org/10.1084/jem.194.11.1639}.

\leavevmode\vadjust pre{\hypertarget{ref-Rossi2009}{}}%
Rossi, Davide, Michaela Cerri, Clara Deambrogi, Elisa Sozzi, Stefania Cresta, Silvia Rasi, Lorenzo De Paoli, et al. 2009. {``{The prognostic value of TP53 mutations in chronic lymphocytic leukemia is independent of Del17p13: Implications for overall survival and chemorefractoriness}.''} \emph{Clinical Cancer Research} 15 (3): 995--1004. \url{https://doi.org/10.1158/1078-0432.CCR-08-1630}.

\leavevmode\vadjust pre{\hypertarget{ref-Rossi2012}{}}%
Rossi, Davide, Silvia Rasi, Giulia Fabbri, Valeria Spina, Marco Fangazio, Francesco Forconi, Roberto Marasca, et al. 2012. {``{Mutations of NOTCH1 are an independent predictor of survival in chronic lymphocytic leukemia}.''} \emph{Blood} 119 (2): 521--29. \url{https://doi.org/10.1182/blood-2011-09-379966}.

\leavevmode\vadjust pre{\hypertarget{ref-Rossi2013}{}}%
Rossi, Davide, Silvia Rasi, Valeria Spina, Alessio Bruscaggin, Sara Monti, Carmela Ciardullo, Clara Deambrogi, et al. 2013. {``{Integrated mutational and cytogenetic analysis identifies new prognostic subgroups in chronic lymphocytic leukemia}.''} \emph{Blood} 121 (8): 1403--12. \url{https://doi.org/10.1182/blood-2012-09-458265}.

\leavevmode\vadjust pre{\hypertarget{ref-Scheffold2020}{}}%
Scheffold, Annika, and Stephan Stilgenbauer. 2020. {``{Revolution of Chronic Lymphocytic Leukemia Therapy: the Chemo-Free Treatment Paradigm}.''} Springer. \url{https://doi.org/10.1007/s11912-020-0881-4}.

\leavevmode\vadjust pre{\hypertarget{ref-Schulz2011}{}}%
Schulz, A., G. Toedt, T. Zenz, S. Stilgenbauer, P. Lichter, and M. Seiffert. 2011. {``{Inflammatory cytokines and signaling pathways are associated with survival of primary chronic lymphocytic leukemia cells in vitro: a dominant role of CCL2}.''} \emph{Haematologica} 96 (3): 408--16. \url{https://doi.org/10.3324/haematol.2010.031377}.

\leavevmode\vadjust pre{\hypertarget{ref-Scielzo2020}{}}%
Scielzo, Cristina, and Paolo Ghia. 2020. {``{Modeling the Leukemia Microenviroment In Vitro}.''} Frontiers Media SA. \url{https://doi.org/10.3389/fonc.2020.607608}.

\leavevmode\vadjust pre{\hypertarget{ref-Scrivener2003}{}}%
Scrivener, S., R. V. Goddard, E. R. Kaminski, and A. G. Prentice. 2003. {``{Abnormal T-cell Function in B-cell Chronic Lymphocytic Leukaemia}.''} \emph{Leukemia \& Lymphoma} 44 (3): 383--89. \url{https://doi.org/10.1080/1042819021000029993}.

\leavevmode\vadjust pre{\hypertarget{ref-Shiloh2013}{}}%
Shiloh, Yosef, and Yael Ziv. 2013. {``{The ATM protein kinase: Regulating the cellular response to genotoxic stress, and more}.''} Nat Rev Mol Cell Biol. \url{https://doi.org/10.1038/nrm3546}.

\leavevmode\vadjust pre{\hypertarget{ref-Shin2004}{}}%
Shin, Chanseok, and James L Manley. 2004. {``{Cell signalling and the control of pre-mRNA splicing}.''} Nat Rev Mol Cell Biol. \url{https://doi.org/10.1038/nrm1467}.

\leavevmode\vadjust pre{\hypertarget{ref-Shlomchik2012}{}}%
Shlomchik, Mark J., and Florian Weisel. 2012. {``{Germinal center selection and the development of memory B and plasma cells}.''} \emph{Immunological Reviews} 247 (1): 52--63. \url{https://doi.org/10.1111/J.1600-065X.2012.01124.X}.

\leavevmode\vadjust pre{\hypertarget{ref-Siegel2012}{}}%
Siegel, Rebecca, Carol DeSantis, Katherine Virgo, Kevin Stein, Angela Mariotto, Tenbroeck Smith, Dexter Cooper, et al. 2012. {``{Cancer treatment and survivorship statistics, 2012}.''} \emph{CA: A Cancer Journal for Clinicians} 62 (4): 220--41. \url{https://doi.org/10.3322/caac.21149}.

\leavevmode\vadjust pre{\hypertarget{ref-Simonetti2014}{}}%
Simonetti, Giorgia, Maria Teresa Sabrina Bertilaccio, Paolo Ghia, and Ulf Klein. 2014. {``{Mouse models in the study of chronic lymphocytic leukemia pathogenesis and therapy}.''} American Society of Hematology. \url{https://doi.org/10.1182/blood-2014-05-577122}.

\leavevmode\vadjust pre{\hypertarget{ref-Skowronska2012}{}}%
Skowronska, Anna, Anton Parker, Gulshanara Ahmed, Ceri Oldreive, Zadie Davis, Sue Richards, Martin Dyer, et al. 2012. {``{Biallelic ATM inactivation significantly reduces survival in patients treated on the United Kingdom leukemia research fund chronic lymphocytic leukemia 4 trial}.''} \emph{Journal of Clinical Oncology} 30 (36): 4524--32. \url{https://doi.org/10.1200/JCO.2011.41.0852}.

\leavevmode\vadjust pre{\hypertarget{ref-Snijder2017}{}}%
Snijder, Berend, Gregory I Vladimer, Nikolaus Krall, Katsuhiro Miura, Ann Sofie Schmolke, Christoph Kornauth, Oscar Lopez de la Fuente, et al. 2017. {``{Image-based ex-vivo drug screening for patients with aggressive haematological malignancies: interim results from a single-arm, open-label, pilot study}.''} \emph{The Lancet Haematology} 4 (12): e595--606. \url{https://doi.org/10.1016/S2352-3026(17)30208-9}.

\leavevmode\vadjust pre{\hypertarget{ref-Srinivasan2020}{}}%
Srinivasan, Vishrut K, Shano Naseem, Neelam Varma, Deepesh P Lad, and Pankaj Malhotra. 2020. {``{Genomic alterations in chronic lymphocytic leukemia and their correlation with clinico-hematological parameters and disease progression}.''} \emph{Blood Research} 55 (3): 131--38. \url{https://doi.org/10.5045/br.2020.2020080}.

\leavevmode\vadjust pre{\hypertarget{ref-Stamatopoulos2012}{}}%
Stamatopoulos, Basile, Nathalie Meuleman, Cécile De Bruyn, Karlien Pieters, Philippe Mineur, Christine Le Roy, Stéphane Saint-Georges, et al. 2012. {``{AMD3100 disrupts the cross-talk between chronic lymphocytic leukemia cells and a mesenchymal stromal or nurse-like cell-based microenvironment: pre-clinical evidence for its association with chronic lymphocytic leukemia treatments.}''} \emph{Haematologica} 97 (4): 608--15. \url{https://doi.org/10.3324/haematol.2011.052779}.

\leavevmode\vadjust pre{\hypertarget{ref-Stamatopoulos2015}{}}%
Stamatopoulos, Basile, Michaël Van Damme, Emerence Crompot, Barbara Dessars, Hakim El Housni, Philippe Mineur, Nathalie Meuleman, Dominique Bron, and Laurence Lagneaux. 2015. {``{Opposite Prognostic Significance of Cellular and Serum Circulating MicroRNA-150 in Patients with Chronic Lymphocytic Leukemia}.''} \emph{Molecular Medicine} 21 (1): 123--33. \url{https://doi.org/10.2119/molmed.2014.00214}.

\leavevmode\vadjust pre{\hypertarget{ref-Stamatopoulos2010}{}}%
Stamatopoulos, B., N. Meuleman, C. De Bruyn, A. Delforge, D. Bron, and L. Lagneaux. 2010. {``{The histone deacetylase inhibitor suberoylanilide hydroxamic acid induces apoptosis, down-regulates the CXCR4 chemokine receptor and impairs migration of chronic lymphocytic leukemia cells}.''} \emph{Haematologica} 95 (7): 1136--43. \url{https://doi.org/10.3324/haematol.2009.013847}.

\leavevmode\vadjust pre{\hypertarget{ref-Stamatopoulos2007}{}}%
Stamatopoulos, Kostas, Chrysoula Belessi, Carol Moreno, Myriam Boudjograh, Giuseppe Guida, Tatjana Smilevska, Lynda Belhoul, et al. 2007. \emph{Blood} 109 (1): 259--70. \url{https://doi.org/10.1182/blood-2006-03-012948}.

\leavevmode\vadjust pre{\hypertarget{ref-Stankovic2014}{}}%
Stankovic, Tatjana, and Anna Skowronska. 2014. {``{The role of ATM mutations and 11q deletions in disease progression in chronic lymphocytic leukemia}.''} Leuk Lymphoma. \url{https://doi.org/10.3109/10428194.2013.829919}.

\leavevmode\vadjust pre{\hypertarget{ref-Stevenson2004}{}}%
Stevenson, Freda K, and Federico Caligaris-Cappio. 2004. {``{Chronic lymphocytic leukemia: Revelations from the B-cell receptor}.''} Blood. \url{https://doi.org/10.1182/blood-2003-12-4312}.

\leavevmode\vadjust pre{\hypertarget{ref-Sthoeger1989}{}}%
Sthoeger, Z. M., M. Wakai, D. B. Tse, V. P. Vinciguerra, S. L. Allen, D. R. Budman, S. M. Lichtman, P. Schulman, L. R. Weiselberg, and N. Chiorazzi. 1989. {``{Production of autoantibodies by CD5-expressing B lymphocytes from patients with chronic lymphocytic leukemia}.''} \emph{Journal of Experimental Medicine} 169 (1): 255--68. \url{https://doi.org/10.1084/jem.169.1.255}.

\leavevmode\vadjust pre{\hypertarget{ref-Stilgenbauer2016}{}}%
Stilgenbauer, Stephan, Barbara Eichhorst, Johannes Schetelig, Steven Coutre, John F Seymour, Talha Munir, Soham D Puvvada, et al. 2016. {``{Venetoclax in relapsed or refractory chronic lymphocytic leukaemia with 17p deletion: a multicentre, open-label, phase 2 study}.''} \emph{The Lancet Oncology} 17 (6): 768--78. \url{https://doi.org/10.1016/S1470-2045(16)30019-5}.

\leavevmode\vadjust pre{\hypertarget{ref-Stilgenbauer2014}{}}%
Stilgenbauer, Stephan, Andrea Schnaiter, Peter Paschka, Thorsten Zenz, Marianna Rossi, Konstanze Döhner, Andreas Bühler, et al. 2014. {``{Gene mutations and treatment outcome in chronic lymphocytic leukemia: results from the CLL8 trial.}''} \emph{Blood} 123 (21): 3247--54. \url{https://doi.org/10.1182/blood-2014-01-546150}.

\leavevmode\vadjust pre{\hypertarget{ref-R-survival}{}}%
Therneau, Terry M. 2021. \emph{Survival: Survival Analysis}. \url{https://github.com/therneau/survival}.

\leavevmode\vadjust pre{\hypertarget{ref-Tibshirani1996}{}}%
Tibshirani, Robert. 1996. {``{Regression Shrinkage and Selection via the Lasso}.''} 1. Vol. 58. \url{https://www.jstor.org/stable/pdf/2346178.pdf}.

\leavevmode\vadjust pre{\hypertarget{ref-Timp2013}{}}%
Timp, Winston, and Andrew P Feinberg. 2013. {``{Cancer as a dysregulated epigenome allowing cellular growth advantage at the expense of the host}.''} Nat Rev Cancer. \url{https://doi.org/10.1038/nrc3486}.

\leavevmode\vadjust pre{\hypertarget{ref-deTotero2006}{}}%
Totero, Daniela de, Raffaella Meazza, Simona Zupo, Giovanna Cutrona, Serena Matis, Monica Colombo, Enrico Balleari, et al. 2006. {``{Interleukin-21 receptor (IL-21R) is up-regulated by CD40 triggering and mediates proapoptotic signals in chronic lymphocytic leukemia B cells.}''} \emph{Blood} 107 (9): 3708--15. \url{https://doi.org/10.1182/blood-2005-09-3535}.

\leavevmode\vadjust pre{\hypertarget{ref-Trbusek2011}{}}%
Trbusek, Martin, Jana Smardova, Jitka Malcikova, Ludmila Sebejova, Petr Dobes, Miluse Svitakova, Vladimira Vranova, et al. 2011. {``{Missense mutations located in structural p53 DNA-binding motifs are associated with extremely poor survival in chronic lymphocytic leukemia}.''} \emph{Journal of Clinical Oncology} 29 (19): 2703--8. \url{https://doi.org/10.1200/JCO.2011.34.7872}.

\leavevmode\vadjust pre{\hypertarget{ref-Trentin1996}{}}%
Trentin, Livio, Andrea Cerutti, Renato Zambello, Rosaria Sancetta, Cristina Tassinari, Monica Facco, Fausto Adami, Francesco Rodeghiero, Carlo Agostini, and Gianpietro Semenzato. 1996. {``{Interleukin-15 promotes the growth of leukemic cells of patients with B-cell chronic lymphoproliferative disorders}.''} \emph{Blood} 87 (8): 3327--35. \url{https://doi.org/10.1182/blood.v87.8.3327.bloodjournal8783327}.

\leavevmode\vadjust pre{\hypertarget{ref-Trimarco2015}{}}%
Trimarco, Valentina, Elisa Ave, Monica Facco, Giorgia Chiodin, Federica Frezzato, Veronica Martini, Cristina Gattazzo, et al. 2015. {``{Cross-talk between chronic lymphocytic leukemia (CLL) tumor b cells and mesenchymal stromal cells (MSCs): Implications for neoplastic cell survival}.''} \emph{Oncotarget} 6 (39): 42130--49. \url{https://doi.org/10.18632/oncotarget.6239}.

\leavevmode\vadjust pre{\hypertarget{ref-Tsukada2002}{}}%
Tsukada, Nobuhiro, Jan A. Burger, Nathan J. Zvaifler, and Thomas J. Kipps. 2002. {``{Distinctive features of "nurselike" cells that differentiate in the context of chronic lymphocytic leukemia}.''} \emph{Blood} 99 (3): 1030--37. \url{https://doi.org/10.1182/blood.V99.3.1030}.

\leavevmode\vadjust pre{\hypertarget{ref-Turkistany2011}{}}%
Turkistany, Shereen A, and Rodney P Dekoter. 2011. {``{The transcription factor PU.1 is a critical regulator of cellular communication in the immune system}.''} \url{https://doi.org/10.1007/s00005-011-0147-9}.

\leavevmode\vadjust pre{\hypertarget{ref-Tyner2013}{}}%
Tyner, Jeffrey W, Wayne F Yang, Armand Bankhead, Guang Fan, Luke B Fletcher, Jade Bryant, Jason M Glover, et al. 2013. {``{Kinase pathway dependence in primary human leukemias determined by rapid inhibitor screening}.''} \emph{Cancer Research} 73 (1): 285--96. \url{https://doi.org/10.1158/0008-5472.CAN-12-1906}.

\leavevmode\vadjust pre{\hypertarget{ref-Vangapandu2017}{}}%
Vangapandu, Hima V., Mary L. Ayres, Christopher A. Bristow, William G. Wierda, Michael J. Keating, Kumudha Balakrishnan, Christine M. Stellrecht, and Varsha Gandhi. 2017. {``{The Stromal Microenvironment Modulates Mitochondrial Oxidative Phosphorylation in Chronic Lymphocytic Leukemia Cells}.''} \emph{Neoplasia (United States)} 19 (10): 762--71. \url{https://doi.org/10.1016/j.neo.2017.07.004}.

\leavevmode\vadjust pre{\hypertarget{ref-Vogelstein2013}{}}%
Vogelstein, Bert, Nickolas Papadopoulos, Victor E Velculescu, Shibin Zhou, Luis A Diaz, and Kenneth W Kinzler. 2013. {``{Cancer genome landscapes}.''} Science. \url{https://doi.org/10.1126/science.1235122}.

\leavevmode\vadjust pre{\hypertarget{ref-Wahlfors1992}{}}%
Wahlfors, J., H. Hiltunen, K. Heinonen, E Hamalainen, L. Alhonen, and J. Janne. 1992. {``{Genomic hypomethylation in human chronic lymphocytic leukemia}.''} \emph{Blood} 80 (8): 2074--80. \url{https://doi.org/10.1182/blood.v80.8.2074.2074}.

\leavevmode\vadjust pre{\hypertarget{ref-Walker2018}{}}%
Walker, Jeffrey A. 2018. {``{Elementary Statistical Modeling for Applied Biostatistics}.''} \url{https://www.middleprofessor.com/files/applied-biostatistics_bookdown/_book/index.html\#what-is-unusual-about-this-book}.

\leavevmode\vadjust pre{\hypertarget{ref-Walsby2014}{}}%
Walsby, Elisabeth, Andrea Buggins, Stephen Devereux, Ceri Jones, Guy Pratt, Paul Brennan, Chris Fegan, and Chris Pepper. 2014. {``{Development and characterization of a physiologically relevant model of lymphocyte migration in chronic lymphocytic leukemia}.''} \emph{Blood} 123 (23): 3607--17. \url{https://doi.org/10.1182/blood-2013-12-544569}.

\leavevmode\vadjust pre{\hypertarget{ref-Wang2011}{}}%
Wang, Lili, Michael S. Lawrence, Youzhong Wan, Petar Stojanov, Carrie Sougnez, Kristen Stevenson, Lillian Werner, et al. 2011. {``{SF3B1 and Other Novel Cancer Genes in Chronic Lymphocytic Leukemia}.''} \emph{New England Journal of Medicine} 365 (26): 2497--2506. \url{https://doi.org/10.1056/nejmoa1109016}.

\leavevmode\vadjust pre{\hypertarget{ref-Widhopf2004}{}}%
Widhopf, George F., Laura Z. Rassenti, Traci L. Toy, John G. Gribben, William G. Wierda, and Thomas J. Kipps. 2004. \emph{Blood} 104 (8): 2499--2504. \url{https://doi.org/10.1182/blood-2004-03-0818}.

\leavevmode\vadjust pre{\hypertarget{ref-R-ConsensusClusterPlus}{}}%
Wilkerson, Matt, and Peter Waltman. 2021. \emph{ConsensusClusterPlus: ConsensusClusterPlus}.

\leavevmode\vadjust pre{\hypertarget{ref-Willis2017}{}}%
Willis, Simon N., Julie Tellier, Yang Liao, Stephanie Trezise, Amanda Light, Kristy O'Donnell, Lee Ann Garrett-Sinha, Wei Shi, David M. Tarlinton, and Stephen L. Nutt. 2017. {``{Environmental sensing by mature B cells is controlled by the transcription factors PU.1 and SpiB}.''} \emph{Nature Communications} 8 (1): 1426. \url{https://doi.org/10.1038/s41467-017-01605-1}.

\leavevmode\vadjust pre{\hypertarget{ref-Woyach2014}{}}%
Woyach, Jennifer A., Richard R. Furman, Ta-Ming Liu, Hatice Gulcin Ozer, Marc Zapatka, Amy S. Ruppert, Ling Xue, et al. 2014. {``{Resistance Mechanisms for the Bruton's Tyrosine Kinase Inhibitor Ibrutinib}.''} \emph{New England Journal of Medicine} 370 (24): 2286--94. \url{https://doi.org/10.1056/nejmoa1400029}.

\leavevmode\vadjust pre{\hypertarget{ref-Woyach2015}{}}%
Woyach, Jennifer A., and Amy J. Johnson. 2015. {``{Targeted therapies in CLL: Mechanisms of resistance and strategies for management}.''} \emph{Blood} 126 (4): 471--77. \url{https://doi.org/10.1182/blood-2015-03-585075}.

\leavevmode\vadjust pre{\hypertarget{ref-Xu2018}{}}%
Xu, Zhenshu, Donglian Xiong, Jushun Zhang, Jingyan Zhang, Xiuli Chen, Zhizhe Chen, and Rong Zhan. 2018. {``{Bone marrow stromal cells enhance the survival of chronic lymphocytic leukemia cells by regulating HES-1 gene expression and H3K27me3 demethylation}.''} \emph{Oncology Letters} 15 (2): 1937--42. \url{https://doi.org/10.3892/ol.2017.7450}.

\leavevmode\vadjust pre{\hypertarget{ref-Yosifov2019}{}}%
Yosifov, Deyan Y, Christine Wolf, Stephan Stilgenbauer, and Daniel Mertens. 2019. {``{From biology to therapy: The CLL success story}.''} Wolters Kluwer Health. \url{https://doi.org/10.1097/HS9.0000000000000175}.

\leavevmode\vadjust pre{\hypertarget{ref-R-ChIPseeker}{}}%
Yu, Guangchuang. 2021a. \emph{ChIPseeker: ChIPseeker for ChIP Peak Annotation, Comparison, and Visualization}. \url{https://guangchuangyu.github.io/software/ChIPseeker}.

\leavevmode\vadjust pre{\hypertarget{ref-R-clusterProfiler}{}}%
---------. 2021b. \emph{clusterProfiler: A Universal Enrichment Tool for Interpreting Omics Data}. \url{https://yulab-smu.top/biomedical-knowledge-mining-book/}.

\leavevmode\vadjust pre{\hypertarget{ref-Yu2017}{}}%
Yu, Lijian, Haesook T Kim, Siddha N. Kasar, Parul Benien, Wei Du, Kevin Hoang, Andrew Aw, et al. 2017. {``{Survival of Del17p CLL depends on genomic complexity and somatic mutation}.''} \emph{Clinical Cancer Research} 23 (3): 735--45. \url{https://doi.org/10.1158/1078-0432.CCR-16-0594}.

\leavevmode\vadjust pre{\hypertarget{ref-Zenz2010}{}}%
Zenz, Thorsten, Barbara Eichhorst, Raymonde Busch, Tina Denzel, Sonja Häbe, Dirk Winkler, Andreas Bühler, et al. 2010. {``{TP53 mutation and survival in chronic lymphocytic leukemia.}''} \emph{Journal of Clinical Oncology : Official Journal of the American Society of Clinical Oncology} 28 (29): 4473--79. \url{https://doi.org/10.1200/JCO.2009.27.8762}.

\leavevmode\vadjust pre{\hypertarget{ref-Zenz2008}{}}%
Zenz, Thorsten, Alexander Kröber, Katrin Scherer, Sonja Häbe, Andreas Bühler, Axel Benner, Tina Denzel, et al. 2008. {``{Monoallelic TP53 inactivation is associated with poor prognosis in chronic lymphocytic leukemia: Results from a detailed genetic characterization with long-term follow-up}.''} \emph{Blood} 112 (8): 3322--29. \url{https://doi.org/10.1182/blood-2008-04-154070}.

\leavevmode\vadjust pre{\hypertarget{ref-Zenz2019}{}}%
Zenz, Thorsten, Almut Luetge, Junyan Lu, Huellein Jennifer, Sascha Dietrich, Leopold Sellner, and Wolfgang Huber. 2019. {``{Transcriptional Profiling Reveals Strong Impact of Major Molecular Disease Subgroups and Mixed Epistasis in Chronic Lymphocytic Leukemia}.''} \emph{Blood} 134 (Supplement\_1): 1742--42. \url{https://doi.org/10.1182/blood-2019-126388}.

\leavevmode\vadjust pre{\hypertarget{ref-Zhang2012}{}}%
Zhang, Wan, Dunyaporn Trachootham, Jinyun Liu, Gang Chen, Helene Pelicano, Celia Garcia-Prieto, Weiqin Lu, et al. 2012. {``{Stromal control of cystine metabolism promotes cancer cell survival in chronic lymphocytic leukaemia}.''} \emph{Nature Cell Biology} 14 (3): 276--86. \url{https://doi.org/10.1038/ncb2432}.

\leavevmode\vadjust pre{\hypertarget{ref-Ziller2013}{}}%
Ziller, Michael J., Hongcang Gu, Fabian Müller, Julie Donaghey, Linus T.-Y. Tsai, Oliver Kohlbacher, Philip L. De Jager, et al. 2013. {``{Charting a dynamic DNA methylation landscape of the human genome}.''} \emph{Nature} 500 (7463): 477--81. \url{https://doi.org/10.1038/nature12433}.

\end{CSLReferences}
\indent
\setlength{\parindent}{17pt}
\setlength{\leftskip}{0pt}
\setlength{\parskip}{0pt}

\newpage

\hypertarget{appendix}{%
\chapter*{Appendix}\label{appendix}}
\addcontentsline{toc}{chapter}{Appendix}

\hypertarget{figures}{%
\section*{Figures}\label{figures}}
\addcontentsline{toc}{section}{Figures}


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{figures/graph} 

}

\caption{Analysis of ATACseq dataset of two trisomy 12 and two non-trisomy 12 untreated CLL PMBC samples. The volcano plot depicts change in TF activity (x axis) versus BH-adjusted p-values (y axis) for 636 TFs, comparing trisomy 12 and non-trisomy 12 samples. The (\protect\hyperlink{ref-diffTF}{\textbf{diffTF?}}){[}REFERENCE{]} package was ran in analytical mode to calculate TF activity, measured as weighted mean difference. TFs are labeled if adjusted p-value \textless{} 0.01 and absolute weighted mean difference \textgreater{} 0.15.}\label{fig:diffTFsmallvolPlot}
\end{figure}
(ref:drugcytGeneIntAll)
\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/drugcytGeneIntAll-1} 

}

\caption{(ref:drugcytGeneIntAll)}\label{fig:drugcytGeneIntAll}
\end{figure}

\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/NutlinPredictors-1} 

}

\caption{ADD NUTLIN-3A Predictor profile}\label{fig:NutlinPredictors}
\end{figure}
\hypertarget{tables}{%
\section*{Tables}\label{tables}}
\addcontentsline{toc}{section}{Tables}


\begin{table}
\centering
\begin{tabular}{l|l|l|l}
\hline
Cytokine & Receptor & ENSEMBLID & Kegg\\
\hline
\cellcolor[HTML]{E2E868}{SDF-1a} & \cellcolor[HTML]{E2E868}{CXCR 4} & \cellcolor[HTML]{E2E868}{ENSG00000121966} & \cellcolor[HTML]{E2E868}{7852}\\
\hline
SDF-1a & CXCR 7 & ENSG00000144476 & 57007\\
\hline
\cellcolor[HTML]{E2E868}{IL-6} & \cellcolor[HTML]{E2E868}{IL-6 R} & \cellcolor[HTML]{E2E868}{ENSG00000160712} & \cellcolor[HTML]{E2E868}{3570}\\
\hline
IL-6 & IL-6 ST & ENSG00000134352 & 3572\\
\hline
\cellcolor[HTML]{E2E868}{IL-15} & \cellcolor[HTML]{E2E868}{IL-15 R alpha} & \cellcolor[HTML]{E2E868}{ENSG00000134470} & \cellcolor[HTML]{E2E868}{3601}\\
\hline
IL-15 & IL-2 R beta & ENSG00000100385 & 3560\\
\hline
\cellcolor[HTML]{E2E868}{IL-15} & \cellcolor[HTML]{E2E868}{IL-2 R gamma} & \cellcolor[HTML]{E2E868}{ENSG00000147168} & \cellcolor[HTML]{E2E868}{3561}\\
\hline
IL-21 & IL-21 R & ENSG00000103522 & 50615\\
\hline
\cellcolor[HTML]{E2E868}{IL-21} & \cellcolor[HTML]{E2E868}{IL-21 R gamma} & \cellcolor[HTML]{E2E868}{ENSG00000147168} & \cellcolor[HTML]{E2E868}{3561}\\
\hline
IL-2 & IL-2 R alpha & ENSG00000134460 & 3559\\
\hline
\cellcolor[HTML]{E2E868}{IL-2} & \cellcolor[HTML]{E2E868}{IL-2 R beta} & \cellcolor[HTML]{E2E868}{ENSG00000100385} & \cellcolor[HTML]{E2E868}{3560}\\
\hline
IL-2 & IL-2 R gamma & ENSG00000147168 & 3561\\
\hline
\cellcolor[HTML]{E2E868}{IL-10} & \cellcolor[HTML]{E2E868}{IL-10 R alpha} & \cellcolor[HTML]{E2E868}{ENSG00000110324} & \cellcolor[HTML]{E2E868}{3587}\\
\hline
IL-10 & IL-10 R beta & ENSG00000243646 & 3588\\
\hline
\cellcolor[HTML]{E2E868}{IL-4} & \cellcolor[HTML]{E2E868}{IL-4 R} & \cellcolor[HTML]{E2E868}{ENSG00000077238} & \cellcolor[HTML]{E2E868}{3566}\\
\hline
IL-4 & IL-2 R gamma & ENSG00000147168 & 3561\\
\hline
\cellcolor[HTML]{E2E868}{IL-4} & \cellcolor[HTML]{E2E868}{IL-13 R A1} & \cellcolor[HTML]{E2E868}{ENSG00000131724} & \cellcolor[HTML]{E2E868}{3597}\\
\hline
soluble anti-IgM & B cell receptor & ENSG00000275063 & 102723407\\
\hline
\cellcolor[HTML]{E2E868}{soluble anti-IgM} & \cellcolor[HTML]{E2E868}{CD79A} & \cellcolor[HTML]{E2E868}{ENSG00000105369} & \cellcolor[HTML]{E2E868}{973}\\
\hline
soluble anti-IgM & CD79B & ENSG00000007312 & 974\\
\hline
\cellcolor[HTML]{E2E868}{Resiquimod} & \cellcolor[HTML]{E2E868}{TLR 7} & \cellcolor[HTML]{E2E868}{ENSG00000196664} & \cellcolor[HTML]{E2E868}{51284 }\\
\hline
Resiquimod & TLR 8 & ENSG00000101916 & 51311\\
\hline
\cellcolor[HTML]{E2E868}{CpG ODN} & \cellcolor[HTML]{E2E868}{TLR9} & \cellcolor[HTML]{E2E868}{ENSG00000239732} & \cellcolor[HTML]{E2E868}{54106}\\
\hline
BAFF & BAFF R & ENSG00000159958 & 115650\\
\hline
\cellcolor[HTML]{E2E868}{BAFF} & \cellcolor[HTML]{E2E868}{TACI} & \cellcolor[HTML]{E2E868}{ENSG00000240505} & \cellcolor[HTML]{E2E868}{23495}\\
\hline
BAFF & BCMA & ENSG00000048462 & 608\\
\hline
\cellcolor[HTML]{E2E868}{IL-1b} & \cellcolor[HTML]{E2E868}{IL-1 R1} & \cellcolor[HTML]{E2E868}{ENSG00000115594} & \cellcolor[HTML]{E2E868}{3554}\\
\hline
IL-1b & IL-1 R2 & ENSG00000115590 & 7850\\
\hline
\cellcolor[HTML]{E2E868}{sCD40L} & \cellcolor[HTML]{E2E868}{CD40} & \cellcolor[HTML]{E2E868}{ENSG00000101017} & \cellcolor[HTML]{E2E868}{958}\\
\hline
Interferon gamma & IFN-gamma R1 & ENSG00000027697 & 3459\\
\hline
\cellcolor[HTML]{E2E868}{Interferon gamma} & \cellcolor[HTML]{E2E868}{IFN-gamma R2} & \cellcolor[HTML]{E2E868}{ENSG00000159128} & \cellcolor[HTML]{E2E868}{3460}\\
\hline
TGF-b1 & TGF-beta R1 & ENSG00000106799 & 7046\\
\hline
\cellcolor[HTML]{E2E868}{TGF-b1} & \cellcolor[HTML]{E2E868}{TGF-beta R2} & \cellcolor[HTML]{E2E868}{ENSG00000163513} & \cellcolor[HTML]{E2E868}{7048}\\
\hline
\end{tabular}
\end{table}
\begin{table}[ht]
    \begin{center}
        {\footnotesize
        \begin{tabular}{l|cccccccccc}
        \hline \hline
                        & 3m    & 6m    & 1yr   & 2yr   & 3yr   & 5yr   & 7yr   & 10yr  & 12yr  & 15yr   \\
            \hline
                Mean   & 3.138 & 3.191 & 3.307 & 3.544 & 3.756 & 4.093 & 4.354 & 4.621 & 4.741 & 4.878  \\
                Median & 3.013 & 3.109 & 3.228 & 3.490 & 3.680 & 3.906 & 4.117 & 4.420 & 4.575 & 4.759  \\
                Min    & 1.984 & 1.950 & 1.956 & 2.010 & 2.240 & 2.615 & 2.850 & 3.120 & 3.250 & 3.395  \\
                Max    & 5.211 & 5.274 & 5.415 & 5.583 & 5.698 & 5.805 & 5.900 & 6.031 & 6.150 & 6.295  \\
                StD    & 0.915 & 0.919 & 0.935 & 0.910 & 0.876 & 0.825 & 0.803 & 0.776 & 0.768 & 0.762  \\
            \hline \hline
        \end{tabular}}
    \end{center}
    \caption{Detailed descriptive statistics of location and dispersion for
    2100 observed swap rates for the period from
    February 15, 1999 to March 2, 2007. Swap rates measured as 3.12 (instead of 0.0312).}
    \label{tab:apptable}
\end{table}
\newpage

\end{document}
