\documentclass[11pt, a4paper, twosided]{book}
\usepackage[hmarginratio=1:1]{geometry} % equal right and left raitios
\usepackage{amsmath,amssymb,amsfonts,amsthm}    % Typical maths resource packages
\usepackage{graphicx}                           % Packages to allow inclusion of graphics
\usepackage[authoryear]{natbib}                 % literature reference style
\usepackage[bf]{caption}
\usepackage{textcomp}                           % For single quotes
\usepackage{floatrow}                           % For image and table position
\usepackage{booktabs}                           % For tables
% \usepackage[colorlinks=true]{hyperref}
% \usepackage[bottom]{footmisc}
\usepackage[bottom, flushmargin]{footmisc}                   % For footnotes
\usepackage[citebordercolor={0 1 0}]{hyperref}  % For creating hyperlinks in cross references
\usepackage{footnotebackref}
\usepackage[utf8]{inputenc}
\usepackage[T1]{fontenc}
% i have added the below, so that kable works
\usepackage{colortbl}
\usepackage{longtable}
\usepackage{array}
\usepackage{multirow}
\usepackage{wrapfig}
\usepackage{float}
\usepackage{pdflscape}
\usepackage{tabu}
\usepackage{threeparttable}
\usepackage{threeparttablex}
\usepackage[normalem]{ulem}
\usepackage{makecell}
\usepackage{xcolor}
\usepackage[gen]{eurosym}
\usepackage{breqn} %to break up equations accross multiple lines, where neccesary 
\usepackage{helvet} %to set font as helvetica 

       % For being able to set `echo = TRUE` in code chunk options, and thus in-line representation of code

%--------------------------------------------
% define line spacing = 1.5
\renewcommand{\baselinestretch}{1.2}

%define font
\renewcommand{\familydefault}{\sfdefault}

% define position of graphics
\floatsetup[figure]{capposition=bottom}
\floatsetup[table]{capposition=top}
\floatplacement{figure}{ht}
\floatplacement{table}{ht}

% define CSL references (required)
\newlength{\cslhangindent}
\setlength{\cslhangindent}{1.5em}
\newenvironment{CSLReferences}%
  {}%
  {\par}

% save thesis parameters for later
\newcommand{\thesistype}{Dissertation}
\newcommand{\thesisauthor}{Holly Amelia Rebecca Giles}
\newcommand{\thesisdate}{Oct 01, 2021?}

% Added by CII (Thanks, Hadley!)
% Use ref for internal links
\renewcommand{\hyperref}[2][???]{\autoref{#1}}
\def\chapterautorefname{Chapter}
\def\sectionautorefname{Section}
\def\subsectionautorefname{Subsection}
% End of CII addition

% Added by CII
\usepackage{caption}
\captionsetup{width=5in,font={small}}
% End of CII addition

% Added by CII
%%% Copied from knitr
%% maxwidth is the original width if it's less than linewidth
%% otherwise use linewidth (to make sure the graphics do not exceed the margin)
\makeatletter
\def\maxwidth{ %
  \ifdim\Gin@nat@width>\linewidth
    \linewidth
  \else
    \Gin@nat@width
  \fi
}
\makeatother

\renewcommand{\contentsname}{Table of Contents}
% End of CII addition

\setlength{\parskip}{0pt}

% Added by CII
  %\setlength{\parskip}{\baselineskip}
  \usepackage[parfill]{parskip}

\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}

	\usepackage{float}
% End of CII addition
%%
%% End Preamble
%%
%
\begin{document}

% ------------------------------------
% ---- frontmatter: First Page -------
% ------------------------------------

\thispagestyle{empty}
  \begin{center}
  {\Large{Dissertation\\
    submitted to the\\
    Combined Faculty of Natural Sciences and Mathematics\\
    of the Ruperto Carola University Heidelberg, Germany\\
    for the degree of\\
    Doctor of Natural Sciences\\}} \vspace*{\fill}

  \Large{Presented by} \\\vspace{0.5cm}
  \Large{M.A. Holly Amelia Rebecca Giles} \\
  \Large{born in Portsmouth} \\
  \Large{Oral examination: Dec 15, 2021?}

\end{center}
\newpage\null\thispagestyle{empty}\newpage

% ------------------------------------
% --- frontmatter: Secound Page ------
% ------------------------------------

\pagebreak
\thispagestyle{empty}
\null
\begin{center}
\Large{Drug - microenvironment - gene interplay in chronic lymphocytic leukemia} \\
\vspace*{\fill}
\begin{tabular}{ c c }
Referees: & Dr.~Judith Zaugg \\
          & Prof.~Dr.~Michael Boutros
\end{tabular}
\end{center}
\newpage\null\thispagestyle{empty}\newpage

% --------------------------------------------
% -------- Declaration of Authorship ---------
% --------------------------------------------

\newpage
\thispagestyle{empty}
% \hypertarget{declaration-of-authorship}{%
% \section*{Declaration of Authorship}\label{declaration-of-authorship}}
%
% I hereby confirm that I have authored this Dissertation independently and
% without the use of others sources than the ones indicated. I have not yet
% presented this thesis or parts thereof to a university as part of an
% examination or degree. This work was carried out in the Genome Biology at
% the European Molecular Biology Laboratory (EMBL) in the group of Dr.~Wolfgang Huber and Dr.~Sascha Dietrich.
% \vspace{1cm}
%
% \noindent Heidelberg, \thesisdate{}
% \vspace{3cm}
%
% \noindent . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
% \vspace{0.1cm}
%
% \noindent \thesisauthor{}
% \newpage
% \thispagestyle{empty}
% \vspace*{\fill}
% Holly Amelia Rebecca Giles, \textit{Drug - microenvironment - gene interplay in chronic lymphocytic leukemia}, \textcopyright\, \thesisdate{}
% \newpage

%\frontmatter % this stuff will be roman-numbered
\pagestyle{empty} % this removes page numbers from the frontmatter



% edit style below this point? 


% --------------------------------------------
% --------------- Dedication -----------------
% --------------------------------------------
\vspace*{\fill}
\begin{center}
\textit{To \ldots{}}
\end{center}
\vspace*{\fill}
\newpage\null\thispagestyle{empty}\newpage

% ------------------------------------
% --- frontmatter: Acknowledgement ---
% ------------------------------------
\pagestyle{plain}
\pagenumbering{roman}   %  define page number in roman style
\setcounter{page}{1}    % start page numbering anew
\newpage
\hypertarget{acknowledgements}{%
\chapter*{Acknowledgements}\label{acknowledgements}}
\addcontentsline{toc}{chapter}{Acknowledgements}

I want to thank a few people.
\newpage\null\newpage


% --------------------------------------------
% ---------- List of publications ------------
% --------------------------------------------
    \hypertarget{list-of-publications}{%
    \chapter*{List of publications}\label{list-of-publications}}
    \addcontentsline{toc}{chapter}{List of publications}

    \hypertarget{thesis-related}{%
    \subsubsection*{Thesis related}\label{thesis-related}}
    \addcontentsline{toc}{subsubsection}{Thesis related}
    \begin{itemize}
        \item  Peter-Martin Bruch\ast,  \textbf{\underline{Holly A. R. Giles}}\ast, Carolin Kolb, Sophie Herbst, Tina Becirovic, Tobias Roider, Junyan Lu, Sebastian Scheinost, Lena Wagner, Jennifer Huellein, Ivan Berest, Mark Kriegsmann, Katharina Kriegsmann, Christiane Zgorzelski, Peter Dreger, Judith Zaugg, Carsten Mueller-Tidow, Thorsten Zenz, Wolfgang Huber, Sascha Dietrich et al. \textit{in preparation.}  "Mapping drug-microenvironment-genetic interplay in CLL reveals trisomy 12 as a modulator of microenvironment."\textbf{\textit{bioRxiv}}. [doi: 10.1101/2021.07.23.453514](https://doi.org/10.1101/2021.07.23.453514)
    \end{itemize}
    \hypertarget{other-contributions}{%
    \subsubsection*{Other contributions}\label{other-contributions}}
    \addcontentsline{toc}{subsubsection}{Other contributions}

    The author of this thesis also contributed to a number of other projects throughout the PhD. The following have been published:
    \begin{itemize}
        \item Berest, Ivan\ast, Christian Arnold\ast, Armando Reyes-Palomares, Giovanni Palla, Kasper Dindler Rasmussen, \textbf{\underline{Holly Giles}}, and Peter-Martin Bruch et al. 2019. "Quantification Of Differential Transcription Factor Activity And Multiomics-Based Classification Into Activators And Repressors: DiffTF". \textit{Cell Reports} 29 (10): 3147-3159.e12. [doi: 10.1016/j.celrep.2019.10.106](https://doi.org/10.1016/j.celrep.2019.10.106)
        
      \item Lu, Junyan\ast, Ester Cannizzaro\ast, Fabienne Meier-Abt, Sebastian Scheinost, Peter-Martin Bruch, \textbf{\underline{Holly A. R. Giles}}, and Almut LÃ¼tge et al. 2021. "Multi-Omics Reveals Clinically Relevant Proliferative Drive Associated With Mtor-MYC-OXPHOS Activity In Chronic Lymphocytic Leukemia". \textbf{\textit{Nature Cancer}}. [doi: 10.1038/s43018-021-00216-6](https://doi.org/10.1038/s43018-021-00216-6)
    \end{itemize}
\newpage\null\newpage

% -------------------------------------
% --- frontmatter: Zusammendfassung ---
% -------------------------------------
    \hypertarget{zusammenfassung}{%
    \chapter*{Zusammenfassung}\label{zusammenfassung}}
    \addcontentsline{toc}{chapter}{Zusammenfassung}

    Abstract in German
    Aktuelle Entwicklungen im Bereich der ``omik''-Technologien tragen wesentlich zurBeschleunigung des Fortschritts in der Krebsforschung bei etc
\newpage\null\newpage

% -----------------------------
% --- frontmatter: Abstract ---
% -----------------------------

    \hypertarget{abstract}{%
    \chapter*{Abstract}\label{abstract}}
    \addcontentsline{toc}{chapter}{Abstract}

    The tumour microenvironment and genetic alterations collectively influence disease biology and drug resistance in Chronic Lymphocytic Leukaemia (CLL). To establish an integrative understanding of these factors in CLL biology, we performed a combinatorial assay using 12 drugs individually co-applied with each of 17 microenvironmental stimuli on 192 samples of CLL peripheral blood mononucleated cells. We examined microenvironment response across a heterogeneous patient cohort and identified four distinct CLL subgroups that differed in their response landscapes and in patient outcomes. By combining our data with whole-exome sequencing, DNA-methylation, RNA-sequencing and copy number variant data of the same tumours, we systematically searched for molecular determinants of stimulus response and found trisomy 12 as a key modulator. Our data suggest that the amplifying effect of trisomy 12 on the response to environmental signals is mediated by the transcription factors Spi-B and PU.1. We generated a comprehensive map of drug-microenvironment interactions in CLL, and profiled the modulating impact of genetic features on these antagonistic and synergistic effects. Interleukin (IL) 4 and Toll-Like Receptor (TLR) 7/8/9 stimuli showed the most interactions. Both pathways were more active in CLL-infiltrated lymph nodes than in healthy samples (p\textless0.001), and high IL4 activity in lymph nodes correlated with shorter survival (p=0.038). We provide a multi-layered resource to investigate microenvironmental and drug interplay in CLL (\href{https://git.embl.de/giles/cllcytokinescreen2021}{Repository} \& \href{https://www.google.com/url?q=https://www.imbi.uni-heidelberg.de/dietrichlab/CLL_Microenvironment/\&sa=D\&source=editors\&ust=1625764640786000\&usg=AOvVaw01HOFP7k2BJRLTteCXa8qy}{Shiny}). Our results highlight the importance of cell-extrinsic influences on drug response and disease progression, and how these further depend on molecular features.
\newpage\null\newpage

% -----------------------------
% --- frontmatter: Contents ---
% -----------------------------
\newpage
\tableofcontents
\clearpage

% ----------------------------------------------------------
% --- frontmatter: List of Abbreviations (not mandatory) ---
% ----------------------------------------------------------
\newpage
\hypertarget{list-of-abbreviations}{%
\chapter*{List of Abbreviations}\label{list-of-abbreviations}}
\addcontentsline{toc}{chapter}{List of Abbreviations}
\begin{tabular}{rp{0.2cm}lp{1cm}rp{0.2cm}l}
    AKT & & Protein kinase B \\
    AUC & & Area under curve \\
    BCR & & B-cell receptor \\
    BH & & Benjamini-Hochberg \\
    BTK & & Brutonâs tyrosine kinase \\
    CDF & & Cumulative Distribution Function \\
    CLL & & Chronic lymphocytic leukemia \\
    CNV & & Copy number variation \\
    CpG ODN & & CpG oligodeoxynucleotides \\
    ERK/MAPK signalling & & Mitogen-activated protein kinase signalling \\
    FBS & & Fetal bovine serum \\
    FDR & & False discovery rate \\
    GSEA & & Gene set enrichment analysis  \\
    IFN$\gamma$ & & Interferon$\gamma$   \\ 
    Ig & & Immunoglobulin \\
    IGHV & & Immunoglobulin heavy chain variable region \\
    IL & & Interleukin \\
    JAK & & Janus kinase \\
    M-CLL & & CLL with somatic hypermutations in IGHV loci \\
    NF$\kappa$B & & Nuclear factor kappa-light-chain-enhancer of activated B cells \\
    NOTCH & & Neurogenic locus notch  \\
    OS & & Overall survival \\
    PBMC & & Peripheral blood mononuclear cells \\
    PCA & & Principal component analysis \\
    PI3K & & Phosphoinositide 3-kinase \\
    U-CLL & & CLL without somatic hypermutations in IGHV loci \\
    TGF$\beta$ & & Transforming growth factor $\beta$ \\
    TF & & Transcription factor \\
    TLR & & Toll-like receptor \\ 
    TSS & & Transcription start site \\
    TTT & & Time to next treatment \\
    SF3B1 & & Splicing factor 3B subunit 1 \\
    STAT & & Signal transducer and activator of transcription \\
    SYK & & Spleen tyrosine kinase \\
    VST & & Variance Stabilising Transformation \\
    WES & & Whole exome sequencing \\
    WGS & & Whole genome sequencing \\
    WT & & Wild type \\
    ZAP70 & & Tyrosine-protein kinase ZAP-70 \\

    
   
\end{tabular}
% ----------------------------------------------------
% --- frontmatter: List of Figures (not mandatory) ---
% ----------------------------------------------------
\newpage
\listoffigures
\addcontentsline{toc}{section}{List of Figures}

% ---------------------------------------------------
% --- frontmatter: List of Tables (not mandatory) ---
% ---------------------------------------------------
\newpage
\listoftables
\addcontentsline{toc}{section}{List of Tables}

% edit here to affect style of main document 
%\mainmatter % here the regular arabic numbering starts
\newpage
\pagestyle{plain} % use this to set the heading style 
\setcounter{page}{1}    % start page numbering anew
\pagenumbering{arabic}  % page numbers in arabic style

\hypertarget{introduction}{%
\chapter{Introduction}\label{introduction}}

\hypertarget{cll}{%
\section{CLL}\label{cll}}

\hypertarget{disease-characteristics}{%
\subsection{Disease characteristics}\label{disease-characteristics}}

\ref{fig:CLLmorphology} \_Figure from (\protect\hyperlink{ref-Kipps2017_}{\textbf{Kipps2017\_?}}).
\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{figures/CLL_morphology} 

}

\caption{(ref:CLLmorphology)}\label{fig:CLLmorphology}
\end{figure}
\ref{fig:proliferationCentres} \_Figure from (\protect\hyperlink{ref-Kipps2017_}{\textbf{Kipps2017\_?}}).
\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{figures/proliferation_centres} 

}

\caption{(ref:proliferationCentres)}\label{fig:proliferationCentres}
\end{figure}
\hypertarget{ighv-status-and-cell-of-origin}{%
\subsection{IGHV status and cell of origin}\label{ighv-status-and-cell-of-origin}}

\ref{fig:IGHVstatus} Taken from Kipps et al 2017: ``Normal naive B cells that have undergone successful V(D)J recombination and express functional B cell receptors that are capable of binding to antigen interact with CD4+ T cells and accessory cells, which aggregate to form follicles that become germinal centres. Germinal cells each have a dark zone, comprising rapidly dividing B cells, and a light zone, comprising B cells mixed with follicular dendritic cells (FDCs), macrophages and helper T cells (TH cells). The B cells enter the dark zone of the germinal centre where they experience rapid proliferation and somatic hypermutation (SHM) in the genes encoding the immunoglobulin variable regions of the heavy chain (IGHV) and the light chain (IGVL). As they pass through to the light zone, the B cells that express the fittest B cell receptors for binding antigen are selected and may undergo immunoglobulin class-switch recombination. Chronic lymphocytic leukaemia (CLL) cells that use unmutated IGHV apparently originate from CD5+ B cells prior to experiencing SHM, whereas CLL cells that use mutated IGHV most likely originate from CD5+ B cells that have passed through and differentiated in the germinal centre. Some CLL cells might be derived from B cells that also have undergone immunoglobulin class-switch recombination and express immunoglobulin isotypes other than IgM and IgD, for example, IgG or IgA. Another subset is one with CLL cells that express immunoglobulin with only modest somatic mutations, such as CLL cells that use IGHV3-21 with \textasciitilde97\% homology to the inherited IGHV3-21 gene and an immunoglobulin light chain encoded by an unmutated IGLV3-21; these cells might derive from a B cell that has had constrained SHM, possibly owing to a limited need for immunoglobulin somatic diveresification and selection. Dashed arrows indicate speculated pathways.'' \_Figure from (\protect\hyperlink{ref-Kipps2017_}{\textbf{Kipps2017\_?}}).
\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{figures/IGHV_status} 

}

\caption{(ref:IGHVstatus)}\label{fig:IGHVstatus}
\end{figure}
\hypertarget{the-central-role-of-bcr-signalling}{%
\subsection{The central role of BCR signalling}\label{the-central-role-of-bcr-signalling}}

\ref{fig:BCRsignalling} Adapted from: Kipps et al 2017: ``B cell receptor (BCR) signalling is initiated by SRC-family kinase-dependent phosphorylation (mainly LYN) of CD79A and CD79B that creates a docking site for the binding and activation of spleen tyrosine kinase (SYK). SYK then triggers the formation of a multi-component `signalosome,' comprising Bruton tyrosine kinase (BTK), AKT, phosphoinositide 3-kinase (PI3K), phospholipase CÎ³2 (PLCÎ³2) and B cell-linker protein (BLNK), among others. CD19 is a co-receptor for BCR and is important for PI3K activation, which recruits and activates PLCÎ³2, BTK and AKT. PLCÎ³2 generates diacylglycerol (DAG) and inositol-1,4,5-trisphosphate (Ins(1,4,5)P3), which triggers Ca2+ release from the endoplasmic reticulum, leading to the activation of the MEK--extracellular signal-regulated kinase (ERK) and nuclear factor-ÎºB (NF-ÎºB) signalling pathways. Other effects of BCR signalling include activation of mechanistic target of rapamycin complex 1 (mTORC1) and of Rho-family GTPases, RAC1 and RHOA, which can affect the cytoskeleton. Inhibitors of SYK, PI3K and BTK are shown. Note that this figure describes the main molecules and interactions that are involved in positive BCR signalling, but is not an exhaustive description of all signalling pathways or molecules activated. IKK, IÎºB kinase; PKC, protein kinase C.''
\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{figures/BCR_signalling} 

}

\caption{(ref:BCRsignalling)}\label{fig:BCRsignalling}
\end{figure}
\hypertarget{recurrent-genetic-features-in-cll}{%
\subsection{Recurrent genetic features in CLL}\label{recurrent-genetic-features-in-cll}}

\ref{fig:CLLmutations} Taken from Kipps et al 2017: ``Genes that are mutated in chronic lymphocytic leukaemia (CLL) are involved in several cellular pathways (blue boxes). As such, mutations in these genes can lead to a range of cellular consequences, such as aberrant DNA repair and B cell receptor (BCR) signalling, among others51,213. The minus sign from GBN1 to the MAPK--ERK pathway indicates negative regulation. *For more detail of the BCR and its associated signalling, see FIG. 3. ASXL1, additional sex combs-like protein 1; ATM, ataxia telangiectasia mutated; BAZ2A, bromodomain adjacent to zinc-finger domain protein 2A; BCOR, BCL-6 co-repressor; BIRC3, baculoviral IAP repeat-containing protein 3; BRCC3, BRCA1/BRCA2-containing complex subunit 3; C-NOTCH, carboxy-terminal domain of NOTCH; CARD11, caspase recruitment domain-containing protein 11; CHD2, chromodomainhelicase-DNA-binding protein 2; CHK2, checkpoint kinase 2; Co-A, co-activator; CSL, CBF1--Suppressor of Hairless--LAG1 (also known as RBPJ); DDX3X, ATP-dependent RNA helicase DDX3X; DYRK1A, dual-specificity tyrosine-phosphorylation-regulated kinase 1A; EGR2, early growth response 2; ELF4, ETS-related transcription factor Elf-4; ERK, extracellular signal-regulated kinase; EWSR1, Ewing sarcoma breakpoint region 1 protein; FBXW7, F-box/WD repeat-containing protein 7; FUBP1, far upstream element-binding protein 1; GNB1, guanine nucleotide-binding protein G(I)/G(S)/G(T) subunit Î²1; H3K4, histone H3 lysine 4; IC, intracellular domain; IKZF3, Ikaros family zinc-finger protein 3; IL-1R, IL-1 receptor; IRF4, interferon regulatory factor 4; ITPKB, inositol-trisphosphate 3-kinase B; LRP, low-density lipoprotein receptor-related protein; MAP2K1, dual-specificity mitogen-activated protein kinase kinase 1; MAPK, mitogen-activated protein kinase; MED12, Mediator of RNA polymerase II transcription subunit 12; MGA, MAX gene-associated protein; MYD88, myeloid differentiation primary response protein MyD88; NF-ÎºB, nuclear factor-ÎºB; NXF1, nuclear RNA export factor 1; P, phosphate; POT1, protection of telomeres protein 1; PTPN11, tyrosine-protein phosphatase non-receptor type 11; RIPK1, receptor-interacting serine/threonine-protein kinase 1; RPS15, 40S ribosomal protein S15; SAMHD1, SAM domain and HD domain-containing protein 1; SF3B1, splicing factor 3B subunit 1; SHP1, Src homology region 2 domain-containing phosphatase 1 (also known as PTPN6); SYK, spleen tyrosine kinase; TCF/LEF, T cell factor/lymphoid enhancer factor; TLR8, Toll-like receptor 8; TNFR1, tumour necrosis factor receptor 1 (also known as TNFRSF1A); TRAF, TNFR-associated factor; XPO, exportin; ZMYM3, zinc-finger MYM-type protein 3. Adapted with permission from REF. 51, Macmillan Publishers Limited.'' \_Figure from (\protect\hyperlink{ref-Kipps2017_}{\textbf{Kipps2017\_?}})
\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{figures/genetic_mutations} 

}

\caption{(ref:CLLmutations)}\label{fig:CLLmutations}
\end{figure}
\hypertarget{intro-trisomy12}{%
\subsection{The incompletely understood role of trisomy12}\label{intro-trisomy12}}

\hypertarget{stratification-of-cll-patients}{%
\subsection{Stratification of CLL patients}\label{stratification-of-cll-patients}}

\ref{fig:CLLstratification} Taken from Kipps et al 2017: ``Indications for therapy of patients with chronic lymphocytic leukaemia (CLL) include late-stage disease, evidence for rapid disease progression or disease-related symptoms. Patients with del(17p) or mutated TP53 should be treated with therapy that does not require functional TP53, such as ibrutinib (a Bruton tyrosine kinase (BTK) inhibitor), given the relatively poor outcome for such patients with chemotherapy. For patients without del(17p) or known mutations in TP53, immunoglobulin heavy-chain variable region (IGHV) mutational status can help to define the treatment strategy; patients with unmutated IGHV could be considered for therapy with a BTK inhibitor (such as ibrutinib) and patients with mutated IGHV might be good candidates for chemoimmunotherapy (CIT), if amenable. Indeed, patients with mutated IGHV can have excellent outcomes with CIT regimens, such as fludarabine, cyclophosphamide and rituximab, with \textgreater50\% of patients having a median progression-free survival of \textgreater10 years, including the potential for cure. If the patient is amenable to CIT, age, medical comorbidities and myeloid reserve should be taken into consideration. Patients \textgreater65 years of age commonly have medical comorbidities and are less able to tolerate myelosuppressive regimens, such as fludarabine, cyclophosphamide and rituximab. Thus, considerations should be given to using reduced dose or less myelosuppressive chemotherapy regimens, such as chlorambucil or reduced-dose bendamustine and an anti-CD20 monoclonal antibody for patients with limited myeloid reserve. Patients who either do not respond, have a poor tolerance to CIT or relapse following CIT, should be re-evaluated for del(17p) or TP53 mutations. Patients who develop de novo del(17p) or TP53 mutations, or have known del(17p) and/or TP53 mutations, or who develop resistance or intolerance to ibrutinib, could be considered for therapy with idelalisib and rituximab or the BCL-2 inhibitor venetoclax. Patients treated with CIT who do not have del(17p) or TP53 mutations could be considered for repeat CIT if their progression-free survival after CIT is \textgreater2 years and the patient has sufficient myeloid reserve. Such patients also might be treated with a BTK inhibitor or a phosphoinositide 3-kinase (PI3K) inhibitor, which also could be considered for patients who develop intolerance or resistance to therapy with ibrutinib. Patients who develop resistance or intolerance to inhibitors of BTK, PI3K and/or BCL-2 should be considered for clinical trials or alternative agents. LDT, lymphocyte doubling time.'' \emph{Figure from \protect\hyperlink{ref-Kipps2017}{Kipps et al.} (\protect\hyperlink{ref-Kipps2017}{2017}) }.
\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{figures/disease_stratification} 

}

\caption{(ref:CLLstratification)}\label{fig:CLLstratification}
\end{figure}
\hypertarget{the-tumour-microenvironment-in-cll}{%
\section{The tumour microenvironment in CLL}\label{the-tumour-microenvironment-in-cll}}

\hypertarget{the-concept-of-the-protective-niche}{%
\subsection{The concept of the protective niche}\label{the-concept-of-the-protective-niche}}

\ref{fig:microenvironmentTissues} \emph{Adapted from Davids et al 2012}
\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{figures/microenvironmentTissues} 

}

\caption{(ref:microenvironmentTissues)}\label{fig:microenvironmentTissues}
\end{figure}
\hypertarget{intro-ME-pathways}{%
\subsection{Microenvironmental pathways and cross-talk}\label{intro-ME-pathways}}

\ref{fig:microenvironmentOverview} \emph{Adapted from: Weistner et al 2015}
\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{figures/microenvironmentCrosstalk} 

}

\caption{(ref:microenvironmentOverview)}\label{fig:microenvironmentCrosstalk}
\end{figure}
\ref{fig:microenvironmentSignalling} Taken from Kipps et al 2017; ``Migration of chronic lymphocytic leukaemia (CLL) cells into the lyphoid tissu is primarily mediated through CXC-chemokine receptor 4 (CXCR4) in response to CXC-chemokine ligand 12 (CXCL12), which is secreted mainly by nurse-like cells (NLCs) and mesenchymal-derived stromal cells. Migration of CLL cells into lymph nodes also occurs via CC-chemokine receptor 7 (CCR7) in response to CC-chemokine ligand 19 (CCL19) and CCL21, which are produced by the endothelial cells of high endothelial venules (HEVs). HEV endothelial cells also express hyaluronan, which can interact with CD44, to facilitate B cell signalling and might enhance the production of active matrix metalloproteinase 9 (MMP9). Once in tissues, several chemokines promote B cell survival, including CXCL12, B cell-activating factor (BAFF; also known as TNFSF13B) and a proliferation-inducing ligand (APRIL; also known as TNFSF13). In addition, CLL cell survival can be promoted through cognate interactions between CD31 and CD38, and the production by stromal cells of WNT factors, which can interact with ROR1, ROR2 and/or various Frizzled receptors. CLL cell contact with mesenchymal stromal cells can also be established through vascular cell adhesion protein 1 (VCAM1)-- Î±4Î² 1 integrin interactions that contribute to CLL cell survival. In turn, CLL cells can secrete chemokines, such as CCL3 and CCL4, which can recruit T cells and NLC-precursor cells (monocytes) to the CLL microenvironment. Activated T cells can provide CLL cells with proliferative signals through CD40 ligand (CD40L)-CD40 interactions and the secretion of several cytokines, such as IL-2, IL-4 and IL-10. In turn, activated CLL cells secrete CCL12 and CCL22, which attract more T cells into the CLL microenvironment. In tissues, CLL cells can be exposed to environmental and/or self-antigens that might trigger B cell activation through interactions with the surface immunoglobulin; this could amplify the responsiveness of CLL cells to the signals and factors that are provided by the CLL microenvironment. BAFFR, BAFF receptor (also known as TNFRSF13C); BCMA, B cell maturation protein (also known as TNFRSF17); BCR, B cell receptor; TACI, transmembrane activator and CAML interactor (also known as TNFRSF13B). \emph{Figure from \protect\hyperlink{ref-Kipps2017}{Kipps et al.} (\protect\hyperlink{ref-Kipps2017}{2017}) .}''
\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{figures/microenivironmental_signalling} 

}

\caption{(ref:microenvironmentSignalling)}\label{fig:microenvironmentSignalling}
\end{figure}
\hypertarget{the-impact-of-the-microenvironment-on-drug-response}{%
\subsection{The impact of the microenvironment on drug response}\label{the-impact-of-the-microenvironment-on-drug-response}}

\hypertarget{modelling-the-tumour-microenvironment}{%
\subsection{Modelling the tumour microenvironment}\label{modelling-the-tumour-microenvironment}}

\hypertarget{therapies-in-cll}{%
\section{Therapies in CLL}\label{therapies-in-cll}}

\hypertarget{new-targeted-therapies}{%
\subsection{New Targeted therapies}\label{new-targeted-therapies}}

\hypertarget{drug-resistances-in-cll}{%
\subsection{Drug resistances in CLL}\label{drug-resistances-in-cll}}

incl ME -mediated and ibrutinib - IL4

\hypertarget{bet-inhibitors-and-their-potential-for-the-treatment-of-leukemia}{%
\subsection{BET inhibitors, and their potential for the treatment of leukemia}\label{bet-inhibitors-and-their-potential-for-the-treatment-of-leukemia}}

\hypertarget{background-to-the-approaches-used-in-this-thesis}{%
\section{Background to the approaches used in this thesis}\label{background-to-the-approaches-used-in-this-thesis}}

\hypertarget{ex-vivo-drug-sensitivity-screens}{%
\subsection{Ex-vivo drug sensitivity screens}\label{ex-vivo-drug-sensitivity-screens}}

\hypertarget{multiomics-datasets-to-study-cll}{%
\subsection{Multiomics datasets to study CLL}\label{multiomics-datasets-to-study-cll}}

The power of multi-omics approaches especially the use of ATAC sequencing and combining ATAC and RNA
ATAC sequncing as a proxy for TF activity justification \{\#intro-ATACseq\}
PUt with section on this?

\hypertarget{cll-as-a-model-for-studying-tumour-biology}{%
\subsection{CLL as a model for studying tumour biology}\label{cll-as-a-model-for-studying-tumour-biology}}

\hypertarget{measurement-of-outcomes}{%
\subsection{Measurement of outcomes?}\label{measurement-of-outcomes}}

PUt with section on this?

\hypertarget{intro-multivariate-modelling}{%
\subsection{Mathematical modelling}\label{intro-multivariate-modelling}}

\hypertarget{aims-of-this-thesis}{%
\section{Aims of this thesis}\label{aims-of-this-thesis}}

\hypertarget{overview-of-the-approach}{%
\section{Overview of the approach}\label{overview-of-the-approach}}

\hypertarget{note-on-contributions}{%
\section{Note on contributions}\label{note-on-contributions}}

\#\#Reading list before defence0:

\hypertarget{methods}{%
\chapter{Methods}\label{methods}}

Methods in quotation marks are taken from Bruch \& Giles et al.~2021 and I have authored the original text, unless stated otherwise. Paragraphs without quotation marks were rewritten for the thesis.

Routine:
all headings should be here
at the end of each chapter, make sure all methods are included
Copy things in and put in quotation marks if from Bruch \& Giles et al.~2021, mark if they were not written by me, or if they were written jointly
Add REFERENCE and TABLE for now
Return to this later and neaten / expand on methods, and provide all referencing and tables
-fine to use quotations (although check plaigerism email), suggested rewriting as i / he / pmb throughout

\hypertarget{experimental-methods}{%
\section{Experimental methods:}\label{experimental-methods}}

\hypertarget{drug---stimulus-combinatorial-pertubation-assay}{%
\subsection{Drug - Stimulus Combinatorial Pertubation Assay}\label{drug---stimulus-combinatorial-pertubation-assay}}

\hypertarget{patient-sample-preparation}{%
\subsubsection{Patient Sample Preparation}\label{patient-sample-preparation}}

\emph{with Peter-Martin Bruch} Original
Peripheral Blood was taken from 192 patients for the initial drug-stimulation assay. Blood was separated by Ficoll gradient (GE Healthcare) and mononuclear cells were cryopreserved.

\hypertarget{preparation-of-screening-plates}{%
\subsection{Preparation of screening plates}\label{preparation-of-screening-plates}}

\emph{with Peter-Martin Bruch} Original
``Sample preparation, cell-culture, drug-stimulation profiling and genomic annotation was performed on 192 CLL patient samples as previously described{[}REFERENCE{]} with the following adjustments. Stimuli and drugs were mixed and preplated in the culture plates directly before adding the cell suspensions. RPMI-1640 and supplements were acquired from Gibco by Life Technologies, human serum was acquired from PAN Biotech (Cat.No. P40-2701, Lot.No:P-020317). Luminescence was measured after 48h on a Perkin Elmer EnVision.''

\emph{with Peter-Martin Bruch} Original
Compounds were obtained from Selleckchem, MedChemExpress and Sigma-Aldrich, dissolved in DMSO and stored at -\(20^\circ\)C. 12 drugs were used in two concentrations {[}TABLE{]}. Final DMSO concentration did not exceed 0.3\% in all experiments. Carfilzomib, Panobinostat and Venetoclax were removed from the analysis as they showed inconsistent toxicity depending on used media.
Insert table

Recombinant cytokines and stimulatory agents were dissolved according to manufacturer's protocol. 17 stimuli were selected {[}TABLE{]}. HS-5 conditioned medium was produced by incubating HS-5 stroma cell line (gifted by Martina Seiffert, DKFZ, Heidelberg) for 4 days at 37\(^\circ\)C and 5\% CO2, after which the supernatant was centrifuged and stored at -20\(^\circ\)C, the final concentration of HS-5 CM was 20\%. Bead immobilised anti-IgM was removed from the analysis due to storage instability. ''

Paper:
``Compounds and stimulatory agents were dissolved, stored and diluted according to manufacturer's protocol. HS-5 conditioned medium was produced by incubating HS-5 stromal cell line to \textgreater80\% confluency and cell removal by centrifugation. For a detailed list of stimuli and drugs and associated concentrations see Supp. Tables 1 and 2. Final DMSO concentration did not exceed 0.3\%.''

Original
Compounds were preplated in 384-well polypropylene storage plates (Greiner Bio-One Cat.No.:781271), which were stored at -20\(^\circ\)C. For each batch of samples tested on the same day a new storage plate was thawed and diluted in serum free RPMI, with or without stimuli. 5\(\mu\)L of drug-stimulation dilution were added into each well of the assay plates, 20 \(\mu\)L of cell suspension were added. The final cell concentration was 8*10\^{}5 cells/ml. Cells were thawed as previously described{[}REFERENCE{]}.

\hypertarget{drug-stimulation-assay}{%
\section{Drug-Stimulation assay}\label{drug-stimulation-assay}}

\emph{with Peter-Martin Bruch} Original
Drug-stimulation assays were performed with RPMI-1640 (Gibco by Life Technologies) supplemented with Penicillin Streptomycin (Gibco) final concentration of 100 Units/ml and 100 \(\mu\)g/ml respectively, L-Glutamine (Gibco) final concentration 2mM, and 10\% pooled, heat-inactivated and sterile filtered human type AB male off the clot serum (PAN Biotech, Cat.No. P40-2701, Lot.No:P-020317).

Drugs and stimuli were added to the cells simultaneously. Culture was performed in 384-well plates (Greiner Bio-One Cat.No.: 781904) and cells were incubated at 37\(^\circ\)C and 5\% CO2 for 48h.

Cell Viability was determined using the ATP-based CellTiter-Glo assay (Promega, Cat.No.:G7573). Luminescence was measured for the drug-stimulation assays using a Perkin Elmer EnVision and for the drug-drug interaction experiments using a Perkin Elmer EnSight, with a measurement time of 100ms per well.

\hypertarget{wes-rna-sequencing-targeted-sequencing-and-dna-copy-number-variants}{%
\section{WES, RNA Sequencing, Targeted Sequencing and DNA Copy number variants}\label{wes-rna-sequencing-targeted-sequencing-and-dna-copy-number-variants}}

\emph{Myself} Original
Sequencing data generated by Dietrich et al{[}REFERENCE{]}, and all sequencing and data processing were performed as described there. For

\hypertarget{follow---up-investigations}{%
\subsection{Follow - up investigations}\label{follow---up-investigations}}

\hypertarget{lymphocyte-doubling-time}{%
\subsubsection{Lymphocyte doubling time}\label{lymphocyte-doubling-time}}

\hypertarget{survival-data}{%
\subsubsection{Survival data}\label{survival-data}}

\hypertarget{atacsequencing-of-4-cll-samples}{%
\subsubsection{ATACsequencing of 4 CLL samples}\label{atacsequencing-of-4-cll-samples}}

\emph{with Peter-Martin Bruch (mostly edited by me)} Paper
``Peripheral blood was taken from 4 CLL patients and separated by Ficoll gradient (GE Healthcare), mononuclear cells were cryopreserved on liquid nitrogen. Samples were later thawed from frozen as previously described{[}REFERENCE{]} and MACS sorted for CD19 positive cells (Milteny autoMACS). The cells were resuspended in RPMI (GIBCO, Cat.No. 21875-034), with the addition of 2mM glutamine (GIBCO, Cat.No. 25030-24), 1\% Pen/Strep (GIBCO, Cat.No. 15140-122) and 10\% pooled, heat-inactivated and sterile filtered human type AB male off the clot serum (PAN Biotech, Cat.No. P40-2701, Lot.No:P-020317). 5ml of cell suspension was cultured in 6-well plates (Greiner Bio-One Cat.No. 657160). After thawing, cells were incubated at 37\(^\circ\)C and 5\% CO2 for 6 hours in 0.2\% DMSO. The final cell concentration was 2x10Ë6 cells/ml. Cell viability and purity was assessed using FACS. All samples had a viability over 90\% and over 95\% of CD19+/CD5+/CD3- cells.''

\emph{with Nayara} Paper
``ATAC-seq libraries were generated as described previously{[}REFERENCE{]}. Cell preparation and transposition was performed according to the protocol, starting with 5x10Ë4 cells per sample. Purified DNA was stored at â20\(^\circ\)C until library preparation was performed. To generate multiplexed libraries, the transposed DNA was initially amplified for 5x PCR cycles using 2.5 \(\mu\)L each of 25 Î¼M PCR Primer 1 and 2.5 \(\mu\)L of 25 \(\mu\)M Barcoded PCR Primer 2 (included in the Nextera index kit, Illumina, San Diego, CA, USA), 25 \(\mu\)L of NEBNext High-Fidelity 2x PCR Master Mix (New England Biolabs, Boston, Massachusetts) in a total volume of 50 \(\mu\)L. 5 \(\mu\)L of the amplified DNA was used to determine the appropriate number of additional PCR cycles using qPCR. Additional number of cycles was calculated through the plotting of the linear Rn versus cycle, and corresponds to one-third of the maximum fluorescent intensity. Finally, amplification was performed on the remaining 45 \(\mu\)L of the PCR reaction using the optimal number of cycles determined for each library by qPCR (max. 13 cycles in total). The amplified fragments were purified with two rounds of SPRI bead clean-up (1.4x). The size distribution of the libraries was assessed on Bioanalyzer with a DNA High Sensitivity kit (Agilent Technologies, Santa Clara, CA), concentration was measured with Qubit DNA High Sensitivity kit in Qubit 2.0 Flurometer (Life Technologies, Carlsbad, CA). Sequencing was performed on NextSeq 500 (Illumina, San Diego, CA, USA) using 75bp paired-end sequencing, generating â¼450 million paired-reads per run, with an average of 55 million reads per sample.''

\hypertarget{spi-b-and-pu.1-shrna-knockdowns}{%
\subsubsection{Spi-B and PU.1 shRNA Knockdowns}\label{spi-b-and-pu.1-shrna-knockdowns}}

\hypertarget{immunohistochemistry-of-patient-lymph-nodes}{%
\subsubsection{Immunohistochemistry of patient lymph nodes}\label{immunohistochemistry-of-patient-lymph-nodes}}

\emph{Peter-Martin Bruch} Paper
``LN biopsies of CLL-infiltrated and non-neoplastic lymph nodes were Paraffin fixed and arranged in Tissue Microarrays. Consecutively they were stained for PAX-5 (790-4420, Roche), CD3 (790-4341, Roche), pSTAT6 (ab28829, abcam), STAT6 (519-4290, Zytomed Systems) and pIRAK4 (ab216513, abcam). The slides were analysed using Qupath 50 and the recommended protocol. Cell based data on mean staining intensity was exported and further analysed using R.''

\hypertarget{ibrutinib---il4---stat6i-interaction-assay}{%
\subsubsection{Ibrutinib - IL4 - STAT6i interaction assay}\label{ibrutinib---il4---stat6i-interaction-assay}}

\hypertarget{investigation-of-ibrutinib-il4-ibet-762}{%
\subsection{Investigation of Ibrutinib + IL4 + IBET-762}\label{investigation-of-ibrutinib-il4-ibet-762}}

\hypertarget{ibrutinib---il4---ibet-762-interaction-assay}{%
\subsubsection{Ibrutinib - IL4 - IBET-762 interaction assay}\label{ibrutinib---il4---ibet-762-interaction-assay}}

\hypertarget{ibrutinib---il4---ibet--762-treatment}{%
\subsubsection{Ibrutinib - IL4 - IBET -762 treatment}\label{ibrutinib---il4---ibet--762-treatment}}

\hypertarget{atacsequencing}{%
\subsubsection{ATACsequencing}\label{atacsequencing}}

\emph{with Peter Martin Bruch} get from diffTF
Peripheral blood was taken from 4 CLL patients and separated by Ficoll gradient (GE Healthcare), mononuclear cells were cryopreserved on liquid nitrogen. Samples were later thawed from frozen as previously described{[}REFERENCE{]} and MACS sorted for CD19 positive cells (Milteny autoMACS). The cells were resuspended in RPMI (GIBCO, Cat.No. 21875-034), with the addition of 2mM glutamine (GIBCO, Cat.No. 25030-24), 1\% Pen/Strep (GIBCO, Cat.No. 15140-122) and 10\% pooled, heat-inactivated and sterile filtered human type AB male off the clot serum (PAN Biotech, Cat.No. P40-2701, Lot.No:P-020317). 5ml of cell suspension was cultured in 6-well plates (Greiner Bio-One Cat.No. 657160). After thawing, cells were incubated at 37\(^\circ\)C and 5\% CO2 for 6 hours in 0.2\% DMSO. The final cell concentration was 2x10Ë6 cells/ml. Cell viability and purity was assessed using FACS. All samples had a viability over 90\% and over 95\% of CD19+/CD5+/CD3- cells.

\emph{with Nayara} get from diffTF
ATAC-seq libraries were generated as described previously{[}REFERENCE{]}. Cell preparation and transposition was performed according to the protocol, starting with 5x10Ë4 cells per sample. Purified DNA was stored at â20\(^\circ\)C until library preparation was performed. To generate multiplexed libraries, the transposed DNA was initially amplified for 5x PCR cycles using 2.5 \(\mu\)L each of 25 \(\mu\)M PCR Primer 1 and 2.5 \(\mu\)L of 25 \(\mu\)M Barcoded PCR Primer 2 (included in the Nextera index kit, Illumina, San Diego, CA, USA), 25 \(\mu\)L of NEBNext High-Fidelity 2x PCR Master Mix (New England Biolabs, Boston, Massachusetts) in a total volume of 50\(\mu\)L. 5 \(\mu\) L of the amplified DNA was used to determine the appropriate number of additional PCR cycles using qPCR. Additional number of cycles was calculated through the plotting of the linear Rn versus cycle, and corresponds to one-third of the maximum fluorescent intensity. Finally, amplification was performed on the remaining 45 \(\mu\)L of the PCR reaction using the optimal number of cycles determined for each library by qPCR (max. 13 cycles in total). The amplified fragments were purified with two rounds of SPRI bead clean-up (1.4x). The size distribution of the libraries was assessed on Bioanalyzer with a DNA High Sensitivity kit (Agilent Technologies, Santa Clara, CA), concentration was measured with Qubit DNA High Sensitivity kit in Qubit 2.0 Flurometer (Life Technologies, Carlsbad, CA). Sequencing was performed on NextSeq 500 (Illumina, San Diego, CA, USA) using 75bp paired-end sequencing, generating â¼450 million paired-reads per run, with an average of 55 million reads per sample.

\hypertarget{rnasequencing}{%
\subsubsection{RNAsequencing}\label{rnasequencing}}

\hypertarget{proteomics}{%
\subsubsection{Proteomics}\label{proteomics}}

\hypertarget{data-availability}{%
\section{Data availability}\label{data-availability}}

European Genome-Phenome Archive (EGA) accession \ldots. The data for \ldots. and computational analysis code used in this study are available from \ldots.

Additionally, we obtained CLL ATACseq data{[}REFERENCE{]} from the European Genome-phenome Archive (EGA: EGAD00001002110) and from (Bruch \& Giles et al.~2021)
For the ChIPseq analysis of SPIB binding sites, we accessed the data 37 from the NCBI GEO database 51, accession GEO: GSE56857. We made use of the data for SPIB in the OCILY3 DLBCL cell line (GSM1370276).

\hypertarget{statistical-analysis}{%
\section{Statistical Analysis}\label{statistical-analysis}}

Mixutre of paper and original
Integrative data analysis of screening data, DNA and RNA sequencing, CNV and methylation profiles and follow up experiments was performed using R version 4 (REFERENCE R Core Team, 2018) with the RStudio interface (REFERENCE RStudio Team, 2016) and using packages that included (\protect\hyperlink{ref-R-DESeq2}{Love, Anders, and Huber 2021}), (\protect\hyperlink{ref-R-survival}{Therneau 2021}), (\protect\hyperlink{ref-R-glmnet}{Friedman et al. 2021}), (\protect\hyperlink{ref-R-ConsensusClusterPlus}{M. Wilkerson and Waltman 2021}), (\protect\hyperlink{ref-R-clusterProfiler}{Yu 2021b}), (\protect\hyperlink{ref-R-ChIPseeker}{Yu 2021a}), (\protect\hyperlink{ref-R-genomation}{Akalin et al. 2021}) and (\protect\hyperlink{ref-R-BloodCancerMultiOmics2017}{Oles et al. 2021}) to perform univariate association tests, multivariate regression with and without lasso penalization, Cox regression, generalised linear modelling and clustering.

\hypertarget{data-processing}{%
\subsection{Data processing}\label{data-processing}}

\hypertarget{processing-of-screening-data}{%
\subsubsection{Processing of screening data}\label{processing-of-screening-data}}

\emph{with Peter-Martin Bruch} Paper
To quantify the response of a patient sample to each treatment condition, we used viability relative to the control, i.e., the CellTiter Glo luminescence readout of the respective well divided by the median of luminescence readouts of the DMSO control wells on the same plate.

\hypertarget{quality-assessment-and-control}{%
\subsubsection{Quality Assessment and Control}\label{quality-assessment-and-control}}

\hypertarget{rna-data}{%
\subsubsection{RNA data}\label{rna-data}}

RNA data was taken from the PACE repository (\protect\hyperlink{ref-PACE}{\textbf{PACE?}}). For detailed explanaition of processing see (\protect\hyperlink{ref-PACE}{\textbf{PACE?}}).

\hypertarget{atacsequencing-1}{%
\subsubsection{ATACsequencing}\label{atacsequencing-1}}

\hypertarget{data-from-pace}{%
\subsubsection{Data from PACE}\label{data-from-pace}}

\hypertarget{analysis-of-screening-data}{%
\subsection{Analysis of Screening Data}\label{analysis-of-screening-data}}

\hypertarget{correlations}{%
\subsubsection{Drug-drug and stimulus - stimulus correlations}\label{correlations}}

\emph{myself} original complete
Section \ref{cytokine-profiling}. Pearson correlation coefficients were calculated for each drug - drug and stimulus - stimulus pair, using the \texttt{cor} functions of the (\protect\hyperlink{ref-R-base}{R Core Team 2021}) package with log transformed viability values which were normalised to untreated controls.

\hypertarget{stimulus-heatmap-method}{%
\subsubsection{Characterising stimulus responses across all samples}\label{stimulus-heatmap-method}}

\emph{myself} original complete (just check)
Section \ref{clusters}. For the heatmap in figure \ref{fig:stimuliHeatmap}, the viability data is represented as z scores. Log transformed viability values, normalised to DMSO controls, are row-scaled according to the Median Absolute Deviance. Limits were then applied to this row scaling factor for optimal visualisation. The matrix of z scores was plotted using the (\protect\hyperlink{ref-R-pheatmap}{Kolde 2019}) package. The ordering of the columns (patient samples) was obtained from the dendrogram that resulted from running \texttt{ConsensusClusterPlus}, from the (\protect\hyperlink{ref-R-ConsensusClusterPlus}{M. Wilkerson and Waltman 2021}) package. The rows were ordered using the dendrogram order produced by \texttt{hclust} with default branch arrangement.

ADD detial on clustering

\hypertarget{univariate-gene-stimulus-associations-method}{%
\subsubsection{Univariate analysis of gene - stimulus and gene - drug response assosciations}\label{univariate-gene-stimulus-associations-method}}

\emph{myself} writtten for thesis complete
Section \ref{univariate-gene-stimulus-associations} \& \ref{univariate-gene-drug-associations}. Two-sided Student's t-tests, with equal variance were performed for IGHV status and somatic mutations and copy number aberrations with â¥3 patient samples in each group (n = 63/54). A 10\% FDR cut off was used to determine significance.

\hypertarget{multivariate-gene-stimulus-assosciations-method}{%
\subsubsection{Penalised multivariate regression of gene - stimulus and gene - drug assosciations}\label{multivariate-gene-stimulus-assosciations-method}}

\emph{myself} written for thesis (but some overlap with JCI)
Section \ref{multivariate-gene-stimulus-assosciations} \& \ref{multivariate-gene-drug-assosciations}.\\
ADAPT FOR DRUGS ALSO A Gaussian linear model with L1-penalty implemented in the \texttt{R} package \texttt{glmnet} (\protect\hyperlink{ref-R-glmnet}{Friedman et al. 2021}), with mixing parameter alpha = 1,was used to identify gene-stimulus associations. Matrix of genetic features (p=39), IGHV status (encoded as M = 1 and U = 0), and Methylation Cluster (encoded as 0, 0.5, 1) were used to identify multivariate predictors. All features were thus encoded on a similar scale to ensure equal treatment by lasso constraint in model fitting. Genetic features with more than 20\% missing values were excluded from the analysis, and samples without complete annotation for remaining features were removed, resulting in n = 129 samples. The matrix of control-normalised log-transformed viability values for these 129 samples was provided as the dependent variable. Using 3-fold cross-validation, the optimal penalty parameter \(\lambda\) was selected so as to minimise the cross-validated R2. The reduction in cross-validated mean squared error compared to the null model was used as loss. The model was fitted for 30 bootstrapped repeats, and the resulting coefficients are the mean of those coefficients that were selected in \textgreater75\% model fits.

\hypertarget{correlation-of-rna-receptor-expression-with-viability}{%
\subsubsection{Correlation of RNA-Receptor Expression with viability}\label{correlation-of-rna-receptor-expression-with-viability}}

\emph{myself} original complete
RNA count data for matched samples was available for 49 patients and was transformed using the variance stabilising transformation. Stimulus - receptor pairs were defined using the available literature, see table \ref{tab:receptorPairs}. For each stimulus, pearson correlation coefficient were calculated between the control-normalised log-transformed viability value of a sample and its expression of the corresponding stimulus receptor.

\hypertarget{drug-stimulus-linear-model-method}{%
\subsubsection{Linear modelling of drug - stimulus interactions}\label{drug-stimulus-linear-model-method}}

\emph{myself} original
Section \ref{drug-stimulus-linear-model} Linear models were fitted for each drug - stimulus combination, to extract a \(\beta{int}\) term and assossciated p value for each combinatorial treatment. Linear model was fitted using equation \eqref{eq:drugCytInt}, using the \texttt{lm} function of the \texttt{R} package stats (\protect\hyperlink{ref-R-base}{R Core Team 2021}):
\begin{equation}
            log(V) = \beta_{drug}X_{drug} + \beta_{stimulus}X_{stimulus} + \beta_{int}X_{drug}X_{stimulus} + \epsilon
                                       \label{eq:drugCytInt}
    \end{equation}
\emph{where \(V\) is the predicted viability with a given treatment,} \(\beta_{drug}\), \(\beta_{stimulus}\) \emph{and} \(\beta_{int}\) \emph{are coefficients for the drug, stimulus and combinatorial terms and} \(X_{drug}\) \emph{and} \(X_{stimulus}\) \emph{are indicator variables (0 or 1) for the presence or absence of a drug/stimulus.} \(\epsilon\) \emph{is the vector of model residuals. Equation from Bruch \& Giles et al.~2021.}

To fit model, the matrix of log-transformed viability values, for control, single and combinatorial treatments was used. Interactions were defined as significant if p value for \(\beta{int}\) \textless{} 0.05. The matrix of significant \(\beta{int}\) was then plotted as a heatmap, with the package \texttt{pheatmap}(\protect\hyperlink{ref-R-pheatmap}{Kolde 2019}) where the rows (stimuli) and columns (drugs) were ordered according to the dendrogram order produced by \texttt{hclust} (\protect\hyperlink{ref-R-base}{R Core Team 2021}) using default branch arrangement.

To define the four interaction categories, drug - stimulus combinations were first divided with respect to the sign of \(\beta{int}\), whereby a positive \(\beta{int}\) indicates that the viability with combinatorial treatment is higher than would be expected based on additive effects alone and vice versa. The groups were further divided into synergies and antagonisms according to the values of the model coefficients (\(\beta{drug}\), \(\beta{stimulus}\) and \(\beta{int}\)). Synergisms were assigned when coefficients for single treatments (\(\beta{drug}\) and \(\beta{stimulus}\)) were both greater than, or both less than, the observed coefficient for the combinatorial treatment (i.e.~\(\beta{drug}\) + \(\beta{stimulus}\) + \(\beta{int}\)). For positive antagonisms, \(\beta{drug}\) + \(\beta{stimulus}\) + \(\beta{int}\) was less than either \(\beta{drug}\) or \(\beta{stimulus}\). For negative antagonisms, \(\beta{drug}\) + \(\beta{stimulus}\) + \(\beta{int}\) was greater than either \(\beta{drug}\) or \(\beta{stimulus}\). All drug - stimulus interactions, for which p value for \(\beta{int}\) \textless0.05 fit into one of these groups.

\hypertarget{drug-stimulus-gene-interactions-method}{%
\subsubsection{Modelling of drug - stimulus - gene interactions}\label{drug-stimulus-gene-interactions-method}}

\emph{myself} original
Section \ref{drug-stimulus-gene-interactions}. To identify drug - stimulus interactions that were dependent on genetic features, we first fitted linear models to the viability matrix, for each drug - stimulus combination, using the formula in Eqn2. We extracted drug - stimulus interaction coefficients for each patient to generate a response matrix for n = 137 patients. Next, we performed multivariate regression using a Gaussian linear model with L1-penalty (i.e., lasso regression) as implemented in the R package \texttt{glmnet}(\protect\hyperlink{ref-R-glmnet}{Friedman et al. 2021}). As the dependent variable, the matrix of drug - stimulus interaction coefficients for each patient sample was used. As input to the model, genetic features with more than 20\% missing values were excluded, and only patients with complete annotation were included in the model ( n= 137 ). As predictors, the genetic mutations and CNVs (p= 39), IGHV status (coded as 0-1) and Methylation CLuster (coded as 0, 0.5, 1) were used, using 3-fold cross-validation. Misclassification error was used as loss for cross- validation.

Equation \eqref{eq:drugCytGeneInt} quantifies how the viability with any combination can be predicted:
\begin{equation}
            log(V) = \beta_{drug}X_{drug} + \beta_{stimulus}X_{stimulus} + \beta_{patient}X_{patient} + \beta_{drug-stimulus}X_{drug}X_{stimulus} + \beta_{drug-patient}X_{drug}X_{patient} + \beta_{stimulus-patient}X_{stimulus}X_{patient} +\beta_{int}X_{drug}X_{stimulus}X_{patient} + \epsilon
                                       \label{eq:drugCytGeneInt}
    \end{equation}
\emph{where \(V\) is the predicted viability of a patient sample with a given treatment,} \(\beta_{drug}\), \(\beta_{stimulus}\),\(\beta_{int}\), \(\beta_{drug-stimulus}\),\(\beta_{drug-patient}\),\(\beta_{drug-stimulus}\) \emph{and} \(\beta_{int}X_{drug}\) \emph{are regression coefficients for the drug, stimulus, patient sample and combinatorial terms and} \(X_{drug}\), \(X_{stimulus}\) \emph{and} \(X_{patient}\) \emph{are indicator variables (0 or 1) for the presence or absence of a drug/stimulus/patient sample.} \(\epsilon\) \emph{is the vector of model residuals. See also Methods section \ref{drug-stimulus-gene-interactions-method}. Equation from Bruch \& Giles et al.~2021.}

This generates a higher order interaction term \(\beta_{int}X_{drug}X_{stimulus}X_{patient}\). This term represents a \emph{patient sample-specific} \(\beta{int}\) for each drug - stimulus combination, quantifying the size of an interaction between a drug and stimulus in each patient genetic background.

With these patient sample-specific \(\beta{int}\) terms, it was possible to search for associations between the size of \(\beta_{int}\)) and genetic features. The aim was to screen for molecular features that increased or decreased the size of a drug - stimulus interactions, using multivariate regression with L1 (lasso) regularisation.

As input to the model, the response matrix was composed of the sample - specific \(\beta_{int}\) values for each drug-stimulus combination. To generate the feature matrix (137 samples versus 40 features), I excluded genetic features for which \textgreater20\% of the values were missing, and patient samples with incomplete annotation. As predictors, I included genetic mutations and CNVs (p=39) and IGHV status (coded as 0-1). I ran lasso regression, as implemented in the \texttt{R} package \texttt{glmnet}(\protect\hyperlink{ref-R-glmnet}{Friedman et al. 2021}), using three-fold cross-validation with misclassification error as loss. The resulting predictors are the mean of those coefficients that were selected in at least 90\% of 30 bootstrapped repeats.

\^{}elaborate?
\#\#\# Follow up
\#\#\#\# Lympocyte doubling times \{\#LDT-method\}
\emph{Herbst et al., 2020c} - update
Section \ref{cluster-survival}. ``Patients which had lymphocyte counts available for less than 4 time points between the sample
collection date and the time of the next treatment and patients currently in treatment were ex-cluded. Thus, \ldots{} patients with enough data remained. Lymphocyte growth rates were calculated by fitting a linear model to the log10 transformed lymphocyte counts of all timepoints between the sample collection date and the time of the next treatment versus the period of time.''

\hypertarget{survival-method}{%
\subsubsection{Survival analyses}\label{survival-method}}

\emph{myself} original
Section \ref{cluster-survival}. Survival analyses were performed using TTT, TFT and OS as metrics. Follow-up information to calculate OS was available for all 192 CLL patients. For 188 of 192 CLL patients treatment information after sample collection was available.

\emph{Herbst et al}
''Time to next treatment (TTT) was calculated from the date of sample collection to subsequent
treatment initiation. Patients without treatment initiation during the observation time and
patients who died before treatment initiation were censored at the latest follow-up contact.''

\emph{myself} original
For for the purpose of visualising Kaplan-Meier plots , optimal cut-points of staining levels were calculated using maximally selected rank statistics as computed by the \texttt{R} package \texttt{maxstat} (\protect\hyperlink{ref-R-maxstat}{\textbf{R-maxstat?}}). Based on these cut points, patients were split into two groups, and their survival data were plotted using the Kaplan-Meier method, using the R package \texttt{survminer} (\protect\hyperlink{ref-R-survminer}{Kassambara, Kosinski, and Biecek 2021}).

For Cox proportional hazards regression, the \texttt{coxph} function of the \texttt{R} package \texttt{survival} (\protect\hyperlink{ref-R-survival}{Therneau 2021}) was used. Expand

\hypertarget{cluster-genetics-method}{%
\subsubsection{Penalised multivariate regression genetic predictors of cluster membership}\label{cluster-genetics-method}}

\emph{Myself} original (edit so not similar to JCI)
Section \ref{genetics-method}. I used a multinomial linear model with L1-penalty, implemented in the \texttt{glmnet} (\protect\hyperlink{ref-R-glmnet}{Friedman et al. 2021}) package. As the dependent variable, the cluster assignment for each patient was used. As input to the model ,genetic features with more than 20\% missing values were excluded, and only patients with complete annotation were included in the model ( n= 137 ). As predictors, the genetic mutations and CNVs (p= 39) and IGHV status (coded as 0-1) were used, using 3-fold cross-validation. Misclassification error was used as loss for cross- validation. The resulting coefficients indicated associations between genetic features and each cluster.

\hypertarget{cluster-rna-method}{%
\subsubsection{Gene expression and gene set enrichment analysis between clusters}\label{cluster-rna-method}}

\emph{myself} original
Section \ref{cluster-rna}. For the n=49 patient samples for which viability data and RNA--Seq data for matching samples was available, the \texttt{R} package \texttt{DESeq2} (\protect\hyperlink{ref-R-DESeq2}{Love, Anders, and Huber 2021}) was used to search for associations of these two data types.RNA-Seq read count data was regressed on to the patient clusters C3, C4 (design formula \textasciitilde{} IGHV.status + Cluster). Genes were ranked by their test statistics and Gene Set Enrichment Analysis (GSEA) (implementing the fgsea algorithm with the \texttt{clusterProfiler} (\protect\hyperlink{ref-R-clusterProfiler}{Yu 2021b}) package) was applied to the ranked lists with the KEGG pathway gene set selections from the MSigDBdatabase{[}REFERENCE{]}.

\hypertarget{associations-of-ex-vivo-stimulus-responses-with-genomic-features}{%
\subsubsection{Associations of ex-vivo stimulus responses with genomic features}\label{associations-of-ex-vivo-stimulus-responses-with-genomic-features}}

We tested for associations between stimulus viability assay results and genomic features by Student's t-tests (two-sided, with equal variance). We tested somatic mutations (aggregated at the gene level), copy number aberrations and IGHV status. We restricted the analysis to features that were present in at least 3 patient samples (63 features). p-values were adjusted for multiple testing by applying the Benjamini-Hochberg procedure.

\hypertarget{gene-dosage-effects-method}{%
\subsubsection{Analysis of differential gene dosage in trisomy 12 CLL}\label{gene-dosage-effects-method}}

\emph{myself} written for thesis
Section \ref{gene-dosage-effects}. For all RNA samples available in PACE (i.e.~not just those that matched the samples in the screen), differential expression was called using the \texttt{DESeq2} package (\protect\hyperlink{ref-R-DESeq2}{Love, Anders, and Huber 2021}), with the design formula \textasciitilde trisomy12. Raw RNA counts were visualised if the gene had BH-adjusted p \textless{} 0.1 and belonged to TGF\(\beta\), JAK-STAT or TLR pathways genesets, as defined in the KEGG database (\protect\hyperlink{ref-Kegg}{\textbf{Kegg?}}) downloaded using the \texttt{msigdbr} package (\protect\hyperlink{ref-R-msigdbr}{Dolgalev 2021}). Proteomic abundance data was also plotted, provided by Sophie Herbst (\protect\hyperlink{ref-HerbstThesis}{Herbst 2020}). Proteomics data partially overlaps with RNAseq data. Protein quantification was performed as described in \protect\hyperlink{ref-HerbstThesis}{Herbst} (\protect\hyperlink{ref-HerbstThesis}{2020}).

\hypertarget{trisomy12-classifier-method}{%
\subsubsection{Generation of a Trisomy 12 classifier}\label{trisomy12-classifier-method}}

\emph{edit and finish}
Section \ref{trisomy12-classifier}. To build the classifier, a binomial linear model with L1-penalty implemented in the \texttt{R} package \texttt{glmnet} (\protect\hyperlink{ref-R-glmnet}{Friedman et al. 2021}), with mixing parameter alpha = 1, was used. To identify coefficients that predict trisomy 12 status, the viability matrix for all stimuli, as described in section \ref{stimulus-heatmap-method} was used, i.e.~z scores of the control-normalised log-transformed viability values.

Matrix of genetic features (p=39), IGHV status (encoded as M = 1 and U = 0), and Methylation Cluster (encoded as 0, 0.5, 1) were used to identify multivariate predictors. All features were thus encoded on a similar scale to ensure equal treatment by lasso constraint in model fitting. Genetic features with more than 20\% missing values were excluded from the analysis, and samples without complete annotation for remaining features were removed, resulting in n = 129 samples. The matrix of control-normalised log-transformed viability values for these 129 samples was provided as the dependent variable. Using 3-fold cross-validation, the optimal penalty parameter \(\lambda\) was selected so as to minimise the cross-validated R2. The reduction in cross-validated mean squared error compared to the null model was used as loss. The model was fitted for 30 bootstrapped repeats, and the resulting coefficients are the mean of those coefficients that were selected in \textgreater75\% model fits.

The classifier was built using binomial regression, with lasso penalisation, as implemented in the \texttt{R} package \texttt{glmnet} (\protect\hyperlink{ref-R-glmnet}{Friedman et al. 2021}). The feature matrix consisted of z scores of the viability values after treatment with each stimulus, and was used to predict the response (trisomy 12 status). I ran the model for 50 bootstrapped repeats, using three-fold cross-validation and mean absolute error as loss. Resiquimod, sCD40L+IL4 and TGF\(\beta\) were selected as coefficients that predict trisomy 12 status (Figure \ref{fig:tri12Classfier}, as would be expected bed on the observations in section \ref{trisomy12-modulator}.

\hypertarget{atacseq-processing}{%
\subsubsection{ATACseq processing}\label{atacseq-processing}}

\emph{with Ivan Berest} original (update to paper)
We downloaded CLL data (\protect\hyperlink{ref-Rendeiro2016}{Rendeiro et al. 2016}) from the European Genome-phenome Archive (EGA: EGAD00001002110). Original dataset had 88 ATAC-seq samples from 55 patients, however we used for the analysis only one sample per patient passing quality checks, resulting in 52 samples. We used an in-house constructed ATAC-seq processing pipeline (\protect\hyperlink{ref-Berest2019}{Berest et al. 2019}) to obtain final bam files mapped to the hg19 annotation genome that were corrected for CG bias.

\hypertarget{annotation-of-trisomy-12-status}{%
\subsubsection{Annotation of trisomy 12 status}\label{annotation-of-trisomy-12-status}}

\emph{with Ivan Berest} original (update to paper)
As trisomy 12 status was not included in the original metadata, we used a mean amount of reads in the chromatin accessible peaks for each sample to distinguish trisomy 12 patients. All samples containing 1.4 times more reads in the peaks located on chromosome 12, compared to the peaks on all other chromosomes, were classified as trisomy 12 patients.

\hypertarget{difftf-analysis-of-tf-activity-in-trisomy-12-cll}{%
\subsubsection{diffTF analysis of TF activity in trisomy 12 CLL}\label{difftf-analysis-of-tf-activity-in-trisomy-12-cll}}

\emph{with Ivan Berest} original (update to paper)
FOr the larger ATACseq dataset, we ran analytical mode of diffTF (\protect\hyperlink{ref-R-diffTF}{\textbf{R-diffTF?}}) pipeline to investigate differential TF activity between trisomy 12 and non-trisomy 12 patients using the HOCOMOCO v10 database (\protect\hyperlink{ref-HOCOMOCO}{Kulakovskiy et al. 2016}) with the following design formula: ``\textasciitilde{} Batch + Gender + IGHV status + trisomy12 status.''

for tirsomy 12 software uses the ATACseq counts at each of the binding sites across the genome to generate a distribution of fold changes between trisomy 12 and WT samples. This fold change distribution is compared to a background distribution of fold changes, calculated similarly using ATACseq counts for a GC-matched motif, that does not contain the TF binding motif. The TF is classified as differentially active between trisomy 12 and WT samples where these two distributions are significantly different. Each TF is assigned a weighted mean difference value, which quantifies the change in activity, and a p value.

For the smaller ATACseq dataset, ATAC-seq data generated from our CLL samples were processed similarly with the in-house ATAC-seq pipeline, as described previously (\protect\hyperlink{ref-Berest2019}{Berest et al. 2019}), with the only exception that we didn't use CG bias correction step. We used a similarly analytical mode of diffTF with HOCOMOCO v10 database (\protect\hyperlink{ref-HOCOMOCO}{Kulakovskiy et al. 2016}) using the following parameters: minOverlap = 1; design formula = ``\textasciitilde{} Patient + Trisomy 12 status.''

\hypertarget{functional-enrichment-analysis-of-spib-chipseq-data}{%
\subsubsection{Functional enrichment analysis of SPIB ChIPseq data}\label{functional-enrichment-analysis-of-spib-chipseq-data}}

\emph{myeslf} original (update to paper)
We downloaded SPIB ChIPseq data {[}REFERENCE{]} from the NCBI GEO database{[}REFERENCE{]}, accession \url{GEO:GSE56857}. We made use of the data for SPIB in the OCILY3 DLBCL cell line (GSM1370276). SPIB ChIP peaks were filtered for significance (q value\textless0.05). We used the annotatePeaks function from the package clusterProfiler{[}REFERENCE{]} to annotate the nearest gene for each ChIPpeak. We filtered peaks within Â±1kb of a TSS of a gene and performed over-representation of KEGG and Reactome pathways (using the clusterProfiler package{[}REFERENCE{]}) amongst the resulting list of genes.

\hypertarget{drug-drug-interaction-assay}{%
\subsubsection{Drug-Drug interaction assay}\label{drug-drug-interaction-assay}}

An independent patient cohort of 16 patients was used in the drug-drug interaction experiments.

\hypertarget{survival-analysis-of-immunohistory-chemistry-data}{%
\subsubsection{Survival analysis of immunohistory chemistry data}\label{survival-analysis-of-immunohistory-chemistry-data}}

\hypertarget{ibrutinib---ibet-762---il-4-analysis}{%
\subsection{Ibrutinib - IBET-762 - IL-4 analysis}\label{ibrutinib---ibet-762---il-4-analysis}}

\hypertarget{rnaseq-processing}{%
\subsubsection{RNAseq processing}\label{rnaseq-processing}}

\hypertarget{deseq2-analysis}{%
\subsubsection{Deseq2 analysis}\label{deseq2-analysis}}

\hypertarget{atacseq-processing-1}{%
\subsubsection{ATACseq processing}\label{atacseq-processing-1}}

\hypertarget{difftf-analysis}{%
\subsubsection{diffTF analysis}\label{difftf-analysis}}

\hypertarget{generation-of-gene-regulatory-network}{%
\subsubsection{Generation of Gene Regulatory Network}\label{generation-of-gene-regulatory-network}}

\hypertarget{proteomics-processing}{%
\subsubsection{Proteomics Processing}\label{proteomics-processing}}

\hypertarget{proteomics-analysis}{%
\subsubsection{Proteomics Analysis}\label{proteomics-analysis}}

\hypertarget{data}{%
\chapter{Data}\label{data}}

Characterisation of primary CLL samples by high-throughput combinatorial screening and multi-omic profiling. An introduction to the dataset that my PhD is based on.

\hypertarget{drug-screens-and-experiments}{%
\section{Drug screens and experiments}\label{drug-screens-and-experiments}}

\hypertarget{high-throughput-combinatorial-pertubation-assay}{%
\subsection{High-throughput combinatorial pertubation assay}\label{high-throughput-combinatorial-pertubation-assay}}

We measured the effects of 17 cytokines and microenvironmental stimuli on cell viability in 192 primary CLL samples and combined each one with 12 drugs to investigate the influence on spontaneous and drug-induced apoptosis (Figure \ref{fig:studyOverview}). Viability was assessed by ATP measurement via CellTiterGlo after 48 h of culture and normalised to untreated controls8.


\begin{figure}

{\centering \includegraphics[width=0.7px]{figures/studyOverview} 

}

\caption{Schematic of experimental protocol. By combining 12 drugs and 17 stimuli, we systematically queried the effects of simultaneous stimulation and inhibition of critical pathways in CLL (n=192). Integrating functional drug-stimulus response profiling with four additional omics layers, we identified pro-survival pathways, underlying molecular modulators of drug and microenvironment reponses, and drug-stimulus interactions in CLL.}\label{fig:studyOverview}
\end{figure}
\hypertarget{validation-experiments}{%
\subsection{Validation experiments}\label{validation-experiments}}

In addition to the screen, we acquired the following validatory data (Figure \ref{fig:additionalData}).

Proper explamantion of data srouces and how we acuried them, the format they are in


\begin{figure}

{\centering \includegraphics[width=0.7px]{figures/studyOverview} 

}

\caption{Schematic of validatory data.}\label{fig:additionalData}
\end{figure}
\hypertarget{characteristics-of-drugs-used-in-the-screen}{%
\section{Characteristics of drugs used in the screen}\label{characteristics-of-drugs-used-in-the-screen}}

\hypertarget{drug-pathways}{%
\subsection{Drug pathways}\label{drug-pathways}}

We screened 17 different drugs (Figure \ref{fig:drugCategories}).


\begin{figure}

{\centering \includegraphics[width=1px]{thesis_files/figure-latex/drugCategories-1} 

}

\caption{Bar plot of the drugs used in screen.}\label{fig:drugCategories}
\end{figure}
\hypertarget{drug-responses}{%
\subsection{Drug responses}\label{drug-responses}}

The drug responses were as follows (Figure \ref{fig:drugResponses}).


\begin{figure}

{\centering \includegraphics[width=0.5px]{thesis_files/figure-latex/drugResponses-1} 

}

\caption{Log transformed viability values for all drugs that were included in the screen after qulaity control. p values from student's t test.}\label{fig:drugResponses}
\end{figure}
\hypertarget{genetic-predictors-of-drug-responses}{%
\subsection{Genetic predictors of drug responses}\label{genetic-predictors-of-drug-responses}}

\hypertarget{drug---drug-correlations}{%
\subsection{Drug - Drug Correlations}\label{drug---drug-correlations}}

\hypertarget{characteristics-of-stimuli-used-in-the-screen}{%
\section{Characteristics of stimuli used in the screen}\label{characteristics-of-stimuli-used-in-the-screen}}

\hypertarget{stimulus-responses}{%
\subsection{Stimulus responses}\label{stimulus-responses}}

\hypertarget{stimulus---stimulus-correlations}{%
\subsection{Stimulus - Stimulus Correlations}\label{stimulus---stimulus-correlations}}

\hypertarget{characteristics-of-patient-samples-used-in-the-screen}{%
\section{Characteristics of patient samples used in the screen}\label{characteristics-of-patient-samples-used-in-the-screen}}

\hypertarget{genetic-data-available-for-each-patient}{%
\subsection{Genetic Data available for each patient}\label{genetic-data-available-for-each-patient}}

WES, CNVs, Methylation, Transcriptomic, ATACseq, LDT, survT, IHC
PLus Ibrutinib + IBET + IL4 treated samples - ATACseq, RNAseq, proteomics

\hypertarget{processing-of-raw-values-obtained-from-cell-viability-assay}{%
\section{Processing of raw values obtained from cell viability assay}\label{processing-of-raw-values-obtained-from-cell-viability-assay}}

\hypertarget{data-normalization-and-quality-control}{%
\subsection{Data normalization and quality control}\label{data-normalization-and-quality-control}}

\hypertarget{how-viabilty-is-calucalted-why-we-use-log-values-etc}{%
\subsection{How viabilty is calucalted, why we use log values etc}\label{how-viabilty-is-calucalted-why-we-use-log-values-etc}}

\hypertarget{heterogeneity-of-response-is-not-caused-by-differences-in-receptor-expression}{%
\subsection{Heterogeneity of response is not caused by differences in receptor expression}\label{heterogeneity-of-response-is-not-caused-by-differences-in-receptor-expression}}

Next, we tested whether the observed heterogeneity of response was caused by differences in receptor expression. We calculated Pearson correlation coefficients comparing control - normalised log viability values for each stimulus with transformed RNA counts of the corresponding stimulus receptor(s). No stimulus-receptor pair showed a Pearson coefficient greater than 0.4, confirming that the heterogeneity of response was not caused by differential receptor expression.



\hypertarget{generation-of-public-resource}{%
\section{Generation of public resource}\label{generation-of-public-resource}}

\hypertarget{shiny-app}{%
\subsection{Shiny app}\label{shiny-app}}

\hypertarget{package}{%
\subsection{Package}\label{package}}
\begin{itemize}
\tightlist
\item
  Present the underlying economic model/theory and give reasons why it is
  suitable to answer the given problem\footnote{Here is an example of a footnote.}.
\end{itemize}
\hypertarget{chapter4}{%
\chapter{Ex-vivo sensitivity to microenvironmental stimulation in primary CLL cells}\label{chapter4}}

The screen included 17 cytokines and microenvironmental stimuli, which were selected based on evidence in the literature that each stimulus had been shown to impact on CLL viability \emph{in vitro}. We attempted to put together the largest panel of stimuli for an assay of this kind in CLL, aiming to minimise redundancy among the compounds (Bruch \& Giles et al.~2021).

CLL cells do not proliferate \emph{in vitro}, but rather undergo spontaneous apoptosis in the absence of stimulation (\protect\hyperlink{ref-Collins1989}{Collins et al. 1989}). The effect of each stimuli on CLL viability was thus quantified by comparing ATP counts in treated primary samples, compared with those in DMSO wells where a positive value indicates that the sample viability was increased relative to control.

The assay represents a reductionist model of microenvironmental signalling, making it possible to dissect the effect of individual stimuli on baseline viability. The large patient cohort also enabled us to study the differential impact of these stimuli across heterogeneous samples, and identify distinct subgroups of patients whose samples show similar response profiles to the panel of stimuli.

\hypertarget{prolifing-responses-to-the-panel-of-stimuli}{%
\section{Prolifing responses to the panel of stimuli}\label{prolifing-responses-to-the-panel-of-stimuli}}

\hypertarget{cytokine-profiling}{%
\subsection{\texorpdfstring{\emph{ex vivo} assay demonstrated functional diversity of cytokines and microenvironmental stimuli}{ex vivo assay demonstrated functional diversity of cytokines and microenvironmental stimuli}}\label{cytokine-profiling}}

To investigate heterogeneity amongst responses to the stimuli, I calculated Pearson correlation coefficients for each stimulus pair, using the log-transformed normalised viabilities (Bruch \& Giles et al.~2021). The resulting coefficients were ordered using hierarchical clustering and visualised in a symmetrical heatmap (Figure \ref{fig:stimuliCorrelations}). The hierarchical clustering distinguished clusters of stimuli, including a larger group corresponding to agonists of TLR and Nf\(\kappa\)B pathways and a smaller group encompassing IL4 and TLR stimuli.


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/stimuliCorrelations-1} 

}

\caption{Heatmap of Pearson correlation coefficients. Coefficients for each pair of stimuli were calculated using log transformed viability values normalised to untreated control, and ordered according to hierarchical clustering. See Methods section \ref{correlations}. \emph{Figure adapted from Bruch \& Giles et al.~2021.}}\label{fig:stimuliCorrelations}
\end{figure}
98.5\% of stimulus pairs showed little correlation (R \textless{} 0.6), including those that targeted similar downstream pathways, indicating a high degree of functional diversity amongst microenvironmental signals. For example, JAK-STAT agonists such as IL4 and IL6 showed little correlation (Figure \ref{fig:correlationsScatter}A).

Only two stimulus pairs showed correlations where R \textgreater{} 0.6, and in both cases these targeted near identical receptors or downstream pathways. These included CpG ODN (TLR 9) and Resiquimod (TLR 7 and 8) (Figure \ref{fig:correlationsScatter}B), and IL4 and IL4 + CD40L which primarily target JAK3 - STAT6.

Repeating the analysis to correlate drug - drug pairs demonstrated that drugs targeting components of the same pathway were highly correlated. For quality control purposes, this indicated that our data sensitively and specifically reflect inter-individual differences in pathway dependencies (\protect\hyperlink{ref-JCIpaper}{Dietrich et al. 2017}).


\begin{figure}

{\centering \includegraphics{thesis_files/figure-latex/correlationsScatter-1} 

}

\caption{Scatter plot of log-transformed viability values, normalised to DMSO controls, for (A) treatment with JAK-STAT agonists IL4 and IL6 and (B) treatment with TLR agonists CpG ODN and Resiquimod.}\label{fig:correlationsScatter}
\end{figure}
Microenvironmental stimulation induced diverse phenotypes between patient samples, and across different stimuli. To gain a global overview of these phenotypes, I visualised log-transformed viability values normalised to DMSO controls for all patient samples and all stimuli (Figure @ref(fig:stimuliBeeswarms, (Bruch \& Giles et al.~2021).


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/stimuliBeeswarms-1} 

}

\caption{Log transformed viabilities after treatment with each stimulus. Where stimuli decreased viability relative to control, points are shown in blue, whilst increased viability is shown in red. Figure adapted from Bruch \& Giles et al.~2021.}\label{fig:stimuliBeeswarms}
\end{figure}
The majority of the stimuli increased viability, underlining the supportive nature of the microenvironment in CLL. However, three out of 17 reduced CLL viability relative to control, namely IL6, TGF\(\beta\) and TLR 7/8/9 agonists in IGHV-mutated (IGHV-M) samples.

IL4 and TLR7/8/9 agonists Resiquimod and CpG ODN induced the strongest responses, an indication of their potency in modulating CLL cell survival. Notably, TLR agonists increased viability in certain samples, in most cases IGHV-U, and decreased viability in others, mostly IGHV-M. Both IL4 and TLR7/8/9 emerged as the key pathways in the screen, and play an important role throughout the results of this thesis.

\hypertarget{clusters}{%
\subsection{Microenvironmental response profiling identifies discrete patient subgroups}\label{clusters}}

To further investigate the variability in responses across the cohort, we visualised z scores of the log-transformed viabilities and performed consensus clustering on the resulting matrix to generate a heatmap of all stimuli responses across all samples (Figure \ref{fig:stimuliHeatmap}).


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/stimuliHeatmap-1} 

}

\caption{The heatmap matrix shows the viability measurements for 192 samples (columns) and 17 stimuli (rows). Viability was measured via ATP-based assay after 48h stimulation. The data are shown normalised to DMSO-treated controls, and scaled within each row according to the Median Absolute Deviation (MAD). Limits were applied to scaling factor for optimal visualisation. The colour bars to the right show sample annotations. Consensus Clustering was used to define column tree layout, using hierarchical clustering with the Euclidean metric. See Methods section \ref{stimulus-heatmap-method}. \emph{Figure from Bruch \& Giles et al.~2021.}}\label{fig:stimuliHeatmap}
\end{figure}
The heatmap reflects heterogeneity in responses across samples and stimuli, once again underlining the potency of IL4 in increasing sample viability across diverse genetic backgrounds and the diversity in responses to TLR stimulation by Resiquimod and CpG ODN.

To generate the column-wise clustering, consensus clustering was performed on the matrix of z scores. Consensus clustering allows the user to subsample from the matrix of values, to generate hierarchical clustering for a given number of clusters, k. From this, it is possible to calculate a consensus matrix for each value of k, indicating for each pair of values the proportion of time they occupy the same cluster when subsampled together.

I visualised the clustered heatmap in Figure \ref{fig:stimuliHeatmap} for different values of k, and concluded on the existence of four robust clusters within the cohort (add these to appendix?). Each cluster shows a unique response profile to the panel of stimuli. We termed the clusters C1 to C4: C1 and C2 were enriched for IGHV-U whilst the samples in C3 and C4 were mostly IGHV-M.

To validate the choice of four clusters, I visualised the summaries of the consensus matrix, using the \texttt{ConsensusClusterPlus} package (\protect\hyperlink{ref-ConsensusClusterPlus}{M. D. Wilkerson and Hayes 2010}), to quantify the degree of confidence in the clusters for different values of k.


\begin{figure}

{\centering \includegraphics[width=0.45\linewidth]{figures/consensus008} \includegraphics[width=0.45\linewidth]{figures/consensus009} 

}

\caption{Summaries of the CDFs of the consensus matrices. Consensus CDF graphic showing the CDFs of the consensus matrix for k = 2 - 7, as indicated in the legend, estimated using 100 bin histogram (left). Relative change in area under the CDF curve, for k = 2 - 7, to compare k with k - 1. In the case of k = 2, there is no k - 1, so the total area is plotted. Line shows relative increase in consensus between each value of k (right). See Methods section \ref{stimulus-heatmap-method}.}\label{fig:consensusClusteringA}
\end{figure}
The graph of the CDFs of the consensus matrix for each k indicated that the CDF reaches a maximum and cluster confidence is maximised at k = 7 , though above k = 4 there is little appreciable increase. Figure \ref{fig:consensusClusteringA}. This is confirmed in the graph showing relative change in the area under the CDF curve, showing there is only a small increase in consensus between k = 4 and k =5, supporting the choice of k = 4. The cluster tracking plot depicts how each patient sample is assigned for each value of k. For k = 4, the plot indicates that C3 and 4 in particular are highly stable (Figure \ref{fig:consensusClusteringB}).


\begin{figure}

{\centering \includegraphics[width=0.45\linewidth]{figures/consensus010} 

}

\caption{Assignment of patient samples (columns), to each cluster, for k = 1 - 7 (rows) to demonstrate stability of cluster membership. Cluster colour for k = 4 match those in heatmap in \ref{fig:stimuliHeatmap}. See Methods section \ref{stimulus-heatmap-method}.}\label{fig:consensusClusteringB}
\end{figure}
\hypertarget{functional-characterisation-of-patient-clusters}{%
\section{Functional characterisation of patient clusters}\label{functional-characterisation-of-patient-clusters}}

\hypertarget{c1---c4-showed-distinct-response-profiles-with-the-panel-of-stimuli}{%
\subsection{C1 - C4 showed distinct response profiles with the panel of stimuli}\label{c1---c4-showed-distinct-response-profiles-with-the-panel-of-stimuli}}

The heatmap in Figure \ref{fig:stimuliHeatmap} demonstrated that each cluster responded differently to the panel of stimuli.

Amongst the IGHV-U enriched C1 and C2, both showed strong, positive responses to IL4 and TLR7/8/9 stimulation. C2 could be distinguished by stronger responses to the stimuli overall, in particular to NF\(\kappa\)B agonists IL1\(\beta\), anti-IgM, BAFF and sCD40L. Amongst the IGHV-M enriched clusters, C3 showed weaker responses to the majority of stimuli, and C4 was defined by a negative response to TLR7/8/9 stimulation (Bruch \& Giles et al.~2021). Figure \ref{fig:clusterStimuliResponse} summarises these findings in more detail, showing responses stratified by cluster for a subset of the stimuli.


\begin{verbatim}
cairo_pdf 
        2 
\end{verbatim}
\hypertarget{cluster-survival}{%
\subsection{The clusters show differences in disease dynamics}\label{cluster-survival}}

To validate the potential biological significance of these four clusters, we investigated whether the groups showed differential \emph{in vivo} disease progression (Bruch \& Giles et al.~2021). The study design was such that not all patients in the cohort were treatment - free, which confounded the analysis. For that reason, lymphocyte doubling time (LDT) and time to next treatment (TTT) were used to quantify CLL proliferative capacity, independently of treatment.

C1 and C2 showed a shorter LDT than C3 and C4, which is expected due to the differential proportions of IGHV-U and M patient samples in these groups (Figure \ref{fig:clusterLDTTTT}A). Notably, within the IGHV-M enriched clusters C3 and C4, samples in C3 showed a significantly shorter LDT (Student's t-test, p-value = 0.025).

To further validate this, we observed that TTT in the IGHV-M enriched C3 was significantly shorter than C4 and comparable to the progression dynamics of IGHV-U enriched C1 and 2 (Figure \ref{fig:clusterLDTTTT}B).


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/clusterLDTTTT-1} 

}

\caption{(A) Lymphocyte doubling time (LDT) stratified by cluster, p-values from Student's t-test. (B) Kaplan-Meier curves to show TTT for each cluster. p-values from univariate Cox proportional hazard models comparing IGHV-U enriched C1 with C2, and IGHV-M enriched C3 with C4. See Methods sections \ref{LDT-method} and \ref{survival-method}. \emph{Figure from Bruch \& Giles et al.~2021.}}\label{fig:clusterLDTTTT}
\end{figure}
The difference in disease progression between the clusters indicated that microenvironmental response represents an additional biological layer, holding information relevant to disease dynamics. To validate that these clusters were not simply an indication of any underlying genetic features, we checked whether the observed differences in progression dynamics could be explained by other prognostic markers (Bruch \& Giles et al.~2021).

A multivariate Cox proportional hazard model accounting for IGHV status, trisomy 12 and TP53 in addition to the cluster assignment indicated an independent prognostic value for cluster assignment between C3 and C4 (p= 0.039, Table \ref{tab:clusterCox}).


\begin{table}

\caption{\label{tab:clusterCox}Table depicting results of Multivariate Cox Proportional Hazard Model to test prognostic value of key genetic features and clusters using Time to Next Treatment and C3 as reference. }
\centering
\begin{tabular}[t]{l|r|r|r|r|l}
\hline
Factor & coef & exp(coef) & se(coef) & z & p.value\\
\hline
\cellcolor[HTML]{E2E868}{Cluster 3 vs Cluster 1} & \cellcolor[HTML]{E2E868}{-0.03979} & \cellcolor[HTML]{E2E868}{0.96099} & \cellcolor[HTML]{E2E868}{0.29813} & \cellcolor[HTML]{E2E868}{-0.13347} & \cellcolor[HTML]{E2E868}{0.89382}\\
\hline
Cluster 3 vs Cluster 2 & 0.51595 & 1.67522 & 0.37741 & 1.36708 & 0.1716\\
\hline
\cellcolor[HTML]{E2E868}{Cluster 3 vs Cluster 4} & \cellcolor[HTML]{E2E868}{-0.82011} & \cellcolor[HTML]{E2E868}{0.44038} & \cellcolor[HTML]{E2E868}{0.39760} & \cellcolor[HTML]{E2E868}{-2.06267} & \cellcolor[HTML]{E2E868}{0.03914}\\
\hline
IGHV.status & 0.55192 & 1.73658 & 0.27253 & 2.02513 & 0.04285\\
\hline
\cellcolor[HTML]{E2E868}{trisomy 12} & \cellcolor[HTML]{E2E868}{-0.13357} & \cellcolor[HTML]{E2E868}{0.87496} & \cellcolor[HTML]{E2E868}{0.35617} & \cellcolor[HTML]{E2E868}{-0.37503} & \cellcolor[HTML]{E2E868}{0.70764}\\
\hline
TP53 & 1.38977 & 4.01395 & 0.26072 & 5.33058 & <0.0001\\
\hline
\end{tabular}
\end{table}
\hypertarget{the-clusters-showed-differential-responses-to-drugs-in-vitro}{%
\subsection{\texorpdfstring{The clusters showed differential responses to drugs \emph{in vitro}}{The clusters showed differential responses to drugs in vitro}}\label{the-clusters-showed-differential-responses-to-drugs-in-vitro}}

The potential clinical relevance of the clusters was underlined by my observation that the samples within each group showed differential responses to drugs \emph{in vitro} (Figure \ref{fig:clusterDrugResponse}).

As expected, the IGHV-U enriched clusters C1 and 2 were more sensitive to BCR inhibition by ibrutinib, idelalisib and PRT062607, than C3 and 4. Between C1 and C2, C2 was more sensitive to a number of the drugs, including idelalisib (SYK) (p-value = 0.012), everolimus (mTOR) (p-value = 0.02) and the chemotherapeutics fludarabine (p-value = 0.031) and nutlin-3a (p-value = 0.042). Amongst C3 and C4, C3 showed lower sensitivity to everolimus (p-value = 0.051) and to fludarabine (p-value \textless{} 0.001) and nutlin-3a (p-value = 0.01). This aligns with the observation that patients in C3 have a poorer prognosis, despite most of these samples annotated as IGHV-M. C4 also showed a positive repsonse to Nf\(\kappa\)B inhibition by BAY-11-7085, and p38 MAPK inhibition by Ralimetinib.

Such differential drug response patterns suggests that microenvironmental response may reflect disease-specific CLL biology, in the same way as molecular profiling, and thus may have the potential to guide therapy decisions in future.


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/clusterDrugResponse-1} 

}

\caption{Log-transformed normalised viability values, stratified by cluster, for each drug. Drugs targeting the same pathway are grouped together. P-values from Student't t-test.}\label{fig:clusterDrugResponse}
\end{figure}
\hypertarget{cluster-genetics}{%
\subsection{The clusters are enriched for different genetic features}\label{cluster-genetics}}

Next we assessed differences in the molecular profiles of samples within each cluster. Visually, it appeared that certain clusters were enriched or depleted for various genetic features recurrent in CLL (Figure \ref{fig:clusterMatrix}).


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/clusterMatrix-1} 

}

\caption{Distribution of selected genetic features (rows) within each cluster for all patient samples (columns). Where a patient sample is not annotated for a feature, this is marked in white.}\label{fig:clusterMatrix}
\end{figure}
To quantify this, I ran a multinomial model, with lasso regularisation, to predict cluster membership (C1 -4) based on the matrix of genetic features for all the patient samples (Bruch \& Giles et al 2021). The model assigned coefficients to genetic features, where a positive coefficient indicated that this feature was enriched in the cluster, and a negative coefficient indicated it was depleted.

The approach used 3-fold cross validation, selecting the optimal model using lamda min. To ensure that the resulting coefficients were robust, we selected coefficients that satisfied certain cut-offs. Coefficients were selected if they were assigned in \textgreater{} 60\% of 50 bootstrapped repeats, and were larger than 0.35. Figure \ref{fig:clusterEnrichment} shows the mean coefficients and associated standard deviation, for each genetic feature that met these criteria in each cluster.


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/clusterEnrichment-1} 

}

\caption{Multinomial regression with lasso penalisation to identify enrichment or depletion of genetic features within each cluster. Matrix of genetic features (p=39), and IGHV status (encoded as M = 1 and U = 0) were used to identify multivariate predictors of cluster assignment. x axis shows genetic predictors, y axis indicates value and sign of coefficient assigned to feature, for each cluster (positive coefficients are enriched in the cluster, negative coefficients are depleted). Coefficients shown are mean coefficients from 50 bootstrapped repeats and error bars represent the mean Â± standard deviation. Genetic features with \textgreater20\% missing values were excluded, and only patients with complete annotation were included in the model (n=137). See Methods section \ref{cluster-genetics-method}. \emph{Figure from Bruch \& Giles et al 2021. }}\label{fig:clusterEnrichment}
\end{figure}
As we expected, IGHV status was the main feature that predicted cluster membership. Beyond IGHV status, trisomy 12 and \emph{SF3B1} mutations were enriched in C2, which showed enhanced responses to many stimuli. C4, which was associated with slow in-vivo progression, showed depletion of \emph{TP53}, \emph{ATM}, RAS/RAF mutations and gain8q.

\hypertarget{cluster-rna}{%
\subsection{GSEA of DE genes between subgroups}\label{cluster-rna}}

In addition to genetic features, I investigated differential expression of genes within each cluster. For n = 49 samples, RNAseq data was available for matched PBMC samples. I focused on the difference between clusters 3 and 4, for which 21 RNAseq samples were available (Bruch \& Giles et al.~2021).

To quantify differential gene expression, I began by filtering out immunoglobulin genes, including genes at the heavy, light and kappa loci that encode the antigen receptor of B cells. The clusters each show differential enrichment of IGHV-M and U samples, and thus the differential expression analysis would otherwise be dominated by immunoglobulin genes that are well known to affected by this biomarker.

I followed the Deseq2 protocol using a design formula to quantify the difference between clusters, and accounting for the confounding effect of IGHV status. 87 genes were differentially expressed (adjusted p \textless{} 0.05) between C3 and 4 (Figure \ref{fig:clusterDE}).


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/clusterDE-1} 

}

\caption{Volcano plot of differentially expressed genes between C3 and C4. X axis indicates log2 fold change values, calculated using the \texttt{DESeq2} package (\protect\hyperlink{ref-DESeq2}{\textbf{DESeq2?}}), y axis gives corresponding -log10(adjusted p value). P values adjusted using BH method. Genes are labelled where adjusted p \textless{} 0.05. Methods section \ref{cluster-rna-method} \emph{Figure from Bruch \& Giles et al.~2021.}}\label{fig:clusterDE}
\end{figure}
To assign biological meaning to the differentially expressed genes, I quantified the enrichment of Hallmark pathways amongst the genes. I ranked the genes based on the Wald statistic, and then ran GSEA was using the fgsea algorithm (Figure \ref{fig:clusterGSEA}) (Bruch \& Giles et al.~2021).

Several pathways were upregulated amongst samples in C3, compared to C4, indicating that these pathways may relate in some way to the shorter TTT and LDT of patients within this cluster. Pathways associated with higher disease aggression were regulated in C3 including genesets relating to stress response (Unfolded Protein Response, UV Response Up, P53 Pathway), metabolism (Oxidative Phosphorylation) and proliferation (G2M Checkpoint, MYC Targets V1, MTORC1 Signaling, E2F Targets) (Figure \ref{fig:GSEApathways}).


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/clusterGSEA-1} 

}

\caption{Gene set enrichment analysis (GSEA) to compare expression of genes in samples from C3 and C4 reveals upregulation of Hallmark pathways involved in microenvironmental signalling, stress response, metabolism and proliferation in C3. Normalised enrichment scores (NES) are shown for top 10 most significant pathways upregulated in C3 versus C4. Bars coloured according to adjusted p-value. Genes are ranked based on Wald statistic, calculated using the Deseq2 package and GSEA performed using the fgsea algorithm. See Methods section \ref{cluster-rna-method}. \emph{Figure from Bruch \& Giles et al.~2021. }}\label{fig:clusterGSEA}
\end{figure}
In addition, C3 showed upregulation of microenvironmental signalling pathways relative to C4, including TNFa Signalling via NFKB and Interferon Gamma Response (Figure \ref{fig:GSEApathways}) (Bruch \& Giles et al.~2021). This finding underlines our hypothesis that differential activity of microenvironmental signalling, both \emph{in vivo} and \emph{ex vivo} may be relevant to disease prognosis.


\begin{figure}

{\centering \includegraphics{thesis_files/figure-latex/GSEApathways-1} 

}

\caption{Enrichment plots of selected pathways. Gene set enrichment analysis (GSEA) was performed with the Hallmark gene sets from the GSEA Molecular Signatures Database. Wald statistic was used to rank the genes. The green curve corresponds to the Enrichment Score curve, which is the running sum of the weighted enrichment score obtained from GSEA software. See Methods section \ref{cluster-rna-method}. \emph{Figure from Bruch \& Giles et al.~2021.}}\label{fig:GSEApathways}
\end{figure}
\hypertarget{additional-analysis}{%
\section{Additional Analysis}\label{additional-analysis}}

Can microenvironmental response aid prognostic models? Run a Multivariate model to predict survival/drug response with genetic matrix and stimuli response matrix?

\hypertarget{summary}{%
\section{Summary}\label{summary}}

The screen represents an attempt to comprehensively dissect the impact of individual microenvironmental pathways on CLL viability. Our assay has enabled us to highlight key, broad spectrum signals such as IL4. Our heterogeneous cohort also reveals pathways that operate in subsets of patients, such as TLR. IL4 and TLR represent the key pathways that we focus on throughout the rest of this thesis.

In addition, this approach enables us to use microenvironmental response profiles, as an alternative to molecular profiling, to look for the existence of subgroups.

We uncovered four such subgroup with distinct response profiles and molecular properties and clinical outcomes, suggesting that microenvironmental response holds biologically significant information that may be relevant to prognosis and treatment decision making.

\hypertarget{discussion}{%
\section{Discussion}\label{discussion}}

Gosia suggested discussing each results section individually?

\hypertarget{contributions-statement}{%
\section{Contributions statement}\label{contributions-statement}}

The author of this thesis performed statistical inference and regression modelling outlined in this chapter. In essence, she performed data visualisation, optimised and interpreted the clustering of the heatmap to identify four patient subgroups, and investigated the molecular and phenotypic profiles of the four subgroups.

Peter-Martin Bruch generated the viability screening data, and curated and analysed the clinical data on LDT and TTT. The genetic annotation data was taken from (\protect\hyperlink{ref-JCIpaper}{Dietrich et al. 2017}). The heatmap in section was generated collectively with Peter-Martin Bruch, Wolfgang Huber and Sascha Dietrich, for the paper Bruch \& Giles et al 2021. The author also received support from Junyan Lu for the generation of the multivariate model, and from Peter-Martin Bruch and Sophie Herbst for the consensus clustering analysis.

\hypertarget{chapter5}{%
\chapter{Genetic modulators of responses to microenvironmental stimulation}\label{chapter5}}

Profiling the effects of the panel of stimuli revealed the heterogenous nature of responses to stimulation. The analysis revealed four patient sample groups, with distinct responses to the stimuli, unique molecular profiles, and differential disease progression. Two pathways emerged as the most potent modulators of CLL viability, namely IL4 and TLR. These too induced heterogeneous responses across the cohort, in particular stimulation of TLR 7/8/9 increased viability in some samples, whilst it reduced viability in others.

Next we asked to what extent this heterogeneity of response relates to the molecular profiles of the tumours. We sought to understand which underlying genetic features might modulate responses to external signals, and how these interactions may occur. We combined the screening dataset with multi-omics profiles of the patient samples taken from the PACE repository (\protect\hyperlink{ref-JCIpaper}{Dietrich et al. 2017}; \protect\hyperlink{ref-R-BloodCancerMultiOmics2017}{Oles et al. 2021}) and performed a systematic survey of molecular determinants of stimulus response, using whole-exome sequencing, DNA-methylation, RNA-sequencing and copy number variant data. In addition, for a small subset of patients we also generated ATAC sequencing and Mass Spectrometry data.

Collectively, these data enabled us to probe genetic and epigenetic modulators of microenvironmental signalling, in a heterogeneous cohort that encompasses the clinical and molecular diversity of CLL. In the first part of this chapter, I apply a broad systematic approach to identify important genetic modulators, and in the second, I outline my follow-up investigations into the impact of trisomy 12 on stimulus response.

\hypertarget{systematic-analysis-of-the-effect-of-genetic-features-on-responses-to-stimuli}{%
\section{Systematic analysis of the effect of genetic features on responses to stimuli}\label{systematic-analysis-of-the-effect-of-genetic-features-on-responses-to-stimuli}}

\hypertarget{univariate-gene-stimulus-assosciations}{%
\subsection{Univariate analysis identifies IGHV status and trisomy 12 as key modulators of microenvironmental response}\label{univariate-gene-stimulus-assosciations}}

To begin, I ran a univariate analysis to compare viability values post-stimulation for patient samples with and without each genetic feature. In total 63 genetic features were surveyed, including IGHV status, somatic gene mutations and structural variants, where there were at least three patient samples in each group (Figure \ref{fig:stimuliGeneAssosciations}).


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/stimuliGeneAssosciations-1} 

}

\caption{Plot showing BH-adjusted p values from Student's t-tests (two-sided, with equal variance), for all tested gene-stimulus associations. Tests performed for IGHV status and somatic mutations and copy number aberrations with â¥3 patient samples in each group (n = 63 or is it 54??). Each circle represents a gene-stimulus association. Associations that meet 10\% FDR cut off are indicated in colour, where the colour denotes the genetic feature. \emph{Figure from Bruch \& Giles et al.~2021.} See Methods section \ref{univariate-gene-stimulus-associations-method}.}\label{fig:stimuliGeneAssosciations}
\end{figure}
This analysis revealed the extent to which genetic features modulate microenvironmental response: for ten out of 17 stimuli, at least one genetic feature determined response and for six out of 17 stimuli, two or more genetic features significantly altered the response (Student's t-tests, FDR = 10\%, Bruch \& Giles et al.~2021). The most common features were IGHV status and trisomy 12. Del(11q) also affected response to several stimuli. Notably, del(13q) and del(17p), which like trisomy 12 and del(11q) are the most common aberrations in CLL and act as prognostic markers (\protect\hyperlink{ref-Dohner2000}{DÃ¶hner et al. 2000}), had no impact on the responses to the panel of stimuli.

\hypertarget{multivariate-gene-stimulus-assosciations}{%
\subsection{Multivariate analysis of gene - stimulus assosciations confirms IGHV status and trisomy 12 as key modulators of stimulus response}\label{multivariate-gene-stimulus-assosciations}}

It was possible that there may be interplay between genetic factors in determining responses to external signals. To address this, I applied multivariate modelling to integrate the influence of genetic features, IGHV status and DNA methylation on the size of response (Bruch \& Giles et al.~2021). I used a Gaussian linear model with L1-penalty (i.e., lasso regression), to derive a predictor for each stimulus, comprised of these covariates. For background on this approach, see section \ref{intro-multivariate-modelling}.

As input to the model, the response matrix was composed of the log transformed viability values for each stimulus. To generate the feature matrix (137 samples versus 41 features), I excluded genetic features for which \textgreater20\% of the values were missing, and patient samples with incomplete annotation. As predictors, I included genetic mutations and CNVs (p= 39), IGHV status (coded as 0-1) and Methylation Cluster (coded as 0, 0.5, 1). I ran lasso regression, as implemented in the \texttt{R} package \texttt{glmnet}(\protect\hyperlink{ref-R-glmnet}{Friedman et al. 2021}), using three-fold cross-validation with misclassification error as loss. The resulting predictors are the mean of those coefficients that were selected in at least 75\% of 30 bootstrapped repeats.

Using the output of the regression, I generated predictor profiles for each stimulus. For 5 / 17 stimuli, there was at least one genetic predictor that met the cut-offs (a selection are shown in figure \ref{fig:stimuliGeneAssosciationsMulti}).


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/stimuliGeneAssosciationsMulti-1} 

}

\caption{Predictor profiles for IL4 and CpG ODN depicting gene - stimulus associations. Bar plots (left) show size and sign of assigned coefficients from Gaussian linear modelling. A positive coefficient indicates that stimulated increase in viability is larger when feature is present. Scatter plots (bottom) and corresponding heatmaps above show how presence of selected genetic feature relates to sample viabilities. Scatter plots show ranked log(viability) values for each sample and heatmaps show mutation status for each predictor, for corresponding sample in scatter plot. See method \ref{multivariate-gene-stimulus-assosciations-method}. \emph{Figure adapted from Bruch \& Giles et al.~2021}.}\label{fig:stimuliGeneAssosciationsMulti}
\end{figure}
The multivariate analysis demonstrated that responses to IL4 and sCD40L + IL4, TLR and TGF\(\beta\) were all affected by multiple genetic features (Bruch \& Giles et al.~2021). For example, higher viability in response to IL4 stimulation was associated with trisomy 12 and unmutated IGHV. In contrast, IL4 generated little or no increase in viability in samples with a mutation in KRAS, NRAS or BRAF. These tumours benefited less from the anti-apoptotic effects of IL4, indicating that the ability of IL4 to increase viability could be dependent on signalling via Ras-Raf-MEK-ERK, which is already active in mutant samples.

Discussion that Inhibition of ERK (p38) also increased IL4 protective effect

\hypertarget{response-to-tlr-stimulation-is-dependent-on-ighv-status-trisomy-12-and-mutations-in-dna-damage-response-genes}{%
\subsection{Response to TLR stimulation is dependent on IGHV status, trisomy 12 and mutations in DNA Damage Response genes}\label{response-to-tlr-stimulation-is-dependent-on-ighv-status-trisomy-12-and-mutations-in-dna-damage-response-genes}}

The multivariate analysis highlighted that TLR stimulation by CpG ODN and Resiquimod both showed the largest number of predictors, reflecting the multiple layers of biology involved here. These included del(11q) and \emph{ATM} , del(17p) and \emph{TP53}, IGHV status, trisomy 12 and \emph{SF3B1} (Bruch \& Giles et al.~2021). Amongst the screening cohort, TLR stimulation increased viability in certain samples and decreased in others. The clustering groups identified in section \ref{clustering} also showed highly divergent responses to TLR, and this related to disease progression, thus warranting a more detailed look at the underlying features modulating TLR response (Figure \ref{fig:TLRGenes}).


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/TLRGenes-1} 

}

\caption{Control-normalised log transformed viability values after treatment with Resiquimod (TLR 7/8), stratified by named genetic features.}\label{fig:TLRGenes}
\end{figure}
IGHV status (along with Methylation Cluster, which is related to IGHV) and trisomy 12 had the strongest effect on TLR response. \protect\hyperlink{ref-Chatzouli2014}{Chatzouli et al.} (\protect\hyperlink{ref-Chatzouli2014}{2014}) have previously shown that TLR responses is dependent on IGHV status (see also, section \ref{intro-ME-pathways}. We noted trisomy 12 as an additional, novel determinant of TLR response (Bruch \& Giles et al.~2021). In samples that do not have trisomy 12, TLR stimulation increases the viability of IGHV-U samples, whilst it decreases viability in IGHV-M samples, as expected. In contrast, in samples with trisomy 12, TLR stimulation increases viability regardless of IGHV status (Student's t-test, p\textless0.001 and p=0.018, Figure \ref{fig:TLRIHGVtri12}).

\protect\hyperlink{ref-Chatzouli2014}{Chatzouli et al.} (\protect\hyperlink{ref-Chatzouli2014}{2014})'s observations point to the existence of specific types of BCR/TLR collaboration in CLL, leading to activation of pro-survival pathways, or apoptosis depending on the IGHV status of the tumour. Our results suggest that synergy between BCR and TLR may also be dependent on trisomy 12 status.


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/TLRIHGVtri12-1} 

}

\caption{Beeswarm-boxplot showing control-normalised log transformed viability values, after treatment with Resiquimod, stratified by trisomy 12 and IGHV status. p-values from Student's t-tests. \emph{Figure from (Bruch \& Giles et al.~2021)}}\label{fig:TLRIHGVtri12}
\end{figure}
In addition to trisomy 12, I noted that TLR stimulation also increased viability in samples with mutations in the DNA damage response pathway, namely del(11q), del(17p), and \emph{ATM} and \emph{TP53}. Del(17p) occurs in 7\% of CLL cases and is associated with loss of TP53, a tumour suppressor gene involved in DNA damage and repair (\protect\hyperlink{ref-Zenz2010}{Zenz et al. 2010}). Del(11q) is more common (18\% of cases) and is associated with loss of ATM, another protein involved in DNA repair(\protect\hyperlink{ref-Kipps2017}{Kipps et al. 2017}). This suggests that there may also be cross-talk between the TLR and DNA damage response pathway, and that this affects the outcome of TLR stimulation.

The effect of TLR stimulation on CLL viability is dependent on the molecular make-up of the tumour. The nature of TLR response may also be important in prognosis, underlined by our the observation in section \ref{clustering} that a subgroup of patients (C3) shows slower disease progression, and is the only group in which TLR induces apoptosis. The role of TLR signalling in pathogenesis and prognosis may thus so far be under appreciated.

Collectively, this work represents the first large-scale attempt to profile the integrative effects of cell-extrinsic signals and cell-intrinsic features in lymphoma on a large scale. This systematic approach highlighted trisomy 12 as the most common feature to modulate responses. The rest of this chapter outlines my work to investigate the role of trisomy 12 in microenvironmental response.

\hypertarget{investigating-trisomy-12-as-a-modulator-of-microenvironmental-response}{%
\section{Investigating Trisomy 12 as a modulator of microenvironmental response}\label{investigating-trisomy-12-as-a-modulator-of-microenvironmental-response}}

\hypertarget{trisomy12-modulator}{%
\subsection{Trisomy 12 is a modulator of microenvironmental response}\label{trisomy12-modulator}}

Trisomy 12 modulated responses to IL4, TGF\(\beta\), soluble CD40L + IL4 and TLR stimuli (Bruch \& Giles et al.~2021, Figure \ref{fig:tri12cytResponse}). For example, the increase in viability induced by IL4 was enhanced in trisomy 12 samples, as was the decrease in viability in response to TGF\(\beta\) stimulation.

Trisomy 12 is commonly mutated in CLL (15\% of patients)(\protect\hyperlink{ref-Dohner2000}{DÃ¶hner et al. 2000}). Until recently, this genetic lesion was regarded as conferring intermediate risk, though novel therapies have improved outcomes for trisomy 12 patients (\protect\hyperlink{ref-Bosch2019}{Bosch and Dalla-Favera 2019}). However, the functional explanation for its recurrence is incompletely understood. Previous work has shown a role for gene dosage effects in the pathogenic mechanism: \protect\hyperlink{ref-Kienle2005}{Kienle et al.} (\protect\hyperlink{ref-Kienle2005}{2005}) show that overexpression of genes including CDK4 and E2F1 leads to increased cell cycling and higher proliferative capacity. Likewise, \protect\hyperlink{ref-HerbstThesis}{Herbst} (\protect\hyperlink{ref-HerbstThesis}{2020}) showed that BCR signalling proteins are also upregulated, which may also contribute to the increased proliferative capacity. Despite this, outcomes in trisomy 12 CLL are more favourable, and these cases show increased susceptibility to BCR inhibition (\protect\hyperlink{ref-JCIpaper}{Dietrich et al. 2017}) (see also, section \ref{intro-trisomy12}).

In the following, I outline my work to investigate the incompletely understood role of trisomy 12 in CLL.


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/tri12cytResponse-1} 

}

\caption{Control-normalised log transformed viability values after treatment with IL4, sCD40L + IL4 and TGF\(\beta\), stratified by trisomy 12.}\label{fig:tri12cytResponse}
\end{figure}
\hypertarget{gene-dosage-effects}{%
\subsection{STAT6, IRAK4 and SMAD3 are more highly expressed in trisomy 12 CLL}\label{gene-dosage-effects}}

Trisomy 12 samples contain a third copy of chromosome 12: transcriptomic and proteomic profiling of CLL samples with this lesion have demonstrated that this has a major impact on gene expression and protein abundances (\protect\hyperlink{ref-Abruzzo2018}{Abruzzo et al. 2018}; \protect\hyperlink{ref-HerbstThesis}{Herbst 2020}; \protect\hyperlink{ref-MeierAbt2021}{Meier-Abt et al. 2021}). It has also been shown that BCR signalling proteins are amongst those with increased expression in trisomy 12 (\protect\hyperlink{ref-HerbstThesis}{Herbst 2020}), and this is thought to contribute to increased susceptibility to BCR inhibition in trisomy 12 CLL (\protect\hyperlink{ref-JCIpaper}{Dietrich et al. 2017}).

Guided by this observation, I began by investigating RNA and protein expression levels amongst genes involved in the TGF\(\beta\), IL4 and TLR pathways, to which trisomy 12 samples respond more strongly. The aim was to determine whether proteins in these pathways are more abundant in trisomy 12 CLL, thus contributing to the enhanced response.

I ran differential expression analysis to compare CLL samples from the screen with and without trisomy 12. Next, I filtered the differentially expressed genes (adjusted p \textless{} 0.1) for those belonging to the TGF\(\beta\), JAK-STAT and TLR pathways genesets, from the KEGG database. I visualised the RNA counts and protein abundances (\protect\hyperlink{ref-HerbstThesis}{Herbst 2020}) for those genes.

Only a small proportion of the differentially expressed genes belonged to the TGF\(\beta\), JAK-STAT and TLR pathways genesets. However of those that were, several genes were key downstream mediators of these pathways (Figure \ref{fig:geneDosage}. Amongst TGF\(\beta\) signalling genes, 7 / 95 were upregulated in trisomy 12, including SMAD3. 12 / 160 IL4 signalling genes were differentially expressed, including STAT2. Only 2 / 116 genes in the TLR geneset were differentially expressed, though this included IRAK4 on chromosome 12. Amongst these key mediators SMAD3, STAT2, and IRAK4, all showed higher protein abundance in addition to increased RNA expression. Notably, STAT6, the key downstream mediator of IL4 signalling, was not differentially expressed, but also showed differential protein abundance. Interestingly, many of the over-expressed genes are not located on chromosome 12, indicating the extent to which the differential dosage of this chromosome has on the expression of the entire genome.


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/geneDosage-1} 

}

\caption{Beeswarm-boxplots showing RNA counts and protein abundances stratified by trisomy 12 status. P values from Student's t-test. \_Proteomics dataset from (\protect\hyperlink{ref-HerbstThesis_}{\textbf{HerbstThesis\_?}}) Method: \ref{gene-dosage-effects-method}.}\label{fig:geneDosage}
\end{figure}
Next I looked for further evidence to directly link the higher gene dosage of cytokine signalling genes, to enhanced responses to cytokine signals.

\hypertarget{trisomy12-classifier}{%
\subsection{Classification analysis identifies trisomy 12 phenocopies that show increased expression of IRAK4 and SMAD3}\label{trisomy12-classifier}}

I next investigated whether any non-trisomy 12 samples might display a trisomy 12-like phenotype (referred to as a phenocopy), in that they respond in a similar way to the panel of stimuli. The aim was to identify trisomy 12 phenocopies and to isolate the feature of these samples that might explain the underlying cause of enhanced response to external signals in trisomy 12 CLL. In particular, I was interested to see if these phenocopies showed higher expression, or even gene amplification, of the signalling genes identified in section \ref{gene-dosage-effects}.

To identify trisomy 12 phenocopies, I began by generating a classifier to predict the trisomy 12 status of a sample from its stimulus response matrix. I aimed to find non-trisomy 12 samples that were consistently misclassified as trisomy 12.

The classifier was built using binomial regression, with lasso penalisation, as implemented in the \texttt{R} package \texttt{glmnet} (\protect\hyperlink{ref-R-glmnet}{Friedman et al. 2021}). The feature matrix consisted of z scores of the viability values after treatment with each stimulus, and was used to predict the response (trisomy 12 status). I ran the model for 50 bootstrapped repeats, using three-fold cross-validation and mean absolute error as loss. Resiquimod, sCD40L+IL4 and TGF\(\beta\) were selected as coefficients that predict trisomy 12 status (Figure \ref{fig:tri12Classfier}, as would be expected bed on the observations in section \ref{trisomy12-modulator}.


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/tri12Classifier-1} 

}

\caption{Predictor profile showing coefficients selected as predictors of trisomy 12 status. Here, binomial modelling with L1-penalty was used to identify associations between responses to stimuli and trisomy 12 status. Bar plots (left) show size and sign of assigned coefficients named on the right. A positive coefficient indicates that higher viability upon treatment named stimulus is associated with trisomy 12. Facet labels (top) and corresponding heatmap below show how the viability with each named stimulus relates to trisomy 12 status. The model shown was selected from 50 bootstrapped repeats, based on maximal AUC. Method \ref{trisomy12-classifier-method}.}\label{fig:tri12Classifier}
\end{figure}
I then predicted trisomy 12 status with the same viability matrix, using each of the 50 bootstrapped. I compared the results of the classification with the true trisomy 12 status. Two patient samples were misclassified as trisomy 12 in more than 50\% of repeats; I refer to these as patient sample A and B in the below. Patient sample A showed the lowest viability with TGF\(\beta\) (Figure \ref{fig:tri12PhenocopiesCytResponse}, and Patient sample B showed the highest viability with IL4, and the second highest with Resiquimod.


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/tri12PhenocopiesCytResponse-1} 

}

\caption{Control-normalised log-transformed viability values, for all samples after treatment with Resiquimod, sCD40L + IL4 and TGF\(\beta\). Patient sample A and B are indicated in blue and red, respectively.}\label{fig:tri12PhenocopiesCytResponse}
\end{figure}
I next investigated whether either of these non-trisomy 12 samples contained any regional amplifications on chromosome 12, which may help to isolate the gene(s) causing the tumour to respond more strongly to these signals. The whole exome sequencing for both patient samples indicated that Patient sample A had several amplified regions at 12p13.31 (42 copies), 12q24.13 (10 copies) and 12q24.33 (21 copies). An examination of the genes in these regions indicated that none of the signalling genes identified in \ref{gene-dosage-effects} could be found in these regions, and no clear candidate gene(s) emerged that may be involved in responses to external signals.

In the absence of gene amplification, I next checked patient samples A and B might show higher expression of the signalling genes identified in \ref{gene-dosage-effects} by visualising the RNA expression levels for SMAD3, IRAK4 and STAT6 for these patients (Figure \ref{fig:tri12PhenocopiesRNA}. Patient sample A, which responded most strongly to Resiquimod, showed high levels of IRAK4 compared to the other non-triomsy 12 samples. Patient sample B, which responded strongly to TGF\(\beta\) showed the highest level of SMAD3 expression amongst the non-trisomy 12 samples. Thus, whilst neither patient sample contained an amplicon of SMAD3 or IRAK4, its possible that increased expression of these proteins enables a stronger response to the corresponding pathway.


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/tri12PhenocopiesRNA-1} 

}

\caption{Raw RNA counts for SMAD3, IRAK4 and STAT6, stratified by trisomy 12 status. Patient sample A and B are indicated in blue and red, respectively. P values from Student's t-test.}\label{fig:tri12PhenocopiesRNA}
\end{figure}
These results collectively suggest that increased gene dosage of key genes in the IL4, TLR and TGF\(\beta\) pathways may underlie the increased response of trisomy 12 samples to these pathways. This is also reflected in the patient samples A and B, which responded more strongly to the TGF\(\beta\) and TLR pathways, respectively, and correspondingly higher levels of SMAD3 and STAT6 proteins. However, computational techniques can only go to far in providing biological proof, and more work is needed here to confirm this finding. In addition to gene dosage effects, I also investigated differences in transcription factor activity in trisomy 12. The results of this warranted further follow up, and this forms the focus of rest of this chapter.

\hypertarget{spi-b-and-pu.1-tfs-show-higher-activity-in-trisomy-12-cll}{%
\subsection{Spi-B and PU.1 TFs show higher activity in trisomy 12 CLL}\label{spi-b-and-pu.1-tfs-show-higher-activity-in-trisomy-12-cll}}

Trisomy 12 has been well-studied at both the transcriptomic and proteomic level, and yet the cause of its recurrence in CLL is not fully understood. Thus, I next decided to investigate the impact of trisomy 12 on the epigenetic landscape of CLL, which is less well studied. In particular, I wanted to investigate differential transcription factor activity in CLL, which would give an indicator of which pathways are differentially active.

We acquired two independent ATAC sequencing datasets. The first, generated and processed in our lab, consisted of two WT and two trisomy 12 samples (Bruch \& Giles et al.~2021). The second, taken from (\protect\hyperlink{ref-Rendeiro2017}{\textbf{Rendeiro2017?}}) and processed in our lab, comprised 43 WT and nine trisomy 12 samples.

In the external dataset, trisomy 12 status was not annotated. To do this, we used the ATACseq reads to call trisomy 12 in samples that contained \textgreater{} 1.4 times more reads per peak (i.e.~genomic region) on average in chromosome 12, compared to peaks on other chromosomes.

Next, we used the \texttt{R} package \texttt{diffTF} (\protect\hyperlink{ref-R-diffTF}{\textbf{R-diffTF?}}) to identify TFs that showed differential binding site accessibility between the WT and trisomy 12 samples (Figure \ref{fig:diffTFexplainer}. \texttt{diffTF} enables comparison of TF activity between two conditions, using chromatin accessibility data, see section \ref{intro-ATACseq}. We provided a list of 636 TFs, from the HOCOMOCO v10 database (\protect\hyperlink{ref-HOCOMOCO}{Kulakovskiy et al. 2016}) to measure their change in actiitvy between trisomy 12 and WT samples (Bruch \& Giles et al.~2021).


\begin{figure}

{\centering \includegraphics[width=0.5\linewidth]{figures/diffTF_method} 

}

\caption{diffTF workflow from \protect\hyperlink{ref-Berest2019}{Berest et al.} (\protect\hyperlink{ref-Berest2019}{2019}). diffTF accepts a list of TFs along with the genomic locations of their bindings sites. For each TF, the software computes the distribution of fold changes between the trisomy 12 and WT samples, using the ATACseq peaks at each TF binding site in each condition. The software compares this distribution to a background set of fold changes generated using GC content-matched loci that do not contain the same TF binding site motif. Each TF is thus assigned assigned a weighted mean difference value, which quantifies the change in activity, and a p value.}\label{fig:diffTFexplainer}
\end{figure}
Running this analysis on both ATACseq datasets indicated that the binding sites of nine TFs were more accessible (p\textless0.05) in the trisomy 12 samples of the larger, external dataset (Figure \ref{fig:tri12diffTF}. In the smaller in-house dataset, the binding sites of 92 TFs were likewise more accessible (Supp. Figure \ref{fig:diffTFsmallvolPlot}). RNA and proteomics data show different abundances of transcripts and proteins in trisomy 12 CLL: ATACseq data here shows that this also corresponds to a specific signalling signature in trisomy 12 CLL.

We confirmed that the results were not affected by the additional copy of chromosome 12. Rerunning diffTF analysis without the ATACseq reads from chromosome 12 had minor impact on the significant TFs.

In both datasets, the TFs with the largest increase in activity in trisomy 12 were Spi-B and/or PU.1 (Bruch \& Giles et al.~2021). Both TFs share similar binding motifs and exhibit functional redundancy (\protect\hyperlink{ref-GarrettSinha2001}{\textbf{GarrettSinha2001?}}), which make it difficult to distinguish from ATACseq data alone whether either or both are more active.

Spi-B and PU.1 are haematopoetic regulators that are known to be key regulators of healthy B-cell function (\protect\hyperlink{ref-Turkistany2011}{Turkistany and Dekoter 2011}){[}READ MORE{]}, controlling B-cell responses to environmental cues including CD40L, TLR ligands and IL4 (\protect\hyperlink{ref-Willis2017}{Willis et al. 2017}).


\begin{figure}

{\centering \includegraphics[width=0.8\linewidth]{thesis_files/figure-latex/tri12diffTF-1} 

}

\caption{Bar plot showing the results of the diffTF analysis for the \protect\hyperlink{ref-Rendeiro2016}{Rendeiro et al.} (\protect\hyperlink{ref-Rendeiro2016}{2016}) dataset. y axis shows change in TF activity (weighted mean difference) between trisomy 12 (n = 9) and non-trisomy 12 samples (n = 43),x axis indicates TF names. 17 / 636 TFs, with BH adjusted p \textless0.05 are shown. p values generated by diffTF in permutation mode. TF binding sites defined in HOCOMOCO v10 (\protect\hyperlink{ref-HOCOMOCO}{Kulakovskiy et al. 2016}).}\label{fig:tri12diffTF}
\end{figure}
Spi-B and PU.1 appeared to be upregulated in trisomy 12 CLL, and evidence in the literature indicated that these may regulate environmental sensing genes, providing a link between trisomy 12, and enhanced responses to external signals.

To provide further evidence of this, I next aimed to profile the downstream effects of Spi-B and PU.1 in lymphoma.

\hypertarget{spi-b-and-pu.1-targets-are-enriched-for-immune-signalling-pathways}{%
\subsection{Spi-B and PU.1 targets are enriched for immune signalling pathways}\label{spi-b-and-pu.1-targets-are-enriched-for-immune-signalling-pathways}}

I hypothesised that Spi-B and PU.1 might coordinate transcriptional response to external signals, thus modulating CLL proliferation in response to the microenvironment. To identify Spi-B and PU.1 target genes specifically in lymphoma, I acquired a ChIPseq dataset (\protect\hyperlink{ref-Care2014}{Care et al. 2014}) containing data on Spi-B and PU.1 binding in lymphoma cell lines. I used this dataset to test for functional enrichment of immune signalling pathways amongst the TF targets.

To define TF targets, I took the closest gene to each significant ChIP peak (q value\textless0.05), and within \(\pm\)kb of TSS. I then tested for over-representation of these TF targets amongst selected KEGG(\protect\hyperlink{ref-KEGG}{\textbf{KEGG?}}) and Reactome (\protect\hyperlink{ref-Reactome}{\textbf{Reactome?}}) genesets, using the `R' package \texttt{clusterProfiler} (\protect\hyperlink{ref-R-clusterProfiler}{Yu 2021b}). This method corresponded to a one-sided version of Fisher's exact test. This analysis showed TLR, BCR and TGF\(\beta\) signalling genes to be enriched (p\textless0.01) amongst Spi-B and PU.1 targets (Figure \ref{fig:SpiBChipSeq}).


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/SpiBChipSeq-1} 

}

\caption{Table shows results from over-representation tests of selected KEGG and Reactome pathways amongst Spi-B and PU.1 targets. Columns show geneset pathways, corresponding database, the number of genes within geneset, the number of Spi-B and PU.1 target genes within geneset (total number of target genes defined also shown), and p-value from over-representation test. TF targets defined as closest gene to each significant ChIP peak (q value\textless0.05), and within \$\pm\$1kb of TSS. ChIPseq data from \protect\hyperlink{ref-Care2014}{Care et al.} (\protect\hyperlink{ref-Care2014}{2014}). Over-representation tests run using (\protect\hyperlink{ref-clusterProfiler}{\textbf{clusterProfiler?}}) package, method corresponds to one-sided version of Fisher's exact test. \emph{Figure and caption from Bruch \& Giles et al 2021.}}\label{fig:SpiBChipSeq}
\end{figure}
\hypertarget{double-knockdown-of-spi-b-and-pu.1-reduces-profileration-of-trisomy-12-cell-lines}{%
\subsection{Double knockdown of Spi-B and PU.1 reduces profileration of trisomy 12+ cell lines}\label{double-knockdown-of-spi-b-and-pu.1-reduces-profileration-of-trisomy-12-cell-lines}}

To establish the functional impact of Spi-B and PU.1 inhibition in trisomy 12+ lymphoma, we tested th impact of inhibiting these TFs on proliferation of lymphoma cell lines (Figure \ref{fig:SpiBshRNAKD}. We generated single and double shRNA knockdowns in lymphoma cell lines, namely SU-DHL4 and SU-DHL5 (trisomy 12) and SU-DHL2 (no trisomy 12) and then measured cell counts at 24 hour intervals (Bruch \& Giles et al.~2021).


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/SpiBshRNAKD-1} 

}

\caption{Knockdown of Spi-B and PU.1 led to growth restriction and cell death. Data shown is the mean of three technical replicates. While SU-DHL 5 was already growth impaired after Spi-B KD, SU-DHL 2 and SU-DHL 4 were only impaired after double knock-down. Double knockdown lead to rapid cell death in SU-DHL 5 (data not shown). Cell count normalised to seeded cell number of diffuse large b cell lymphoma cell lines after knockdown of Spi-B or double knockdown of Spi-B and PU.1 or shRNA as control. \emph{Figure and caption from Bruch \& Giles et al.~2021.}}\label{fig:SpiBshRNAKD}
\end{figure}
The single knockdowns had a small impact on proliferation: Spi-B inhibition reduced proliferation in SU-DHL5, as did PU.1 to a lesser extent. Double knockdown of both TFs markedly reduced proliferation in SU-DHL2 and SU-DHL4, and was lethal in SU-DHL5. This result suggested that both these TFs play an important role in the proliferative capacity of the tumour cells, and that there is functional redundancy between Spi-B and PU.1 in this context.

Collectively, these results demonstrate that trisomy 12 modulates responses to microenvironmental signals. Trisomy 12 appears to increase Spi-B and PU.1 activity, which regulate genes relating to environmental sensing, and reduce proliferation of cell lines when inhibited.

\hypertarget{summary-1}{%
\section{Summary}\label{summary-1}}

In this chapter, we perform a systematic survey of genetic determinants of microenvironmental response, leading to two key findings. Firstly, TLR signalling has a range of effects on CLL viability, and this is determined by many genetic features, including IGHV status, trisomy 12 and mutation in the DNA Damage Response pathway. Secondly, we identify trisomy 12 as a modulator of microenvironmental response, and show that higher activity of Spi-B and PU.1 may mediate this effect.

\hypertarget{discussion-1}{%
\section{Discussion}\label{discussion-1}}

Some do affect, some dont at all
Notably, del(13q) and del(17p), which like trisomy 12 and del(11q) are the most common aberrations in CLL and act as prognostic markers (\protect\hyperlink{ref-Dohner2000}{DÃ¶hner et al. 2000}), had no impact on the responses to the panel of stimuli.
ie some are really affecting signalling, others are not

TLR and BCR
\protect\hyperlink{ref-Chatzouli2014}{Chatzouli et al.} (\protect\hyperlink{ref-Chatzouli2014}{2014}) show that stimulation of TLR in IGHV-U samples leads to pronounced p-ERK induction, whereas in M-CLL, stimulation of TLR concomitantly with BCR stimulation is required to induce a smaller increase in p-ERK. TLR stimulation without BCR stimulation in IGHV-M samples leads to an induction of caspase-8 and apoptosis, much more so than in IGHV-U samples(\protect\hyperlink{ref-Chatzouli2014}{Chatzouli et al. 2014}).

TLR as a prognostic tool?\textless- does this integrate multiple risk markers?
Reelvance in lymhp node??

Papeer whosing TLR response and survival that i found ages ago

REad up on trisomy 12
\protect\hyperlink{ref-JCIpaper}{Dietrich et al.} (\protect\hyperlink{ref-JCIpaper}{2017}).

Whats knwon about pathogenetic mechanims:
DOI: 10.1200/JCO.2005.02.568 Journal of Clinical Oncology 23, no. 16 (June 01, 2005) 3780-3792.

whole chromososm - is it lots of things - IRAK4, STAT6, gen dosge, SPIB and PU1 activity, BCR signalling?

gene dosage effects are obviosuly quite important here

the power of bioiformatics techqnieus , as well as traditioanl techinuqes, in bioogical discover, to save time and resouce

Read up on SpiB and PU1
SPib an dpU 1 regulate BTK: \url{https://pubmed.ncbi.nlm.nih.gov/8934542/}
SPIB and PU1 activity in B cells: \url{https://pubmed.ncbi.nlm.nih.gov/15936902/}
(hese data indicate that Spi-1 and Spi-B directly regulate the expression of Grap2 and that Grap2 functions to modulate BCR signaling, but that reduced Grap2 expression is unlikely to account for the BCR signaling defects observed in Spi1+/- SpiB-/- B cells.)
PU1 and SPIb and higher in Bc cells- this paper indicates this is to do with SP1 binding (which is on chromosome 12 and is hgiher ) \url{https://pubmed.ncbi.nlm.nih.gov/7566969/}
PU1 and PSIb bdinign in mouse lympohma cell ine \url{https://pubmed.ncbi.nlm.nih.gov/25765478/}
Spib and PU1 in B and T cell development \url{https://pubmed.ncbi.nlm.nih.gov/8691135/}

TLR signalling:
REad (\protect\hyperlink{ref-Chatzouli2014}{Chatzouli et al. 2014})
TLR stimulation can lead to both acitvation of MAPK and increase in viability or activation of apoptosis. If mutatiosn can affect which pathway is active, this poitns to aditoanl mechsnsim by which these mutations contribtue to CLL drive. It also points to TLR response as a useful prognostic marker.

POwer of multivariate modellings;
we need mor ltieracy about these things
discuovery of mulitple layers of biology rather thatn jsut using t tests

IL4 and KRAS - IL4 and ERK apthway reading

BCR singalling - which is important iwtih IHGV and tirsomy 12, is evleated in teh LN accoridng to gene sigantures:
\url{https://ashpublications.org/blood/article/117/2/563/28102/The-lymph-node-microenvironment-promotes-B-cell}

\hypertarget{contributions-statement-1}{%
\section{Contributions statement}\label{contributions-statement-1}}

The author of this thesis performed statistical inference and regression modelling outlined in this chapter. In essence, she performed the univariate and multivariate survey of genetic determinants of stimulus response. She identified trisomy 12 as a modulator of microenvironmental response, and conceptualised and performed the computational follow-up work on this finding, with support from Peter-Martin Bruch, Ivan Berest and Tina Becirovic.

This chapter integrates a number of datasets from the following sources. The screening data was generated by Peter-Martin Bruch. The associated patient genetic data, including WES, WGS, CNV profiles, DNA methylation data and RNA was generated and processed by \protect\hyperlink{ref-JCIpaper}{Dietrich et al.} (\protect\hyperlink{ref-JCIpaper}{2017}). The proteomics dataset was generated by \protect\hyperlink{ref-HerbstThesis}{Herbst} (\protect\hyperlink{ref-HerbstThesis}{2020}), and kindly shared for the purpose of this thesis. The larger ATACseq dataset was generated by (\protect\hyperlink{ref-Rendeiro2017}{\textbf{Rendeiro2017?}}), and processed by Ivan Berest. The smaller ATACseq dataset was planned by the author and Peter-Martin Bruch, and generated by Peter-Martin Bruch. The data was processed by the author, with support from Ivan Berest. The Spi-B and PU.1 knockdown data was generated by Tina Bericoviv and published in Bruch \& Giles et al.~2021.

In addition, the author received support on the following analysis work. Junyan Lu provided advice and support with multivariate modelling approaches. Ivan Berest provided support with running diffTF.

used I versus we and credited datasets, and my paper for figures and factrs - main analysis nad key finding, this hsould be routine
Then outline in detail at the bottom

\hypertarget{chapter6}{%
\chapter{Molecular and microenvironmental modulators of drug response in CLL}\label{chapter6}}

Chapters \ref{chapter4} and \ref{chapter5} provide a systematic exploration of the effect of microenvironmental stimulation on \emph{ex vivo} CLL biology, and how these pathways can be modulated by genetic features. Collectively, these results underline the fact that cancer biology is determined by an integrated network of cell-intrinsic molecular aberrations and cell-intrinsic signals generated by cell-cell contacts and soluble factors. The comprehensive nature of this dataset enables the study of these multiple axis.

In this chapter, we examine the third axis of the dataset: drug response. As well as providing protective signals that increase CLL viability and support tumour growth, the microenvironment is also believed to interfere with drug action. Several studies have demonstrated the impact of individual microenvironmental signals on drug response \emph{ex vivo}, and is also thought to occur within the protective niche (\protect\hyperlink{ref-Ahn2018}{\textbf{Ahn2018?}}). Guided by this observation, we next wanted to investigate the impact of each of the stimuli on responses to the panel of drugs, and how this can be further modulated by genetic features.

\^{}sharpen this once more

In section \ref{mapping-interactions}, the impact of the stimuli on drug response is quantified by identifying drug - stimulus interactions. In section \ref{mapping-genetic-modulators}, the impact of gneetic features on drug response is explored. Finally, in section \ref{drug-stimulus-gene-interactions}, multivariate modelling is apploed to integrate all three axes of the dataset, to investigate how drug - microenvironment interactions can be further modulated by genetic features. These analyses benefit from linear regression and generalised linear modelling approaches, described in more detail in section \ref{intro-multivariate-modelling}.

\hypertarget{mapping-interactions}{%
\section{Mapping drug - stimulus interactions}\label{mapping-interactions}}

\hypertarget{drug-stimulus-linear-model}{%
\subsection{Linear modelling quantifies interactions between drugs and stimuli}\label{drug-stimulus-linear-model}}

To begin, I evaluated computational methods for quantifying the impact of the stimuli on drug action. The aim was to screen for cases where the stimulus pathway specifically interacted with the target pathway of the drug.

When a stimulus is individually applied to a CLL sample, the stimulus activates signalling cascades that modulate CLl viability and impact the rate of spontaneous apoptosis. When a stimulus is co-applied with a drug, the stimulus will continue to impact upon baseline viability, at the same time as the drug inhibits baseline viability. Assuming the drug and stimulus do not interact, the viability of the cells with the combinatorial treatment will equate to the additive impact of both these compounds. In the case that there is some interaction between the stimulus pathway and drug target pathway, the resulting viability will be more complex. The difference between the additive, or expected viability, and the true measured viability, can be quantified by an interaction factor.

I fitted a linear model (Equation \eqref{eq:drugCytInt}) to quantify how each stimulus modulates drug efficacy beyond its individual effect (Bruch \& Giles et al.~2021). I used the \texttt{lm} function implemented in the \texttt{R} package \texttt{stats}(\protect\hyperlink{ref-R-base}{R Core Team 2021}). The model accounts for the individual effects of the drug and stimulus treatment on spontaneous apoptosis and baseline viability, in addition to an interaction term, named \(\beta{int}\). \(\beta{int}\), the interaction factor, quantifies how the combined treatment effect differs from the sum of the individual treatments.

Equation \eqref{eq:drugCytInt} quantifies how the viability with any combination can be predicted:
\begin{equation}
            log(V) = \beta_{drug}X_{drug} + \beta_{stimulus}X_{stimulus} + \beta_{int}X_{drug}X_{stimulus} + \epsilon
                                       \label{eq:drugCytInt}
    \end{equation}
\emph{where \(V\) is the predicted viability with a given treatment,} \(\beta_{drug}\), \(\beta_{stimulus}\) \emph{and} \(\beta_{int}\) \emph{are coefficients for the drug, stimulus and combinatorial terms and} \(X_{drug}\) \emph{and} \(X_{stimulus}\) \emph{are indicator variables (0 or 1) for the presence or absence of a drug/stimulus.} \(\epsilon\) \emph{is the vector of model residuals. See also Methods section \ref{drug-stimulus-linear-model-method}. Equation from Bruch \& Giles et al.~2021.}

This approach identified 45 drug-stimulus combinations (out of 204), where \(\beta{int}\) had p\textless0.05, i.e.~the stimulus had a specific effect on the efficacy of the drug, highlighting the extent to which drug action can be modulated by cell-external signals \emph{ex vivo}.

These interactions are of potential clinical interest, to support our understanding of \emph{in vivo} drug response, and the development of strategies to target drug resistance pathways. They are also relevant to laboratory studies of drug action, which are very often performed in the absence of such signals.

To establish which interactions may be important in these contexts, we began by classifying the drug - stimulus interactions into four categories (Bruch \& Giles et al.~2021). The categories were firstly defined based on whether the interaction was antagonistic or synergistic, i.e.~the stimulus acted to oppose or reinforce the activity of the drug. These were further divided based on the sign of \(\beta{int}\) i.e.~whether the combinatorial viability was higher (postive) or lower (negative) than would be expected based on additive effects (Figure \ref{fig:drugStimulusDummyPLots}A).


\begin{figure}[H]

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/drugStimulusDummyPLots-1} 

}

\caption{(A) Graphical line plots representing typical response patterns for each of the four drug - stimulus interaction categories (types I - IV). x-axis shows treatment, y-axis shows log transformed viability values, facet labels indicate interaction type. Blue and black horizontal lines demonstrate effect of interaction on viability with combinatorial treatment. Blue line shows predicted viability based on additive effects alone and black line shows viability accounting for additive effects and interaction. (B) Histogram showing number of drug-stimulus interactions within each category, where p value for \(\beta_{int}\) is \textless0.05. See Methods section \ref{drug-stimulus-linear-model-method}. \emph{Figure and caption from Bruch \& Giles et al.~2021.}}\label{fig:drugStimulusDummyPLots}
\end{figure}
These categories encompassed the following interaction types. Type I defined positive antagonisms in which stimuli reversed drug action, leading to decreased drug efficacy and increased CLL viability. This group highlights drug - stimulus interactions that could be relevant to treatment resistance pathways \emph{in vivo}. Type II negative antagonisms highlighted cases where the drug inhibited the stimulus, reducing the effect of the stimulus on baseline viability, and decreasing CLL viability. These interactions include drugs that provide potential therapies to target treatment - resistance pathways \emph{in vivo}. Type III and type IV interactions cover synergies, in which the drug and stimulus increase the effect of the other. Type III positive synergies lead to increased CLL viability with the combinatorial treatment, whilst type IV negative synergies lead to increased drug efficacy and lower viability. In the latter case, these may highlight cases where drug efficacy is dependent on external pathways, and may be useful for understanding \emph{ex vivo} drug action. Figure \ref{fig:drugStimulusDummyPLots}B quantifies the number of interactions in each category.

\hypertarget{drug-stimulus-interaction-examples}{%
\subsection{\texorpdfstring{INF\(\gamma\) induces resistance to ibrutinib \emph{in vitro}}{INF\textbackslash gamma induces resistance to ibrutinib in vitro}}\label{drug-stimulus-interaction-examples}}

Using the linear modelling approach in combination with our classification system, we attempted to generate the first comprehensive map of drug - stimulus interactions in CLL (Bruch \& Giles et al.~2021). This map provides a resource to explore the impact of the microenvironment on the majority of drugs that are clinically-licensed in CLL (Figure \ref{fig:drugStimulusIntMapt}). We use this resource to investigate a number of drug - stimulus interactions here, and in section \ref{chapter7}.


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/drugStimulusIntMapt-1} 

}

\caption{Heatmap of all \(\beta_{int}\) values for all drug - stimulus combinations where p for \(\beta_{int}\) \textless0.05, annotated with interaction type. Scale indicates size and sign of \(\beta_{int}\). Rows and columns clustered according to hierarchical clustering. See Methods section \ref{drug-stimulus-linear-model-method}. \emph{Figure and caption from Bruch \& Giles et al.~2021.}}\label{fig:drugStimulusIntMapt}
\end{figure}
To complement this broader approach, I next examined each interaction individually, to select the most clinically interesting examples (Figure \ref{fig:drugStimulusIntEx}).

Positive antagonistic interactions were the most common, in which stimuli reversed drug action. This group included resistance mechanisms such as the inactivation of ibrutinib by IL4 stimulation (Figure \ref{fig:drugStimulusIntEx}A)(\protect\hyperlink{ref-Aguilar-Hernandez2016}{Aguilar-Hernandez et al. 2016}), indicating that the modelling and classification approach successfully recapitulated known drug - stimulus interactions. Our approach demonstrated that IL4 induced resistance to a range of drugs, beyond ibrutinib, and showed the highest number of positive antagonistic interactions of all of the stimuli.

Amongst the positive antagonistic interactions, we noted reduced efficacy of ibrutinib in the context of IFN\(\gamma\) stimulation (Figure \ref{fig:drugStimulusIntEx}B, Bruch \& Giles et al.~2021). {[}ADD FOLLOW UP / STAT1 experiment{]}.

There were six cases of negative antagonisms, whereby drug action inhibited the pro-survival effect of the stimulus. As IL4 appeared to be the most potent conferrer of drug resistance, it was of greatest interest to identify drugs that may inhibit this pathway. For example, the Pan-JAK inhibitor pyridone-6 inhibited the increase in viability with sCD40L + IL4 stimulation (Figure \ref{fig:drugStimulusIntEx}C, Bruch \& Giles et al.~2021).

The model identified a single positive synergism, in which IFN\(\gamma\) treatment in combination with ralimetinib, a p38 MAPK inhibitor, stimulated a large increase in viability that was not observed in each of the single treatments (Figure \ref{fig:drugStimulusIntEx}D, Bruch \& Giles et al.~2021). This finding suggests a potential inhibitory effect of p38 MAPK activity on signalling via IFN\(\gamma\). {[}ADD TO THIS NOS THEORY{]}

16 drug - stimulus combinations demonstrated negative synergistic interactions. This implied that the drug efficacy was somehow dependent upon, or increased by, activation of the stimulus pathway. For example, luminespib, a HSP90 inhibitor, showed higher efficacy with a number of stimuli, including soluble anti-IgM, indicating that \emph{ex vivo} measurements of luminespib action are likely to be affected by the presence or absence of microenvironmental signals (Figure \ref{fig:drugStimulusIntEx}F, Bruch \& Giles et al.~2021).

Collectively, these results represent a comprehensive and systematic study of the influence of the microenvironment on drug efficacy in CLL. This work highlights key resistance pathways, strategies for targeting microenvironmental resistance, and underlines important pathways in \emph{ex vivo} studies of drug efficacy. In this thesis, I have selected some of the more interesting examples, such as resistance to Ibrutinib induced by IL4 and IFN\(\gamma\), to investigate in more detail here and in Chapter \ref{chapter7}. This work also aims to serve as a resource for others, to provide a basis for further follow-up studies beyond the ones in this thesis. It also aims to provide a proof-of-principle highlighting the value of combinatorial screening, and linear modelling approaches, to study the impact of the microenvironment in cancers in general. All drug - stimulus combinations and interactions can be explored on the \href{https://www.imbi.uni-heidelberg.de/dietrichlab/CLL_Microenvironment/}{online shiny app}, and the whole analysis can be replicated, or adapted, from the \href{https://github.com/Huber-group-EMBL/CLLCytokineScreen2021}{online code repository}, published alongside Bruch \& Giles et al.~2021.


\begin{figure}[H]

{\centering \includegraphics[width=0.8\linewidth]{thesis_files/figure-latex/drugStimulusIntEx-1} 

}

\caption{Line plots showing examples of drug-stimulus interactions. x axis indicates treatment, y axis shows log transformed viability values, with matching samples linked across treatments. Black horizontal lines represent predicted viability for each treatment, using coefficients from linear model fit. For combinatorial treatment, blue line indicates predicted viability based on additive effect alone, and black line indicates predicted viability accounting for additive effects plus interaction. See Methods section \ref{drug-stimulus-linear-model-method}. \emph{Figure and caption from Bruch \& Giles et al.~2021.}}\label{fig:drugStimulusIntEx}
\end{figure}
\hypertarget{mapping-genetic-modulators}{%
\section{Genetic modulators of drug response}\label{mapping-genetic-modulators}}

\hypertarget{univariate-gene-drug-associations}{%
\subsection{Univariate analysis identifies genetic modulators of drug response}\label{univariate-gene-drug-associations}}

Most mutations in cancer have not been linked to drug response (\protect\hyperlink{ref-JCIpaper}{Dietrich et al. 2017}). CLL has provided an powerful model system in which to study the link between recurrent genetic features and drug response: liquid biopsies are easy to obtain and sequence, and screen \emph{ex vivo} and it is much easier than in many cancers to take samples over a time course. Previous work in our lab (\protect\hyperlink{ref-JCIpaper}{Dietrich et al. 2017}) has taken advantage of this, generating the largest survey of molecular determinants of response in CLL to date.

Before investigating the impact of mutations on drug - stimulus interactions, I performed a similar analysis of genetic determinants of drug response, to first establish the effects of mutations independently of stimuli on drug response. I used the genetic data, including all mutations, copy number variants and IGHV status for which there were greater than 3 cases in the cohort (n = 54), to perform t-tests to identify molecular features that impact on drug response (Figure \ref{fig:drugGeneIntMap}).

This analysis recapitulated the findings of previous work (\protect\hyperlink{ref-JCIpaper}{Dietrich et al. 2017}), including the impact of IGHV status on responses to BCR inhibitors ibrutinib (BTK) and idelalisib (Pi3K), and the impact of TP53 and del(17p) mutations on Nutlin-3a (MDM2) and Fludarabine (Purine analogue).


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/drugGeneIntMap-1} 

}

\caption{Plot showing BH-adjusted p values from Student's t-tests (two-sided, with equal variance), for all tested drug-gene associations. Tests performed for IGHV status and somatic mutations and copy number aberrations (n=54). Each coloured circle represents a gene-stimulus association meeting 10\% FDR cut off. Method \ref{univariate-gene-stimulus-associations-method} \emph{Figure from Bruch \& Giles et al.~2021.}}\label{fig:drugGeneIntMap}
\end{figure}
In addition, it also highlighted the broad impact of trisomy 12 status on drug response \emph{ex vivo} (Figure \ref{fig:drugGeneIntEx}). While trisomy 12 is known to accelerate disease progression, tumours with trisomy 12 are more treatable due to higher sensitivity to chemotherapeutics and ibrutinib. In my results, trisomy 12 increased sensitivity to all drugs except ------fludarabine, BAY-11-7085 and Everolimus. These included increased sensitivity to p38 MAPK inhibition by ralimetinib, agreeing with previous work demonstrating higher trisomy 12 sensitivity with a range of MEK and ERK inhibitors, suggesting an essential role for MEK/ERK signalling in trisomy 12 CLL (\protect\hyperlink{ref-JCIpaper}{Dietrich et al. 2017}). Trisomy 12 samples also showed higher sensitivity to Nutlin-3a, an MDM2 inhibitor. Interesting, the MDM2 gene is located on chromosome 12, and is more abundant in trisomy 12 samples {[}ADD SUPP FIG{]}, which may explain this. I also noted increased trisomy 12 sensitivity to IBET-762, a BET inhibitor that has been shown to suppress transcriptional responses to cytokine signalling via JAK-STAT (\protect\hyperlink{ref-Chan2015}{Chan et al. 2015}). If trisomy 12 samples are more dependent on cytokine signals, as the results in section \ref{trisomy12-modulator} suggest, this makes sense.


\begin{figure}[H]

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/drugGeneIntEx-1} 

}

\caption{Beeswarm-boxplots showing control-normalised log transformed viability values after treatment with (A) Ralimetinib, (B) Nutlin-3a and (C) IBET-762, stratified by trisomy 12.}\label{fig:drugGeneIntEx}
\end{figure}
\hypertarget{drug-stimulus-gene-interactions}{%
\section{The modulatory effect of mutations on drug - stimuli interactions}\label{drug-stimulus-gene-interactions}}

\hypertarget{patient-specific-linear-modelling-identifies-drug---stimulus---gene-interactions}{%
\subsection{Patient-specific linear modelling identifies drug - stimulus - gene interactions}\label{patient-specific-linear-modelling-identifies-drug---stimulus---gene-interactions}}

Sections \ref{mapping-interactions} \& \ref{mapping-genetic-modulators}, explore the effects of microenvironmental signals and molecular features on drug response independently. I next aimed to investigate their collective effect on \emph{in vitro} drug efficacy in CLL, by quantifying the extent to which genetic driver mutations modulated the interactions between stimuli and drugs (Bruch \& Giles et al.~2021).

I began by conceptualising methods to quantify the collective effect of the drugs, stimuli and genetic features on CLL viability and biology. I decided to adapt the linear model in Equation \eqref{eq:drugCytInt}, by fitting this model in a patient sample - specific manner (Bruch \& Giles et al.~2021):

Equation \eqref{eq:drugCytGeneInt} quantifies how the viability with any combination can be predicted:
\begin{equation}
            log(V) = \beta_{drug}X_{drug} + \beta_{stimulus}X_{stimulus} + \beta_{patient}X_{patient} + \beta_{drug-stimulus}X_{drug}X_{stimulus} + \beta_{drug-patient}X_{drug}X_{patient} + \beta_{stimulus-patient}X_{stimulus}X_{patient} +\beta_{int}X_{drug}X_{stimulus}X_{patient} + \epsilon
                                       \label{eq:drugCytGeneInt}
    \end{equation}
\emph{where \(V\) is the predicted viability of a patient sample with a given treatment,} \(\beta_{drug}\), \(\beta_{stimulus}\), \(\beta_{int}\), \(\beta_{drug-stimulus}\), \(\beta_{drug-patient}\), \(\beta_{drug-stimulus}\) \emph{and} \(\beta_{int}X_{drug}\) \emph{are regression coefficients for the drug, stimulus, patient sample and combinatorial terms and} \(X_{drug}\), \(X_{stimulus}\) \emph{and} \(X_{patient}\) \emph{are indicator variables (0 or 1) for the presence or absence of a drug/stimulus/patient sample.} \(\epsilon\) \emph{is the vector of model residuals. See also Methods section \ref{drug-stimulus-gene-interactions-method}. Equation from Bruch \& Giles et al.~2021.}

This generates a higher order interaction term \(\beta_{int}X_{drug}X_{stimulus}X_{patient}\). This term represents a \emph{patient sample-specific} \(\beta{int}\) for each drug - stimulus combination, quantifying the size of an interaction between a drug and stimulus in each patient genetic background.

With these patient sample-specific \(\beta{int}\) terms, it was possible to search for associations between the size of \(\beta_{int}\)) and genetic features. The aim was to screen for molecular features that increased or decreased the size of a drug - stimulus interactions, using multivariate regression with L1 (lasso) regularisation.

As input to the model, the response matrix was composed of the sample - specific \(\beta_{int}\) values for each drug-stimulus combination. To generate the feature matrix (137 samples versus 40 features), I excluded genetic features for which \textgreater20\% of the values were missing, and patient samples with incomplete annotation. As predictors, I included genetic mutations and CNVs (p=39) and IGHV status (coded as 0-1). I ran lasso regression, as implemented in the \texttt{R} package \texttt{glmnet}(\protect\hyperlink{ref-R-glmnet}{Friedman et al. 2021}), using three-fold cross-validation with misclassification error as loss. The resulting predictors are the mean of those coefficients that were selected in at least 90\% of 30 bootstrapped repeats.

This analysis revealed that 60/204 drug - stimulus interactions were modulated by at least one genetic feature (Figure \ref{fig:drugCytGeneIntMap}, Appendix Figure \ref{fig:drugcytGeneIntAll}). A positive coefficient here indicates that the presence of the genetic feature is associated with more positive \(\beta{int}\), in other words, the viability with the drug and stimulus is higher than expected in the presence of the genetic feature.

\emph{Have to make sure this figure takes a whole page but doesn't float to the bottom}


\begin{figure}

{\centering \includegraphics[width=0.5\linewidth]{thesis_files/figure-latex/drugCytGeneIntMap-1} 

}

\caption{Heatmap summarising genetic predictors of drug - stimulus interactions. Each row depicts a single drug - stimulus combination, and each tile indicates that \(\beta{int}\) for given drug and stimulus combination is modulated by corresponding genetic feature. Thus each row represents the output of a single model fit (as in Figure \ref{fig:FluCpGExA}). Colour of tile indicates size and sign of coefficient assigned to genetic feature, where a positive coefficient corresponds to a more positive \(\beta{int}\) if the feature is present. Only top 8 most commonly selected genetic features are shown. Drug - stimulus combinations with no genetic predictors of \(\beta{int}\) amongst top 8 shown are omitted for clarity. See Methods section \ref{drug-stimulus-gene-interactions-method}. \emph{Figure from Bruch \& Giles et al.~2021.}}\label{fig:drugCytGeneIntMap}
\end{figure}
Applying this broad scale screening approach established some wider trends. Firstly, Trisomy 12 and IGHV status impacted the largest number of drug - stimulus interactions,indicating that their impact on stimulus and drug response individually also extends to drug-stimulus interactions. Secondly, out of all the stimuli, IL4 and sCD40L + IL4-based interactions were modulated by the largest number of features. IL4 alone increased viability uniformly accross genetic backgrounds; its interaction with drugs may be more influenced by genetic features. Notably, the presence of trisomy 12 acted to increase IL4-induced drug resistance, in almost all drugs for which \(\beta{int}\)\textgreater0.05. Interestingly, the presence of TP53 and del(17q) reduced the degree to which IL4 induced resistance to the chemotherapeutics Nutlin-3a and Fludarabine, most likely because both these mutations already induce resistance alone.

\hypertarget{patient---specific-drug---stimulus-interactions-may-have-clinical-significance}{%
\subsection{Patient - specific drug - stimulus interactions may have clinical significance}\label{patient---specific-drug---stimulus-interactions-may-have-clinical-significance}}

This broad scale screening approach was not perfect in itself. Modelling can only go so far, and it is likely that some cases identified here are artefacts of the modelling approach. Therefore, I decided to investigate each individual case one - by - one to establish which drug - stimulus - gene interactions are the most biologically interesting. This analysis highlighted several drug - stimulus - gene interactions that may have clinical importance and warrant further investigation (Bruch \& Giles et al.~2021).

The first of these concerned the interaction between fludarabine and CpG ODN. The value of \(\beta_{int}\) for fludarabine and CpG ODN was modulated by six genetic factors; most strongly by IGHV status, del(11q) and trisomy 12 (Figure \ref{fig:FluCpGExA}, Bruch \& Giles et al.~2021).


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/FluCpGExA-1} 

}

\caption{Genetic features that modulate the size of \(\beta{int}\) between fludarabine and CpG ODN. Scatter plot, heatmap and bar plots indicate inputs and outputs of equation \eqref{eq:drugCytGeneInt}. Bar plots indicate size and sign of coefficients assigned to genetic features named on right. Scatter plot depicts \(\beta{int}\) values for each patient sample i.e.response matrix. Heatmap tiles indicate mutation status for the selected genetic features (i.e.~feature matrix) corresponding to same sample in scatter plot below, to show how size of \(\beta{int}\) varies with each feature. See Methods section \ref{drug-stimulus-gene-interactions-method}. \emph{Figure and caption from Bruch \& Giles et al.~2021.}}\label{fig:FluCpGExA}
\end{figure}
To look in more detail at what this means, I visualised sample responses to CpG ODN and fludarabine, stratified by IGHV status and trisomy 12 (Figure \ref{fig:FluCpGExB}, Bruch \& Giles et al.~2021). In IGHV-M non-trisomy 12 samples, TLR stimulation increased fludarabine efficacy. In samples that were either IGHV-U or trisomy 12+, TLR stimulation induced resistance to fludarabine. TLR stimulation \emph{ex vivo} can increase or decrease viability, depending on the genetic background and this also impacts on drug response. It would be useful to establish whether these dynamics are replicated within the \emph{in vivo} niche.


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/FluCpGExB-1} 

}

\caption{Beeswarm-boxplots of log(viability) values for fludarabine and CpG ODN single and combinatorial treatments, faceted by IGHV status and trisomy 12 status. P-values from paired Student's t-tests. \emph{Figure and caption from Bruch \& Giles et al.~2021.}}\label{fig:FluCpGExB}
\end{figure}
TLR-induced drug resistance in certain genetic backgrounds extended to other chemotherapeutic drugs. Nutlin-3a is an MDM2 inhibitor, which in our dataset shows higher efficacy in certain backgrounds including IGHV-U, trisomy12, and del(11q) (Figure \ref{fig:drugGeneIntMap}, Figure \ref{fig:drugGeneIntEx}, Appendix Figure \ref{fig:NutlinPredictors}). Despite this, TLR stimulation reduces Nutlin-3a toxicity in these same genetic backgrounds, such that treatment with Nutlin-3a and CpG ODN is more toxic to IGHV-M, non-trisomy 12 and non-del(11q) samples i.e.~the opposite(Figure \ref{fig:nutlinCpgEx}, Figure \ref{fig:TLRCpGIGHVdel11q}). This observation underlines the need to observe drug activity \emph{in vitro} in the context of microenvironmental signals, in CLL and in other cancers.


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/nutlinCpgEx-1} 

}

\caption{Predictor profile depicting genetic features that modulate the size of \(\beta{int}\) between Nutlin-3a (MDM2) and CpG ODN. Scatter plot (bottom) and heatmap (right) and bar plots (left) indicate inputs and outputs of fitting model in Equation \eqref{eq:drugCytGeneInt}. Bar plots indicate size and sign of coefficients assigned to genetic features named on the right. Scatter plot depicts ranked patient-specific \(\beta{int}\) values i.e.~the response matrix. Heatmap above indicates patient mutation status for the selected genetic features i.e.~feature matrix. Heatmap tiles correspond to points in scatter plot (i.e.~patient data is aligned), to indicate how the size of \(\beta{int}\) varies with selected genetic feature. See Methods section \ref{drug-stimulus-gene-interactions-method}. \emph{Figure and caption from Bruch \& Giles et al.~2021.}}\label{fig:nutlinCpgEx}
\end{figure}

\begin{figure}

{\centering \includegraphics[width=0.8\linewidth]{thesis_files/figure-latex/TLRCpGIGHVdel11q-1} 

}

\caption{Beeswarm-boxplot showing control-normalised log transformed viability values, after treatment with CpG ODN + Nutlin-3a, stratified by del(11q) and IGHV status. p-values from Student's t-tests. \emph{Figure from (Bruch \& Giles et al.~2021)}}\label{fig:TLRCpGIGHVdel11q}
\end{figure}
The modelling approach also identified interactions, where the differential size of \(\beta{int}\) was driven by the single treatment effect, rather than the combination. For example, if a drug is more efficacious in a certain genetic background, but a stimulus nonetheless eradicates drug toxicity, the size of the interaction will be larger in this genetic background. This is because the increase in expected viability in these samples (quantified by \(\beta{int}\)) would be even higher than in other genetic backgrounds. This is the case with Ibrutinib + IL4 for instance (Bruch \& Giles et al.~2021). Ibrutinib showed higher efficacy in trisomy 12 and IGHV-U samples and IL4 induced complete resistance to ibrutinib independently of genetic background. Thus in trisomy 12 and IGHV-U samples treated with ibrutinib, the increase in viability in the context of IL4 stimulation is larger than in non-trisomy 12 and IGHV-M samples, and thus these features are assigned positive coefficients. This highlights the breadth of IL4-induced resistance to ibrutinib in CLL across genetic backgrounds, even where genetic features would indicate that ibrutinib efficacy should be high, and underlines the importance of understanding the activity of this resistance pathway \emph{in vivo} (Figure \ref{fig:ibrIl4ExA}, Figure \ref{fig:ibrIl4ExB}, Bruch \& Giles et al.~2021).




\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/ibrIl4ExA-1} 

}

\caption{Predictor profile depicting genetic features that modulate the size of \(\beta{int}\) between ibrutinib (BTK) and IL4. Plot generated as in Figure \ref{fig:FluCpGExA}. See also Methods section \ref{drug-stimulus-gene-interactions-method}. \emph{Figure from Bruch \& Giles et al.~2021.}}\label{fig:ibrIl4ExA}
\end{figure}
\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/ibrIl4ExB-1} 

}

\caption{Beeswarm boxplots of log(viability) values for ibrutinib and IL4 single and combinatorial treatments, faceted by IGHV status and trisomy 12 status. P-values from paired Student's t-tests. \emph{Figure and caption from Bruch \& Giles et al.~2021.}}\label{fig:ibrIl4ExB}
\end{figure}
this is a resource, all models , and drug - stimulu resposnes stratifed by genetic features can be explored:

\hypertarget{summary-2}{%
\section{Summary}\label{summary-2}}

This work represents the first large - scale attempt to integrate the effects of mutations and the microenvironment in drug response in lymphoma, and to the best of my knowledge, in other cancers. The results provide a proof-of-principle of using combinatorial screening, molecular profiling and statistical modelling can generate more complex biological insights to guide further clinical studies.

More specifically, these results identify a number of key drug-resistance pathways, including the impact of IFN\(\gamma\) on ibrutinib. I highlight the importance of the NOS pathway in IL4 and IFN\(\gamma\) inhibitor of apoptosis. I also show how some of these resistance pathways are context-dependent. In particular, the ability of TLR stimulation to induce resistance to chemotherapeutics is modulated by many genetic feautres. THis warrants further follow-up.

Overall these findings illustrate how the presence of known genetic alterations determines how drugs and external stimuli interact with each other.

\hypertarget{discussion-2}{%
\section{Discussion}\label{discussion-2}}

IL4and ibrutinib- what's knwon
implications of ifn gamma and also p38 and ralitmetinib and ifn gamma

strategies for targeting these - pyridone 6, but not licensned? BET inihibition - paper showing that bet specfically inhibits ifn and il4 targets - so maybe somehting more braod specturm is useful

implciatiosnfor studying drug action ex vivo
In addition, we aimed to demonstrate that combinatorial screening in conjunction with linear modelling is a powerful tool to study the impact of the microenvironment on drug response in cancer, which is often under-appreciated relative to the impact of molecular features.

In addition, we aimed to demonstrate that combinatorial screening in combination with linear modelling is a powerful approach to study the impact of the microenvironment on drug response in cancer, which is often under-appreciated relative to the impact of molecular features.

How each category might be useful

the importanceof the resource here

OUr approach distinguishedbetween baseline and interaction s- this is a powerful proof of rpinciple
cancer biology is integratively influenced by cell-intrinsic and cell-extrinsic features \textless- what attempts hhave their been, persuade whythis needs to continue, how canthis behepful in the clinic

MUST READ: BET inhibitors and supression of cytokine signalling : BET Bromodomain Inhibition Suppresses Transcriptional Responses to Cytokine-Jak-STAT Signaling in a Gene-specific Manner in Human Monocytes

Nutline and IBET
nutlin-3ain trisomy 12 - MDM2 is higher

also IBET -
these also correlate - its possible that they work via a simliar mechanism -ibet inhibits brd4 whcih is inhibtiing p53 targets, so p53 is more active in thesecels

\url{https://www.nature.com/articles/s41467-020-20378-8}

NOS pathway

TLR - wahts going on in vivo, is it reallyhaving thhis bipolar effect in vivo
Flduarabine response - does this help to explain - more work needed

In Vitro Sensitivity of CLL Cells to Fludarabine May Be Modulated by the Stimulation of Toll-like Receptors
Eleonora Fonte

IL4-induced resistance- this is clearly important accross a rnage of drufs adn genetic background, more needed in vivo, possible combintatorial therapies

in vivo IL4 evidence:
from T-cells in chronic lymphocytic leukemia: Guardians or drivers of disease?
Philipp M. Roessner \& Martina Seiffert

Leukemia volume 34, pages 2012--2024 (2020)
In 2005, GÃ¶rgÃ¼n and colleagues investigated T-cell function in CLL. Gene expression profiling (GEP) of sorted T-cells from peripheral blood mononuclear cells (PBMCs) of CLL patients suggested that CD4+ T-cells are skewed toward a Th2 phenotype due to decreased expression of c-Jun NH2-terminal kinase (JNK) and p38 mitogen-activated protein kinase (MAPK) pathway-related genes {[}12{]}. This is supported by an increased number of interleukin (IL)-4 secreting CD4+ T-cells in PBMCs of CLL patients in comparison to cells of healthy controls (HC). Studies using the immunocompetent EÂµ-TCL1 mouse model of CLL and after adoptive transfer of TCL1 leukemia (TCL1 AT) into BL/6 wild-type (WT) mice revealed more IL-4 secretion of CD4+ T-cells in comparison to control mice {[}13, 14{]}. Similarly, conditioned medium of CLL cells was able to induce secretion of IL-4 in allogeneic mixed lymphocyte reactions (MLR) of healthy individuals, suggesting that soluble factors secreted by CLL cells trigger Th2 T-cell polarization in vitro {[}15{]}.

in other cancers we shoudl integrate drugs - stimul and genetics

flurdarabine and Cpg, different outcomes in IGHV M and U patients

the imrpotatnce of IL4 in all patietn backgrounds

TLR Signaling Is Activated in Lymph Node--Resident CLL Cells and Is Only Partially Inhibited by Ibrutinib
Eman L. Dadashian
\url{https://cancerres.aacrjournals.org/content/79/2/360.short}

There are models to investigate ibrutinib resistance in vivo, but they are lacking in for ME:
Development and characterization of prototypes for in vitro and in vivo mouse models of ibrutinib-resistant CLL
Burcu Aslan

also check out discussion from :\url{https://www.sciencedirect.com/science/article/pii/S0006497119402206}

\hypertarget{contributions-statement-2}{%
\section{Contributions Statement}\label{contributions-statement-2}}

The author of this thesis performed the statistical inference and regression modelling outlined in this chapter. In essence, she conceptualised and performed the linear modelling and multivariate modelling in Equations \ldots. In addition, she performed univariate analysis of genetic determinants of drug response and data visualisation and investigation of hits.

Peter-Martin Bruch generated the viability screening data used in the model. The genetic annotation data was taken from (\protect\hyperlink{ref-JCIpaper}{Dietrich et al. 2017}). The categorisation of drug - stimulus interactions, and investigation of examples of drug - stimulus interactions was performed in collaboration with Peter-Martin Bruch for the paper Bruch \& Giles et al 2021. The author also received support from Frederich Zwiebell and Junyan Lu, who provided advice with regards to modelling approaches.

-check pdf and tidy
referencing
linear model
figures tidy
extra text

-check of text
= 6 done!

\hypertarget{chapter7}{%
\chapter{Results}\label{chapter7}}

\hypertarget{IL4resistance}{%
\subsection{IL4 induces resistance to BCR inhibitors and chemotherapeutics}\label{IL4resistance}}

(Figure \ref{fig:IL4Res}).


\begin{figure}

{\centering \includegraphics[width=0.8\linewidth]{thesis_files/figure-latex/IL4Res-1} 

}

\caption{Examples of IL4-induced resistance. Plots show log transformed viability values with each treatment, for all samples. Matching samples are linked across treatments. Black and blue horizontal lines indicate the predicted viability from the linear model for each single and combinatorial treatment. In combinatorial treatment, both the expected viability based on the additive effect of drug and stimulus (blue), and the viability with interaction (black) are shown, to indicate the impact of the interaction. Plots show IL4 + (A) Idelalisib (Pi3K), (B) PRT062607 (SYK) (C) Everolimus (mTOR), and (D) Selumetinib (p38 MAPK). See Methods section \ref{drug-stimulus-linear-model-method}. \emph{Figure from Bruch \& Giles et al.~2021.}}\label{fig:IL4Res}
\end{figure}
\begin{itemize}
\item
  Organize material and present results.
\item
  Use tables, figures (but prefer visual presentation):
  \begin{itemize}
  \item
    Tables and figures should supplement (and not duplicate) the text.
  \item
    Tables and figures should be provided with legends.
  \item
    \emph{Figure \ref{fig:graph} shows how to include and reference graphics.
    The graphic must be labelled before. Files must be in \textbf{.eps} format. You
    can do this really easily in R Markdown with \texttt{knitr::include\_graphics()}}!
  \item
    Figures can be referenced with \texttt{\textbackslash{}@ref(fig:\textless{}name\textgreater{})}, where \texttt{\textless{}name\textgreater{}} is the
    name of the code chunk.
  \end{itemize}
\end{itemize}
\begin{figure}

{\centering \includegraphics[width=0.5\linewidth]{figures/graph} 

}

\caption{Estimated residuals from model XXX4. ...}\label{fig:graph4}
\end{figure}
\begin{itemize}
\item
  Tables and graphics may appear in the text or in the appendix, especially if
  there are many simulation results tabulated, but is also depends on the study
  and number of tables resp. figures. The key graphs and tables must appear in
  the text!
\item
  R Markdown can also supports math equations just like \emph{LaTeX}!
  \begin{itemize}
  \item
    \emph{Equation \eqref{eq:SpecDens} represents the ACs of a stationary
    stochastic process:}
    \begin{equation}
            f_y(\lambda) = (2\pi)^{-1} \sum_{j=-\infty}^{\infty}
                           \gamma_j e^{-i\lambda j}
                         =(2\pi)^{-1}\left(\gamma_0 + 2 \sum_{j=1}^{\infty}
        \gamma_j \cos(\lambda j)\right)
                                       \label{eq:SpecDens}
    \end{equation}
    \emph{where \(i=\sqrt{-1}\) is the imaginary unit, \(\lambda \in [-\pi, \pi]\) is the
    frequency and the \(\gamma_j\) are the autocovariances of \(y_t\).}
  \item
    Equations can be referenced with \texttt{\textbackslash{}@ref(eq:\textless{}name\textgreater{})}, where name is defined
    by adding \texttt{(\textbackslash{}\#eq:\textless{}name\textgreater{})} in the line immediately before \texttt{\textbackslash{}end\{equation\}}.
  \end{itemize}
\end{itemize}
\hypertarget{review-of-results}{%
\section{Review of Results}\label{review-of-results}}
\begin{itemize}
\item
  Do the results support or do they contradict economic theory ?
\item
  What does the reader learn from the results?
\item
  Try to give an intuition for your results.
\item
  Provide robustness checks.
\item
  Compare to previous research.
\end{itemize}
\hypertarget{discussion-3}{%
\chapter{Discussion}\label{discussion-3}}

TLR modulation could be prognostic - it is modualted by ATM and del11q- this coudl explain:
Austen, B., Powell, J. E., Alvi, A., Edwards, I., Hooper, L., Starczynski, J., Taylor, A. M. R.,
Fegan, C., Moss, P. and Stankovic, T. (2005). Mutations in the ATM gene lead to impaired
overall and treatment-free survival that is independent of IGVH mutation status in patients
with B-CLL. Blood 106, 3175--82

TLR Signaling Is Activated in Lymph Node--Resident CLL Cells and Is Only Partially Inhibited by Ibrutinib
Eman L. Dadashian
\url{https://cancerres.aacrjournals.org/content/79/2/360.short}

tlr and fludarabine:
n Vitro Sensitivity of CLL Cells to Fludarabine May Be Modulated by the Stimulation of Toll-like Receptors
Eleonora Fonte

Expression and function of toll like receptors in chronic lymphocytic leukaemia cells
Marta Muzio 1 , Cristina Scielzo, Maria T S Bertilaccio, Michela Frenquelli, Paolo Ghia, Federico Caligaris-Cappio

Other ME singals startify patients: The functional in vitro response to CD40 ligation reflects a different clinical outcome in patients with chronic lymphocytic leukemia \url{https://www.nature.com/articles/leu2011149}

GEnetic resistance to BCR inhibition is well reserached, ME-induced resistance much less so:
(\protect\hyperlink{ref-tenHacken2016}{\textbf{tenHacken2016?}})
Mutations in BTK and PLCÎ³2 were identified through whole-exome sequencing of peripheral blood samples from patients experiencing relapse after ibrutinib treatment, including a cysteine-to-serine mutation at position 481 in BTK (C481S) leading to a protein product with reduced kinase activity, that is only partially inhibited by ibrutinib {[}179, 180{]}, and three putative gain-of-function mutations in PLCÎ³2, including arginine-to-tryptophan at position 665 (R665W), leucine to phenylalanine at position 845 (L845F) and serine to tyrosine at position 707 (S707Y) {[}179{]}.

Woyach JA, Furman RR, Liu TM, Ozer HG, Zapatka M, Ruppert AS, Xue L, Li DH, Steggerda SM, Versele M, Dave SS, Zhang J, Yilmaz AS, Jaglowski SM, Blum KA, Lozanski A, Lozanski G, James DF, Barrientos JC, Lichter P, Stilgenbauer S, Buggy JJ, Chang BY, Johnson AJ, Byrd JC. Resistance mechanisms for the Bruton's tyrosine kinase inhibitor ibrutinib. The New England journal of medicine. 2014;370:2286--2294.
180. Fegan C, Bagshawe J, Salles G, Karlin L, Rule S, Shah N, Morschhauser F, Terriou L, Dyer MJ, Walter H, Hutchinson C, Cartron G, Sharpe J, Duffy K, Nishimura A, Abe S, Honda H, Yasuhiro T, Yoshizawa T, Birkett J, Courtenay-Luck N. The Bruton's Tyrosine Kinase (BTK) Inhibitor ONO-4059: Promising Single Agent Activity and Well Tolerated in Patients with High Risk Chronic Lymphocytic Leukaemia (CLL) Blood (ASH annual meeting abstracts) 2014;124:3328.

\newpage

\hypertarget{references}{%
\chapter*{References}\label{references}}
\addcontentsline{toc}{chapter}{References}

\noindent

\setlength{\parindent}{-0.5cm}
\setlength{\leftskip}{0.5cm}
\setlength{\parskip}{8pt}

\hypertarget{refs}{}
\begin{CSLReferences}{1}{0}
\leavevmode\vadjust pre{\hypertarget{ref-Abruzzo2018}{}}%
Abruzzo, Lynne V., Carmen D. Herling, George A. Calin, Christopher Oakes, Lynn L. Barron, Haley E. Banks, Vikram Katju, Michael J. Keating, and Kevin R. Coombes. 2018. {``{Trisomy 12 chronic lymphocytic leukemia expresses a unique set of activated and targetable pathways}.''} \emph{Haematologica} 103 (12): 2069--78. \url{https://doi.org/10.3324/haematol.2018.190132}.

\leavevmode\vadjust pre{\hypertarget{ref-Aguilar-Hernandez2016}{}}%
Aguilar-Hernandez, Maria M, Matthew D Blunt, Rachel Dobson, Alison Yeomans, Stephen Thirdborough, Marta Larrayoz, Lindsay D Smith, et al. 2016. {``{IL-4 enhances expression and function of surface IgM in CLL cells}.''} \url{https://doi.org/10.1182/blood-2015-11}.

\leavevmode\vadjust pre{\hypertarget{ref-R-genomation}{}}%
Akalin, Altuna, Vedran Franke, Katarzyna Wreczycka, Alexander Gosdschan, Liz Ing-Simmons, and Bozena Mika-Gospodorz. 2021. \emph{Genomation: Summary, Annotation and Visualization of Genomic Data}. \url{http://bioinformatics.mdc-berlin.de/genomation/}.

\leavevmode\vadjust pre{\hypertarget{ref-Berest2019}{}}%
Berest, Ivan, Christian Arnold, Armando Reyes-Palomares, Giovanni Palla, Kasper Dindler Rasmussen, Holly Amelia Rebecca Giles, Peter-Martin Bruch, et al. 2019. {``{Quantification of Differential Transcription Factor Activity and Multiomics-Based Classification into Activators and Repressors: diffTF}.''} \emph{Cell Reports} 29 (10): 3147--3159.e12. \url{https://doi.org/10.1016/J.CELREP.2019.10.106}.

\leavevmode\vadjust pre{\hypertarget{ref-Bosch2019}{}}%
Bosch, Francesc, and Riccardo Dalla-Favera. 2019. {``{Chronic lymphocytic leukaemia: from genetics to treatment}.''} \emph{Nature Reviews Clinical Oncology} 16 (11): 684--701. \url{https://doi.org/10.1038/s41571-019-0239-8}.

\leavevmode\vadjust pre{\hypertarget{ref-Care2014}{}}%
Care, Matthew A., Mario Cocco, Jon P. Laye, Nicholas Barnes, Yuanxue Huang, Ming Wang, Sharon Barrans, et al. 2014. {``{SPIB and BATF provide alternate determinants of IRF4 occupancy in diffuse large B-cell lymphoma linked to disease heterogeneity}.''} \emph{Nucleic Acids Research} 42 (12): 7591--7610. \url{https://doi.org/10.1093/nar/gku451}.

\leavevmode\vadjust pre{\hypertarget{ref-Chan2015}{}}%
Chan, Chun Hin, Celestia Fang, Anna Yarilina, Rab K Prinjha, Yu Qiao, and Lionel B Ivashkiv. 2015. {``{BET bromodomain inhibition suppresses transcriptional responses to cytokine-Jak-STAT signaling in a gene-specific manner in human monocytes.}''} \emph{European Journal of Immunology} 45 (1): 287--97. \url{https://doi.org/10.1002/eji.201444862}.

\leavevmode\vadjust pre{\hypertarget{ref-Chatzouli2014}{}}%
Chatzouli, Maria, Stavroula Ntoufa, Nikos Papakonstantinou, Elisavet Chartomatsidou, Achilles Anagnostopoulos, Panagoula Kollia, Paolo Ghia, Marta Muzio, Kostas Stamatopoulos, and Chrysoula Belessi. 2014. {``{Heterogeneous Functional Effects of Concomitant B Cell Receptor and TLR Stimulation in Chronic Lymphocytic Leukemia with Mutated versus Unmutated Ig Genes}.''} \emph{The Journal of Immunology} 192 (10): 4518--24. \url{https://doi.org/10.4049/jimmunol.1302102}.

\leavevmode\vadjust pre{\hypertarget{ref-Collins1989}{}}%
Collins, Russell J., Louise A. Verschuer, Brian V. Harmon, Roger L. Prentice, John H. Pope, and John F. R. Kerr. 1989. {``{Spontaneous programmed death (apoptosis) of Bâchronic lymphocytic leukaemia cells following their culture in vitro}.''} \emph{British Journal of Haematology} 71 (3): 343--50. \url{https://doi.org/10.1111/j.1365-2141.1989.tb04290.x}.

\leavevmode\vadjust pre{\hypertarget{ref-JCIpaper}{}}%
Dietrich, Sascha, MaÅgorzata OleÅ, Junyan Lu, Leopold Sellner, Simon Anders, Britta Velten, Bian Wu, et al. 2017. {``{Drug-perturbation-based stratification of blood cancer}.''} \emph{Journal of Clinical Investigation} 128 (1): 427--45. \url{https://doi.org/10.1172/JCI93801}.

\leavevmode\vadjust pre{\hypertarget{ref-Dohner2000}{}}%
DÃ¶hner, H, Stephan Stilgenbauer, Axel Benner, Elke Leupolt, Alexander KrÃ¶ber, Lars Bullinger, Konstanze DÃ¶hner, Martin Bentz, and Peter Lichter. 2000. {``{Genomic aberrations and survival in chronic lymphocytic leukemia.}''} \emph{The New England Journal of Medicine} 343 (26): 1910--16. \url{https://doi.org/10.1056/NEJM200012283432602}.

\leavevmode\vadjust pre{\hypertarget{ref-R-msigdbr}{}}%
Dolgalev, Igor. 2021. \emph{Msigdbr: MSigDB Gene Sets for Multiple Organisms in a Tidy Data Format}. \url{https://igordot.github.io/msigdbr/}.

\leavevmode\vadjust pre{\hypertarget{ref-R-glmnet}{}}%
Friedman, Jerome, Trevor Hastie, Rob Tibshirani, Balasubramanian Narasimhan, Kenneth Tay, and Noah Simon. 2021. \emph{Glmnet: Lasso and Elastic-Net Regularized Generalized Linear Models}. \url{https://CRAN.R-project.org/package=glmnet}.

\leavevmode\vadjust pre{\hypertarget{ref-HerbstThesis}{}}%
Herbst, Sophie. 2020. {``{Systematic analysis of cell-intrinsic and extrinsic factors in chronic lymphocytic leukemia to understand functional consequences for drug response and clinical outcome}.''} PhD diss., University of Heidelberg. \url{https://archiv.ub.uni-heidelberg.de/volltextserver/27438/1/Doktorarbeit.pdf}.

\leavevmode\vadjust pre{\hypertarget{ref-R-survminer}{}}%
Kassambara, Alboukadel, Marcin Kosinski, and Przemyslaw Biecek. 2021. \emph{Survminer: Drawing Survival Curves Using Ggplot2}. \url{https://rpkgs.datanovia.com/survminer/index.html}.

\leavevmode\vadjust pre{\hypertarget{ref-Kienle2005}{}}%
Kienle, Dirk L, Christian Korz, Beate Hosch, Axel Benner, Daniel Mertens, Annett Habermann, Alexander KrÃ¶ber, et al. 2005. {``{Evidence for distinct pathomechanisms in genetic subgroups of chronic lymphocytic leukemia revealed by quantitative expression analysis of cell cycle, activation, and apoptosis-associated genes.}''} \emph{Journal of Clinical Oncology : Official Journal of the American Society of Clinical Oncology} 23 (16): 3780--92. \url{https://doi.org/10.1200/JCO.2005.02.568}.

\leavevmode\vadjust pre{\hypertarget{ref-Kipps2017}{}}%
Kipps, Thomas J, Freda K Stevenson, Catherine J Wu, Carlo M Croce, Graham Packham, William G Wierda, Susan O'Brien, John Gribben, and Kanti Rai. 2017. {``{Chronic lymphocytic leukaemia.}''} \emph{Nature Reviews. Disease Primers} 3: 16096. \url{https://doi.org/10.1038/nrdp.2016.96}.

\leavevmode\vadjust pre{\hypertarget{ref-R-pheatmap}{}}%
Kolde, Raivo. 2019. \emph{Pheatmap: Pretty Heatmaps}. \url{https://CRAN.R-project.org/package=pheatmap}.

\leavevmode\vadjust pre{\hypertarget{ref-HOCOMOCO}{}}%
Kulakovskiy, Ivan V, Ilya E Vorontsov, Ivan S Yevshin, Anastasiia V Soboleva, Artem S Kasianov, Haitham Ashoor, Wail Ba-Alawi, et al. 2016. {``{HOCOMOCO: expansion and enhancement of the collection of transcription factor binding sites models.}''} \emph{Nucleic Acids Research} 44 (D1): D116--25. \url{https://doi.org/10.1093/nar/gkv1249}.

\leavevmode\vadjust pre{\hypertarget{ref-R-DESeq2}{}}%
Love, Michael, Simon Anders, and Wolfgang Huber. 2021. \emph{DESeq2: Differential Gene Expression Analysis Based on the Negative Binomial Distribution}. \url{https://github.com/mikelove/DESeq2}.

\leavevmode\vadjust pre{\hypertarget{ref-MeierAbt2021}{}}%
Meier-Abt, Fabienne, Junyan Lu, Ester Cannizzaro, Marcel F. Pohly, Sandra Kummer, Sibylle Pfammatter, Laura Kunz, et al. 2021. {``{The Protein Landscape of Chronic Lymphocytic Leukemia (CLL)}.''} \emph{Blood}, June. \url{https://doi.org/10.1182/blood.2020009741}.

\leavevmode\vadjust pre{\hypertarget{ref-R-BloodCancerMultiOmics2017}{}}%
Oles, Malgorzata, Sascha Dietrich, Junyan Lu, Britta Velten, Andreas Mock, Vladislav Kim, and Wolfgang Huber. 2021. \emph{BloodCancerMultiOmics2017: "Drug-Perturbation-Based Stratification of Blood Cancer" by Dietrich s, Oles m, Lu j Et Al. - Experimental Data and Complete Analysis}.

\leavevmode\vadjust pre{\hypertarget{ref-R-base}{}}%
R Core Team. 2021. \emph{R: A Language and Environment for Statistical Computing}. Vienna, Austria: R Foundation for Statistical Computing. \url{https://www.R-project.org/}.

\leavevmode\vadjust pre{\hypertarget{ref-Rendeiro2016}{}}%
Rendeiro, AndrÃ© F., Christian Schmidl, Jonathan C. Strefford, Renata Walewska, Zadie Davis, Matthias Farlik, David Oscier, and Christoph Bock. 2016. {``{Chromatin accessibility maps of chronic lymphocytic leukaemia identify subtype-specific epigenome signatures and transcription regulatory networks}.''} \emph{Nature Communications} 7 (1): 11938. \url{https://doi.org/10.1038/ncomms11938}.

\leavevmode\vadjust pre{\hypertarget{ref-R-survival}{}}%
Therneau, Terry M. 2021. \emph{Survival: Survival Analysis}. \url{https://github.com/therneau/survival}.

\leavevmode\vadjust pre{\hypertarget{ref-Turkistany2011}{}}%
Turkistany, Shereen A, and Rodney P Dekoter. 2011. {``{The transcription factor PU.1 is a critical regulator of cellular communication in the immune system}.''} \url{https://doi.org/10.1007/s00005-011-0147-9}.

\leavevmode\vadjust pre{\hypertarget{ref-ConsensusClusterPlus}{}}%
Wilkerson, Matthew D, and D Neil Hayes. 2010. {``{ConsensusClusterPlus: a class discovery tool with confidence assessments and item tracking.}''} \emph{Bioinformatics (Oxford, England)} 26 (12): 1572--73. \url{https://doi.org/10.1093/bioinformatics/btq170}.

\leavevmode\vadjust pre{\hypertarget{ref-R-ConsensusClusterPlus}{}}%
Wilkerson, Matt, and Peter Waltman. 2021. \emph{ConsensusClusterPlus: ConsensusClusterPlus}.

\leavevmode\vadjust pre{\hypertarget{ref-Willis2017}{}}%
Willis, Simon N., Julie Tellier, Yang Liao, Stephanie Trezise, Amanda Light, Kristy O'Donnell, Lee Ann Garrett-Sinha, Wei Shi, David M. Tarlinton, and Stephen L. Nutt. 2017. {``{Environmental sensing by mature B cells is controlled by the transcription factors PU.1 and SpiB}.''} \emph{Nature Communications} 8 (1): 1426. \url{https://doi.org/10.1038/s41467-017-01605-1}.

\leavevmode\vadjust pre{\hypertarget{ref-R-ChIPseeker}{}}%
Yu, Guangchuang. 2021a. \emph{ChIPseeker: ChIPseeker for ChIP Peak Annotation, Comparison, and Visualization}. \url{https://guangchuangyu.github.io/software/ChIPseeker}.

\leavevmode\vadjust pre{\hypertarget{ref-R-clusterProfiler}{}}%
---------. 2021b. \emph{clusterProfiler: A Universal Enrichment Tool for Interpreting Omics Data}. \url{https://yulab-smu.top/biomedical-knowledge-mining-book/}.

\leavevmode\vadjust pre{\hypertarget{ref-Zenz2010}{}}%
Zenz, Thorsten, Daniel Mertens, Ralf KÃ¼ppers, Hartmut DÃ¶hner, and Stephan Stilgenbauer. 2010. {``{From pathogenesis to treatment of chronic lymphocytic leukaemia}.''} \emph{Nature Reviews Cancer} 10 (1): 37--50. \url{https://doi.org/10.1038/nrc2764}.

\end{CSLReferences}
\indent
\setlength{\parindent}{17pt}
\setlength{\leftskip}{0pt}
\setlength{\parskip}{0pt}

\newpage

\hypertarget{appendix}{%
\chapter*{Appendix}\label{appendix}}
\addcontentsline{toc}{chapter}{Appendix}

\hypertarget{figures}{%
\section*{Figures}\label{figures}}
\addcontentsline{toc}{section}{Figures}


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{figures/graph} 

}

\caption{Analysis of ATACseq dataset of two trisomy 12 and two non-trisomy 12 untreated CLL PMBC samples. The volcano plot depicts change in TF activity (x axis) versus BH-adjusted p-values (y axis) for 636 TFs, comparing trisomy 12 and non-trisomy 12 samples. The (\protect\hyperlink{ref-diffTF}{\textbf{diffTF?}}){[}REFERENCE{]} package was ran in analytical mode to calculate TF activity, measured as weighted mean difference. TFs are labeled if adjusted p-value \textless{} 0.01 and absolute weighted mean difference \textgreater{} 0.15.}\label{fig:diffTFsmallvolPlot}
\end{figure}
(ref:drugcytGeneIntAll)
\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/drugcytGeneIntAll-1} 

}

\caption{(ref:drugcytGeneIntAll)}\label{fig:drugcytGeneIntAll}
\end{figure}

\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/NutlinPredictors-1} 

}

\caption{ADD NUTLIN-3A Predictor profile}\label{fig:NutlinPredictors}
\end{figure}
\hypertarget{tables}{%
\section*{Tables}\label{tables}}
\addcontentsline{toc}{section}{Tables}


\begin{table}
\centering
\begin{tabular}{l|l|l|l}
\hline
Cytokine & Receptor & ENSEMBLID & Kegg\\
\hline
\cellcolor[HTML]{E2E868}{SDF-1a} & \cellcolor[HTML]{E2E868}{CXCR 4} & \cellcolor[HTML]{E2E868}{ENSG00000121966} & \cellcolor[HTML]{E2E868}{7852}\\
\hline
SDF-1a & CXCR 7 & ENSG00000144476 & 57007\\
\hline
\cellcolor[HTML]{E2E868}{IL-6} & \cellcolor[HTML]{E2E868}{IL-6 R} & \cellcolor[HTML]{E2E868}{ENSG00000160712} & \cellcolor[HTML]{E2E868}{3570}\\
\hline
IL-6 & IL-6 ST & ENSG00000134352 & 3572\\
\hline
\cellcolor[HTML]{E2E868}{IL-15} & \cellcolor[HTML]{E2E868}{IL-15 R alpha} & \cellcolor[HTML]{E2E868}{ENSG00000134470} & \cellcolor[HTML]{E2E868}{3601}\\
\hline
IL-15 & IL-2 R beta & ENSG00000100385 & 3560\\
\hline
\cellcolor[HTML]{E2E868}{IL-15} & \cellcolor[HTML]{E2E868}{IL-2 R gamma} & \cellcolor[HTML]{E2E868}{ENSG00000147168} & \cellcolor[HTML]{E2E868}{3561}\\
\hline
IL-21 & IL-21 R & ENSG00000103522 & 50615\\
\hline
\cellcolor[HTML]{E2E868}{IL-21} & \cellcolor[HTML]{E2E868}{IL-21 R gamma} & \cellcolor[HTML]{E2E868}{ENSG00000147168} & \cellcolor[HTML]{E2E868}{3561}\\
\hline
IL-2 & IL-2 R alpha & ENSG00000134460 & 3559\\
\hline
\cellcolor[HTML]{E2E868}{IL-2} & \cellcolor[HTML]{E2E868}{IL-2 R beta} & \cellcolor[HTML]{E2E868}{ENSG00000100385} & \cellcolor[HTML]{E2E868}{3560}\\
\hline
IL-2 & IL-2 R gamma & ENSG00000147168 & 3561\\
\hline
\cellcolor[HTML]{E2E868}{IL-10} & \cellcolor[HTML]{E2E868}{IL-10 R alpha} & \cellcolor[HTML]{E2E868}{ENSG00000110324} & \cellcolor[HTML]{E2E868}{3587}\\
\hline
IL-10 & IL-10 R beta & ENSG00000243646 & 3588\\
\hline
\cellcolor[HTML]{E2E868}{IL-4} & \cellcolor[HTML]{E2E868}{IL-4 R} & \cellcolor[HTML]{E2E868}{ENSG00000077238} & \cellcolor[HTML]{E2E868}{3566}\\
\hline
IL-4 & IL-2 R gamma & ENSG00000147168 & 3561\\
\hline
\cellcolor[HTML]{E2E868}{IL-4} & \cellcolor[HTML]{E2E868}{IL-13 R A1} & \cellcolor[HTML]{E2E868}{ENSG00000131724} & \cellcolor[HTML]{E2E868}{3597}\\
\hline
soluble anti-IgM & B cell receptor & ENSG00000275063 & 102723407\\
\hline
\cellcolor[HTML]{E2E868}{soluble anti-IgM} & \cellcolor[HTML]{E2E868}{CD79A} & \cellcolor[HTML]{E2E868}{ENSG00000105369} & \cellcolor[HTML]{E2E868}{973}\\
\hline
soluble anti-IgM & CD79B & ENSG00000007312 & 974\\
\hline
\cellcolor[HTML]{E2E868}{Resiquimod} & \cellcolor[HTML]{E2E868}{TLR 7} & \cellcolor[HTML]{E2E868}{ENSG00000196664} & \cellcolor[HTML]{E2E868}{51284Â }\\
\hline
Resiquimod & TLR 8 & ENSG00000101916 & 51311\\
\hline
\cellcolor[HTML]{E2E868}{CpG ODN} & \cellcolor[HTML]{E2E868}{TLR9} & \cellcolor[HTML]{E2E868}{ENSG00000239732} & \cellcolor[HTML]{E2E868}{54106}\\
\hline
BAFF & BAFF R & ENSG00000159958 & 115650\\
\hline
\cellcolor[HTML]{E2E868}{BAFF} & \cellcolor[HTML]{E2E868}{TACI} & \cellcolor[HTML]{E2E868}{ENSG00000240505} & \cellcolor[HTML]{E2E868}{23495}\\
\hline
BAFF & BCMA & ENSG00000048462 & 608\\
\hline
\cellcolor[HTML]{E2E868}{IL-1b} & \cellcolor[HTML]{E2E868}{IL-1 R1} & \cellcolor[HTML]{E2E868}{ENSG00000115594} & \cellcolor[HTML]{E2E868}{3554}\\
\hline
IL-1b & IL-1 R2 & ENSG00000115590 & 7850\\
\hline
\cellcolor[HTML]{E2E868}{sCD40L} & \cellcolor[HTML]{E2E868}{CD40} & \cellcolor[HTML]{E2E868}{ENSG00000101017} & \cellcolor[HTML]{E2E868}{958}\\
\hline
Interferon gamma & IFN-gamma R1 & ENSG00000027697 & 3459\\
\hline
\cellcolor[HTML]{E2E868}{Interferon gamma} & \cellcolor[HTML]{E2E868}{IFN-gamma R2} & \cellcolor[HTML]{E2E868}{ENSG00000159128} & \cellcolor[HTML]{E2E868}{3460}\\
\hline
TGF-b1 & TGF-beta R1 & ENSG00000106799 & 7046\\
\hline
\cellcolor[HTML]{E2E868}{TGF-b1} & \cellcolor[HTML]{E2E868}{TGF-beta R2} & \cellcolor[HTML]{E2E868}{ENSG00000163513} & \cellcolor[HTML]{E2E868}{7048}\\
\hline
\end{tabular}
\end{table}
\begin{table}[ht]
    \begin{center}
        {\footnotesize
        \begin{tabular}{l|cccccccccc}
        \hline \hline
                        & 3m    & 6m    & 1yr   & 2yr   & 3yr   & 5yr   & 7yr   & 10yr  & 12yr  & 15yr   \\
            \hline
                Mean   & 3.138 & 3.191 & 3.307 & 3.544 & 3.756 & 4.093 & 4.354 & 4.621 & 4.741 & 4.878  \\
                Median & 3.013 & 3.109 & 3.228 & 3.490 & 3.680 & 3.906 & 4.117 & 4.420 & 4.575 & 4.759  \\
                Min    & 1.984 & 1.950 & 1.956 & 2.010 & 2.240 & 2.615 & 2.850 & 3.120 & 3.250 & 3.395  \\
                Max    & 5.211 & 5.274 & 5.415 & 5.583 & 5.698 & 5.805 & 5.900 & 6.031 & 6.150 & 6.295  \\
                StD    & 0.915 & 0.919 & 0.935 & 0.910 & 0.876 & 0.825 & 0.803 & 0.776 & 0.768 & 0.762  \\
            \hline \hline
        \end{tabular}}
    \end{center}
    \caption{Detailed descriptive statistics of location and dispersion for
    2100 observed swap rates for the period from
    February 15, 1999 to March 2, 2007. Swap rates measured as 3.12 (instead of 0.0312).}
    \label{tab:apptable}
\end{table}
\newpage

\end{document}
