\documentclass[11pt, a4paper, twosided]{book}
\usepackage[hmarginratio=1:1]{geometry} % equal right and left raitios
\usepackage{amsmath,amssymb,amsfonts,amsthm}    % Typical maths resource packages
\usepackage{graphicx}                           % Packages to allow inclusion of graphics
\usepackage[authoryear]{natbib}                 % literature reference style
\usepackage[bf]{caption}
\usepackage{textcomp}                           % For single quotes
\usepackage{floatrow}                           % For image and table position
\usepackage{booktabs}                           % For tables
% \usepackage[colorlinks=true]{hyperref}
% \usepackage[bottom]{footmisc}
\usepackage[bottom, flushmargin]{footmisc}                   % For footnotes
\usepackage[citebordercolor={0 1 0}]{hyperref}  % For creating hyperlinks in cross references
\usepackage{footnotebackref}
\usepackage[utf8]{inputenc}
\usepackage[T1]{fontenc}
% i have added the below, so that kable works
\usepackage{colortbl}
\usepackage{longtable}
\usepackage{array}
\usepackage{multirow}
\usepackage{wrapfig}
\usepackage{float}
\usepackage{pdflscape}
\usepackage{tabu}
\usepackage{threeparttable}
\usepackage{threeparttablex}
\usepackage[normalem]{ulem}
\usepackage{makecell}
\usepackage{xcolor}

       % For being able to set `echo = TRUE` in code chunk options, and thus in-line representation of code

%--------------------------------------------
% define line spacing = 1.5
\renewcommand{\baselinestretch}{1.5}

% define position of graphics
\floatsetup[figure]{capposition=bottom}
\floatsetup[table]{capposition=top}
\floatplacement{figure}{ht}
\floatplacement{table}{ht}

% define CSL references (required)
\newlength{\cslhangindent}
\setlength{\cslhangindent}{1.5em}
\newenvironment{CSLReferences}%
  {}%
  {\par}

% save thesis parameters for later
\newcommand{\thesistype}{Dissertation}
\newcommand{\thesisauthor}{Holly Amelia Rebecca Giles}
\newcommand{\thesisdate}{Oct 01, 2021?}

% Added by CII (Thanks, Hadley!)
% Use ref for internal links
\renewcommand{\hyperref}[2][???]{\autoref{#1}}
\def\chapterautorefname{Chapter}
\def\sectionautorefname{Section}
\def\subsectionautorefname{Subsection}
% End of CII addition

% Added by CII
\usepackage{caption}
\captionsetup{width=5in}
% End of CII addition

% Added by CII
%%% Copied from knitr
%% maxwidth is the original width if it's less than linewidth
%% otherwise use linewidth (to make sure the graphics do not exceed the margin)
\makeatletter
\def\maxwidth{ %
  \ifdim\Gin@nat@width>\linewidth
    \linewidth
  \else
    \Gin@nat@width
  \fi
}
\makeatother

\renewcommand{\contentsname}{Table of Contents}
% End of CII addition

\setlength{\parskip}{0pt}

% Added by CII
  %\setlength{\parskip}{\baselineskip}
  \usepackage[parfill]{parskip}

\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}

	\usepackage{float}
% End of CII addition
%%
%% End Preamble
%%
%
\begin{document}

% ------------------------------------
% ---- frontmatter: First Page -------
% ------------------------------------

\thispagestyle{empty}
  \begin{center}
  {\Large{Dissertation\\
    submitted to the\\
    Combined Faculty of Natural Sciences and Mathematics\\
    of the Ruperto Carola University Heidelberg, Germany\\
    for the degree of\\
    Doctor of Natural Sciences\\}} \vspace*{\fill}

  \Large{Presented by} \\\vspace{0.5cm}
  \Large{M.A. Holly Amelia Rebecca Giles} \\
  \Large{born in Portsmouth} \\
  \Large{Oral examination: Dec 15, 2021?}

\end{center}
\newpage\null\thispagestyle{empty}\newpage

% ------------------------------------
% --- frontmatter: Secound Page ------
% ------------------------------------

\pagebreak
\thispagestyle{empty}
\null
\begin{center}
\Large{Drug - microenvironment - gene interplay in chronic lymphocytic leukemia} \\
\vspace*{\fill}
\begin{tabular}{ c c }
Referees: & Dr.~Judith Zaugg \\
          & Prof.~Dr.~Michael Boutros
\end{tabular}
\end{center}
\newpage\null\thispagestyle{empty}\newpage

% --------------------------------------------
% -------- Declaration of Authorship ---------
% --------------------------------------------

\newpage
\thispagestyle{empty}
% \hypertarget{declaration-of-authorship}{%
% \section*{Declaration of Authorship}\label{declaration-of-authorship}}
%
% I hereby confirm that I have authored this Dissertation independently and
% without the use of others sources than the ones indicated. I have not yet
% presented this thesis or parts thereof to a university as part of an
% examination or degree. This work was carried out in the Genome Biology at
% the European Molecular Biology Laboratory (EMBL) in the group of Dr.~Wolfgang Huber and Dr.~Sascha Dietrich.
% \vspace{1cm}
%
% \noindent Heidelberg, \thesisdate{}
% \vspace{3cm}
%
% \noindent . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
% \vspace{0.1cm}
%
% \noindent \thesisauthor{}
% \newpage
% \thispagestyle{empty}
% \vspace*{\fill}
% Holly Amelia Rebecca Giles, \textit{Drug - microenvironment - gene interplay in chronic lymphocytic leukemia}, \textcopyright\, \thesisdate{}
% \newpage

%\frontmatter % this stuff will be roman-numbered
\pagestyle{empty} % this removes page numbers from the frontmatter



% edit style below this point? 


% --------------------------------------------
% --------------- Dedication -----------------
% --------------------------------------------
\vspace*{\fill}
\begin{center}
\textit{To \ldots{}}
\end{center}
\vspace*{\fill}
\newpage\null\thispagestyle{empty}\newpage

% ------------------------------------
% --- frontmatter: Acknowledgement ---
% ------------------------------------
\pagestyle{plain}
\pagenumbering{roman}   %  define page number in roman style
\setcounter{page}{1}    % start page numbering anew
\newpage
\hypertarget{acknowledgements}{%
\chapter*{Acknowledgements}\label{acknowledgements}}
\addcontentsline{toc}{chapter}{Acknowledgements}

I want to thank a few people.
\newpage\null\newpage


% --------------------------------------------
% ---------- List of publications ------------
% --------------------------------------------
    \hypertarget{list-of-publications}{%
    \chapter*{List of publications}\label{list-of-publications}}
    \addcontentsline{toc}{chapter}{List of publications}

    \hypertarget{thesis-related}{%
    \subsubsection*{Thesis related}\label{thesis-related}}
    \addcontentsline{toc}{subsubsection}{Thesis related}
    \begin{itemize}
        \item  Peter-Martin Bruch\ast,  \textbf{\underline{Holly A. R. Giles}}\ast, Carolin Kolb, Sophie Herbst, Tina Becirovic, Tobias Roider, Junyan Lu, Sebastian Scheinost, Lena Wagner, Jennifer Huellein, Ivan Berest, Mark Kriegsmann, Katharina Kriegsmann, Christiane Zgorzelski, Peter Dreger, Judith Zaugg, Carsten Mueller-Tidow, Thorsten Zenz, Wolfgang Huber, Sascha Dietrich et al. \textit{in preparation.}  "Mapping drug-microenvironment-genetic interplay in CLL reveals trisomy 12 as a modulator of microenvironment."\textbf{\textit{bioRxiv}}. [doi: 10.1101/2021.07.23.453514](https://doi.org/10.1101/2021.07.23.453514)
    \end{itemize}
    \hypertarget{other-contributions}{%
    \subsubsection*{Other contributions}\label{other-contributions}}
    \addcontentsline{toc}{subsubsection}{Other contributions}

    The author of this thesis also contributed to a number of other projects throughout the PhD. The following have been published:
    \begin{itemize}
        \item Berest, Ivan\ast, Christian Arnold\ast, Armando Reyes-Palomares, Giovanni Palla, Kasper Dindler Rasmussen, \textbf{\underline{Holly Giles}}, and Peter-Martin Bruch et al. 2019. "Quantification Of Differential Transcription Factor Activity And Multiomics-Based Classification Into Activators And Repressors: DiffTF". \textit{Cell Reports} 29 (10): 3147-3159.e12. [doi: 10.1016/j.celrep.2019.10.106](https://doi.org/10.1016/j.celrep.2019.10.106)
        
      \item Lu, Junyan\ast, Ester Cannizzaro\ast, Fabienne Meier-Abt, Sebastian Scheinost, Peter-Martin Bruch, \textbf{\underline{Holly A. R. Giles}}, and Almut Lütge et al. 2021. "Multi-Omics Reveals Clinically Relevant Proliferative Drive Associated With Mtor-MYC-OXPHOS Activity In Chronic Lymphocytic Leukemia". \textbf{\textit{Nature Cancer}}. [doi: 10.1038/s43018-021-00216-6](https://doi.org/10.1038/s43018-021-00216-6)
    \end{itemize}
\newpage\null\newpage

% -------------------------------------
% --- frontmatter: Zusammendfassung ---
% -------------------------------------
    \hypertarget{zusammenfassung}{%
    \chapter*{Zusammenfassung}\label{zusammenfassung}}
    \addcontentsline{toc}{chapter}{Zusammenfassung}

    Abstract in German
    Aktuelle Entwicklungen im Bereich der ``omik''-Technologien tragen wesentlich zurBeschleunigung des Fortschritts in der Krebsforschung bei etc
\newpage\null\newpage

% -----------------------------
% --- frontmatter: Abstract ---
% -----------------------------

    \hypertarget{abstract}{%
    \chapter*{Abstract}\label{abstract}}
    \addcontentsline{toc}{chapter}{Abstract}

    The tumour microenvironment and genetic alterations collectively influence disease biology and drug resistance in Chronic Lymphocytic Leukaemia (CLL). To establish an integrative understanding of these factors in CLL biology, we performed a combinatorial assay using 12 drugs individually co-applied with each of 17 microenvironmental stimuli on 192 samples of CLL peripheral blood mononucleated cells. We examined microenvironment response across a heterogeneous patient cohort and identified four distinct CLL subgroups that differed in their response landscapes and in patient outcomes. By combining our data with whole-exome sequencing, DNA-methylation, RNA-sequencing and copy number variant data of the same tumours, we systematically searched for molecular determinants of stimulus response and found trisomy 12 as a key modulator. Our data suggest that the amplifying effect of trisomy 12 on the response to environmental signals is mediated by the transcription factors Spi-B and PU.1. We generated a comprehensive map of drug-microenvironment interactions in CLL, and profiled the modulating impact of genetic features on these antagonistic and synergistic effects. Interleukin (IL) 4 and Toll-Like Receptor (TLR) 7/8/9 stimuli showed the most interactions. Both pathways were more active in CLL-infiltrated lymph nodes than in healthy samples (p\textless0.001), and high IL4 activity in lymph nodes correlated with shorter survival (p=0.038). We provide a multi-layered resource to investigate microenvironmental and drug interplay in CLL (\href{https://git.embl.de/giles/cllcytokinescreen2021}{Repository} \& \href{https://www.google.com/url?q=https://www.imbi.uni-heidelberg.de/dietrichlab/CLL_Microenvironment/\&sa=D\&source=editors\&ust=1625764640786000\&usg=AOvVaw01HOFP7k2BJRLTteCXa8qy}{Shiny}). Our results highlight the importance of cell-extrinsic influences on drug response and disease progression, and how these further depend on molecular features.
\newpage\null\newpage

% -----------------------------
% --- frontmatter: Contents ---
% -----------------------------
\newpage
\tableofcontents
\clearpage

% ----------------------------------------------------------
% --- frontmatter: List of Abbreviations (not mandatory) ---
% ----------------------------------------------------------
\newpage
\hypertarget{list-of-abbreviations}{%
\chapter*{List of Abbreviations}\label{list-of-abbreviations}}
\addcontentsline{toc}{chapter}{List of Abbreviations}
\begin{tabular}{rp{0.2cm}lp{1cm}rp{0.2cm}l}
    AKT & & Protein kinase B \\
    AUC & & Area under curve \\
    BCR & & B-cell receptor \\
    BH & & Benjamini-Hochberg \\
    BTK & & Bruton’s tyrosine kinase \\
    CDF & & Cumulative Distribution Function \\
    CLL & & Chronic lymphocytic leukemia \\
    CNV & & Copy number variation \\
    CpG ODN & & CpG oligodeoxynucleotides \\
    ERK/MAPK signalling & & Mitogen-activated protein kinase signalling \\
    FBS & & Fetal bovine serum \\
    FDR & & False discovery rate \\
    GSEA & & Gene set enrichment analysis  \\
    Ig & & Immunoglobulin \\
    IGHV & & Immunoglobulin heavy chain variable region \\
    IL & & Interleukin \\
    JAK & & Janus kinase \\
    M-CLL & & CLL with somatic hypermutations in IGHV loci \\
    NF$\kappa$B & & Nuclear factor kappa-light-chain-enhancer of activated B cells \\
    NOTCH & & Neurogenic locus notch  \\
    OS & & Overall survival \\
    PBMC & & Peripheral blood mononuclear cells \\
    PCA & & Principal component analysis \\
    PI3K & & Phosphoinositide 3-kinase \\
    U-CLL & & CLL without somatic hypermutations in IGHV loci \\
    TGF$\beta$ & & Transforming growth factor $\beta$ \\
    TF & & Transcription factor \\
    TLR & & Toll-like receptor \\ 
    TSS & & Transcription start site \\
    TTT & & Time to next treatment \\
    SF3B1 & & Splicing factor 3B subunit 1 \\
    STAT & & Signal transducer and activator of transcription \\
    SYK & & Spleen tyrosine kinase \\
    VST & & Variance Stabilising Transformation \\
    WES & & Whole exome sequencing \\
    WGS & & Whole genome sequencing \\
    WT & & Wild type \\
    ZAP70 & & Tyrosine-protein kinase ZAP-70 \\

    
   
\end{tabular}
% ----------------------------------------------------
% --- frontmatter: List of Figures (not mandatory) ---
% ----------------------------------------------------
\newpage
\listoffigures
\addcontentsline{toc}{section}{List of Figures}

% ---------------------------------------------------
% --- frontmatter: List of Tables (not mandatory) ---
% ---------------------------------------------------
\newpage
\listoftables
\addcontentsline{toc}{section}{List of Tables}

% edit here to affect style of main document 
%\mainmatter % here the regular arabic numbering starts
\newpage
\pagestyle{plain}
\setcounter{page}{1}    % start page numbering anew
\pagenumbering{arabic}  % page numbers in arabic style

\hypertarget{introduction}{%
\chapter{Introduction}\label{introduction}}

Next tasks:
Give a bit of time each day for next few weeks, then full tie when chapter 5 and 6 finished
Continue to go through add the next papers
Then go throuhg each section, do a lit review, and read what i need for each, add the text and references and kkep going
Should then be easy to tie this together

(theoretical background)
\#\# CLL Disease biology

\hypertarget{disease-characteristcs}{%
\subsection{Disease Characteristcs}\label{disease-characteristcs}}

CD5, CD19, CD23+, lowlevelsofCD20

Reading:

From Kipps 2017:
Incidence (get for 2021)
Siegel R, et al.~Cancer treatment and survivorship statistics, 2012. CA Cancer J. Clin. 2012; 62:220--241. {[}PubMed: 22700443{]}

\ref{fig:CLLmorphology} Taken from Kipps et al 2017: ``Wright--Giemsa-stained blood smears showing the typical chronic lymphocytic leukaemia (CLL) B lymphocyte (part a). \_Figure from (\protect\hyperlink{ref-Kipps2017._}{\textbf{Kipps2017.\_?}})''
\begin{figure}

{\centering \includegraphics{figures/CLL_morphology} 

}

\caption{(ref:CLLmorphology)}\label{fig:CLLmorphology}
\end{figure}
\emph{Figure taken from Kipps et al 2017}

\ref{fig:proliferationCentres} Taken from Kipps et al 2017: ``a \textbar{} Tissue sections of a lymph node stained with haemotoxylin and eosin showing numerous pale-staining pseudofollicles, which are circled (original magnification ×20).'' \_Figure from (\protect\hyperlink{ref-Kipps2017_}{\textbf{Kipps2017\_?}})
\begin{figure}

{\centering \includegraphics{figures/proliferation_centres} 

}

\caption{(ref:proliferationCentres)}\label{fig:proliferationCentres}
\end{figure}
\emph{Figure taken from Kipps et al 2017}

(\protect\hyperlink{ref-Kipps2017}{Kipps et al. 2017}) 10.1038/nrdp.2016.96.Chronic
``Chronic lymphocytic leukaemia (CLL) is a malignancy of CD5+ B cells that is characterized by the accumulation of small, mature-appearing lymphocytes in the blood, marrow and lymphoid tissues. Signalling via surface immunoglobulin, which constitutes the major part of the B cell receptor, and several genetic alterations play a part in CLL pathogenesis, in addition to interactions between CLL cells and other cell types, such as stromal cells, T cells and nurse-like cells in the lymph nodes. The clinical progression of CLL is heterogeneous and ranges from patients whorequire treatment soon after diagnosis to others who do not require therapy for many years, if at all. Several factors, including the immunoglobulin heavy-chain variable region gene (IGHV) mutational status, genomic changes, patient age and the presence of comorbidities, should be considered when defining the optimal management strategies, which include chemotherapy, chemoimmunotherapy and/or drugs targeting B cell receptor signalling or inhibitors of apoptosis, such as BCL-2. Research on the biology of CLL has profoundly enhanced our ability to identify patients who are at higher risk for disease progression and our capacity to treat patients with drugs that selectively target distinctive phenotypic or physiological features of CLL.''

Get for 2021:
CLL is estimated to account for \textasciitilde19,000 of all newly detected cancers in the United States in 2016 (REF. 11). The average incidence of CLL varies between individuals in different geographical regions and ranges from \textless0.01\% of individuals in eastern Asia to \textasciitilde0.06\% of individuals in Europe and the United States. The risk of developing CLL is about two-times higher for men than for women and increases with age; the median age at diagnosis ranges from 70 to 72 years11--14.
The

\hypertarget{ighv-status-and-the-role-of-bcr-signalling}{%
\subsection{IGHV status and the role of BCR signalling}\label{ighv-status-and-the-role-of-bcr-signalling}}

Reading:
From Kipps 2017:
IGHV status:
Hamblin TJ, Davis Z, Gardiner A, Oscier DG, Stevenson FK. Unmutated Ig V(H) genes are associated with a more aggressive form of chronic lymphocytic leukemia. Blood. 1999; 94:1848-- 1854. {[}PubMed: 10477713{]}

Damle RN, et al.~Ig V gene mutation status and CD38 expression as novel prognostic indicators in chronic lymphocytic leukemia. Blood. 1999; 94:1840--1847. {[}PubMed: 10477712{]} References 1 and 2 are landmark papers that describe two main subsets of patients with different disease progression tendencies based on IGHV mutation status of the immunoglobulins that are expressed by CLL cells.

Restricted repetoire:
Kipps TJ, et al.~Developmentally restricted immunoglobulin heavy chain variable region gene expressed at high frequency in chronic lymphocytic leukemia. Proc. Natl Acad. Sci. USA. 1989; 86:5913--5917. {[}PubMed: 2503826{]} This paper describes the discovery that the immunoglobulin repertoire of CLL cells may be highly restricted, suggesting that the antibodies expressed by CLL cells are most likely selected based on their capacity to bind to some common self-antigens.

Fais F, et al.~Chronic lymphocytic leukemia B cells express restricted sets of mutated and unmutated antigen receptors. J. Clin. Invest. 1998; 102:1515--1525. {[}PubMed: 9788964{]}

Widhopf G, F 2nd, et al.~Chronic lymphocytic leukemia B cells of more than 1\% of patients express virtually identical immunoglobulins. Blood. 2004; 104:2499--2504. {[}PubMed: 15217828{]}

\ref{fig:BCRsignalling} Adapted from: Kipps et al 2017: ``B cell receptor (BCR) signalling is initiated by SRC-family kinase-dependent phosphorylation (mainly LYN) of CD79A and CD79B that creates a docking site for the binding and activation of spleen tyrosine kinase (SYK). SYK then triggers the formation of a multi-component `signalosome,' comprising Bruton tyrosine kinase (BTK), AKT, phosphoinositide 3-kinase (PI3K), phospholipase Cγ2 (PLCγ2) and B cell-linker protein (BLNK), among others. CD19 is a co-receptor for BCR and is important for PI3K activation, which recruits and activates PLCγ2, BTK and AKT. PLCγ2 generates diacylglycerol (DAG) and inositol-1,4,5-trisphosphate (Ins(1,4,5)P3), which triggers Ca2+ release from the endoplasmic reticulum, leading to the activation of the MEK--extracellular signal-regulated kinase (ERK) and nuclear factor-κB (NF-κB) signalling pathways. Other effects of BCR signalling include activation of mechanistic target of rapamycin complex 1 (mTORC1) and of Rho-family GTPases, RAC1 and RHOA, which can affect the cytoskeleton. Inhibitors of SYK, PI3K and BTK are shown. Note that this figure describes the main molecules and interactions that are involved in positive BCR signalling, but is not an exhaustive description of all signalling pathways or molecules activated. IKK, IκB kinase; PKC, protein kinase C.''
\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{figures/BCR_signalling} 

}

\caption{(ref:BCRsignalling)}\label{fig:BCRsignalling}
\end{figure}
\ref{fig:IGHVstatus} Taken from Kipps et al 2017: ``Normal naive B cells that have undergone successful V(D)J recombination and express functional B cell receptors that are capable of binding to antigen interact with CD4+ T cells and accessory cells, which aggregate to form follicles that become germinal centres. Germinal cells each have a dark zone, comprising rapidly dividing B cells, and a light zone, comprising B cells mixed with follicular dendritic cells (FDCs), macrophages and helper T cells (TH cells). The B cells enter the dark zone of the germinal centre where they experience rapid proliferation and somatic hypermutation (SHM) in the genes encoding the immunoglobulin variable regions of the heavy chain (IGHV) and the light chain (IGVL). As they pass through to the light zone, the B cells that express the fittest B cell receptors for binding antigen are selected and may undergo immunoglobulin class-switch recombination. Chronic lymphocytic leukaemia (CLL) cells that use unmutated IGHV apparently originate from CD5+ B cells prior to experiencing SHM, whereas CLL cells that use mutated IGHV most likely originate from CD5+ B cells that have passed through and differentiated in the germinal centre. Some CLL cells might be derived from B cells that also have undergone immunoglobulin class-switch recombination and express immunoglobulin isotypes other than IgM and IgD, for example, IgG or IgA. Another subset is one with CLL cells that express immunoglobulin with only modest somatic mutations, such as CLL cells that use IGHV3-21 with \textasciitilde97\% homology to the inherited IGHV3-21 gene and an immunoglobulin light chain encoded by an unmutated IGLV3-21; these cells might derive from a B cell that has had constrained SHM, possibly owing to a limited need for immunoglobulin somatic diveresification and selection. Dashed arrows indicate speculated pathways.'' \_Figure from (\protect\hyperlink{ref-Kipps2017_}{\textbf{Kipps2017\_?}}).
\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{figures/IGHV_status} 

}

\caption{(ref:IGHVstatus)}\label{fig:IGHVstatus}
\end{figure}
(\protect\hyperlink{ref-Kipps2017}{Kipps et al. 2017})
The BCR is composed of a ligand-binding trans-membrane immunoglobulin molecule (either IgA, IgD, IgE, IgG or IgM) and the signalling Igα (also known as CD79A)--Igβ (also known as CD79B) heterodimer. CLL cells typically co-express IgD and IgM, although at low levels compared with normal B cells; less than a few per cent of CLL cases express class-switched isotypes, most commonly IgG. The CD79A and CD79B molecules contain immunoreceptor tyrosine-based activation motifs, which can be phosphorylated following the crosslinking of surface immunoglobulin, thereby triggering BCR signalling. A functional BCR is required for the survival of mature B cells76 and is maintained in most mature B cell malignancies, including CLL. In CLL, evidence suggests that the surface immunoglobulin of CLL B cells is engaged by autoantigen, which leads to constitutive BCR signalling in vivo77--79. The importance of this interaction is underscored by the clinical success of kinase inhibitors that block BCR signalling (see Management), although effects on other receptors might also have a role80.
Like most cancers, CLL is heterogeneous and the outcome of BCR signalling ranges from enhanced B cell activation to B cell anergy81,82. The main pathways that lead to cell survival and proliferation downstream of the BCR are shown in FIG. 3, along with drugs targeted against key signalling intermediates. BCR signalling that leads to anergy is less well defined, but seems to involve biased activation of inhibitory molecules with only partial activation of the pathways that are typically associated with B cell activation81. One important molecule that may be involved is the inositol lipid phosphatase SHIP1. SHIP1 is activated by the tyrosine-protein kinase LYN and may limit B cell activation by counteracting phosphoinositide 3-kinase (PI3K) activity at both chronically engaged receptors and distant non-ligated BCRs, rendering them insensitive to stimulation82,83.
Enhanced B cell activation is more commonly observed in CLL that expresses unmutated IGHV, whereas anergy predominates in most cases of CLL that express mutated IGHV84. Anergy is a state of cellular lethargy induced by chronic engagement of the surface antigen receptors in the absence of adequate T cell help. Although capable of reversing their phenotype, anergic cells are less likely to proliferate in response to BCR signalling than more activated cells, which might, in part, account for the observation that patients with CLL cells that express mutated IGHV generally have more indolent disease than patients with CLL cells with unmutated IGHV85. The fate of the cell (activation versus anergy) might be influenced by the CLL cell of origin (FIG. 1), as the cell types that can form CLL differ in their patterns of DNA methylation73, and are likely to respond differently to autoantigens. An unresolved question is whether anergy can be reversed in vivo, mirroring what occurs in vitro78.
The BCR also coordinates the activity of other cell surface receptors, including integrins, such as α4β1 integrin. BCR stimulation can result in increased adhesion of CLL cells to α4β1 integrin substrates, for example, fibronectin and vascular cell adhesion protein 1 (REF. 86). By contrast, CXC-chemokine receptor 4 (CXCR4) is downmodulated by BCR engagement and both can trigger `inside-out' signalling, resulting in the activation of α4β1 integrin87,88. Thus, recognized antigen encountered in lymphoid tissue is likely to affect adhesion and migration of CLL cells. Modulation of these pathways, coupled with the role of BTK and PI3K in chemokine receptor signalling89, contribute to the increased lymphocytosis observed in patients upon initiation of treatment with inhibitors of BTK or PI3K (see Management).

(\protect\hyperlink{ref-Kipps2017}{Kipps et al. 2017}) 10.1038/nrdp.2016.96.Chronic
``CLL can be divided into two main subsets, which differ in their clinical behaviour. These subsets are distinguished by whether CLL cells express an unmutated or mutated immunoglobulin heavy-chain variable region gene (IGHV), reflecting the stage of normal B cell differentiation from which they originate1,2. CLL cells that express an unmutated IGHV originate from a B cell that has not undergone differentiation in germinal centres, which are the sites in the lymph nodes where B cells experience somatic hypermutation in their immunoglobulin variable region genes and selection during an immune response. Patients with CLL cells that express an unmutated IGHV typically have more-aggressive disease than patients with CLL cells that express a mutated IGHV. CLL cells with mutated IGHV arise from a post-germinal centre B cell that expresses immunoglobulin that has undergone somatic hypermutation and, in some cases, also immunoglobulin isotype switching (FIG. 1), similar to what occurs in normal B cells during an immune response to antigen. It should be emphasized that the high level of somatic mutations that arise in IGHV in the germinal centre are a natural part of affinity maturation of antibodies and, unlike mutations in other genes, are not pathological. The tumours are simply reflecting the stage of maturation of the parental B cell. In addition, some CLL cells have been described that are similar to unmutated IGHV CLL, but originate from B cells with limited somatic mutation, such as CLL with immunoglobulin heavy chains encoded by mutated IGHV3--21 and immunoglobulin light chains encoded by unmutated IGLV3--21 (REFS 3,4).
The repertoire of immunoglobin molecules produced by the CLL cells of all patients is considerably more limited than the repertoire of immunoglobulin molecules that can be made by the B cells of any one person5,6, reflecting the biased use in CLL of certain IGHV genes that have restricted somatic mutation and limited junctional and heavy-light chain combinatorial diversity. In as many as one-third of patients, the CLL cells express immunoglobulin `stereotypes,' which are stretches of primary structure in the variable region that can also be identified in the immunoglobulins produced by the CLL cells of other patients7. The restricted immunoglobulin repertoire in CLL is underscored by the finding that \textasciitilde1 in 75 patients have CLL cells that express immunoglobulin molecules that are virtually identical8. The limited immunoglobulin diversity provides compelling evidence that CLL B cells are selected based on the binding activity of their expressed surface immunoglobulin, suggesting that B cell receptor (BCR) signalling plays a crucial part in CLL pathogenesis''

\hypertarget{recurrent-genetic-features-in-cll}{%
\subsection{Recurrent genetic features in CLL}\label{recurrent-genetic-features-in-cll}}

Genetic alterations in CLL can include chromosomal alterations, mutations, alterations in the expression of mi RNAs and epigenetic modifications.

(for purpose of TLR sotry)
These included del(11q) and \emph{ATM} ,
del(17p) and \emph{TP53},
\emph{SF3B1}.

REading:
From Kipps et al:
38 Dohner H, et al.~Genomic aberrations and survival in chronic lymphocytic leukemia. N. Engl. J. Med. 2000; 343:1910--1916. {[}PubMed: 11136261{]}

Del(13q)mechanism: 39. Klein U, et al.~The DLEU2/miR-15a/16-1 cluster controls B cell proliferation and its deletion leads to chronic lymphocytic leukemia. Cancer Cell. 2010; 17:28--40. {[}PubMed: 20060366{]}

Del(17p) mechanism 40. Zenz T, Mertens D, Kuppers R, Dohner H, Stilgenbauer S. From pathogenesis to treatment of chronic lymphocytic leukaemia. Nat. Rev.~Cancer. 2010; 10:37--50. {[}PubMed: 19956173{]}

del(11q)

Mutations and outcome:
38: Dohner H, et al.~Genomic aberrations and survival in chronic lymphocytic leukemia. N. Engl. J. Med. 2000; 343:1910--1916. {[}PubMed: 11136261{]}

JIC paper, explaon PACE
Improved in recent years: 41: Van Dyke DL, et al.~The Dohner fluorescence in situ hybridization prognostic classification of chronic lymphocytic leukaemia (CLL): the CLL Research Consortium experience. Br. J. Haematol. 2016; 173:105--113. {[}PubMed: 26848054{]}

CLl ahrbours a high degree of genetic diversity:
Fabbri G, et al.~Analysis of the chronic lymphocytic leukemia coding genome: role of NOTCH1 mutational activation. J. Exp. Med. 2011; 208:1389--1401. {[}PubMed: 21670202{]}
43. Pleasance ED, et al.~A comprehensive catalogue of somatic mutations from a human cancer genome. Nature. 2010; 463:191--196. {[}PubMed: 20016485{]}
44. Puente XS, et al.~Whole-genome sequencing identifies recurrent mutations in chronic lymphocytic leukaemia. Nature. 2011; 475:101--105. {[}PubMed: 21642962{]}
45. Wang L, et al.~SF3B1 and other novel cancer genes in chronic lymphocytic leukemia. N. Engl. J. Med. 2011; 365:2497--2506. {[}PubMed: 22150006{]}

Functional role of mutations:
WNT: 47 Wang L, et al.~Somatic mutation as a mechanism of Wnt/beta-catenin pathway activation in CLL. Blood. 2014; 124:1089--1098. {[}PubMed: 24778153{]}

POT1: reference thereview

SF3B1:
45 Wang L, et al.~SF3B1 and other novel cancer genes in chronic lymphocytic leukemia. N. Engl. J. Med. 2011; 365:2497--2506. {[}PubMed: 22150006{]}
48 Ferreira PG, et al.~Transcriptome characterization by RNA sequencing identifies a major molecular and clinical subdivision in chronic lymphocytic leukemia. Genome Res. 2014; 24:212--226. {[}PubMed: 24265505{]}
\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\setcounter{enumi}{48}
\tightlist
\item
  Quesada V, et al.~Exome sequencing identifies recurrent mutations of the splicing factor SF3B1 gene in chronic lymphocytic leukemia. Nat. Genet. 2012; 44:47--52.
\item
  Te Raa GD, et al.~The impact of SF3B1 mutations in CLL on the DNA-damage response. Leukemia. 2015; 29:1133--1142. {[}PubMed: 25371178{]}
\end{enumerate}
Seqencing studies:
two seminal studies:
51. Landau DA, et al.~Mutations driving CLL and their evolution in progression and relapse. Nature. 2015; 526:525--530. {[}PubMed: 26466571{]}
52. Puente XS, et al.~Non-coding recurrent mutations in chronic lymphocytic leukaemia. Nature. 2015; 526:519--524. {[}PubMed: 26200345{]} References 51 and 52 describe landmark studies on whole-exome sequencing of CLL cells obtained from a large cohort of patients, reporting new driver mutations in CLL.

Epigenetic changes:
Global hypomethylation and local hypermethylation
66. Cahill N, et al.~450K--array analysis of chronic lymphocytic leukemia cells reveals global DNA
methylation to be relatively stable over time and similar in resting and proliferative compartments. Leukemia. 2013; 27:150--158. {[}PubMed: 22922567{]}
67. Wahlfors J, et al.~Genomic hypomethylation in human chronic lymphocytic leukemia. Blood. 1992; 80:2074--2080. {[}PubMed: 1382719{]}
68. Ziller MJ, et al.~Charting a dynamic DNA methylation landscape of the human genome. Nature. 2013; 500:477--481. {[}PubMed: 23925113{]}

Methylation for classification:
69. Kulis M, et al.~Epigenomic analysis detects widespread gene-body DNA hypomethylation in chronic lymphocytic leukemia. Nat. Genet. 2012; 44:1236--1242. {[}PubMed: 23064414{]} This paper describes an analysis of DNA methylation in CLL cases, showing that the two molecular subtypes of CLL have differing DNA methylomes that seem to represent epigenetic imprints from distinct normal B cell subpopulations.
\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\setcounter{enumi}{73}
\tightlist
\item
  Bhoi S, et al.~Prognostic impact of epigenetic classification in chronic lymphocytic leukemia: the
  case of subset \#2. Epigenetics. 2016; 11:449--455. {[}PubMed: 27128508{]}
\end{enumerate}
\ref{fig:CLLmutations} Taken from Kipps et al 2017: ``Genes that are mutated in chronic lymphocytic leukaemia (CLL) are involved in several cellular pathways (blue boxes). As such, mutations in these genes can lead to a range of cellular consequences, such as aberrant DNA repair and B cell receptor (BCR) signalling, among others51,213. The minus sign from GBN1 to the MAPK--ERK pathway indicates negative regulation. *For more detail of the BCR and its associated signalling, see FIG. 3. ASXL1, additional sex combs-like protein 1; ATM, ataxia telangiectasia mutated; BAZ2A, bromodomain adjacent to zinc-finger domain protein 2A; BCOR, BCL-6 co-repressor; BIRC3, baculoviral IAP repeat-containing protein 3; BRCC3, BRCA1/BRCA2-containing complex subunit 3; C-NOTCH, carboxy-terminal domain of NOTCH; CARD11, caspase recruitment domain-containing protein 11; CHD2, chromodomainhelicase-DNA-binding protein 2; CHK2, checkpoint kinase 2; Co-A, co-activator; CSL, CBF1--Suppressor of Hairless--LAG1 (also known as RBPJ); DDX3X, ATP-dependent RNA helicase DDX3X; DYRK1A, dual-specificity tyrosine-phosphorylation-regulated kinase 1A; EGR2, early growth response 2; ELF4, ETS-related transcription factor Elf-4; ERK, extracellular signal-regulated kinase; EWSR1, Ewing sarcoma breakpoint region 1 protein; FBXW7, F-box/WD repeat-containing protein 7; FUBP1, far upstream element-binding protein 1; GNB1, guanine nucleotide-binding protein G(I)/G(S)/G(T) subunit β1; H3K4, histone H3 lysine 4; IC, intracellular domain; IKZF3, Ikaros family zinc-finger protein 3; IL-1R, IL-1 receptor; IRF4, interferon regulatory factor 4; ITPKB, inositol-trisphosphate 3-kinase B; LRP, low-density lipoprotein receptor-related protein; MAP2K1, dual-specificity mitogen-activated protein kinase kinase 1; MAPK, mitogen-activated protein kinase; MED12, Mediator of RNA polymerase II transcription subunit 12; MGA, MAX gene-associated protein; MYD88, myeloid differentiation primary response protein MyD88; NF-κB, nuclear factor-κB; NXF1, nuclear RNA export factor 1; P, phosphate; POT1, protection of telomeres protein 1; PTPN11, tyrosine-protein phosphatase non-receptor type 11; RIPK1, receptor-interacting serine/threonine-protein kinase 1; RPS15, 40S ribosomal protein S15; SAMHD1, SAM domain and HD domain-containing protein 1; SF3B1, splicing factor 3B subunit 1; SHP1, Src homology region 2 domain-containing phosphatase 1 (also known as PTPN6); SYK, spleen tyrosine kinase; TCF/LEF, T cell factor/lymphoid enhancer factor; TLR8, Toll-like receptor 8; TNFR1, tumour necrosis factor receptor 1 (also known as TNFRSF1A); TRAF, TNFR-associated factor; XPO, exportin; ZMYM3, zinc-finger MYM-type protein 3. Adapted with permission from REF. 51, Macmillan Publishers Limited.'' \_Figure from (\protect\hyperlink{ref-Kipps2017_}{\textbf{Kipps2017\_?}})
\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{figures/genetic_mutations} 

}

\caption{(ref:CLLmutations)}\label{fig:CLLmutations}
\end{figure}
\emph{mutations}
(\protect\hyperlink{ref-Kipps2017}{Kipps et al. 2017})
Several large genetic studies have revealed numerous genetic alterations in CLL, including single- nucleotide polymorphisms (SNPs), chromosomal alterations and alterations in non- coding RNA, such as microRNA (miRNA), some of which can be used to determine prognosis and to guide management strategies. Interactions between CLL cells and their microenvironment, including interactions with other cell types, such as T cells, nurse-like cells and stromal cells, can induce B cell proliferation and contribute to disease.

(\protect\hyperlink{ref-Kipps2017}{Kipps et al. 2017})
The altered expression of genes that are located in or near CLL-associated SNPs might contribute to disease development. For example, a SNP in IRF4 is associated with low expression of interferon regulatory factor 4; mice that are deficient in this protein can develop CLL23, partly owing to hyperactivation of Notch signalling24.

23: Shukla V, Ma S, Hardy RR, Joshi SS, Lu R. A role for IRF4 in the development of CLL. Blood. 2013; 122:2848--2855. {[}PubMed: 23926303{]}
24. Shukla V, Shukla A, Joshi SS, Lu R. Interferon regulatory factor 4 attenuates notch signaling to suppress the development of chronic lymphocytic leukemia. Oncotarget. 2016; 7:41081--41094. {[}PubMed: 27232759{]}

(\protect\hyperlink{ref-Kipps2017}{Kipps et al. 2017})
Somatic mutations---The advent of massively parallel sequencing and the application of whole-exome sequencing to CLL have transformed our understanding of the genetic heterogeneity of CLL and have established that CLL harbours a high degree of genetic variability42--45 (FIG. 2). From these studies, recurrent somatic mutations have been consistently observed in genes that have a role in DNA damage (for example, TP53 and ATM), mRNA processing (for example, SF3B1 and XPO1), chromatin modification (for example, HIST1H1E, CHD2 and ZMYM3), WNT signalling, Notch signalling (for example, NOTCH1) and inflammatory pathways (for example, MYD88). Other mutations, such as those found in EGR2 or BRAF, can affect B cell-related signalling and transcription46 (FIG. 2).
The functional role of several putative driver mutations has been confirmed; for example, silencing mutated WNT pathway genes in primary CLL cells resulted in decreased cell viability47. Mutations in POT1, which has a role in the protection of telomeres, prevented the binding of protection of telomeres protein 1 to telomeric DNA, resulting in numerous chromosomal abnormalities, in addition to the development of abnormal telomeres. Mutations in SF3B1 have been found to be associated with aberrant RNA splicing45,48,49 and an altered DNA-damage response50. SAMHD1 encodes a protein involved in the regulation of intracellular deoxy-nucleotide pools, which are recruited to the site of DNA damage and are probably involved in the response to DNA double-strand breaks50.
The detection of somatic mutations and their relative frequencies is variable, which possibly reflects differences in the composition of the cohorts studied worldwide. Two seminal studies have provided the largest sequenced collections to date51,52, in which \textgreater500 CLL samples were characterized by whole-exome sequencing or whole- genome sequencing. The clinical and/or biological features of the patients examined in each study were notably distinct; one study analysed matched pretreatment samples from patients who required initial treatment and noted mutations in SF3B1 (21\% of patients), ATM (15\% of patients), TP53 (7\% of patients), NOTCH1 (6\% of patients) or BIRC3 (4\% of patients). The other study assessed patients with earlystage CLL and patients with monoclonal B cell lymphocytosis and identified NOTCH1 (12.6\% of patients), A T M (11\% of patients), BIRC3 (8.8\% of patients) and SF3B1 (8.6\% of patients) as the most frequently mutated genes.
These large sample cohorts have provided the sensitivity to discover novel candidate cancer genes that are altered in CLL. Both studies also identified somatic mutations in MGA and PTPN11, which encode modulators of MYC, IKZF3, which encodes a key transcription factor, and RPS15, which encodes 40S ribosomal protein S15 and is recurrently mutated in \textasciitilde20\% of patients who relapse after combination chemotherapy53. Other recurrent somatic mutations include those in the 3′ untranslated region of NOTCH1 and a PAX5 enhancer, which increases the expression of these B cell-associated transcription factors54,55. Patients with mutations in the 3′ untranslated region of NOTCH1 have a shorter time from diagnosis to treatment and poorer overall survival, similar to that of patients with non- synonymous mutations, which alter the amino acid sequence of NOTCH1.
Next-generation sequencing has revealed intra-tumoural heterogeneity in CLL. Some somatic mutations, such as those in MYD88, or chromosomal abnormalities, such trisomy 12 or del(13q), are most often found in all the CLL cells of any one patient, indicating that these genetic alterations occurred early in the evolution of the leukaemia. Other mutations, such as those found in SF3B1 or NOTCH1, or chromosomal alterations, such as del(17p), are typically found in only a fraction of the leukaemia cells and thus represent subclonal events, which occur after the development of CLL. Across studies, subclonal driver mutations are associated with more-aggressive disease, particularly when two or more are found concurrent in a leukaemia cell population51,56,57. In addition, studies have demonstrated that large clonal shifts can occur following chemotherapy, owing to increases in the proportions of CLL cells that have a TP53 mutation or del(17p), indicating that such genetic changes provide a strong fitness advantage in the setting of therapy51. By contrast, one study of CLL cells from patients treated with ibrutinib (an inhibitor of Bruton tyrosine kinase (BTK)), revealed mutations associated with drug resistance that were distinct from those observed in CLL cells of patients treated with standard chemotherapy58. CLL was the first human disease that was found to be associated with alterations in miRNA.

(can compare size of large studies with ours)

\emph{copy number variants}
(\protect\hyperlink{ref-Kipps2017}{Kipps et al. 2017})
Approximately 80\% of patients with CLL carry at least one of four common chromosomal alterations: a deletion in chromosome 13q14.3 (del(13q)), del(11q), del(17p) and trisomy 12 (REF. 38). Del(13q) is the most common chromosomal alteration, evident in \textgreater50\% of patients, and is associated with favourable prognosis. Within this deleted region is the DLEU2-mir-15--16 cluster, which regulates the expression of proteins that can inhibit apoptosis or that are involved in cell cycle progression39 (see Supplementary information S2 (table)). Del(17p) is found in 7\% of patients and is associated with loss of the tumour suppressor gene TP53 (REF. 40), whereas del(11q) is found in 18\% of patients and is often associated with alterations in ATM (which encodes a protein involved in DNA repair); each of these chromosomal alterations is associated with adverse clinical outcome38, although this has improved in recent years41. Trisomy 12 is found in 16\% of patients with CLL and is associated with an intermediate prognosis. Unlike the neoplastic B cells in mantle cell lymphoma, CLL cells do not have the translocation t(11;14) (q13;q32) or other genetic alterations that enhance the expression of CCND1, which encodes the cell cycle regulator cyclin D1

\emph{Epigenetic changes}
The CLL epigenome shows global hypomethylation combined with local hypermethylation, as has been observed in other cancers66--68. Indeed, comprehensive methylation profiling has demonstrated substantial intra-tumoural methylation heterogeneity52,69--73. Increasing methylation heterogeneity has consistently been associated with increased genetic complexity owing to the acquisition of subclonal mutations, thus linking genomic and methylomic evolution in CLL52,70,71. Indeed, locally disordered methylation in CLL might enhance the evolutionary adaptive capacity of CLL cells by increasing the background `noise' of the genome, thereby providing increased opportunities for somatic mutations within the leukaemia clone. In support of this notion is the observed association between methylation evolution and adverse clinical outcome52,70,71.

(\protect\hyperlink{ref-Kipps2017}{Kipps et al. 2017})
Methylation signatures can classify distinct clinical CLL subgroups69,74. As these methylation patterns are a heritable trait, they have been used to `trace' back to the type of normal B cell from which the CLL cells were derived75. These studies revealed that the CLL cells of different patients derive from a continuum of B cell maturation states, which are not restricted to discrete maturation stages. Nevertheless, CLL cells that use unmutated IGHV versus mutated IGHV generally have distinctive methylation patterns, which respectively approximate to those of pre-germinal centre versus post-germinal centre memory B cells, as depicted in FIG. 1. The diversity in the likely cells of origin of CLL cells highlights the biological and phenotypic heterogeneity of this disease. These findings also suggest that epigenetic programming that is dependent of transcription factors has a potentially important role in the development of CLL.

\hypertarget{intro-trisomy12}{%
\subsection{The incompletely understood role of trisomy12}\label{intro-trisomy12}}

Bosch, F., Dalla-Favera, R. Chronic lymphocytic leukaemia: from genetics to treatment. Nat Rev Clin Oncol 16, 684--701 (2019). \url{https://doi.org/10.1038/s41571-019-0239-8}

Trisomy 12 chronic lymphocytic leukemia expresses a unique set of activated and targetable pathways
DOI: \url{https://doi.org/10.3324/haematol.2018.190132}
Issue: Vol. 103 No.~12 (2018): December, 2018

The Protein Landscape of Chronic Lymphocytic Leukemia (CLL)
Fabienne Meier-Abt \url{https://doi.org/10.1182/blood.2020009741}

what's understood about the overespssopn of genes and proliferative drive;
DOI: 10.1200/JCO.2005.02.568 Journal of Clinical Oncology 23, no. 16 (June 01, 2005) 3780-3792.

\hypertarget{stratification-of-cll-patients}{%
\subsection{Stratification of CLL patients}\label{stratification-of-cll-patients}}

Reading:
(from Kipps)
\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\setcounter{enumi}{11}
\tightlist
\item
  Nabhan C, et al.~The impact of race, ethnicity, age and sex on clinical outcome in chronic lymphocytic leukemia: a comprehensive Surveillance, Epidemiology, and End Results analysis in the modern era. Leuk. Lymphoma. 2014; 55:2778--2784. {[}PubMed: 24588735{]}
\item
  Li Y, Wang Y, Wang Z, Yi D, Ma S. Racial differences in three major NHL subtypes: descriptive epidemiology. Cancer Epidemiol. 2015; 39:8--13. {[}PubMed: 25560974{]}
\item
  Pulte D, Redaniel MT, Bird J, Jeffreys M. Survival for patients with chronic leukemias in the US and Britain: age-related disparities and changes in the early 21st century. Eur. J. Haematol. 2015; 94:540--545. {[}PubMed: 25315799{]}
\end{enumerate}
\ref{fig:CLLstratification} Taken from Kipps et al 2017: ``Indications for therapy of patients with chronic lymphocytic leukaemia (CLL) include late-stage disease, evidence for rapid disease progression or disease-related symptoms. Patients with del(17p) or mutated TP53 should be treated with therapy that does not require functional TP53, such as ibrutinib (a Bruton tyrosine kinase (BTK) inhibitor), given the relatively poor outcome for such patients with chemotherapy. For patients without del(17p) or known mutations in TP53, immunoglobulin heavy-chain variable region (IGHV) mutational status can help to define the treatment strategy; patients with unmutated IGHV could be considered for therapy with a BTK inhibitor (such as ibrutinib) and patients with mutated IGHV might be good candidates for chemoimmunotherapy (CIT), if amenable. Indeed, patients with mutated IGHV can have excellent outcomes with CIT regimens, such as fludarabine, cyclophosphamide and rituximab, with \textgreater50\% of patients having a median progression-free survival of \textgreater10 years, including the potential for cure. If the patient is amenable to CIT, age, medical comorbidities and myeloid reserve should be taken into consideration. Patients \textgreater65 years of age commonly have medical comorbidities and are less able to tolerate myelosuppressive regimens, such as fludarabine, cyclophosphamide and rituximab. Thus, considerations should be given to using reduced dose or less myelosuppressive chemotherapy regimens, such as chlorambucil or reduced-dose bendamustine and an anti-CD20 monoclonal antibody for patients with limited myeloid reserve. Patients who either do not respond, have a poor tolerance to CIT or relapse following CIT, should be re-evaluated for del(17p) or TP53 mutations. Patients who develop de novo del(17p) or TP53 mutations, or have known del(17p) and/or TP53 mutations, or who develop resistance or intolerance to ibrutinib, could be considered for therapy with idelalisib and rituximab or the BCL-2 inhibitor venetoclax. Patients treated with CIT who do not have del(17p) or TP53 mutations could be considered for repeat CIT if their progression-free survival after CIT is \textgreater2 years and the patient has sufficient myeloid reserve. Such patients also might be treated with a BTK inhibitor or a phosphoinositide 3-kinase (PI3K) inhibitor, which also could be considered for patients who develop intolerance or resistance to therapy with ibrutinib. Patients who develop resistance or intolerance to inhibitors of BTK, PI3K and/or BCL-2 should be considered for clinical trials or alternative agents. LDT, lymphocyte doubling time.'' \emph{Figure from \protect\hyperlink{ref-Kipps2017}{Kipps et al.} (\protect\hyperlink{ref-Kipps2017}{2017}) }.
\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{figures/disease_stratification} 

}

\caption{(ref:CLLstratification)}\label{fig:CLLstratification}
\end{figure}
Disease staging
(\protect\hyperlink{ref-Kipps2017}{Kipps et al. 2017}) Two clinical staging systems are widely used to divide patients with CLL into three broad prognostic groups114,115. The Rai staging system (TABLE 1) is more commonly used in the United States, whereas the Binet classification (TABLE 2) is more commonly used in Europe. The staging systems each recognize the importance of marrow function and define late-stage, or high-risk, disease by the presence of pronounced anaemia or thrombocytopenia.

(\protect\hyperlink{ref-Kipps2017}{Kipps et al. 2017})
Prognostic factors that can help to identify patients who require therapy relatively soon after diagnosis include certain clinical features and genetic, molecular and biochemical characteristics of the CLL cell. Multivariable models, prognostic indexes116--118 and nomograms119 have been developed to consolidate such prognostic factors so that they can more robustly predict clinical outcome. Commonly used parameters that are associated with poorer outcome are male sex,
65 years of age, poor performance status due to medical
comorbidities, certain CLL cell characteristics, such as the expression of unmutated IGHV1,2, ZAP70 (REFS 120,121), CD49d (also known as integrin α4)122 or CD38 (REF.2), the presence of del(17p)38 or del(11q)123, high serum levels of β2-microglobulin (\textgreater3.5 mg per l)124, complex karyotype (that is, the presence of three or more chromosomal aberrations observed on a karyotype test)125,126, or a high absolute lymphocyte count (\textgreater50,000 cells per µl) and/or late-stage disease at initial presentation. Del(17p) is often associated with inactivating mutations in TP53 and is a predictor of poor outcome to treatment with regimens that involve conventional chemotherapy127.

(\protect\hyperlink{ref-Kipps2017}{Kipps et al. 2017})
Generally, indications to initiate therapy include pronounced disease-related anaemia or thrombocytopenia (patients with Rai stage III or stage IV disease, or Binet stage C disease), symptomatic lymphadenopathy and/or symptoms that are associated with active disease, such as night sweats, fatigue, unintentional weight loss or fever without evidence of infection107. However, when basing a treatment decision on constitutional symptoms alone, the physician should consider other medical conditions, such as hypothyroidism, hyperthyroidism, hypoglycaemia, chronic inflammation, uncommon opportunistic infections or sleep disorders, including sleep apnoea.

For patients who need treatment, the presence of del(17p) or mutated TP53 are the most important features that are currently directing the choice of therapy (FIG. 8). Next, advanced age of \textgreater65 years, the presence of medical comorbidities and the objectives of treatment have substantial bearing on the choice of therapy. Increasingly, IGHV mutational status is considered as a parameter when determining the type of therapy; for example, chemotherapy-based regimens are reserved for patients with CLL and mutated IGHV. Conversely, the specific Rai or Binet stage of the patient who requires treatment does not necessarily influence the choice of therapy.
Systemic

\hypertarget{prognosis-and-where-the-problems-lie}{%
\subsection{Prognosis and where the problems lie}\label{prognosis-and-where-the-problems-lie}}

\emph{harsh treatments, especially when most are elderly}
\emph{not a proper cure - combo therapies may cover all bases}
\emph{heterogenous - combo therapies may cover all bases}
The clinical course of newly diagnosed CLL is extremely variable; some patients remain free of symptoms and are fully active for decades, whereas others rapidly become symptomatic or develop high-risk disease, which requires treatment soon after diagnosis and might result in death due to therapy-related and/or disease-related complications. However, most patients have a clinical course that is in between these two extremes.

(\protect\hyperlink{ref-Kipps2017}{Kipps et al. 2017})
Adverse effects of ibrutinib include fatigue, diarrhoea, bleeding, ecchymoses, rash, arthralgia, myalgia, increased blood pressure and atrial fibrillation. Clinical trials are currently evaluating second-generation BTK inhibitors (for example, acalabrutinib148, ONO/ GS-4059 (REF. 149) or BGB-3111) to determine whether any one of these drugs has a superior therapeutic index than that of ibrutinib150.

(\protect\hyperlink{ref-Kipps2017}{Kipps et al. 2017}) Toxicities of venetoclax include gastrointestinal disturbances, neutropenia and tumour lysis syndrome159, which is characterized by hyperkalaemia, hyper-uricaemia and/or azotaemia. Tumour lysis syndrome results from the rapid destruction of cancer cells and the release of their cellular contents into the blood. Tumour lysis syndrome typically occurs when initiating venetoclax therapy or when dosing is increased. Thus, patients start venetoclax with a low daily dose, which is escalated each week over 5 weeks to mitigate the risk of developing tumour lysis syndrome. Even with this strategy, patients who are at high risk for tumour lysis syndrome because of bulky lymphadenopathy and/or lymphocytosis of \textgreater25,000 cells per µl must be hydrated and closely monitored during therapy initiation and during dose escalation.

(\protect\hyperlink{ref-Kipps2017}{Kipps et al. 2017})
The outlook for patients with CLL has improved substantially over the past several years. Through research on the immune biology and genetics of CLL, patients can be stratified into subgroups with distinctive clinical features, which has improved our capacity to assess prognosis or govern therapy. However, an understanding of the mechanisms that contribute to immune dysfunction or how it contributes to autoimmune disease, such as autoimmune haemolytic anaemia, therapy resistance or therapy-related complications is unknown. Whether tyrosine kinase inhibitors can affect clonal evolution, induce and/or select for drug resistance, or can achieve complete responses if used earlier in the course of the disease is also unknown.

New therapies:\\
(\protect\hyperlink{ref-Kipps2017}{Kipps et al. 2017})
Several therapies are currently under preclinical and clinical investigation for the treatment of patients with CLL, including new drugs and treatment modalities that can modulate the immune system, and cell transplantation.

\emph{Car-T Therapies}
(\protect\hyperlink{ref-Kipps2017}{Kipps et al. 2017})
CD19-targeted CAR T cells have yielded long-term PFS and relapse-free survival durations in patients with CLL; in 14 patients with relapsed or refractory CLL, four patients achieved a complete response and four patients achieved a partial response197. None of the complete responders had MRD and none relapsed (a median follow-up of 19 months). However, the efficacy of CAR T cell therapy in patients with CLL has been modest compared with that in patients with acute lymphoblastic leukaemia; this might be owing to qualitative defects in the T cells of patients with CLL, who are generally older than patients with acute lymphoblastic leukaemia and already have immune dysfunction that reflects disease-associated anergy (see BCR and B cell signalling). Ibrutinib might partially correct some of these defects198. Larger

\emph{Allogenic stem cell transplantation}
(\protect\hyperlink{ref-Kipps2017}{Kipps et al. 2017})
Allogeneic stem cell transplantation is a potentially curative strategy for patients with relapsed or refractory CLL, including patients with high-risk features such as del(17p).Research efforts are ongoing to develop better-tolerated cell-based therapy with a similar curative potential that can be used without the immunosuppression and associated long-term morbidity and mortality of allogeneic stem cell transplantation.
Donor availability, advanced patient age, associated toxicities of myelosuppression, graft- versus-host disease and impaired resistance to infections limit the application of allogeneic stem cell transplantation in patients with CLL. In addition, the advent of BCR signalling inhibitors and BCL-2 inhibitors provide multiple treatment options that afford well- tolerated, long-term disease control, making allogeneic stem cell transplantation the least desirable option for most patients.

\emph{Immune-modulatory drugs}
(\protect\hyperlink{ref-Kipps2017}{Kipps et al. 2017})
Immune-modulatory drugs, such as thalidomide and lenalidomide, are approved for the treatment of patients with multiple myeloma, mantle cell lymphoma or myelodysplastic disease. Although these drugs have clinical activity in patients with CLL, they have had limited application unless used in combination with an anti-CD20 monoclonal antibody166,186. In CLL, lenalidomide can induce the expression of p21WAF1/CIP1, which inhibits cyclin-dependent kinase and CLL cell proliferation187, and can improve immune synapse formation, potentially enhancing immune function.

\emph{Immune Checkpoint inhibitors}
(\protect\hyperlink{ref-Kipps2017}{Kipps et al. 2017})
The immune checkpoint receptor PD-1, and its ligands PD-L1 and PD-L2, is the most important cognate receptor involved in the suppression of cellular immune activation. CLL cells express high levels of PD-L1 and PD-L2 and can suppress the responses of PD-1- expressing effector T cells106, leading to an exhausted (that is, no longer functional) T cell phenotype. Preclinical studies in mouse models have demonstrated that checkpoint inhibitors can reactivate immune effector cells to have anti-leukaemia activity202. However,

\emph{Combination targeted therapies}
(\protect\hyperlink{ref-Kipps2017}{Kipps et al. 2017})
We can now target the distinctive phenotypic or physiological features of CLL with targeted therapeutic agents, which have a higher therapeutic index than standard chemotherapy. Through the use of combination therapy, which targets different B cell survival signalling pathways and/or achieves better eradication of CLL cells, we might be able to define curative treatments for most patients with this disease.
Research on leukaemia cell survival signalling pathways, such as those governed by interactions between leukaemia cells and cells or secreted factors within the microenvironment (FIG. 4), might identify pathways that are not affected by BCR inhibitors. For example, BCR inhibitors, such as ibrutinib, cannot block ROR1-dependent WNT5A signalling, which enhances CLL cell proliferation, migration and survival93; as such, antibodies that block ROR1-dependent signalling could potentially have synergistic activity when used in combination with BCR inhibitors203. Furthermore, the interaction between CLL cells and accessory cells in the microenvironment might enhance CLL cell expression of anti-apoptotic proteins other than BCL-2, such as MCL1, thereby contributing to therapy resistance. As such, the therapeutic use of a selective BCL-2 antagonist, such as venetoclax, might be more effective when used in combination with BCL-2 inhibitors161,162, which also interfere with the homing of CLL cells to the microenvironment. Conceivably, combination target therapy with agents that have synergistic activity will provide highly effective and potentially curative treatment of patients with CLL.

\hypertarget{the-tumour-microenvironment-in-cll}{%
\section{The tumour microenvironment in CLL}\label{the-tumour-microenvironment-in-cll}}

\hypertarget{the-concept-of-the-protective-niche}{%
\subsection{The concept of the protective niche}\label{the-concept-of-the-protective-niche}}

\ref{fig:microenvironmentTissues} \emph{Adapted from Davids et al 2012}
\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{figures/microenvironmentTissues} 

}

\caption{(ref:microenvironmentTissues)}\label{fig:microenvironmentTissues}
\end{figure}
Lymph nodes etc

(\protect\hyperlink{ref-Kipps2017}{Kipps et al. 2017})
CLL cells depend on survival signals that they receive in lymphoid tissues from neighbouring non-neoplastic cells within the so-called cancer microenvironment. CLL cells follow chemokine gradients into lymph nodes, where they form `proliferation centres' (REF. 77), as opposed to normal germinal centres. In these proliferation centres, the CLL cells contact non- malignant stromal cells, nurselike cells (also known as lymphoma-associated macrophages), T cells and mesenchymal-derived stromal cells (FIG. 4). Engagement with autoantigen may occur during this transit, thereby stimulating CLL cell activation and proliferation if sufficient T cell help is available. Only a few per cent of the CLL cells undergo proliferation at any one time; the remainder of the cells are either unstimulated or driven into anergy84. However, within such proliferation centres, all CLL cells are exposed to chemokines, integrins, cytokines and survival factors.

\hypertarget{intro-ME-pathways}{%
\subsection{Microenvironmental pathways and cross-talk}\label{intro-ME-pathways}}

pathways esp IL4 and TLR\\
The microenvironment in CLL constitutes cross talk via soluble factors and cell-cell contacts, in the blood, bone marrow and lymph nodes.

\ref{fig:microenvironmentOverview} \emph{Adapted from: Weistner et al 2015}
\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{figures/microenvironmentCrosstalk} 

}

\caption{(ref:microenvironmentOverview)}\label{fig:microenvironmentCrosstalk}
\end{figure}
\ref{fig:microenvironmentSignalling} Taken from Kipps et al 2017; ``Migration of chronic lymphocytic leukaemia (CLL) cells into the lyphoid tissu is primarily mediated through CXC-chemokine receptor 4 (CXCR4) in response to CXC-chemokine ligand 12 (CXCL12), which is secreted mainly by nurse-like cells (NLCs) and mesenchymal-derived stromal cells. Migration of CLL cells into lymph nodes also occurs via CC-chemokine receptor 7 (CCR7) in response to CC-chemokine ligand 19 (CCL19) and CCL21, which are produced by the endothelial cells of high endothelial venules (HEVs). HEV endothelial cells also express hyaluronan, which can interact with CD44, to facilitate B cell signalling and might enhance the production of active matrix metalloproteinase 9 (MMP9). Once in tissues, several chemokines promote B cell survival, including CXCL12, B cell-activating factor (BAFF; also known as TNFSF13B) and a proliferation-inducing ligand (APRIL; also known as TNFSF13). In addition, CLL cell survival can be promoted through cognate interactions between CD31 and CD38, and the production by stromal cells of WNT factors, which can interact with ROR1, ROR2 and/or various Frizzled receptors. CLL cell contact with mesenchymal stromal cells can also be established through vascular cell adhesion protein 1 (VCAM1)-- α4β 1 integrin interactions that contribute to CLL cell survival. In turn, CLL cells can secrete chemokines, such as CCL3 and CCL4, which can recruit T cells and NLC-precursor cells (monocytes) to the CLL microenvironment. Activated T cells can provide CLL cells with proliferative signals through CD40 ligand (CD40L)-CD40 interactions and the secretion of several cytokines, such as IL-2, IL-4 and IL-10. In turn, activated CLL cells secrete CCL12 and CCL22, which attract more T cells into the CLL microenvironment. In tissues, CLL cells can be exposed to environmental and/or self-antigens that might trigger B cell activation through interactions with the surface immunoglobulin; this could amplify the responsiveness of CLL cells to the signals and factors that are provided by the CLL microenvironment. BAFFR, BAFF receptor (also known as TNFRSF13C); BCMA, B cell maturation protein (also known as TNFRSF17); BCR, B cell receptor; TACI, transmembrane activator and CAML interactor (also known as TNFRSF13B). \emph{Figure from \protect\hyperlink{ref-Kipps2017}{Kipps et al.} (\protect\hyperlink{ref-Kipps2017}{2017}) .}''
\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{figures/microenivironmental_signalling} 

}

\caption{(ref:microenvironmentSignalling)}\label{fig:microenvironmentSignalling}
\end{figure}
\hypertarget{therapies-in-cll}{%
\section{Therapies in CLL}\label{therapies-in-cll}}

\hypertarget{clinical-regimes}{%
\subsection{Clinical regimes}\label{clinical-regimes}}

(\protect\hyperlink{ref-Kipps2017}{Kipps et al. 2017})
The treatment of patients with CLL can include chemotherapy, a combination of chemotherapy and immunotherapy, or drugs that target the signalling pathways that promote the growth and/or survival of CLL cells (for example, BCR signalling and BCL-2)128,129.

Chemotherapy---Chemotherapy has been the mainstay of therapy for the past 50 years. Purine analogues (most commonly fludarabine, but also pentostatin or cladribine) and alkylating agents (including chlorambucil, cyclophosphamide or bendamustine) are used in the treatment of CLL130--132. Chemotherapy-based regimens can cause myelosuppression, an increased risk of infections and, in a small subset of patients, post-therapy myelodysplasia or secondary cancers, such as acute myeloid leukaemia (see Secondary cancers).

Chemoimmunotherapy---Phase III clinical trials have validated the benefit of anti-CD20 monoclonal antibodies, such as rituximab, obinutuzumab or ofatumumab, in combination with chemotherapy for the treatment of patients with CLL. In one trial (the CLL8 trial of the German CLL Study Group), patients who received fludarabine and cyclophosphamide with rituximab had higher response rates and a longer median progression-free survival (PFS) than patients who were treated with fludarabine and cyclophosphamide133. In a separate study (the CLL11 trial), patients \textgreater65 years of age with medical comorbidities who were treated with chlorambucil and either obinutuzumab or rituximab had improved response rates and longer median PFS than patients who were treated with chlorambucil alone134. However, the median PFS was significantly longer for patients who received obinutuzumab (26.7 months) than in those who received rituximab (11.1 months). In a third phase III trial, median PFS significantly improved from 13.1 months for patients treated with chlorambucil to only 22.4 months for patients treated with chlorambucil and ofatumumab135. As a consequence of these three trials, the US FDA approved the use of rituximab, obinutuzumab or ofatumumab in combination with chemotherapy for the first-line treatment of patients with CLL. The FDA also approved the use of ofatumumab as a single agent for the treatment of patients with relapsed or refractory disease based on data from a phase II study136.

\hypertarget{new-targeted-therapies}{%
\subsection{New Targeted therapies}\label{new-targeted-therapies}}

in particular Ibrutinib and idelalisib

(\protect\hyperlink{ref-Kipps2017}{Kipps et al. 2017})
Three main classes of drugs that each can inhibit BCR signalling have been evaluated in patients with CLL: BTK inhibitors, PI3K inhibitors and spleen tyrosine kinase (SYK) inhibitors86,143. CLL cells with unmutated IGHV seem to be more sensitive to inhibitors of BCR signalling than CLL cells with mutated IGHV144, but whether inhibitors, such as ibrutinib, are more effective in patients with CLL and unmutated IGHV, remains to be validated in clinical trials.
Ibrutinib has been approved in the United States and Europe for use as initial therapy, as well as in patients with relapsed disease, which followed results from a randomized trial that showed a significantly higher response rate to therapy with ibrutinib than with ofatumumab145. In addition, with continuous therapy, patients treated with ibrutinib had a significantly longer median PFS and overall survival than patients treated for 8 months with ofatumumab. Approval of ibrutinib as initial therapy was based on the results of a randomized trial that showed a significant improvement in median PFS and overall survival in patients 65 years of age without del(17p) who were treated indefinitely with ibrutinib than in patients treated for up to 48 weeks with chlorambucil146.
Upon initiation of treatment with ibrutinib, lymphadenopathy is rapidly reduced, which is associated with a concomitant increase in absolute lymphocyte count147. The rise in absolute lymphocyte count is related to the inhibition of chemokine receptor signalling, which inhibits the migration of CLL cells from the blood into the lymphoid tissues. This resulting lymphocytosis should not be considered a sign of progression; over time, the lymphocytosis subsides as the overall tumour burden decreases with continued therapy.

(\protect\hyperlink{ref-Kipps2017}{Kipps et al. 2017})
PI3K inhibitors include idelalisib, duvelisib (also known as IPI-145), TGR-1022 and ACP-319 (also known as AMG-319)151; the latter three drugs are being evaluated in clinical trials, whereas idelalisib was approved in the United States and Europe for the treatment of patients with relapsed CLL

(\protect\hyperlink{ref-Kipps2017}{Kipps et al. 2017})
Phase I/II clinical trials of fostamatinib, an oral SYK inhibitor, caused reduction in lymphadenopathy with concomitant lymphocytosis, an improvement in disease-related cytopenias and relief of disease-related symptoms in most of the treated patients with CLL156. However, dose-limiting toxicities of fostamatinib treatment include neutropenia,thrombocytopenia and diarrhoea. Other inhibitors of SYK, such as entospletinib, are being evaluated in preclinical and clinical studies.

(\protect\hyperlink{ref-Kipps2017}{Kipps et al. 2017})
BCL-2 inhibitors---Venetoclax is a small molecule that functions as a BH3 mimetic to inhibit BCL-2 (REF. 157). This drug is highly potent in inducing apoptosis in CLL cells, possibly by diminishing the capacity of BCL-2 to sequester the pro-apoptotic molecule BCL-2-interacting mediator of cell death (BIM; also known as BCL2L11)158. As such, venetoclax is effective in patients with relapsed and/or refractory disease159 or in patients with relapsed disease and del(17p)160. Indeed, the overall response rate for patients with relapsed disease and del(17p) was 79\%, with 8\% achieving a complete response. In addition, the estimated 12-month PFS was 72\% and overall survival was 87\%. On the basis of these results, the FDA approved the use of venetoclax for patients with relapsed disease and del(17p). Ongoing studies have shown that venetoclax can be safely combined with rituximab or obinutuzumab. Moreover, studies are examining the use of venetoclax with or without an anti-CD20 monoclonal antibody, and with or without ibrutinib161,162, which might provide higher response rates to therapy than that with venetoclax alone.

\hypertarget{drug-resistances-in-cll}{%
\subsection{Drug resistances in CLL}\label{drug-resistances-in-cll}}

incl ME -mediated and ibrutinib - IL4

\hypertarget{bet-inhibitors-and-their-potential-for-the-treatment-of-leukemia}{%
\subsection{BET inhibitors, and their potential for the treatment of leukemia}\label{bet-inhibitors-and-their-potential-for-the-treatment-of-leukemia}}

\hypertarget{background-to-the-approache-used-in-this-thesis}{%
\section{Background to the approache used in this thesis}\label{background-to-the-approache-used-in-this-thesis}}

\hypertarget{ex-vivo-drug-sensitivity-screens}{%
\subsection{Ex-vivo drug sensitivity screens}\label{ex-vivo-drug-sensitivity-screens}}

\hypertarget{studies-of-the-microenivornment-and-attempts-to-replicate-the-impact-of-the-microenvironment}{%
\subsection{Studies of the microenivornment, and attempts to replicate the impact of the microenvironment}\label{studies-of-the-microenivornment-and-attempts-to-replicate-the-impact-of-the-microenvironment}}

\hypertarget{intro-ATACseq}{%
\subsection{ATAC sequncing as a proxy for TF activity justification}\label{intro-ATACseq}}

\hypertarget{measurement-of-outcomes}{%
\subsection{Measurement of outcomes?}\label{measurement-of-outcomes}}

LDT, TTT, TFT, OS

(\protect\hyperlink{ref-Kipps2017}{Kipps et al. 2017})
Currently, the most reliable prognostic models are those developed for treatment-free survival, as evolving treatments have yet to change the indications for therapy. Predictive models to define overall survival with a given type of therapy are challenged by the chronicity of CLL and the fact that patients often receive serial treatments, each of which can affect outcome; moreover, death might be due to an indirect or unrelated cause. Furthermore, treatment options are changing, with newly identified, highly effective agents that are clearly prolonging survival and have activity among patients who would have been considered high risk when the only option was conventional chemotherapy.

(\protect\hyperlink{ref-Kipps2017}{Kipps et al. 2017})
Historically, a favourable response to therapy has been defined as a partial remission or complete remission. Partial remission requires a 50\% reduction in tumour bulk (for example, lymphadenopathy and splenomegaly), a 50\% reduction in lymphocytosis, and platelet counts of \textgreater100,000 cells per µl (or 50\% improvement over baseline) or a haemoglobin level of \textgreater11 g per dl (or 50\% improvement over baseline) without requiring transfusions or exogenous growth factors107. Complete remission requires the normalization of blood counts, resolution in lymphadenopathy and splenomegaly, and normal marrow function. The use of CT to assess response in CLL is becoming more common, particularly in clinical trials. However, the benefit of using repeated CT scans to monitor disease is uncertain, and seems unlikely to change patient outcome. Because of the distinct pattern of response observed with BCR inhibitors, a new response category, namely, partial response with lymphocytosis, has been described. Partial response with lymphocytosis is defined as a \textgreater50\% reduction in lymphadenopathy and splenomegaly, with persistent lymphocytosis; often the blood lymphocyte counts are equal to or greater than those observed prior to therapy.

In clinical trials, it is becoming more common to evaluate for MRD with 0.01\% of CLL cells among the total population of mononuclear cells in the blood or marrow. MRD can be measured by flow cytometry or PCR with next-generation sequencing of the clonal immunoglobulin variable region gene rearrangements163. In most clinical trials for patients with CLL, particularly those conducted in Europe, evaluation of MRD has been performed by flow cytometry of mononuclear cells from the marrow aspirate (the preferred method) or from the peripheral blood. In the 6 months following anti-CD20 monoclonal antibody treatment, the assessment of MRD is more sensitive on the mononuclear cells of the marrow aspirate than on cells that are isolated from the blood, which will often lack detectable CLL even when they are readily found in the marrow. Beyond a complete response, the best predictor of long-term PFS and overall survival is the achievement of a complete remission without evidence for MRD.

\hypertarget{multiomics-datasets-to-study-cll}{%
\subsection{Multiomics datasets to study CLL}\label{multiomics-datasets-to-study-cll}}

The power of multi-omics approaches especially the use of ATAC sequencing and combining ATAC and RNA

\hypertarget{intro-multivariate-modelling}{%
\subsection{Mathematical modelling}\label{intro-multivariate-modelling}}

Mathematical background in particular, lasso-regularised regression, and the caveats associated with feature selection
wrt cluster assignment

use method, explain how it works
we need mor ltieracy about these things
discuovery of mulitple layers of biology rather thatn jsut using t tests

how this can be used to build classfiers

\hypertarget{cll-as-a-model-for-studying-tumour-biology}{%
\subsection{CLL as a model for studying tumour biology}\label{cll-as-a-model-for-studying-tumour-biology}}

\hypertarget{aims-of-this-thesis}{%
\section{Aims of this thesis}\label{aims-of-this-thesis}}

\hypertarget{overview-of-the-approach}{%
\section{Overview of the approach}\label{overview-of-the-approach}}

Including diagrams

\hypertarget{note-on-contributions}{%
\section{Note on contributions}\label{note-on-contributions}}

Many of the figures, adn their cpations are taken directly from the amunscript. Where this si the case, it is indicated in the figure caption.

\hypertarget{methods}{%
\chapter{Methods}\label{methods}}

Methods in quotation marks are taken from Bruch \& Giles et al.~2021 and I have authored the original text, unless stated otherwise. Paragraphs without quotation marks were rewritten for the thesis.

Routine:
all headings should be here
at the end of each chapter, make sure all methods are included
Copy things in and put in quotation marks if from Bruch \& Giles et al.~2021, mark if they were not written by me, or if they were written jointly
Add REFERENCE and TABLE for now
Return to this later and neaten / expand on methods, and provide all referencing and tables
-fine to use quotations (although check plaigerism email), suggested rewriting as i / he / pmb throughout

\hypertarget{experimental-methods}{%
\section{Experimental methods:}\label{experimental-methods}}

\hypertarget{drug---stimulus-combinatorial-pertubation-assay}{%
\subsection{Drug - Stimulus Combinatorial Pertubation Assay}\label{drug---stimulus-combinatorial-pertubation-assay}}

\hypertarget{patient-sample-preparation}{%
\subsubsection{Patient Sample Preparation}\label{patient-sample-preparation}}

\emph{with Peter-Martin Bruch} Original
Peripheral Blood was taken from 192 patients for the initial drug-stimulation assay. Blood was separated by Ficoll gradient (GE Healthcare) and mononuclear cells were cryopreserved.

\hypertarget{preparation-of-screening-plates}{%
\subsection{Preparation of screening plates}\label{preparation-of-screening-plates}}

\emph{with Peter-Martin Bruch} Original
``Sample preparation, cell-culture, drug-stimulation profiling and genomic annotation was performed on 192 CLL patient samples as previously described{[}REFERENCE{]} with the following adjustments. Stimuli and drugs were mixed and preplated in the culture plates directly before adding the cell suspensions. RPMI-1640 and supplements were acquired from Gibco by Life Technologies, human serum was acquired from PAN Biotech (Cat.No. P40-2701, Lot.No:P-020317). Luminescence was measured after 48h on a Perkin Elmer EnVision.''

\emph{with Peter-Martin Bruch} Original
Compounds were obtained from Selleckchem, MedChemExpress and Sigma-Aldrich, dissolved in DMSO and stored at -\(20^\circ\)C. 12 drugs were used in two concentrations {[}TABLE{]}. Final DMSO concentration did not exceed 0.3\% in all experiments. Carfilzomib, Panobinostat and Venetoclax were removed from the analysis as they showed inconsistent toxicity depending on used media.
Insert table

Recombinant cytokines and stimulatory agents were dissolved according to manufacturer's protocol. 17 stimuli were selected {[}TABLE{]}. HS-5 conditioned medium was produced by incubating HS-5 stroma cell line (gifted by Martina Seiffert, DKFZ, Heidelberg) for 4 days at 37\(^\circ\)C and 5\% CO2, after which the supernatant was centrifuged and stored at -20\(^\circ\)C, the final concentration of HS-5 CM was 20\%. Bead immobilised anti-IgM was removed from the analysis due to storage instability. ''

Paper:
``Compounds and stimulatory agents were dissolved, stored and diluted according to manufacturer's protocol. HS-5 conditioned medium was produced by incubating HS-5 stromal cell line to \textgreater80\% confluency and cell removal by centrifugation. For a detailed list of stimuli and drugs and associated concentrations see Supp. Tables 1 and 2. Final DMSO concentration did not exceed 0.3\%.''

Original
Compounds were preplated in 384-well polypropylene storage plates (Greiner Bio-One Cat.No.:781271), which were stored at -20\(^\circ\)C. For each batch of samples tested on the same day a new storage plate was thawed and diluted in serum free RPMI, with or without stimuli. 5\(\mu\)L of drug-stimulation dilution were added into each well of the assay plates, 20 \(\mu\)L of cell suspension were added. The final cell concentration was 8*10\^{}5 cells/ml. Cells were thawed as previously described{[}REFERENCE{]}.

\hypertarget{drug-stimulation-assay}{%
\section{Drug-Stimulation assay}\label{drug-stimulation-assay}}

\emph{with Peter-Martin Bruch} Original
Drug-stimulation assays were performed with RPMI-1640 (Gibco by Life Technologies) supplemented with Penicillin Streptomycin (Gibco) final concentration of 100 Units/ml and 100 \(\mu\)g/ml respectively, L-Glutamine (Gibco) final concentration 2mM, and 10\% pooled, heat-inactivated and sterile filtered human type AB male off the clot serum (PAN Biotech, Cat.No. P40-2701, Lot.No:P-020317).

Drugs and stimuli were added to the cells simultaneously. Culture was performed in 384-well plates (Greiner Bio-One Cat.No.: 781904) and cells were incubated at 37\(^\circ\)C and 5\% CO2 for 48h.

Cell Viability was determined using the ATP-based CellTiter-Glo assay (Promega, Cat.No.:G7573). Luminescence was measured for the drug-stimulation assays using a Perkin Elmer EnVision and for the drug-drug interaction experiments using a Perkin Elmer EnSight, with a measurement time of 100ms per well.

\hypertarget{wes-rna-sequencing-targeted-sequencing-and-dna-copy-number-variants}{%
\section{WES, RNA Sequencing, Targeted Sequencing and DNA Copy number variants}\label{wes-rna-sequencing-targeted-sequencing-and-dna-copy-number-variants}}

\emph{Myself} Original
Sequencing data generated by Dietrich et al{[}REFERENCE{]}, and all sequencing and data processing were performed as described there. For

\hypertarget{follow---up-investigations}{%
\subsection{Follow - up investigations}\label{follow---up-investigations}}

\hypertarget{lymphocyte-doubling-time}{%
\subsubsection{Lymphocyte doubling time}\label{lymphocyte-doubling-time}}

\hypertarget{survival-data}{%
\subsubsection{Survival data}\label{survival-data}}

\hypertarget{atacsequencing-of-4-cll-samples}{%
\subsubsection{ATACsequencing of 4 CLL samples}\label{atacsequencing-of-4-cll-samples}}

\emph{with Peter-Martin Bruch (mostly edited by me)} Paper
``Peripheral blood was taken from 4 CLL patients and separated by Ficoll gradient (GE Healthcare), mononuclear cells were cryopreserved on liquid nitrogen. Samples were later thawed from frozen as previously described{[}REFERENCE{]} and MACS sorted for CD19 positive cells (Milteny autoMACS). The cells were resuspended in RPMI (GIBCO, Cat.No. 21875-034), with the addition of 2mM glutamine (GIBCO, Cat.No. 25030-24), 1\% Pen/Strep (GIBCO, Cat.No. 15140-122) and 10\% pooled, heat-inactivated and sterile filtered human type AB male off the clot serum (PAN Biotech, Cat.No. P40-2701, Lot.No:P-020317). 5ml of cell suspension was cultured in 6-well plates (Greiner Bio-One Cat.No. 657160). After thawing, cells were incubated at 37\(^\circ\)C and 5\% CO2 for 6 hours in 0.2\% DMSO. The final cell concentration was 2x10ˆ6 cells/ml. Cell viability and purity was assessed using FACS. All samples had a viability over 90\% and over 95\% of CD19+/CD5+/CD3- cells.''

\emph{with Nayara} Paper
``ATAC-seq libraries were generated as described previously{[}REFERENCE{]}. Cell preparation and transposition was performed according to the protocol, starting with 5x10ˆ4 cells per sample. Purified DNA was stored at −20\(^\circ\)C until library preparation was performed. To generate multiplexed libraries, the transposed DNA was initially amplified for 5x PCR cycles using 2.5 \(\mu\)L each of 25 μM PCR Primer 1 and 2.5 \(\mu\)L of 25 \(\mu\)M Barcoded PCR Primer 2 (included in the Nextera index kit, Illumina, San Diego, CA, USA), 25 \(\mu\)L of NEBNext High-Fidelity 2x PCR Master Mix (New England Biolabs, Boston, Massachusetts) in a total volume of 50 \(\mu\)L. 5 \(\mu\)L of the amplified DNA was used to determine the appropriate number of additional PCR cycles using qPCR. Additional number of cycles was calculated through the plotting of the linear Rn versus cycle, and corresponds to one-third of the maximum fluorescent intensity. Finally, amplification was performed on the remaining 45 \(\mu\)L of the PCR reaction using the optimal number of cycles determined for each library by qPCR (max. 13 cycles in total). The amplified fragments were purified with two rounds of SPRI bead clean-up (1.4x). The size distribution of the libraries was assessed on Bioanalyzer with a DNA High Sensitivity kit (Agilent Technologies, Santa Clara, CA), concentration was measured with Qubit DNA High Sensitivity kit in Qubit 2.0 Flurometer (Life Technologies, Carlsbad, CA). Sequencing was performed on NextSeq 500 (Illumina, San Diego, CA, USA) using 75bp paired-end sequencing, generating ∼450 million paired-reads per run, with an average of 55 million reads per sample.''

\hypertarget{spi-b-and-pu.1-shrna-knockdowns}{%
\subsubsection{Spi-B and PU.1 shRNA Knockdowns}\label{spi-b-and-pu.1-shrna-knockdowns}}

\hypertarget{immunohistochemistry-of-patient-lymph-nodes}{%
\subsubsection{Immunohistochemistry of patient lymph nodes}\label{immunohistochemistry-of-patient-lymph-nodes}}

\emph{Peter-Martin Bruch} Paper
``LN biopsies of CLL-infiltrated and non-neoplastic lymph nodes were Paraffin fixed and arranged in Tissue Microarrays. Consecutively they were stained for PAX-5 (790-4420, Roche), CD3 (790-4341, Roche), pSTAT6 (ab28829, abcam), STAT6 (519-4290, Zytomed Systems) and pIRAK4 (ab216513, abcam). The slides were analysed using Qupath 50 and the recommended protocol. Cell based data on mean staining intensity was exported and further analysed using R.''

\hypertarget{ibrutinib---il4---stat6i-interaction-assay}{%
\subsubsection{Ibrutinib - IL4 - STAT6i interaction assay}\label{ibrutinib---il4---stat6i-interaction-assay}}

\hypertarget{investigation-of-ibrutinib-il4-ibet-762}{%
\subsection{Investigation of Ibrutinib + IL4 + IBET-762}\label{investigation-of-ibrutinib-il4-ibet-762}}

\hypertarget{ibrutinib---il4---ibet-762-interaction-assay}{%
\subsubsection{Ibrutinib - IL4 - IBET-762 interaction assay}\label{ibrutinib---il4---ibet-762-interaction-assay}}

\hypertarget{ibrutinib---il4---ibet--762-treatment}{%
\subsubsection{Ibrutinib - IL4 - IBET -762 treatment}\label{ibrutinib---il4---ibet--762-treatment}}

\hypertarget{atacsequencing}{%
\subsubsection{ATACsequencing}\label{atacsequencing}}

\emph{with Peter Martin Bruch} get from diffTF
Peripheral blood was taken from 4 CLL patients and separated by Ficoll gradient (GE Healthcare), mononuclear cells were cryopreserved on liquid nitrogen. Samples were later thawed from frozen as previously described{[}REFERENCE{]} and MACS sorted for CD19 positive cells (Milteny autoMACS). The cells were resuspended in RPMI (GIBCO, Cat.No. 21875-034), with the addition of 2mM glutamine (GIBCO, Cat.No. 25030-24), 1\% Pen/Strep (GIBCO, Cat.No. 15140-122) and 10\% pooled, heat-inactivated and sterile filtered human type AB male off the clot serum (PAN Biotech, Cat.No. P40-2701, Lot.No:P-020317). 5ml of cell suspension was cultured in 6-well plates (Greiner Bio-One Cat.No. 657160). After thawing, cells were incubated at 37\(^\circ\)C and 5\% CO2 for 6 hours in 0.2\% DMSO. The final cell concentration was 2x10ˆ6 cells/ml. Cell viability and purity was assessed using FACS. All samples had a viability over 90\% and over 95\% of CD19+/CD5+/CD3- cells.

\emph{with Nayara} get from diffTF
ATAC-seq libraries were generated as described previously{[}REFERENCE{]}. Cell preparation and transposition was performed according to the protocol, starting with 5x10ˆ4 cells per sample. Purified DNA was stored at −20\(^\circ\)C until library preparation was performed. To generate multiplexed libraries, the transposed DNA was initially amplified for 5x PCR cycles using 2.5 \(\mu\)L each of 25 \(\mu\)M PCR Primer 1 and 2.5 \(\mu\)L of 25 \(\mu\)M Barcoded PCR Primer 2 (included in the Nextera index kit, Illumina, San Diego, CA, USA), 25 \(\mu\)L of NEBNext High-Fidelity 2x PCR Master Mix (New England Biolabs, Boston, Massachusetts) in a total volume of 50\(\mu\)L. 5 \(\mu\) L of the amplified DNA was used to determine the appropriate number of additional PCR cycles using qPCR. Additional number of cycles was calculated through the plotting of the linear Rn versus cycle, and corresponds to one-third of the maximum fluorescent intensity. Finally, amplification was performed on the remaining 45 \(\mu\)L of the PCR reaction using the optimal number of cycles determined for each library by qPCR (max. 13 cycles in total). The amplified fragments were purified with two rounds of SPRI bead clean-up (1.4x). The size distribution of the libraries was assessed on Bioanalyzer with a DNA High Sensitivity kit (Agilent Technologies, Santa Clara, CA), concentration was measured with Qubit DNA High Sensitivity kit in Qubit 2.0 Flurometer (Life Technologies, Carlsbad, CA). Sequencing was performed on NextSeq 500 (Illumina, San Diego, CA, USA) using 75bp paired-end sequencing, generating ∼450 million paired-reads per run, with an average of 55 million reads per sample.

\hypertarget{rnasequencing}{%
\subsubsection{RNAsequencing}\label{rnasequencing}}

\hypertarget{proteomics}{%
\subsubsection{Proteomics}\label{proteomics}}

\hypertarget{data-availability}{%
\section{Data availability}\label{data-availability}}

European Genome-Phenome Archive (EGA) accession \ldots. The data for \ldots. and computational analysis code used in this study are available from \ldots.

Additionally, we obtained CLL ATACseq data{[}REFERENCE{]} from the European Genome-phenome Archive (EGA: EGAD00001002110) and from (Bruch \& Giles et al.~2021)
For the ChIPseq analysis of SPIB binding sites, we accessed the data 37 from the NCBI GEO database 51, accession GEO: GSE56857. We made use of the data for SPIB in the OCILY3 DLBCL cell line (GSM1370276).

\hypertarget{statistical-analysis}{%
\section{Statistical Analysis}\label{statistical-analysis}}

Mixutre of paper and original
Integrative data analysis of screening data, DNA and RNA sequencing, CNV and methylation profiles and follow up experiments was performed using R version 4 (REFERENCE R Core Team, 2018) with the RStudio interface (REFERENCE RStudio Team, 2016) and using packages that included (\protect\hyperlink{ref-R-DESeq2}{Love, Anders, and Huber 2021}), (\protect\hyperlink{ref-R-survival}{Therneau 2021}), (\protect\hyperlink{ref-R-glmnet}{Friedman et al. 2021}), (\protect\hyperlink{ref-R-ConsensusClusterPlus}{M. Wilkerson and Waltman 2021}), (\protect\hyperlink{ref-R-clusterProfiler}{Yu 2021b}), (\protect\hyperlink{ref-R-ChIPseeker}{Yu 2021a}), (\protect\hyperlink{ref-R-genomation}{Akalin et al. 2021}) and (\protect\hyperlink{ref-R-BloodCancerMultiOmics2017}{Oles et al. 2021}) to perform univariate association tests, multivariate regression with and without lasso penalization, Cox regression, generalised linear modelling and clustering.

\hypertarget{data-processing}{%
\subsection{Data processing}\label{data-processing}}

\hypertarget{processing-of-screening-data}{%
\subsubsection{Processing of screening data}\label{processing-of-screening-data}}

\emph{with Peter-Martin Bruch} Paper
To quantify the response of a patient sample to each treatment condition, we used viability relative to the control, i.e., the CellTiter Glo luminescence readout of the respective well divided by the median of luminescence readouts of the DMSO control wells on the same plate.

\hypertarget{quality-assessment-and-control}{%
\subsubsection{Quality Assessment and Control}\label{quality-assessment-and-control}}

\hypertarget{rna-data}{%
\subsubsection{RNA data}\label{rna-data}}

RNA data was taken from the PACE repository (\protect\hyperlink{ref-PACE}{\textbf{PACE?}}). For detailed explanaition of processing see (\protect\hyperlink{ref-PACE}{\textbf{PACE?}}).

\hypertarget{atacsequencing-1}{%
\subsubsection{ATACsequencing}\label{atacsequencing-1}}

\hypertarget{data-from-pace}{%
\subsubsection{Data from PACE}\label{data-from-pace}}

\hypertarget{analysis-of-screening-data}{%
\subsection{Analysis of Screening Data}\label{analysis-of-screening-data}}

\hypertarget{drug-drug-and-stimulus---stimulus-correlations}{%
\subsubsection{Drug-drug and stimulus - stimulus correlations}\label{drug-drug-and-stimulus---stimulus-correlations}}

\emph{myself} original complete
Pearson correlation coefficients were calculated for each drug - drug and stimulus - stimulus pair, using the \texttt{cor} functions of the (\protect\hyperlink{ref-R-stats}{\textbf{R-stats?}}) package with log transformed viability values which were normalised to untreated controls.

\hypertarget{stimulus-heatmap-method}{%
\subsubsection{Characterising stimulus responses across all samples}\label{stimulus-heatmap-method}}

\emph{myself} original complete (just check)
For the heatmap in figure \ref{fig:stimuliHeatmap}, the viability data is represented as z scores. Log transformed viability values, normalised to DMSO controls, are row-scaled according to the Median Absolute Deviance. Limits were then applied to this row scaling factor for optimal visualisation. The matrix of z scores was plotted using the (\protect\hyperlink{ref-R-pheatmap}{Kolde 2019}) package. The ordering of the columns (patient samples) was obtained from the dendrogram that resulted from running \texttt{ConsensusClusterPlus}, from the (\protect\hyperlink{ref-R-ConsensusClusterPlus}{M. Wilkerson and Waltman 2021}) package. The rows were ordered using the dendrogram order produced by \texttt{hclust} with default branch arrangement.

\hypertarget{univariate-gene-stimulus-associations-method}{%
\subsubsection{Univariate analysis of gene - stimulus and gene - drug response assosciations}\label{univariate-gene-stimulus-associations-method}}

\emph{myself} writtten for thesis complete
Section \ref{univariate-gene-stimulus-associations} \& \ref{univariate-gene-drug-associations}. Two-sided Student's t-tests, with equal variance were performed for IGHV status and somatic mutations and copy number aberrations with ≥3 patient samples in each group (n = 63/54). A 10\% FDR cut off was used to determine significance.

\hypertarget{multivariate-gene-stimulus-assosciations-method}{%
\subsubsection{Penalised multivariate regression of gene - stimulus and gene - drug assosciations}\label{multivariate-gene-stimulus-assosciations-method}}

\emph{myself} written for thesis (but some overlap with JCI)
Section \ref{multivariate-gene-stimulus-assosciations} \& \ref{multivariate-gene-drug-assosciations}.\\
ADAPT FOR DRUGS ALSO A Gaussian linear model with L1-penalty implemented in the \texttt{R} package \texttt{glmnet} (\protect\hyperlink{ref-R-glmnet}{Friedman et al. 2021}), with mixing parameter alpha = 1,was used to identify gene-stimulus associations. Matrix of genetic features (p=39), IGHV status (encoded as M = 1 and U = 0), and Methylation Cluster (encoded as 0, 0.5, 1) were used to identify multivariate predictors. All features were thus encoded on a similar scale to ensure equal treatment by lasso constraint in model fitting. Genetic features with more than 20\% missing values were excluded from the analysis, and samples without complete annotation for remaining features were removed, resulting in n = 129 samples. The matrix of control-normalised log-transformed viability values for these 129 samples was provided as the dependent variable. Using 3-fold cross-validation, the optimal penalty parameter \(\lambda\) was selected so as to minimise the cross-validated R2. The reduction in cross-validated mean squared error compared to the null model was used as loss. The model was fitted for 30 bootstrapped repeats, and the resulting coefficients are the mean of those coefficients that were selected in \textgreater75\% model fits.

\hypertarget{correlation-of-rna-receptor-expression-with-viability}{%
\subsubsection{Correlation of RNA-Receptor Expression with viability}\label{correlation-of-rna-receptor-expression-with-viability}}

\emph{myself} original complete
RNA count data for matched samples was available for 49 patients and was transformed using the variance stabilising transformation. Stimulus - receptor pairs were defined using the available literature, see table \ref{tab:receptorPairs}. For each stimulus, pearson correlation coefficient were calculated between the control-normalised log-transformed viability value of a sample and its expression of the corresponding stimulus receptor.

\hypertarget{linear-modelling-of-drug---stimulus-interactions}{%
\subsubsection{Linear modelling of drug - stimulus interactions}\label{linear-modelling-of-drug---stimulus-interactions}}

\hypertarget{drug-stimulus-gene-interactions-method}{%
\subsubsection{Modelling of drug - stimulus - gene interactions}\label{drug-stimulus-gene-interactions-method}}

\emph{myself} original
Section \ref{drug-stimulus-gene-interactions}. To identify drug - stimulus interactions that were dependent on genetic features, we first fitted linear models to the viability matrix, for each drug - stimulus combination, using the formula in Eqn2. We extracted drug - stimulus interaction coefficients for each patient to generate a response matrix for n = 137 patients. Next, we performed multivariate regression using a Gaussian linear model with L1-penalty (i.e., lasso regression) as implemented in the R package \texttt{glmnet}(\protect\hyperlink{ref-R-glmnet}{Friedman et al. 2021}). As the dependent variable, the matrix of drug - stimulus interaction coefficients for each patient sample was used. As input to the model, genetic features with more than 20\% missing values were excluded, and only patients with complete annotation were included in the model ( n= 137 ). As predictors, the genetic mutations and CNVs (p= 39), IGHV status (coded as 0-1) and Methylation CLuster (coded as 0, 0.5, 1) were used, using 3-fold cross-validation. Misclassification error was used as loss for cross- validation.

\hypertarget{follow-up}{%
\subsection{Follow up}\label{follow-up}}

\hypertarget{lympocyte-doubling-times}{%
\subsubsection{Lympocyte doubling times}\label{lympocyte-doubling-times}}

\emph{Herbst et al., 2020c} - update
``Patients which had lymphocyte counts available for less than 4 timepoints between the sample
collection date and the time of the next treatment and patients currently in treatment were ex-cluded. Thus, \ldots{} patients with enough data remained. Lymphocyte growth rates were calculated by fitting a linear model to the log10 transformed lymphocyte counts of all timepoints between the sample collection date and the time of the next treatment versus the period of time.''

\hypertarget{survival-analyses}{%
\subsubsection{Survival analyses}\label{survival-analyses}}

\emph{myself} original
Survival analyses were performed using TTT, TFT and OS as metrics. Follow-up information to calculate OS was available for all 192 CLL patients. For 188 of 192 CLL patients treatment information after sample collection was available.

\emph{Herbst et al}
''Time to next treatment (TTT) was calculated from the date of sample collection to subsequent
treatment initiation. Patients without treatment initiation during the observation time and
patients who died before treatment initiation were censored at the latest follow-up contact.''

\emph{myself} original
For for the purpose of visualising Kaplan-Meier plots , optimal cut-points of staining levels were calculated using maximally selected rank statistics as computed by the \texttt{R} package \texttt{maxstat} (\protect\hyperlink{ref-R-maxstat}{\textbf{R-maxstat?}}). Based on these cut points, patients were split into two groups, and their survival data were plotted using the Kaplan-Meier method, using the R package \texttt{survminer} (\protect\hyperlink{ref-R-survminer}{Kassambara, Kosinski, and Biecek 2021}).

For Cox proportional hazards regression, the \texttt{coxph} function of the \texttt{R} package \texttt{survival} (\protect\hyperlink{ref-R-survival}{Therneau 2021}) was used. Expand

\hypertarget{penalised-multivariate-regression-genetic-predictors-of-cluster-membership}{%
\subsubsection{Penalised multivariate regression genetic predictors of cluster membership}\label{penalised-multivariate-regression-genetic-predictors-of-cluster-membership}}

\emph{Myself} original (edit so not similar to JCI)
I used a multinomial linear model with L1-penalty, implemented in the \texttt{glmnet} (\protect\hyperlink{ref-R-glmnet}{Friedman et al. 2021}) package. As the dependent variable, the cluster assignment for each patient was used. As input to the model ,genetic features with more than 20\% missing values were excluded, and only patients with complete annotation were included in the model ( n= 137 ). As predictors, the genetic mutations and CNVs (p= 39) and IGHV status (coded as 0-1) were used, using 3-fold cross-validation. Misclassification error was used as loss for cross- validation. The resulting coefficients indicated associations between genetic features and each cluster.

\hypertarget{gene-expression-and-gene-set-enrichment-analysis-between-clusters}{%
\subsubsection{Gene expression and gene set enrichment analysis between clusters}\label{gene-expression-and-gene-set-enrichment-analysis-between-clusters}}

\emph{myself} original
For the n=49 patient samples for which viability data and RNA--Seq data for matching samples was available, the \texttt{R} package \texttt{DESeq2} (\protect\hyperlink{ref-R-DESeq2}{Love, Anders, and Huber 2021}) was used to search for associations of these two data types.RNA-Seq read count data was regressed on to the patient clusters C3, C4 (design formula \textasciitilde{} IGHV.status + Cluster). Genes were ranked by their test statistics and Gene Set Enrichment Analysis (GSEA) (implementing the fgsea algorithm with the \texttt{clusterProfiler} (\protect\hyperlink{ref-R-clusterProfiler}{Yu 2021b}) package) was applied to the ranked lists with the KEGG pathway gene set selections from the MSigDBdatabase{[}REFERENCE{]}.

\hypertarget{associations-of-ex-vivo-stimulus-responses-with-genomic-features}{%
\subsubsection{Associations of ex-vivo stimulus responses with genomic features}\label{associations-of-ex-vivo-stimulus-responses-with-genomic-features}}

We tested for associations between stimulus viability assay results and genomic features by Student's t-tests (two-sided, with equal variance). We tested somatic mutations (aggregated at the gene level), copy number aberrations and IGHV status. We restricted the analysis to features that were present in at least 3 patient samples (63 features). p-values were adjusted for multiple testing by applying the Benjamini-Hochberg procedure.

\hypertarget{gene-dosage-effects-method}{%
\subsubsection{Analysis of differential gene dosage in trisomy 12 CLL}\label{gene-dosage-effects-method}}

\emph{myself} written for thesis
Section \ref{gene-dosage-effects}. For all RNA samples available in PACE (i.e.~not just those that matched the samples in the screen), differential expression was called using the \texttt{DESeq2} package (\protect\hyperlink{ref-R-DESeq2}{Love, Anders, and Huber 2021}), with the design formula \textasciitilde trisomy12. Raw RNA counts were visualised if the gene had BH-adjusted p \textless{} 0.1 and belonged to TGF\(\beta\), JAK-STAT or TLR pathways genesets, as defined in the KEGG database (\protect\hyperlink{ref-Kegg}{\textbf{Kegg?}}) downloaded using the \texttt{msigdbr} package (\protect\hyperlink{ref-R-msigdbr}{Dolgalev 2021}). Proteomic abundance data was also plotted, provided by Sophie Herbst (\protect\hyperlink{ref-HerbstThesis}{Herbst 2020}). Proteomics data partially overlaps with RNAseq data. Protein quantification was performed as described in \protect\hyperlink{ref-HerbstThesis}{Herbst} (\protect\hyperlink{ref-HerbstThesis}{2020}).

\hypertarget{trisomy12-classifier-method}{%
\subsubsection{Generation of a Trisomy 12 classifier}\label{trisomy12-classifier-method}}

\emph{edit and finish}
Section \ref{trisomy12-classifier}. To build the classifier, a binomial linear model with L1-penalty implemented in the \texttt{R} package \texttt{glmnet} (\protect\hyperlink{ref-R-glmnet}{Friedman et al. 2021}), with mixing parameter alpha = 1, was used. To identify coefficients that predict trisomy 12 status, the viability matrix for all stimuli, as described in section \ref{stimulus-heatmap-method} was used, i.e.~z scores of the control-normalised log-transformed viability values.

Matrix of genetic features (p=39), IGHV status (encoded as M = 1 and U = 0), and Methylation Cluster (encoded as 0, 0.5, 1) were used to identify multivariate predictors. All features were thus encoded on a similar scale to ensure equal treatment by lasso constraint in model fitting. Genetic features with more than 20\% missing values were excluded from the analysis, and samples without complete annotation for remaining features were removed, resulting in n = 129 samples. The matrix of control-normalised log-transformed viability values for these 129 samples was provided as the dependent variable. Using 3-fold cross-validation, the optimal penalty parameter \(\lambda\) was selected so as to minimise the cross-validated R2. The reduction in cross-validated mean squared error compared to the null model was used as loss. The model was fitted for 30 bootstrapped repeats, and the resulting coefficients are the mean of those coefficients that were selected in \textgreater75\% model fits.

The classifier was built using binomial regression, with lasso penalisation, as implemented in the \texttt{R} package \texttt{glmnet} (\protect\hyperlink{ref-R-glmnet}{Friedman et al. 2021}). The feature matrix consisted of z scores of the viability values after treatment with each stimulus, and was used to predict the response (trisomy 12 status). I ran the model for 50 bootstrapped repeats, using three-fold cross-validation and mean absolute error as loss. Resiquimod, sCD40L+IL4 and TGF\(\beta\) were selected as coefficients that predict trisomy 12 status (Figure \ref{fig:tri12Classfier}, as would be expected bed on the observations in section \ref{trisomy12-modulator}.

\hypertarget{atacseq-processing}{%
\subsubsection{ATACseq processing}\label{atacseq-processing}}

\emph{with Ivan Berest} original (update to paper)
We downloaded CLL data (\protect\hyperlink{ref-Rendeiro2016}{Rendeiro et al. 2016}) from the European Genome-phenome Archive (EGA: EGAD00001002110). Original dataset had 88 ATAC-seq samples from 55 patients, however we used for the analysis only one sample per patient passing quality checks, resulting in 52 samples. We used an in-house constructed ATAC-seq processing pipeline (\protect\hyperlink{ref-Berest2019}{Berest et al. 2019}) to obtain final bam files mapped to the hg19 annotation genome that were corrected for CG bias.

\hypertarget{annotation-of-trisomy-12-status}{%
\subsubsection{Annotation of trisomy 12 status}\label{annotation-of-trisomy-12-status}}

\emph{with Ivan Berest} original (update to paper)
As trisomy 12 status was not included in the original metadata, we used a mean amount of reads in the chromatin accessible peaks for each sample to distinguish trisomy 12 patients. All samples containing 1.4 times more reads in the peaks located on chromosome 12, compared to the peaks on all other chromosomes, were classified as trisomy 12 patients.

\hypertarget{difftf-analysis-of-tf-activity-in-trisomy-12-cll}{%
\subsubsection{diffTF analysis of TF activity in trisomy 12 CLL}\label{difftf-analysis-of-tf-activity-in-trisomy-12-cll}}

\emph{with Ivan Berest} original (update to paper)
FOr the larger ATACseq dataset, we ran analytical mode of diffTF (\protect\hyperlink{ref-R-diffTF}{\textbf{R-diffTF?}}) pipeline to investigate differential TF activity between trisomy 12 and non-trisomy 12 patients using the HOCOMOCO v10 database (\protect\hyperlink{ref-HOCOMOCO}{Kulakovskiy et al. 2016}) with the following design formula: ``\textasciitilde{} Batch + Gender + IGHV status + trisomy12 status.''

for tirsomy 12 software uses the ATACseq counts at each of the binding sites across the genome to generate a distribution of fold changes between trisomy 12 and WT samples. This fold change distribution is compared to a background distribution of fold changes, calculated similarly using ATACseq counts for a GC-matched motif, that does not contain the TF binding motif. The TF is classified as differentially active between trisomy 12 and WT samples where these two distributions are significantly different. Each TF is assigned a weighted mean difference value, which quantifies the change in activity, and a p value.

For the smaller ATACseq dataset, ATAC-seq data generated from our CLL samples were processed similarly with the in-house ATAC-seq pipeline, as described previously (\protect\hyperlink{ref-Berest2019}{Berest et al. 2019}), with the only exception that we didn't use CG bias correction step. We used a similarly analytical mode of diffTF with HOCOMOCO v10 database (\protect\hyperlink{ref-HOCOMOCO}{Kulakovskiy et al. 2016}) using the following parameters: minOverlap = 1; design formula = ``\textasciitilde{} Patient + Trisomy 12 status.''

\hypertarget{functional-enrichment-analysis-of-spib-chipseq-data}{%
\subsubsection{Functional enrichment analysis of SPIB ChIPseq data}\label{functional-enrichment-analysis-of-spib-chipseq-data}}

\emph{myeslf} original (update to paper)
We downloaded SPIB ChIPseq data {[}REFERENCE{]} from the NCBI GEO database{[}REFERENCE{]}, accession \url{GEO:GSE56857}. We made use of the data for SPIB in the OCILY3 DLBCL cell line (GSM1370276). SPIB ChIP peaks were filtered for significance (q value\textless0.05). We used the annotatePeaks function from the package clusterProfiler{[}REFERENCE{]} to annotate the nearest gene for each ChIPpeak. We filtered peaks within ±1kb of a TSS of a gene and performed over-representation of KEGG and Reactome pathways (using the clusterProfiler package{[}REFERENCE{]}) amongst the resulting list of genes.

\hypertarget{drug-stimulus-linear-model-method}{%
\subsubsection{Linear modelling of drug - stimulus interactions}\label{drug-stimulus-linear-model-method}}

\emph{myself} original
Section \ref{drug-stimulus-linear-model} To identify drug-stimulus interactions, we fitted linear models to the viability matrix, for each drug - stimulus combination, using the formula in Eqn1. Drug - stimulus interaction coefficients and associated p values were extracted and used to define significant interactions.

Incl defining interaction types

\hypertarget{drug-drug-interaction-assay}{%
\subsubsection{Drug-Drug interaction assay}\label{drug-drug-interaction-assay}}

An independent patient cohort of 16 patients was used in the drug-drug interaction experiments.

\hypertarget{survival-analysis-of-immunohistory-chemistry-data}{%
\subsubsection{Survival analysis of immunohistory chemistry data}\label{survival-analysis-of-immunohistory-chemistry-data}}

\hypertarget{ibrutinib---ibet-762---il-4-analysis}{%
\subsection{Ibrutinib - IBET-762 - IL-4 analysis}\label{ibrutinib---ibet-762---il-4-analysis}}

\hypertarget{rnaseq-processing}{%
\subsubsection{RNAseq processing}\label{rnaseq-processing}}

\hypertarget{deseq2-analysis}{%
\subsubsection{Deseq2 analysis}\label{deseq2-analysis}}

\hypertarget{atacseq-processing-1}{%
\subsubsection{ATACseq processing}\label{atacseq-processing-1}}

\hypertarget{difftf-analysis}{%
\subsubsection{diffTF analysis}\label{difftf-analysis}}

\hypertarget{generation-of-gene-regulatory-network}{%
\subsubsection{Generation of Gene Regulatory Network}\label{generation-of-gene-regulatory-network}}

\hypertarget{proteomics-processing}{%
\subsubsection{Proteomics Processing}\label{proteomics-processing}}

\hypertarget{proteomics-analysis}{%
\subsubsection{Proteomics Analysis}\label{proteomics-analysis}}

\hypertarget{data}{%
\chapter{Data}\label{data}}

Characterisation of primary CLL samples by high-throughput combinatorial screening and multi-omic profiling. An introduction to the dataset that my PhD is based on.

\hypertarget{drug-screens-and-experiments}{%
\section{Drug screens and experiments}\label{drug-screens-and-experiments}}

\hypertarget{high-throughput-combinatorial-pertubation-assay}{%
\subsection{High-throughput combinatorial pertubation assay}\label{high-throughput-combinatorial-pertubation-assay}}

We measured the effects of 17 cytokines and microenvironmental stimuli on cell viability in 192 primary CLL samples and combined each one with 12 drugs to investigate the influence on spontaneous and drug-induced apoptosis (Figure \ref{fig:studyOverview}). Viability was assessed by ATP measurement via CellTiterGlo after 48 h of culture and normalised to untreated controls8.


\begin{figure}

{\centering \includegraphics[width=0.7px]{figures/studyOverview} 

}

\caption{Schematic of experimental protocol. By combining 12 drugs and 17 stimuli, we systematically queried the effects of simultaneous stimulation and inhibition of critical pathways in CLL (n=192). Integrating functional drug-stimulus response profiling with four additional omics layers, we identified pro-survival pathways, underlying molecular modulators of drug and microenvironment reponses, and drug-stimulus interactions in CLL.}\label{fig:studyOverview}
\end{figure}
\hypertarget{validation-experiments}{%
\subsection{Validation experiments}\label{validation-experiments}}

In addition to the screen, we acquired the following validatory data (Figure \ref{fig:additionalData}).

Proper explamantion of data srouces and how we acuried them, the format they are in


\begin{figure}

{\centering \includegraphics[width=0.7px]{figures/studyOverview} 

}

\caption{Schematic of validatory data.}\label{fig:additionalData}
\end{figure}
\hypertarget{characteristics-of-drugs-used-in-the-screen}{%
\section{Characteristics of drugs used in the screen}\label{characteristics-of-drugs-used-in-the-screen}}

\hypertarget{drug-pathways}{%
\subsection{Drug pathways}\label{drug-pathways}}

We screened 17 different drugs (Figure \ref{fig:drugCategories}).


\begin{figure}

{\centering \includegraphics[width=1px]{thesis_files/figure-latex/drugCategories-1} 

}

\caption{Bar plot of the drugs used in screen.}\label{fig:drugCategories}
\end{figure}
\hypertarget{drug-responses}{%
\subsection{Drug responses}\label{drug-responses}}

The drug responses were as follows (Figure \ref{fig:drugResponses}).


\begin{figure}

{\centering \includegraphics[width=0.5px]{thesis_files/figure-latex/drugResponses-1} 

}

\caption{Log transformed viability values for all drugs that were included in the screen after qulaity control. p values from student's t test.}\label{fig:drugResponses}
\end{figure}
\hypertarget{genetic-predictors-of-drug-responses}{%
\subsection{Genetic predictors of drug responses}\label{genetic-predictors-of-drug-responses}}

\hypertarget{drug---drug-correlations}{%
\subsection{Drug - Drug Correlations}\label{drug---drug-correlations}}

\hypertarget{characteristics-of-stimuli-used-in-the-screen}{%
\section{Characteristics of stimuli used in the screen}\label{characteristics-of-stimuli-used-in-the-screen}}

\hypertarget{stimulus-responses}{%
\subsection{Stimulus responses}\label{stimulus-responses}}

\hypertarget{stimulus---stimulus-correlations}{%
\subsection{Stimulus - Stimulus Correlations}\label{stimulus---stimulus-correlations}}

\hypertarget{characteristics-of-patient-samples-used-in-the-screen}{%
\section{Characteristics of patient samples used in the screen}\label{characteristics-of-patient-samples-used-in-the-screen}}

\hypertarget{genetic-data-available-for-each-patient}{%
\subsection{Genetic Data available for each patient}\label{genetic-data-available-for-each-patient}}

WES, CNVs, Methylation, Transcriptomic, ATACseq, LDT, survT, IHC
PLus Ibrutinib + IBET + IL4 treated samples - ATACseq, RNAseq, proteomics

\hypertarget{processing-of-raw-values-obtained-from-cell-viability-assay}{%
\section{Processing of raw values obtained from cell viability assay}\label{processing-of-raw-values-obtained-from-cell-viability-assay}}

\hypertarget{data-normalization-and-quality-control}{%
\subsection{Data normalization and quality control}\label{data-normalization-and-quality-control}}

\hypertarget{how-viabilty-is-calucalted-why-we-use-log-values-etc}{%
\subsection{How viabilty is calucalted, why we use log values etc}\label{how-viabilty-is-calucalted-why-we-use-log-values-etc}}

\hypertarget{heterogeneity-of-response-is-not-caused-by-differences-in-receptor-expression}{%
\subsection{Heterogeneity of response is not caused by differences in receptor expression}\label{heterogeneity-of-response-is-not-caused-by-differences-in-receptor-expression}}

Next, we tested whether the observed heterogeneity of response was caused by differences in receptor expression. We calculated Pearson correlation coefficients comparing control - normalised log viability values for each stimulus with transformed RNA counts of the corresponding stimulus receptor(s). No stimulus-receptor pair showed a Pearson coefficient greater than 0.4, confirming that the heterogeneity of response was not caused by differential receptor expression.



\hypertarget{generation-of-public-resource}{%
\section{Generation of public resource}\label{generation-of-public-resource}}

\hypertarget{shiny-app}{%
\subsection{Shiny app}\label{shiny-app}}

\hypertarget{package}{%
\subsection{Package}\label{package}}
\begin{itemize}
\tightlist
\item
  Present the underlying economic model/theory and give reasons why it is
  suitable to answer the given problem\footnote{Here is an example of a footnote.}.
\end{itemize}
\hypertarget{ex-vivo-sensitivity-to-microenvironmental-stimulation-in-primary-cll-cells}{%
\chapter{Ex-vivo sensitivity to microenvironmental stimulation in primary CLL cells}\label{ex-vivo-sensitivity-to-microenvironmental-stimulation-in-primary-cll-cells}}

The screen included 17 cytokines and microenvironmental stimuli, which were selected based on evidence in the literature that each stimulus had been shown to impact on CLL viability \emph{in vitro}. We attempted to put together the largest panel of stimuli for an assay of this kind in CLL, aiming to minimise redundancy among the compounds (Bruch \& Giles et al.~2021).

CLL cells do not proliferate \emph{in vitro}, but rather undergo spontaneous apoptosis in the absence of stimulation (\protect\hyperlink{ref-Collins1989}{Collins et al. 1989}). The effect of each stimuli on CLL viability was thus quantified by comparing ATP counts in treated primary samples, compared with those in DMSO wells where a positive value indicates that the sample viability was increased relative to control.

The assay represents a reductionist model of microenvironmental signalling, making it possible to dissect the effect of individual stimuli on baseline viability. The large patient cohort also enabled us to study the differential impact of these stimuli across heterogeneous samples, and identify distinct subgroups of patients whose samples show similar response profiles to the panel of stimuli.

\hypertarget{prolifing-responses-to-the-panel-of-stimuli}{%
\section{Prolifing responses to the panel of stimuli}\label{prolifing-responses-to-the-panel-of-stimuli}}

\hypertarget{ex-vivo-assay-demonstrated-functional-diversity-of-cytokines-and-microenvironmental-stimuli}{%
\subsection{\texorpdfstring{\emph{ex vivo} assay demonstrated functional diversity of cytokines and microenvironmental stimuli}{ex vivo assay demonstrated functional diversity of cytokines and microenvironmental stimuli}}\label{ex-vivo-assay-demonstrated-functional-diversity-of-cytokines-and-microenvironmental-stimuli}}

To investigate heterogeneity amongst responses to the stimuli, I calculated Pearson correlation coefficients for each stimulus pair, using the log-transformed normalised viabilities (Bruch \& Giles et al.~2021). The resulting coefficients were ordered using hierarchical clustering and visualised in a symmetrical heatmap (Figure \ref{fig:stimuliCorrelations}). The hierarchical clustering distinguished clusters of stimuli, including a larger group corresponding to agonists of TLR and Nf\(\kappa\)B pathways and a smaller group encompassing IL4 and TLR stimuli.


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/stimuliCorrelations-1} 

}

\caption{Heatmap of Pearson correlation coefficients. Coefficients for each pair of stimuli were calculated using log transformed viability values normalised to untreated control, and ordered according to hierarchical clustering. Figure adapted from Bruch \& Giles et al.~2021.}\label{fig:stimuliCorrelations}
\end{figure}
98.5\% of stimulus pairs showed little correlation (R \textless{} 0.6), including those that targeted similar downstream pathways, indicating a high degree of functional diversity amongst microenvironmental signals. For example, JAK-STAT agonists such as IL4 and IL6 showed little correlation (Figure \ref{fig:correlationsScatter}A).

Only two stimulus pairs showed correlations where R \textgreater{} 0.6, and in both cases these targeted near identical receptors or downstream pathways. These included CpG ODN (TLR 9) and Resiquimod (TLR 7 and 8) (Figure \ref{fig:correlationsScatter}B), and IL4 and IL4 + CD40L which primarily target JAK3 - STAT6.

Repeating the analysis to correlate drug - drug pairs demonstrated that drugs targeting components of the same pathway were highly correlated. For quality control purposes, this indicated that our data sensitively and specifically reflect inter-individual differences in pathway dependencies (\protect\hyperlink{ref-JCIpaper}{Dietrich et al. 2017}).


\begin{figure}

{\centering \includegraphics{thesis_files/figure-latex/correlationsScatter-1} 

}

\caption{Scatter plot of log-transformed viability values, normalised to DMSO controls, for (A) treatment with JAK-STAT agonists IL4 and IL6 and (B) treatment with TLR agonists CpG ODN and Resiquimod.}\label{fig:correlationsScatter}
\end{figure}
Microenvironmental stimulation induced diverse phenotypes between patient samples, and across different stimuli. To gain a global overview of these phenotypes, I visualised log-transformed viability values normalised to DMSO controls for all patient samples and all stimuli (Figure \ref{fig:stimuliBeeswarms} (Bruch \& Giles et al.~2021).


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/stimuliBeeswarms-1} 

}

\caption{Log transformed viabilities after treatment with each stimulus. Where stimuli decreased viability relative to control, points are shown in blue, whilst increased viability is shown in red. Figure adapted from Bruch \& Giles et al.~2021.}\label{fig:stimuliBeeswarms}
\end{figure}
The majority of the stimuli increased viability, underlining the supportive nature of the microenvironment in CLL. However, three out of 17 reduced CLL viability relative to control, namely IL6, TGF\(\beta\) and TLR 7/8/9 agonists in IGHV-mutated (IGHV-M) samples.

IL4 and TLR7/8/9 agonists Resiquimod and CpG ODN induced the strongest responses, an indication of their potency in modulating CLL cell survival. Notably, TLR agonists increased viability in certain samples, in most cases IGHV-U, and decreased viability in others, mostly IGHV-M. Both IL4 and TLR7/8/9 emerged as the key pathways in the screen, and play an important role throughout the results of this thesis.

\hypertarget{clusters}{%
\subsection{Microenvironmental response profiling identifies discrete patient subgroups}\label{clusters}}

To further investigate the variability in responses across the cohort, we visualised z scores of the log-transformed viabilities and performed consensus clustering on the resulting matrix to generate a heatmap of all stimuli responses across all samples (Figure \ref{fig:stimuliHeatmap}.


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/stimuliHeatmap-1} 

}

\caption{The heatmap matrix shows the viability measurements for 192 samples (columns) and 17 stimuli (rows). Viability was measured via ATP-based assay after 48h stimulation. The data are shown normalised to DMSO-treated controls, and scaled within each row according to the Median Absolute Deviation (MAD). Limits were applied to scaling factor for optimal visualisation. The colour bars to the right show sample annotations. Consensus Clustering was used to define column tree layout, using hierarchical clustering with the Euclidean metric. Figure from Bruch \& Giles et al.~2021.}\label{fig:stimuliHeatmap}
\end{figure}
The heatmap reflects heterogeneity in responses across samples and stimuli, once again underlining the potency of IL4 in increasing sample viability across diverse genetic backgrounds and the diversity in responses to TLR stimulation by Resiquimod and CpG ODN.

To generate the column-wise clustering, consensus clustering was performed on the matrix of z scores. Consensus clustering allows the user to subsample from the matrix of values, to generate hierarchical clustering for a given number of clusters, k. From this, it is possible to calculate a consensus matrix for each value of k, indicating for each pair of values the proportion of time they occupy the same cluster when subsampled together.

I visualised the clustered heatmap in Figure \ref{fig:stimuliHeatmap} for different values of k, and concluded on the existence of four robust clusters within the cohort (add these to appendix?). Each cluster shows a unique response profile to the panel of stimuli. We termed the clusters C1 to C4: C1 and C2 were enriched for IGHV-U whilst the samples in C3 and C4 were mostly IGHV-M.

To validate the choice of four clusters, I visualised the summaries of the consensus matrix, using the \texttt{ConsensusClusterPlus} package (\protect\hyperlink{ref-ConsensusClusterPlus}{M. D. Wilkerson and Hayes 2010}), to quantify the degree of confidence in the clusters for different values of k.


\begin{figure}

{\centering \includegraphics[width=0.45\linewidth]{figures/consensus008} \includegraphics[width=0.45\linewidth]{figures/consensus009} 

}

\caption{Summaries of the CDFs of the consensus matrices. Consensus CDF graphic showing the CDFs of the consensus matrix for k = 2 - 7, as indicated in the legend, estimated using 100 bin histogram (left). Relative change in area under the CDF curve, for k = 2 - 7, to compare k with k - 1. In the case of k = 2, there is no k - 1, so the total area is plotted. Line shows relative increase in consensus between each value of k (right).}\label{fig:consensusClusteringA}
\end{figure}
The graph of the CDFs of the consensus matrix for each k indicated that the CDF reaches a maximum and cluster confidence is maximised at k = 7 , though above k = 4 there is little appreciable increase. Figure \ref{fig:consensusClusteringA}. This is confirmed in the graph showing relative change in the area under the CDF curve, showing there is only a small increase in consensus between k = 4 and k =5, supporting the choice of k = 4. The cluster tracking plot depicts how each patient sample is assigned for each value of k. For k = 4, the plot indicates that C3 and 4 in particular are highly stable (Figure \ref{fig:consensusClusteringB}).


\begin{figure}

{\centering \includegraphics[width=0.45\linewidth]{figures/consensus010} 

}

\caption{Assignment of patient samples (columns), to each cluster, for k = 1 - 7 (rows) to demonstrate stability of cluster membership. Cluster colour for k = 4 match those in heatmap in \ref{fig:stimuliHeatmap}.}\label{fig:consensusClusteringB}
\end{figure}
\hypertarget{functional-characterisation-of-patient-clusters}{%
\section{Functional characterisation of patient clusters}\label{functional-characterisation-of-patient-clusters}}

\hypertarget{c1---c4-showed-distinct-response-profiles-with-the-panel-of-stimuli}{%
\subsection{C1 - C4 showed distinct response profiles with the panel of stimuli}\label{c1---c4-showed-distinct-response-profiles-with-the-panel-of-stimuli}}

The heatmap in Figure \ref{fig:stimuliHeatmap} demonstrated that each cluster responded differently to the panel of stimuli.

Amongst the IGHV-U enriched C1 and C2, both showed strong, positive responses to IL4 and TLR7/8/9 stimulation. C2 could be distinguished by stronger responses to the stimuli overall, in particular to NF\(\kappa\)B agonists IL1\(\beta\), anti-IgM, BAFF and sCD40L. Amongst the IGHV-M enriched clusters, C3 showed weaker responses to the majority of stimuli, and C4 was defined by a negative response to TLR7/8/9 stimulation (Bruch \& Giles et al.~2021). Figure \ref{fig:clusterStimuliResponse} summarises these findings in more detail, showing responses stratified by cluster for a subset of the stimuli.


\begin{verbatim}
cairo_pdf 
        2 
\end{verbatim}
\hypertarget{the-clusters-show-differences-in-disease-dynamics}{%
\subsection{The clusters show differences in disease dynamics}\label{the-clusters-show-differences-in-disease-dynamics}}

To validate the potential biological significance of these four clusters, we investigated whether the groups showed differential \emph{in vivo} disease progression (Bruch \& Giles et al.~2021). The study design was such that not all patients in the cohort were treatment - free, which confounded the analysis. For that reason, lymphocyte doubling time (LDT) and time to next treatment (TTT) were used to quantify CLL proliferative capacity, independently of treatment.

C1 and C2 showed a shorter LDT than C3 and C4, which is expected due to the differential proportions of IGHV-U and M patient samples in these groups (Figure \ref{fig:clusterLDTTTT}A). Notably, within the IGHV-M enriched clusters C3 and C4, samples in C3 showed a significantly shorter LDT (Student's t-test, p-value = 0.025).

To further validate this, we observed that TTT in the IGHV-M enriched C3 was significantly shorter than C4 and comparable to the progression dynamics of IGHV-U enriched C1 and 2 (Figure \ref{fig:clusterLDTTTT}B).


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/clusterLDTTTT-1} 

}

\caption{(A) Lymphocyte doubling time (LDT) stratified by cluster, p-values from Student's t-test. (B) Kaplan-Meier curves to show TTT for each cluster. p-values from univariate Cox proportional hazard models comparing IGHV-U enriched C1 with C2, and IGHV-M enriched C3 with C4. Figure from Bruch \& Giles et al.~2021.}\label{fig:clusterLDTTTT}
\end{figure}
The difference in disease progression between the clusters indicated that microenvironmental response represents an additional biological layer, holding information relevant to disease dynamics. To validate that these clusters were not simply an indication of any underlying genetic features, we checked whether the observed differences in progression dynamics could be explained by other prognostic markers (Bruch \& Giles et al.~2021).

A multivariate Cox proportional hazard model accounting for IGHV status, trisomy 12 and TP53 in addition to the cluster assignment indicated an independent prognostic value for cluster assignment between C3 and C4 (p= 0.039, Table \ref{tab:clusterCox}).


\begin{table}

\caption{\label{tab:clusterCox}Table depicting results of Multivariate Cox Proportional Hazard Model to test prognostic value of key genetic features and clusters using Time to Next Treatment and C3 as reference. }
\centering
\begin{tabular}[t]{l|r|r|r|r|l}
\hline
Factor & coef & exp(coef) & se(coef) & z & p.value\\
\hline
\cellcolor[HTML]{E2E868}{Cluster 3 vs Cluster 1} & \cellcolor[HTML]{E2E868}{-0.03979} & \cellcolor[HTML]{E2E868}{0.96099} & \cellcolor[HTML]{E2E868}{0.29813} & \cellcolor[HTML]{E2E868}{-0.13347} & \cellcolor[HTML]{E2E868}{0.89382}\\
\hline
Cluster 3 vs Cluster 2 & 0.51595 & 1.67522 & 0.37741 & 1.36708 & 0.1716\\
\hline
\cellcolor[HTML]{E2E868}{Cluster 3 vs Cluster 4} & \cellcolor[HTML]{E2E868}{-0.82011} & \cellcolor[HTML]{E2E868}{0.44038} & \cellcolor[HTML]{E2E868}{0.39760} & \cellcolor[HTML]{E2E868}{-2.06267} & \cellcolor[HTML]{E2E868}{0.03914}\\
\hline
IGHV.status & 0.55192 & 1.73658 & 0.27253 & 2.02513 & 0.04285\\
\hline
\cellcolor[HTML]{E2E868}{trisomy 12} & \cellcolor[HTML]{E2E868}{-0.13357} & \cellcolor[HTML]{E2E868}{0.87496} & \cellcolor[HTML]{E2E868}{0.35617} & \cellcolor[HTML]{E2E868}{-0.37503} & \cellcolor[HTML]{E2E868}{0.70764}\\
\hline
TP53 & 1.38977 & 4.01395 & 0.26072 & 5.33058 & <0.0001\\
\hline
\end{tabular}
\end{table}
\hypertarget{the-clusters-showed-differential-responses-to-drugs-in-vitro}{%
\subsection{\texorpdfstring{The clusters showed differential responses to drugs \emph{in vitro}}{The clusters showed differential responses to drugs in vitro}}\label{the-clusters-showed-differential-responses-to-drugs-in-vitro}}

The potential clinical relevance of the clusters was underlined by my observation that the samples within each group showed differential responses to drugs \emph{in vitro} (Figure \ref{fig:clusterDrugResponse}).

As expected, the IGHV-U enriched clusters C1 and 2 were more sensitive to BCR inhibition by ibrutinib, idelalisib and PRT062607, than C3 and 4. Between C1 and C2, C2 was more sensitive to a number of the drugs, including idelalisib (SYK) (p-value = 0.012), everolimus (mTOR) (p-value = 0.02) and the chemotherapeutics fludarabine (p-value = 0.031) and nutlin-3a (p-value = 0.042). Amongst C3 and C4, C3 showed lower sensitivity to everolimus (p-value = 0.051) and to fludarabine (p-value \textless{} 0.001) and nutlin-3a (p-value = 0.01). This aligns with the observation that patients in C3 have a poorer prognosis, despite most of these samples annotated as IGHV-M. C4 also showed a positive repsonse to Nf\(\kappa\)B inhibition by BAY-11-7085, and p38 MAPK inhibition by Ralimetinib.

Such differential drug response patterns suggests that microenvironmental response may reflect disease-specific CLL biology, in the same way as molecular profiling, and thus may have the potential to guide therapy decisions in future.


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/clusterDrugResponse-1} 

}

\caption{Log-transformed normalised viability values, stratified by cluster, for each drug. Drugs targeting the same pathway are grouped together. P-values from Student't t-test.}\label{fig:clusterDrugResponse}
\end{figure}
\hypertarget{the-clusters-are-enriched-for-different-genetic-features}{%
\subsection{The clusters are enriched for different genetic features}\label{the-clusters-are-enriched-for-different-genetic-features}}

Next we assessed differences in the molecular profiles of samples within each cluster. Visually, it appeared that certain clusters were enriched or depleted for various genetic features recurrent in CLL (Figure \ref{fig:clusterMatrix}.


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/clusterMatrix-1} 

}

\caption{Distribution of selected genetic features (rows) within each cluster for all patient samples (columns). Where a patient sample is not annotated for a feature, this is marked in white.}\label{fig:clusterMatrix}
\end{figure}
To quantify this, I ran a multinomial model, with lasso regularisation, to predict cluster membership (C1 -4) based on the matrix of genetic features for all the patient samples (Bruch \& Giles et al 2021). The model assigned coefficients to genetic features, where a positive coefficient indicated that this feature was enriched in the cluster, and a negative coefficient indicated it was depleted.

The approach used 3-fold cross validation, selecting the optimal model using lamda min. To ensure that the resulting coefficients were robust, we selected coefficients that satisfied certain cut-offs. Coefficients were selected if they were assigned in \textgreater{} 60\% of 50 bootstrapped repeats, and were larger than 0.35. Figure \ref{fig:clusterEnrichment} shows the mean coefficients and associated standard deviation, for each genetic feature that met these criteria in each cluster.


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/clusterEnrichment-1} 

}

\caption{Multinomial regression with lasso penalisation to identify enrichment or depletion of genetic features within each cluster. Matrix of genetic features (p=39), and IGHV status (encoded as M = 1 and U = 0) were used to identify multivariate predictors of cluster assignment. x axis shows genetic predictors, y axis indicates value and sign of coefficient assigned to feature, for each cluster (positive coefficients are enriched in the cluster, negative coefficients are depleted). Coefficients shown are mean coefficients from 50 bootstrapped repeats and error bars represent the mean ± standard deviation. Genetic features with \textgreater20\% missing values were excluded, and only patients with complete annotation were included in the model (n=137). Figure from Bruch \& Giles et al 2021.}\label{fig:clusterEnrichment}
\end{figure}
As we expected, IGHV status was the main feature that predicted cluster membership. Beyond IGHV status, trisomy 12 and \emph{SF3B1} mutations were enriched in C2, which showed enhanced responses to many stimuli. C4, which was associated with slow in-vivo progression, showed depletion of \emph{TP53}, \emph{ATM}, RAS/RAF mutations and gain8q.

\hypertarget{gsea-of-de-genes-between-subgroups}{%
\subsection{GSEA of DE genes between subgroups}\label{gsea-of-de-genes-between-subgroups}}

In addition to genetic features, I investigated differential expression of genes within each cluster. For n = 49 samples, RNAseq data was available for matched PBMC samples. I focused on the difference between clusters 3 and 4, for which 21 RNAseq samples were available (Bruch \& Giles et al.~2021).

To quantify differential gene expression, I began by filtering out immunoglobulin genes, including genes at the heavy, light and kappa loci that encode the antigen receptor of B cells. The clusters each show differential enrichment of IGHV-M and U samples, and thus the differential expression analysis would otherwise be dominated by immunoglobulin genes that are well known to affected by this biomarker.

I followed the Deseq2 protocol using a design formula to quantify the difference between clusters, and accounting for the confounding effect of IGHV status. 87 genes were differentially expressed (adjusted p \textless{} 0.05) between C3 and 4 (Figure \ref{fig:clusterDE}).


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/clusterDE-1} 

}

\caption{Volcano plot of differentially expressed genes between C3 and C4. X axis indicates log2 fold change values, calculated using the \texttt{DESeq2} package (\protect\hyperlink{ref-DESeq2}{\textbf{DESeq2?}}), y axis gives corresponding -log10(adjusted p value). P values adjusted using BH method. Genes are labeled where ajdusted p \textless{} 0.05. Figure from Bruch \& Giles et al.~2021.}\label{fig:clusterDE}
\end{figure}
To assign biological meaning to the differentially expressed genes, I quantified the enrichment of Hallmark pathways amongst the genes. I ranked the genes based on the Wald statistic, and then ran GSEA was using the fgsea algorithm (Figure \ref{fig:clusterGSEA}) (Bruch \& Giles et al.~2021).

Several pathways were upregulated amongst samples in C3, compared to C4, indicating that these pathways may relate in some way to the shorter TTT and LDT of patients within this cluster. Pathways associated with higher disease aggression were regulated in C3 including genesets relating to stress response (Unfolded Protein Response, UV Response Up, P53 Pathway), metabolism (Oxidative Phosphorylation) and proliferation (G2M Checkpoint, MYC Targets V1, MTORC1 Signaling, E2F Targets) (Figure \ref{fig:GSEApathways}).


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/clusterGSEA-1} 

}

\caption{Gene set enrichment analysis (GSEA) to compare expression of genes in samples from C3 and C4 reveals upregulation of Hallmark pathways involved in microenvironmental signalling, stress response, metabolism and proliferation in C3. Normalised enrichment scores (NES) are shown for top 10 most significant pathways upregulated in C3 versus C4. Bars coloured according to adjusted p-value. Genes are ranked based on Wald statistic, calculated using the Deseq2 package and GSEA performed using the fgsea algorithm. Figure from Bruch \& Giles et al.~2021.}\label{fig:clusterGSEA}
\end{figure}
In addition, C3 showed upregulation of microenvironmental signalling pathways relative to C4, including TNFa Signalling via NFKB and Interferon Gamma Response (Figure \ref{fig:GSEApathways}) (Bruch \& Giles et al.~2021). This finding underlines our hypothesis that differential activity of microenvironmental signalling, both \emph{in vivo} and \emph{ex vivo} may be relevant to disease prognosis.


\begin{figure}

{\centering \includegraphics{thesis_files/figure-latex/GSEApathways-1} 

}

\caption{Enrichment plots of selected pathways. Gene set enrichment analysis (GSEA) was performed with the Hallmark gene sets from the GSEA Molecular Signatures Database. Wald statistic was used to rank the genes. The green curve corresponds to the Enrichment Score curve, which is the running sum of the weighted enrichment score obtained from GSEA software. Figure from Bruch \& Giles et al.~2021}\label{fig:GSEApathways}
\end{figure}
\hypertarget{additional-analysis}{%
\section{Additional Analysis}\label{additional-analysis}}

Can microenvironmental response aid prognostic models? Run a Multivariate model to predict survival/drug response with genetic matrix and stimuli response matrix?

\hypertarget{summary}{%
\section{Summary}\label{summary}}

The screen represents an attempt to comprehensively dissect the impact of individual microenvironmental pathways on CLL viability. Our assay has enabled us to highlight key, broad spectrum signals such as IL4. Our heterogeneous cohort also reveals pathways that operate in subsets of patients, such as TLR. IL4 and TLR represent the key pathways that we focus on throughout the rest of this thesis.

In addition, this approach enables us to use microenvironmental response profiles, as an alternative to molecular profiling, to look for the existence of subgroups.

We uncovered four such subgroup with distinct response profiles and molecular properties and clinical outcomes, suggesting that microenvironmental response holds biologically significant information that may be relevant to prognosis and treatment decision making.

\hypertarget{discussion}{%
\section{Discussion}\label{discussion}}

Gosia suggested discussing each results section individually?

\hypertarget{contributions-statement}{%
\section{Contributions statement}\label{contributions-statement}}

The author of this thesis performed statistical inference and regression modelling outlined in this chapter. In essence, she performed data visualisation, optimised and interpreted the clustering of the heatmap to identify four patient subgroups, and investigated the molecular and phenotypic profiles of the four subgroups.

Peter-Martin Bruch generated the viability screening data, and curated and analysed the clinical data on LDT and TTT. The genetic annotation data was taken from (\protect\hyperlink{ref-JCIpaper}{Dietrich et al. 2017}). The heatmap in section was generated collectively with Peter-Martin Bruch, Wolfgang Huber and Sascha Dietrich, for the paper Giles \& Bruch et al 2021. The author also received support from Junyan Lu for the generation of the multivariate model, and from Peter-Martin Bruch and Sophie Herbst for the consensus clustering analysis.

\hypertarget{genetic-modulators-of-responses-to-microenvironmental-stimulation}{%
\chapter{Genetic modulators of responses to microenvironmental stimulation}\label{genetic-modulators-of-responses-to-microenvironmental-stimulation}}

Profiling the effects of the panel of stimuli revealed the heterogenous nature of responses to stimulation. The analysis revealed four patient sample groups, with distinct responses to the stimuli, unique molecular profiles, and differential disease progression. Two pathways emerged as the most potent modulators of CLL viability, namely IL4 and TLR. These too induced heterogeneous responses across the cohort, in particular stimulation of TLR 7/8/9 increased viability in some samples, whilst it reduced viability in others.

Next we asked to what extent this heterogeneity of response relates to the molecular profiles of the tumours. We sought to understand which underlying genetic features might modulate responses to external signals, and how these interactions may occur. We combined the screening dataset with multi-omics profiles of the patient samples taken from the PACE repository (\protect\hyperlink{ref-JCIpaper}{Dietrich et al. 2017}; \protect\hyperlink{ref-R-BloodCancerMultiOmics2017}{Oles et al. 2021}) and performed a systematic survey of molecular determinants of stimulus response, using whole-exome sequencing, DNA-methylation, RNA-sequencing and copy number variant data. In addition, for a small subset of patients we also generated ATAC sequencing and Mass Spectrometry data.

Collectively, these data enabled us to probe genetic and epigenetic modulators of microenvironmental signalling, in a heterogeneous cohort that encompasses the clinical and molecular diversity of CLL. In the first part of this chapter, I apply a broad systematic approach to identify important genetic modulators, and in the second, I outline my follow-up investigations into the impact of trisomy 12 on stimulus response.

\hypertarget{systematic-analysis-of-the-effect-of-genetic-features-on-responses-to-stimuli}{%
\section{Systematic analysis of the effect of genetic features on responses to stimuli}\label{systematic-analysis-of-the-effect-of-genetic-features-on-responses-to-stimuli}}

\hypertarget{univariate-gene-stimulus-assosciations}{%
\subsection{Univariate analysis identifies IGHV status and trisomy 12 as key modulators of microenvironmental response}\label{univariate-gene-stimulus-assosciations}}

To begin, I ran a univariate analysis to compare viability values post-stimulation for patient samples with and without each genetic feature. In total 63 genetic features were surveyed, including IGHV status, somatic gene mutations and structural variants, where there were at least three patient samples in each group (Figure \ref{fig:stimuliGeneAssosciations}).


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/stimuliGeneAssosciations-1} 

}

\caption{Plot showing BH-adjusted p values from Student's t-tests (two-sided, with equal variance), for all tested gene-stimulus associations. Tests performed for IGHV status and somatic mutations and copy number aberrations with ≥3 patient samples in each group (n = 63). Each circle represents a gene-stimulus association. Associations that meet 10\% FDR cut off are indicated in colour, where the colour denotes the genetic feature. \emph{Figure from Bruch \& Giles et al.~2021.} See Methods section \ref{univariate-gene-stimulus-associations-method}.}\label{fig:stimuliGeneAssosciations}
\end{figure}
This analysis revealed the extent to which genetic features modulate microenvironmental response: for ten out of 17 stimuli, at least one genetic feature determined response and for six out of 17 stimuli, two or more genetic features significantly altered the response (Student's t-tests, FDR = 10\%, Bruch \& Giles et al.~2021). The most common features were IGHV status and trisomy 12. Del(11q) also affected response to several stimuli. Notably, del(13q) and del(17p), which like trisomy 12 and del(11q) are the most common aberrations in CLL and act as prognostic markers (\protect\hyperlink{ref-Dohner2000}{Döhner et al. 2000}), had no impact on the responses to the panel of stimuli.

\hypertarget{multivariate-gene-stimulus-assosciations}{%
\subsection{Multivariate analysis of gene - stimulus assosciations confirms IGHV status and trisomy 12 as key modulators of stimulus response}\label{multivariate-gene-stimulus-assosciations}}

It was possible that there may be interplay between genetic factors in determining responses to external signals. To address this, I applied multivariate modelling to integrate the influence of genetic features, IGHV status and DNA methylation on the size of response (Bruch \& Giles et al.~2021). I used a Gaussian linear model with L1-penalty (i.e., lasso regression), to derive a predictor for each stimulus, comprised of these covariates. For background on this approach, see section \ref{intro-multivariate-modelling}.

As input to the model, the response matrix was composed of the log transformed viability values for each stimulus. To generate the feature matrix (137 samples versus 39 features), I excluded genetic features for which \textgreater20\% of the values were missing, and patient samples with incomplete annotation. As predictors, I included genetic mutations and CNVs (p= 39), IGHV status (coded as 0-1) and Methylation Cluster (coded as 0, 0.5, 1). I ran lasso regression, as implemented in the \texttt{R} package \texttt{glmnet}(\protect\hyperlink{ref-R-glmnet}{Friedman et al. 2021}), using three-fold cross-validation with misclassification error as loss. The resulting predictors are the mean of those coefficients that were selected in at least 75\% of 30 bootstrapped repeats.

Using the output of the regression, I generated predictor profiles for each stimulus. For 5 / 17 stimuli, there was at least one genetic predictor that met the cut-offs (a selection are shown in figure \ref{fig:stimuliGeneAssosciationsMulti}).


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/stimuliGeneAssosciationsMulti-1} 

}

\caption{Predictor profiles for IL4 and CpG ODN depicting gene - stimulus associations. Bar plots (left) show size and sign of assigned coefficients from Gaussian linear modelling. A positive coefficient indicates that stimulated increase in viability is larger when feature is present. Scatter plots (bottom) and corresponding heatmaps above show how presence of selected genetic feature relates to sample viabilities. Scatter plots show ranked log(viability) values for each sample and heatmaps show mutation status for each predictor, for corresponding sample in scatter plot. See method \ref{multivariate-gene-stimulus-assosciations-method}. \emph{Figure adapted from Bruch \& Giles et al.~2021}.}\label{fig:stimuliGeneAssosciationsMulti}
\end{figure}
The multivariate analysis demonstrated that responses to IL4 and sCD40L + IL4, TLR and TGF\(\beta\) were all affected by multiple genetic features (Bruch \& Giles et al.~2021). For example, higher viability in response to IL4 stimulation was associated with trisomy 12 and unmutated IGHV. In contrast, IL4 generated little or no increase in viability in samples with a mutation in KRAS, NRAS or BRAF. These tumours benefited less from the anti-apoptotic effects of IL4, indicating that the ability of IL4 to increase viability could be dependent on signalling via Ras-Raf-MEK-ERK, which is already active in mutant samples.

Discussion that Inhibition of ERK (p38) also increased IL4 protective effect

\hypertarget{response-to-tlr-stimulation-is-dependent-on-ighv-status-trisomy-12-and-mutations-in-dna-damage-response-genes}{%
\subsection{Response to TLR stimulation is dependent on IGHV status, trisomy 12 and mutations in DNA Damage Response genes}\label{response-to-tlr-stimulation-is-dependent-on-ighv-status-trisomy-12-and-mutations-in-dna-damage-response-genes}}

The multivariate analysis highlighted that TLR stimulation by CpG ODN and Resiquimod both showed the largest number of predictors, reflecting the multiple layers of biology involved here. These included del(11q) and \emph{ATM} , del(17p) and \emph{TP53}, IGHV status, trisomy 12 and \emph{SF3B1} (Bruch \& Giles et al.~2021). Amongst the screening cohort, TLR stimulation increased viability in certain samples and decreased in others. The clustering groups identified in section \ref{clustering} also showed highly divergent responses to TLR, and this related to disease progression, thus warranting a more detailed look at the underlying features modulating TLR response (Figure \ref{fig:TLRGenes}).


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/TLRGenes-1} 

}

\caption{Control-normalised log transformed viability values after treatment with Resiquimod (TLR 7/8), stratified by named genetic features.}\label{fig:TLRGenes}
\end{figure}
IGHV status (along with Methylation Cluster, which is related to IGHV) and trisomy 12 had the strongest effect on TLR response. \protect\hyperlink{ref-Chatzouli2014}{Chatzouli et al.} (\protect\hyperlink{ref-Chatzouli2014}{2014}) have previously shown that TLR responses is dependent on IGHV status (see also, section \ref{intro-ME-pathways}. We noted trisomy 12 as an additional, novel determinant of TLR response (Bruch \& Giles et al.~2021). In samples that do not have trisomy 12, TLR stimulation increases the viability of IGHV-U samples, whilst it decreases viability in IGHV-M samples, as expected. In contrast, in samples with trisomy 12, TLR stimulation increases viability regardless of IGHV status (Student's t-test, p\textless0.001 and p=0.018, Figure \ref{fig:TLRIHGVtri12}).

\protect\hyperlink{ref-Chatzouli2014}{Chatzouli et al.} (\protect\hyperlink{ref-Chatzouli2014}{2014})'s observations point to the existence of specific types of BCR/TLR collaboration in CLL, leading to activation of pro-survival pathways, or apoptosis depending on the IGHV status of the tumour. Our results suggest that synergy between BCR and TLR may also be dependent on trisomy 12 status.


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/TLRIHGVtri12-1} 

}

\caption{Beeswarm-boxplot showing control-normalised log transformed viability values, after treatment with Resiquimod, stratified by trisomy 12 and IGHV status. p-values from Student's t-tests. \emph{Figure from (Bruch \& Giles et al.~2021)}}\label{fig:TLRIHGVtri12}
\end{figure}
In addition to trisomy 12, I noted that TLR stimulation also increased viability in samples with mutations in the DNA damage response pathway, namely del(11q), del(17p), and \emph{ATM} and \emph{TP53}. Del(17p) occurs in 7\% of CLL cases and is associated with loss of TP53, a tumour suppressor gene involved in DNA damage and repair (\protect\hyperlink{ref-Zenz2010}{Zenz et al. 2010}). Del(11q) is more common (18\% of cases) and is associated with loss of ATM, another protein involved in DNA repair(\protect\hyperlink{ref-Kipps2017}{Kipps et al. 2017}). This suggests that there may also be cross-talk between the TLR and DNA damage response pathway, and that this affects the outcome of TLR stimulation.

The effect of TLR stimulation on CLL viability is dependent on the molecular make-up of the tumour. The nature of TLR response may also be important in prognosis, underlined by our the observation in section \ref{clustering} that a subgroup of patients (C3) shows slower disease progression, and is the only group in which TLR induces apoptosis. The role of TLR signalling in pathogenesis and prognosis may thus so far be under appreciated.

Collectively, this work represents the first large-scale attempt to profile the integrative effects of cell-extrinsic signals and cell-intrinsic features in lymphoma on a large scale. This systematic approach highlighted trisomy 12 as the most common feature to modulate responses. The rest of this chapter outlines my work to investigate the role of trisomy 12 in microenvironmental response.

\hypertarget{investigating-trisomy-12-as-a-modulator-of-microenvironmental-response}{%
\section{Investigating Trisomy 12 as a modulator of microenvironmental response}\label{investigating-trisomy-12-as-a-modulator-of-microenvironmental-response}}

\hypertarget{trisomy12-modulator}{%
\subsection{Trisomy 12 is a modulator of microenvironmental response}\label{trisomy12-modulator}}

Trisomy 12 modulated responses to IL4, TGF\(\beta\), soluble CD40L + IL4 and TLR stimuli (Bruch \& Giles et al.~2021, Figure \ref{fig:tri12cytResponse}). For example, the increase in viability induced by IL4 was enhanced in trisomy 12 samples, as was the decrease in viability in response to TGF\(\beta\) stimulation.

Trisomy 12 is commonly mutated in CLL (15\% of patients)(\protect\hyperlink{ref-Dohner2000}{Döhner et al. 2000}). Until recently, this genetic lesion was regarded as conferring intermediate risk, though novel therapies have improved outcomes for trisomy 12 patients (\protect\hyperlink{ref-Bosch2019}{Bosch and Dalla-Favera 2019}). However, the functional explanation for its recurrence is incompletely understood. Previous work has shown a role for gene dosage effects in the pathogenic mechanism: \protect\hyperlink{ref-Kienle2005}{Kienle et al.} (\protect\hyperlink{ref-Kienle2005}{2005}) show that overexpression of genes including CDK4 and E2F1 leads to increased cell cycling and higher proliferative capacity. Likewise, \protect\hyperlink{ref-HerbstThesis}{Herbst} (\protect\hyperlink{ref-HerbstThesis}{2020}) showed that BCR signalling proteins are also upregulated, which may also contribute to the increased proliferative capacity. Despite this, outcomes in trisomy 12 CLL are more favourable, and these cases show increased susceptibility to BCR inhibition (\protect\hyperlink{ref-JCIpaper}{Dietrich et al. 2017}) (see also, section \ref{intro-trisomy12}).

In the following, I outline my work to investigate the incompletely understood role of trisomy 12 in CLL.


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/tri12cytResponse-1} 

}

\caption{Control-normalised log transformed viability values after treatment with IL4, sCD40L + IL4 and TGF\(\beta\), stratified by trisomy 12.}\label{fig:tri12cytResponse}
\end{figure}
\hypertarget{gene-dosage-effects}{%
\subsection{STAT6, IRAK4 and SMAD3 are more highly expressed in trisomy 12 CLL}\label{gene-dosage-effects}}

Trisomy 12 samples contain a third copy of chromosome 12: transcriptomic and proteomic profiling of CLL samples with this lesion have demonstrated that this has a major impact on gene expression and protein abundances (\protect\hyperlink{ref-Abruzzo2018}{Abruzzo et al. 2018}; \protect\hyperlink{ref-HerbstThesis}{Herbst 2020}; \protect\hyperlink{ref-MeierAbt2021}{\textbf{MeierAbt2021?}}). It has also been shown that BCR signalling proteins are amongst those with increased expression in trisomy 12 (\protect\hyperlink{ref-HerbstThesis}{Herbst 2020}), and this is thought to contribute to increased susceptibility to BCR inhibition in trisomy 12 CLL (\protect\hyperlink{ref-JCIpaper}{Dietrich et al. 2017}).

Guided by this observation, I began by investigating RNA and protein expression levels amongst genes involved in the TGF\(\beta\), IL4 and TLR pathways, to which trisomy 12 samples respond more strongly. The aim was to determine whether proteins in these pathways are more abundant in trisomy 12 CLL, thus contributing to the enhanced response.

I ran differential expression analysis to compare CLL samples from the screen with and without trisomy 12. Next, I filtered the differentially expressed genes (adjusted p \textless{} 0.1) for those belonging to the TGF\(\beta\), JAK-STAT and TLR pathways genesets, from the KEGG database. I visualised the RNA counts and protein abundances (\protect\hyperlink{ref-HerbstThesis}{Herbst 2020}) for those genes.

Only a small proportion of the differentially expressed genes belonged to the TGF\(\beta\), JAK-STAT and TLR pathways genesets. However of those that were, several genes were key downstream mediators of these pathways (Figure \ref{fig:geneDosage}. Amongst TGF\(\beta\) signalling genes, 7 / 95 were upregulated in trisomy 12, including SMAD3. 12 / 160 IL4 signalling genes were differentially expressed, including STAT2. Only 2 / 116 genes in the TLR geneset were differentially expressed, though this included IRAK4 on chromosome 12. Amongst these key mediators SMAD3, STAT2, and IRAK4, all showed higher protein abundance in addition to increased RNA expression. Notably, STAT6, the key downstream mediator of IL4 signalling, was not differentially expressed, but also showed differential protein abundance. Interestingly, many of the over-expressed genes are not located on chromosome 12, indicating the extent to which the differential dosage of this chromosome has on the expression of the entire genome.


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/geneDosage-1} 

}

\caption{Beeswarm-boxplots showing RNA counts and protein abundances stratified by trisomy 12 status. P values from Student's t-test. \_Proteomics dataset from (\protect\hyperlink{ref-HerbstThesis_}{\textbf{HerbstThesis\_?}}) Method: \ref{gene-dosage-effects-method}.}\label{fig:geneDosage}
\end{figure}
Next I looked for further evidence to directly link the higher gene dosage of cytokine signalling genes, to enhanced responses to cytokine signals.

\hypertarget{trisomy12-classifier}{%
\subsection{Classification analysis identifies trisomy 12 phenocopies that show increased expression of IRAK4 and SMAD3}\label{trisomy12-classifier}}

I next investigated whether any non-trisomy 12 samples might display a trisomy 12-like phenotype (refered to as a phenocopy), in that they respond in a similar way to the panel of stimuli. The aim was to identify trisomy 12 phenocopies and to isolate the feature of these samples that might explain the underlying cause of enhanced response to external signals in trisomy 12 CLL. In particular, I was interested to see if these phenocopies showed higher expression, or even gene amplification, of the signalling genes identified in section \ref{gene-dosage-effects}.

To identify trisomy 12 phenocopies, I began by generating a classifier to predict the trisomy 12 status of a sample from its stimulus response matrix. I aimed to find non-trisomy 12 samples that were consistently misclassified as trisomy 12.

The classifier was built using binomial regression, with lasso penalisation, as implemented in the \texttt{R} package \texttt{glmnet} (\protect\hyperlink{ref-R-glmnet}{Friedman et al. 2021}). The feature matrix consisted of z scores of the viability values after treatment with each stimulus, and was used to predict the response (trisomy 12 status). I ran the model for 50 bootstrapped repeats, using three-fold cross-validation and mean absolute error as loss. Resiquimod, sCD40L+IL4 and TGF\(\beta\) were selected as coefficients that predict trisomy 12 status (Figure \ref{fig:tri12Classfier}, as would be expected bed on the observations in section \ref{trisomy12-modulator}.


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/tri12Classifier-1} 

}

\caption{Predictor profile showing coefficients selected as predictors of trisomy 12 status. Here, binomial modelling with L1-penalty was used to identify associations between responses to stimuli and trisomy 12 status. Bar plots (left) show size and sign of assigned coefficients named on the right. A positive coefficient indicates that higher viability upon treatment named stimulus is associated with trisomy 12. Facet labels (top) and corresponding heatmap below show how the viability with each named stimulus relates to trisomy 12 status. The model shown was selected from 50 bootstrapped repeats, based on maximal AUC. Method \ref{trisomy12-classifier-method}.}\label{fig:tri12Classifier}
\end{figure}
I then predicted trisomy 12 status with the same viability matrix, using each of the 50 bootstrapped. I compared the results of the classification with the true trisomy 12 status. Two patient samples were misclassified as trisomy 12 in more than 50\% of repeats; I refer to these as patient sample A and B in the below. Patient sample A showed the lowest viability with TGF\(\beta\) (Figure \ref{fig:tri12PhenocopiesCytResponse}, and Patient sample B showed the highest viability with IL4, and the second highest with Resiquimod.


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/tri12PhenocopiesCytResponse-1} 

}

\caption{Control-normalised log-transformed viability values, for all samples after treatment with Resiquimod, sCD40L + IL4 and TGF\(\beta\). Patient sample A and B are indicated in blue and red, respectively.}\label{fig:tri12PhenocopiesCytResponse}
\end{figure}
I next investigated whether either of these non-trisomy 12 samples contained any regional amplifications on chromosome 12, which may help to isolate the gene(s) causing the tumour to respond more strongly to these signals. The whole exome sequencing for both patient samples indicated that Patient sample A had several amplified regions at 12p13.31 (42 copies), 12q24.13 (10 copies) and 12q24.33 (21 copies). An examination of the genes in these regions indicated that none of the signalling genes identified in \ref{gene-dosage-effects} could be found in these regions, and no clear candidate gene(s) emerged that may be involved in responses to external signals.

In the absence of gene amplification, I next checked patient samples A and B might show higher expression of the signalling genes identified in \ref{gene-dosage-effects} by visualising the RNA expression levels for SMAD3, IRAK4 and STAT6 for these patients (Figure \ref{fig:tri12PhenocopiesRNA}. Patient sample A, which responded most strongly to Resiquimod, showed high levels of IRAK4 compared to the other non-triomsy 12 samples. Patient sample B, which responded strongly to TGF\(\beta\) showed the highest level of SMAD3 expression amongst the non-trisomy 12 samples. Thus, whilst neither patient sample contained an amplicon of SMAD3 or IRAK4, its possible that increased expression of these proteins enables a stronger response to the corresponding pathway.


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/tri12PhenocopiesRNA-1} 

}

\caption{Raw RNA counts for SMAD3, IRAK4 and STAT6, stratified by trisomy 12 status. Patient sample A and B are indicated in blue and red, respectively. P values from Student's t-test.}\label{fig:tri12PhenocopiesRNA}
\end{figure}
These results collectively suggest that increased gene dosage of key genes in the IL4, TLR and TGF\(\beta\) pathways may underlie the increased response of trisomy 12 samples to these pathways. This is also reflected in the patient samples A and B, which responded more strongly to the TGF\(\beta\) and TLR pathways, respectively, and correspondingly higher levels of SMAD3 and STAT6 proteins. However, computational techniques can only go to far in providing biological proof, and more work is needed here to confirm this finding. In addition to gene dosage effects, I also investigated differences in transcription factor activity in trisomy 12. The results of this warranted further follow up, and this forms the focus of rest of this chapter.

\hypertarget{spi-b-and-pu.1-tfs-show-higher-activity-in-trisomy-12-cll}{%
\subsection{Spi-B and PU.1 TFs show higher activity in trisomy 12 CLL}\label{spi-b-and-pu.1-tfs-show-higher-activity-in-trisomy-12-cll}}

Trisomy 12 has been well-studied at both the transcriptomic and proteomic level, and yet the cause of its recurrence in CLL is not fully understood. Thus, I next decided to investigate the impact of trisomy 12 on the epigenetic landscape of CLL, which is less well studied. In particular, I wanted to investigate differential transcription factor activity in CLL, which would give an indicator of which pathways are differentially active.

We acquired two independent ATAC sequencing datasets. The first, generated and processed in our lab, consisted of two WT and two trisomy 12 samples (Bruch \& Giles et al.~2021). The second, taken from (\protect\hyperlink{ref-Rendeiro2017}{\textbf{Rendeiro2017?}}) and processed in our lab, comprised 43 WT and nine trisomy 12 samples.

In the external dataset, trisomy 12 status was not annotated. To do this, we used the ATACseq reads to call trisomy 12 in samples that contained \textgreater{} 1.4 times more reads per peak (i.e.~genomic region) on average in chromosome 12, compared to peaks on other chromosomes.

Next, we used the \texttt{R} package \texttt{diffTF} (\protect\hyperlink{ref-R-diffTF}{\textbf{R-diffTF?}}) to identify TFs that showed differential binding site accessibility between the WT and trisomy 12 samples (Figure \ref{fig:diffTFexplainer}. \texttt{diffTF} enables comparison of TF activity between two conditions, using chromatin accessibility data, see section \ref{intro-ATACseq}. We provided a list of 636 TFs, from the HOCOMOCO v10 database (\protect\hyperlink{ref-HOCOMOCO}{Kulakovskiy et al. 2016}) to measure their change in actiitvy between trisomy 12 and WT samples (Bruch \& Giles et al.~2021).


\begin{figure}

{\centering \includegraphics[width=0.5\linewidth]{figures/diffTF_method} 

}

\caption{diffTF workflow from \protect\hyperlink{ref-Berest2019}{Berest et al.} (\protect\hyperlink{ref-Berest2019}{2019}). diffTF accepts a list of TFs along with the genomic locations of their bindings sites. For each TF, the software computes the distribution of fold changes between the trisomy 12 and WT samples, using the ATACseq peaks at each TF binding site in each condition. The software compares this distribution to a background set of fold changes generated using GC content-matched loci that do not contain the same TF binding site motif. Each TF is thus assigned assigned a weighted mean difference value, which quantifies the change in activity, and a p value.}\label{fig:diffTFexplainer}
\end{figure}
Running this analysis on both ATACseq datasets indicated that the binding sites of nine TFs were more accessible (p\textless0.05) in the trisomy 12 samples of the larger, external dataset (Figure \ref{fig:tri12diffTF}. In the smaller in-house dataset, the binding sites of 92 TFs were likewise more accessible (Supp. Figure \ref{fig:diffTFsmallvolPlot}). RNA and proteomics data show different abundances of transcripts and proteins in trisomy 12 CLL: ATACseq data here shows that this also corresponds to a specific signalling signature in trisomy 12 CLL.

We confirmed that the results were not affected by the additional copy of chromosome 12. Rerunning diffTF analysis without the ATACseq reads from chromosome 12 had minor impact on the significant TFs.

In both datasets, the TFs with the largest increase in activity in trisomy 12 were Spi-B and/or PU.1 (Bruch \& Giles et al.~2021). Both TFs share similar binding motifs and exhibit functional redundancy (\protect\hyperlink{ref-GarrettSinha2001}{\textbf{GarrettSinha2001?}}), which make it difficult to distinguish from ATACseq data alone whether either or both are more active.

Spi-B and PU.1 are haematopoetic regulators that are known to be key regulators of healthy B-cell function (\protect\hyperlink{ref-Turkistany2011}{Turkistany and Dekoter 2011}){[}READ MORE{]}, controlling B-cell responses to environmental cues including CD40L, TLR ligands and IL4 (\protect\hyperlink{ref-Willis2017}{Willis et al. 2017}).


\begin{figure}

{\centering \includegraphics[width=0.8\linewidth]{thesis_files/figure-latex/tri12diffTF-1} 

}

\caption{Bar plot showing the results of the diffTF analysis for the \protect\hyperlink{ref-Rendeiro2016}{Rendeiro et al.} (\protect\hyperlink{ref-Rendeiro2016}{2016}) dataset. y axis shows change in TF activity (weighted mean difference) between trisomy 12 (n = 9) and non-trisomy 12 samples (n = 43),x axis indicates TF names. 17 / 636 TFs, with BH adjusted p \textless0.05 are shown. p values generated by diffTF in permutation mode. TF binding sites defined in HOCOMOCO v10 (\protect\hyperlink{ref-HOCOMOCO}{Kulakovskiy et al. 2016}).}\label{fig:tri12diffTF}
\end{figure}
Spi-B and PU.1 appeared to be upregulated in trisomy 12 CLL, and evidence in the literature indicated that these may regulate environmental sensing genes, providing a link between trisomy 12, and enhanced responses to external signals.

To provide further evidence of this, I next aimed to profile the downstream effects of Spi-B and PU.1 in lymphoma.

\hypertarget{spi-b-and-pu.1-targets-are-enriched-for-immune-signalling-pathways}{%
\subsection{Spi-B and PU.1 targets are enriched for immune signalling pathways}\label{spi-b-and-pu.1-targets-are-enriched-for-immune-signalling-pathways}}

I hypothesised that Spi-B and PU.1 might coordinate transcriptional response to external signals, thus modulating CLL proliferation in response to the microenvironment. To identify Spi-B and PU.1 target genes specifically in lymphoma, I acquired a ChIPseq dataset (\protect\hyperlink{ref-Care2014}{Care et al. 2014}) containing data on Spi-B and PU.1 binding in lymphoma cell lines. I used this dataset to test for functional enrichment of immune signalling pathways amongst the TF targets.

To define TF targets, I took the closest gene to each significant ChIP peak (q value\textless0.05), and within \(\pm\)kb of TSS. I then tested for over-representation of these TF targets amongst selected KEGG(\protect\hyperlink{ref-KEGG}{\textbf{KEGG?}}) and Reactome (\protect\hyperlink{ref-Reactome}{\textbf{Reactome?}}) genesets, using the `R' package \texttt{clusterProfiler} (\protect\hyperlink{ref-R-clusterProfiler}{Yu 2021b}). This method corresponded to a one-sided version of Fisher's exact test. This analysis showed TLR, BCR and TGF\(\beta\) signalling genes to be enriched (p\textless0.01) amongst Spi-B and PU.1 targets (Figure \ref{fig:SpiBChipSeq}).


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/SpiBChipSeq-1} 

}

\caption{Table shows results from over-representation tests of selected KEGG and Reactome pathways amongst Spi-B and PU.1 targets. Columns show geneset pathways, corresponding database, the number of genes within geneset, the number of Spi-B and PU.1 target genes within geneset (total number of target genes defined also shown), and p-value from over-representation test. TF targets defined as closest gene to each significant ChIP peak (q value\textless0.05), and within \$\pm\$1kb of TSS. ChIPseq data from \protect\hyperlink{ref-Care2014}{Care et al.} (\protect\hyperlink{ref-Care2014}{2014}). Over-representation tests run using (\protect\hyperlink{ref-clusterProfiler}{\textbf{clusterProfiler?}}) package, method corresponds to one-sided version of Fisher's exact test. \emph{Figure and caption from Bruch \& Giles et al 2021.}}\label{fig:SpiBChipSeq}
\end{figure}
\hypertarget{double-knockdown-of-spi-b-and-pu.1-reduces-profileration-of-trisomy-12-cell-lines}{%
\subsection{Double knockdown of Spi-B and PU.1 reduces profileration of trisomy 12+ cell lines}\label{double-knockdown-of-spi-b-and-pu.1-reduces-profileration-of-trisomy-12-cell-lines}}

To establish the functional impact of Spi-B and PU.1 inhibition in trisomy 12+ lymphoma, we tested th impact of inhibiting these TFs on proliferation of lymphoma cell lines (Figure \ref{fig:SpiBshRNAKD}. We generated single and double shRNA knockdowns in lymphoma cell lines, namely SU-DHL4 and SU-DHL5 (trisomy 12) and SU-DHL2 (no trisomy 12) and then measured cell counts at 24 hour intervals (Bruch \& Giles et al.~2021).


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/SpiBshRNAKD-1} 

}

\caption{Knockdown of Spi-B and PU.1 led to growth restriction and cell death. Data shown is the mean of three technical replicates. While SU-DHL 5 was already growth impaired after Spi-B KD, SU-DHL 2 and SU-DHL 4 were only impaired after double knock-down. Double knockdown lead to rapid cell death in SU-DHL 5 (data not shown). Cell count normalised to seeded cell number of diffuse large b cell lymphoma cell lines after knockdown of Spi-B or double knockdown of Spi-B and PU.1 or shRNA as control. \emph{Figure and caption from Bruch \& Giles et al.~2021.}}\label{fig:SpiBshRNAKD}
\end{figure}
The single knockdowns had a small impact on proliferation: Spi-B inhibition reduced proliferation in SU-DHL5, as did PU.1 to a lesser extent. Double knockdown of both TFs markedly reduced proliferation in SU-DHL2 and SU-DHL4, and was lethal in SU-DHL5. This result suggested that both these TFs play an important role in the proliferative capacity of the tumour cells, and that there is functional redundancy between Spi-B and PU.1 in this context.

Collectively, these results demonstrate that trisomy 12 modulates responses to microenvironmental signals. Trisomy 12 appears to increase Spi-B and PU.1 activity, which regulate genes relating to environmental sensing, and reduce proliferation of cell lines when inhibited.

\hypertarget{summary-1}{%
\section{Summary}\label{summary-1}}

In this chapter, we perform a systematic survey of genetic determinants of microenvironmental response, leading to two key findings. Firstly, TLR signalling has a range of effects on CLL viability, and this is determined by many genetic features, including IGHV status, trisomy 12 and mutation in the DNA Damage Response pathway. Secondly, we identify trisomy 12 as a modulator of microenvironmental response, and show that higher activity of Spi-B and PU.1 may mediate this effect.

\hypertarget{discussion-1}{%
\section{Discussion}\label{discussion-1}}

Some do affect, some dont at all
Notably, del(13q) and del(17p), which like trisomy 12 and del(11q) are the most common aberrations in CLL and act as prognostic markers (\protect\hyperlink{ref-Dohner2000}{Döhner et al. 2000}), had no impact on the responses to the panel of stimuli.
ie some are really affecting signalling, others are not

TLR and BCR
\protect\hyperlink{ref-Chatzouli2014}{Chatzouli et al.} (\protect\hyperlink{ref-Chatzouli2014}{2014}) show that stimulation of TLR in IGHV-U samples leads to pronounced p-ERK induction, whereas in M-CLL, stimulation of TLR concomitantly with BCR stimulation is required to induce a smaller increase in p-ERK. TLR stimulation without BCR stimulation in IGHV-M samples leads to an induction of caspase-8 and apoptosis, much more so than in IGHV-U samples(\protect\hyperlink{ref-Chatzouli2014}{Chatzouli et al. 2014}).

TLR as a prognostic tool?\textless- does this integrate multiple risk markers?
Reelvance in lymhp node??

Papeer whosing TLR response and survival that i found ages ago

REad up on trisomy 12
\protect\hyperlink{ref-JCIpaper}{Dietrich et al.} (\protect\hyperlink{ref-JCIpaper}{2017}).

Whats knwon about pathogenetic mechanims:
DOI: 10.1200/JCO.2005.02.568 Journal of Clinical Oncology 23, no. 16 (June 01, 2005) 3780-3792.

whole chromososm - is it lots of things - IRAK4, STAT6, gen dosge, SPIB and PU1 activity, BCR signalling?

gene dosage effects are obviosuly quite important here

the power of bioiformatics techqnieus , as well as traditioanl techinuqes, in bioogical discover, to save time and resouce

Read up on SpiB and PU1
SPib an dpU 1 regulate BTK: \url{https://pubmed.ncbi.nlm.nih.gov/8934542/}
SPIB and PU1 activity in B cells: \url{https://pubmed.ncbi.nlm.nih.gov/15936902/}
(hese data indicate that Spi-1 and Spi-B directly regulate the expression of Grap2 and that Grap2 functions to modulate BCR signaling, but that reduced Grap2 expression is unlikely to account for the BCR signaling defects observed in Spi1+/- SpiB-/- B cells.)
PU1 and SPIb and higher in Bc cells- this paper indicates this is to do with SP1 binding (which is on chromosome 12 and is hgiher ) \url{https://pubmed.ncbi.nlm.nih.gov/7566969/}
PU1 and PSIb bdinign in mouse lympohma cell ine \url{https://pubmed.ncbi.nlm.nih.gov/25765478/}
Spib and PU1 in B and T cell development \url{https://pubmed.ncbi.nlm.nih.gov/8691135/}

TLR signalling:
REad (\protect\hyperlink{ref-Chatzouli2014}{Chatzouli et al. 2014})
TLR stimulation can lead to both acitvation of MAPK and increase in viability or activation of apoptosis. If mutatiosn can affect which pathway is active, this poitns to aditoanl mechsnsim by which these mutations contribtue to CLL drive. It also points to TLR response as a useful prognostic marker.

POwer of multivariate modellings;
we need mor ltieracy about these things
discuovery of mulitple layers of biology rather thatn jsut using t tests

IL4 and KRAS - IL4 and ERK apthway reading

\hypertarget{contributions-statement-1}{%
\section{Contributions statement}\label{contributions-statement-1}}

The author of this thesis performed statistical inference and regression modelling outlined in this chapter. In essence, she performed the univariate and multivariate survey of genetic determinants of stimulus response. She identified trisomy 12 as a modulator of microenvironmental response, and conceptualised and performed the computational follow-up work on this finding, with support from Peter-Martin Bruch, Ivan Berest and Tina Becirovic.

This chapter integrates a number of datasets from the following sources. The screening data was generated by Peter-Martin Bruch. The associated patient genetic data, including WES, WGS, CNV profiles, DNA methylation data and RNA was generated and processed by \protect\hyperlink{ref-JCIpaper}{Dietrich et al.} (\protect\hyperlink{ref-JCIpaper}{2017}). The proteomics dataset was generated by \protect\hyperlink{ref-HerbstThesis}{Herbst} (\protect\hyperlink{ref-HerbstThesis}{2020}), and kindly shared for the purpose of this thesis. The larger ATACseq dataset was generated by (\protect\hyperlink{ref-Rendeiro2017}{\textbf{Rendeiro2017?}}), and processed by Ivan Berest. The smaller ATACseq dataset was planned by the author and Peter-Martin Bruch, and generated by Peter-Martin Bruch. The data was processed by the author, with support from Ivan Berest. The Spi-B and PU.1 knockdown data was generated by Tina Bericoviv and published in Bruch \& Giles et al.~2021.

In addition, the author received support on the following analysis work. Junyan Lu provided advice and support with multivariate modelling approaches. Ivan Berest provided support with running diffTF.

used I versus we and credited datasets, and my paper for figures and factrs - main analysis nad key finding, this hsould be routine
Then outline in detail at the bottom

\hypertarget{molecular-and-microenvironmental-modulators-of-drug-response-in-cll}{%
\chapter{Molecular and microenvironmental modulators of drug response in CLL}\label{molecular-and-microenvironmental-modulators-of-drug-response-in-cll}}

So far, we have observed that the panel of stimuli have a range of ects on Cll vibaliyt, some higlhy potent. Form this analysis, IL4 and TLR pathways emerge as kye modualtors of CLL biology, in some genetic backgrounds hguley incresaing CLL viability. We have also observed hos these signals can be modualted by genetic features, underlying the fact that cancer biology is integratively influenec by cll-intrinsc and cell-extrisnic feautres. Trisomy 12 and IGHV status emerge as key modualtors that impact on resposnes to exernal signals.

The microenvironment is known to reduce treatment efficacy within the protective niche {[}REFERENCE Ahn IE, Farooqui MZH, Tian X, et al.~Depth and durability of response to ibrutinib in CLL: 5-year follow-up of a phase 2 study. Blood. 2018;131(21):2357--2366.
Abstract/FREE Full TextGoogle Scholar{]}. Guided by this obesrvation,

In the second part of this chapter, we intergrate all three axes of the dataset, to ivnesrgatied how drug - microenivronment interactions can be further modulated by genetic features.

Outline method

INtor sentence

Tasks for chapter
Sort figures {[}{]}
Write all text
Reread and check story and narrative
Reference
Credit and contirbutiosn statement
Dump in disucssion and intro as I go
Do a intro, discussion and methods dump

Intro text

Process data

\hypertarget{mapping-drug---stimulus-interactions}{%
\section{Mapping drug - stimulus interactions}\label{mapping-drug---stimulus-interactions}}

\hypertarget{drug-stimulus-linear-model}{%
\subsection{Linear modelling quantifies interactions between drugs and stimuli}\label{drug-stimulus-linear-model}}

Principle

Equation \eqref{eq:drugCytInt} quantifies how the viability with any combination can be predicted:
\begin{equation}
            log(V) = \beta_{drug}X_{drug} + \beta_{stimulus}X_{stimulus} + \beta_{int}X_{drug}X_{stimulus} + \epsilon
                                       \label{eq:drugCytInt}
    \end{equation}
\emph{where \(V\) is the viability with a given treatment,} \(\beta_{drug}\), \(\beta_{stimulus}\) \emph{and} \(\beta_{int}\) \emph{are coefficients for the drug, stimulus and combinatorial terms and} \(X_{drug}\) \emph{and} \(X_{stimulus}\) \emph{are indicator variables (0 or 1) for the presence or absence of a drug/stimulus.} \(\epsilon\) \emph{is the vector of model residuals See also Methods section \ref{drug-stimulus-linear-model-method}.}

Different types of interaction\\
(Figure \ref{fig:drugStimulusDummyPLots}).


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/drugStimulusDummyPLots-1} 

}

\caption{Graphical representation of the four drug - stimulus interaction categories. Categories are defined according to (i) the nature of the interaction (synergistic or antagonistic), and (ii) whether \(\beta_{int}\) is positive or negative i.e.~the viability with the combinatorial treatment is higher or lower than would be expected based on additive effects alone. x-axis shows treatment type, y-axis shows log transformed viability values. Red and blue lines depict representative treatment response patterns for given interaction type. To demonstrate the effect of each each interaction on the expected viability, blue and black horizontal lines represent the expected viability for combinatorial treatment based on additive effects alone (blue) and the actual viability accounting for additive effects and the interaction (black). See Methods section \ref{drug-stimulus-linear-model-method}. \emph{Figure and caption from Bruch \& Giles et al.~2021.}}\label{fig:drugStimulusDummyPLots}
\end{figure}
Quantify how many fit into each category\\
(Figure \ref{fig:barPlotInteractions}).


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/barPlotInteractions-1} 

}

\caption{Histogram showing the number of drug-stimulus interactions within each category, where p value for \(\beta_{int}\) is \textless0.05. See Methods section \ref{drug-stimulus-linear-model-method}. \emph{Figure and caption from Bruch \& Giles et al.~2021.}}\label{fig:barPlotInteractions}
\end{figure}
\hypertarget{ID}{%
\subsection{Linear modelling identifies drug - stimulus interactions of potential clinical importance}\label{ID}}

SHow all actual interactions
(Figure \ref{fig:drugStimulusIntMapt}).


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/drugStimulusIntMapt-1} 

}

\caption{Heatmap of all \(\beta_{int}\) values for all drug - stimulus combinations where p for \(\beta_{int}\) \textless0.05, annotated with interaction type. Scale indicates size and sign of \(\beta_{int}\). Rows and columns clustered according to hierarchical clustering. See Methods section \ref{drug-stimulus-linear-model-method}. \emph{Figure and caption from Bruch \& Giles et al.~2021.}}\label{fig:drugStimulusIntMapt}
\end{figure}
Discuss specific examples
(find two replacements for ibr and ifn gamma)
Ibrutinib + IL4 Ibrutinib + IFNgamma, Ralimetinib + IFNgamma, Ibrutinib + Resiquimod - show in grid graphic
big up interferon gamma

(Figure \ref{fig:drugStimulusIntEx}).


\begin{figure}

{\centering \includegraphics[width=0.8\linewidth]{thesis_files/figure-latex/drugStimulusIntEx-1} 

}

\caption{Examples of drug-stimulus interactions, for each category. Plots show log transformed viability values with each treatment, for all samples. Matching samples are linked across treatments. Black and blue horizontal lines indicate the predicted viability from the linear model for each single and combinatorial treatment. In combinatorial treatment, both the expected viability based on the additive effect of drug and stimulus (blue), and the viability with interaction (black) are shown, to indicate the impact of the interaction. (A+B) Ibrutinib, a clinically-used BTK inhibitor, is blocked by IL4 and IFN\(\gamma\). (C) The JAK inhibitor pyridone-6 inhibits the pro-survival effect of sCD40L + IL4 stimulation. (D) The p38 inhibitor ralimetinib and IFN\(\gamma\) show a synergistic pro-survival effect not seen in either single treatment. (E) TLR agonists, including CpG ODN (shown) increase sensitivity to BTK inhibition by ibrutinib, despite increasing viability in isolation. (F) Soluble anti-IgM sensitises CLL samples to HSP90 inhibition by luminespib. See Methods section \ref{drug-stimulus-linear-model-method}. \emph{Figure and caption from Bruch \& Giles et al.~2021.}}\label{fig:drugStimulusIntEx}
\end{figure}
\hypertarget{IL4resistance}{%
\subsection{IL4 induces resistance to BCR inhibitors and chemotherapeutics}\label{IL4resistance}}

(Figure \ref{fig:IL4Res}).


\begin{figure}

{\centering \includegraphics[width=0.8\linewidth]{thesis_files/figure-latex/IL4Res-1} 

}

\caption{Examples of IL4-induced resistance. Plots show log transformed viability values with each treatment, for all samples. Matching samples are linked across treatments. Black and blue horizontal lines indicate the predicted viability from the linear model for each single and combinatorial treatment. In combinatorial treatment, both the expected viability based on the additive effect of drug and stimulus (blue), and the viability with interaction (black) are shown, to indicate the impact of the interaction. Plots show IL4 + (A) Idelalisib (Pi3K), (B) PRT062607 (SYK) (C) Everolimus (mTOR), and (D) Selumetinib (p38 MAPK). See Methods section \ref{drug-stimulus-linear-model-method}. \emph{Figure from Bruch \& Giles et al.~2021.}}\label{fig:IL4Res}
\end{figure}
\hypertarget{genetic-modulators-of-drug-response}{%
\section{Genetic modulators of drug response}\label{genetic-modulators-of-drug-response}}

\hypertarget{univariate-gene-drug-associations}{%
\subsection{Univariate analysis identifies genetic modualtors of drug response}\label{univariate-gene-drug-associations}}

(Figure \ref{fig:drugGeneIntMap}).


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/drugGeneIntMap-1} 

}

\caption{Plot showing BH-adjusted p values from Student's t-tests (two-sided, with equal variance), for all tested drug-gene associations. Tests performed for IGHV status and somatic mutations and copy number aberrations with ≥3 patient samples in each group (n=54). Each circle represents a gene-stimulus association. Associations that meet 10\% FDR cut off are indicated in colour, where the colour denotes the genetic feature. Method \ref{univariate-gene-stimulus-associations-method} \emph{Figure from Bruch \& Giles et al.~2021.}}\label{fig:drugGeneIntMap}
\end{figure}
\hypertarget{multivariate-gene-drug-assosciations}{%
\subsection{Multivariate analysis identifies genetic modulators of drug response}\label{multivariate-gene-drug-assosciations}}

(Figure \ref{fig:drugGeneIntMulti}).


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/drugGeneIntMulti-1} 

}

\caption{Example predictor profile for IBET-762, depicting drug - gene associations from multivariate modelling. Bar plots (left) show size and sign of assigned coefficients from Gaussian linear modelling. A positive coefficient indicates that drug has lower efficacy when feature is present. Scatter plots (bottom) and corresponding heatmaps above show how presence of selected genetic feature relates to sample viabilities. Scatter plots show ranked log(viability) values for each sample and heatmaps show mutation status for each predictor, for corresponding sample in scatter plot. See Methods section \ref{multivariate-gene-stimulus-assosciations-method}. \emph{Figure from Bruch \& Giles et al.~2021.}}\label{fig:drugGeneIntMulti}
\end{figure}
\hypertarget{specific-examples}{%
\subsection{Specific examples}\label{specific-examples}}

(Figure \ref{fig:drugGeneIntEx}).


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/drugGeneIntEx-1} 

}

\caption{Beeswarm-boxplots showing control-normalised log transformed viability values after treatment with (A) IBET-762, (B) Nutlin-3a and (C) Ralimetinib, stratified by trisomy 12. \emph{Figure from Bruch \& Giles et al.~2021.}}\label{fig:drugGeneIntEx}
\end{figure}
in a grid:
IBET-762 and trisomy 12
Nutlin-3a and trisomy 12 and gene dosage effect of MDM2 in trisomy12 CLL
Correlation of IBET-762 and Nutlin-3a
Ralimetinib and trisomy 12

\hypertarget{drug-stimulus-gene-interactions}{%
\section{The modulatory effect of mutations on drug - stimuli interactions}\label{drug-stimulus-gene-interactions}}

\hypertarget{fitting-patient-specific-linear-models-identfies-drug---stimulus---gene-interactions}{%
\subsection{Fitting patient-specific linear models identfies drug - stimulus - gene interactions}\label{fitting-patient-specific-linear-models-identfies-drug---stimulus---gene-interactions}}

(Figure \ref{fig:drugCytGeneIntMap}).

Also Appendix figure \ref{fig:drugcytGeneIntAll}

Have to make sure this figure takes a whol page but doesnt float to the bottom


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/drugCytGeneIntMap-1} 

}

\caption{Heatmap depicting overview of genetic predictors of drug - stimulus interactions (each row represents the coefficients from fitting a single multivariate model, as in Figure \ref{fig:FluCpGExA}). Stimuli are shown on left, and corresponding drugs on right. Coloured fields indicate that \(\beta{int}\) for given drug and stimulus is modulated by corresponding genetic feature. Positive coefficients are shown in red, indicating a more positive \(\beta{int}\) if the feature is present. Only top 8 most commonly selected genetic features are shown. Drug - stimulus combinations with no genetic predictors of \(\beta{int}\) amongst top 8 shown are omitted for clarity. See Methods section \ref{drug-stimulus-gene-interactions-method}. \emph{Figure from Bruch \& Giles et al.~2021.}}\label{fig:drugCytGeneIntMap}
\end{figure}
\hypertarget{patient---specific-drug---stimulus-interactions-may-have-clinical-significance}{%
\subsection{Patient - specific drug - stimulus interactions may have clinical significance}\label{patient---specific-drug---stimulus-interactions-may-have-clinical-significance}}

(Figure \ref{fig:FluCpGExA}).



(Figure \ref{fig:FluCpGExB}).


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/FluCpGExA-1} 

}

\caption{Predictor profile depicting genetic features that modulate the size of \(\beta{int}\) between fludarabine and CpG ODN. To generate predictor profile, linear model in Eqn. \eqref{eq:drugCytInt} was fitted in a sample - specific manner, to calculate drug-stimulus interaction coefficients (\(\beta{int}\)) for each patient sample. Ranked patient-specific \(\beta{int}\) values are shown in lower scatter plot. Associations between the size of \(\beta{int}\) and genetic features were identified using multivariate regression with L1 (lasso) regularisation, with gene mutations (p = 39) and IGHV status as predictors, and selecting coefficients that were chosen in \textgreater90\% of bootstrapped model fits. The horizontal bars on left show the size of fitted coefficients assigned to genetic features. Matrix above scatter plot indicates patient mutation status for the selected genetic features. Matrix fields correspond to points in scatter plot (ie patient data is aligned), to indicate how the size of \(\beta{int}\) varies with selected genetic feature. Grey lines indicate presence of genetic feature/IGHV mutated. See Methods section \ref{drug-stimulus-gene-interactions-method}. \emph{Figure and caption from Bruch \& Giles et al.~2021.}}\label{fig:FluCpGExA}
\end{figure}
\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/FluCpGExB-1} 

}

\caption{Beeswarm boxplots of log(viability) values for fludarabine and CpG ODN single and combinatorial treatments, faceted by IGHV status and trisomy 12 status. P-values from paired Student's t-tests. \emph{Figure and caption from Bruch \& Giles et al.~2021.}}\label{fig:FluCpGExB}
\end{figure}
(Figure \ref{fig:ibrIl4ExA}).


(Figure \ref{fig:ibrIl4ExB}).
(ref:fig:ibrIl4ExB) Beeswarm boxplots of log(viability) values for ibrutinib and IL4 single and combinatorial treatments, faceted by IGHV status and trisomy 12 status. P-values from paired Student's t-tests. \emph{Figure and caption from Bruch \& Giles et al.~2021.}
\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/ibrIl4ExA-1} 

}

\caption{Predictor profile depicting genetic features that modulate the size of \(\beta{int}\) between ibrutinib (BTK) and IL4. Plot generated as in Figure \ref{fig:FluCpGExA}. See also Methods section \ref{drug-stimulus-gene-interactions-method}. \emph{Figure from Bruch \& Giles et al.~2021.}}\label{fig:ibrIl4ExA}
\end{figure}
\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/fig:ibrIl4ExB-1} 

}

\caption{(ref:fig:ibrIl4ExB)}(\#fig:fig:ibrIl4ExB)
\end{figure}
(Figure \ref{fig:nutlinCpgEx}).

\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/nutlinCpgEx-1} 

}

\caption{Predictor profile depicting genetic features that modulate the size of \(\beta{int}\) between Nutlin-3a (MDM2) and CpG ODN. Plot generated as in Figure \ref{fig:FluCpGExA}. See also Methods section \ref{drug-stimulus-gene-interactions-method}.}\label{fig:nutlinCpgEx}
\end{figure}
(Figure \ref{fig:ralIl4ExA}).

\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/ralIl4ExA-1} 

}

\caption{Predictor profile depicting genetic features that modulate the size of \(\beta{int}\) between Ralimetinib (p38 MAPK) and IL4. Plot generated as in Figure \ref{fig:FluCpGExA}. See also Methods section \ref{drug-stimulus-gene-interactions-method}.}\label{fig:ralIl4ExA}
\end{figure}
(Figure \ref{fig:ralIl4ExB}).


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/ralIl4ExB-1} 

}

\caption{Beeswarm boxplots showing log-transformed viabilty values after treatment with IL4 and Ralimetinib + IL4, faceted by trisomy 12 status. P values from paired Student's t-tests.}\label{fig:ralIl4ExB}
\end{figure}
Responses to chemotherapeutic drugs in the context of TLR stimulation are modulated by del11q and ATM mutations,
Trisomy12 enhances pro-survival synergy between Ralimetinib + IFNgamma
Ralimetinib treatment increases the pro-survival effect of IL4, and this is further enhanced in NOTCH1 and Ras/Raf mutants
Fludarabine and TLR
Ibrutinib and IL4
Discuss examples in drug- cyt -gene examples script

\hypertarget{using-stimuli-responses-in-combination-with-genetic-profiles-to-predict-in-vivo-drug-response}{%
\section{Using stimuli responses in combination with genetic profiles to predict in vivo drug response}\label{using-stimuli-responses-in-combination-with-genetic-profiles-to-predict-in-vivo-drug-response}}

Can stimuli + genes predict drug response - this would be useful in the clinic. Especially if its just TLR/IL4 response, are specific signals useful?

(Figure \ref{fig:drugCytGeneIntClinic}).



\hypertarget{summary-2}{%
\section{Summary}\label{summary-2}}

\hypertarget{discussion-2}{%
\section{Discussion}\label{discussion-2}}

\hypertarget{contirbutions-statement}{%
\section{Contirbutions Statement}\label{contirbutions-statement}}

\hypertarget{results}{%
\chapter{Results}\label{results}}
\begin{itemize}
\item
  Organize material and present results.
\item
  Use tables, figures (but prefer visual presentation):
  \begin{itemize}
  \item
    Tables and figures should supplement (and not duplicate) the text.
  \item
    Tables and figures should be provided with legends.
  \item
    \emph{Figure \ref{fig:graph} shows how to include and reference graphics.
    The graphic must be labelled before. Files must be in \textbf{.eps} format. You
    can do this really easily in R Markdown with \texttt{knitr::include\_graphics()}}!
  \item
    Figures can be referenced with \texttt{\textbackslash{}@ref(fig:\textless{}name\textgreater{})}, where \texttt{\textless{}name\textgreater{}} is the
    name of the code chunk.
  \end{itemize}
\end{itemize}
\begin{figure}

{\centering \includegraphics[width=0.5\linewidth]{figures/graph} 

}

\caption{Estimated residuals from model XXX4. ...}\label{fig:graph4}
\end{figure}
\begin{itemize}
\item
  Tables and graphics may appear in the text or in the appendix, especially if
  there are many simulation results tabulated, but is also depends on the study
  and number of tables resp. figures. The key graphs and tables must appear in
  the text!
\item
  R Markdown can also supports math equations just like \emph{LaTeX}!
  \begin{itemize}
  \item
    \emph{Equation \eqref{eq:SpecDens} represents the ACs of a stationary
    stochastic process:}
    \begin{equation}
            f_y(\lambda) = (2\pi)^{-1} \sum_{j=-\infty}^{\infty}
                           \gamma_j e^{-i\lambda j}
                         =(2\pi)^{-1}\left(\gamma_0 + 2 \sum_{j=1}^{\infty}
        \gamma_j \cos(\lambda j)\right)
                                       \label{eq:SpecDens}
    \end{equation}
    \emph{where \(i=\sqrt{-1}\) is the imaginary unit, \(\lambda \in [-\pi, \pi]\) is the
    frequency and the \(\gamma_j\) are the autocovariances of \(y_t\).}
  \item
    Equations can be referenced with \texttt{\textbackslash{}@ref(eq:\textless{}name\textgreater{})}, where name is defined
    by adding \texttt{(\textbackslash{}\#eq:\textless{}name\textgreater{})} in the line immediately before \texttt{\textbackslash{}end\{equation\}}.
  \end{itemize}
\end{itemize}
\hypertarget{review-of-results}{%
\section{Review of Results}\label{review-of-results}}
\begin{itemize}
\item
  Do the results support or do they contradict economic theory ?
\item
  What does the reader learn from the results?
\item
  Try to give an intuition for your results.
\item
  Provide robustness checks.
\item
  Compare to previous research.
\end{itemize}
\hypertarget{discussion-3}{%
\chapter{Discussion}\label{discussion-3}}

\newpage

\hypertarget{references}{%
\chapter*{References}\label{references}}
\addcontentsline{toc}{chapter}{References}

\noindent

\setlength{\parindent}{-0.5cm}
\setlength{\leftskip}{0.5cm}
\setlength{\parskip}{8pt}

\hypertarget{refs}{}
\begin{CSLReferences}{1}{0}
\leavevmode\vadjust pre{\hypertarget{ref-Abruzzo2018}{}}%
Abruzzo, Lynne V., Carmen D. Herling, George A. Calin, Christopher Oakes, Lynn L. Barron, Haley E. Banks, Vikram Katju, Michael J. Keating, and Kevin R. Coombes. 2018. {``{Trisomy 12 chronic lymphocytic leukemia expresses a unique set of activated and targetable pathways}.''} \emph{Haematologica} 103 (12): 2069--78. \url{https://doi.org/10.3324/haematol.2018.190132}.

\leavevmode\vadjust pre{\hypertarget{ref-R-genomation}{}}%
Akalin, Altuna, Vedran Franke, Katarzyna Wreczycka, Alexander Gosdschan, Liz Ing-Simmons, and Bozena Mika-Gospodorz. 2021. \emph{Genomation: Summary, Annotation and Visualization of Genomic Data}. \url{http://bioinformatics.mdc-berlin.de/genomation/}.

\leavevmode\vadjust pre{\hypertarget{ref-Berest2019}{}}%
Berest, Ivan, Christian Arnold, Armando Reyes-Palomares, Giovanni Palla, Kasper Dindler Rasmussen, Holly Amelia Rebecca Giles, Peter-Martin Bruch, et al. 2019. {``{Quantification of Differential Transcription Factor Activity and Multiomics-Based Classification into Activators and Repressors: diffTF}.''} \emph{Cell Reports} 29 (10): 3147--3159.e12. \url{https://doi.org/10.1016/J.CELREP.2019.10.106}.

\leavevmode\vadjust pre{\hypertarget{ref-Bosch2019}{}}%
Bosch, Francesc, and Riccardo Dalla-Favera. 2019. {``{Chronic lymphocytic leukaemia: from genetics to treatment}.''} \emph{Nature Reviews Clinical Oncology} 16 (11): 684--701. \url{https://doi.org/10.1038/s41571-019-0239-8}.

\leavevmode\vadjust pre{\hypertarget{ref-Care2014}{}}%
Care, Matthew A., Mario Cocco, Jon P. Laye, Nicholas Barnes, Yuanxue Huang, Ming Wang, Sharon Barrans, et al. 2014. {``{SPIB and BATF provide alternate determinants of IRF4 occupancy in diffuse large B-cell lymphoma linked to disease heterogeneity}.''} \emph{Nucleic Acids Research} 42 (12): 7591--7610. \url{https://doi.org/10.1093/nar/gku451}.

\leavevmode\vadjust pre{\hypertarget{ref-Chatzouli2014}{}}%
Chatzouli, Maria, Stavroula Ntoufa, Nikos Papakonstantinou, Elisavet Chartomatsidou, Achilles Anagnostopoulos, Panagoula Kollia, Paolo Ghia, Marta Muzio, Kostas Stamatopoulos, and Chrysoula Belessi. 2014. {``{Heterogeneous Functional Effects of Concomitant B Cell Receptor and TLR Stimulation in Chronic Lymphocytic Leukemia with Mutated versus Unmutated Ig Genes}.''} \emph{The Journal of Immunology} 192 (10): 4518--24. \url{https://doi.org/10.4049/jimmunol.1302102}.

\leavevmode\vadjust pre{\hypertarget{ref-Collins1989}{}}%
Collins, Russell J., Louise A. Verschuer, Brian V. Harmon, Roger L. Prentice, John H. Pope, and John F. R. Kerr. 1989. {``{Spontaneous programmed death (apoptosis) of B‐chronic lymphocytic leukaemia cells following their culture in vitro}.''} \emph{British Journal of Haematology} 71 (3): 343--50. \url{https://doi.org/10.1111/j.1365-2141.1989.tb04290.x}.

\leavevmode\vadjust pre{\hypertarget{ref-JCIpaper}{}}%
Dietrich, Sascha, Małgorzata Oleś, Junyan Lu, Leopold Sellner, Simon Anders, Britta Velten, Bian Wu, et al. 2017. {``{Drug-perturbation-based stratification of blood cancer}.''} \emph{Journal of Clinical Investigation} 128 (1): 427--45. \url{https://doi.org/10.1172/JCI93801}.

\leavevmode\vadjust pre{\hypertarget{ref-Dohner2000}{}}%
Döhner, H, Stephan Stilgenbauer, Axel Benner, Elke Leupolt, Alexander Kröber, Lars Bullinger, Konstanze Döhner, Martin Bentz, and Peter Lichter. 2000. {``{Genomic aberrations and survival in chronic lymphocytic leukemia.}''} \emph{The New England Journal of Medicine} 343 (26): 1910--16. \url{https://doi.org/10.1056/NEJM200012283432602}.

\leavevmode\vadjust pre{\hypertarget{ref-R-msigdbr}{}}%
Dolgalev, Igor. 2021. \emph{Msigdbr: MSigDB Gene Sets for Multiple Organisms in a Tidy Data Format}. \url{https://igordot.github.io/msigdbr/}.

\leavevmode\vadjust pre{\hypertarget{ref-R-glmnet}{}}%
Friedman, Jerome, Trevor Hastie, Rob Tibshirani, Balasubramanian Narasimhan, Kenneth Tay, and Noah Simon. 2021. \emph{Glmnet: Lasso and Elastic-Net Regularized Generalized Linear Models}. \url{https://CRAN.R-project.org/package=glmnet}.

\leavevmode\vadjust pre{\hypertarget{ref-HerbstThesis}{}}%
Herbst, Sophie. 2020. {``{Systematic analysis of cell-intrinsic and extrinsic factors in chronic lymphocytic leukemia to understand functional consequences for drug response and clinical outcome}.''} PhD diss., University of Heidelberg. \url{https://archiv.ub.uni-heidelberg.de/volltextserver/27438/1/Doktorarbeit.pdf}.

\leavevmode\vadjust pre{\hypertarget{ref-R-survminer}{}}%
Kassambara, Alboukadel, Marcin Kosinski, and Przemyslaw Biecek. 2021. \emph{Survminer: Drawing Survival Curves Using Ggplot2}. \url{https://rpkgs.datanovia.com/survminer/index.html}.

\leavevmode\vadjust pre{\hypertarget{ref-Kienle2005}{}}%
Kienle, Dirk L, Christian Korz, Beate Hosch, Axel Benner, Daniel Mertens, Annett Habermann, Alexander Kröber, et al. 2005. {``{Evidence for distinct pathomechanisms in genetic subgroups of chronic lymphocytic leukemia revealed by quantitative expression analysis of cell cycle, activation, and apoptosis-associated genes.}''} \emph{Journal of Clinical Oncology : Official Journal of the American Society of Clinical Oncology} 23 (16): 3780--92. \url{https://doi.org/10.1200/JCO.2005.02.568}.

\leavevmode\vadjust pre{\hypertarget{ref-Kipps2017}{}}%
Kipps, Thomas J, Freda K Stevenson, Catherine J Wu, Carlo M Croce, Graham Packham, William G Wierda, Susan O'Brien, John Gribben, and Kanti Rai. 2017. {``{Chronic lymphocytic leukaemia.}''} \emph{Nature Reviews. Disease Primers} 3: 16096. \url{https://doi.org/10.1038/nrdp.2016.96}.

\leavevmode\vadjust pre{\hypertarget{ref-R-pheatmap}{}}%
Kolde, Raivo. 2019. \emph{Pheatmap: Pretty Heatmaps}. \url{https://CRAN.R-project.org/package=pheatmap}.

\leavevmode\vadjust pre{\hypertarget{ref-HOCOMOCO}{}}%
Kulakovskiy, Ivan V, Ilya E Vorontsov, Ivan S Yevshin, Anastasiia V Soboleva, Artem S Kasianov, Haitham Ashoor, Wail Ba-Alawi, et al. 2016. {``{HOCOMOCO: expansion and enhancement of the collection of transcription factor binding sites models.}''} \emph{Nucleic Acids Research} 44 (D1): D116--25. \url{https://doi.org/10.1093/nar/gkv1249}.

\leavevmode\vadjust pre{\hypertarget{ref-R-DESeq2}{}}%
Love, Michael, Simon Anders, and Wolfgang Huber. 2021. \emph{DESeq2: Differential Gene Expression Analysis Based on the Negative Binomial Distribution}. \url{https://github.com/mikelove/DESeq2}.

\leavevmode\vadjust pre{\hypertarget{ref-R-BloodCancerMultiOmics2017}{}}%
Oles, Malgorzata, Sascha Dietrich, Junyan Lu, Britta Velten, Andreas Mock, Vladislav Kim, and Wolfgang Huber. 2021. \emph{BloodCancerMultiOmics2017: "Drug-Perturbation-Based Stratification of Blood Cancer" by Dietrich s, Oles m, Lu j Et Al. - Experimental Data and Complete Analysis}.

\leavevmode\vadjust pre{\hypertarget{ref-Rendeiro2016}{}}%
Rendeiro, André F., Christian Schmidl, Jonathan C. Strefford, Renata Walewska, Zadie Davis, Matthias Farlik, David Oscier, and Christoph Bock. 2016. {``{Chromatin accessibility maps of chronic lymphocytic leukaemia identify subtype-specific epigenome signatures and transcription regulatory networks}.''} \emph{Nature Communications} 7 (1): 11938. \url{https://doi.org/10.1038/ncomms11938}.

\leavevmode\vadjust pre{\hypertarget{ref-R-survival}{}}%
Therneau, Terry M. 2021. \emph{Survival: Survival Analysis}. \url{https://github.com/therneau/survival}.

\leavevmode\vadjust pre{\hypertarget{ref-Turkistany2011}{}}%
Turkistany, Shereen A, and Rodney P Dekoter. 2011. {``{The transcription factor PU.1 is a critical regulator of cellular communication in the immune system}.''} \url{https://doi.org/10.1007/s00005-011-0147-9}.

\leavevmode\vadjust pre{\hypertarget{ref-ConsensusClusterPlus}{}}%
Wilkerson, Matthew D, and D Neil Hayes. 2010. {``{ConsensusClusterPlus: a class discovery tool with confidence assessments and item tracking.}''} \emph{Bioinformatics (Oxford, England)} 26 (12): 1572--73. \url{https://doi.org/10.1093/bioinformatics/btq170}.

\leavevmode\vadjust pre{\hypertarget{ref-R-ConsensusClusterPlus}{}}%
Wilkerson, Matt, and Peter Waltman. 2021. \emph{ConsensusClusterPlus: ConsensusClusterPlus}.

\leavevmode\vadjust pre{\hypertarget{ref-Willis2017}{}}%
Willis, Simon N., Julie Tellier, Yang Liao, Stephanie Trezise, Amanda Light, Kristy O'Donnell, Lee Ann Garrett-Sinha, Wei Shi, David M. Tarlinton, and Stephen L. Nutt. 2017. {``{Environmental sensing by mature B cells is controlled by the transcription factors PU.1 and SpiB}.''} \emph{Nature Communications} 8 (1): 1426. \url{https://doi.org/10.1038/s41467-017-01605-1}.

\leavevmode\vadjust pre{\hypertarget{ref-R-ChIPseeker}{}}%
Yu, Guangchuang. 2021a. \emph{ChIPseeker: ChIPseeker for ChIP Peak Annotation, Comparison, and Visualization}. \url{https://guangchuangyu.github.io/software/ChIPseeker}.

\leavevmode\vadjust pre{\hypertarget{ref-R-clusterProfiler}{}}%
---------. 2021b. \emph{clusterProfiler: A Universal Enrichment Tool for Interpreting Omics Data}. \url{https://yulab-smu.top/biomedical-knowledge-mining-book/}.

\leavevmode\vadjust pre{\hypertarget{ref-Zenz2010}{}}%
Zenz, Thorsten, Daniel Mertens, Ralf Küppers, Hartmut Döhner, and Stephan Stilgenbauer. 2010. {``{From pathogenesis to treatment of chronic lymphocytic leukaemia}.''} \emph{Nature Reviews Cancer} 10 (1): 37--50. \url{https://doi.org/10.1038/nrc2764}.

\end{CSLReferences}
\indent
\setlength{\parindent}{17pt}
\setlength{\leftskip}{0pt}
\setlength{\parskip}{0pt}

\newpage

\hypertarget{appendix}{%
\chapter*{Appendix}\label{appendix}}
\addcontentsline{toc}{chapter}{Appendix}

\hypertarget{figures}{%
\section*{Figures}\label{figures}}
\addcontentsline{toc}{section}{Figures}


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{figures/graph} 

}

\caption{Analysis of ATACseq dataset of two trisomy 12 and two non-trisomy 12 untreated CLL PMBC samples. The volcano plot depicts change in TF activity (x axis) versus BH-adjusted p-values (y axis) for 636 TFs, comparing trisomy 12 and non-trisomy 12 samples. The (\protect\hyperlink{ref-diffTF}{\textbf{diffTF?}}){[}REFERENCE{]} package was ran in analytical mode to calculate TF activity, measured as weighted mean difference. TFs are labeled if adjusted p-value \textless{} 0.01 and absolute weighted mean difference \textgreater{} 0.15.}\label{fig:diffTFsmallvolPlot}
\end{figure}
(ref:drugcytGeneIntAll)
\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/drugcytGeneIntAll-1} 

}

\caption{(ref:drugcytGeneIntAll)}\label{fig:drugcytGeneIntAll}
\end{figure}
\hypertarget{tables}{%
\section*{Tables}\label{tables}}
\addcontentsline{toc}{section}{Tables}


\begin{table}
\centering
\begin{tabular}{l|l|l|l}
\hline
Cytokine & Receptor & ENSEMBLID & Kegg\\
\hline
\cellcolor[HTML]{E2E868}{SDF-1a} & \cellcolor[HTML]{E2E868}{CXCR 4} & \cellcolor[HTML]{E2E868}{ENSG00000121966} & \cellcolor[HTML]{E2E868}{7852}\\
\hline
SDF-1a & CXCR 7 & ENSG00000144476 & 57007\\
\hline
\cellcolor[HTML]{E2E868}{IL-6} & \cellcolor[HTML]{E2E868}{IL-6 R} & \cellcolor[HTML]{E2E868}{ENSG00000160712} & \cellcolor[HTML]{E2E868}{3570}\\
\hline
IL-6 & IL-6 ST & ENSG00000134352 & 3572\\
\hline
\cellcolor[HTML]{E2E868}{IL-15} & \cellcolor[HTML]{E2E868}{IL-15 R alpha} & \cellcolor[HTML]{E2E868}{ENSG00000134470} & \cellcolor[HTML]{E2E868}{3601}\\
\hline
IL-15 & IL-2 R beta & ENSG00000100385 & 3560\\
\hline
\cellcolor[HTML]{E2E868}{IL-15} & \cellcolor[HTML]{E2E868}{IL-2 R gamma} & \cellcolor[HTML]{E2E868}{ENSG00000147168} & \cellcolor[HTML]{E2E868}{3561}\\
\hline
IL-21 & IL-21 R & ENSG00000103522 & 50615\\
\hline
\cellcolor[HTML]{E2E868}{IL-21} & \cellcolor[HTML]{E2E868}{IL-21 R gamma} & \cellcolor[HTML]{E2E868}{ENSG00000147168} & \cellcolor[HTML]{E2E868}{3561}\\
\hline
IL-2 & IL-2 R alpha & ENSG00000134460 & 3559\\
\hline
\cellcolor[HTML]{E2E868}{IL-2} & \cellcolor[HTML]{E2E868}{IL-2 R beta} & \cellcolor[HTML]{E2E868}{ENSG00000100385} & \cellcolor[HTML]{E2E868}{3560}\\
\hline
IL-2 & IL-2 R gamma & ENSG00000147168 & 3561\\
\hline
\cellcolor[HTML]{E2E868}{IL-10} & \cellcolor[HTML]{E2E868}{IL-10 R alpha} & \cellcolor[HTML]{E2E868}{ENSG00000110324} & \cellcolor[HTML]{E2E868}{3587}\\
\hline
IL-10 & IL-10 R beta & ENSG00000243646 & 3588\\
\hline
\cellcolor[HTML]{E2E868}{IL-4} & \cellcolor[HTML]{E2E868}{IL-4 R} & \cellcolor[HTML]{E2E868}{ENSG00000077238} & \cellcolor[HTML]{E2E868}{3566}\\
\hline
IL-4 & IL-2 R gamma & ENSG00000147168 & 3561\\
\hline
\cellcolor[HTML]{E2E868}{IL-4} & \cellcolor[HTML]{E2E868}{IL-13 R A1} & \cellcolor[HTML]{E2E868}{ENSG00000131724} & \cellcolor[HTML]{E2E868}{3597}\\
\hline
soluble anti-IgM & B cell receptor & ENSG00000275063 & 102723407\\
\hline
\cellcolor[HTML]{E2E868}{soluble anti-IgM} & \cellcolor[HTML]{E2E868}{CD79A} & \cellcolor[HTML]{E2E868}{ENSG00000105369} & \cellcolor[HTML]{E2E868}{973}\\
\hline
soluble anti-IgM & CD79B & ENSG00000007312 & 974\\
\hline
\cellcolor[HTML]{E2E868}{Resiquimod} & \cellcolor[HTML]{E2E868}{TLR 7} & \cellcolor[HTML]{E2E868}{ENSG00000196664} & \cellcolor[HTML]{E2E868}{51284 }\\
\hline
Resiquimod & TLR 8 & ENSG00000101916 & 51311\\
\hline
\cellcolor[HTML]{E2E868}{CpG ODN} & \cellcolor[HTML]{E2E868}{TLR9} & \cellcolor[HTML]{E2E868}{ENSG00000239732} & \cellcolor[HTML]{E2E868}{54106}\\
\hline
BAFF & BAFF R & ENSG00000159958 & 115650\\
\hline
\cellcolor[HTML]{E2E868}{BAFF} & \cellcolor[HTML]{E2E868}{TACI} & \cellcolor[HTML]{E2E868}{ENSG00000240505} & \cellcolor[HTML]{E2E868}{23495}\\
\hline
BAFF & BCMA & ENSG00000048462 & 608\\
\hline
\cellcolor[HTML]{E2E868}{IL-1b} & \cellcolor[HTML]{E2E868}{IL-1 R1} & \cellcolor[HTML]{E2E868}{ENSG00000115594} & \cellcolor[HTML]{E2E868}{3554}\\
\hline
IL-1b & IL-1 R2 & ENSG00000115590 & 7850\\
\hline
\cellcolor[HTML]{E2E868}{sCD40L} & \cellcolor[HTML]{E2E868}{CD40} & \cellcolor[HTML]{E2E868}{ENSG00000101017} & \cellcolor[HTML]{E2E868}{958}\\
\hline
Interferon gamma & IFN-gamma R1 & ENSG00000027697 & 3459\\
\hline
\cellcolor[HTML]{E2E868}{Interferon gamma} & \cellcolor[HTML]{E2E868}{IFN-gamma R2} & \cellcolor[HTML]{E2E868}{ENSG00000159128} & \cellcolor[HTML]{E2E868}{3460}\\
\hline
TGF-b1 & TGF-beta R1 & ENSG00000106799 & 7046\\
\hline
\cellcolor[HTML]{E2E868}{TGF-b1} & \cellcolor[HTML]{E2E868}{TGF-beta R2} & \cellcolor[HTML]{E2E868}{ENSG00000163513} & \cellcolor[HTML]{E2E868}{7048}\\
\hline
\end{tabular}
\end{table}
\begin{table}[ht]
    \begin{center}
        {\footnotesize
        \begin{tabular}{l|cccccccccc}
        \hline \hline
                        & 3m    & 6m    & 1yr   & 2yr   & 3yr   & 5yr   & 7yr   & 10yr  & 12yr  & 15yr   \\
            \hline
                Mean   & 3.138 & 3.191 & 3.307 & 3.544 & 3.756 & 4.093 & 4.354 & 4.621 & 4.741 & 4.878  \\
                Median & 3.013 & 3.109 & 3.228 & 3.490 & 3.680 & 3.906 & 4.117 & 4.420 & 4.575 & 4.759  \\
                Min    & 1.984 & 1.950 & 1.956 & 2.010 & 2.240 & 2.615 & 2.850 & 3.120 & 3.250 & 3.395  \\
                Max    & 5.211 & 5.274 & 5.415 & 5.583 & 5.698 & 5.805 & 5.900 & 6.031 & 6.150 & 6.295  \\
                StD    & 0.915 & 0.919 & 0.935 & 0.910 & 0.876 & 0.825 & 0.803 & 0.776 & 0.768 & 0.762  \\
            \hline \hline
        \end{tabular}}
    \end{center}
    \caption{Detailed descriptive statistics of location and dispersion for
    2100 observed swap rates for the period from
    February 15, 1999 to March 2, 2007. Swap rates measured as 3.12 (instead of 0.0312).}
    \label{tab:apptable}
\end{table}
\newpage

\end{document}
