\documentclass[11pt, a4paper, twosided]{book}
\usepackage[hmarginratio=1:1]{geometry} % equal right and left raitios
\usepackage{amsmath,amssymb,amsfonts,amsthm}    % Typical maths resource packages
\usepackage{graphicx}                           % Packages to allow inclusion of graphics
\usepackage[authoryear]{natbib}                 % literature reference style
\usepackage[bf]{caption}
\usepackage{textcomp}                           % For single quotes
\usepackage{floatrow}                           % For image and table position
\usepackage{booktabs}                           % For tables
% \usepackage[colorlinks=true]{hyperref}
% \usepackage[bottom]{footmisc}
\usepackage[bottom, flushmargin]{footmisc}                   % For footnotes
\usepackage[citebordercolor={0 1 0}]{hyperref}  % For creating hyperlinks in cross references
\usepackage{footnotebackref}
\usepackage[utf8]{inputenc}
\usepackage[T1]{fontenc}
% i have added the below, so that kable works
\usepackage{colortbl}
\usepackage{longtable}
\usepackage{array}
\usepackage{multirow}
\usepackage{wrapfig}
\usepackage{float}
\usepackage{pdflscape}
\usepackage{tabu}
\usepackage{threeparttable}
\usepackage{threeparttablex}
\usepackage[normalem]{ulem}
\usepackage{makecell}
\usepackage{xcolor}

       % For being able to set `echo = TRUE` in code chunk options, and thus in-line representation of code

%--------------------------------------------
% define line spacing = 1.5
\renewcommand{\baselinestretch}{1.5}

% define position of graphics
\floatsetup[figure]{capposition=bottom}
\floatsetup[table]{capposition=top}
\floatplacement{figure}{ht}
\floatplacement{table}{ht}

% define CSL references (required)
\newlength{\cslhangindent}
\setlength{\cslhangindent}{1.5em}
\newenvironment{CSLReferences}%
  {}%
  {\par}

% save thesis parameters for later
\newcommand{\thesistype}{Dissertation}
\newcommand{\thesisauthor}{Holly Amelia Rebecca Giles}
\newcommand{\thesisdate}{Oct 01, 2021?}

% Added by CII (Thanks, Hadley!)
% Use ref for internal links
\renewcommand{\hyperref}[2][???]{\autoref{#1}}
\def\chapterautorefname{Chapter}
\def\sectionautorefname{Section}
\def\subsectionautorefname{Subsection}
% End of CII addition

% Added by CII
\usepackage{caption}
\captionsetup{width=5in}
% End of CII addition

% Added by CII
%%% Copied from knitr
%% maxwidth is the original width if it's less than linewidth
%% otherwise use linewidth (to make sure the graphics do not exceed the margin)
\makeatletter
\def\maxwidth{ %
  \ifdim\Gin@nat@width>\linewidth
    \linewidth
  \else
    \Gin@nat@width
  \fi
}
\makeatother

\renewcommand{\contentsname}{Table of Contents}
% End of CII addition

\setlength{\parskip}{0pt}

% Added by CII
  %\setlength{\parskip}{\baselineskip}
  \usepackage[parfill]{parskip}

\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}

	\usepackage{float}
% End of CII addition
%%
%% End Preamble
%%
%
\begin{document}

% ------------------------------------
% ---- frontmatter: First Page -------
% ------------------------------------

\thispagestyle{empty}
  \begin{center}
  {\Large{Dissertation\\
    submitted to the\\
    Combined Faculty of Natural Sciences and Mathematics\\
    of the Ruperto Carola University Heidelberg, Germany\\
    for the degree of\\
    Doctor of Natural Sciences\\}} \vspace*{\fill}

  \Large{Presented by} \\\vspace{0.5cm}
  \Large{M.A. Holly Amelia Rebecca Giles} \\
  \Large{born in Portsmouth} \\
  \Large{Oral examination: Dec 15, 2021?}

\end{center}
\newpage\null\thispagestyle{empty}\newpage

% ------------------------------------
% --- frontmatter: Secound Page ------
% ------------------------------------

\pagebreak
\thispagestyle{empty}
\null
\begin{center}
\Large{Drug - microenvironment - gene interplay in chronic lymphocytic leukemia} \\
\vspace*{\fill}
\begin{tabular}{ c c }
Referees: & Dr.~Judith Zaugg \\
          & Prof.~Dr.~Michael Boutros
\end{tabular}
\end{center}
\newpage\null\thispagestyle{empty}\newpage

% --------------------------------------------
% -------- Declaration of Authorship ---------
% --------------------------------------------

\newpage
\thispagestyle{empty}
% \hypertarget{declaration-of-authorship}{%
% \section*{Declaration of Authorship}\label{declaration-of-authorship}}
%
% I hereby confirm that I have authored this Dissertation independently and
% without the use of others sources than the ones indicated. I have not yet
% presented this thesis or parts thereof to a university as part of an
% examination or degree. This work was carried out in the Genome Biology at
% the European Molecular Biology Laboratory (EMBL) in the group of Dr.~Wolfgang Huber and Dr.~Sascha Dietrich.
% \vspace{1cm}
%
% \noindent Heidelberg, \thesisdate{}
% \vspace{3cm}
%
% \noindent . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
% \vspace{0.1cm}
%
% \noindent \thesisauthor{}
% \newpage
% \thispagestyle{empty}
% \vspace*{\fill}
% Holly Amelia Rebecca Giles, \textit{Drug - microenvironment - gene interplay in chronic lymphocytic leukemia}, \textcopyright\, \thesisdate{}
% \newpage

%\frontmatter % this stuff will be roman-numbered
\pagestyle{empty} % this removes page numbers from the frontmatter



% edit style below this point? 


% --------------------------------------------
% --------------- Dedication -----------------
% --------------------------------------------
\vspace*{\fill}
\begin{center}
\textit{To \ldots{}}
\end{center}
\vspace*{\fill}
\newpage\null\thispagestyle{empty}\newpage

% ------------------------------------
% --- frontmatter: Acknowledgement ---
% ------------------------------------
\pagestyle{plain}
\pagenumbering{roman}   %  define page number in roman style
\setcounter{page}{1}    % start page numbering anew
\newpage
\hypertarget{acknowledgements}{%
\chapter*{Acknowledgements}\label{acknowledgements}}
\addcontentsline{toc}{chapter}{Acknowledgements}

I want to thank a few people.
\newpage\null\newpage


% --------------------------------------------
% ---------- List of publications ------------
% --------------------------------------------
    \hypertarget{list-of-publications}{%
    \chapter*{List of publications}\label{list-of-publications}}
    \addcontentsline{toc}{chapter}{List of publications}

    \hypertarget{thesis-related}{%
    \subsubsection*{Thesis related}\label{thesis-related}}
    \addcontentsline{toc}{subsubsection}{Thesis related}
    \begin{itemize}
        \item  Peter-Martin Bruch\ast,  \textbf{\underline{Holly A. R. Giles}}\ast, Carolin Kolb, Sophie Herbst, Tina Becirovic, Tobias Roider, Junyan Lu, Sebastian Scheinost, Lena Wagner, Jennifer Huellein, Ivan Berest, Mark Kriegsmann, Katharina Kriegsmann, Christiane Zgorzelski, Peter Dreger, Judith Zaugg, Carsten Mueller-Tidow, Thorsten Zenz, Wolfgang Huber, Sascha Dietrich et al. \textit{in preparation.}  "Mapping drug-microenvironment-genetic interplay in CLL reveals trisomy 12 as a modulator of microenvironment."\textbf{\textit{Journal}} \textbf{\textit{bioRxiv}}. doi:
    \end{itemize}
    \hypertarget{other-contributions}{%
    \subsubsection*{Other contributions}\label{other-contributions}}
    \addcontentsline{toc}{subsubsection}{Other contributions}

    The author of this thesis also contributed to a number of other projects throughout the PhD. The following have been published:
    \begin{itemize}
        \item Berest, Ivan\ast, Christian Arnold\ast, Armando Reyes-Palomares, Giovanni Palla, Kasper Dindler Rasmussen, \textbf{\underline{Holly Giles}}, and Peter-Martin Bruch et al. 2019. "Quantification Of Differential Transcription Factor Activity And Multiomics-Based Classification Into Activators And Repressors: DiffTF". \textit{Cell Reports} 29 (10): 3147-3159.e12. doi:10.1016/j.celrep.2019.10.106.
        
      \item Lu, Junyan\ast, Ester Cannizzaro\ast, Fabienne Meier-Abt, Sebastian Scheinost, Peter-Martin Bruch, \textbf{\underline{Holly A. R. Giles}}, and Almut Lütge et al. 2021. "Multi-Omics Reveals Clinically Relevant Proliferative Drive Associated With Mtor-MYC-OXPHOS Activity In Chronic Lymphocytic Leukemia". \textbf{\textit{Nature Cancer}}. doi:10.1038/s43018-021-00216-6.
    \end{itemize}
\newpage\null\newpage

% -------------------------------------
% --- frontmatter: Zusammendfassung ---
% -------------------------------------
    \hypertarget{zusammenfassung}{%
    \chapter*{Zusammenfassung}\label{zusammenfassung}}
    \addcontentsline{toc}{chapter}{Zusammenfassung}

    Abstract in German
    Aktuelle Entwicklungen im Bereich der ``omik''-Technologien tragen wesentlich zurBeschleunigung des Fortschritts in der Krebsforschung bei etc
\newpage\null\newpage

% -----------------------------
% --- frontmatter: Abstract ---
% -----------------------------

    \hypertarget{abstract}{%
    \chapter*{Abstract}\label{abstract}}
    \addcontentsline{toc}{chapter}{Abstract}

    The tumour microenvironment and genetic alterations collectively influence disease biology and drug resistance in Chronic Lymphocytic Leukaemia (CLL). To establish an integrative understanding of these factors in CLL biology, we performed a combinatorial assay using 12 drugs individually co-applied with each of 17 microenvironmental stimuli on 192 samples of CLL peripheral blood mononucleated cells. We examined microenvironment response across a heterogeneous patient cohort and identified four distinct CLL subgroups that differed in their response landscapes and in patient outcomes. By combining our data with whole-exome sequencing, DNA-methylation, RNA-sequencing and copy number variant data of the same tumours, we systematically searched for molecular determinants of stimulus response and found trisomy 12 as a key modulator. Our data suggest that the amplifying effect of trisomy 12 on the response to environmental signals is mediated by the transcription factors Spi-B and PU.1. We generated a comprehensive map of drug-microenvironment interactions in CLL, and profiled the modulating impact of genetic features on these antagonistic and synergistic effects. Interleukin (IL) 4 and Toll-Like Receptor (TLR) 7/8/9 stimuli showed the most interactions. Both pathways were more active in CLL-infiltrated lymph nodes than in healthy samples (p\textless0.001), and high IL4 activity in lymph nodes correlated with shorter survival (p=0.038). We provide a multi-layered resource to investigate microenvironmental and drug interplay in CLL (\href{https://git.embl.de/giles/cllcytokinescreen2021}{Repository} \& \href{https://www.google.com/url?q=https://www.imbi.uni-heidelberg.de/dietrichlab/CLL_Microenvironment/\&sa=D\&source=editors\&ust=1625764640786000\&usg=AOvVaw01HOFP7k2BJRLTteCXa8qy}{Shiny}). Our results highlight the importance of cell-extrinsic influences on drug response and disease progression, and how these further depend on molecular features.
\newpage\null\newpage

% -----------------------------
% --- frontmatter: Contents ---
% -----------------------------
\newpage
\tableofcontents
\clearpage

% ----------------------------------------------------------
% --- frontmatter: List of Abbreviations (not mandatory) ---
% ----------------------------------------------------------
\newpage
\hypertarget{list-of-abbreviations}{%
\chapter*{List of Abbreviations}\label{list-of-abbreviations}}
\addcontentsline{toc}{chapter}{List of Abbreviations}
\begin{tabular}{rp{0.2cm}lp{1cm}rp{0.2cm}l}
    AKT & & Protein kinase B \\
    BCR & & B-cell receptor \\
    BTK & & Bruton’s tyrosine kinase \\
    CDF & & Cumulative Distribution Function \\
    CLL & & Chronic lymphocytic leukemia \\
    CNV & & Copy number variation \\
    CpG ODN & & CpG oligodeoxynucleotides \\
    ERK/MAPK signalling & & Mitogen-activated protein kinase signalling \\
    FBS & & Fetal bovine serum \\
    FDR & & False discovery rate \\
    GSEA & & Gene set enrichment analysis  \\
    Ig & & Immunoglobulin \\
    IGHV & & Immunoglobulin heavy chain variable region \\
    IL & & Interleukin \\
    JAK & & Janus kinase \\
    M-CLL & & CLL with somatic hypermutations in IGHV loci \\
    NF$\kappa$B & & Nuclear factor kappa-light-chain-enhancer of activated B cells \\
    NOTCH & & Neurogenic locus notch  \\
    OS & & Overall survival \\
    PBMC & & Peripheral blood mononuclear cells \\
    PCA & & Principal component analysis \\
    PI3K & & Phosphoinositide 3-kinase \\
    U-CLL & & CLL without somatic hypermutations in IGHV loci \\
    TGF$\beta$ & & Transforming growth factor $\beta$ \\
    TLR & & Toll-like receptor \\ 
    TTT & & Time to next treatment \\
    SF3B1 & & Splicing factor 3B subunit 1 \\
    STAT & & Signal transducer and activator of transcription \\
    SYK & & Spleen tyrosine kinase \\
    ZAP70 & & Tyrosine-protein kinase ZAP-70 \\

    
   
\end{tabular}
% ----------------------------------------------------
% --- frontmatter: List of Figures (not mandatory) ---
% ----------------------------------------------------
\newpage
\listoffigures
\addcontentsline{toc}{section}{List of Figures}

% ---------------------------------------------------
% --- frontmatter: List of Tables (not mandatory) ---
% ---------------------------------------------------
\newpage
\listoftables
\addcontentsline{toc}{section}{List of Tables}

% edit here to affect style of main document 
%\mainmatter % here the regular arabic numbering starts
\newpage
\pagestyle{plain}
\setcounter{page}{1}    % start page numbering anew
\pagenumbering{arabic}  % page numbers in arabic style

\hypertarget{introduction}{%
\chapter{Introduction}\label{introduction}}

(theoretical background)
\#\# CLL Disease biology

\hypertarget{disease-characteristcs}{%
\subsection{Disease Characteristcs}\label{disease-characteristcs}}

\protect\hyperlink{ref-JCIpaper}{Dietrich et al.} (\protect\hyperlink{ref-JCIpaper}{2017}) shows that\ldots{} (\protect\hyperlink{ref-JCIpaper}{Dietrich et al. 2017}).

\hypertarget{ighv-status}{%
\subsection{IGHV status}\label{ighv-status}}

\hypertarget{the-role-of-bcr-signalling}{%
\subsection{The role of BCR signalling}\label{the-role-of-bcr-signalling}}

\hypertarget{recurrent-genetic-features-in-cll}{%
\subsection{Recurrent genetic features in CLL}\label{recurrent-genetic-features-in-cll}}

-mutations

-copy number variants

\hypertarget{the-incompletely-understood-role-of-trisomy12}{%
\subsection{The incompletely understood role of trisomy12}\label{the-incompletely-understood-role-of-trisomy12}}

\hypertarget{stratification-of-cll-patients}{%
\subsection{Stratification of CLL patients}\label{stratification-of-cll-patients}}

\hypertarget{the-tumour-microenvironment-in-cll}{%
\section{The tumour microenvironment in CLL}\label{the-tumour-microenvironment-in-cll}}

\hypertarget{the-concept-of-the-protective-niche}{%
\subsection{The concept of the protective niche}\label{the-concept-of-the-protective-niche}}

Lymph nodes etc

\hypertarget{microenvironmeental-pathways-and-cross-talk}{%
\subsection{Microenvironmeental pathways and cross-talk}\label{microenvironmeental-pathways-and-cross-talk}}

pathways esp IL4 and TLR\\
The microenvironment in CLL constitutes cross talk via soluble factors and cell-cell contacts, in the blood, bone marrow and lymph nodes.

\ref{fig:microenvironmentOverview}

\emph{Figure adapted from \ldots{}}

\ref{fig:microenvironmentTissues}

\emph{Figure adapted from \ldots{}}

\hypertarget{therapies-in-cll}{%
\section{Therapies in CLL}\label{therapies-in-cll}}

\hypertarget{clinical-regimes}{%
\subsection{Clinical regimes}\label{clinical-regimes}}

\hypertarget{new-targeted-therapies}{%
\subsection{New Targeted therapies}\label{new-targeted-therapies}}

in particular Ibrutinib and idelalisib

\hypertarget{drug-resistances-in-cll}{%
\subsection{Drug resistances in CLL}\label{drug-resistances-in-cll}}

incl ME -mediated and ibrutinib - IL4

\hypertarget{bet-inhibitors-and-their-potential-for-the-treatment-of-leukemia}{%
\subsection{BET inhibitors, and their potential for the treatment of leukemia}\label{bet-inhibitors-and-their-potential-for-the-treatment-of-leukemia}}

\hypertarget{background-to-the-approache-used-in-this-thesis}{%
\section{Background to the approache used in this thesis}\label{background-to-the-approache-used-in-this-thesis}}

\hypertarget{ex-vivo-drug-sensitivity-screens}{%
\subsection{Ex-vivo drug sensitivity screens}\label{ex-vivo-drug-sensitivity-screens}}

\hypertarget{studies-of-the-microenivornment-and-attempts-to-replicate-the-impact-of-the-microenvironment}{%
\subsection{Studies of the microenivornment, and attempts to replicate the impact of the microenvironment}\label{studies-of-the-microenivornment-and-attempts-to-replicate-the-impact-of-the-microenvironment}}

\hypertarget{measurement-of-outcomes}{%
\subsection{Measurement of outcomes?}\label{measurement-of-outcomes}}

LDT, TTT, TFT, OS

\hypertarget{multiomics-datasets-to-study-cll}{%
\subsection{Multiomics datasets to study CLL}\label{multiomics-datasets-to-study-cll}}

The power of multi-omics approaches especially the use of ATAC sequencing and combining ATAC and RNA

\hypertarget{mathematical-modelling}{%
\subsection{Mathematical modelling}\label{mathematical-modelling}}

Mathematical background in particular, lasso-regularised regression, and the caveats associated with feature selection
wrt cluster assignment

\hypertarget{cll-as-a-model-for-studying-tumour-biology}{%
\subsection{CLL as a model for studying tumour biology}\label{cll-as-a-model-for-studying-tumour-biology}}

\hypertarget{aims-of-this-thesis}{%
\section{Aims of this thesis}\label{aims-of-this-thesis}}

\hypertarget{overview-of-the-approach}{%
\section{Overview of the approach}\label{overview-of-the-approach}}

Including diagrams

\hypertarget{methods}{%
\chapter{Methods}\label{methods}}

Methods in quotation marks are taken from Bruch \& Giles et al.~2021 and I have authored the original text, unless stated otherwise. Paragraphs without quotation marks were rewritten for the thesis.

Routine:
all headings should be here
at the end of each chapter, make sure all methods are included
Copy things in and put in quotation marks if from Bruch \& Giles et al.~2021, mark if they were not written by me, or if they were written jointly
Add REFERENCE and TABLE for now
Return to this later and neaten / expand on methods, and provide all referencing and tables

\hypertarget{experimental-methods}{%
\section{Experimental methods:}\label{experimental-methods}}

\hypertarget{drug---stimulus-combinatorial-pertubation-assay}{%
\subsection{Drug - Stimulus Combinatorial Pertubation Assay}\label{drug---stimulus-combinatorial-pertubation-assay}}

\hypertarget{patient-sample-preparation}{%
\subsubsection{Patient Sample Preparation}\label{patient-sample-preparation}}

\emph{with Peter-Martin Bruch} Original
Peripheral Blood was taken from 192 patients for the initial drug-stimulation assay. Blood was separated by Ficoll gradient (GE Healthcare) and mononuclear cells were cryopreserved.

\hypertarget{preparation-of-screening-plates}{%
\subsection{Preparation of screening plates}\label{preparation-of-screening-plates}}

\emph{with Peter-Martin Bruch} Original
``Sample preparation, cell-culture, drug-stimulation profiling and genomic annotation was performed on 192 CLL patient samples as previously described{[}REFERENCE{]} with the following adjustments. Stimuli and drugs were mixed and preplated in the culture plates directly before adding the cell suspensions. RPMI-1640 and supplements were acquired from Gibco by Life Technologies, human serum was acquired from PAN Biotech (Cat.No. P40-2701, Lot.No:P-020317). Luminescence was measured after 48h on a Perkin Elmer EnVision.''

\emph{with Peter-Martin Bruch} Original
Compounds were obtained from Selleckchem, MedChemExpress and Sigma-Aldrich, dissolved in DMSO and stored at -\(20^\circ\)C. 12 drugs were used in two concentrations {[}TABLE{]}. Final DMSO concentration did not exceed 0.3\% in all experiments. Carfilzomib, Panobinostat and Venetoclax were removed from the analysis as they showed inconsistent toxicity depending on used media.
Insert table

Recombinant cytokines and stimulatory agents were dissolved according to manufacturer's protocol. 17 stimuli were selected {[}TABLE{]}. HS-5 conditioned medium was produced by incubating HS-5 stroma cell line (gifted by Martina Seiffert, DKFZ, Heidelberg) for 4 days at 37\(^\circ\)C and 5\% CO2, after which the supernatant was centrifuged and stored at -20\(^\circ\)C, the final concentration of HS-5 CM was 20\%. Bead immobilised anti-IgM was removed from the analysis due to storage instability. ''

Paper:
``Compounds and stimulatory agents were dissolved, stored and diluted according to manufacturer's protocol. HS-5 conditioned medium was produced by incubating HS-5 stromal cell line to \textgreater80\% confluency and cell removal by centrifugation. For a detailed list of stimuli and drugs and associated concentrations see Supp. Tables 1 and 2. Final DMSO concentration did not exceed 0.3\%.''

Original
Compounds were preplated in 384-well polypropylene storage plates (Greiner Bio-One Cat.No.:781271), which were stored at -20\(^\circ\)C. For each batch of samples tested on the same day a new storage plate was thawed and diluted in serum free RPMI, with or without stimuli. 5\(\mu\)L of drug-stimulation dilution were added into each well of the assay plates, 20 \(\mu\)L of cell suspension were added. The final cell concentration was 8*10\^{}5 cells/ml. Cells were thawed as previously described{[}REFERENCE{]}.

\hypertarget{drug-stimulation-assay}{%
\section{Drug-Stimulation assay}\label{drug-stimulation-assay}}

\emph{with Peter-Martin Bruch} Original
Drug-stimulation assays were performed with RPMI-1640 (Gibco by Life Technologies) supplemented with Penicillin Streptomycin (Gibco) final concentration of 100 Units/ml and 100 \(\mu\)g/ml respectively, L-Glutamine (Gibco) final concentration 2mM, and 10\% pooled, heat-inactivated and sterile filtered human type AB male off the clot serum (PAN Biotech, Cat.No. P40-2701, Lot.No:P-020317).

Drugs and stimuli were added to the cells simultaneously. Culture was performed in 384-well plates (Greiner Bio-One Cat.No.: 781904) and cells were incubated at 37\(^\circ\)C and 5\% CO2 for 48h.

Cell Viability was determined using the ATP-based CellTiter-Glo assay (Promega, Cat.No.:G7573). Luminescence was measured for the drug-stimulation assays using a Perkin Elmer EnVision and for the drug-drug interaction experiments using a Perkin Elmer EnSight, with a measurement time of 100ms per well.

\hypertarget{wes-rna-sequencing-targeted-sequencing-and-dna-copy-number-variants}{%
\section{WES, RNA Sequencing, Targeted Sequencing and DNA Copy number variants}\label{wes-rna-sequencing-targeted-sequencing-and-dna-copy-number-variants}}

\emph{Myself} Original
Sequencing data generated by Dietrich et al{[}REFERENCE{]}, and all sequencing and data processing were performed as described there. For

\hypertarget{follow---up-investigations}{%
\subsection{Follow - up investigations}\label{follow---up-investigations}}

\hypertarget{lymphocyte-doubling-time}{%
\subsubsection{Lymphocyte doubling time}\label{lymphocyte-doubling-time}}

\hypertarget{survival-data}{%
\subsubsection{Survival data}\label{survival-data}}

\hypertarget{atacsequencing-of-4-cll-samples}{%
\subsubsection{ATACsequencing of 4 CLL samples}\label{atacsequencing-of-4-cll-samples}}

\emph{with Peter-Martin Bruch (mostly edited by me)} Paper
``Peripheral blood was taken from 4 CLL patients and separated by Ficoll gradient (GE Healthcare), mononuclear cells were cryopreserved on liquid nitrogen. Samples were later thawed from frozen as previously described{[}REFERENCE{]} and MACS sorted for CD19 positive cells (Milteny autoMACS). The cells were resuspended in RPMI (GIBCO, Cat.No. 21875-034), with the addition of 2mM glutamine (GIBCO, Cat.No. 25030-24), 1\% Pen/Strep (GIBCO, Cat.No. 15140-122) and 10\% pooled, heat-inactivated and sterile filtered human type AB male off the clot serum (PAN Biotech, Cat.No. P40-2701, Lot.No:P-020317). 5ml of cell suspension was cultured in 6-well plates (Greiner Bio-One Cat.No. 657160). After thawing, cells were incubated at 37\(^\circ\)C and 5\% CO2 for 6 hours in 0.2\% DMSO. The final cell concentration was 2x10ˆ6 cells/ml. Cell viability and purity was assessed using FACS. All samples had a viability over 90\% and over 95\% of CD19+/CD5+/CD3- cells.''

\emph{with Nayara} Paper
``ATAC-seq libraries were generated as described previously{[}REFERENCE{]}. Cell preparation and transposition was performed according to the protocol, starting with 5x10ˆ4 cells per sample. Purified DNA was stored at −20\(^\circ\)C until library preparation was performed. To generate multiplexed libraries, the transposed DNA was initially amplified for 5x PCR cycles using 2.5 \(\mu\)L each of 25 μM PCR Primer 1 and 2.5 \(\mu\)L of 25 \(\mu\)M Barcoded PCR Primer 2 (included in the Nextera index kit, Illumina, San Diego, CA, USA), 25 \(\mu\)L of NEBNext High-Fidelity 2x PCR Master Mix (New England Biolabs, Boston, Massachusetts) in a total volume of 50 \(\mu\)L. 5 \(\mu\)L of the amplified DNA was used to determine the appropriate number of additional PCR cycles using qPCR. Additional number of cycles was calculated through the plotting of the linear Rn versus cycle, and corresponds to one-third of the maximum fluorescent intensity. Finally, amplification was performed on the remaining 45 \(\mu\)L of the PCR reaction using the optimal number of cycles determined for each library by qPCR (max. 13 cycles in total). The amplified fragments were purified with two rounds of SPRI bead clean-up (1.4x). The size distribution of the libraries was assessed on Bioanalyzer with a DNA High Sensitivity kit (Agilent Technologies, Santa Clara, CA), concentration was measured with Qubit DNA High Sensitivity kit in Qubit 2.0 Flurometer (Life Technologies, Carlsbad, CA). Sequencing was performed on NextSeq 500 (Illumina, San Diego, CA, USA) using 75bp paired-end sequencing, generating ∼450 million paired-reads per run, with an average of 55 million reads per sample.''

\hypertarget{spi-b-and-pu.1-shrna-knockdowns}{%
\subsubsection{Spi-B and PU.1 shRNA Knockdowns}\label{spi-b-and-pu.1-shrna-knockdowns}}

\hypertarget{immunohistochemistry-of-patient-lymph-nodes}{%
\subsubsection{Immunohistochemistry of patient lymph nodes}\label{immunohistochemistry-of-patient-lymph-nodes}}

\emph{Peter-Martin Bruch} Paper
``LN biopsies of CLL-infiltrated and non-neoplastic lymph nodes were Paraffin fixed and arranged in Tissue Microarrays. Consecutively they were stained for PAX-5 (790-4420, Roche), CD3 (790-4341, Roche), pSTAT6 (ab28829, abcam), STAT6 (519-4290, Zytomed Systems) and pIRAK4 (ab216513, abcam). The slides were analysed using Qupath 50 and the recommended protocol. Cell based data on mean staining intensity was exported and further analysed using R.''

\hypertarget{ibrutinib---il4---stat6i-interaction-assay}{%
\subsubsection{Ibrutinib - IL4 - STAT6i interaction assay}\label{ibrutinib---il4---stat6i-interaction-assay}}

\hypertarget{investigation-of-ibrutinib-il4-ibet-762}{%
\subsection{Investigation of Ibrutinib + IL4 + IBET-762}\label{investigation-of-ibrutinib-il4-ibet-762}}

\hypertarget{ibrutinib---il4---ibet-762-interaction-assay}{%
\subsubsection{Ibrutinib - IL4 - IBET-762 interaction assay}\label{ibrutinib---il4---ibet-762-interaction-assay}}

\hypertarget{ibrutinib---il4---ibet--762-treatment}{%
\subsubsection{Ibrutinib - IL4 - IBET -762 treatment}\label{ibrutinib---il4---ibet--762-treatment}}

\hypertarget{atacsequencing}{%
\subsubsection{ATACsequencing}\label{atacsequencing}}

\emph{with Peter Martin Bruch} get from diffTF
Peripheral blood was taken from 4 CLL patients and separated by Ficoll gradient (GE Healthcare), mononuclear cells were cryopreserved on liquid nitrogen. Samples were later thawed from frozen as previously described{[}REFERENCE{]} and MACS sorted for CD19 positive cells (Milteny autoMACS). The cells were resuspended in RPMI (GIBCO, Cat.No. 21875-034), with the addition of 2mM glutamine (GIBCO, Cat.No. 25030-24), 1\% Pen/Strep (GIBCO, Cat.No. 15140-122) and 10\% pooled, heat-inactivated and sterile filtered human type AB male off the clot serum (PAN Biotech, Cat.No. P40-2701, Lot.No:P-020317). 5ml of cell suspension was cultured in 6-well plates (Greiner Bio-One Cat.No. 657160). After thawing, cells were incubated at 37\(^\circ\)C and 5\% CO2 for 6 hours in 0.2\% DMSO. The final cell concentration was 2x10ˆ6 cells/ml. Cell viability and purity was assessed using FACS. All samples had a viability over 90\% and over 95\% of CD19+/CD5+/CD3- cells.

\emph{with Nayara} get from diffTF
ATAC-seq libraries were generated as described previously{[}REFERENCE{]}. Cell preparation and transposition was performed according to the protocol, starting with 5x10ˆ4 cells per sample. Purified DNA was stored at −20\(^\circ\)C until library preparation was performed. To generate multiplexed libraries, the transposed DNA was initially amplified for 5x PCR cycles using 2.5 \(\mu\)L each of 25 \(\mu\)M PCR Primer 1 and 2.5 \(\mu\)L of 25 \(\mu\)M Barcoded PCR Primer 2 (included in the Nextera index kit, Illumina, San Diego, CA, USA), 25 \(\mu\)L of NEBNext High-Fidelity 2x PCR Master Mix (New England Biolabs, Boston, Massachusetts) in a total volume of 50\(\mu\)L. 5 \(\mu\) L of the amplified DNA was used to determine the appropriate number of additional PCR cycles using qPCR. Additional number of cycles was calculated through the plotting of the linear Rn versus cycle, and corresponds to one-third of the maximum fluorescent intensity. Finally, amplification was performed on the remaining 45 \(\mu\)L of the PCR reaction using the optimal number of cycles determined for each library by qPCR (max. 13 cycles in total). The amplified fragments were purified with two rounds of SPRI bead clean-up (1.4x). The size distribution of the libraries was assessed on Bioanalyzer with a DNA High Sensitivity kit (Agilent Technologies, Santa Clara, CA), concentration was measured with Qubit DNA High Sensitivity kit in Qubit 2.0 Flurometer (Life Technologies, Carlsbad, CA). Sequencing was performed on NextSeq 500 (Illumina, San Diego, CA, USA) using 75bp paired-end sequencing, generating ∼450 million paired-reads per run, with an average of 55 million reads per sample.

\hypertarget{rnasequencing}{%
\subsubsection{RNAsequencing}\label{rnasequencing}}

\hypertarget{proteomics}{%
\subsubsection{Proteomics}\label{proteomics}}

\hypertarget{data-availability}{%
\section{Data availability}\label{data-availability}}

European Genome-Phenome Archive (EGA) accession \ldots. The data for \ldots. and computational analysis code used in this study are available from \ldots.

Additionally, we obtained CLL ATACseq data{[}REFERENCE{]} from the European Genome-phenome Archive (EGA: EGAD00001002110).

For the ChIPseq analysis of SPIB binding sites, we accessed the data 37 from the NCBI GEO database 51, accession GEO: GSE56857. We made use of the data for SPIB in the OCILY3 DLBCL cell line (GSM1370276).

\hypertarget{statistical-analysis}{%
\section{Statistical Analysis}\label{statistical-analysis}}

Mixutre of paper and original
Integrative data analysis of screening data, DNA and RNA sequencing, CNV and methylation profiles and follow up experiments was performed using R version 4 (REFERENCE R Core Team, 2018) with the RStudio interface (REFERENCE RStudio Team, 2016) and using packages that included (\protect\hyperlink{ref-DESeq2}{\textbf{DESeq2?}}){[}REFERENCE{]}, (\protect\hyperlink{ref-survival}{\textbf{survival?}}){[}REFERENCE{]}, (\protect\hyperlink{ref-Glmnet}{\textbf{Glmnet?}}){[}REFERENCE{]}, (\protect\hyperlink{ref-ConsensusClusterPlus}{Wilkerson and Hayes 2010}){[}REFERENCE{]}, (\protect\hyperlink{ref-clusterProfiler}{\textbf{clusterProfiler?}}){[}REFERENCE{]}, (\protect\hyperlink{ref-ChIPseeker}{\textbf{ChIPseeker?}}){[}REFERENCE{]}, (\protect\hyperlink{ref-genomation}{\textbf{genomation?}}){[}REFERENCE{]} and (\protect\hyperlink{ref-BloodCancerMultiOmics2017}{\textbf{BloodCancerMultiOmics2017?}}){[}REFERENCE{]} to perform univariate association tests, multivariate regression with and without lasso penalization, Cox regression, generalised linear modelling and clustering.

\hypertarget{screening-data}{%
\subsection{Screening Data}\label{screening-data}}

\hypertarget{processing-of-screening-data}{%
\subsubsection{Processing of screening data}\label{processing-of-screening-data}}

\emph{with Peter-Martin Bruch} Paper
To quantify the response of a patient sample to each treatment condition, we used viability relative to the control, i.e., the CellTiter Glo luminescence readout of the respective well divided by the median of luminescence readouts of the DMSO control wells on the same plate.

\hypertarget{quality-assessment-and-control}{%
\subsubsection{Quality Assessment and Control}\label{quality-assessment-and-control}}

\hypertarget{drug--drug-and-stimulus---stimulus-correlations}{%
\subsubsection{Drug -drug and stimulus - stimulus correlations}\label{drug--drug-and-stimulus---stimulus-correlations}}

\emph{myself} original complete
Pearson correlation coefficients were calculated for each drug - drug and stimulus - stimulus pair, using the \texttt{cor} functions of the (\protect\hyperlink{ref-stats}{\textbf{stats?}}){[}REFERENCE{]} package with log transformed viability values which were normalised to untreated controls.

\hypertarget{characterising-stimulus-responses-across-all-samples}{%
\subsubsection{Characterising stimulus responses across all samples}\label{characterising-stimulus-responses-across-all-samples}}

\emph{myself} original complete (just check)
For the heatmap in figure \ref{fig:stimuliHeatmap}, the viability data is represented as z scores. Log transformed viability values, normalised to DMSO controls, are row-scaled according to the Median Absolute Deviance. Limits were then applied to this row scaling factor for optimal visualisation. The matrix of z scores was plotted using the (\protect\hyperlink{ref-pheatmap}{\textbf{pheatmap?}}){[}REFERENCE{]} package. The ordering of the columns (patient samples) was obtained from the dendrogram that resulted from running \texttt{ConsensusClusterPlus}, from the (\protect\hyperlink{ref-ConsensusClusterPlus}{Wilkerson and Hayes 2010}) {[}REFERENCE{]} package. The rows were ordered using the dendrogram order produced by \texttt{hclust} with default branch arrangement.

\hypertarget{univariate-gene---repsonse-testing}{%
\subsubsection{Univariate gene - repsonse testing}\label{univariate-gene---repsonse-testing}}

\hypertarget{penalised-multivariate-regression-of-gene---response}{%
\subsubsection{Penalised multivariate regression of gene - response}\label{penalised-multivariate-regression-of-gene---response}}

\emph{myself} original
For the plots in Figure 3B, we used a Gaussian linear model with L1-penalty. As the dependent variable, the normalised viability value was used for all stimuli. We used the same features as above, plus Methylation Cluster(coded as 0, 0.5, 1). As a measure of explained variance the reduction in cross-validated mean squared error relative to the null model was calculated.

\hypertarget{correlation-of-rna-receptor-expression-with-viability}{%
\subsubsection{Correlation of RNA-Receptor Expression with viability}\label{correlation-of-rna-receptor-expression-with-viability}}

\hypertarget{linear-modelling-of-drug---stimulus-interactions}{%
\subsubsection{Linear modelling of drug - stimulus interactions}\label{linear-modelling-of-drug---stimulus-interactions}}

\hypertarget{modelling-of-drug---stimulus---gene-interactions}{%
\subsubsection{Modelling of drug - stimulus - gene interactions}\label{modelling-of-drug---stimulus---gene-interactions}}

\emph{myself} original
To identify drug - stimulus interactions that were dependent on genetic features, we first fitted linear models to the viability matrix, for each drug - stimulus combination, using the formula in Eqn2. We extracted drug - stimulus interaction coefficients for each patient to generate a response matrix for n = 137 patients. Next, we performed multivariate regression using a Gaussian linear model with L1-penalty (i.e., lasso regression) as implemented in the R package glmnet{[}REFERENCE{]}. As the dependent variable, the matrix of drug - stimulus interaction coefficients for each patient sample was used. As input to the model, genetic features with more than 20\% missing values were excluded, and only patients with complete annotation were included in the model ( n= 137 ). As predictors, the genetic mutations and CNVs (p= 39), IGHV status (coded as 0-1) and Methylation CLuster (coded as 0, 0.5, 1) were used, using 3-fold cross-validation. Misclassification error was used as loss for cross- validation.

\hypertarget{follow-up}{%
\subsection{Follow up}\label{follow-up}}

\hypertarget{lympocyte-doubling-times}{%
\subsubsection{Lympocyte doubling times}\label{lympocyte-doubling-times}}

\emph{Herbst et al., 2020c} - update
``Patients which had lymphocyte counts available for less than 4 timepoints between the sample
collection date and the time of the next treatment and patients currently in treatment were ex-cluded. Thus, \ldots{} patients with enough data remained. Lymphocyte growth rates were calculated by fitting a linear model to the log10 transformed lymphocyte counts of all timepoints between the sample collection date and the time of the next treatment versus the period of time.''

\hypertarget{survival-analyses}{%
\subsubsection{Survival analyses}\label{survival-analyses}}

\emph{myself} original
Survival analyses were performed using TTT, TFT and OS as metrics. Follow-up information to calculate OS was available for all 192 CLL patients. For 188 of 192 CLL patients treatment information after sample collection was available.

\emph{Herbst et al}
''Time to next treatment (TTT) was calculated from the date of sample collection to subsequent
treatment initiation. Patients without treatment initiation during the observation time and
patients who died before treatment initiation were censored at the latest follow-up contact.''

\emph{myself} original
For for the purpose of visualising Kaplan-Meier plots , optimal cut-points of staining levels were calculated using maximally selected rank statistics as computed by the package (\protect\hyperlink{ref-maxstat}{\textbf{maxstat?}}) \href{Hothorn,2017}{REFERENCE}. Based on these cut points, patients were split into two groups, and their survival data were plotted using the Kaplan-Meier method, using the R package (\protect\hyperlink{ref-survminer}{\textbf{survminer?}}) \href{Kassambara\%20et\%20al.,\%202019}{REFERENCE} .

For Cox proportional hazards regression, the \texttt{coxph} function of the R package survival {[}REFERENCE{]} (Therneau, 2020) was used. Expand

\hypertarget{penalised-multivariate-regression-genetic-predictors-of-cluster-membership}{%
\subsubsection{Penalised multivariate regression genetic predictors of cluster membership}\label{penalised-multivariate-regression-genetic-predictors-of-cluster-membership}}

\emph{Myself} original (edit so not similar to JCI)
I used a multinomial linear model with L1-penalty, implemented in the (\protect\hyperlink{ref-cvglmnet}{\textbf{cvglmnet?}}) package{[}REFERENCE{]}. As the dependent variable, the cluster assignment for each patient was used. As input to the model ,genetic features with more than 20\% missing values were excluded, and only patients with complete annotation were included in the model ( n= 137 ). As predictors, the genetic mutations and CNVs (p= 39) and IGHV status (coded as 0-1) were used, using 3-fold cross-validation. Misclassification error was used as loss for cross- validation. The resulting coefficients indicated associations between genetic features and each cluster.

\hypertarget{gene-expression-and-gene-set-enrichment-analysis-between-clusters}{%
\subsubsection{Gene expression and gene set enrichment analysis between clusters}\label{gene-expression-and-gene-set-enrichment-analysis-between-clusters}}

\emph{myself} original
For the n=49 patient samples for which viability data and RNA--Seq data for matching samples was available, the R package (\protect\hyperlink{ref-DESeq2}{\textbf{DESeq2?}}){[}REFERENCE{]} was used to search for associations of these two data types.RNA-Seq read count data was regressed on to the patient clusters C3, C4 (design formula \textasciitilde{} IGHV.status + Cluster). Genes were ranked by their test statistics and Gene Set Enrichment Analysis (GSEA) (implementing the fgsea algorithm with the (\protect\hyperlink{ref-clusterProfiler}{\textbf{clusterProfiler?}}) package{[}REFERENCE{]}) was applied to the ranked lists with the KEGG pathway gene set selections from the MSigDBdatabase{[}REFERENCE{]}.

\hypertarget{associations-of-ex-vivo-stimulus-responses-with-genomic-features}{%
\subsubsection{Associations of ex-vivo stimulus responses with genomic features}\label{associations-of-ex-vivo-stimulus-responses-with-genomic-features}}

We tested for associations between stimulus viability assay results and genomic features by Student's t-tests (two-sided, with equal variance). We tested somatic mutations (aggregated at the gene level), copy number aberrations and IGHV status. We restricted the analysis to features that were present in at least 3 patient samples (63 features). p-values were adjusted for multiple testing by applying the Benjamini-Hochberg procedure.

\hypertarget{atacseq-processing}{%
\subsubsection{ATACseq processing}\label{atacseq-processing}}

\emph{with Ivan Berest} original (update to paper)
We downloaded CLL data{[}REFERENCE{]} from the European Genome-phenome Archive (EGA: EGAD00001002110). Original dataset had 88 ATAC-seq samples from 55 patients, however we used for the analysis only one sample per patient passing quality checks, resulting in 52 samples. We used an in-house constructed ATAC-seq processing pipeline{[}REFERENCE{]} to obtain final bam files mapped to the hg19 annotation genome that were corrected for CG bias.

\hypertarget{annotation-of-trisomy-12-status}{%
\subsubsection{Annotation of trisomy 12 status}\label{annotation-of-trisomy-12-status}}

\emph{with Ivan Berest} original (update to paper)
As trisomy 12 status was not included in the original metadata, we used a mean amount of reads in the chromatin accessible peaks for each sample to distinguish trisomy 12 patients. All samples containing 1.4 times more reads in the peaks located on chromosome 12, compared to the peaks on all other chromosomes, were classified as trisomy 12 patients.

\hypertarget{difftf-analysis-of-tf-activity-in-trisomy-12-cll}{%
\subsubsection{diffTF analysis of TF activity in trisomy 12 CLL}\label{difftf-analysis-of-tf-activity-in-trisomy-12-cll}}

\emph{with Ivan Berest} original (update to paper)
FOr the larger ATACseq dataset, we ran analytical mode of diffTF{[}REFERENCE{]} pipeline to investigate differential TF activity between trisomy 12 and non-trisomy 12 patients using the HOCOMOCO v10 database {[}REFERENCE{]} with the following design formula: ``\textasciitilde{} Batch + Gender + IGHV status + trisomy12 status.''

For the smaller ATACseq dataset, ATAC-seq data generated from our CLL samples were processed similarly with the in-house ATAC-seq pipeline, as described previously{[}REFERENCE{]}, with the only exception that we didn't use CG bias correction step. We used a similarly analytical mode of diffTF with HOCOMOCO v10 database{[}REFERENCE{]} using the following parameters: minOverlap = 1; design formula = ``\textasciitilde{} Patient + Trisomy 12 status.''

\hypertarget{functional-enrichment-analysis-of-spib-chipseq-data}{%
\subsubsection{Functional enrichment analysis of SPIB ChIPseq data}\label{functional-enrichment-analysis-of-spib-chipseq-data}}

\emph{myeslf} original (update to paper)
We downloaded SPIB ChIPseq data{[}REFERENCE{]} from the NCBI GEO database{[}REFERENCE{]}, accession \url{GEO:GSE56857}. We made use of the data for SPIB in the OCILY3 DLBCL cell line (GSM1370276). SPIB ChIP peaks were filtered for significance (q value\textless0.05). We used the annotatePeaks function from the package clusterProfiler{[}REFERENCE{]} to annotate the nearest gene for each ChIPpeak. We filtered peaks within ±1kb of a TSS of a gene and performed over-representation of KEGG and Reactome pathways (using the clusterProfiler package{[}REFERENCE{]}) amongst the resulting list of genes.

\hypertarget{linear-modelling-of-drug---stimulus-interactions-1}{%
\subsubsection{Linear modelling of drug - stimulus interactions}\label{linear-modelling-of-drug---stimulus-interactions-1}}

\emph{myself} original
To identify drug-stimulus interactions, we fitted linear models to the viability matrix, for each drug - stimulus combination, using the formula in Eqn1. Drug - stimulus interaction coefficients and associated p values were extracted and used to define significant interactions.

\hypertarget{drug-drug-interaction-assay}{%
\subsubsection{Drug-Drug interaction assay}\label{drug-drug-interaction-assay}}

An independent patient cohort of 16 patients was used in the drug-drug interaction experiments.

\hypertarget{survival-analysis-of-immunohistory-chemistry-data}{%
\subsubsection{Survival analysis of immunohistory chemistry data}\label{survival-analysis-of-immunohistory-chemistry-data}}

\hypertarget{ibrutinib---ibet-762---il-4-analysis}{%
\subsection{Ibrutinib - IBET-762 - IL-4 analysis}\label{ibrutinib---ibet-762---il-4-analysis}}

\hypertarget{rnaseq-processing}{%
\subsubsection{RNAseq processing}\label{rnaseq-processing}}

\hypertarget{deseq2-analysis}{%
\subsubsection{Deseq2 analysis}\label{deseq2-analysis}}

\hypertarget{atacseq-processing-1}{%
\subsubsection{ATACseq processing}\label{atacseq-processing-1}}

\hypertarget{difftf-analysis}{%
\subsubsection{diffTF analysis}\label{difftf-analysis}}

\hypertarget{generation-of-gene-regulatory-network}{%
\subsubsection{Generation of Gene Regulatory Network}\label{generation-of-gene-regulatory-network}}

\hypertarget{proteomics-processing}{%
\subsubsection{Proteomics Processing}\label{proteomics-processing}}

\hypertarget{proteomics-analysis}{%
\subsubsection{Proteomics Analysis}\label{proteomics-analysis}}

\hypertarget{data}{%
\chapter{Data}\label{data}}

Characterisation of primary CLL samples by high-throughput combinatorial screening and multi-omic profiling. An introduction to the dataset that my PhD is based on.

\hypertarget{drug-screens-and-experiments}{%
\section{Drug screens and experiments}\label{drug-screens-and-experiments}}

\hypertarget{high-throughput-combinatorial-pertubation-assay}{%
\subsection{High-throughput combinatorial pertubation assay}\label{high-throughput-combinatorial-pertubation-assay}}

We measured the effects of 17 cytokines and microenvironmental stimuli on cell viability in 192 primary CLL samples and combined each one with 12 drugs to investigate the influence on spontaneous and drug-induced apoptosis (Figure \ref{fig:studyOverview}). Viability was assessed by ATP measurement via CellTiterGlo after 48 h of culture and normalised to untreated controls8.


\begin{figure}

{\centering \includegraphics[width=0.7px]{figures/studyOverview} 

}

\caption{Schematic of experimental protocol. By combining 12 drugs and 17 stimuli, we systematically queried the effects of simultaneous stimulation and inhibition of critical pathways in CLL (n=192). Integrating functional drug-stimulus response profiling with four additional omics layers, we identified pro-survival pathways, underlying molecular modulators of drug and microenvironment reponses, and drug-stimulus interactions in CLL.}\label{fig:studyOverview}
\end{figure}
\hypertarget{validation-experiments}{%
\subsection{Validation experiments}\label{validation-experiments}}

In addition to the screen, we acquired the following validatory data (Figure \ref{fig:additionalData}).


\begin{figure}

{\centering \includegraphics[width=0.7px]{figures/studyOverview} 

}

\caption{Schematic of validatory data.}\label{fig:additionalData}
\end{figure}
\hypertarget{characteristics-of-drugs-used-in-the-screen}{%
\section{Characteristics of drugs used in the screen}\label{characteristics-of-drugs-used-in-the-screen}}

\hypertarget{drug-pathways}{%
\subsection{Drug pathways}\label{drug-pathways}}

We screened 17 different drugs (Figure \ref{fig:drugCategories}).


\begin{figure}

{\centering \includegraphics[width=1px]{thesis_files/figure-latex/drugCategories-1} 

}

\caption{Bar plot of the drugs used in screen.}\label{fig:drugCategories}
\end{figure}
\hypertarget{drug-responses}{%
\subsection{Drug responses}\label{drug-responses}}

The drug responses were as follows (Figure \ref{fig:drugResponses}).


\begin{figure}

{\centering \includegraphics[width=0.5px]{thesis_files/figure-latex/drugResponses-1} 

}

\caption{Log transformed viability values for all drugs that were included in the screen after qulaity control. p values from student's t test.}\label{fig:drugResponses}
\end{figure}
\hypertarget{genetic-predictors-of-drug-responses}{%
\subsection{Genetic predictors of drug responses}\label{genetic-predictors-of-drug-responses}}

\hypertarget{drug---drug-correlations}{%
\subsection{Drug - Drug Correlations}\label{drug---drug-correlations}}

\hypertarget{characteristics-of-stimuli-used-in-the-screen}{%
\section{Characteristics of stimuli used in the screen}\label{characteristics-of-stimuli-used-in-the-screen}}

\hypertarget{stimulus-responses}{%
\subsection{Stimulus responses}\label{stimulus-responses}}

\hypertarget{stimulus---stimulus-correlations}{%
\subsection{Stimulus - Stimulus Correlations}\label{stimulus---stimulus-correlations}}

\hypertarget{characteristics-of-patient-samples-used-in-the-screen}{%
\section{Characteristics of patient samples used in the screen}\label{characteristics-of-patient-samples-used-in-the-screen}}

\hypertarget{genetic-data-available-for-each-patient}{%
\subsection{Genetic Data available for each patient}\label{genetic-data-available-for-each-patient}}

WES, CNVs, Methylation, Transcriptomic, ATACseq, LDT, survT, IHC
PLus Ibrutinib + IBET + IL4 treated samples - ATACseq, RNAseq, proteomics

\hypertarget{processing-of-raw-values-obtained-from-cell-viability-assay}{%
\section{Processing of raw values obtained from cell viability assay}\label{processing-of-raw-values-obtained-from-cell-viability-assay}}

\hypertarget{data-normalization-and-quality-control}{%
\subsection{Data normalization and quality control}\label{data-normalization-and-quality-control}}

\hypertarget{generation-of-public-resource}{%
\section{Generation of public resource}\label{generation-of-public-resource}}

\hypertarget{shiny-app}{%
\subsection{Shiny app}\label{shiny-app}}

\hypertarget{package}{%
\subsection{Package}\label{package}}
\begin{itemize}
\tightlist
\item
  Present the underlying economic model/theory and give reasons why it is
  suitable to answer the given problem\footnote{Here is an example of a footnote.}.
\end{itemize}
\hypertarget{ex-vivo-sensitivity-to-microenvironmental-stimulation-in-primary-cll-cells}{%
\chapter{Ex-vivo sensitivity to microenvironmental stimulation in primary CLL cells}\label{ex-vivo-sensitivity-to-microenvironmental-stimulation-in-primary-cll-cells}}

The screen included 17 cytokines and microenvironmental stimuli, which were selected based on evidence in the literature that each stimulus had been shown to impact on CLL viability \emph{in vitro}. We attempted to put together the largest panel of stimuli for an assay of this kind in CLL, aiming to minimise redundancy among the compounds (Bruch \& Giles et al.~2021).

CLL cells do not proliferate \emph{in vitro}, but rather undergo spontaneous apoptosis in the absence of stimulation (\protect\hyperlink{ref-Collins1989}{Collins et al. 1989}). The effect of each stimuli on CLL viability was thus quantified by comparing ATP counts in treated primary samples, compared with those in DMSO wells where a positive value indicates that the sample viability was increased relative to control.

The assay represents a reductionist model of microenvironmental signalling, making it possible to dissect the effect of individual stimuli on baseline viability. The large patient cohort also enabled us to study the differential impact of these stimuli across heterogeneous samples, and identify distinct subgroups of patients whose samples show similar response profiles to the panel of stimuli.

\hypertarget{prolifing-responses-to-the-panel-of-stimuli}{%
\section{Prolifing responses to the panel of stimuli}\label{prolifing-responses-to-the-panel-of-stimuli}}

\hypertarget{ex-vivo-assay-demonstrated-functional-diversity-of-cytokines-and-microenvironmental-stimuli}{%
\subsection{\texorpdfstring{\emph{ex vivo} assay demonstrated functional diversity of cytokines and microenvironmental stimuli}{ex vivo assay demonstrated functional diversity of cytokines and microenvironmental stimuli}}\label{ex-vivo-assay-demonstrated-functional-diversity-of-cytokines-and-microenvironmental-stimuli}}

To investigate heterogeneity amongst responses to the stimuli, I calculated Pearson correlation coefficients for each stimulus pair, using the log-transformed normalised viabilities (Bruch \& Giles et al.~2021). The resulting coefficients were ordered using hierarchical clustering and visualised in a symmetrical heatmap (Figure \ref{fig:stimuliCorrelations}). The hierarchical clustering distinguished clusters of stimuli, including a larger group corresponding to agonists of TLR and Nf\(\kappa\)B pathways and a smaller group encompassing IL4 and TLR stimuli.


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/stimuliCorrelations-1} 

}

\caption{Heatmap of Pearson correlation coefficients. Coefficients for each pair of stimuli were calculated using log transformed viability values normalised to untreated control, and ordered according to hierarchical clustering. Figure adapted from Bruch \& Giles et al.~2021.}\label{fig:stimuliCorrelations}
\end{figure}
98.5\% of stimulus pairs showed little correlation (R \textless{} 0.6), including those that targeted similar downstream pathways, indicating a high degree of functional diversity amongst microenvironmental signals. For example, JAK-STAT agonists such as IL4 and IL6 showed little correlation (Figure \ref{fig:correlationsScatter}A).

Only two stimulus pairs showed correlations where R \textgreater{} 0.6, and in both cases these targeted near identical receptors or downstream pathways. These included CpG ODN (TLR 9) and Resiquimod (TLR 7 and 8) (Figure \ref{fig:correlationsScatter}B), and IL4 and IL4 + CD40L which primarily target JAK3 - STAT6.

Repeating the analysis to correlate drug - drug pairs demonstrated that drugs targeting components of the same pathway were highly correlated. For quality control purposes, this indicated that our data sensitively and specifically reflect inter-individual differences in pathway dependencies (\protect\hyperlink{ref-JCIpaper}{Dietrich et al. 2017}).


\begin{figure}

{\centering \includegraphics{thesis_files/figure-latex/correlationsScatter-1} 

}

\caption{Scatter plot of log-transformed viability values, normalised to DMSO controls, for (A) treatment with JAK-STAT agonists IL4 and IL6 and (B) treatment with TLR agonists CpG ODN and Resiquimod.}\label{fig:correlationsScatter}
\end{figure}
Microenvironmental stimulation induced diverse phenotypes between patient samples, and across different stimuli. To gain a global overview of these phenotypes, I visualised log-transformed viability values normalised to DMSO controls for all patient samples and all stimuli (Figure \ref{fig:stimuliBeeswarms} (Bruch \& Giles et al.~2021).


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/stimuliBeeswarms-1} 

}

\caption{Log transformed viabilities after treatment with each stimulus. Where stimuli decreased viability relative to control, points are shown in blue, whilst increased viability is shown in red. Figure adapted from Bruch \& Giles et al.~2021.}\label{fig:stimuliBeeswarms}
\end{figure}
The majority of the stimuli increased viability, underlining the supportive nature of the microenvironment in CLL. However, three out of 17 reduced CLL viability relative to control, namely IL6, TGF\(\beta\) and TLR 7/8/9 agonists in IGHV-mutated (IGHV-M) samples.

IL4 and TLR7/8/9 agonists Resiquimod and CpG ODN induced the strongest responses, an indication of their potency in modulating CLL cell survival. Notably, TLR agonists increased viability in certain samples, in most cases IGHV-U, and decreased viability in others, mostly IGHV-M. Both IL4 and TLR7/8/9 emerged as the key pathways in the screen, and play an important role throughout the results of this thesis.

\hypertarget{microenvironmental-response-profiling-identifies-discrete-patient-subgroups}{%
\subsection{Microenvironmental response profiling identifies discrete patient subgroups}\label{microenvironmental-response-profiling-identifies-discrete-patient-subgroups}}

To further investigate the variability in responses across the cohort, we visualised z scores of the log-transformed viabilities and performed consensus clustering on the resulting matrix to generate a heatmap of all stimuli responses across all samples (Figure \ref{fig:stimuliHeatmap}.


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/stimuliHeatmap-1} 

}

\caption{The heatmap matrix shows the viability measurements for 192 samples (columns) and 17 stimuli (rows). Viability was measured via ATP-based assay after 48h stimulation. The data are shown normalised to DMSO-treated controls, and scaled within each row according to the Median Absolute Deviation (MAD). Limits were applied to scaling factor for optimal visualisation. The colour bars to the right show sample annotations. Consensus Clustering was used to define column tree layout, using hierarchical clustering with the Euclidean metric. Figure from Bruch \& Giles et al.~2021.}\label{fig:stimuliHeatmap}
\end{figure}
The heatmap reflects heterogeneity in responses across samples and stimuli, once again underlining the potency of IL4 in increasing sample viability across diverse genetic backgrounds and the diversity in responses to TLR stimulation by Resiquimod and CpG ODN.

To generate the column-wise clustering, consensus clustering was performed on the matrix of z scores. Consensus clustering allows the user to subsample from the matrix of values, to generate hierarchical clustering for a given number of clusters, k. From this, it is possible to calculate a consensus matrix for each value of k, indicating for each pair of values the proportion of time they occupy the same cluster when subsampled together.

I visualised the clustered heatmap in Figure \ref{fig:stimuliHeatmap} for different values of k, and concluded on the existence of four robust clusters within the cohort (add these to appendix?). Each cluster shows a unique response profile to the panel of stimuli. We termed the clusters C1 to C4: C1 and C2 were enriched for IGHV-U whilst the samples in C3 and C4 were mostly IGHV-M.

To validate the choice of four clusters, I visualised the summaries of the consensus matrix, using the \texttt{ConsensusClusterPlus} package (\protect\hyperlink{ref-ConsensusClusterPlus}{Wilkerson and Hayes 2010}), to quantify the degree of confidence in the clusters for different values of k.


\begin{figure}

{\centering \includegraphics[width=0.45\linewidth]{figures/consensus008} \includegraphics[width=0.45\linewidth]{figures/consensus009} 

}

\caption{Summaries of the CDFs of the consensus matrices. Consensus CDF graphic showing the CDFs of the consensus matrix for k = 2 - 7, as indicated in the legend, estimated using 100 bin histogram (left). Relative change in area under the CDF curve, for k = 2 - 7, to compare k with k - 1. In the case of k = 2, there is no k - 1, so the total area is plotted. Line shows relative increase in consensus between each value of k (right).}\label{fig:consensusClusteringA}
\end{figure}
The graph of the CDFs of the consensus matrix for each k indicated that the CDF reaches a maximum and cluster confidence is maximised at k = 7 , though above k = 4 there is little appreciable increase. Figure \ref{fig:consensusClusteringA}. This is confirmed in the graph showing relative change in the area under the CDF curve, showing there is only a small increase in consensus between k = 4 and k =5, supporting the choice of k = 4. The cluster tracking plot depicts how each patient sample is assigned for each value of k. For k = 4, the plot indicates that C3 and 4 in particular are highly stable (Figure \ref{fig:consensusClusteringB}).


\begin{figure}

{\centering \includegraphics[width=0.45\linewidth]{figures/consensus010} 

}

\caption{Assignment of patient samples (columns), to each cluster, for k = 1 - 7 (rows) to demonstrate stability of cluster membership. Cluster colour for k = 4 match those in heatmap in \ref{fig:stimuliHeatmap}.}\label{fig:consensusClusteringB}
\end{figure}
\hypertarget{functional-characterisation-of-patient-clusters}{%
\section{Functional characterisation of patient clusters}\label{functional-characterisation-of-patient-clusters}}

\hypertarget{c1---c4-showed-distinct-response-profiles-with-the-panel-of-stimuli}{%
\subsection{C1 - C4 showed distinct response profiles with the panel of stimuli}\label{c1---c4-showed-distinct-response-profiles-with-the-panel-of-stimuli}}

The heatmap in Figure \ref{fig:stimuliHeatmap} demonstrated that each cluster responded differently to the panel of stimuli.

Amongst the IGHV-U enriched C1 and C2, both showed strong, positive responses to IL4 and TLR7/8/9 stimulation. C2 could be distinguished by stronger responses to the stimuli overall, in particular to NF\(\kappa\)B agonists IL1\(\beta\), anti-IgM, BAFF and sCD40L. Amongst the IGHV-M enriched clusters, C3 showed weaker responses to the majority of stimuli, and C4 was defined by a negative response to TLR7/8/9 stimulation (Bruch \& Giles et al.~2021). Figure \ref{fig:clusterStimuliResponse} summarises these findings in more detail, showing responses stratified by cluster for a subset of the stimuli.


\begin{verbatim}
cairo_pdf 
        2 
\end{verbatim}
\hypertarget{the-clusters-show-differences-in-disease-dynamics}{%
\subsection{The clusters show differences in disease dynamics}\label{the-clusters-show-differences-in-disease-dynamics}}

To validate the potential biological significance of these four clusters, we investigated whether the groups showed differential \emph{in vivo} disease progression (Bruch \& Giles et al.~2021). The study design was such that not all patients in the cohort were treatment - free, which confounded the analysis. For that reason, lymphocyte doubling time (LDT) and time to next treatment (TTT) were used to quantify CLL proliferative capacity, independently of treatment.

C1 and C2 showed a shorter LDT than C3 and C4, which is expected due to the differential proportions of IGHV-U and M patient samples in these groups (Figure \ref{fig:clusterLDTTTT}A). Notably, within the IGHV-M enriched clusters C3 and C4, samples in C3 showed a significantly shorter LDT (Student's t-test, p-value = 0.025).

To further validate this, we observed that TTT in the IGHV-M enriched C3 was significantly shorter than C4 and comparable to the progression dynamics of IGHV-U enriched C1 and 2 (Figure \ref{fig:clusterLDTTTT}B).


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/clusterLDTTTT-1} 

}

\caption{(A) Lymphocyte doubling time (LDT) stratified by cluster, p-values from Student's t-test. (B) Kaplan-Meier curves to show TTT for each cluster. p-values from univariate Cox proportional hazard models comparing IGHV-U enriched C1 with C2, and IGHV-M enriched C3 with C4. Figure from Bruch \& Giles et al.~2021.}\label{fig:clusterLDTTTT}
\end{figure}
The difference in disease progression between the clusters indicated that microenvironmental response represents an additional biological layer, holding information relevant to disease dynamics. To validate that these clusters were not simply an indication of any underlying genetic features, we checked whether the observed differences in progression dynamics could be explained by other prognostic markers (Bruch \& Giles et al.~2021).

A multivariate Cox proportional hazard model accounting for IGHV status, trisomy 12 and TP53 in addition to the cluster assignment indicated an independent prognostic value for cluster assignment between C3 and C4 (p= 0.039, Table \ref{tab:clusterCox}).


\begin{table}

\caption{\label{tab:clusterCox}Table depicting results of Multivariate Cox Proportional Hazard Model to test prognostic value of key genetic features and clusters using Time to Next Treatment and C3 as reference. }
\centering
\begin{tabular}[t]{l|r|r|r|r|l}
\hline
Factor & coef & exp(coef) & se(coef) & z & p.value\\
\hline
\cellcolor[HTML]{E2E868}{Cluster 3 vs Cluster 1} & \cellcolor[HTML]{E2E868}{-0.03979} & \cellcolor[HTML]{E2E868}{0.96099} & \cellcolor[HTML]{E2E868}{0.29813} & \cellcolor[HTML]{E2E868}{-0.13347} & \cellcolor[HTML]{E2E868}{0.89382}\\
\hline
Cluster 3 vs Cluster 2 & 0.51595 & 1.67522 & 0.37741 & 1.36708 & 0.1716\\
\hline
\cellcolor[HTML]{E2E868}{Cluster 3 vs Cluster 4} & \cellcolor[HTML]{E2E868}{-0.82011} & \cellcolor[HTML]{E2E868}{0.44038} & \cellcolor[HTML]{E2E868}{0.39760} & \cellcolor[HTML]{E2E868}{-2.06267} & \cellcolor[HTML]{E2E868}{0.03914}\\
\hline
IGHV.status & 0.55192 & 1.73658 & 0.27253 & 2.02513 & 0.04285\\
\hline
\cellcolor[HTML]{E2E868}{trisomy 12} & \cellcolor[HTML]{E2E868}{-0.13357} & \cellcolor[HTML]{E2E868}{0.87496} & \cellcolor[HTML]{E2E868}{0.35617} & \cellcolor[HTML]{E2E868}{-0.37503} & \cellcolor[HTML]{E2E868}{0.70764}\\
\hline
TP53 & 1.38977 & 4.01395 & 0.26072 & 5.33058 & <0.0001\\
\hline
\end{tabular}
\end{table}
\hypertarget{the-clusters-showed-differential-responses-to-drugs-in-vitro}{%
\subsection{\texorpdfstring{The clusters showed differential responses to drugs \emph{in vitro}}{The clusters showed differential responses to drugs in vitro}}\label{the-clusters-showed-differential-responses-to-drugs-in-vitro}}

The potential clinical relevance of the clusters was underlined by my observation that the samples within each group showed differential responses to drugs \emph{in vitro} (Figure \ref{fig:clusterDrugResponse}).

As expected, the IGHV-U enriched clusters C1 and 2 were more sensitive to BCR inhibition by ibrutinib, idelalisib and PRT062607, than C3 and 4. Between C1 and C2, C2 was more sensitive to a number of the drugs, including idelalisib (SYK) (p-value = 0.012), everolimus (mTOR) (p-value = 0.02) and the chemotherapeutics fludarabine (p-value = 0.031) and nutlin-3a (p-value = 0.042). Amongst C3 and C4, C3 showed lower sensitivity to everolimus (p-value = 0.051) and to fludarabine (p-value \textless{} 0.001) and nutlin-3a (p-value = 0.01). This aligns with the observation that patients in C3 have a poorer prognosis, despite most of these samples annotated as IGHV-M. C4 also showed a positive repsonse to Nf\(\kappa\)B inhibition by BAY-11-7085, and p38 MAPK inhibition by Ralimetinib.

Such differential drug response patterns suggests that microenvironmental response may reflect disease-specific CLL biology, in the same way as molecular profiling, and thus may have the potential to guide therapy decisions in future.


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/clusterDrugResponse-1} 

}

\caption{Log-transformed normalised viability values, stratified by cluster, for each drug. Drugs targeting the same pathway are grouped together. P-values from Student't t-test.}\label{fig:clusterDrugResponse}
\end{figure}
\hypertarget{the-clusters-are-enriched-for-different-genetic-features}{%
\subsection{The clusters are enriched for different genetic features}\label{the-clusters-are-enriched-for-different-genetic-features}}

Next we assessed differences in the molecular profiles of samples within each cluster. Visually, it appeared that certain clusters were enriched or depleted for various genetic features recurrent in CLL (Figure \ref{fig:clusterMatrix}.


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/clusterMatrix-1} 

}

\caption{Distribution of selected genetic features (rows) within each cluster for all patient samples (columns). Where a patient sample is not annotated for a feature, this is marked in white.}\label{fig:clusterMatrix}
\end{figure}
To quantify this, I ran a multinomial model, with lasso regularisation, to predict cluster membership (C1 -4) based on the matrix of genetic features for all the patient samples (Bruch \& Giles et al 2021). The model assigned coefficients to genetic features, where a positive coefficient indicated that this feature was enriched in the cluster, and a negative coefficient indicated it was depleted.

The approach used 3-fold cross validation, selecting the optimal model using lamda min. To ensure that the resulting coefficients were robust, we selected coefficients that satisfied certain cut-offs. Coefficients were selected if they were assigned in \textgreater{} 60\% of 50 bootstrapped repeats, and were larger than 0.35. Figure \ref{fig:clusterEnrichment} shows the mean coefficients and associated standard deviation, for each genetic feature that met these criteria in each cluster.


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/clusterEnrichment-1} 

}

\caption{Multinomial regression with lasso penalisation to identify enrichment or depletion of genetic features within each cluster. Matrix of genetic features (p=39), and IGHV status (encoded as M = 1 and U = 0) were used to identify multivariate predictors of cluster assignment. x axis shows genetic predictors, y axis indicates value and sign of coefficient assigned to feature, for each cluster (positive coefficients are enriched in the cluster, negative coefficients are depleted). Coefficients shown are mean coefficients from 50 bootstrapped repeats and error bars represent the mean ± standard deviation. Genetic features with \textgreater20\% missing values were excluded, and only patients with complete annotation were included in the model (n=137). Figure from Bruch \& Giles et al 2021.}\label{fig:clusterEnrichment}
\end{figure}
As we expected, IGHV status was the main feature that predicted cluster membership. Beyond IGHV status, trisomy 12 and \emph{SF3B1} mutations were enriched in C2, which showed enhanced responses to many stimuli. C4, which was associated with slow in-vivo progression, showed depletion of \emph{TP53}, \emph{ATM}, RAS/RAF mutations and gain8q.

\hypertarget{gsea-of-de-genes-between-subgroups}{%
\subsection{GSEA of DE genes between subgroups}\label{gsea-of-de-genes-between-subgroups}}

In addition to genetic features, I investigated differential expression of genes within each cluster. For n = 49 samples, RNAseq data was available for matched PBMC samples. I focused on the difference between clusters 3 and 4, for which 21 RNAseq samples were available (Bruch \& Giles et al.~2021).

To quantify differential gene expression, I began by filtering out immunoglobulin genes, including genes at the heavy, light and kappa loci that encode the antigen receptor of B cells. The clusters each show differential enrichment of IGHV-M and U samples, and thus the differential expression analysis would otherwise be dominated by immunoglobulin genes that are well known to affected by this biomarker.

I followed the Deseq2 protocol using a design formula to quantify the difference between clusters, and accounting for the confounding effect of IGHV status. 87 genes were differentially expressed (adjusted p \textless{} 0.05) between C3 and 4 (Figure \ref{fig:clusterDE}).


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/clusterDE-1} 

}

\caption{Volcano plot of differentially expressed genes between C3 and C4. X axis indicates log2 fold change values, calculated using the \texttt{DESeq2} package (\protect\hyperlink{ref-DESeq2}{\textbf{DESeq2?}}), y axis gives corresponding -log10(adjusted p value). P values adjusted using BH method. Genes are labeled where ajdusted p \textless{} 0.05. Figure from Bruch \& Giles et al.~2021.}\label{fig:clusterDE}
\end{figure}
To assign biological meaning to the differentially expressed genes, I quantified the enrichment of Hallmark pathways amongst the genes. I ranked the genes based on the Wald statistic, and then ran GSEA was using the fgsea algorithm (Figure \ref{fig:clusterGSEA}) (Bruch \& Giles et al.~2021).

Several pathways were upregulated amongst samples in C3, compared to C4, indicating that these pathways may relate in some way to the shorter TTT and LDT of patients within this cluster. Pathways associated with higher disease aggression were regulated in C3 including genesets relating to stress response (Unfolded Protein Response, UV Response Up, P53 Pathway), metabolism (Oxidative Phosphorylation) and proliferation (G2M Checkpoint, MYC Targets V1, MTORC1 Signaling, E2F Targets) (Figure \ref{fig:GSEApathways}).


\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/clusterGSEA-1} 

}

\caption{Gene set enrichment analysis (GSEA) to compare expression of genes in samples from C3 and C4 reveals upregulation of Hallmark pathways involved in microenvironmental signalling, stress response, metabolism and proliferation in C3. Normalised enrichment scores (NES) are shown for top 10 most significant pathways upregulated in C3 versus C4. Bars coloured according to adjusted p-value. Genes are ranked based on Wald statistic, calculated using the Deseq2 package and GSEA performed using the fgsea algorithm. Figure from Bruch \& Giles et al.~2021.}\label{fig:clusterGSEA}
\end{figure}
In addition, C3 showed upregulation of microenvironmental signalling pathways relative to C4, including TNFa Signalling via NFKB and Interferon Gamma Response (Figure \ref{fig:GSEApathways}) (Bruch \& Giles et al.~2021). This finding underlines our hypothesis that differential activity of microenvironmental signalling, both \emph{in vivo} and \emph{ex vivo} may be relevant to disease prognosis.


\begin{figure}

{\centering \includegraphics{thesis_files/figure-latex/GSEApathways-1} 

}

\caption{Enrichment plots of selected pathways. Gene set enrichment analysis (GSEA) was performed with the Hallmark gene sets from the GSEA Molecular Signatures Database. Wald statistic was used to rank the genes. The green curve corresponds to the Enrichment Score curve, which is the running sum of the weighted enrichment score obtained from GSEA software. Figure from Bruch \& Giles et al.~2021}\label{fig:GSEApathways}
\end{figure}
\hypertarget{additional-analysis}{%
\section{Additional Analysis}\label{additional-analysis}}

Can microenvironmental response aid prognostic models? Run a Multivariate model to predict survival/drug response with genetic matrix and stimuli response matrix?

\hypertarget{summary}{%
\section{Summary}\label{summary}}

The screen represents an attempt to comprehensively dissect the impact of individual microenvironmental pathways on CLL viability. Our assay has enabled us to highlight key, broad spectrum signals such as IL4. Our heterogeneous cohort also reveals pathways that operate in subsets of patients, such as TLR. IL4 and TLR represent the key pathways that we focus on throughout the rest of this thesis.

In addition, this approach enables us to use microenvironmental response profiles, as an alternative to molecular profiling, to look for the existence of subgroups.

We uncovered four such subgroup with distinct response profiles and molecular properties and clinical outcomes, suggesting that microenvironmental response holds biologically significant information that may be relevant to prognosis and treatment decision making.

\hypertarget{discussion}{%
\section{Discussion}\label{discussion}}

Gosia suggested discussing each results section individually?

\hypertarget{genetic-modulators-of-responses-to-microenvironmental-stimulation}{%
\chapter{Genetic modulators of responses to microenvironmental stimulation}\label{genetic-modulators-of-responses-to-microenvironmental-stimulation}}

To continue:
add the additional plots
text and plot arrangement also update the cytokine labelling and gene labelling
any additional plots
text dump

Continute to figure 6 (see at clsoe of tomorrow if you are better of leaveing this till i get back, then only one week behind)

\hypertarget{systematic-analysis-of-the-effect-of-genetic-features-on-responses-to-stimuli}{%
\section{Systematic analysis of the effect of genetic features on responses to stimuli:}\label{systematic-analysis-of-the-effect-of-genetic-features-on-responses-to-stimuli}}

\hypertarget{univariate-analysis-identifies-ighv-status-and-tirsomy-12-as-key-modulators}{%
\subsection{Univariate analysis identifies IGHV status and tirsomy 12 as key modulators}\label{univariate-analysis-identifies-ighv-status-and-tirsomy-12-as-key-modulators}}

(Figure \ref{fig:stimuliGeneAssosciations}).

(ref:stimuliGeneAssosciations)
\begin{figure}

{\centering \includegraphics[width=0.7\linewidth]{thesis_files/figure-latex/stimuliGeneAssosciations-1} 

}

\caption{(ref:stimuliGeneAssosciations)}\label{fig:stimuliGeneAssosciations}
\end{figure}
\hypertarget{multivariate-analysis-demosntrates-multiple-layers-of-biology-involved}{%
\subsection{Multivariate analysis demosntrates multiple layers of biology involved}\label{multivariate-analysis-demosntrates-multiple-layers-of-biology-involved}}

(Figure \ref{fig:stimuliGeneAssosciationsMulti}).

(ref:stimuliGeneAssosciationsMulti)
\begin{figure}

{\centering \includegraphics[width=0.7\linewidth]{thesis_files/figure-latex/stimuliGeneAssosciationsMulti-1} 

}

\caption{(ref:stimuliGeneAssosciationsMulti)}\label{fig:stimuliGeneAssosciationsMulti}
\end{figure}
\hypertarget{individual-cytokine-gene-interactions-of-note}{%
\subsection{Individual cytokine -- gene interactions of note}\label{individual-cytokine-gene-interactions-of-note}}

Role of IGHV status and BCR signalling in Cytokine response
(Figure \ref{fig:IHGVBCR}).

(ref:IHGVBCR)
\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/IHGVBCR-1} 

}

\caption{(ref:IHGVBCR)}\label{fig:IHGVBCR}
\end{figure}
TLR response stratified by IGHV status (and trisomy 12) and discussion of interactions between BCR and TLR pathways
(Figure \ref{fig:TLRIHGVtri12}).

(ref:TLRIHGVtri12)
\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/TLRIHGVtri12-1} 

}

\caption{(ref:TLRIHGVtri12)}\label{fig:TLRIHGVtri12}
\end{figure}
-Modulators of TLR signalling incl.~effect of del11q and ATM mutations on TLR response
(Figure \ref{fig:TLRGenes}).

(ref:TLRGenes)

Potentially show effect of IGHV, then ivnestgiation of TLR combined boave and belwo, then look intro tri 12

These plots are not working right now - can work on them if i haev time
\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/TLRGenes-1} 

}

\caption{(ref:TLRGenes)}\label{fig:TLRGenes}
\end{figure}
-samples with KRAS mutations benefit less from IL4 stimulation

(Figure \ref{fig:IL4KRAS}).

(ref:IL4KRAS)
\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/IL4KRAS-1} 

}

\caption{(ref:IL4KRAS)}\label{fig:IL4KRAS}
\end{figure}
\hypertarget{profiling-the-effects-of-trisomy-12-on-cytokine-response-and-how-this-may-be-possible}{%
\section{Profiling the effects of trisomy 12 on cytokine response, and how this may be possible}\label{profiling-the-effects-of-trisomy-12-on-cytokine-response-and-how-this-may-be-possible}}

\hypertarget{trisomy-12-modulates-responses-to-il4-tgfbeta-and-tlr-stimuli}{%
\subsection{Trisomy 12 modulates responses to IL4, TGFbeta and TLR stimuli}\label{trisomy-12-modulates-responses-to-il4-tgfbeta-and-tlr-stimuli}}

(Figure \ref{fig:SMAD3geneDosage}).

(ref:SMAD3geneDosage)
\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/tri12cytResponse-1} 

}

\caption{(ref:tri12cytResponse)}\label{fig:tri12cytResponse}
\end{figure}
\hypertarget{gene-dosage-effects}{%
\subsection{Gene dosage effects}\label{gene-dosage-effects}}

(Figure \ref{fig:tri12geneDosage}.

(ref:tri12geneDosage)

\emph{The Proteomics dataset was kindly shared by Sophie Herbst, from her paper\ldots{}}
\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/SMAD3geneDosage-1} 

}

\caption{(ref:SMAD3geneDosage)}\label{fig:SMAD3geneDosage}
\end{figure}
(Figure \ref{fig:STATgeneDosage}).

(ref:STATgeneDosage)
\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/STATgeneDosage-1} 

}

\caption{(ref:STATgeneDosage)}\label{fig:STATgeneDosage}
\end{figure}
(Figure \ref{fig:IRAK4geneDosage}).

(ref:IRAK4geneDosage)
\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/IRAK4geneDosage-1} 

}

\caption{(ref:IRAK4geneDosage)}\label{fig:IRAK4geneDosage}
\end{figure}
\hypertarget{investigation-of-non-trisomy12-samples-that-are-phenocopies-of-trisomy12-cll-based-on-their-responses-to-stimuli}{%
\subsection{Investigation of non-trisomy12 samples that are phenocopies of trisomy12 CLL+ (based on their responses to stimuli)}\label{investigation-of-non-trisomy12-samples-that-are-phenocopies-of-trisomy12-cll-based-on-their-responses-to-stimuli}}

(Figure \ref{fig:tri12Classfier}.

(ref:tri12Classfier)
\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/tri12Classfier-1} 

}

\caption{(ref:tri12Classfier)}\label{fig:tri12Classfier}
\end{figure}
(Figure \ref{fig:tri12PhenocopiesCytResponse}.

(ref:tri12PhenocopiesCytResponse)
\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{thesis_files/figure-latex/tri12PhenocopiesCytResponse-1} 

}

\caption{(ref:tri12PhenocopiesCytResponse)}\label{fig:tri12PhenocopiesCytResponse}
\end{figure}
(Figure \ref{fig:tri12PhenocopiesRNA}.

(ref:tri12PhenocopiesRNA)
\begin{figure}

{\centering \includegraphics[width=0.8\linewidth]{thesis_files/figure-latex/tri12PhenocopiesRNA-1} 

}

\caption{(ref:tri12PhenocopiesRNA)}\label{fig:tri12PhenocopiesRNA}
\end{figure}
Pat033 has duplicated regions:
-p13.31 - 42 copies
-q24.13 - 10 copies
-q24.33 - 21 copies
See what Junyan says about this and potentially add a figure

(Figure \ref{fig:tri12PhenocopiesCNV}.

(ref:tri12PhenocopiesCNV)

\hypertarget{investigation-of-tf-activity-in-trisomy12-cll-and-finding-that-spib-and-pu1-tfs-are-upregulated-in-trisomy12-cll}{%
\subsection{Investigation of TF activity in trisomy12+ CLL and finding that SPIB and PU1 TFs are upregulated in trisomy12+ CLL}\label{investigation-of-tf-activity-in-trisomy12-cll-and-finding-that-spib-and-pu1-tfs-are-upregulated-in-trisomy12-cll}}

tri 12 annotation

running diffTF, add a diagram to explain?

(Figure \ref{fig:diffTFexplainer}.

(ref:diffTFexplainer)
\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{figures/diffTF_method} 

}

\caption{(ref:diffTFexplainer)}\label{fig:diffTFexplainer}
\end{figure}
visualisation of results

(Figure \ref{fig:tri12annotation}.

(ref:tri12annotation)

(Figure \ref{fig:tri12diffTF}.

(ref:tri12diffTF)

Also refer to : (Figure \ref{fig:diffTFsmallvolPlot}
\begin{figure}

{\centering \includegraphics[width=0.8\linewidth]{thesis_files/figure-latex/tri12diffTF-1} 

}

\caption{(ref:tri12diffTF)}\label{fig:tri12diffTF}
\end{figure}
\hypertarget{profiling-the-downstream-effects-of-spib-and-pu1}{%
\section{Profiling the downstream effects of SPIB and PU1:}\label{profiling-the-downstream-effects-of-spib-and-pu1}}

\hypertarget{over-representation-of-bcr-tlr-interleukin-and-tgfbeta-response-genes-amongst-spib-and-pu1-targets-in-chipseq-data}{%
\subsection{Over-representation of BCR, TLR, interleukin and TGFbeta response genes amongst SPIB and PU1 targets in ChIPseq data}\label{over-representation-of-bcr-tlr-interleukin-and-tgfbeta-response-genes-amongst-spib-and-pu1-targets-in-chipseq-data}}

(Figure \ref{fig:SpiBChipSeq}.

(ref:SpiBChipSeq)

\hypertarget{visualisation-of-spib-and-pu1-binding-sites-from-chipseq-data-in-bcr-tlr-interleukin-and-tgfbeta-signalling-genes}{%
\subsection{Visualisation of SPIB and PU1 binding sites from ChIPseq data in BCR, TLR, interleukin and TGFbeta signalling genes}\label{visualisation-of-spib-and-pu1-binding-sites-from-chipseq-data-in-bcr-tlr-interleukin-and-tgfbeta-signalling-genes}}

(Figure \ref{fig:SpiBBindingSites}.

(ref:SpiBBindingSites)

\hypertarget{investigation-of-de-genes-within-il4-and-tgfbeta-pathways-in-spibpu1-dko-mice}{%
\subsection{Investigation of DE genes within IL4 and TGFbeta pathways in SPIB/PU1 DKO mice}\label{investigation-of-de-genes-within-il4-and-tgfbeta-pathways-in-spibpu1-dko-mice}}

(Figure \ref{fig:SpiBDEgenes}.

(ref:SpiBDEgenes)

\hypertarget{spib-and-pu1-linked-genes}{%
\subsection{SPIb and PU1 linked genes}\label{spib-and-pu1-linked-genes}}

Key result: generation of GRN to identify SPIB and PU1 regulated genes, and investigation of associated log2FCs of these genes in trisomy 12 vs non-trisomy12 samples (including CD79A and B)

(Figure \ref{fig:SpiBGRN}.

(ref:SpiBGRN)

\hypertarget{spibpu1-dkos-in-trisomy-12-cell-lines}{%
\subsection{SPIB/PU1 DKOs in trisomy 12+ cell lines}\label{spibpu1-dkos-in-trisomy-12-cell-lines}}

(Figure \ref{fig:SpiBshRNAKD}.

(ref:SpiBshRNAKD)

\emph{The data here was generated by Tina Bericoviv and was included in Bruch \& Giles et al.~2021. It is included in this thesis for completeness.}
\begin{figure}

{\centering \includegraphics[width=1\linewidth]{thesis_files/figure-latex/SpiBshRNAKD-1} 

}

\caption{(ref:SpiBshRNAKD)}\label{fig:SpiBshRNAKD}
\end{figure}
\hypertarget{conclusions}{%
\section{Conclusions}\label{conclusions}}

so everything is here
focus on the content you have and finihsh - add additioanl if time

\hypertarget{results}{%
\chapter{Results}\label{results}}
\begin{itemize}
\item
  Organize material and present results.
\item
  Use tables, figures (but prefer visual presentation):
  \begin{itemize}
  \item
    Tables and figures should supplement (and not duplicate) the text.
  \item
    Tables and figures should be provided with legends.
  \item
    \emph{Figure \ref{fig:graph} shows how to include and reference graphics.
    The graphic must be labelled before. Files must be in \textbf{.eps} format. You
    can do this really easily in R Markdown with \texttt{knitr::include\_graphics()}}!
  \item
    Figures can be referenced with \texttt{\textbackslash{}@ref(fig:\textless{}name\textgreater{})}, where \texttt{\textless{}name\textgreater{}} is the
    name of the code chunk.
  \end{itemize}
\end{itemize}
\begin{figure}

{\centering \includegraphics[width=0.5\linewidth]{figures/graph} 

}

\caption{Estimated residuals from model XXX3. ...}\label{fig:graph3}
\end{figure}
\begin{itemize}
\item
  Tables and graphics may appear in the text or in the appendix, especially if
  there are many simulation results tabulated, but is also depends on the study
  and number of tables resp. figures. The key graphs and tables must appear in
  the text!
\item
  R Markdown can also supports math equations just like \emph{LaTeX}!
  \begin{itemize}
  \item
    \emph{Equation \eqref{eq:SpecDens} represents the ACs of a stationary
    stochastic process:}
    \begin{equation}
            f_y(\lambda) = (2\pi)^{-1} \sum_{j=-\infty}^{\infty}
                           \gamma_j e^{-i\lambda j}
                         =(2\pi)^{-1}\left(\gamma_0 + 2 \sum_{j=1}^{\infty}
        \gamma_j \cos(\lambda j)\right)
                                       \label{eq:SpecDens}
    \end{equation}
    \emph{where \(i=\sqrt{-1}\) is the imaginary unit, \(\lambda \in [-\pi, \pi]\) is the
    frequency and the \(\gamma_j\) are the autocovariances of \(y_t\).}
  \item
    Equations can be referenced with \texttt{\textbackslash{}@ref(eq:\textless{}name\textgreater{})}, where name is defined
    by adding \texttt{(\textbackslash{}\#eq:\textless{}name\textgreater{})} in the line immediately before \texttt{\textbackslash{}end\{equation\}}.
  \end{itemize}
\end{itemize}
\hypertarget{review-of-results}{%
\section{Review of Results}\label{review-of-results}}
\begin{itemize}
\item
  Do the results support or do they contradict economic theory ?
\item
  What does the reader learn from the results?
\item
  Try to give an intuition for your results.
\item
  Provide robustness checks.
\item
  Compare to previous research.
\end{itemize}
\hypertarget{results-1}{%
\chapter{Results}\label{results-1}}
\begin{itemize}
\item
  Organize material and present results.
\item
  Use tables, figures (but prefer visual presentation):
  \begin{itemize}
  \item
    Tables and figures should supplement (and not duplicate) the text.
  \item
    Tables and figures should be provided with legends.
  \item
    \emph{Figure \ref{fig:graph} shows how to include and reference graphics.
    The graphic must be labelled before. Files must be in \textbf{.eps} format. You
    can do this really easily in R Markdown with \texttt{knitr::include\_graphics()}}!
  \item
    Figures can be referenced with \texttt{\textbackslash{}@ref(fig:\textless{}name\textgreater{})}, where \texttt{\textless{}name\textgreater{}} is the
    name of the code chunk.
  \end{itemize}
\end{itemize}
\begin{figure}

{\centering \includegraphics[width=0.5\linewidth]{figures/graph} 

}

\caption{Estimated residuals from model XXX4. ...}\label{fig:graph4}
\end{figure}
\begin{itemize}
\item
  Tables and graphics may appear in the text or in the appendix, especially if
  there are many simulation results tabulated, but is also depends on the study
  and number of tables resp. figures. The key graphs and tables must appear in
  the text!
\item
  R Markdown can also supports math equations just like \emph{LaTeX}!
  \begin{itemize}
  \item
    \emph{Equation \eqref{eq:SpecDens} represents the ACs of a stationary
    stochastic process:}
    \begin{equation}
            f_y(\lambda) = (2\pi)^{-1} \sum_{j=-\infty}^{\infty}
                           \gamma_j e^{-i\lambda j}
                         =(2\pi)^{-1}\left(\gamma_0 + 2 \sum_{j=1}^{\infty}
        \gamma_j \cos(\lambda j)\right)
                                       \label{eq:SpecDens}
    \end{equation}
    \emph{where \(i=\sqrt{-1}\) is the imaginary unit, \(\lambda \in [-\pi, \pi]\) is the
    frequency and the \(\gamma_j\) are the autocovariances of \(y_t\).}
  \item
    Equations can be referenced with \texttt{\textbackslash{}@ref(eq:\textless{}name\textgreater{})}, where name is defined
    by adding \texttt{(\textbackslash{}\#eq:\textless{}name\textgreater{})} in the line immediately before \texttt{\textbackslash{}end\{equation\}}.
  \end{itemize}
\end{itemize}
\hypertarget{review-of-results-1}{%
\section{Review of Results}\label{review-of-results-1}}
\begin{itemize}
\item
  Do the results support or do they contradict economic theory ?
\item
  What does the reader learn from the results?
\item
  Try to give an intuition for your results.
\item
  Provide robustness checks.
\item
  Compare to previous research.
\end{itemize}
\hypertarget{discussion-1}{%
\chapter{Discussion}\label{discussion-1}}

\newpage

\hypertarget{references}{%
\chapter*{References}\label{references}}
\addcontentsline{toc}{chapter}{References}

\noindent

\setlength{\parindent}{-0.5cm}
\setlength{\leftskip}{0.5cm}
\setlength{\parskip}{8pt}

\hypertarget{refs}{}
\begin{CSLReferences}{1}{0}
\leavevmode\vadjust pre{\hypertarget{ref-Collins1989}{}}%
Collins, Russell J., Louise A. Verschuer, Brian V. Harmon, Roger L. Prentice, John H. Pope, and John F. R. Kerr. 1989. {``{Spontaneous programmed death (apoptosis) of B‐chronic lymphocytic leukaemia cells following their culture in vitro}.''} \emph{British Journal of Haematology} 71 (3): 343--50. \url{https://doi.org/10.1111/j.1365-2141.1989.tb04290.x}.

\leavevmode\vadjust pre{\hypertarget{ref-JCIpaper}{}}%
Dietrich, Sascha, Małgorzata Oleś, Junyan Lu, Leopold Sellner, Simon Anders, Britta Velten, Bian Wu, et al. 2017. {``{Drug-perturbation-based stratification of blood cancer}.''} \emph{Journal of Clinical Investigation} 128 (1): 427--45. \url{https://doi.org/10.1172/JCI93801}.

\leavevmode\vadjust pre{\hypertarget{ref-ConsensusClusterPlus}{}}%
Wilkerson, Matthew D, and D Neil Hayes. 2010. {``{ConsensusClusterPlus: a class discovery tool with confidence assessments and item tracking.}''} \emph{Bioinformatics (Oxford, England)} 26 (12): 1572--73. \url{https://doi.org/10.1093/bioinformatics/btq170}.

\end{CSLReferences}
\indent
\setlength{\parindent}{17pt}
\setlength{\leftskip}{0pt}
\setlength{\parskip}{0pt}

\newpage

\hypertarget{appendix}{%
\chapter*{Appendix}\label{appendix}}
\addcontentsline{toc}{chapter}{Appendix}

\hypertarget{figures}{%
\section*{Figures}\label{figures}}
\addcontentsline{toc}{section}{Figures}


\begin{figure}

{\centering \includegraphics[width=0.6\linewidth]{figures/graph} 

}

\caption{Analysis of ATACseq dataset of two trisomy 12 and two non-trisomy 12 untreated CLL PMBC samples. The volcano plot depicts change in TF activity (x axis) versus BH-adjusted p-values (y axis) for 636 TFs, comparing trisomy 12 and non-trisomy 12 samples. The (\protect\hyperlink{ref-diffTF}{\textbf{diffTF?}}){[}REFERENCE{]} package was ran in analytical mode to calculate TF activity, measured as weighted mean difference. TFs are labeled if adjusted p-value \textless{} 0.01 and absolute weighted mean difference \textgreater{} 0.15.}\label{fig:diffTFsmallvolPlot}
\end{figure}
\hypertarget{tables}{%
\section*{Tables}\label{tables}}
\addcontentsline{toc}{section}{Tables}
\begin{table}[ht]
    \begin{center}
        {\footnotesize
        \begin{tabular}{l|cccccccccc}
        \hline \hline
                        & 3m    & 6m    & 1yr   & 2yr   & 3yr   & 5yr   & 7yr   & 10yr  & 12yr  & 15yr   \\
            \hline
                Mean   & 3.138 & 3.191 & 3.307 & 3.544 & 3.756 & 4.093 & 4.354 & 4.621 & 4.741 & 4.878  \\
                Median & 3.013 & 3.109 & 3.228 & 3.490 & 3.680 & 3.906 & 4.117 & 4.420 & 4.575 & 4.759  \\
                Min    & 1.984 & 1.950 & 1.956 & 2.010 & 2.240 & 2.615 & 2.850 & 3.120 & 3.250 & 3.395  \\
                Max    & 5.211 & 5.274 & 5.415 & 5.583 & 5.698 & 5.805 & 5.900 & 6.031 & 6.150 & 6.295  \\
                StD    & 0.915 & 0.919 & 0.935 & 0.910 & 0.876 & 0.825 & 0.803 & 0.776 & 0.768 & 0.762  \\
            \hline \hline
        \end{tabular}}
    \end{center}
    \caption{Detailed descriptive statistics of location and dispersion for
    2100 observed swap rates for the period from
    February 15, 1999 to March 2, 2007. Swap rates measured as 3.12 (instead of 0.0312).}
    \label{tab:apptable}
\end{table}
\newpage

\end{document}
