# Introduction

```{r loadLibraries, echo = FALSE}
library(ggplot2)
```

## CLL
### Disease characteristics

\@ref(fig:CLLmorphology)  _Figure from @Kipps2017_. 

```{r CLLmorphology, fig.cap='(ref:CLLmorphology)', out.width="60%", echo = FALSE}

knitr::include_graphics("figures/CLL_morphology.eps")

```

\@ref(fig:proliferationCentres)  _Figure from @Kipps2017_. 

```{r proliferationCentres, fig.cap='(ref:proliferationCentres)', out.width="60%", echo = FALSE}

knitr::include_graphics("figures/proliferation_centres.eps")

```

### IGHV status and cell of origin
\@ref(fig:IGHVstatus) Taken from Kipps et al 2017: "Normal naive B cells that have undergone successful V(D)J recombination and express functional B cell receptors that are capable of binding to antigen interact with CD4+ T cells and accessory cells, which aggregate to form follicles that become germinal centres. Germinal cells each have a dark zone, comprising rapidly dividing B cells, and a light zone, comprising B cells mixed with follicular dendritic cells (FDCs), macrophages and helper T cells (TH cells). The B cells enter the dark zone of the germinal centre where they experience rapid proliferation and somatic hypermutation (SHM) in the genes encoding the immunoglobulin variable regions of the heavy chain (IGHV) and the light chain (IGVL). As they pass through to the light zone, the B cells that express the fittest B cell receptors for binding antigen are selected and may undergo immunoglobulin class-switch recombination. Chronic lymphocytic leukaemia (CLL) cells that use unmutated IGHV apparently originate from CD5+ B cells prior to experiencing SHM, whereas CLL cells that use mutated IGHV most likely originate from CD5+ B cells that have passed through and differentiated in the germinal centre. Some CLL cells might be derived from B cells that also have undergone immunoglobulin class-switch recombination and express immunoglobulin isotypes other than IgM and IgD, for example, IgG or IgA. Another subset is one with CLL cells that express immunoglobulin with only modest somatic mutations, such as CLL cells that use IGHV3-21 with ~97% homology to the inherited IGHV3-21 gene and an immunoglobulin light chain encoded by an unmutated IGLV3-21; these cells might derive from a B cell that has had constrained SHM, possibly owing to a limited need for immunoglobulin somatic diveresification and selection. Dashed arrows indicate speculated pathways." _Figure from @Kipps2017_. 

```{r IGHVstatus, fig.cap='(ref:IGHVstatus)', out.width="60%", echo = FALSE}

knitr::include_graphics("figures/IGHV_status.eps")

```



### The central role of BCR signalling

\@ref(fig:BCRsignalling) Adapted from: Kipps et al 2017: "B cell receptor (BCR) signalling is initiated by SRC-family kinase-dependent phosphorylation (mainly LYN) of CD79A and CD79B that creates a docking site for the binding and activation of spleen tyrosine kinase (SYK). SYK then triggers the formation of a multi-component ‘signalosome’, comprising Bruton tyrosine kinase (BTK), AKT, phosphoinositide 3-kinase (PI3K), phospholipase Cγ2 (PLCγ2) and B cell-linker protein (BLNK), among others. CD19 is a co-receptor for BCR and is important for PI3K activation, which recruits and activates PLCγ2, BTK and AKT. PLCγ2 generates diacylglycerol (DAG) and inositol-1,4,5-trisphosphate (Ins(1,4,5)P3), which triggers Ca2+ release from the endoplasmic reticulum, leading to the activation of the MEK–extracellular signal-regulated kinase (ERK) and nuclear factor-κB (NF-κB) signalling pathways. Other effects of BCR signalling include activation of mechanistic target of rapamycin complex 1 (mTORC1) and of Rho-family GTPases, RAC1 and RHOA, which can affect the cytoskeleton. Inhibitors of SYK, PI3K and BTK are shown. Note that this figure describes the main molecules and interactions that are involved in positive BCR signalling, but is not an exhaustive description of all signalling pathways or molecules activated. IKK, IκB kinase; PKC, protein kinase C."

```{r BCRsignalling, fig.cap='(ref:BCRsignalling)', out.width = "60%",  echo = FALSE}

knitr::include_graphics("figures/BCR_signalling.eps")

```



### Recurrent genetic features in CLL
\@ref(fig:CLLmutations) Taken from Kipps et al 2017: "Genes that are mutated in chronic lymphocytic leukaemia (CLL) are involved in several cellular pathways (blue boxes). As such, mutations in these genes can lead to a range of cellular consequences, such as aberrant DNA repair and B cell receptor (BCR) signalling, among others51,213. The minus sign from GBN1 to the MAPK–ERK pathway indicates negative regulation. *For more detail of the BCR and its associated signalling, see FIG. 3. ASXL1, additional sex combs-like protein 1; ATM, ataxia telangiectasia mutated; BAZ2A, bromodomain adjacent to zinc-finger domain protein 2A; BCOR, BCL-6 co-repressor; BIRC3, baculoviral IAP repeat-containing protein 3; BRCC3, BRCA1/BRCA2-containing complex subunit 3; C-NOTCH, carboxy-terminal domain of NOTCH; CARD11, caspase recruitment domain-containing protein 11; CHD2, chromodomainhelicase-DNA-binding protein 2; CHK2, checkpoint kinase 2; Co-A, co-activator; CSL, CBF1–Suppressor of Hairless–LAG1 (also known as RBPJ); DDX3X, ATP-dependent RNA helicase DDX3X; DYRK1A, dual-specificity tyrosine-phosphorylation-regulated kinase 1A; EGR2, early growth response 2; ELF4, ETS-related transcription factor Elf-4; ERK, extracellular signal-regulated kinase; EWSR1, Ewing sarcoma breakpoint region 1 protein; FBXW7, F-box/WD repeat-containing protein 7; FUBP1, far upstream element-binding protein 1; GNB1, guanine nucleotide-binding protein G(I)/G(S)/G(T) subunit β1; H3K4, histone H3 lysine 4; IC, intracellular domain; IKZF3, Ikaros family zinc-finger protein 3; IL-1R, IL-1 receptor; IRF4, interferon regulatory factor 4; ITPKB, inositol-trisphosphate 3-kinase B; LRP, low-density lipoprotein receptor-related protein; MAP2K1, dual-specificity mitogen-activated protein kinase kinase 1; MAPK, mitogen-activated protein kinase; MED12, Mediator of RNA polymerase II transcription subunit 12; MGA, MAX gene-associated protein; MYD88, myeloid differentiation primary response protein MyD88; NF-κB, nuclear factor-κB; NXF1, nuclear RNA export factor 1; P, phosphate; POT1, protection of telomeres protein 1; PTPN11, tyrosine-protein phosphatase non-receptor type 11; RIPK1, receptor-interacting serine/threonine-protein kinase 1; RPS15, 40S ribosomal protein S15; SAMHD1, SAM domain and HD domain-containing protein 1; SF3B1, splicing factor 3B subunit 1; SHP1, Src homology region 2 domain-containing phosphatase 1 (also known as PTPN6); SYK, spleen tyrosine kinase; TCF/LEF, T cell factor/lymphoid enhancer factor; TLR8, Toll-like receptor 8; TNFR1, tumour necrosis factor receptor 1 (also known as TNFRSF1A); TRAF, TNFR-associated factor; XPO, exportin; ZMYM3, zinc-finger MYM-type protein 3. Adapted with permission from REF. 51, Macmillan Publishers Limited." _Figure from @Kipps2017_

```{r CLLmutations, fig.cap='(ref:CLLmutations)', out.width="60%", echo = FALSE}

knitr::include_graphics("figures/genetic_mutations.eps")

```

### The incompletely understood role of trisomy12 {#intro-trisomy12}

### Stratification of CLL patients

\@ref(fig:CLLstratification) Taken from Kipps et al 2017: "Indications for therapy of patients with chronic lymphocytic leukaemia (CLL) include late-stage disease, evidence for rapid disease progression or disease-related symptoms. Patients with del(17p) or mutated TP53 should be treated with therapy that does not require functional TP53, such as ibrutinib (a Bruton tyrosine kinase (BTK) inhibitor), given the relatively poor outcome for such patients with chemotherapy. For patients without del(17p) or known mutations in TP53, immunoglobulin heavy-chain variable region (IGHV) mutational status can help to define the treatment strategy; patients with unmutated IGHV could be considered for therapy with a BTK inhibitor (such as ibrutinib) and patients with mutated IGHV might be good candidates for chemoimmunotherapy (CIT), if amenable. Indeed, patients with mutated IGHV can have excellent outcomes with CIT regimens, such as fludarabine, cyclophosphamide and rituximab, with >50% of patients having a median progression-free survival of >10 years, including the potential for cure. If the patient is amenable to CIT, age, medical comorbidities and myeloid reserve should be taken into consideration. Patients >65 years of age commonly have medical comorbidities and are less able to tolerate myelosuppressive regimens, such as fludarabine, cyclophosphamide and rituximab. Thus, considerations should be given to using reduced dose or less myelosuppressive chemotherapy regimens, such as chlorambucil or reduced-dose bendamustine and an anti-CD20 monoclonal antibody for patients with limited myeloid reserve. Patients who either do not respond, have a poor tolerance to CIT or relapse following CIT, should be re-evaluated for del(17p) or TP53 mutations. Patients who develop de novo del(17p) or TP53 mutations, or have known del(17p) and/or TP53 mutations, or who develop resistance or intolerance to ibrutinib, could be considered for therapy with idelalisib and rituximab or the BCL-2 inhibitor venetoclax. Patients treated with CIT who do not have del(17p) or TP53 mutations could be considered for repeat CIT if their progression-free survival after CIT is >2 years and the patient has sufficient myeloid reserve. Such patients also might be treated with a BTK inhibitor or a phosphoinositide 3-kinase (PI3K) inhibitor, which also could be considered for patients who develop intolerance or resistance to therapy with ibrutinib. Patients who develop resistance or intolerance to inhibitors of BTK, PI3K and/or BCL-2 should be considered for clinical trials or alternative agents. LDT, lymphocyte doubling time." _Figure from @Kipps2017 _.

```{r CLLstratification, fig.cap='(ref:CLLstratification)', out.width = '60%', echo = FALSE}

knitr::include_graphics("figures/disease_stratification.eps")

```


## The tumour microenvironment in CLL 

### The concept of the protective niche 


\@ref(fig:microenvironmentTissues) _Adapted from Davids et al 2012_

```{r microenvironmentTissues, fig.cap='(ref:microenvironmentTissues)',out.width = "60%",  echo = FALSE}

knitr::include_graphics("figures/microenvironmentTissues.eps")

```

### Microenvironmental pathways and cross-talk {#intro-ME-pathways}

\@ref(fig:microenvironmentOverview) _Adapted from: Weistner et al 2015_

```{r microenvironmentCrosstalk, fig.cap='(ref:microenvironmentOverview)', out.width = "60%", echo = FALSE}

knitr::include_graphics("figures/microenvironmentCrosstalk.eps")

```

\@ref(fig:microenvironmentSignalling) Taken from Kipps et al 2017; "Migration of chronic lymphocytic leukaemia (CLL) cells into the lyphoid tissu is primarily mediated through CXC-chemokine receptor 4 (CXCR4) in response to CXC-chemokine ligand 12 (CXCL12), which is secreted mainly by nurse-like cells (NLCs) and mesenchymal-derived stromal cells. Migration of CLL cells into lymph nodes also occurs via CC-chemokine receptor 7 (CCR7) in response to CC-chemokine ligand 19 (CCL19) and CCL21, which are produced by the endothelial cells of high endothelial venules (HEVs). HEV endothelial cells also express hyaluronan, which can interact with CD44, to facilitate B cell signalling and might enhance the production of active matrix metalloproteinase 9 (MMP9). Once in tissues, several chemokines promote B cell survival, including CXCL12, B cell-activating factor (BAFF; also known as TNFSF13B) and a proliferation-inducing ligand (APRIL; also known as TNFSF13). In addition, CLL cell survival can be promoted through cognate interactions between CD31 and CD38, and the production by stromal cells of WNT factors, which can interact with ROR1, ROR2 and/or various Frizzled receptors. CLL cell contact with mesenchymal stromal cells can also be established through vascular cell adhesion protein 1 (VCAM1)– α4β 1 integrin interactions that contribute to CLL cell survival. In turn, CLL cells can secrete chemokines, such as CCL3 and CCL4, which can recruit T cells and NLC-precursor cells (monocytes) to the CLL microenvironment. Activated T cells can provide CLL cells with proliferative signals through CD40 ligand (CD40L)-CD40 interactions and the secretion of several cytokines, such as IL-2, IL-4 and IL-10. In turn, activated CLL cells secrete CCL12 and CCL22, which attract more T cells into the CLL microenvironment. In tissues, CLL cells can be exposed to environmental and/or self-antigens that might trigger B cell activation through interactions with the surface immunoglobulin; this could amplify the responsiveness of CLL cells to the signals and factors that are provided by the CLL microenvironment. BAFFR, BAFF receptor (also known as TNFRSF13C); BCMA, B cell maturation protein (also known as TNFRSF17); BCR, B cell receptor; TACI, transmembrane activator and CAML interactor (also known as TNFRSF13B). _Figure from @Kipps2017 ._ "

```{r microenvironmentSignalling, fig.cap='(ref:microenvironmentSignalling)', out.width = "60%", echo = FALSE}

knitr::include_graphics("figures/microenivironmental_signalling.eps")

```


### The impact  of the microenvironment on drug response

### Modelling the tumour microenvironment


## Therapies in CLL 

### New Targeted therapies

### Drug resistances in CLL
incl ME -mediated and ibrutinib - IL4 

### BET inhibitors, and their potential for the treatment of leukemia

## Background to the approaches used in this thesis
### Ex-vivo drug sensitivity screens

### Multiomics datasets to study CLL 
The power of multi-omics approaches especially the use of ATAC sequencing and combining ATAC and RNA
ATAC sequncing as a proxy for TF activity justification {#intro-ATACseq}
PUt with section on this? 

### CLL as a model for studying tumour biology

### Measurement of outcomes? 
PUt with section on this? 
 
### Mathematical modelling {#intro-multivariate-modelling}


## Aims of this thesis

## Overview of the approach

## Note on contributions


##Reading list before defence0: 


