# Introduction {#introduction}

```{r loadLibraries, echo = FALSE}
library(ggplot2)
```

## Chronic Lymphocytic Leukaemia (CLL)
### Disease characteristics {#intro-disease-characteristics}
Chronic lymphocytic leukaemia (CLL) is a malignancy of mature B cells, characterised by the progressive accumulation of malignant lymphocytes in the blood, bone marrow and lymph nodes [@Kipps2017]. The malignant cells can be distinguished by expression of CD5, CD19, and CD23, and by lower levels of membrane IgM, IgD, and CD79B [@Matutes1994; @Moreau1997; @Chiorazzi2005] (Figure \@ref(fig:CLLmorphology)).

The disease pathogenesis is driven by multiple factors, including molecular features, signalling via the B cell receptor (BCR) and interactions with non-neoplastic cells within the lymphoid tissues (known as the tumour microenvironment) [@Kipps2017].  Whilst the majority of CLL cells are in a resting state [@Messmer2005; @Defoiche2008],  evidence suggests that CLL cells migrate towards lymph nodes where they form proliferation centres [@Granziero2001; @Messmer2005], similar to germinal centres in healthy lymph nodes, which can show a daily birth rate of up to 3.3% of the tumour [@Herndon2017]. 

(ref:CLLmorphology) Wright–Giemsa-stained blood smear of a CLL malignant B cell, depicting typical morphology. _Figure reproduced with permission from @Kipps2017 _. 

```{r CLLmorphology, fig.cap='(ref:CLLmorphology)', out.width="40%", echo = FALSE}

knitr::include_graphics("figures/CLL_morphology.eps")

```

CLL is the most common leukaemia in the West, accounting for 37% of leukaemia cases and ~19,000 newly detected cancers in the US in 2016 [@Kipps2017;@Dubois2020]. The risk of developing CLL is twice as high for men than women, and more likely with increasing age [@Siegel2012; @Nabhan2014; @Li2015; @Pulte2015]. Chemotherapy [@Robak2005; @Chang2012; @Lukenbill2013] and chemoimmunotherapy [@Hallek2010; @Goede2014; @Hillmen2015] has been the mainstay of therapy for many years. More recently, the central role of BCR signalling in disease pathogenesis has been appreciated and new drugs targeting this pathway have improved patient outcomes.  The disease course is highly heterogeneous, and the overall 5-years relative survival for CLL is 84% [@Miller2019]. 

In spite of this, CLL is widely considered to be incurable [@Bosch2019], and treatment regimens induce significant side-effects. Moreover, the disease is characterised by its clinical heterogeneity [@Miller2019] and there is significant variation in disease progression amongst patients. Some patients harbour indolent disease, and can be followed by a watch-and-wait approach, sometimes going decades without requiring treatment. Others may need immediate treatment and survival can be short [@Miller2019]. There is a clinical need to understand the underlying biology of the disease, with a view to understanding the causes of this clinical heterogeneity and to improving patient outcomes and experiences. 

### Cell of Origin {#intro-cell-of-origin}
Identifying the cell of origin of a cancer can help explain disease pathogenesis, and understand the different subtypes, and their associated prognosis. Various studies have established mature CD5+ B cells as the cell of origin in CLL  [@Bosch2019]. 

In healthy tissues, these mature B cells are derived from haematopoietic stem cells in the bone marrow [@Kondo2010; @Fischer2020]. These stem cells develop in multiple stages, where each stage is defined by rearrangements within the immunoglobulin heavy chain and light chain loci [@Pelanda2012] that encode components of the BCR. The aim of this process is to generate a large BCR repertoire, capable of recognising a range of foreign antigens. 

Developing B cells in the bone marrow undergo positive and negative selection, ensuring that each BCR binds effectively to foreign antigen, whilst eliminating those that bind strongly to self-antigen [@Lebien2008; @Martensson2010]. These immature B cells then migrate to the lymph nodes and spleen where they differentiate into mature cells that are considered to be antigen “naive” [@Chung2003].

B cells are later activated when they encounter their respective antigen. When activated, the B cells form germinal centres, which are specialised microenvironments within the lymph node that facilitate extensive proliferation. Here they undergo a process called affinity maturation, in which the loci encoding components of the BCR undergo somatic hypermutation to optimise BCR antigen specificity [@Shlomchik2012]. This process generates short-lived plasmablasts which provide immediate protection, along with plasma cells and memory B cells procuring longer-term immunity [@Nutt2015]. 

### IGHV status {#intro-ighv-status}
The two major subtypes of CLL can be defined by whether the B cell of origin has undergone this process of somatic hypermutation within a germinal centre. This is reflected in the degree of mutation of the immunoglobulin heavy chain variable region (IGHV) genes. Evidence suggests that IGHV-mutated (IGHV-M) CLLs (60% of cases) are derived from antigen-experienced B cells. On the other hand, there is continued debate as to whether IGHV-unmutated (IGHV-U) CLLs derivate from naive B cells, or germinal centre, antigen-experienced B cells [@Klein2001]. 

The two subtypes show distinct clinical and molecular properties. In contrast to IGHV-M CLLs, IGHV-U CLLs have a higher proportion of high-risk genetic lesions, more commonly undergo clonal evolution and consequently have a shorter time to first treatment (TTFT) and less favourable overall survival (OS) [@Hamblin1999; @Damle1999; @Landau2013; @Puente2015; @Bosch2019]. Their differential antigen experience also affects how they respond to signals from the microenvironment, including stimulation of the BCR. 

### The central role of BCR signalling {#intro-bcr-signalling}
The BCR pathway is central to the process of selection, development, proliferation and survival of CLL clones [@Chiorazzi2003; @Stevenson2004; @Agathangelidis2012; @Burger2013; @Iacovelli2015; @Dubois2020]. Evidence of the importance of this pathway comes from the observation that BCRs of CLL cells show a biased selection of IGHV and immunoglobulin light chain variable region (IGLV) $\kappa$/$\lambda$ genes. This generates BCRs that are remarkably similar across patients [@Kipps1989; @Fais1998; @Widhopf2004; @Messmer2004; @Stamatopoulos2007],  suggesting that BCR binding to certain antigens may drive selection and proliferation of CLL clones. Other compelling studies have shown that BCR signalling genes are upregulated in CLL cells taken from lymph nodes [@Herishanu2011] and that cells from IGHV-U CLL patients (with poorer outcomes), also show activation of BCR-related genes [@Rosenwald2001]. The success of BCR inhibitors in the clinic [@Byrd2013], also underlines the critical importance of this pathway in CLL.

Figure \@ref(fig:BCRsignalling) shows a schematic of the BCR pathway itself. The BCR is a multimeric complex consisting of surface immunoglobulin (Ig), which recognises antigen, plus the Ig-$\alpha$/Ig-$\beta$ hetero-dimers, known as CD79A and CD79B. The BCR may bind to external antigens within the microenvironment [@Binder2010] or intra-BCR self-antigens [@Minden2012], which in turn recruits SYK and the Src kinase LYN. These kinases phosphorylate motifs located on the cytoplasmic tails of CD79A and CD79B, initiating a signalling cascade involving a number of proteins and pathways. These include the proteins BTK [@Herman2011] and PI3K [@Longo2007] which activate a number of downstream pathways and players including PLC$\gamma$ 2, calcium signalling, PKC, NF$\kappa$B signalling, ERK and MAPKs, and nuclear transcription [@Burger2013; @Dubois2020]. 

There are two types of BCR signalling in healthy and CLL B cells: ligand-dependent “active” signalling that relies on antigen-binding, and ligand-independent “tonic” signalling [@Lam1997; @Kraus2004]. Whilst active signalling engages the entire signalling cascade described above, tonic signalling activates only a subset of these. @Kraus2004 showed that tonic signalling is important for prolonged B cell survival whereby PI3K signalling is thought to play a key role in delivering survival signals [@Srinivasan2009; @Pula2019]. Both these modes of BCR signalling are believed to impact of the survival and growth of the tumour, although the dominant mode remains a matter of debate [@Burger2013].

The different clinical properties of IGHV-M and IGHV-U CLLs are believed to be determined by in part by their differential response to BCR stimulation, which is influenced by their cell of origin [@Burger2013]. IGHV-U CLLs have not undergone somatic hypermutation and consequently express low-affinity BCRs that are frequently activated by numerous antigens and auto-antigens in the microenvironment [@Borche1990; @Broker1988; @Sthoeger1989; @Herve2005; @Lanemo2008; @Chu2008; @Binder2010; @Krysov2010; @Kostareli2012]. In contrast, IGHV-M BCRs only recognise highly specific antigens, which either occur infrequently or induce anergy due to high-affinity binding [@Chiorazzi2003; @Stevenson2004; @Chiorazzi2005; @Burger2013]. IGHV-M CLL clones therefore are more stable and expand at a slower rate. 

(ref:BCRsignalling) Graphical depiction of the BCR pathway. _Figure reproduced with permission from @Kipps2017 _

```{r BCRsignalling, fig.cap='(ref:BCRsignalling)', out.width = "60%",  echo = FALSE}

knitr::include_graphics("figures/BCR_signalling.eps")

```



### Recurrent genetic features in CLL {#intro-genetic-features}
The genomic landscape of CLL has been thoroughly characterised in a number of studies, including two seminal papers involving >500 CLL samples [@Landau2015; @Puente2015]. Such studies have indicated that CLL shows a lower mutational load than other lymphoid neoplasms [@Vogelstein2013; @Alexandrov2013; @Bosch2019], in which a relatively large number of genes are rarely mutated [@Fabbri2016]. A small number of driver genes are mutated in a significant proportion of cases, though a common genetic event which can accounts for most cases of CLL has not been identified [@Fabbri2016]. These genetic alterations encompass chromosomal alterations, mutations, alterations in miRNA expression and epigenetic modifications [@Kipps2017].


#### Somatic Mutations {#intro-mutations}
There are many recurrent somatic mutations in CLL, and these centre on several major pathways and functions that are frequently altered (Figure \@ref(fig:CLLmutations)) [@Puente2015; @Fabbri2016; @Kipps2017]. These pathways include Notch signalling, DNA damage response, RNA processing, NF$\kappa$B signalling, BCR signalling, WNT signalling and chromatin modification [@Kipps2017; @Bosch2019]. 

Within these pathways, most mutations are rare and only a few occur at a frequency >5% [@Bosch2019]. One landmark study indicated the most frequent mutations occur in *NOTCH1* (12.6% of cases), *ATM* (11%), *BIRC3* (8.8%) and *SF3B1* (8.6%), although these frequencies are dependent disease stage and treatment status [@Puente2015]. The functional role and prognostic importance of a number of these putative driver mutations has been established, and are discussed below. 

**Notch signalling**
The Notch pathway activates genes required for proliferation, metabolism and survival, including *MYC*, via the activation of the NOTCH1 transmembrane receptor [@Guruharsha2012]. Mutations in *NOTCH1* are very common in CLL (~4-20%) [@Fabbri2011; @Puente2011; @Landau2013; @Puente2015; @Landau2015]. A number of other recurrent mutations also centre on Notch deregulation, indicating that this pathway is disrupted in many CLL cases [@Fabbri2016]. *NOTCH1* mutations occur more often in IGVH-U CLLs and are associated with a less favourable OS [@Fabbri2011]. *NOTCH1* mutants also respond less-well to anti-CD20-based therapies, due to decreased surface expression of CD20 [@Fabbri2011; @Rossi2012]. 

**DNA Damage Response**
A number of frequently occurring mutations disrupt the DNA damage response, the most clinically important of which is *TP53*. *TP53* is known as the “guardian of the genome”, owing to its role as a tumour suppressor gene protecting genome integrity by preventing mutation. Despite being a tumour suppressor, *TP53* mutations can have a dominant negative effect on function, such that loss of genomic stability occurs even when a single allele is mutated [@Zenz2008].  *TP53* is an important prognostic marker, as dysregulation of its function is associated with resistance to DNA-damaging agents (chemotherapy and radiotherapy) [@Zenz2008; @Dicker2009; @Rossi2009], and cases with *TP53* mutations at diagnosis often show shorter TTFT and a less favourable OS [@Dohner2000; @Zenz2008; @Rossi2013; @Puente2015]. 

Upstream of *TP53*, the tumour suppressor *ATM* is also frequently mutated in CLL [@Bosch2019]. *ATM* activates the DNA damage response upon recognition of DNA double strand breaks [@Austen2005; @Shiloh2013]. Similarly to *TP53* mutants, *ATM*-disrupted CLLs also show genomic instability and these mutations are associated with shorter TTFT and OS, and chemoresistance [@Austen2005; @Stankovic2014]. 

*POT1* is also involved in genomic stability, and is mutated in around 3 – 7% of cases [@Bosch2019; @Ramsay2013b]. *POT1* mutations disrupt telomere protection, leading to a increased in structural aberrations and chromosomal breaks [@Ramsay2013b]. Mutations in *POT1* are associated with IGHV-U CLL and advanced clinical stage [@Ramsay2013b].

**RNA processing**
30% of CLL tumours harbour mutations disrupting RNA processing and the spliceosome machinery [@Puente2011; @Wang2011; @Quesada2012; @Puente2015; @Fabbri2016].
*SF3B1* mutations are the most frequent to occur, and are found in ∼10% of cases, usually in IGHV-U CLLs [ @Puente2011; @Wang2011; @Quesada2012; @Puente2015; @Fabbri2016]. The *SF3B1* gene encodes part of the U2 snRNP complex which is involved in RNA splicing [@Shin2004], though the functional implication of *SF3B1* mutations are yet to be established, and many transcripts show abnormal splicing in*SF3B1* cases [@Quesada2012]. The presence of *SF3B1* mutations is associated with a decreased TTFT and unfavourable OS [@Bosch2019].

**NF$\kappa$B signalling**
A wide range of mutations across various pathways converge on the activation of NF$\kappa$B [@Fabbri2011; @Puente2011; @Wang2011; @Quesada2012; @Puente2015; @Landau2015], including *BIRC3* and *MYD88*. For example, certain mutations in *MYD88* result in increased binding to IRAK1 and higher activation of NF$\kappa$B [@Puente2011]. However, the role and prognostic importance of NF$\kappa$B activation in CLL is still unclear [@Bosch2019].

(ref:CLLmutations)  Graphical depiction of commonly mutated genes in CLL, grouped into cellular pathways.  Minus sign indicates negative regulation. _Figure reproduced with permission from @Kipps2017 ._

```{r CLLmutations, fig.cap='(ref:CLLmutations)', out.width="100%", echo = FALSE}

knitr::include_graphics("figures/genetic_mutations.eps")

```

#### Structural Aberrations {#intro-structural-abberations}
In addition to the aforementioned mutations, a number of common structural aberrations confer similar disruption to normal B cell function. The key structural aberrations and their prognostic value were set out in a landmark paper by @Dohner2000, as follows. 

**del(11q)**
Deletions of chromosome 11q (del(11q)) are fairly common (~10%) in CLL, and are believed to target the *ATM* gene in the 11q22-23 region [@Dohner2000; @Austen2005; @Bosch2019]. del(11q) is usually monoallelic, but can also be associated with mutations in the remaining ATM allele (~30% of cases) [@Austen2005]. In certain instances, the deleted region does not include *ATM* but rather *BIRC3*, a negative regulator of the NF$\kappa$B pathway [@Rossi2012]. Cases with del(11q) or *ATM* lesions have a shorter TTFT and OS, especially if the lesion is biallelic [@Austen2005; @Skowronska2012; @Stankovic2014; @Nadeu2016]. 

**del(17p)**
Deletion of chromosomal region 17p13 (del(17p)) is found in 1 – 20% of cases, depending on the stage of the disease and most common in chemo-refractory cases [@Dohner2000; @Zenz2008; @Zenz2010; @Stilgenbauer2014]. The target of this lesion is thought to be *TP53*; the deleted region consistently includes the *TP53* locus [@Dohner1995], and around 80% del(17p) cases also have missense mutations in the second *TP53* allele [@Zenz2008; @Gonzalez2011; @Trbusek2011]. Del(17p) CLLs show increased genomic instability [@Yu2017], resistance to chemotherapy and, correspondingly, a shorter TTFT and a less favourable OS [@Dohner2000; @Zenz2008; @Rossi2013; @Puente2015]. 
   
**del(13q)**
Deletion in the 13q14 region (del(13q)) is the most common genetic lesion in CLL (~50–60% of cases) [@Dohner2000]. Experiments to determine the minimal deleted region identified that this invariably contains *DLEU1* and *DLEU2*, two long non-coding RNA genes, and the microRNA gene cluster *MIR15A–MIR16-1* [@Kalachikov1997; @Migliazza2001; @Calin2002;  @Palamarchuk2010]. *in vitro* studies have demonstrated the role of these genes in regulation of the cell cycle and apoptosis [@Cimmino2005; @Bosch2019]. In some CLL cases, del(13q) is the sole genetic abnormality, indicating that this lesion may be involved in early CLL development [@Dohner2000;  @Landau2015]. Moreover, conditional deletion of the equivalent minimal deleted region in mice recapitulated CLL initiation and progression, and these mice developed clonal lymphoproliferations [@Migliazza2001].  del(13q) CLLs have the best prognosis, with prolonged TTFT, and OS compared to tumours with other lesions [@Dohner2000; @Rossi2013]. 

#### The incompletely understood role of trisomy12 {#intro-trisomy12}
Complete duplication of chromosome 12 (trisomy 12) is observed in ~15% of CLL cases at diagnosis [@Dohner2000]. Despite its recurrence, there is currently no functional explanation for this lesion [@Bosch2019], although a number of features have been observed. Trisomy 12 is more common in IGHV-M CLLs than IGHV-U [@Hamblin1999], and is thought to confer an abnormal cellular morphology [@Bosch2019]. Previous work in our lab has also demonstrated that trisomy 12 CLLs show a specific signalling signature and distinct transcriptomic [@JCIpaper] and proteomic profiles [@HerbstThesis; @MeierAbt2021], including differential expression of genes within the BCR, PI3K, AKT, and mTOR signaling and chemokine signaling pathways. Moreover, trisomy 12 CLLs show higher sensitivity to BCR inhibitors, indicating that BCR signalling may be amplified in these cases [@JCIpaper]. 

Traditionally trisomy 12 has been classified as an intermediate-risk lesion [@Dohner2000]: these cases have a higher proliferative capacity, but are more treatable with chemotherapeutics and BCR inhibitors. However, *NOTCH1* mutations are frequently observed in trisomy 12 cases, and this is associated with poorer outcomes [@Balatti2012; @Giudice2012]. 
  
#### Epigenetic alterations {#intro-epigenetic-alterations}
In addition to genetic lesions, the epigenome is also modified in CLL and samples typically show global hypomethylation combined with local hypermethylation [@Wahlfors1992; @Cahill2013; @Ziller2013; @Kipps2017; @Bosch2019]. Studies investigating the epigenome in CLL have proven revealing, in particular, higher levels of intra-sample methylation heterogeneity have been shown to be associated with high-risk genetic lesions and poorer prognosis [@Landau2014]. 

Moreover, methylation signatures have been used to classify distinct clinical CLL subgroups [@Kulis2012; @Bhoi2016], as they are useful to trace the cell of origin. For example, CLL cells from distinct patients originate from many different B cell maturation states, possibly reflecting the biological and phenotypic heterogeneity of CLL [@Oakes2016]. Accordingly, IGHV-U CLLs have a distinct methylation signature to IGHV-M CLLs, and these patterns correspond approximately to those of pre-germinal centre or post-germinal centre memory B cells, respectively [@Kulis2012; @Oakes2016]. Epigenetic studies have also revealed the relationship between certain genetic lesions and specific epigenetic signatures, for example *MYD88* mutations and trisomy 12 [@Beekman2018].  The CLL epigenome can also be modulated by drugs and thus is of increasing clinical interest [@Timp2013; @Beekman2018; @Gaiti2019].

## Therapies in CLL {#intro-therapies}
Extensive work to uncover the molecular drivers of CLL has led to the development of several therapeutic strategies. CLL represents a successful example of how developing a complex understanding of the biological characteristics of a disease can lead to significantly improved patient outcomes [@Yosifov2019]. 
Treatment of CLL patients can be via chemotherapy, chemoimmunotherapy or targeted therapies that inhibit specific pathways [@Kipps2017; @Burger2020]. Additionally, allogeneic stem cell transplantation is sometimes considered [@Kipps2017].  

### Chemotherapy and chemoimmunotherapy {#intro-chemotherapy}
Chemotherapy has been the standard of care for CLL for many decades, either with purine analogues such as fludarabine or alkylating agents such as chlorambucil [@Robak2005; @Lukenbill2013]. Chemoimmunotherapy has also benefited many patients owing to rapid improvements to monoclonal antibody technology and the development of anti-CD20 treatments, such as rituximab [@Yosifov2019; @Robak2010]. However, patients with higher risk lesions such as *TP53* and del(17p) do not respond well to chemoimmunotherapy, and require alternative therapeutic options [@Zenz2010].

### Targeted therapies {#intro-targeted-therapies}
More recently, the importance of BCR signalling and upregulation of anti-apoptotic proteins in CLL expansion has been increasingly realised and led to the development of therapies targeting these pathways. Drugs targeting BCR signalling and BCL-2 have changed the treatment landscape dramatically [@Scheffold2020]. Three main drug classes that target BCR signalling have been developed for CLL: BTK inhibitors, PI3K inhibitors and SYK inhibitors [@Rooij2012; @Kipps2017]. 

Ibrutinib is a BTK inhibitor approved for use as an initial therapy and for patients who are refractory to chemoimmunotherapy [@Byrd2013; @Byrd2014]. Despite such success, complete remission is rare and many patients continue to harbour minimal residual disease within the bone marrow [@Byrd2013], requiring continued therapy for years [@Byrd2013; @Woyach2015]. Resistance can also occur via the acquisition of mutations in BTK or PLC$\gamma$ 2 [@Woyach2015] genes. Treatment initiation with ibrutinib is associated with a concomitant increase in the absolute lymphocyte count in the blood [@Woyach2014], thought to caused by the inhibition of chemokine receptor signalling leading to the release of malignant B cells from the lymph nodes into the peripheral blood. 

Idelalisib also acts to inhibits BCR signalling and chemokine signalling [@Hoellenriegel2011], via inhibition of PI3K. Whilst the drug is highly efficacious in CLL [@Furman2014; @Brown2014], idelalisib demonstrates more toxicities and lower efficacy than BTK inhibitors and thus is generally used as an alternative therapy in patients for whom BTK inhibitors are unsuitable [@Ghia2020; @Burger2020].

Other drugs targeting SYK, downstream of BTK, have also shown promise in Phase I/II clinical trials [@Friedberg2010], though none are licenced as yet.

Aside from BCR inhibitors, BCL-2 inhibitors such as venetoclax, have also shown good efficacy in CLL. Venetoclax is thought to induce apoptosis in CLL cells, by acting as a BH3 mimetic interfering with the ability of BCL-2 to sequester BIM [@Moore2007]. 

CLL is thus a manageable disease, with a well-established arsenal of treatments and associated management algorithm. However, treatments are harsh, and the disease is still considered incurable [@Bosch2019]. Many cases develop resistance to therapy, both via acquired mutations and through survival signals provided by the microenvironment (see Section \@ref(intro-microenvironment-drug-response)) and minimal residual disease within the bone marrow is common. Collectively, these lead to relapse or required prolonged therapy and its associated toxicities. There remains a clinical need to improve patient outcomes and experience.

## The tumour microenvironment in CLL {#intro-tumour-microenvironment}

### The role of the tumour microenvironment in CLL {#intro-tumour-microenvironment-role}
In addition to genetic aberrations, the tumour microenvironment is an important driver of disease pathogenesis in CLL [@tenHacken2016]. The term microenvironment encompasses the set of non-neoplastic cells within the lymphoid tissues, including the bone marrow and lymph nodes, that provide survival signals to the tumour, leading to clonal expansion and drug resistance [@Burger2014].  Malignant B cells engage in a dialogue with the non-neoplastic cells, via cell-cell contacts and soluble factors, including chemokines, integrins, cytokines and survival factors, centring on a number of important pathways including BCR signalling and tissue homing chemokine receptors [@tenHacken2016].

The importance of the microenvironment in CLL pathogenesis was first recognised in studies that showed CLL cells rapidly undergo apoptosis *in vitro*, whilst their survival can be extended by stimulation or by co-culture with nurselike cells (NLCs) or mesenchymal bone marrow stromal cells (BMSCs) [@Collins1989; @Burger2000; @Kurtova2009; @Deaglio2009; @Purroy2015]. This observation indicated that the ability of CLL to progressively accumulate *in vivo* may be highly dependent on external stimulation, rather than some cell-intrinsic feature of the tumour. Building on these observations, further *in vitro* studies have shown a number of cell types and soluble factors belonging to the microenvironment are also capable of protecting the tumour cells from drugs and chemotherapeutic agents. Many CLL patients continue to harbour minimal residual disease (MRD), in which a fraction of the malignant cells remain whilst the patient is in remission and eventually lead to relapse [@Hayden2012].  It is believed that the tumour microenvironment provides a sanctuary for the malignant B cells to shield from the effects of therapy [@Dubois2020]. 


### Components of the tumour microenvironment {#intro-TM-components}
**Lymph Nodes**
Over the last two decades, significant progress has been made in unravelling this complex cross-talk and many of the important cellular and molecular components have been defined and studied. Malignant B cells circulate through the blood in a resting state, and follow chemokine gradients towards the lymph nodes to form "profileration centres" (Figure \@ref(fig:proliferationCentres)), similar to germinal centres [@Herishanu2011]. Studies using detuerated water labelling have shown that up to 3% of the clone is actively proliferating within the lymph node [@Messmer2005; @Herndon2017]. Cross-talk with the non-neoplastic cells in the lymph node  shapes the transcriptomic profile of the malignant B cells, and leads to upregulation of the BCR pathway [@Herishanu2011, @Mittal2014], a central driver of CLL pathogenesis. 

(ref:proliferationCentres) Haemotoxylin and eosin stain of CLL-infiltrated lymph node tissue section, showing pale-staining profileration centres (circled). _Figure reproduced with permission from @Kipps2017 _. 

```{r proliferationCentres, fig.cap='(ref:proliferationCentres)', out.width="50%", echo = FALSE}

knitr::include_graphics("figures/proliferation_centres.eps")

```

**Bone marrow**
The bone marrow is also known to be important, and several studies have shown *ex vivo* BMSCs to protect CLL cells against the drug toxicity  [@Kay2007; @Kurtova2009]. However, gene expression changes are less pronounced within the bone marrow compared to the lymph node [@Herishanu2011].

**Cellular components**
Within these compartments, the CLL cells engage in a dialogue with mesenchymal stromal cells (MSCs), NLCs and follicular dendritic cells (FDCs), in concert with T cells, natural killer cells (NK cells) and components of the extracellular matrix [@tenHacken2016]. NLCs are of monocytic origin: their critical role in CLL was first demonstrated by the observation that peripheral blood-derived monocytes differentiate into NLCs, and that these cells prolong CLL cell survival *ex vivo* [@Burger2000]. They are also found within the lymphoid tissues of CLL patients [@Tsukada2002; @Burkle2007]. MSCs, which include BMSCs, are frequently observed within the secondary lymphatic tissues of CLL patients. Many studies have demonstrated the ability of these cells to inhibit spontaneous and drug-induced apoptosis in *in vitro* CLL co-cultures [@Panayiotidis1996; @Lagneaux1998, @Kurtova2009; @Stamatopoulos2010; @Stamatopoulos2012]. 

FDCs are important for tissue homing and retention of CLL cells within tissues. In healthy tissues, they are usually found within germinal centres [@Allen2008], and present unprocessed antigen to B cells. In CLL they play an important role within the secondary lymphoid organs, having a protective effect on CLL via cytokine secreton, adhesion molecules and the activation of BCR signalling [@Dubois2020]. CLL co-culture with FDCs leads to inhibition of spontaneous apoptosis and upregulation of anti-apoptotic MCL-1 [@Pedersen2002]. 

The T cell compartment is also altered in CLL, first described by @Scrivener2003. T cells have been observed to have pro-tumour and anti-tumour behaviour. On the one hand,  higher numbers of CD4+ T-helper (Th) cells are seen in CLL patient blood samples [@Palma2017; @Elston2020], and in line with this, Th cell cytokines have been shown to provide pro-survival signals *in vitro*, for example IL4 from Th2 cells [@Dancescu1992; @Bhattacharya2015; @AguilarHernandez2016]. Activated CD4+ T-cells in murine xenograft models [@Bagnara2011; @Os2013] of CLL have also been shown to increase survival and growth of the tumour. On the other hand, there is evidence of increased antigen-experienced CD8+ T cells in CLL, which control tumour growth in a CLL mouse model [@Roessner2020; @Grioni2021].


### Microenvironmental pathways {#intro-microenvironmental-pathways}
Cross-talk between these non-neoplastic cells and the malignant B cells can occur directly, via cell-cell contacts and adhesion molecules, indirectly, via soluble factors that bind to receptors on the CLL cells, or through the exchange of material held in extracellular vesicles [@Guarini2008; @Oppezzo2013; @Crompot2017]. Collectively these induce pathway activation (most importantly BCR and NF$\kappa$B [@Herishanu2011]) and gene expression modifications with the CLL cells, leading to chemotaxis, homing to lymphoid tissues and survival of the tumour cells [@Dubois2020]. Figure \@ref(fig:microenvironmentSignalling) depicts an overview of this cross-talk.

(ref:microenvironmentSignalling) Graphic summarising soluble factors and cell-cell contacts involved in cross-talk between CLL cells and non-neoplastic cells of the tumour microenvironment. _Figure reproduced with permission from @Kipps2017 ._ 

```{r microenvironmentSignalling, fig.cap='(ref:microenvironmentSignalling)', out.width = "80%", echo = FALSE}

knitr::include_graphics("figures/microenivironmental_signalling.eps")

```

**Cell-cell contacts**
The importance of direct contact between cells became clear with observations that the ability of MSCs to provide efficient rescue from spontaneous and drug-induced apoptosis is dependent on direct contact and can be blocked by separation through a filter [@Lagneaux1998; @Burger2000; @Kay2007; @Kurtova2009; @Ding2009]. Likewise, FDCs also operate through direct contact, as evidenced by the observation that contact with HK cells (an FDC cell line) protects against CLL cells from apoptosis [@Pedersen2002].

This direct contact operates through a number of receptor-receptor interactions. For example, CLL-stromal cell binding involves $\beta$ 1 integrin (ITGB1, or CD29) and $\beta$ 2 integrin (ITGB2, or CD18) [@Lagneaux1999].  VLA-4 is also an important integrin for retention of CLL cells within the lymph nodes and bone marrow, by interacting with its ligand VCAM-1 (or CD106) on stromal cells [@Burger2001]. 

These cell-cell interactions then lead to pathway activation (including BCR [@Stamatopoulos2015] and TLR [@Schulz2011]),  gene expression changes and epigenetic changes  [@Vangapandu2017; @Xu2018] within the tumour cells. For example, contact between CLL B cells and MSCs alters the transcriptomic profile of the cells [@Schulz2011; @Mangolini2018], leading to increased expression of anti-apoptotic proteins such as BCL-2 [@Nwabo2012; @Patel2014], BCL-XL [@Patel2014; @AmigoJimenez2015], MCL-1[@Kurtova2009; @AmigoJimenez2015], and $\beta$-catenin [@Mangolini2018]. 

**Soluble Factors**
NLCs, MSCs, FDCs and T cells also secrete soluble factors that have a protective effect on the tumour. For example, MSCs secrete a number of cytokines. One of the most widely studied is SDF1-$\alpha$ (or CXCL12), which interacts with CXCR4 on CLL cells [@Burger1999; @Burger2001; @Kay2007], stimulating the PI3K [@Burger2005], STAT3 [@Burger1999], and p44/42 MAPK [@Burger2000] pathways which activates BTK [@Montresor2018], ERK [@Messmer2011], and AKT [@OHayre2010].  FDCs on the other hand secrete B cell-activating factor (BAFF), which has been shown to increase survival of CLL cells through the activation of canonical NF$\kappa$B signalling [@Nishio2005]. A number of Th T-cell-derived cytokines have also been shown to increase CLL viability *in vitro*, including IFN$\gamma$ [@Buschle1993], IL15 [@Trentin1996], IL21 [@deTotero2006; @Pascutti2013], IL4 [@Dancescu1992; @Bhattacharya2015; @AguilarHernandez2016],  IL2 [@Decker2010] and CD40L [@Kitada1999; @Pascutti2013; @Bhattacharya2015]. Certain soluble factors can also increase apoptosis and act against the tumour, including TGF$\beta$ [@Lotz1994]. 

### The influence of the microenvironment on drug response {#intro-microenvironment-drug-response}

In addition to their effect of spontaneous apoptosis, the cell types and soluble factors outlined above have also been shown to impact on drug-induced apoptosis *in vitro*. For example, NLCs and stromal cells have been shown to meditate ibrutinib resistance [@Cheng2014; @Guo2017], and the chemotherapeutics fludarabine, oxaliplatin, chlorambucil, cyclophosphamide and doxorubicine show reduced efficacy in stromal cell co-cultures [@Kay2007; @Kurtova2009; @Mraz2011; @Zhang2012]. A number of soluble factors also induce resistance to drugs *in vitro*. These include decreased efficacy of fludarabine and venetoclax in the presence of TLR stimulation [@Fonte2013; @Jayappa2017] and reduced sensitivity to ibrutinib in the presence of IL4 [@AguilarHernandez2016] and BAFF [@McWilliams2019].  

Evidence of drug resistance  induced by the microenvironment *in vivo* is less prevalent, although there is widespread consensus that the microenvironment, in particular the lymph node, plays an important role in patient outcomes. Low rates of complete response and the inevitability of relapse in CLL have implicated the protective niche in enabling MRD [@OBrien2011; @Hayden2012]. A number of studies have shown enlarged lymph nodes are associated with MRD [@Moreton2005], in particular, incomplete response to ibrutinib is associated with persistently enlarged lymph nodes [@Ahn2018]. MSCs have also been shown to protect CLL cells taken from patients before and after *in vivo* fludarabine therapy [@Trimarco2015]. 

In light of this, an important goal in CLL research is to develop strategies to overcome drug resistance induced by the microenvironment. Targeting microenvironmental signalling in the lymph node tissue could be key to achieving long term remission and cure [@Hayden2012] and thus there is a need for combinatorial therapies that aim to eliminate CLL cells in the lymph node and reduce CLL load in the peripheral blood. For example, @Guo2017 have proposed cerdulatinib as a potential CLL therapy. Cerdulatinib is a dual inhibitor of the BCR pathway and the JAK-STAT pathway, capable of inducing cell death whilst also inhibiting the protective effects from the microenvironment. 

The development of rational strategies to target the microenvironment requires a more comprehensive understanding of drug – microenvironment interactions, and how these interplay with molecular features. Some studies have worked in this direction, for example @Jayappa2018 tested the impact of several agonists on ibrutinib and venetoclax. In a larger scale approach, @Gimenez2020 applied machine learning to identify drugs targeting proteins involved in microenvironmental signalling, and later screened these drugs in combination with venetoclax and ibrutinib, for activity against CLL in the presence of stromal cells.

Previous work in our lab probed drug activity in CLL peripheral blood mononuclear cell  (PBMC) samples in the context of BMSC co-culture in a large-scale screen of 81 CLL patients [@HerbstThesis]. Similar larger scale systematic studies of drug – microenvironment interactions, particularly in the context of molecular features, are required.

### Modelling the tumour microenvironment {#modelling-microenvironment-intro}
A major goal of current research in CLL is to unravel the complexity of CLL-microenvironment cross-talk and its role in drug response. Many studies have applied a range of strategies to mimic the pro-survival effect of the microenvironment, each with their own advantages [@Crassini2017; @Scielzo2020].

**Stimulation with soluble factors** 
One such strategy is to stimulate individual pathways in CLL samples *ex vivo* in order to elucidate their impact on CLL survival and drug response. For example, studies of BCR, TLR, CD40L and interleukin stimulation [@Muzio2009; @Crassini2017; @Scielzo2020] have proven critical in demonstrating the marked effect each of these have on CLL survival and the key downstream pathways involved (in particular NF$\kappa$B and MAPK) [@Crassini2017]. This strategy allows a direct understanding of cause and effect, with the caveat that accurately mimicking cytokine concentrations *in vitro* is challenging, and that the activity of certain stimuli may be altered in the absence of other signals. 

**Co-culture**
Stimulation studies omit the impact of cell-cell contacts and thus co-culturing CLL cells with cell lines can provide a more complete picture. Various co-culture systems have been developed in order to mimic different components of the microenvironment, including stromal cells, T cells, endothelial cells, NLCs and FDCs [@Panayiotidis1996; @Lagneaux1998; @Pedersen2002; @Kurtova2009; @Stamatopoulos2010; @Stamatopoulos2012; @Asslaber2013; @Hamilton2012; @Crassini2017].

**3D Models**
In recent years, interest has developed in the use of 3D culture systems, to create yet more *in vivo*-like models of the microenvironment [@Jensen2020; @Scielzo2020]. Static 3D approaches involve the use of scaffolds or the generation of spheroids, recapitulating the complexity of the protective niche to a greater degree [@Farinello2018; @Scielzo2020]. More ambitious still is the development of dynamic 3D cultures, through the use of bioreactors and microfluidics. These systems attempt to capture the influences of gravity, flow and mechanical stresses, to study the phenotypic changes that occur as CLL cells traffic through and communicate with the non-neoplastic tissues [@Walsby2014; @Scielzo2020].  

**in vivo murine models**
Murine models to investigate CLL-microenvironment interactions are also possible [@Lu2004; @Enzler2009; @Herishanu2011; @Fedorchenko2013; @Simonetti2014; @Crassini2017], though the value of these models can be hampered by species-specific biological differences [@Simonetti2014] and *in vitro* modelling is often a more accurate approach [@Crassini2017].  

Of the many strategies to model the microenvironment, the reductionist approach of stimulating individual pathways is a useful tool to demonstrate direct causal relationships between signal and response. So far, most of these studies have investigated individual stimuli. Larger scale systematic studies of stimuli in other lymphomas have proven successful, such as work by @Carey2017 to functionally screen many immune stimuli in Acute Myeloid Leukaemia (AML), suggesting that similar approaches could be valuable in CLL. 

Moreover, most of these studies have been performed in smaller patient cohorts, omitting the influence of the molecular heterogeneity of CLL. Indeed, integrative studies of the interplay between external stimuli and cell-intrinsic features in CLL are lacking. A few studies have identified interactions between genetic features and the microenvironment, for example, @Martinez2016 have established a link between MYD88 mutations and TLR response, and @Chatzouli2014 demonstrated a link between IGHV status and the response to TLR activation. In addition, @Mansouri2016 have discussed the convergence of mutations and external signals on the NF$\kappa$B pathway.  

The importance of interplay between microenvironment and molecular features in CLL survival and drug response is abundantly clear. However, a systematic study of the integrative influence of mutations and signals, particularly in the context of drug response, is missing in CLL. 


## CLL as a model for studying tumour biology {#CLL-model-intro}
CLL represents a valuable model system in cancer and studies of CLL can offer proof-of-principle for the application of new approaches in other entities. Primary PMBC samples are relatively simple to obtain, as CLL is the most common leukaemia and biopsies are performed by taking blood samples rather than intrusive operations. In addition, multiple biopsies cab be taken over the course of a patient’s monitoring and therapy. The samples themselves can survive *ex vivo* for several days, even in basic culture conditions, making it relatively simple to perform experiments on primary samples. 

Moreover, CLL is widely viewed as the prototypic disease for studying the integrative role of cell-intrinsic and cell-extrinsic features in disease initiation, expansion and drug response [@Mansouri2016; @Srinivasan2020; @Oppezzo2021].  Thus, studies seeking to integrate the impact of cell-intrinsic and cell-extrinsic features on CLL survival and drug response are important in the study of CLL and beyond. 

For example, the work by @Lu2021 to decipher a new multi-omic marker of disease aggression not only represents an important advance in our understanding of CLL drive, it also demonstrates an integrative approach to the study of cancer which could be useful in other cancers, particular where the heterogeneity of outcome remains unexplained. 

## Background to the approaches used in this thesis {#background-approaches-intro}
This thesis explores drug - microenvironment - gene interplay in CLL through the analysis of *ex vivo* perturbation assays combined with multi-omic profiling of patient samples. Background information on these experimental approaches, and associated data analysis, is outlined below. 

### *Ex-vivo* drug pertubation screens {#drug-perturbation-screens-intro}
Drug perturbation screens have been invaluable in identifying pathway dependencies, biomarkers and potential therapies in CLL [@Bosch2019]. Drug perturbation screens are usually performed in microtiter plates that contain a grid of wells suitable for performing an array of pharmacological or genetic experiments [@Letai2017]. Tumour cells, either cell lines or primary samples, can be deposited in each well to test their sensitivity to a set of compounds of interest. Tumour cells are incubated with each of the compounds, commonly dissolved in an aqueous solution of dimethyl sulfoxide (DMSO). After a set amount of time has passed to allow the cells to respond to the compound, the effect of each compound is measured. This “read-out” can take a number of forms.  For example the morphology of the cells can imaged [@Snijder2017; @HerbstThesis], or the cell viability can be measured via the number of cells or the level of adenosine triphosphate (ATP) in the well [@JCIpaper]. 

High-throughput drug screens of cancer cell lines have been widely used to link drug responses to molecular features [@Barretina2012; @Basu2013; @Garnett2012; @Iorio2016]. However, cell lines do not capture the genetic heterogeneity of a cancer [@Goodspeed2016], and thus drug screening of primary samples can be more valuable [@JCIpaper; @Tyner2013; @Pemovska2013; @Snijder2017]. In the case of CLL, screening primary samples has the caveat that CLL cells do not proliferate *ex vivo* [@Collins1989] and thus read-outs need to be taken on the basis of the rate of apoptosis relative to controls, rather than proliferation rate. 

A few *ex vivo* perturbation screens have also investigated the impact of microenvironmental stimulation on cancer biology,  for example @Carey2017’s functional screen of 94 cytokines in primary AML samples.  Studies of stimuli are much rarer despite the well-recognised role of the microenvironment across haematological malignancies and other cancers. 

Drug perturbation assays are also suited to combinatorial approaches, most commonly to test the efficacy of pairs of drugs in order to identify synergistic combinations [@Axelrod2014; @Lukas2020]. Combinatorial screening to test drug efficacy in the context of microenvironmental stimulation is also possible, though rare. 

### Multi-omics profiling and integration of multiple datatypes {#multi-omics-cll-intro}
Multi-omics profiling of samples, in combination with *ex vivo* perturbation screening, is a powerful approach to link cell phenotypes with molecular features in cancer. 
 
The disease pathogenesis and drug response of a cancer is determined by complex interactions between mutations, epigenetic alterations, gene expression, metabolic abnormalities, and aberrant signalling functions [@Anda2020]. Thus, the study of tumour biology, and indeed CLL biology, requires integrative methodologies and analyses to decipher this complex network [@Du2015].

Enter multi-omics, an approach to studying biological systems that utilises multiple “omic” layers to study biological entities [@Anda2020; @Menyhart2021]. These “omic” layers can encompass next-generation sequencing technologies, including DNA sequencing (DNAseq) and RNA sequencing (RNAseq) and high-throughput proteomics and metabolomics, along with newer single cell technologies and other sequence-based approaches such as ChIPseq (chromatin immunoprecipitation sequencing) and ATACseq (Assay for Transposase-Accessible Chromatin using sequencing) [@Anda2020]. A number of methods have been built to integrate these diverse data types, calling on tools from statistics, probability, machine learning and network analysis [@Hernandez2013; @Hernandez2014; @Argelaguet2018; @Anda2020]. 

In CLL, studies have profiled each of these layers independently, including the genomic [@Landau2015; @Puente2015], transcriptomic [@Ferreira2014; @Zenz2019], epigenomic [@Rendeiro2016; @Beekman2018; @Mallm2019; @Rendeiro2020] and proteomic landscapes [@HerbstThesis; @MeierAbt2021].

Building on these studies, multi-omics approaches have the power to identify causal relationships between phenotypic layers of CLL, and thus have led to important biological insights and clinical perspectives [@JCIpaper; @Berest2019; @Lipsky2020; @HerbstThesis; @Lu2021].  For example, a recent study in our lab integrated multiple data types to identify a novel biological axis, termed CLL proliferative drive, which is strongly associated with disease outcome [@Lu2021]. Other multi-omics studies have identified markers of drug response [@JCIpaper]. Integration of ATACseq and RNAseq has also been used to determine differences in transcription factor (TF) activity in CLL between the two major subtypes (IGHV-M and IGHV-U) [@Berest2019].  These studies highlight the importance of integrative approaches to gain further insights into CLL pathogenesis, especially with a view towards more personalised treatment strategies for patients. 

### Mathematical modelling {#intro-multivariate-modelling}

A number of mathematical tools are valuable in analysing complex multi-omics datasets. In this thesis, linear regression with and without lasso penalisation is used extensively to model the impact of the microenvironment and molecular features on drug response and a basic background to these approaches is outlined below. 

**Basic linear regression**
Linear regression involves fitting a linear model to a dataset, to model the process that generated the data. For example, equation \@ref(eq:basiclinearmodelsymbols) describes a basic model to map the values of $X$, to the values of $Y$:

 \begin{equation}
            Y = \beta_{0} + \beta_{1}X  + \epsilon
                                       (\#eq:basiclinearmodelsymbols)
    \end{equation}
    
This model specifies two components: the linear predictor  $\beta_{0} + \beta_{1}X$ and the error $\epsilon$. The linear predictor can be compared to the equation for a straight line ($Y = mX + c$), where $\beta_{0}$ represents the intercept ($c$), and $\beta_{1}$, the gradient ($m$). The error $\epsilon$ can be modelled by sampling from a normal distribution e.g. $N(0,\sigma^2)$,  a normal distribution with mean zero and variance $\sigma^2$. 

Thus, we can model a variable $Y$ via an expected value derived from the $X$ independent variable(s), plus a random value derived from normal distribution with specified variance [@Walker2018; @Huber2019].

**Multiple Linear Regression and Interaction Effects**
In certain cases, multiple independent variables may be predictors of the value of $Y$. In these cases, multiple linear regression is required, involving more than one explanatory variable. The basic model, for two independent variables $X_{1}$ and $X_{2}$, is denoted as follows: 

 \begin{equation}
            Y = \beta_{0} + \beta_{1}X_{1} + \beta_{2}X_{2}   + \epsilon
                                       (\#eq:multiplelinearmodel)
    \end{equation}
    
In some cases, the effect of an independent variable on $Y$ may depend on the on the value of another independent variable. For example, the effect of a drug on the viability of a CLL cell may depend on whether *TP53* is mutated. Such “interactions” between independent variables can also be accounted for within linear models.  These interactions are denoted as the product of two or more independent variables: 
   
 \begin{equation}
            Y = \beta_{0} + \beta_{1}X_{1} + \beta_{2}X_{2}  +\beta_{3} X_{1}X_{2}   + \epsilon
                                       (\#eq:interactionlinearmodel)
    \end{equation}
    
Here $X_{1}X_{2}$ represents the interaction, and  $\beta_{3}$ is the associated regression coefficient. Higher-order interactions with more terms are also possible. 

**Generalised Linear Models**
Not all dependent variables ($Y$) can be assumed to derive from sampling a normal distribution, as in equation \@ref(eq:basiclinearmodelsymbols). For example, if $Y$ is binary and takes only the values 0 or 1, a Bernoulli distribution is more appropriate.  Generalised linear modelling builds on linear regression such that the response variable can have an error distribution other than the normal distribution [@Nelder1972]. A number of distributions are possible, including the binomial, Poisson and gamma distributions. Where $Y$ is categorical, binomial or multinomial distributions are valuable; for count data, the Poisson distribution is often used.

**Lasso Regularisation**
When fitting such models to a particular dataset, it is important to avoid overfitting the data. Overfitting occurs when a model conforms too precisely to the dataset in hand. The model may not be representative of other datasets, and thus any predictions made using model are not reliable. Regularisation is an important tool to minimise such issues [@Kumar]. 

Regularisation adds a penalty term to the best fit model. This reduces the influences of dependent variables on the value of $Y$, by compressing the coefficients. This often acts to reduce the number of predictors and to generate a lesser variance with the test dataset.  There are two main methods for this, named L1 Lasso Regression and L2 Ridge Regression. The models described in this thesis use Lasso Regression, which is more interpretable than Ridge Regression. 

Lasso regularisation shrinks coefficients towards a central point, by adding a penalty to each coefficient, equal to the absolute value of its magnitude. This shrinkage approach means than some coefficients are reduced to 0 and are eliminated from the model, generating models that are sparse and have fewer parameters [@Tibshirani1996; @Kumar]. 

This can make models simpler to interpret and can be useful in cases where the dependent variables are highly correlated, as in these cases only one of the correlated variables will usually be assigned a coefficient.  This also has the caveat that the correlated features may each independently effect $Y$, but only one of these variables will be deemed important by the model. It is also often impossible to determine whether one or all the correlated variables are truly influencing $Y$. Thus, it is important to bear in mind that whilst modelling approaches have proven invaluable in advancing our understanding of biological processes, careful interpretation is required. 


## Summary of this thesis
In this thesis, I investigate interplay between molecular features and microenvironmental signals and their influence on viability and drug response in a large CLL patient cohort. Layer by layer, I aim to establish an integrative understanding of survival and drug resistance pathways in CLL.

I employ a large-scale screening dataset of 192 primary CLL samples, which quantifies the effects of 17 microenvironmental stimuli, both alone and in combination with 12 drugs. I combine the screening data with multi-omic profiles for each of the patient samples,  covering point mutations, copy number variations, DNA methylation and mRNA expression. The screen serves as a reductionist model of signalling via soluble factors in the microenvironment, in a cohort of patient samples that encompasses the clinical and molecular heterogeneity of CLL. I investigate the influence of microenvironmental stimulation on spontaneous and drug-induced apoptosis, and how this can be further modulated by molecular features.

Several signalling pathways and genetic features emerge as key players in CLL-microenvironment cross-talk, and in microenvironmental mechanisms of drug resistance. The above experiments are complimented by IHC stains of healthy and CLL-infiltrated lymph nodes to investigate the *in vivo* activity of selected pathways. To supplement the large scale screening approach, I also make use of RNAseq, ATACseq, Mass Spectrometry and ChIPseq datasets of CLL patient samples to validate individual findings at a mechanistic level. 

The thesis is structured as follows:   
   
**Chapter 1: Introduction** Overview of the background information and surrounding literature of the project.    
   
**Chapter 2: Methods** Overview of the experimental and analytical methods used in this thesis. 
   
**Chapter 3: Data** Overview of the datasets employed in this thesis, and the corresponding public code repository and shiny app.  
   
**Chapter 4: *Ex-vivo* responses of primary CLL samples to microenvironmental stimuli **
Phenotypic profiling of the responses of CLL samples to the panel of microenvironmental stimuli. 

**Chapter 5: Genetic modulators of responses to microenvironmental stimulation** Integration of the screening data with patient sample multi-omic profiles to perform a comprehensive survey of molecular determinants of stimulus response.   
   
**Chapter 6: Molecular and microenvironmental modulators of drug response in CLL** Mapping of drug - microenvironment interactions and how these can be further modulated by molecular features   
  
**Chapter 7: Discussion ** A discussion of the findings, the strengths and weakness of the approaches and their context within the field and surrounding literature.    