# Methods {#methods}

Methods in quotation marks are taken from Bruch & Giles et al. 2021 and I have authored the original text, unless stated otherwise. Paragraphs without quotation marks were rewritten for the thesis. 

```{r eval = FALSE, include = FALSE}

#Routine: 
 
#ALL headings
#Go trhough and fill out all methods, add [CHECK if anything left to confrim and collect a list at bottom]
#COPy where i can, put in qutatoationamrks and indicate any additioanl authors
#Rewrtie if necssary
#Use was/were

#at end go through and say which heading belong to who 

```

Experiments - was / were keep a record of who, then make clear when i have strat
Data - make very clear where each additional dataset comes from 
Analysis - was / were, keep a record of who, then make clear 
+ask for permission wrt proteomics dataset 
^then this is adaptable if needed 

## Experimental methods: 
### Combinatorial Drug - Stimulus Perturbation Screen
#### Patient Sample Preparation
_HG wrote_
Peripheral blood samples were obtained from 192 patients. A Ficoll gradient (GE Healthcare) was used to separated the samples and mononuclear cells were then cryopreserved. For the screen, samples were thawed and DMSO was removed.  Primary patient samples were next incubated on a roll mixer at room temperature in cell culture medium for a duration of  three hours. This process ensured that any remaining DMSO was removed and to ensure that cell counts would only include those that survived the freezing process.  For the screen,  15 $\mu$l of each PBMC sample was deposited into each well of the plates with stock solution concentration of 1.3\*10\^6 cells/ml. The final cell concentration was 2\*10\^4 cells per well.
 Viability using Celltiter Glo from Promega and measuring on the PerkinElmer Envision Plate Reader, as this absolute Value varies between Samples and Measuring date (Refreezing of CTG and Incubation Time) the Values were normalized to internal DMSO Values.
 
#### Preparation of screening plates
using the ATP-based CellTiter Glo assay (Promega, Fitchburg, WI, USA) and luminescence level was measured with a Tecan Infinite F200 Microplate Reader(Tecan Group AG, Männedorf, Switzerland).

Compounds were preplated in 384-well polypropylene storage plates (Greiner Bio-One Cat.No.:781271), which were stored at -20$^\circ$C. For each batch of samples tested on the same day a new storage plate was thawed and diluted in serum free RPMI, with or without stimuli. 5$\mu$L of drug-stimulation dilution were then added into each well of the assay plates, 20 $\mu$L of cell suspension were added. The final cell concentration was 8\*10\^5 cells/ml. Cells were thawed as previously described[REFERENCE].

_with Peter-Martin Bruch_ Original 
Stimuli and drugs were mixed and preplated in the culture plates directly before adding the cell suspensions. RPMI-1640 and supplements were acquired from Gibco by Life Technologies, human serum was acquired from PAN Biotech (Cat.No. P40-2701, Lot.No:P-020317). Luminescence was measured after 48h on a Perkin Elmer EnVision." 

_with Peter-Martin Bruch_ Original 
Compounds were obtained from Selleckchem, MedChemExpress and Sigma-Aldrich, dissolved in DMSO and stored at -$20^\circ$C. 12 drugs were used in two concentrations [TABLE]. Final DMSO concentration did not exceed 0.3\% in all experiments. Carfilzomib, Panobinostat and Venetoclax were removed from the analysis as they showed inconsistent toxicity depending on used media.
Insert table

Recombinant cytokines and stimulatory agents were dissolved according to manufacturer's protocol. 17 stimuli were selected [TABLE]. HS-5 conditioned medium was produced by incubating HS-5 stroma cell line (gifted by Martina Seiffert, DKFZ, Heidelberg) for 4 days at 37$^\circ$C and 5% CO2, after which the supernatant was centrifuged and stored at -20$^\circ$C, the final concentration of HS-5 CM was 20\%. Bead immobilised anti-IgM was removed from the analysis due to storage instability. "

Paper: 
"Compounds and stimulatory agents were dissolved, stored and diluted according to manufacturer's protocol. HS-5 conditioned medium was produced by incubating HS-5 stromal cell line to >80\% confluency and cell removal by centrifugation. For a detailed list of stimuli and drugs and associated concentrations see Supp. Tables 1 and 2. Final DMSO concentration did not exceed 0.3\%. "

Original 
Compounds were preplated in 384-well polypropylene storage plates (Greiner Bio-One Cat.No.:781271), which were stored at -20$^\circ$C. For each batch of samples tested on the same day a new storage plate was thawed and diluted in serum free RPMI, with or without stimuli. 5$\mu$L of drug-stimulation dilution were added into each well of the assay plates, 20 $\mu$L of cell suspension were added. The final cell concentration was 8\*10\^5 cells/ml. Cells were thawed as previously described[REFERENCE].

#### Drug-Stimulation assay 
_with Peter-Martin Bruch_ Original 
Drug-stimulation assays were performed with RPMI-1640 (Gibco by Life Technologies) supplemented with Penicillin Streptomycin (Gibco) final concentration of 100 Units/ml and 100 $\mu$g/ml respectively, L-Glutamine (Gibco) final concentration 2mM, and 10\% pooled, heat-inactivated and sterile filtered human type AB male off the clot serum (PAN Biotech, Cat.No. P40-2701, Lot.No:P-020317).

Drugs and stimuli were added to the cells simultaneously. Culture was performed in 384-well plates (Greiner Bio-One Cat.No.: 781904) and cells were incubated at 37$^\circ$C and 5\% CO2 for 48h.

Cell Viability was determined using the ATP-based CellTiter-Glo assay (Promega, Cat.No.:G7573). Luminescence was measured for the drug-stimulation assays using a Perkin Elmer EnVision and for the drug-drug interaction experiments using a Perkin Elmer EnSight, with a measurement time of 100ms per well.

Drugs : ADD


Cytokines : update 

```{r cytTable02, out.width = "30%", fig.cap='(ref:cytTable)', echo = FALSE, eval = FALSE}
#cytokines_clean <- cytokines
cytokines <- cytokines_clean

#tidy names for table
cytokines$stimulus[grepl("IL-4 human recombinant animal component free",cytokines$stimulus)]<-"IL4 human recombinant animal component free"
cytokines$stimulus[grepl("IL-2 human recombinant animal component free",cytokines$stimulus)]<-"IL2 human recombinant animal component free"
cytokines$stimulus[grepl("Human IL-1beta",cytokines$stimulus)]<-"Human IL-1$\\beta$"
cytokines$stimulus[grepl("Human IL-15",cytokines$stimulus)]<-"Human IL15"
cytokines$stimulus[grepl("Human IFN-gamma",cytokines$stimulus)]<-"Human Interferon $\\gamma$"
cytokines$stimulus[grepl("IL-10 human Animal component free",cytokines$stimulus)]<-"IL10 human Animal component free"
cytokines$stimulus[grepl("Human TGF-beta1",cytokines$stimulus)]<-"Human TGF$\\beta$"
cytokines$stimulus[grepl("Human IL-6",cytokines$stimulus)]<-"Human IL6"
cytokines$stimulus[grepl("alpha",cytokines$stimulus)]<-"Human SDF-1$\\alpha$ (CXCL12)"
cytokines$stimulus[grepl("HS-5 konditioniertes Medium",cytokines$stimulus)]<-"HS-5 conditioned medium"

newCytokines <-  
cytokines %>%
  left_join(Cytokine_labels, by = "name") %>%
  dplyr::select(-name, -additional_pathways, -source, -company, -cat_no, -lot_no) %>%
  dplyr::rename(Stimulus=stimulus, 
                Name = Cytlabel, 
                Concentration=conc, 
                Pathway=pathway) %>%

  dplyr::select(Stimulus, Name,  everything())


#update label for latex 

newCytokines$Name[grepl("TGF\u03B21",cytokines$Name)] <- "TGF$\\beta$"
newCytokines$Name[grepl("IL1\u03B2",cytokines$Name)] <- "IL1$\\beta$"
newCytokines$Name[grepl("Interferon \u03B3",cytokines$Name)] <- "Interferon $\\gamma$"
newCytokines$Name[grepl("SDF-1\u03B1",cytokines$Name)] <- "SDF-1$\\alpha$"

newCytokines %>% 
  kable() %>%
  row_spec(0,bold=TRUE) %>%
  kable_styling(latex_options = "striped", 
                stripe_color = colors[7],
                html_font = "Helvetica", 
                font_size = 7)

```

## WES, RNA Sequencing, Targeted Sequencing and DNA Copy number variants 
_Myself_ Original 
Sequencing data generated by Dietrich et al[REFERENCE], and all sequencing and data processing were performed as described there. For 

### Follow - up investigations 
#### Lymphocyte doubling time 
collected as, received from 
#### Survival data 

#### Large Proteomics data set
Provided by Sophie Herbst [@HerbstThesis]

#### ATACsequencing of 4 CLL samples 
_with Peter-Martin Bruch (mostly edited by me)_ Paper 
"Peripheral blood was taken from 4 CLL patients and separated by Ficoll gradient (GE Healthcare), mononuclear cells were cryopreserved on liquid nitrogen. Samples were later thawed from frozen as previously described[REFERENCE] and MACS sorted for CD19 positive cells (Milteny autoMACS). The cells were resuspended in RPMI (GIBCO, Cat.No. 21875-034), with the addition of 2mM glutamine (GIBCO, Cat.No. 25030-24), 1% Pen/Strep (GIBCO, Cat.No. 15140-122) and 10\% pooled, heat-inactivated and sterile filtered human type AB male off the clot serum (PAN Biotech, Cat.No. P40-2701, Lot.No:P-020317). 5ml of cell suspension was cultured in 6-well plates (Greiner Bio-One Cat.No. 657160). After thawing, cells were incubated at 37$^\circ$C and 5\% CO2 for 6 hours in 0.2\% DMSO. The final cell concentration was 2x10ˆ6 cells/ml. Cell viability and purity was assessed using FACS. All samples had a viability over 90\% and over 95\% of CD19+/CD5+/CD3- cells."

_with Nayara_ Paper 
"ATAC-seq libraries were generated as described previously[REFERENCE]. Cell preparation and transposition was performed according to the protocol, starting with 5x10ˆ4 cells per sample. Purified DNA was stored at −20$^\circ$C until library preparation was performed. To generate multiplexed libraries, the transposed DNA was initially amplified for 5x PCR cycles using 2.5 $\mu$L each of 25 μM PCR Primer 1 and 2.5 $\mu$L of 25 $\mu$M Barcoded PCR Primer 2 (included in the Nextera index kit, Illumina, San Diego, CA, USA), 25 $\mu$L of NEBNext High-Fidelity 2x PCR Master Mix (New England Biolabs, Boston, Massachusetts) in a total volume of 50 $\mu$L. 5 $\mu$L of the amplified DNA was used to determine the appropriate number of additional PCR cycles using qPCR. Additional number of cycles was calculated through the plotting of the linear Rn versus cycle, and corresponds to one-third of the maximum fluorescent intensity. Finally, amplification was performed on the remaining 45 $\mu$L of the PCR reaction using the optimal number of cycles determined for each library by qPCR (max. 13 cycles in total). The amplified fragments were purified with two rounds of SPRI bead clean-up (1.4x). The size distribution of the libraries was assessed on Bioanalyzer with a DNA High Sensitivity kit (Agilent Technologies, Santa Clara, CA), concentration was measured with Qubit DNA High Sensitivity kit in Qubit 2.0 Flurometer (Life Technologies, Carlsbad, CA). Sequencing was performed on NextSeq 500 (Illumina, San Diego, CA, USA) using 75bp paired-end sequencing, generating ∼450 million paired-reads per run, with an average of 55 million reads per sample."


#### Spi-B and PU.1 shRNA Knockdowns

#### Immunohistochemistry of patient lymph nodes
_Peter-Martin Bruch_ Paper 
"LN biopsies of CLL-infiltrated and non-neoplastic lymph nodes were Paraffin fixed and arranged in Tissue Microarrays. Consecutively they were stained for PAX-5 (790-4420, Roche), CD3 (790-4341, Roche), pSTAT6 (ab28829, abcam), STAT6 (519-4290, Zytomed Systems) and pIRAK4 (ab216513, abcam). The slides were analysed using Qupath 50 and the recommended protocol. Cell based data on mean staining intensity was exported and further analysed using R. "

#### Ibrutinib - IL4 - STAT6i interaction assay 

### Investigation of Ibrutinib + IL4 + IBET-762
#### Ibrutinib - IL4 - IBET-762 interaction assay

#### Ibrutinib - IL4 - IBET -762 treatment

#### ATACsequencing 
_with Peter Martin Bruch_ get from diffTF 
Peripheral blood was taken from 4 CLL patients and separated by Ficoll gradient (GE Healthcare), mononuclear cells were cryopreserved on liquid nitrogen. Samples were later thawed from frozen as previously described[REFERENCE] and MACS sorted for CD19 positive cells (Milteny autoMACS). The cells were resuspended in RPMI (GIBCO, Cat.No. 21875-034), with the addition of 2mM glutamine (GIBCO, Cat.No. 25030-24), 1\% Pen/Strep (GIBCO, Cat.No. 15140-122) and 10\% pooled, heat-inactivated and sterile filtered human type AB male off the clot serum (PAN Biotech, Cat.No. P40-2701, Lot.No:P-020317). 5ml of cell suspension was cultured in 6-well plates (Greiner Bio-One Cat.No. 657160). After thawing, cells were incubated at 37$^\circ$C and 5\% CO2 for 6 hours in 0.2\% DMSO. The final cell concentration was 2x10ˆ6 cells/ml. Cell viability and purity was assessed using FACS. All samples had a viability over 90% and over 95% of CD19+/CD5+/CD3- cells.

_with Nayara_ get from diffTF 
ATAC-seq libraries were generated as described previously[REFERENCE]. Cell preparation and transposition was performed according to the protocol, starting with 5x10ˆ4 cells per sample. Purified DNA was stored at −20$^\circ$C until library preparation was performed. To generate multiplexed libraries, the transposed DNA was initially amplified for 5x PCR cycles using 2.5 $\mu$L each of 25 $\mu$M PCR Primer 1 and 2.5 $\mu$L of 25 $\mu$M Barcoded PCR Primer 2 (included in the Nextera index kit, Illumina, San Diego, CA, USA), 25 $\mu$L of NEBNext High-Fidelity 2x PCR Master Mix (New England Biolabs, Boston, Massachusetts) in a total volume of 50$\mu$L. 5 $\mu$ L of the amplified DNA was used to determine the appropriate number of additional PCR cycles using qPCR. Additional number of cycles was calculated through the plotting of the linear Rn versus cycle, and corresponds to one-third of the maximum fluorescent intensity. Finally, amplification was performed on the remaining 45 $\mu$L of the PCR reaction using the optimal number of cycles determined for each library by qPCR (max. 13 cycles in total). The amplified fragments were purified with two rounds of SPRI bead clean-up (1.4x). The size distribution of the libraries was assessed on Bioanalyzer with a DNA High Sensitivity kit (Agilent Technologies, Santa Clara, CA), concentration was measured with Qubit DNA High Sensitivity kit in Qubit 2.0 Flurometer (Life Technologies, Carlsbad, CA). Sequencing was performed on NextSeq 500 (Illumina, San Diego, CA, USA) using 75bp paired-end sequencing, generating ∼450 million paired-reads per run, with an average of 55 million reads per sample.

#### RNAsequencing 

#### Proteomics 

## Data availability
European Genome-Phenome Archive (EGA) accession .... The data  for …. and computational analysis code used in this study are available from ….

Additionally, we obtained CLL ATACseq data[REFERENCE] from the European Genome-phenome Archive (EGA: EGAD00001002110) and from (Bruch & Giles et al. 2021)
For the ChIPseq analysis of SPIB binding sites, we accessed the data 37 from the NCBI GEO database 51, accession GEO: GSE56857. We made use of the data for SPIB in the OCILY3 DLBCL cell line (GSM1370276). 


addtioanl i obtained pretoeomics data from: , provided by Sophie Herbst [@HerbstThesis]

## Statistical Analysis 
Mixutre of paper and original 
Integrative data analysis of screening data, DNA and RNA sequencing, CNV and methylation profiles and follow up experiments was performed using R version 4 (REFERENCE R Core Team, 2018) with the RStudio interface (REFERENCE RStudio Team, 2016) and using packages that included [@R-DESeq2], [@R-survival], [@R-glmnet], [@R-ConsensusClusterPlus], [@R-clusterProfiler], [@R-ChIPseeker], [@R-genomation] and [@R-BloodCancerMultiOmics2017] to perform univariate association tests, multivariate regression with and without lasso penalization, Cox regression, generalised linear modelling and clustering. 

### Data processing
#### Processing of screening data
_with Peter-Martin Bruch_ Paper 
To quantify the response of a patient sample to each treatment condition, we used viability relative to the control, i.e., the CellTiter Glo luminescence readout of the respective well divided by the median of luminescence readouts of the DMSO control wells on the same plate.

#### Quality Assessment and Control 

#### RNA data
RNA data was taken from the PACE repository @PACE. For detailed explanaition of processing see @PACE. 

#### ATACsequencing 

#### Data from PACE


### Analysis of Screening Data
#### Drug-drug and stimulus - stimulus correlations {#correlations}
Section \@ref(drug-responses) and  \@ref(cytokine-profiling). Pearson correlation coefficients were calculated for each drug - drug and stimulus - stimulus pair, using the `cor` functions in `R` [@R-base] with log transformed viability values which were normalised to untreated controls (Bruch and Giles et al. 2021).


Contributions: 
Performed with: Peter
Written by: Me
Text source: written for thesis

#### Correlation of RNA-Receptor Expression with viability scores
Section \@ref(rna-correlations). RNA count data for matched samples was available for 49 patients and was transformed using the variance stabilising transformation. Stimulus - receptor pairs were defined using the available literature, see Appendix Table \@ref(tab:receptorPairs).  For each stimulus, a Pearson correlation coefficient was calculated between the log-transformed control-normalised viability values and the expression of the corresponding stimulus receptor for matching samples. Correlation coefficients were visualised in a volcano plot, to determine if any cytokine-receptor pairs showed R>0.4 (Bruch and Giles et al. 2021).

Contributions: 
Performed with: Peter
Written by: Me
Text source: Written for thesis

#### Consensus clustering and visualisation of stimulus responses {#stimulus-heatmap-method}
Section \@ref(clusters). For the heatmap in figure \@ref(fig:stimuliHeatmap), the log transformed  control-normalised viability scores were scaled for optimal visualisation. For each stimulus, viability values  were row-scaled  according to the Median Absolute Deviance, and  limits were then applied  to this row scaling factor for the purposes of visualisation, such that all resulting z scores were between -3 and +3. 

The columns (patient samples) of the resulting matrix were then clustered using the `ConsensusClusterPlus` function, from the [@R-ConsensusClusterPlus] package. The function generated robust clusters for  k (number of clusters) = 2 - 7,  performing heirarchical clustering based on Euclidean distances, with 10,000 repeats. 

To quantify the degree of confidence in the clusters for each k, plots representing the cumulative distribution functions (CDFs) of the consensus matrices for k = 2 - 7, the relative change in area under the CDF curves, and cluster stability were assessed.

The z scores were then visualised for k = 4, using the pheatmap package [@R-pheatmap], whereby the columns were clustered according to the dendrogram resulting from the above,  and the rows (stimuli) were ordered using the dendrogram order produced by `hclust` with default branch arrangement (Bruch and Giles et al. 2021).

Contributions: 
Performed with: Peter
Written by: Me
Text source: Written for thesis

#### Univariate analysis of gene - stimulus and gene - drug response assosciations {#univariate-gene-stimulus-associations-method}

Section \@ref(univariate-gene-stimulus-assosciations) and \@ref(univariate-gene-drug-associations). Two-sided Student's t-tests, with equal variance were performed for IGHV status and somatic mutations and copy number aberrations with at least 3 patient samples in each group (n = 54). Mutations in _KRAS_, _NRAS_ and _BRAF_ were tested in a single group.  p values were adjusted using the BH-adjustment procedure, and a 10% FDR cut off was used to determine significance (Bruch and Giles et al. 2021).  


Contributions: 
Performed with: Peter
Written by: Me
Text source: Written for thesis

#### Penalised multivariate regression of gene - stimulus assosciations {#multivariate-gene-stimulus-assosciations-method}
Section \@ref(multivariate-gene-stimulus-assosciations). To identify gene-stimulus associations, a Gaussian linear model with L1-penalty with mixing parameter alpha = 1 was fitted for each stimulus using the `cv.glmnet` function from the `R` package `glmnet` [@R-glmnet]. The feature matrix consisted of genetic features (p=39), IGHV status (input as M = 1 and U = 0), and Methylation Cluster (input as 0, 0.5, 1). All features were thus encoded on a similar scale to ensure equal treatment by lasso constraint in model fitting.  Where genetic features showed more than 20% missing values, these were excluded from the feature matrix. Samples without complete annotation for remaining features were removed, resulting in n = 129 samples. The matrix of control-normalised log-transformed viability values for these 129 samples was provided as the response matrix. Using three-fold cross-validation, the optimal penalty parameter $\lambda$ was selected so as to minimise the cross-validated R2. The reduction in cross-validated mean squared error compared to the null model was used as loss. The model was fitted for 30 bootstrapped repeats, and the resulting coefficients are the mean of those coefficients that were selected in >75% model fits (Bruch and Giles et al. 2021). 

Contributions: 
Performed with: support from Junyan and JCI paper 
Written by: Me
Text source: Written for thesis

#### Linear modelling of drug - stimulus interactions {#drug-stimulus-linear-model-method}
Section \@ref(drug-stimulus-linear-model). Linear models were fitted for each drug - stimulus combination, to extract a $\beta{int}$ term and associated p value for each combinatorial treatment. Linear model was fitted using equation \@ref(eq:drugCytInt), using the `lm` function of the `R` package stats [@R-base].

To fit model, the matrix of log-transformed viability values, for control, single and combinatorial treatments was used. Interactions were filtered according to whether p value for $\beta{int}$ < 0.05. To generate the map of drug - stimulus interactions, the matrix of resulting $\beta{int}$ was plotted as a heatmap, with the package `pheatmap`[@R-pheatmap] where the rows (stimuli) and columns (drugs) were ordered according to the dendrogram order produced by `hclust` [@R-base] using default branch arrangement.

To define the four interaction categories, drug - stimulus combinations were first divided with respect to the sign of $\beta{int}$, whereby a positive $\beta{int}$ indicates that the viability with combinatorial treatment is higher than would be expected based on additive effects alone and vice versa. The groups were further divided into synergies and antagonisms according to the values of the model coefficients ($\beta{drug}$, $\beta{stimulus}$ and $\beta{int}$). Synergisms were assigned when coefficients for single treatments ($\beta{drug}$ and $\beta{stimulus}$) were both greater than, or both less than, the observed coefficient for the combinatorial treatment (i.e. $\beta{drug}$ + $\beta{stimulus}$ + $\beta{int}$). For positive antagonisms, $\beta{drug}$ + $\beta{stimulus}$ + $\beta{int}$ was less than either $\beta{drug}$ or $\beta{stimulus}$. For negative antagonisms,  $\beta{drug}$ + $\beta{stimulus}$ + $\beta{int}$ was greater than either $\beta{drug}$ or $\beta{stimulus}$. All drug - stimulus interactions for which p value for $\beta{int}$ <0.05 fit into one of these groups (Bruch and Giles et al. 2021).

Contributions: 
Performed with: myself, Frederick helped with script (Sophie did not credit this), Peter did int categories
Written by: Me
Text source: Written for thesis

#### Modelling of drug - stimulus - gene interactions {#drug-stimulus-gene-interactions-method}
Section \@ref(drug-stimulus-gene-interactions). Identification of drug - stimulus interactions that were modulated by genetic features was performed in two steps. 

First the linear model in equation \@ref(eq:drugCytInt) was fitted in a patient sample specific manner i.e.  equation \@ref(eq:drugCytGeneInt) was fit to  the matrix of log-transformed, control-normalised viability scores, for each drug - stimulus combination. 

This resulted in a higher order interaction term for each drug - stimulus - patient combination, named $\beta_{int}X_{drug}X_{stimulus}X_{patient}$. This term represents a _patient sample-specific_ $\beta{int}$ for each drug - stimulus combination, quantifying the size of an interaction between a drug and stimulus in each patient genetic background. 

In the second step,  multivariate regression with L1 (lasso) regularisation was used to identify associations between the size of the patient sample-specific $\beta{int}$ terms and genetic features. As input to the model, the response matrix was composed of the sample - specific $\beta_{int}$ values for each drug-stimulus combination. The feature matrix consisted of genetic features (p=39), IGHV status (input as M = 1 and U = 0), and Methylation Cluster (input as 0, 0.5, 1). All features were thus encoded on a similar scale to ensure equal treatment by lasso constraint in model fitting.  Where genetic features showed more than 20% missing values, these were excluded from the feature matrix. Samples without complete annotation for remaining features were removed, resulting in n = 129 samples.

Using three-fold cross-validation, the optimal penalty parameter $\lambda$ was selected so as to minimise the cross-validated R2. The misclassification error was used as loss. The resulting predictors are the mean of those coefficients that were selected in at least 90% of 30 bootstrapped repeats (Bruch and Giles et al. 2021). 

Contributions: 
Performed with: myself
Written by: Me
Text source: Written for thesis

Now proof this section in pdf and sign off 

### Follow up 
#### Assosciation of clusters and lympocyte doubling times {#LDT-method}
Section \@ref(cluster-survival). Data on lymphocyte growth rates were curated from clinical records.  To calculate growth rates, a linear model was fit to log10 transformed lymphocyte counts for a series of time points starting with the sample collection date and ending with the time of the next treatment. Where less than four time points were recorded, these patients were excluded, resulting in LDT measurements for 115 patient samples. Associations between LDT measurements and patient clusters were assessed using two-sided Student's t-tests. 

[CHECK]
Contributions: 
Performed by: Peter
Written by: Me
Text source: Written for thesis

#### Assosciation of clusters and patient outcomes {#survival-method}
Section \@ref(cluster-survival).  Differential disease progression between patient clusters was measured using TTT as metric. 188 of 192 CLL patients were annotated for treatment information after sample collection. TTT represents the period between the date of sample collection and the data of treatment initiation. TTT was plotted using the Kaplan-Meier method, in which patient samples were stratified by cluster. To calculate significance, a Cox proportional hazards regression model was fitted using  the  `coxph` function of the `R` package `survival` [@R-survival].  

Contributions: 
Performed by: Peter
Written by: Me
Text source: Written for thesis

#### Penalised multivariate regression to identify genetic predictors of cluster membership {#cluster-genetics-method}
Section \@ref(cluster-genetics). Differential enrichment of genetic features amongst the four patient clusters was quantified using a multinomial linear model with L1-penalty, via the `cv.glmnet` function of the `glmnet` package [@R-glmnet]. As input to the model, the discrete response matrix represented the vector of cluster assignments (1-4) for each patient sample. The feature matrix consisted of genetic features (p = 39) and IGHV status (input as M = 1 and U = 0). All features were thus encoded on a similar scale to ensure equal treatment by lasso constraint in model fitting.  Where genetic features showed more than 20% missing values, these were excluded from the feature matrix. Samples without complete annotation for remaining features were removed, resulting in n = 129 samples. Using three-fold cross-validation, the optimal penalty parameter $\lambda$ was selected so as to minimise the cross-validated R2. The misclassification error was used as loss. The resulting coefficients are the mean of 50 bootstrapped repeats, where coefficients were filtered if they were selected in < 60% of cases or were <0.35.  Standard deviations were calculated for each coefficient based on the bootstrapped repeats. (Bruch and Giles et al. 2021). 

Contributions: 
Performed by: Me
Written by: Me
Text source: Written for thesis
 
#### Gene expression and Gene Set Enrichment Analysis (GSEA) between clusters {#cluster-rna-method}

Section \@ref(cluster-rna). To look for associations between stimulus response data and RNA expression data the `R` package `DESeq2` [@R-DESeq2] was used.  RNAseq data was available for 49 matched PBMC samples, 21 of which  belonged to C3 and C4 (Bruch and Giles et al. 2021). 

To quantify differential gene expression between C3 and C4, genes encoding components of the BCR  were first filtered, including genes at the heavy, light and kappa immonglobulin loci. Differential expression was quantified using `DESeq2` protocol [@R-DESeq2] with the design formula ~ IGHV.status + Cluster. Genes were then ranked according to the resulting Wald statistics and GSEA was performed using the the `clusterProfiler` package [@R-clusterProfiler], with the fgsea algorithm and using KEGG pathway gene sets from the MSigDB database [@R-msigdbr].  

Contributions: 
Performed by: Me
Written by: Me
Text source: Written for thesis

#### Analysis of differential gene dosage in trisomy 12 CLL {#gene-dosage-effects-method}
Section \@ref(gene-dosage-effects). For all RNA samples available in PACE (i.e. not just those that matched the samples in the screen), differential expression was called using the `DESeq2` package [@R-DESeq2], with the design formula ~trisomy12.  Raw RNA counts were visualised if the gene had BH-adjusted p < 0.1 and belonged to TGF$\beta$, JAK-STAT or TLR pathways genesets, as defined in the KEGG database [@Kegg] downloaded using the `msigdbr` package [@R-msigdbr]. Proteomic abundance data was also plotted. Samples in proteomics data partially overlap with those in RNAseq data.

Contributions: 
Performed by: Me + use of Sophie's data 
Written by: Me
Text source: Written for thesis

^good to here 
#### Generation of a Trisomy 12 classifier {#trisomy12-classifier-method}
_edit and finish_ 
Section \@ref(trisomy12-classifier). To build the classifier,  a binomial linear model with L1-penalty implemented in the `R` package `glmnet` [@R-glmnet], with mixing parameter alpha = 1, was used. To identify coefficients that predict trisomy 12 status, the viability matrix for all stimuli,  as described in section \@ref(stimulus-heatmap-method) was used, i.e. z scores of the control-normalised log-transformed viability values. 

Matrix of genetic features (p=39), IGHV status (encoded as M = 1 and U = 0), and Methylation Cluster (encoded as 0, 0.5, 1) were used to identify multivariate predictors. All features were thus encoded on a similar scale to ensure equal treatment by lasso constraint in model fitting.  Genetic features with more than 20% missing values were excluded from the analysis, and samples without complete annotation for remaining features were removed, resulting in n = 129 samples. The matrix of control-normalised log-transformed viability values for these 129 samples was provided as the dependent variable. Using 3-fold cross-validation, the optimal penalty parameter $\lambda$  was selected so as to minimise the cross-validated R2. The reduction in cross-validated mean squared error compared to the null model was used as loss. The model was fitted for 30 bootstrapped repeats, and the resulting coefficients are the mean of those coefficients that were selected in >75% model fits. 
The classifier was built using binomial regression, with lasso penalisation,  as implemented in the `R` package `glmnet` [@R-glmnet]. The feature matrix consisted of z scores of the viability values after treatment with each stimulus, and was used to predict the response (trisomy 12 status). I ran the model for 50 bootstrapped repeats, using  three-fold cross-validation and mean absolute error as loss. Resiquimod, sCD40L+IL4 and TGF$\beta$ were selected as coefficients that predict trisomy 12 status (Figure \@ref(fig:tri12Classfier), as would be expected bed on the observations in section \@ref(trisomy12-modulator).


#### ATACseq processing 
_with Ivan Berest_ original (update to paper) 
We downloaded CLL data [@Rendeiro2016] from the European Genome-phenome Archive (EGA: EGAD00001002110). Original dataset had 88 ATAC-seq samples from 55 patients, however we used for the analysis only one sample per patient passing quality checks, resulting in 52 samples. We used an in-house constructed ATAC-seq processing pipeline [@Berest2019] to obtain final bam files mapped to the hg19 annotation genome that were corrected for CG bias. 

#### Annotation of trisomy 12 status
_with Ivan Berest_ original (update to paper) 
As trisomy 12 status was not included in the original metadata, we used a mean amount of reads in the chromatin accessible peaks for each sample to distinguish trisomy 12 patients. All samples containing 1.4 times more reads in the peaks located on chromosome 12, compared to the peaks on all other chromosomes, were classified as trisomy 12 patients.

#### diffTF analysis of TF activity in trisomy 12 CLL
_with Ivan Berest_ original (update to paper) 
FOr the larger ATACseq dataset, we ran analytical mode of diffTF [@R-diffTF] pipeline to investigate differential TF activity between trisomy 12 and non-trisomy 12 patients using the HOCOMOCO v10 database [@HOCOMOCO] with the following design formula: “~ Batch + Gender + IGHV status + trisomy12 status”. 

for tirsomy 12 software uses the ATACseq counts at each of the binding sites across the genome to generate a distribution of fold changes between trisomy 12 and WT samples. This fold change distribution is compared to a background distribution of fold changes, calculated similarly using ATACseq counts for a GC-matched motif, that does not contain the TF binding motif. The TF is classified as differentially active between trisomy 12 and WT samples where these two distributions are significantly different. Each TF is assigned a weighted mean difference value, which quantifies the change in activity, and a p value. 

For the smaller ATACseq dataset, ATAC-seq data generated from our CLL samples were processed similarly with the in-house ATAC-seq pipeline, as described previously [@Berest2019], with the only exception that we didn’t use CG bias correction step. We used a similarly analytical mode of diffTF with HOCOMOCO v10 database [@HOCOMOCO] using the following parameters: minOverlap = 1; design formula = “~ Patient + Trisomy 12 status”.  

#### Functional enrichment analysis of SPIB ChIPseq data
_myeslf_ original (update to paper) 
We downloaded SPIB ChIPseq data [REFERENCE] from the NCBI GEO database[REFERENCE], accession GEO:GSE56857. We made use of the data for SPIB in the OCILY3 DLBCL cell line (GSM1370276). SPIB ChIP peaks were filtered for significance (q value<0.05). We used the annotatePeaks function from the package clusterProfiler[REFERENCE] to annotate the nearest gene for each ChIPpeak. We filtered peaks  within ±1kb of a TSS of a gene and performed over-representation of  KEGG and Reactome pathways (using the clusterProfiler package[REFERENCE])  amongst the resulting list of genes. 


#### Drug-Drug interaction assay

An independent patient cohort of 16 patients was used in the drug-drug interaction experiments.



#### Survival analysis of immunohistory chemistry data 

_myself_ original 
For for the purpose of visualising Kaplan-Meier plots , optimal cut-points of staining levels were calculated using maximally selected rank statistics as computed by the `R` package `maxstat` [@R-maxstat]. Based on these cut points, patients were split into two groups, and their survival data were plotted using the Kaplan-Meier method, using the R package `survminer` [@R-survminer]. 

### Ibrutinib - IBET-762 - IL-4 analysis 
#### RNAseq processing

#### Deseq2 analysis

#### ATACseq processing

#### diffTF analysis 

#### Generation of Gene Regulatory Network 

#### Proteomics Processing

#### Proteomics Analysis 



