---
output:
  pdf_document: default
  html_document: default
---

# Data {#data}
This chapter provides an overview of the datasets that are central to this thesis. It also covers the generation of resources to make the data and associated analysis publicly-available. 

The screening dataset discussed here was produced and published as part of the manuscript Bruch and Giles et al. (2021). Some analysis and figures outlined in this chapter have also been published in Bruch and Giles et al. (2021), and this is clearly stated where this is the case. 

```{r, echo = FALSE}

knitr::opts_chunk$set(
   echo = FALSE, 
   message = FALSE, 
   warning = FALSE,
   fig.align="center"
)

```


```{r setup03, echo = FALSE, message = FALSE, warning = FALSE}
#libraries
library(ggplot2)
library(tidyverse)
library(ggbeeswarm)
library(dplyr)
library(reshape2)
library(kableExtra)
library(BloodCancerMultiOmics2017)
library(gtable)
library(grid)
library(patchwork)
library(DESeq2)
library(ggrepel)
library(biobroom)

```

```{r loadData03}

#Data
load("data/drugs.RData")
load("data/cytokines.RData")
load("data/df.RData")
load("data/patMeta.RData")
load("data/Raw_Screening_Data.RData")
load("data/dds_smp.RData")
load("data/cytReceptors.RData")
load("data/cyt_and_receptors.RData")


#R code
source("R/themes_colors.R")
source("R/plotPathways.R")

#additional aesthetics
bwScale = c("0"="white","1"="black","N.A."="grey90")
lfsize = 16 # legend font size

```

```{r processData03}

#add label column to df
df <- df %>%
  dplyr::mutate(Cytokine = as.character(Cytokine)) %>%
  dplyr::mutate(Cytlabel = ifelse(Cytokine == "IL-2", "IL2",
                            ifelse(Cytokine == "IL-4", "IL4",
                                   ifelse(Cytokine == "IL-6", "IL6",
                                          ifelse(Cytokine == "IL-10", "IL10",

                                                        ifelse(Cytokine == "IL-21", "IL21",
                                                               ifelse(Cytokine == "sCD40L+IL-4","sCD40L + IL4",
                                                                      ifelse(Cytokine == "IL-15", "IL15",
                                                                             ifelse(Cytokine == "Interferon gamma","Interferon \u03B3",
                                                                                    ifelse(Cytokine == "SDF-1a","SDF-1\u03B1",
                                                                                           ifelse(Cytokine == "IL-1b","IL1\u03B2",
                                                                                                  ifelse(Cytokine == "TGF-b1","TGF\u03B21", Cytokine)))))))))))) %>%

  dplyr::mutate(Cytokine = factor(Cytokine))

#generate table of Cytokines and assosciated labels
Cytokine_labels <- df %>% 
  filter(PatientID == "Pat_001", Drug == "DMSO") %>% 
  select( Cytokine, Cytlabel)

colnames(Cytokine_labels) <- c("name", "Cytlabel")


```


## Experimental overview
### Drug-stimulus combinatorial pertubation assay and patient sample multi-omic profiling {#the-screen}
This thesis centres on a dataset of 192 CLL patient samples subjected to functional and molecular profiling. A drug-stimulus combinatorial perturbation assay (refered to below as the screen) measured the effects of 17 cytokines and microenvironmental stimuli alone and in combination with 12 drugs, to investigate the influence on spontaneous and drug-induced apoptosis (Bruch and Giles et al. 2021).  

The screen was primarily designed and performed by Peter-Martin Bruch in the Department of Medicine, University of Heidelberg, and published in the manuscript by Bruch and Giles et al. (2021). Drugs and stimuli were deposited first and patient samples second, using two 384-well plates per patient sample. Each plate contained stimulus, drug and drug - stimulus wells along with DMSO control wells (Appendix Figures \@ref(fig:drugPlot), \@ref(fig:stimulusPlot1) and \@ref(fig:stimulusPlot2)). After 48 hours of incubation at 37$^\circ$C, cell viability was assessed using the CellTiter-Glo Luminescent Cell Viability Assay. This method estimated the number of viable cells in a culture by determining the quantity of ATP present, an indicator the number of metabolically active cells. The drugs and stimuli tested in the screens are outlined in Sections \@ref(drugs) and \@ref(stimuli).
 
Multi-omics profiles for the patient samples were also available from the PACE repository [@R-BloodCancerMultiOmics2017], consisting of whole-exome sequencing, DNA-methylation, RNA-sequencing and copy number variant data. In addition, clinical follow-up data was also available for some patients, including LDT, TTT, TTFT and OS.

Collectively, these data enabled (i) characterisation of responses to microenvironmental stimuli, (ii) definition of functional patient subgroups, (iii)  profiling of molecular determinants of drug and stimulus responses (iv) mapping of drug-stimulus and drug-stimulus-gene interactions, thus shedding light on the heterogeneity of CLL biology and drug response (Figure \@ref(fig:studyOverview)).

(ref:studyOverview) "Schematic of experimental protocol. By combining 12 drugs and 17 stimuli, we systematically queried the effects of simultaneous stimulation and inhibition of critical pathways in CLL (n = 192). Integrating functional drug-stimulus response profiling with four additional omic layers, we identified pro-survival pathways, underlying molecular modulators of drug and microenvironment responses, and drug-stimulus interactions in CLL." _Figure and caption from Bruch and Giles et al. 2021._

```{r studyOverview, out.width = "70%", fig.cap='(ref:studyOverview)', echo = FALSE}

knitr::include_graphics("figures/studyOverview.eps")

```

### Additional datasets
A number of key findings emerged from the above data which warranted further investigation. In addition to the screen, I made use of a number of validatory datasets, from within our lab and from external sources. These are outlined in the Methods (Chapter \@ref(methods)) and in the relevant results chapters. 

## Data Processing {#data-processing}
### Processing the raw values obtained from the screen
This section describes the process of generating viability scores used in downstream analyses (following the same process as published in Bruch and Giles et al. (2021)). Initially, raw luminescence measurements from the experiments were read in using custom-made `R` scripts and functions. Raw values represent the luminescence readout of the CellTiter-Glo Luminescent Cell Viability Assay, which is proportional to the amount of ATP present. The ATP levels are directly proportional to the number of viable cells in the well. 

(ref:rawViabilities) Boxplots of raw viability count data prior to normalisation and log transformation. For each of the 192 patient samples, there were 50 DMSO-treated wells.  _Raw data read in and initial processing performed by Peter-Martin Bruch, figure adapted from Bruch and Giles et al. 2021._ 

```{r rawViabilities, out.width = "60%", fig.cap='(ref:rawViabilities)', echo = FALSE}

patOrder <- 
  Raw_Screening_Data %>% 
  filter(Drug == "DMSO", Cytokine == "No Cytokine")%>% 
  group_by(PatientID) %>% 
  summarize(medianViability = median(Viability)) %>% 
  arrange(medianViability) %>% 
  select(PatientID) %>% 
  unlist()

Raw_Screening_Data %>% 
  left_join(patMeta, by="PatientID") %>% 
  filter(Drug == "DMSO", Cytokine == "No Cytokine") %>% 
  mutate(PatientID = factor(PatientID, levels = patOrder)) %>% 
  
  ggplot(aes(y = Viability, x = PatientID)) +
  geom_boxplot(outlier.size = 1.1, color = colors[1]) +
  scale_y_log10() + 
  #geom_hline(yintercept=1000000, linetype='dotted', col = darkergrey)+
  #geom_hline(yintercept=300000, linetype='dotted', col = darkergrey)+
  t1 +
  theme(axis.text.x = element_blank(), 
        text = element_text(size=20),
        axis.ticks.x=element_blank(),
        panel.grid.major.x = element_blank(),
        legend.position = "bottom") +
  labs(y = "Viability of \nDMSO-treated controls", x = "Patient samples")


```


Figure \@ref(fig:rawViabilities) shows the raw viability values from the DMSO-treated wells for each patient sample. The absolute values vary between patient samples, and between measuring dates, and thus the viability values required additional normalisation. 

Each viability value was normalised to internal DMSO values of the same plate.  Specifically, the well for each stimulus, drug or drug-stimulus treatment was divided by the median of the DMSO negative control wells present on each plate, resulting in viability scores. A value of 0 indicated that all cells were killed by the treatment, and a value of 1 indicated the cells survived as well as in the negative control. 

Given the downstream analysis involved the use of linear modelling, I thus applied a natural logarithm transformation to the viability scores. This generated log-transformed control-normalised viability scores which are used for the majority of the downstream analysis. Here 0 indicates that cells survived as well as negative control. 

### Quality control and data reproducibility 
Next, several additional quality control steps were considered, with the aim of a) adjusting for any spatial effects on each screening plate, b) accounting for any batch effects between screening batches and c) testing data reproducibility. 

First, the screening data for each well on each plate was plotted in a grid corresponding to the plate layout, to visualise whether viability values were affected by their position on the plate. We discussed adjusting for any position effect by fitting a surface to the negative control wells, to generate correction factors for each well for each plate. The resulting correction factor was then subtracted from each of the treatment wells. We decided that the position effect was not sufficient to warrant this adjustment, and continued the analysis with the unadjusted values. 

With regards to batch effects, screening was performed on 16 separate days. Peter-Martin Bruch normalised data to the DMSO-treated wells on each plate and concluded that this sufficiently accounted for any batch effects. With respect to data reproducibility, technical replicates existed for single drug treatments (two) and DMSO wells (100), for each patient sample. Peter-Martin Bruch compared replicates to ensure data was reproducible. 

Thus, the data processing steps generated robust, log-transformed control-normalised viability scores that aim to capture the biological impact of the concomitant application of drugs and stimuli, outlined below. 

## Characteristics of drugs used in the screen {#drugs}
### The panel of drugs 
This study aimed to generate a systematic investigation of the impact of soluble factors on therapies and critical pathways in CLL (Bruch and Giles et al. 2021). To that end, a panel of 17 drugs was constructed, encompassing FDA-approved therapies for CLL, along with a number of drugs in clinical trial or laboratory compounds targeting pathways of interest (Figure \@ref(fig:drugCategories)). These included fludarabine (a frontline chemotherapeutic) and ibrutinib and idelalisib (BCR inhibitors). 

(ref:drugCategories) Bar plot of the drugs used in screen, indicating targets and licencing status. _Screen primarily designed by Peter-Martin Bruch. Figure from Bruch and Giles et al. 2021._

```{r  echo = FALSE}
#add row names to drugs
rownames(drugs) <- drugs$ID

# Categorise the drug into FDA approved / clinical development
drugs$target_category = as.character(drugs$target_category)
drugs$group = NA
drugs$group[which(drugs$approved_012020==1)] = "FDA approved"
drugs$group[which(drugs$devel_012020==1)] = "Clinical development/\ntool compound"

```

```{r drugCategories, fig.cap='(ref:drugCategories)', echo = FALSE, out.width='50%'}

drugCategories <- plotPathways(dat=drugs) 
drugCategories

```

The number and concentration of drugs included was limited by the size of the plate. Thus there is minimal overlap between drug targets, and two concentrations were used for each drug. The choice of concentration was guided by the results of a previous drug screen in CLL patients, performed in our lab [@JCIpaper]. The concentrations used were expected to reduce CLL viability without eliminating all cells. Drug concentrations are shown in Appendix Table \@ref(tab:drugTableAppendix).  


### Assessing drug response {#drug-responses}
To assess the quality of the drug response data, we quantified correlation coefficients for every drug pair (Bruch and Giles et al. 2021). Drugs were highly correlated if they shared identical target pathways, suggesting that the screen captures inter-individual differences in pathway dependencies, both sensitively and specifically. For example, BCR inhibitors ibrutinib, idelalisib, PRT062607 and selumetinib were all highly correlated (Figure \@ref(fig:drugCorrelations)A).

(ref:drugCorrelations) (A) Heatmap of Pearson correlation coefficients of each pair of drugs, based on log transformed viability values. See Methods Section \@ref(correlations). (B) Log transformed control-normalised viability values for all drugs that were included in the screen after quality control. p values from Student's t-test. _Figure (A) adapted from original generated with Peter-Martin Bruch for the manuscript Bruch and Giles et al. (2021)._

```{r drugCorrelations, out.width = "70%", fig.cap='(ref:drugCorrelations)', fig.width  = 12, fig.height = 7, echo = FALSE, eval = TRUE}

# Using high Drug Concentrations
#list of cytokines
thedrugs <- unique(df$Drug) %>% setdiff("DMSO")

corMat <- df %>% 
  filter(Drug %in% thedrugs, Drug_Concentration == "High", Cytokine == "No Cytokine") %>%
  dplyr::select(Log, Drug, PatientID) %>% 
  spread(key = "Drug", value = "Log") %>%
  column_to_rownames("PatientID")

#compute correlation coefficients 
cormat <-  cor(corMat, use="all.obs", method="pearson")
    
# Function to get lower triangle of the correlation matrix
get_lower_tri <- function(cormat){
  cormat[upper.tri(cormat)] <- NA
  return(cormat)
}
  # Function to get upper triangle of the correlation matrix
get_upper_tri <- function(cormat){
  cormat[lower.tri(cormat)]<- NA
  return(cormat)
}

# Function to cluster matrix 
reorder_cormat <- function(cormat){
# Use correlation between variables as distance
dd <- as.dist((1-cormat)/2)
hc <- hclust(dd)
cormat <-cormat[hc$order, hc$order]
}

cormat <- reorder_cormat(cormat)  
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE)

A<- 
ggplot(data = melted_cormat, aes(Var2, Var1, fill = value))+
 geom_tile(color = "white")+
 scale_fill_gradient2(low = palblues[1], high = palreds[7], mid = "white", 
   midpoint = 0, limit = c(-1,1), space = "Lab", 
   name="Pearson\nCorrelation") +
  t1 + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1))+
  xlab("") +
  ylab("") +
 coord_fixed() +
  theme(
  panel.grid.major = element_blank(),
  panel.border = element_blank(),
  panel.background = element_blank(),
  legend.justification = c(1, 0),
  legend.position = c(0.4, 0.7),
  legend.direction = "horizontal")+
  guides(fill = guide_colorbar(barwidth = 5, barheight = 1,
                title.position = "top", title.hjust = 0.2))

orderDrug <-
  df %>%
  dplyr::filter(Drug_Concentration == "High", Cytokine == "No Cytokine") %>%
  group_by(Drug) %>%
  dplyr::summarise(Log = median(Log), .groups="keep") %>%
  arrange(Log) 

orderDrug <- as.character(orderDrug$Drug)

df$Drug <- factor(df$Drug, levels=orderDrug)

B<- 
  df %>%
  dplyr::filter(Drug_Concentration =="High", Cytokine == "No Cytokine") %>%
  ggplot(aes(x = Drug,
             y = Log)) +
  geom_hline(yintercept = 0) +
  geom_violin(cex=0.4, alpha=0.7,color = palblues[1], aes(x = Drug,y = Log)) + 
  scale_colour_manual(name = 'Viability > 0', values = setNames(c(palreds[8], palblues[1]), c(T, F)))+
  t2 +
  guides(size="none", color = "none")+
  theme(axis.text.x = element_text(angle = 45, hjust = 1), text = element_text(size=20))+
  ylab("Log(Viability)") +
  xlab("")

#thedesign <- "
#  AAB
#"

A + B + plot_annotation(tag_levels = c('A')) 
#+  plot_layout(design = thedesign) 

```

In addition, the individual drug response profiles (Figure \@ref(fig:drugCorrelations)B) indicated that each of the drugs decreased CLL viability as expected, and in line with previous CLL drug screens performed in our lab [@JCIpaper].   

## Characteristics of stimuli used in the screen {#stimuli}
### The panel of stimuli {#stimuli-info}
The stimuli selected for the screen, their associated targets and concentrations used are described in Appendix Table \@ref(tab:stimulusTableAppendix), including 16 individual stimuli, plus HS-5 Culture Medium which encompasses the range of soluble factors secreted by the stromal cell line HS-5 (Bruch and Giles et al. 2021).

A number of studies have demonstrated the ability of various soluble factors to increase CLL viability or induce drug resistance *ex vivo* (see Section \@ref(intro-microenvironmental-pathways)). Guided by these observations, the panel of stimuli was selected so as to cover a range of key survival signals in CLL, aiming to minimise redundancy amongst the targeted pathways. The stimuli encompass a cross-section of the complex communication network between CLL cells and non-neoplastic cells, mediated by soluble factors within the tumour microenvironment (Figure \@ref(fig:microenvironmentCrosstalk)).


(ref:microenvironmentCrosstalk) A selection of interactions between CLL cells and components of the microenvironment, covered by the screen. See Section \@ref(intro-microenvironmental-pathways) for more details on these signals.  _Figure adapted from an original published in @Wiestner2015 and reproduced with permission._


```{r microenvironmentCrosstalk, fig.cap='(ref:microenvironmentCrosstalk)', out.width = "60%", echo = FALSE}

knitr::include_graphics("figures/microenvironmentCrosstalk.eps")

```

The stimuli targeted a number of critical pathways in CLL, including BCR, TLR, JAK-STAT, NF$\kappa$B and TGF$\beta$. For more information on the importance of these pathways see Section \@ref(intro-bcr-signalling) and \@ref(intro-microenvironmental-pathways). Amongst these, the roles of BCR, IL4, sCD40L and TLR stimulation were of particular interest. 

(ref:stimuliPathways)  "Overview of stimuli included in the screen and summary of their associated targets. HS-5 Culture Medium is omitted, as no specific target can be shown." _Screen primarily designed by Peter-Martin Bruch. Figure and caption from Bruch and Giles et al. 2021._

```{r stimuliPathways, fig.cap='(ref:stimuliPathways)', out.width = "60%", echo = FALSE}

knitr::include_graphics("figures/stimuliPathways.eps")

```

### Assessing stimulus response {#rna-correlations}
To guide our interpretation of the stimulus responses, it was important to determine whether heterogeneity of response to the stimuli could be caused by differences in receptor expression (Bruch and Giles et al. 2021). 

Pearson correlation coefficients were calculated to compare log-transformed control - normalised  viability values for each stimulus,  with vst-transformed RNA counts of the matching receptor(s). All coefficients were <0.4, indicating that response heterogeneity was not related to receptor expression. 

## Characteristics of patient samples used in the screen
### Overview of the molecular profiles of the patient samples
Multi-omics profiles were available for the patient samples in the screen, taken from PACE [@R-BloodCancerMultiOmics2017]. The PACE repository represents an initiative by our lab to characterise  primary tumour samples from leukemia and lymphoma patients. Patient multi-omic profiles included whole-exome sequencing, DNA-methylation, RNA-sequencing and copy number variant data. In addition, clinical information and follow-up data was also available for some patients, including sex, IGHV status and LDT, TTT, TTFT and OS. Figure \@ref(fig:geneticData) summarises the molecular characteristics of the patient samples. See also Appendix Table \@ref(tab:geneTable). 


```{r}

# select CLL samples
patM = patMeta$PatientID
patMeta <- as.data.frame(patMeta)
rownames(patMeta) <- patM
ighv = factor(setNames(patMeta[patM,"IGHV.status"], nm=patM), levels=c("U","M"))
mut1 = c("del17p", "del11q", "trisomy12", "del13q")
mut2 = c("TP53", "ATM", "SF3B1", "NOTCH1", "MYD88")
mc = patMeta[,c(8:113)] %>% as.matrix()
mc <- matrix(as.numeric(mc),  
             ncol = ncol(mc))
colnames(mc) = colnames(patMeta[,c(8:113)] )
rownames(mc) <- rownames(patMeta[,c(8:113)] )

## SELECTION OF MUTATIONS
# # include mutations with at least incidence of 4
mut2plot = names(which(sort(colSums(mc, na.rm=TRUE), decreasing=TRUE)>3))

# divide mutations into gene mut and cnv
mut2plotSV = mut2plot[grep("[[:lower:]]", mut2plot)]
mut2plotSP = mut2plot[grep("[[:upper:]]", mut2plot)]


# rearrange the top ones to match the order in mut1 and mut2
mut2plotSV = c(mut1, mut2plotSV[!mut2plotSV %in% mut1])
mut2plotSP = c(mut2, mut2plotSP[!mut2plotSP %in% mut2])
factors = data.frame(patMeta[patM, c(mut2plotSV, mut2plotSP)],
                     check.names=FALSE)

# change it to factors
for(i in 1:ncol(factors)) {
  factors[,i] = factor(factors[,i], levels=c(1,0))
}
ord = order(factors[,1], factors[,2], factors[,3], factors[,4], factors[,5],
            factors[,6], factors[,7], factors[,8], factors[,9], factors[,10],
            factors[,11], factors[,12], factors[,13], factors[,14],
            factors[,15], factors[,16], factors[,17], factors[,18],
            factors[,19], factors[,20], factors[,21], factors[,22],
            factors[,23], factors[,24], factors[,25], factors[,26],
            factors[,27], factors[,28], factors[,29], factors[,30],
            factors[,31], factors[,32])
factorsord = factors[ord,]
patM = patM[ord]
#(c(mut2plotSV, mut2plotSP))


```


```{r geneTablePrep,  echo = FALSE, eval = TRUE}



plotDF = meltWholeDF(factorsord)
plotDF$Mut =
  ifelse(sapply(plotDF$X,
                function(x) grep(x, list(mut2plotSV, mut2plotSP)))==1,"SV","SP")
plotDF$Status = "N.A."
plotDF$Status[plotDF$Measure==1 & plotDF$Mut=="SV"] = "1a"
plotDF$Status[plotDF$Measure==1 & plotDF$Mut=="SP"] = "1b"
plotDF$Status[plotDF$Measure==0] = "0"
plotDF$Status = factor(plotDF$Status, levels=c("1a","1b","0","N.A."))
plotDF$Y = factor(plotDF$Y, levels=patM)
plotDF$X = factor(plotDF$X, levels=rev(colnames(factorsord)))

mutPL = ggplotGrob(
  ggplot(data=plotDF, aes(x=Y, y=X, fill=Status)) + geom_tile() +
    scale_fill_manual(
      values=c("0"="white","1a"=colors[1],"1b"=colors[2],"N.A."="grey90"),
      name="Mutation", labels=c("WT","CNV","Gene mutation","NA")) +
    ylab("") + xlab("") +
    geom_vline(xintercept=seq(0.5,length(patM)+1,5), colour="grey60") +
    geom_hline(yintercept=seq(0.5,ncol(factorsord)+1,1), colour="grey60") +
    scale_y_discrete(expand=c(0,0)) + scale_x_discrete(expand=c(0,0)) +
    theme(axis.ticks=element_blank(), axis.text.x=element_blank(),
          axis.text.y=element_text(
            size=60, face=ifelse(levels(plotDF$X) %in% mut2plotSV,
                                 "plain","plain")),
          axis.text=element_text(margin=unit(0.5,"cm"), colour="black"),
          legend.key = element_rect(colour = "black"),
          legend.text=element_text(size=lfsize),
          legend.title=element_text(size=lfsize)))
res = table(plotDF[,c("X","Measure")])

```

```{r sex}
sexDF = data.frame(Factor="Sex", PatientID=factor(patM, levels=patM),
                   Value=patMeta[patM, "gender"])
sexPL = ggplotGrob(
  ggplot(sexDF, aes(x=PatientID, y=Factor, fill=Value)) + geom_tile() +
    scale_fill_manual(values=c("f"=palreds[6],"m"=palblues[5],"N.A."="grey90"),
                      name="Sex", labels=c("Female","Male","NA")) +
    theme(axis.ticks=element_blank(),
          axis.text=element_text(size=60, colour="black",
                                 margin=unit(0.5,"cm")),
          legend.key = element_rect(colour = "black"),
          legend.text=element_text(size=lfsize),
          legend.title=element_text(size=lfsize)))

```

```{r methylationCluster}
mcDF = data.frame(Factor="Methylation", PatientID=factor(patM, levels=patM),Value=patMeta[patM, "Methylation_Cluster"])
mcPL = ggplotGrob(
  ggplot(mcDF, aes(x=PatientID, y=Factor, fill=Value)) + geom_tile() +
    scale_fill_manual(values=c("LP"=palblues[8], "IP"=palblues[6], "HP"=palblues[4],  N.A.="grey90"), name="Methylation Cluster", labels=c("LP","IP","HP", "NA")) +
    theme(axis.ticks=element_blank(),
          axis.text=element_text(size=60, colour="black",
                                 margin=unit(0.5,"cm")),
          legend.key = element_rect(colour = "black"),
          legend.text=element_text(size=lfsize),
          legend.title=element_text(size=lfsize)))

```

```{r treatmentStatus}
treatDF = data.frame(Factor="Treated", PatientID=factor(patM, levels=patM),
                     Value=ifelse(patMeta[patM, "treatment"], 0, 1))
treatDF$Value[is.na(treatDF$Value)] = "N.A."
treatDF$Value = factor(treatDF$Value, levels=c("0","1","N.A."))
treatPL = ggplotGrob(
  ggplot(treatDF, aes(x=PatientID, y=Factor, fill=Value)) +geom_tile() +
    scale_fill_manual(values=bwScale, name="Treated",
                      labels=c("0"="No","1"="Yes","N.A."="NA")) +
    theme(axis.ticks=element_blank(),
          axis.text=element_text(size=60, colour="black",
                                 margin=unit(0.5,"cm")),
          legend.key = element_rect(colour = "black"),
          legend.text=element_text(size=lfsize),
          legend.title=element_text(size=lfsize)))

```

```{r ighvStatus, echo = FALSE}
ighvDF = data.frame(Factor="IGHV", PatientID=factor(patM, levels=patM),
                    Value=patMeta[patM, "IGHV.status"])
ighvDF$Value = ifelse(ighvDF$Value=="M", 1, 0)
ighvDF$Value[is.na(ighvDF$Value)] = "N.A."
ighvDF$Value = factor(ighvDF$Value, levels=c("0","1","N.A."))
ighvPL = ggplotGrob(
  ggplot(ighvDF, aes(x=PatientID, y=Factor, fill=Value)) + geom_tile() +
    scale_fill_manual(values=bwScale, name="IGHV",
                      labels=c("0"="Unmutated","1"="Mutated","N.A."="NA")) +
    theme(axis.ticks=element_blank(), 
          axis.text=element_text(size=60, colour="black", margin=unit(0.5,"cm")),
          legend.key=element_rect(colour = "black"),
          legend.text=element_text(size=lfsize),
          legend.title=element_text(size=lfsize)))

```

```{r echo=FALSE}
nX = length(patM)
nY = ncol(factorsord)
unY1 = 0.6*1.6
unY2 = 0.6*1.8
unX = 0.2
sp = 0.001
wdths = c(6, unX*nX, sp)
hghts = c(sp, unY1,unY1,unY1,unY1, 0.8, sp, sp ,unY2*nY, sp)
gt = gtable(widths=unit(wdths, "in"), heights=unit(hghts, "in"))
# add the plots
gt = gtable_add_grob(gt, sexPL$grobs[[whichInGrob(sexPL, "panel")]], 2, 2)
gt = gtable_add_grob(gt, treatPL$grobs[[whichInGrob(treatPL, "panel")]], 3, 2)
gt = gtable_add_grob(gt, mcPL$grobs[[whichInGrob(mcPL, "panel")]], 4, 2)
gt = gtable_add_grob(gt, ighvPL$grobs[[whichInGrob(ighvPL, "panel")]], 5, 2)
gt = gtable_add_grob(gt, mutPL$grobs[[whichInGrob(mutPL, "panel")]], 9, 2)
# add x axis
gt = gtable_add_grob(gt, mutPL$grobs[[whichInGrob(mutPL, "axis-b")]], 10, 2)
# add y axis
gt = gtable_add_grob(gt, sexPL$grobs[[whichInGrob(sexPL, "axis-l")]], 2, 1)
gt = gtable_add_grob(gt, treatPL$grobs[[whichInGrob(treatPL, "axis-l")]], 3, 1)
gt = gtable_add_grob(gt, mcPL$grobs[[whichInGrob(mcPL, "axis-l")]], 4, 1)
gt = gtable_add_grob(gt, ighvPL$grobs[[whichInGrob(ighvPL, "axis-l")]], 5, 1)
gt = gtable_add_grob(gt, mutPL$grobs[[whichInGrob(mutPL, "axis-l")]], 9, 1)
```

(ref:geneticData) Summary of the genetic characteristics of the patient samples in the screen. Annotations: Grey shows when patient sample is not annotated.  Black indicates treated / IGHV-M,  Green indicates occurrence of CNV, Yellow indicates gene is Mutated.

```{r geneticData, fig.cap='(ref:geneticData)',fig.width=sum(wdths), fig.height=sum(hghts), echo=FALSE, out.width = '70%', fig.pos = "h"}

grid.draw(gt)

```

The findings from this study have potential relevance in a clinical setting. Thus it was important to ensure that the distribution of genetic features amongst the cohort was representative of those observed in clinical practice. Other studies have determined the frequency of many recurrent genetic features in CLL [@Dohner2000; @Puente2015; @Landau2015] : the distribution of molecular lesions in our cohort is comparable to these (Bruch and Giles et al. 2021). 

## Making the data and assosciated analysis available
Collectively, this dataset represents a valuable resource providing the ability to explore genetic, epigenetic and microenvironmental modulators of survival and drug response in a heterogeneous cohort, and how these relate to clinical outcomes. With CLL samples relatively simple to obtain compared to other cancers, this project (in collaboration with the PACE initiative [@R-BloodCancerMultiOmics2017]) represents a considerable dataset containing joint functional and molecular profiling of primary cancer samples. 

An important goal of my work was thus to ensure that this dataset was both publicly available and accessible for a range of users. I aimed to ensure that our analysis was transparent and reproducible, and that others could explore the dataset for the purposes of their own research. To that end, I developed a shiny app to explore the screening dataset, and published all code and data from the manuscript Bruch and Giles et al. (2021) to an online git repository. 

The process of generating these resources is described below. 

### Shiny app
The shiny app can be found at [https://www.imbi.uni-heidelberg.de/dietrichlab/CLL_Microenvironment/](https://www.imbi.uni-heidelberg.de/dietrichlab/CLL_Microenvironment/). Figure \@ref(fig:shinyApp1) shows the home page. It consists of four tabs, covering the following: 

\begin{itemize}
    \item  \textbf{\textit{Drug and stimulus responses}} Explore drug - stimulus interactions and view log-transformed viabilities with single and combinatorial treatments
    \item \textbf{\textit{Effects of mutations on drug and stimulus responses}} Explore how drug and stimulus responses are modulated by genetic features and view log-transformed viability data stratified by mutations
    \item \textbf{\textit{Genetic predictors of drug and stimulus responses}} Explore how drug and stimulus responses are modulated by genetic features with predictor profiles from section \ref{multivariate-gene-stimulus-assosciations}
    \item \textbf{\textit{Genetic predictors of drug and stimulus interactions}} Explore how drug - stimulus interactions are further modulated by mutations, and view predictor profiles from section \ref{drug-stimulus-gene-predictor-profile}

\end{itemize}

(ref:shinyApp1) Home page of the shiny app accompanying this project. The app was published alongside Bruch and Giles et al. 2021. 

```{r shinyApp1, out.width = "100%", fig.cap='(ref:shinyApp1)', echo = FALSE}

knitr::include_graphics("figures/shiny1.eps")

```

To generate the app, I first curated the individual datasets and ran each of the individual analyses required to generate the plots outlined above. I then set up the four tab structure, and adapted the code required to generate plots in a dynamic manner. Finally, I worked on the aesthetics and interface of the app, to ensure that it was both professional and understandable. I then tested the app with number of colleagues, to ensure it was accessible and understandable by a range of users both familiar and unfamiliar with the project. I maintain the app on the university server. 


### Online code repository 
The online repository can be found at [https://github.com/Huber-group-EMBL/CLLCytokineScreen2021](https://github.com/Huber-group-EMBL/CLLCytokineScreen2021).The repository consists of the data and executable transcripts to completely reproduce the analysis described in Bruch and Giles et al. (2021).

To generate the repository, I first curated each of the individual datasets to generate objects containing the screening data, patient genetic meta data, ATACseq processed data, RNAseq counts and clinical data (including LDT, TTT and OS data), plus the follow up data including lymph node IHC experiments, shRNA knockdown experiments, and additional stimulation and inhibition assays. As many of these data contain sensitive information on patients, I anonymised each object by updating the patient IDs and removing potential identifying features such as age. 

I next arranged the analysis into seven separate scripts, one for each figure, such that all individual sections can be rendered into a single html vignette outlining the entire analysis. I ensured that the code in each script was well-annotated and relatively simple to understand and to follow. I shared the code with several colleagues to receive feedback on coding style, and ensured that the analysis could be reproduced by others, and on different operating systems. I published the repository with the along with the preprint (Bruch and Giles et al. 2021). 

These online resources are already beginning to serve as a community resource (e.g. @Lu2021, Nature Cancer), as querying them enables researchers to test new hypotheses within minutes and may obviate the need for certain small-scale experiments.

```{r remove02}

#clear data
rm(list = ls())

```
