---
output:
  pdf_document: default
  html_document: default
---

```{r eval = FALSE, include = FALSE}


#Plan: 
#-headings [x]   
#-add all figures [x]   
#-all text   <- Tomorrow, shape this up, work with what i have 
#-proof figures   
#-proof text   
#-proof numbers  
#-proof code   
#-proof of gene name etc 
#-anything else?  
#Done - send   

#Refer to dataset i used, dont talk about removed drugs,put as oneliner in methods. add mroe if time 
#Now: 
#try to docus and get drugs and cytokines done
#write rest tomorrow
#check over weekend and send off

#discucsion and methods next week

#keep fixing things as i go esp do ackonweldgements on drive up? 


```

# Data {#data}
This chapter provides an overview of the datasets that are central to this thesis, and covers the generation of  resources to make this data and associated analysis publicly-available. 

```{r, echo = FALSE}

knitr::opts_chunk$set(
   echo = FALSE, 
   message = FALSE, 
   warning = FALSE,
   fig.align="center"
)

```


```{r setup03, echo = FALSE, message = FALSE, warning = FALSE}
#libraries
library(ggplot2)
library(tidyverse)
library(ggbeeswarm)
library(dplyr)
library(reshape2)
library(kableExtra)
library(BloodCancerMultiOmics2017)
library(gtable)
library(grid)
library(patchwork)
```

```{r loadData03}

#Data
load("data/drugs.RData")
load("data/cytokines.RData")
load("data/df.RData")
load("data/patMeta.RData")
load("data/Raw_Screening_Data.RData")


#R code
source("R/themes_colors.R")
source("R/plotPathways.R")

#additional aesthetics
bwScale = c("0"="white","1"="black","N.A."="grey90")
lfsize = 16 # legend font size

```


```{r processData03}

#add label column to df
df <- df %>%
  dplyr::mutate(Cytokine = as.character(Cytokine)) %>%
  dplyr::mutate(Cytlabel = ifelse(Cytokine == "IL-2", "IL2",
                            ifelse(Cytokine == "IL-4", "IL4",
                                   ifelse(Cytokine == "IL-6", "IL6",
                                          ifelse(Cytokine == "IL-10", "IL10",

                                                        ifelse(Cytokine == "IL-21", "IL21",
                                                               ifelse(Cytokine == "sCD40L+IL-4","sCD40L + IL4",
                                                                      ifelse(Cytokine == "IL-15", "IL15",
                                                                             ifelse(Cytokine == "Interferon gamma","Interferon \u03B3",
                                                                                    ifelse(Cytokine == "SDF-1a","SDF-1\u03B1",
                                                                                           ifelse(Cytokine == "IL-1b","IL1\u03B2",
                                                                                                  ifelse(Cytokine == "TGF-b1","TGF\u03B21", Cytokine)))))))))))) %>%

  dplyr::mutate(Cytokine = factor(Cytokine))

#generate table of Cytokines and assosciated labels
Cytokine_labels <- df %>% 
  filter(PatientID == "Pat_001", Drug == "DMSO") %>% 
  select( Cytokine, Cytlabel)

colnames(Cytokine_labels) <- c("name", "Cytlabel")


```


```{r defineFunctions3}

plotPathways <-function(dat) 
{
  #Set pathway, number of drugs and group
  Pathway = NULL
  No = NULL
  Group = NULL
  #set ordering of drug cateogries (FDA Approved and  Clinical development/tool compound)
  ordM = sort(table(dat$group), decreasing = TRUE)
  
  #set ordering of pathways within categories
  ordS = tapply(dat$target_category, dat$group, function(pth) {
    sort(table(pth), decreasing = TRUE)
  })
  
  #set overall ordering
  ocur = ordS[names(ordM)]
  
  #get a dataframe of number of drugs for each pathway, split by category
  tmp = do.call(rbind, lapply(names(ocur), function(pathgroup) {
    data.frame(Group = pathgroup, Pathway = names(ocur[[pathgroup]]), 
               No = as.vector(unname(ocur[[pathgroup]])))
  }))
  
  #relevel categories
  tmp$Group = factor(tmp$Group, levels = rev(names(ocur)))
  
  #set plotting order
  lev = sort(tapply(tmp$No, tmp$Pathway, function(x) sum(x)), 
             decreasing = TRUE)
  
  #relevel pathways
  tmp$Pathway = factor(tmp$Pathway, levels = names(lev))
  
  #set height of y axis scale
  widthmax = max(lev) + 1
  
  #make plot
  g = ggplot(tmp, aes(x = Pathway, y = No, fill = Group)) + 
    geom_bar(width = 0.6, stat = "identity") + theme_bw() + 
    scale_fill_manual(values = c(drugpal[1], drugpal[2]), name = "Drug type") + 
    scale_y_continuous(breaks = seq(0, 20, 2), 
                       expand = c(0,0), limits = c(0, widthmax)) + 
    xlab("") + 
    ylab("No. of drugs") + 
    t1 + 
    theme(legend.position = c(0.7, 0.65))
  
}



```


## Experimental overview
### Drug-stimulus combinatorial pertubation assay and patient sample multi-omic profiling
This thesis centres on a cohort of 192 CLL patient samples, which were subjected to functional and molecular profiling. First CLL samples were characterised based on their responses to microenvironmental stimuli and drugs. 
A drug-stimulus combinatorial perturbation assay (refered to below as the screen) measured the effects of 17 cytokines and microenvironmental stimuli alone and in combination with 12 drugs, to investigate the influence on spontaneous and drug-induced apoptosis.  

To prepare 384-well plates for functional profiling, drugs and stimuli were deposited first and patient samples second, using one plate per patient. Each plate accommodated stimulus, drug, drug - stimulus wells along with [CHECK] DMSO control wells. After 48 hours of incubation at 37$^\circ$C, cell viability was assessed using ATP levels in each well. The 12 drugs and 17 stimuli tested in the screens are outlined in sections \@ref(drugs) and \@ref(stimuli).
 
Multi-omics profiles for the patient samples were also available from the PACE repository [@PACE], consisting of whole-exome sequencing, DNA-methylation, RNA-sequencing and copy number variant data. In addition, clinical follow-up data was also available for some patients, including LDT, TTT, TTFT and OS.

Collectively, these data enabled (i)characterisation of responses to microenvironmental stimuli, (ii) definition of functional patient subgroups, (iii)  profiling of molecular determinants of drug and stimulus responses (iv) mapping of drug - stimulus and drug - stimulus - gene interactions, thus shedding light on the heterogeneity of CLL biology and drug response (Figure \@ref(fig:studyOverview)).

(ref:studyOverview) Schematic of experimental protocol. By combining 12 drugs and 17 stimuli, we systematically queried the effects of simultaneous stimulation and inhibition of critical pathways in CLL (n = 192). Integrating functional drug-stimulus response profiling with four additional omics layers, we identified pro-survival pathways, underlying molecular modulators of drug and microenvironment responses, and drug-stimulus interactions in CLL. _Figure and caption from Bruch & Giles et al. 2021._

```{r studyOverview, out.width = "70%", fig.cap='(ref:studyOverview)', echo = FALSE}

knitr::include_graphics("figures/studyOverview.eps")

```

### Additional datasets
A number of key findings emerged from the above data which warranted further investigation. In addition to the screen, this thesis makes use of a number of validatory datasets, from within our lab and from external sources. These are outlined in the Methods (Chapter \@ref(methods)) and in the relevant results chapters. 

## Data Processing
### Processing the raw values obtained from the screen
Raw luminescence measurements  were read in,  normalised  and underwent quality control checks to generate the viability measurements which were used for downstream analysis. Initially, raw data from the experiements were read in using readxl package, fucntions in R were used fro reading in and nroamlsiing. 

Rw data read in via, and processed

Raw values 
- we looked at them 
-some are quite low 
-we cehcked their responses - seem ok 
-reran analysis 
-dwonst seem to changed very much 

(Figure \@ref(fig:rawViabilities)). 

(ref:rawViabilities) Boxplots of raw viability count data prior to normalisation and log transformation. For each of the 192 patient samples, there were  50 DMSO-treated wells. Dashed horizontal lines indicate potential cut-offs for samples that showed low viability with DMSO control treatment. 

```{r rawViabilities, out.width = "70%", fig.cap='(ref:rawViabilities)', echo = FALSE}

patOrder <- 
  Raw_Screening_Data %>% 
  filter(Drug == "DMSO", Cytokine == "No Cytokine")%>% 
  group_by(PatientID) %>% 
  summarize(medianViability = median(Viability)) %>% 
  arrange(medianViability) %>% 
  select(PatientID) %>% 
  unlist()

Raw_Screening_Data %>% 
  left_join(patMeta, by="PatientID") %>% 
  filter(Drug == "DMSO", Cytokine == "No Cytokine") %>% 
  mutate(PatientID = factor(PatientID, levels = patOrder)) %>% 
  
  ggplot(aes(y = Viability, x = PatientID)) +
  geom_boxplot(outlier.size = 1.1, color = colors[1]) +
  scale_y_log10() + 
  geom_hline(yintercept=1000000, linetype='dotted', col = darkergrey)+
  geom_hline(yintercept=300000, linetype='dotted', col = darkergrey)+
  t1 +
  theme(axis.text.x = element_blank(), 
        text = element_text(size=20),
        axis.ticks.x=element_blank(),
        panel.grid.major.x = element_blank(),
        legend.position = "bottom") +
  labs(y = "Viability of \nDMSO-treated controls", x = "Patient samples")


```

In this section, we provide explanation of how we normalized the raw luminescence measurements into the viabilities, which were later used for analysis.

WE divided by the mean of .. . controls, then eprform log tranform . Why did we do this.... 


### Quality control and data reproducibility 
With the nromalised values, we could perfrom additional QC steps: 

Quality control aims to: 
cehcek there are no spatial effects relating to the plate 
check whether ther are bathc effects between plats
is each palte reporuible in prinicple 

1. spatial effects - no show plate plot, bascially we didnt see an edge effect 
2. batch effects (leave for now, ask), 16 batches, didnt see any 
3. data reproducibility (leave for now, ask) merged duplaictes. 

this is waht happened, dont over -empashie, its on Peter really, this isnt my PhD 

= log-trasnforemd nromalised values used in rest of analysis....

## Characteristics of drugs used in the screen {#drugs}
### Drug characteristics
We aimed to generate a systematic study of the impact of soluble factors on therapies in CLL. In addition, we were interested to dissect the effect of simultaneous inhibition and stimuli of critical pathways in CLL. To that end, a panel of 17 drugs was constructed, encompassing FDA-approved therapies for CLL, along with a number of drugs in clinical trial or laboratory compounds targeting pathways of interest (Figure \@ref(fig:drugCategories)). These included fludarabine (a frontline chemotherapeutic) and ibrutinib and idelalisib (newer BCR inhibitors). 

(ref:drugCategories) Bar plot of the drugs used in screen, indicating targets and licencing status. 

```{r  echo = FALSE}
#add row names to drugs
rownames(drugs) <- drugs$ID

# Categorise the drug into FDA approved / clinical development
drugs$target_category = as.character(drugs$target_category)
drugs$group = NA
drugs$group[which(drugs$approved_012020==1)] = "FDA approved"
drugs$group[which(drugs$devel_012020==1)] = "Clinical development/\ntool compound"

```

```{r drugCategories, fig.cap='(ref:drugCategories)', echo = FALSE, out.width='50%'}

drugCategories <- plotPathways(dat=drugs) 
drugCategories

```

The number and concentration of drugs included was limited by the size of the plate. Thus there is minimal overlap between drug targets, and two concentrations were used for each drug. The choice of concentration was guided by the results of a previous drug screen in CLL patients, performed in our lab [@JCIpaper]. The concentrations used were expected to reduce CLL viability without eliminating all cells. Drug concentrations are shown in Table \@ref(tab:drugTable).  

(ref:drugTable) Drug characteristics, including targets, concentrations and licencing status. 

```{r drugTable, out.width = "40%", fig.cap='(ref:drugTable)', echo = FALSE}

drugs %>%
  dplyr::select(-ID, -approved_012020,-devel_012020, -target_category, -distributor, -cat_no) %>%
  dplyr::rename(Drug=Name,
                `Main targets`=main_targets,
                `Drug Group`=group,
                Pathway=pathway) %>%
  kable() %>%
  row_spec(0,bold=TRUE) %>%
kable_styling(latex_options = "striped", 
              stripe_color = colors[7],
                html_font = "Helvetica")

```

### Assessing drug response
To assess the quality of the drug response data, I quantified correlation coefficients for every drug pair. 
Drugs were highly correlated if they shared identical target pathways, suggesting that the screen captures interindividual differences in pathway dependencies, both sensitively and specifically. For example, BCR inhibitors ibrutinib, idelalisib, PRT062607 and selumetinib were all highly correlated (Figure \@ref(fig:drugCorrelations)).

(ref:drugCorrelations) Heatmap of Pearson correlation coefficients of each pair of drugs, based on log transformed viability values. See Methods section \@ref(correlations). _Figure adapted from @Giles2021 ._

```{r drugCorrelations, out.width = "70%", fig.cap='(ref:drugCorrelations)', echo = FALSE, eval = TRUE}

# Using high Drug Concentrations
#list of cytokines
thedrugs <- unique(df$Drug) %>% setdiff("DMSO")

corMat <- df %>% 
  filter(Drug %in% thedrugs, Drug_Concentration == "High", Cytokine == "No Cytokine") %>%
  dplyr::select(Log, Drug, PatientID) %>% 
  spread(key = "Drug", value = "Log") %>%
  column_to_rownames("PatientID")

#compute correlation coefficients 
cormat <-  cor(corMat, use="all.obs", method="pearson")
    
# Function to get lower triangle of the correlation matrix
get_lower_tri <- function(cormat){
  cormat[upper.tri(cormat)] <- NA
  return(cormat)
}
  # Function to get upper triangle of the correlation matrix
get_upper_tri <- function(cormat){
  cormat[lower.tri(cormat)]<- NA
  return(cormat)
}

# Function to cluster matrix 
reorder_cormat <- function(cormat){
# Use correlation between variables as distance
dd <- as.dist((1-cormat)/2)
hc <- hclust(dd)
cormat <-cormat[hc$order, hc$order]
}

cormat <- reorder_cormat(cormat)  
upper_tri <- get_upper_tri(cormat)
melted_cormat <- melt(upper_tri, na.rm = TRUE)


ggplot(data = melted_cormat, aes(Var2, Var1, fill = value))+
 geom_tile(color = "white")+
 scale_fill_gradient2(low = palblues[1], high = palreds[7], mid = "white", 
   midpoint = 0, limit = c(-1,1), space = "Lab", 
   name="Pearson\nCorrelation") +
  t1 + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1))+
  xlab("") +
  ylab("") +
 coord_fixed() +
  theme(
  panel.grid.major = element_blank(),
  panel.border = element_blank(),
  panel.background = element_blank(),
  axis.ticks = element_blank(),
  axis.text.x = element_blank(),
  legend.justification = c(1, 0),
  legend.position = c(0.4, 0.7),
  legend.direction = "horizontal")+
  guides(fill = guide_colorbar(barwidth = 5, barheight = 1,
                title.position = "top", title.hjust = 0.2))

```

In addition,the individual drug response profiles (Figure \@ref(fig:drugResponses)) indicated that each of the drugs decreased CLL viability as expected, and in line with previous CLL drug screens performed in our lab [@JCIpaper].   

(ref:drugResponses) Log transformed viability values for all drugs that were included in the screen after quality control. p values from student's t test. 

```{r drugResponses, fig.cap='(ref:drugResponses)', out.width="70%", echo = FALSE}


orderDrug <-
  df %>%
  dplyr::filter(Drug_Concentration =="High", Cytokine == "No Cytokine") %>%
  group_by(Drug) %>%
  dplyr::summarise(Log = median(Log), .groups="keep") %>%
  arrange(Log) 

orderDrug <- as.character(orderDrug$Drug)

df$Drug <- factor(df$Drug, levels=orderDrug)

  df %>%
  dplyr::filter(Drug_Concentration =="High", Cytokine == "No Cytokine") %>%
  ggplot(aes(x = Drug,
             y = Log)) +
  geom_hline(yintercept = 0) +
  geom_beeswarm(cex=0.4, alpha=0.7, size=2, aes(x = Drug,y = Log, colour = Log>0)) +
  scale_colour_manual(name = 'Viability > 0', values = setNames(c(palreds[8], palblues[1]), c(T, F)))+
  t2 +
  guides(size="none", color = "none")+
  theme(axis.text.x = element_text(angle = 45, hjust = 1), text = element_text(size=20))+
  ylab("Log transformed viability") +
  xlab("")





```

## Characteristics of stimuli used in the screen {#stimuli}
### The panel of stimuli {#stimuli-info}
The stimuli selected for the screen, their associated targets and concentrations used  are described in Table \@ref(tab:cytTable). They include 16 individual stimuli, plus HS-5 Culture Medium which encompasses the range of soluble factors secreted by the stromal cell line HS-5. 

(ref:cytTable) Raw viabilities  

```{r cytTable, out.width = "40%", fig.cap='(ref:cytTable)', echo = FALSE}


#tidy names for table
cytokines$stimulus[grepl("IL-4 human recombinant animal component free",cytokines$stimulus)]<-"IL4 human recombinant animal component free"
cytokines$stimulus[grepl("IL-2 human recombinant animal component free",cytokines$stimulus)]<-"IL2 human recombinant animal component free"
cytokines$stimulus[grepl("Human IL-1beta",cytokines$stimulus)]<-"Human IL-1 beta"
cytokines$stimulus[grepl("Human IL-15",cytokines$stimulus)]<-"Human IL15"
cytokines$stimulus[grepl("Human IFN-gamma",cytokines$stimulus)]<-"Human Interferon gamma"
cytokines$stimulus[grepl("IL-10 human Animal component free",cytokines$stimulus)]<-"IL10 human Animal component free"
cytokines$stimulus[grepl("Human TGF-beta1",cytokines$stimulus)]<-"Human TGF\u03B21"
cytokines$stimulus[grepl("Human IL-6",cytokines$stimulus)]<-"Human IL6"
cytokines$stimulus[grepl("alpha",cytokines$stimulus)]<-"Human SDF1 alpha (CXCL12)"
cytokines$stimulus[grepl("HS-5 konditioniertes Medium",cytokines$stimulus)]<-"HS-5 conditioned medium"

cytokines %>%
  left_join(Cytokine_labels, by = "name") %>%
  dplyr::select(-name, -additional_pathways, -source, -company, -cat_no, -lot_no) %>%
  dplyr::rename(Stimulus=stimulus, Name = Cytlabel, Concentration=conc, Pathway=pathway) %>%

  select(Stimulus, Name,  everything()) %>%
  kable() %>%
  row_spec(0,bold=TRUE) %>%
kable_styling(latex_options = "striped", 
              stripe_color = colors[7],
                html_font = "Helvetica")



```

A number of studies have demonstrated the ability of various soluble factors to increase CLL viability or induce drug resistance *ex vivo* (see section \@ref(intro-microenvironmental-pathways). Guided by these observations, the panel of stimuli was selected so as to cover a range of key survival signals in CLL, aiming to minimise redundancy amongst the targeted pathways. The stimuli encompass a cross-section of the complex communication network between CLL cells and non-neoplastic cells, mediated by soluble factors within the tumour microenvrionment. \@ref(fig:microenvironmentCrosstalk).


(ref:microenvironmentCrosstalk) [CHECK THIS] _Adapted from: Weistner et al 2015_

```{r microenvironmentCrosstalk, fig.cap='(ref:microenvironmentOverview)', out.width = "60%", echo = FALSE}

knitr::include_graphics("figures/microenvironmentCrosstalk.eps")

```

The stimuli activate a number of critical pathways in CLL, including BCR, TLR, JAK-STAT, NF$\kappa$B and TGF$\beta$. For more information on the importance of these pathways see section \@ref(intro-bcr-signalling) and \@ref(intro-microenvironmental-pathways). Amongst these, the roles of BCR, IL4, sCD40L and TLR stimulation were of particular interest. 

(ref:stimuliPathways)  Overview of stimuli included in the screen and summary of their associated targets. HS-5 Culture Medium is omitted, as no specific target can be shown. _Figure and caption from Bruch & Giles et al. 2021._
```{r stimuliPathways, fig.cap='(ref:stimuliPathways)', out.width = "60%", echo = FALSE}

knitr::include_graphics("figures/stimuliPathways.eps")

```

### Assessing stimulus response
Before begining the downstream analysis, it was first important to determine whether  heterogeneity of response to the stimuli could be caused by differences receptor expression, rather than some other cell-intrinsic feature. This would guide our interpretation of the stimulus responses. 

To determine this, we calculated Pearson correlation coefficients to compare log-transformed control - normalised   viability values for each stimulus,  with vst-transformed RNA counts of the matching receptor(s). All Pearson coefficients were less than 0.4, indicating that heterogeneity of response was not related to  differential receptor expression. 

(ref:RNAcorrelations) Volcano plot showing p-values versus Pearson correlation coefficients, for the correlation between control - normalised log viability values for each stimulus and expression of the corresponding stimulus receptor, in matched patient samples. Expression calculated using transformed RNA counts (using VST method) of untreated CLL sample. _Figure and caption from Bruch and Giles et al. 2021_ 

```{r RNAcorrelations, fig.cap='(ref:RNAcorrelations)', message = FALSE,  eval = FALSE, echo = FALSE, fig.height=10, fig.width=7, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

#VST
vsd <- vst(dds_smp, blind=TRUE)

#rename transformed counts
assayNames(vsd) <- "transformedcounts"
ColData <- as.data.frame(colData(dds_smp))
tidy_RNA_data <-
  vsd %>%
  tidy() %>%
  filter(gene %in% cytReceptors$ENSEMBLID) %>%
  dplyr::rename(PatientID=sample) %>%
  left_join(cyt_and_receptors, by=c("gene"="ENSEMBLID"))

RNA_Screen_data <-
  df %>%
  dplyr::filter(Drug=="DMSO", Cytokine %in% c("Resiquimod","IL-4" ,"TGF-b1","IL-1b",
                                     "Interferon gamma","SDF-1a","sCD40L",
                                     "sCD40L+IL-4","soluble anti-IgM","CpG ODN",
                                     "IL-6","IL-10","IL-21","anti-IgM Beads",
                                     "HS-5 CM","IL-15","BAFF","IL-2")) %>%
  
  dplyr::filter(PatientID %in% unique(tidy_RNA_data$PatientID)) %>%
  
  dplyr::select(PatientID, Cytokine, Normalized) %>%
  
  left_join(tidy_RNA_data, by=c("PatientID", "Cytokine"))


Cytokine_Receptor_combinations <-
  RNA_Screen_data %>%
  dplyr::group_by(Cytokine, Receptor) %>%
  dplyr::summarize(.groups = "keep") %>%
  dplyr::filter(!is.na(Receptor))

Cytokine_Receptor_combinations$p.value <- as.numeric(NA)
Cytokine_Receptor_combinations$pearson_R <- as.numeric(NA)

for(i in 1:nrow(Cytokine_Receptor_combinations)) {
  
  x = Cytokine_Receptor_combinations[i,]
  Correlation_data_matrix<-RNA_Screen_data %>%
    mutate(Cytokine_Receptor=paste0(Cytokine,":", Receptor)) %>%
    dplyr::filter(Cytokine_Receptor==paste0(x[1],":",x[2])) %>%
    dplyr::select(Normalized, value) %>%
    as.matrix()
  
  correlation_res <- cor.test(Correlation_data_matrix[,1],Correlation_data_matrix[,2], method = "pearson")
  Cytokine_Receptor_combinations[which(Cytokine_Receptor_combinations$Cytokine==unlist(x[1])&Cytokine_Receptor_combinations$Receptor==unlist(x[2])),]$p.value=correlation_res$p.value
  
  Cytokine_Receptor_combinations[which(Cytokine_Receptor_combinations$Cytokine==unlist(x[1])&Cytokine_Receptor_combinations$Receptor==unlist(x[2])),]$pearson_R<-correlation_res$estimate
  
}
Cytokine_Receptor_combinations <- Cytokine_Receptor_combinations%>%
  mutate(adj.p.value=p.adjust(p.value, method="fdr"))
colnames(Cytokine_labels) <- c("Cytokine", "Cytlabel")
Cytokine_Receptor_combinations <- left_join(Cytokine_Receptor_combinations, Cytokine_labels, by = "Cytokine")
Cytokine_Receptor_combinations$Receptor[grepl("IL-10 R alpha",Cytokine_Receptor_combinations$Receptor)]<-"IL10 R \u03B1"
Cytokine_Receptor_combinations$Receptor[grepl("IL-10 R beta",Cytokine_Receptor_combinations$Receptor)]<-"IL10 R \u03B2"
Cytokine_Receptor_combinations$Receptor[grepl("IL-15 R alpha",Cytokine_Receptor_combinations$Receptor)]<-"IL15 R \u03B1"
Cytokine_Receptor_combinations$Receptor[grepl("IL-2 R beta",Cytokine_Receptor_combinations$Receptor)]<-"IL2 R \u03B2"
Cytokine_Receptor_combinations$Receptor[grepl("IL-2 R gamma",Cytokine_Receptor_combinations$Receptor)]<-"IL2 R \u03B3"
 
Cytokine_Receptor_combinations$Receptor[grepl("IL-1 R1",Cytokine_Receptor_combinations$Receptor)]<-"IL1 R1"
Cytokine_Receptor_combinations$Receptor[grepl("IL-2 R alpha",Cytokine_Receptor_combinations$Receptor)]<-"IL2 R \u03B1"
Cytokine_Receptor_combinations$Receptor[grepl("IL-13 R A1",Cytokine_Receptor_combinations$Receptor)]<-"IL13 R A1"
Cytokine_Receptor_combinations$Receptor[grepl("IFN-gamma R2",Cytokine_Receptor_combinations$Receptor)]<-"IFN \u03B3 R2"
Cytokine_Receptor_combinations$Receptor[grepl("IL-6 R",Cytokine_Receptor_combinations$Receptor)]<-"IL6 R"
Cytokine_Receptor_combinations$Receptor[grepl("IFN gamma R1",Cytokine_Receptor_combinations$Receptor)]<-"IFN \u03B3 R1"
Cytokine_Receptor_combinations$Receptor[grepl("IL-4 R alpha",Cytokine_Receptor_combinations$Receptor)]<-"IL4 R \u03B1"
CR_volcanoplot <-
  ggplot(Cytokine_Receptor_combinations,
         aes(x=pearson_R,
             y=-log10(adj.p.value)))+
  geom_point(data=filter(Cytokine_Receptor_combinations, pearson_R>(-0.4)&pearson_R<0.4), color="grey")+
  geom_point(data=filter(Cytokine_Receptor_combinations, pearson_R<(-0.4)|pearson_R>0.4), color="red")+
   geom_label_repel(data=filter(Cytokine_Receptor_combinations, pearson_R<(-0.4)|pearson_R>0.4), nudge_x = 0.07, color="red", aes(label=paste0(Cytlabel, " | ", Receptor)), size=5)+
  geom_label_repel(data=filter(Cytokine_Receptor_combinations, pearson_R<(-0.1)|pearson_R>0.1, !(pearson_R<(-0.4)|pearson_R>0.4)), nudge_x = 0.07, color="grey", aes(label=paste0(Cytlabel, " | ", Receptor)), size=5) + 
  coord_cartesian(xlim=c(-0.45,0.45)) +
  geom_vline(xintercept = -0.4, linetype=3) +
  geom_vline(xintercept = 0.4, linetype=3) +
  theme_minimal() +
  scale_x_continuous(breaks = c(-0.2,0,0.2)) +
  xlab("Pearson Correlation Coefficient") +
  ylab("-log10(BH adjusted p-value)") +
  t2

CR_volcanoplot

```


## Characteristics of patient samples used in the screen
### Overview of the molecular profiles of the patient samples
Multi-omics profiles were available for the patient samples in the screen, taken from the Primary Cancer Cell Encyclopedia (PACE). The PACE repository represents an initiate by our lab to characterise the multi-omic characteristics of  primary tumour samples from leukemia and lymphoma patients. Patient multi-omic profiles included whole-exome sequencing, DNA-methylation, RNA-sequencing and copy number variant data. In addition, clinical information and follow-up data was also available for some patients, including their sex, IGHV status and LDT, TTT, TTFT and OS. Figure \@ref(fig:geneticData) summarises the molecular characteristics of the patient samples. For a summary of the number of samples with each mutation, see Appendix Table \@ref(tab:geneTable).

The findings from this study have potential relevance in a clinical setting. Thus it was important to ensure that the distribution of genetic features amongst the cohort was representative of those observed in clinical practice. Other studies have determined the expected frequency of many recurrent genetic features in CLL [@Dohner2000;@Puente2015;@Landau2015] : the distribution of molecular lesions in our cohort is comparable to these. 

```{r}

# select CLL samples
patM = patMeta$PatientID
patMeta <- as.data.frame(patMeta)
rownames(patMeta) <- patM
ighv = factor(setNames(patMeta[patM,"IGHV.status"], nm=patM), levels=c("U","M"))
mut1 = c("del17p", "del11q", "trisomy12", "del13q")
mut2 = c("TP53", "ATM", "SF3B1", "NOTCH1", "MYD88")
mc = patMeta[,c(8:113)] %>% as.matrix()
mc <- matrix(as.numeric(mc),  
             ncol = ncol(mc))
colnames(mc) = colnames(patMeta[,c(8:113)] )
rownames(mc) <- rownames(patMeta[,c(8:113)] )

## SELECTION OF MUTATIONS
# # include mutations with at least incidence of 4
mut2plot = names(which(sort(colSums(mc, na.rm=TRUE), decreasing=TRUE)>3))

# divide mutations into gene mut and cnv
mut2plotSV = mut2plot[grep("[[:lower:]]", mut2plot)]
mut2plotSP = mut2plot[grep("[[:upper:]]", mut2plot)]


# rearrange the top ones to match the order in mut1 and mut2
mut2plotSV = c(mut1, mut2plotSV[!mut2plotSV %in% mut1])
mut2plotSP = c(mut2, mut2plotSP[!mut2plotSP %in% mut2])
factors = data.frame(patMeta[patM, c(mut2plotSV, mut2plotSP)],
                     check.names=FALSE)

# change it to factors
for(i in 1:ncol(factors)) {
  factors[,i] = factor(factors[,i], levels=c(1,0))
}
ord = order(factors[,1], factors[,2], factors[,3], factors[,4], factors[,5],
            factors[,6], factors[,7], factors[,8], factors[,9], factors[,10],
            factors[,11], factors[,12], factors[,13], factors[,14],
            factors[,15], factors[,16], factors[,17], factors[,18],
            factors[,19], factors[,20], factors[,21], factors[,22],
            factors[,23], factors[,24], factors[,25], factors[,26],
            factors[,27], factors[,28], factors[,29], factors[,30],
            factors[,31], factors[,32])
factorsord = factors[ord,]
patM = patM[ord]
#(c(mut2plotSV, mut2plotSP))


```


```{r geneTablePrep,  echo = FALSE, eval = TRUE}



plotDF = meltWholeDF(factorsord)
plotDF$Mut =
  ifelse(sapply(plotDF$X,
                function(x) grep(x, list(mut2plotSV, mut2plotSP)))==1,"SV","SP")
plotDF$Status = "N.A."
plotDF$Status[plotDF$Measure==1 & plotDF$Mut=="SV"] = "1a"
plotDF$Status[plotDF$Measure==1 & plotDF$Mut=="SP"] = "1b"
plotDF$Status[plotDF$Measure==0] = "0"
plotDF$Status = factor(plotDF$Status, levels=c("1a","1b","0","N.A."))
plotDF$Y = factor(plotDF$Y, levels=patM)
plotDF$X = factor(plotDF$X, levels=rev(colnames(factorsord)))

mutPL = ggplotGrob(
  ggplot(data=plotDF, aes(x=Y, y=X, fill=Status)) + geom_tile() +
    scale_fill_manual(
      values=c("0"="white","1a"=colors[1],"1b"=colors[2],"N.A."="grey90"),
      name="Mutation", labels=c("WT","CNV","Gene mutation","NA")) +
    ylab("") + xlab("") +
    geom_vline(xintercept=seq(0.5,length(patM)+1,5), colour="grey60") +
    geom_hline(yintercept=seq(0.5,ncol(factorsord)+1,1), colour="grey60") +
    scale_y_discrete(expand=c(0,0)) + scale_x_discrete(expand=c(0,0)) +
    theme(axis.ticks=element_blank(), axis.text.x=element_blank(),
          axis.text.y=element_text(
            size=60, face=ifelse(levels(plotDF$X) %in% mut2plotSV,
                                 "plain","plain")),
          axis.text=element_text(margin=unit(0.5,"cm"), colour="black"),
          legend.key = element_rect(colour = "black"),
          legend.text=element_text(size=lfsize),
          legend.title=element_text(size=lfsize)))
res = table(plotDF[,c("X","Measure")])

```

```{r sex}
sexDF = data.frame(Factor="Sex", PatientID=factor(patM, levels=patM),
                   Value=patMeta[patM, "gender"])
sexPL = ggplotGrob(
  ggplot(sexDF, aes(x=PatientID, y=Factor, fill=Value)) + geom_tile() +
    scale_fill_manual(values=c("f"=palreds[6],"m"=palblues[5],"N.A."="grey90"),
                      name="Sex", labels=c("Female","Male","NA")) +
    theme(axis.ticks=element_blank(),
          axis.text=element_text(size=60, colour="black",
                                 margin=unit(0.5,"cm")),
          legend.key = element_rect(colour = "black"),
          legend.text=element_text(size=lfsize),
          legend.title=element_text(size=lfsize)))

```

```{r methylationCluster}
mcDF = data.frame(Factor="Methylation", PatientID=factor(patM, levels=patM),
                   Value=patMeta[patM, "Methylation_Cluster"])
mcPL = ggplotGrob(
  ggplot(mcDF, aes(x=PatientID, y=Factor, fill=Value)) + geom_tile() +
    scale_fill_manual(values=c("LP"=palblues[8], "IP"=palblues[6], "HP"=palblues[4],  N.A.="grey90"), name="Methylation Cluster", labels=c("LP","IP","HP", "NA")) +
    theme(axis.ticks=element_blank(),
          axis.text=element_text(size=60, colour="black",
                                 margin=unit(0.5,"cm")),
          legend.key = element_rect(colour = "black"),
          legend.text=element_text(size=lfsize),
          legend.title=element_text(size=lfsize)))

```

```{r treatmentStatus}
treatDF = data.frame(Factor="Treated", PatientID=factor(patM, levels=patM),
                     Value=ifelse(patMeta[patM, "treatment"], 0, 1))
treatDF$Value[is.na(treatDF$Value)] = "N.A."
treatDF$Value = factor(treatDF$Value, levels=c("0","1","N.A."))
treatPL = ggplotGrob(
  ggplot(treatDF, aes(x=PatientID, y=Factor, fill=Value)) +geom_tile() +
    scale_fill_manual(values=bwScale, name="Treated",
                      labels=c("0"="No","1"="Yes","N.A."="NA")) +
    theme(axis.ticks=element_blank(),
          axis.text=element_text(size=60, colour="black",
                                 margin=unit(0.5,"cm")),
          legend.key = element_rect(colour = "black"),
          legend.text=element_text(size=lfsize),
          legend.title=element_text(size=lfsize)))

```

```{r ighvStatus}
ighvDF = data.frame(Factor="IGHV", PatientID=factor(patM, levels=patM),
                    Value=patMeta[patM, "IGHV.status"])
ighvDF$Value = ifelse(ighvDF$Value=="M", 1, 0)
ighvDF$Value[is.na(ighvDF$Value)] = "N.A."
ighvDF$Value = factor(ighvDF$Value, levels=c("0","1","N.A."))
ighvPL = ggplotGrob(
  ggplot(ighvDF, aes(x=PatientID, y=Factor, fill=Value)) + geom_tile() +
    scale_fill_manual(values=bwScale, name="IGHV",
                      labels=c("0"="Unmutated","1"="Mutated","N.A."="NA")) +
    theme(axis.ticks=element_blank(), 
          axis.text=element_text(size=60, colour="black", margin=unit(0.5,"cm")),
          legend.key=element_rect(colour = "black"),
          legend.text=element_text(size=lfsize),
          legend.title=element_text(size=lfsize)))

```

```{r echo=FALSE}
nX = length(patM)
nY = ncol(factorsord)
unY1 = 0.6*1.6
unY2 = 0.6*1.8
unX = 0.2
sp = 0.001
wdths = c(6, unX*nX, sp)
hghts = c(sp, unY1,unY1,unY1,unY1, 0.8, sp, sp ,unY2*nY, sp)
gt = gtable(widths=unit(wdths, "in"), heights=unit(hghts, "in"))
# add the plots
gt = gtable_add_grob(gt, sexPL$grobs[[whichInGrob(sexPL, "panel")]], 2, 2)
gt = gtable_add_grob(gt, treatPL$grobs[[whichInGrob(treatPL, "panel")]], 3, 2)
gt = gtable_add_grob(gt, mcPL$grobs[[whichInGrob(mcPL, "panel")]], 4, 2)
gt = gtable_add_grob(gt, ighvPL$grobs[[whichInGrob(ighvPL, "panel")]], 5, 2)
gt = gtable_add_grob(gt, mutPL$grobs[[whichInGrob(mutPL, "panel")]], 9, 2)
# add x axis
gt = gtable_add_grob(gt, mutPL$grobs[[whichInGrob(mutPL, "axis-b")]], 10, 2)
# add y axis
gt = gtable_add_grob(gt, sexPL$grobs[[whichInGrob(sexPL, "axis-l")]], 2, 1)
gt = gtable_add_grob(gt, treatPL$grobs[[whichInGrob(treatPL, "axis-l")]], 3, 1)
gt = gtable_add_grob(gt, mcPL$grobs[[whichInGrob(mcPL, "axis-l")]], 4, 1)
gt = gtable_add_grob(gt, ighvPL$grobs[[whichInGrob(ighvPL, "axis-l")]], 5, 1)
gt = gtable_add_grob(gt, mutPL$grobs[[whichInGrob(mutPL, "axis-l")]], 9, 1)
```


(Figure \@ref(fig:geneticData)). 

(ref:geneticData) Summary of genetic data 

```{r geneticData, out.width = "70%", fig.cap='(ref:geneticData)', echo=FALSE, fig.width=sum(wdths), fig.height=sum(hghts)}


wrap_elements(grid.draw(gt)) / wrap_elements(BloodCancerMultiOmics2017:::drawLegends(plobj=list(sexPL,treatPL,ighvPL,mutPL))) + plot_layout(heights = c(1,0.1))


```


## Publishing the data and assosciated analysis
As explained in section \@ref(microenvironmental-pathways-intro), many studies have characterised the impact of individual omic layers, and individual soluble factors in CLL. However, studies investigating the interplay  cell-intrinsic and cell-extrinsic features in CLL are lacking. Collectively, this dataset represents a valuable resource providing the ability to explore genetic, epigenetic and microenvironmental modulators of survival and drug response in a heterogeneous cohort, and how these relate to clinical outcomes. With CLL samples relatively simple to obtain compared to other cancers, this project (in collaboration with the PACE initiative) represents a considerable dataset of joint functional and molecular profiling of primary cancer samples. 

Thus, an important goal of my work was to ensure that this dataset was both publicly available and accessible for a range of users, including medics and bioinformaticians. I aimed to ensure that our analysis was transparent and reproducible, and that others could explore the dataset for the purposes of their own research. To that end, I developed a shiny app to explore the screening dataset, and published all code and data from the  manuscript  Bruch and Giles et al. 2021 to an online git repository. 

### Shiny app
The shiny app can be found [here](https://www.imbi.uni-heidelberg.de/dietrichlab/CLL_Microenvironment/). Figure \@ref(fig:shinyApp1) shows the home page. It consists of four tabs, covering the following: 
* _Drug and stimulus responses_ Explore drug - stimulus interactions and view log-transformed viabilities with single and combinatorial treatments (Figure \@ref(fig:shinyApp3)).
* _Effects of mutations on drug and stimulus responses_ Explore how drug and stimulus responses are modulated by genetic features and view log-transformed viability data stratified by mutations
* _Genetic predictors of drug and stimulus responses_ Explore how drug and stimulus responses are modulated by genetic features with predictor profiles from section \@ref(multivariate-gene-stimulus-assosciations)
* _Genetic predictors of drug and stimulus interactions_ Explore how drug - stimulus interactions are further modulated by mutations, and view predictor profiles from section \@ref(drug-stimulus-gene-predictor-profile)

(ref:shinyApp1) Home page of the shiny app accompanying this project. The app was published along side @Giles2021. 

```{r shinyApp1, out.width = "100%", fig.cap='(ref:shinyApp1)', echo = FALSE}

knitr::include_graphics("figures/shiny1.eps")

```

To generate the app, I first curated the individual datasets and ran each of the individual analyses required to generate the plots outline above. I then set up the four tab structure, and adapted the code required to generate plots in a dynamic manner. Finally, I worked on the aesthetics and interface of the app, to ensure that it was both professional and understandable. I then tested the app with number of colleagues, to ensure it was accessible and understandable by a range of users both familiar and unfamiliar with the project. I implemented their suggested changes and I have published the app on the university server, and I now continue to maintain this. 

(ref:shinyApp3) Image of one of the shiny tabs, in which the user can explore log-transformed viability values for different drug and stimulus treatments, stratified by genetic features. 

```{r shinyApp3, out.width = "100%", fig.cap='(ref:shinyApp3)', echo = FALSE}

knitr::include_graphics("figures/shiny3.eps")

```

### Online code repository 
The online repository can be found [here](https://github.com/Huber-group-EMBL/CLLCytokineScreen2021).The repository consists of the data and executable transcripts to completely reproduce the analysis described in Bruch and Giles et al. 2021 (Figure \@ref(fig:package1)). 

To generate the repository, I first curated each of the individual datasets, including various objects containing the screening data, patient genetic meta data, ATACseq processed data,  RNAseq counts, clinical data include LDT, TTT and OS data, plus the follow up data including lymph node IHV experiments, shRNA knockdown experiments, and additional stimulation and inhibition assays.  As many of these data contain sensitive information on patients,  I anonymised each object by updating the patient IDs, and generating a key to convert between public and private IDs, whilst also removing potential identifying features such as age. 

I next arranged the analysis into seven separate scripts, one for each figure, such that all individual sections can be rendered into a single html vignette outlining the entire analysis. I ensured that the code in each script was well-annotated and relatively simple to understand and to follow. I shared the code with several colleagues to receive feedback on coding style, and ensured that the analysis could be reproduced by others, and on different operating systems. I published the repository with the along with the preprint (Bruch and Giles et al. 2021), and I continue to maintain the code.

(ref:package1)  Interface of the online code repository 

```{r package1, out.width = "70%", fig.cap='(ref:package1)', echo = FALSE}

knitr::include_graphics("figures/package.eps")

```

These online resources are already beginning to serve as a community resource (e.g.@Lu2021, Nature Cancer), as querying them enables researchers to test new hypotheses within minutes and may obviate the need for certain small-scale experiments.


## Questions
can i just put in methods that we removed
is the section on drug QC sufficeint ? I dnt want to put too much empashsis on this, as it was mostly done bfoe i arrived 
