# Genetic modulators of responses to microenvironmental stimulation

```{r, echo = FALSE}

knitr::opts_chunk$set(
   echo = FALSE, 
   message = FALSE, 
   warning = FALSE,
   fig.align="center"
)

```

```{r setup05, echo = FALSE}
#libraries

library(clusterProfiler)
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
library(ChIPseeker)
library(genomation)
library(DESeq2)
library(ggbeeswarm)
library(ggpubr)
library(ggplot2)
library(magrittr)
library(gtable)
library(glmnet)
library(patchwork)
library(scales)
library(gridExtra)
library(ggrepel)
library(broom)
library(plyr)
library(dplyr)
library(tidyverse)
library(msigdbr)
library(png)
library(cowplot)
library(Hmisc)

```

```{r loadData05, eval = TRUE}

#data("df", "patMeta", "LDT", "survT", "dds_smp")
#df_complete <- df

#Data
#df: tibble containing all screening viability data
load("data/df.RData")
df_complete <- df

#patMeta: tibble containing all patient genetic data
load("data/patMeta.RData")


#diffTF_large: tibble containing diffTF weighted mean difference values for 636 TFs, for trisomy 12 versus non-trisomy 12 CLL ATACseq data
load( "data/diffTF_large.RData")


#diffTF_small: tibble containing diffTF weighted mean difference values for 636 TFs, for trisomy 12 versus non-trisomy 12 CLL ATACseq data
load( "data/diffTF_small.RData")

#ChIPseq data for SPIB and PU1 binding in OCILY3 cell line (downloaded from GEO)
#get info on dataset of interest
chipSeq <- getGEOInfo(genome="hg19", simplify=FALSE) %>% dplyr::filter(series_id=="GSE56857")
#Define gsm id for SPIB and PU1 ChIPseq dataset in OCILY3 cell line (DLBCL)
gsm_spib = "GSM1370276"
gsm_pu1 = "GSM1370275"                                                         
#download bed file if not already there                                          
downloadGSMbedFiles(gsm_spib, destDir="data/")
downloadGSMbedFiles(gsm_pu1, destDir="data/")

#read the data
#ChIPseq data downloaded from GEO Accession Number: GSE56857
peak.spib <-  genomation ::readBed("data/GSM1370276_OCILY3_SPIB_GEM_events.bed.gz", track.line = "auto")
peak.pu1 <-  genomation ::readBed( "data/GSM1370275_OCILY3_PU1_GEM_events.bed.gz", track.line = "auto")


#SPIB_KD_Cellcounts: a tibble containing Normalized Cell Counts for Cell Lines infected with shRNA against SPIB and PU1
load( "data/SPIB_KD_Cellcounts.RData")
```

```{r themes_functions05, echo = FALSE, eval = TRUE}

### ggplot themes
fontsize = 11

## theme for ggplots
t1 <- 
  theme(                              
  plot.background = element_blank(), 
  panel.grid.major = element_line(),
  panel.grid.major.x = element_line(linetype = "dotted", colour = "grey"),
  panel.grid.minor = element_blank(), 
  panel.border = element_blank(), 
  panel.background = element_blank(),
  axis.line = element_line(size=.4),
  axis.line.x = element_line(),
  axis.line.y = element_line(),
  axis.text.x  = element_text(angle=90, size=11, hjust = 1, vjust = 0.4),
  axis.text.y = element_text(size = 11),
  axis.ticks.length = unit(0.3,"cm"),
  axis.title.x = element_text(face="bold", size=13), 
  axis.title.y = element_text(face="bold", size=13),
  plot.title = element_text(face="bold", size=14, hjust = 0.5),
  strip.text = element_text(size = fontsize)
)

t2 <- t1+
  theme( axis.text.x  = element_text(angle=0, size=11, hjust = 0.5, vjust = 1))

## theme for legends
t.leg <-  theme(legend.title = element_text(face='bold', 
                                            hjust = 1, size=11),
                legend.key = element_blank(),
                legend.text = element_text(size=11),
                legend.background = element_rect(color = "black"))



### Set colour palettes

#For Categorical: 
colors <- c("#A1BE1F", #green
            "#F4C61F", #yellow
            "#734595", #purple
            "#D41645", #red
            "#3B6FB6", #blue
            "#B65417", #orange
            "#E2E868", #light green
            "#CBA3D8", #light purple
            "#E58F9E", #light purple
            "#8BB8E8", #light blue
            "#F49E17", #light orange
            "#303030", #black
            "#A8A99E", #grey
            "#007B53") #dark green



#For Divergent: 
Divergent <- c("#003DA5", "#2055B0", "#406EBC", "#6086C7", "#809ED2", "#9FB6DD", "#BFCFE9", "#DFE7F4", "white", "white", "white","#F4E0E7", "#E9C2CF", "#DEA3B6", "#D3849E", "#C76586", "#BC476E", "#B12855", "#A6093D")

#for negatives only: 
palblues <- c("#003DA5", "#2055B0", "#406EBC", "#6086C7", "#809ED2", "#9FB6DD", "#BFCFE9", "#DFE7F4")

#for positives only:
palreds <- c("#F4E0E7", "#E9C2CF", "#DEA3B6", "#D3849E", "#C76586", "#BC476E", "#B12855", "#A6093D")

#For mutations: 
Mutant <- c("#b5b5b5","#373A36")
Sex <- c("#707372","#D0D0CE")
IGHV <- c("#373A36","#D0D0CE")
Methylation_cluster <- c("#373A36","#A8A99E","#D0D0CE")

#For drugs: 
drugpal <- c("#734595", "#CBA3D8") #purples

#For cytokines: 
cytpal <- c("#F49E17", "#EFC06E") #yellows


#neutral
offwhite <- "#f8f8ff"
lightergrey <- "#D0D0CE"
darkergrey <- "#707372"


na_color="#f0f0f0"


###FUNCTIONS
#select lasso or ridge 
runGlm05 <- function(X, y, method = "lasso", repeats=30, folds = 3) {
  
  #set up objects
  modelList <- list()
  lambdaList <- c()
  varExplain <- c()
  
  #set up a matrix for values of coefficients, with a row for each feature, and a column for each repeat
  coefMat <- matrix(NA, ncol(X), repeats)
  
  #make row names = genetic features
  rownames(coefMat) <- colnames(X)
  #set alpha according to selected method
  if (method == "lasso"){
    alpha = 1
  } else if (method == "ridge") {
    alpha = 0
  }
  
  #Run cv.glmnet for chosen number of repeats 
  for (i in seq(repeats)) {
    
    #if there are more than two features, fit a glm
    if (ncol(X) > 2) {
      
      res <- cv.glmnet(X,y, type.measure = "mse", family="gaussian", 
                       nfolds = folds, alpha = alpha, standardize = FALSE)
      
      #add lambda min from this repeat to the list of lambdas
      lambdaList <- c(lambdaList, res$lambda.min)
      
      #put the res object (with lambdas) into the list of models
      modelList[[i]] <- res
      
      #extract the coefficients for each feature, for lambda.min
      coefModel <- coef(res, s = "lambda.min")[-1] #remove intercept row
      
      #put these coefficients into  column of coefMatrix corresponding to the repeat
      coefMat[,i] <- coefModel
      
      #calculate variance explained
      y.pred <- predict(res, s = "lambda.min", newx = X)
      varExp <- cor(as.vector(y),as.vector(y.pred))^2
      varExplain[i] <- ifelse(is.na(varExp), 0, varExp) 
      
     
      
    } else {
      #if there are only two features, fit a linear model
      fitlm<-lm(y~., data.frame(X))
      varExp <- summary(fitlm)$r.squared
      varExplain <- c(varExplain, varExp)
      
    }
  }
  #gather all lists
  list(modelList = modelList, lambdaList = lambdaList, varExplain = varExplain, coefMat = coefMat)
}


makelegends <- function (legendFor, colors) {
    x = NULL
    y = NULL
    colors = colors[names(colors) %in% legendFor]
    nleg = length(colors)
    
    #set up gtable
    wdths = c(0.4,2,2,2,1.5)
    hghts = c(2)
    
    gtl = gtable(widths=unit(wdths, "in"), heights=unit(hghts, "in"))
    n = 2
    
    #legend for Methylation Cluster
    if ("M" %in% names(colors)) {
        Mgg = ggplot(data = data.frame(x = 1, 
                                       y = factor(c("LP", "IP", "HP"), 
                                                  levels = c("LP", "IP", "HP"))), 
                     aes(x = x, y = y, fill = y)) + 
              geom_tile() + 
              scale_fill_manual(name = "Methylation cluster", 
                                values = setNames(colors[["M"]], 
                                                  nm = c("LP", "IP","HP"))) + 
              theme(legend.title = element_text(size = 12), 
                    legend.text = element_text(size = 12))
        
        gtl = gtable_add_grob(gtl, gtable_filter(ggplotGrob(Mgg), "guide-box"), 1, n)
        n = n + 1
    }
    
    #Legend for IGHV status
    if ("I" %in% names(colors)) {
        Igg = ggplot(data = data.frame(x = 1, y = factor(c("Unmutated", 
            "Mutated"), levels = c("Unmutated", "Mutated"))), 
            aes(x = x, y = y, fill = y)) + geom_tile() + scale_fill_manual(name = "IGHV", 
            values = setNames(colors[["I"]], nm = c("Unmutated", "Mutated"))) + 
            theme(legend.title = element_text(size = 12), 
                  legend.text = element_text(size = 12))
        gtl = gtable_add_grob(gtl, gtable_filter(ggplotGrob(Igg), 
            "guide-box"), 1, n)
        n = n + 1
    }
    
    #legend for gene mutations
    if ("G" %in% names(colors)) {
        Ggg = ggplot(data = data.frame(x = 1, y = factor(c("Wild Type", 
            "Mutated"), levels = c("Wild Type", "Mutated"))), 
            aes(x = x, y = y, fill = y)) + geom_tile() + scale_fill_manual(name = "Gene", 
            values = setNames(colors[["G"]], nm = c("Wild Type", "Mutated"))) + 
            theme(legend.title = element_text(size = 12), 
                  legend.text = element_text(size = 12))
        gtl = gtable_add_grob(gtl, gtable_filter(ggplotGrob(Ggg), 
            "guide-box"), 1, n)
        n = n + 1
    }
    
    return(list(plot = gtl, width = sum(wdths), height = sum(hghts)))
}

lassoPlot05 <- function(lassoOut, geneMatrix, viabMatrix, freqCut = 1, coefCut = 0.01) {
  
  plotList <- list()
  
  for (seaName in names(lassoOut)) { 
    ###FOR THE BAR PLOT
    #extract coefMat for each stimulus
    barValue <- rowMeans(lassoOut[[seaName]]$coefMat)
    #extract proportion of repeats for which each coefficient is significant
    freqValue <- rowMeans(abs(sign(lassoOut[[seaName]]$coefMat)))
    #filter out coefficients that don't meet freqCut and coefCut thresholds
    barValue <- barValue[abs(barValue) >= coefCut & freqValue >= freqCut] 
    #arrange the bar values in numerical order
    barValue <- barValue[order(barValue)]
    #if there are no sig coefficients, don't plot
    if(length(barValue) == 0) {
      plotList[[seaName]] <- NA
      next
    }
    ###FOR THE HEATMAP AND SCATTER PLOT
    #get feature matrix and response matrix to plot
    allData <- geneMatrix
    viabValue <- unlist(viabMatrix[seaName,])
    
    #get feature matrix for sig coefficients only
    tabValue <- allData[, names(barValue),drop=FALSE]
    ord <- order(viabValue)
    viabValue <- viabValue[ord]
    tabValue <- tabValue[ord, ,drop=FALSE]
    sampleIDs <- rownames(tabValue)
    tabValue <- as_tibble(tabValue)
    tabValue$Sample <- sampleIDs
    
    
    #annotate mutations by mutation, methylation or IGHV
    matValue <- gather(tabValue, key = "Var",value = "Value", -Sample)
    matValue$Type <- "mut"
    
    #for Methylation Cluster
    matValue$Type[grep("Methylation",matValue$Var)] <- "meth"
    
    #for IGHV status
    matValue$Type[grep("IGHV",matValue$Var)] <- "ighv"
    
    #change the scale of the value so that IGHV, Methylation and Mutation do not overlap
    matValue[matValue$Type == "mut",]$Value = matValue[matValue$Type == "mut",]$Value + 10
    matValue[matValue$Type == "meth",]$Value = matValue[matValue$Type == "meth",]$Value + 20
    matValue[matValue$Type == "ighv",]$Value = matValue[matValue$Type == "ighv",]$Value + 30
    
    #change continuous to categorical
    matValue$Value <- factor(matValue$Value,levels = sort(unique(matValue$Value)))
    
    #arrange order of heatmap
    matValue$Var <- factor(matValue$Var, levels = names(barValue))
    matValue$Sample <- factor(matValue$Sample, levels = names(viabValue))
    
    #change labels if mutation is a Doehner mutation 
    matValue$Var <- revalue(matValue$Var, c("del11q" = "del(11q)", "del13q" = "del(13q)", "del17p" = "del(17p)", "trisomy12" = "trisomy 12")) 
    
   
     #plot the heatmap 
      #title
    if(seaName=="TGF-b1"){
      thetitle <- "TGF-\u03B21"     
      } else {
        if(seaName=='IL-4'){ 
          thetitle <- "IL4"  
          } else {
            thetitle <- seaName
          }
        }
    
      p1 <- ggplot(matValue, aes(x=Sample, y=Var)) + 
            geom_tile(aes(fill=Value), color = "white") + #ghost white
            theme_bw()+
            scale_y_discrete(expand=c(0,0)) + 
            theme(axis.title.y = element_text( size=14),
                  axis.title.x = element_text( size=14),
                  axis.text.x=element_text(hjust=0, size=11),
                  axis.text.y=element_text(hjust=0.1, size=11),
                  axis.ticks=element_blank(),
                  panel.border=element_rect(colour="gainsboro"),  
                  plot.title=element_text(face="bold", size = 18, margin = margin(t = -5, b = 1)), 
                  panel.background=element_blank(),
                  panel.grid.major=element_blank(), 
                  panel.grid.minor=element_blank()) + 
            xlab("Mutation status for each patient") + 
            ylab("") + 
            scale_fill_manual(name="Mutated", 
                              values=c(`10`= offwhite,  #WT
                                       `11`="#373A36", #Mutant
                                       `20`= offwhite, #LP
                                       `20.5`= "#707372", #IP
                                       `21` = "#A8A99E", #HP
                                       `30` = offwhite, #IGHV-U
                                       `31` = "#707372"), #IGHV-M
                                       guide="none") + 
            ggtitle(thetitle)
        
         
    #Plot the bar plot on the left of the heatmap 
    barDF = data.frame(barValue, nm=factor(names(barValue),levels=names(barValue)))
    
    p2 <- ggplot(data=barDF, aes(x=nm, y=barValue)) + 
      geom_bar(stat="identity", 
               fill=ifelse(barValue<0,
                           palblues[6],palreds[8]), 
               colour="black", 
               size=0.3) + 
      scale_x_discrete(expand=c(0,0.5)) + 
      scale_y_continuous(expand=c(0,0)) + 
      coord_flip(ylim=c(-0.3,0.35)) + #changed from min(barValue) and max(barValue)
      theme(panel.grid.major=element_blank(), 
            panel.background=element_blank(), 
            axis.ticks.y = element_blank(),
            panel.grid.minor = element_blank(), 
            axis.text=element_text(size=11, angle = 45, hjust = 1, vjust = 1),
                axis.title = element_text(size=14), 
            panel.border=element_blank()) +
      ylab("Size of predictor") + 
      geom_vline(xintercept=c(0.5), 
                 color="black", 
                 size=0.6)
    
    #Plot the scatter plot under the heatmap
    scatterDF = data.frame(X=factor(names(viabValue), 
                                    levels=names(viabValue)), 
                           Y=unlist(viabValue))
    
    p3 <- 
      ggplot(scatterDF, aes(x=X, y=Y)) + 
      geom_point(shape=21, 
                     fill="dimgrey", 
                     colour="#707372", #dark grey
                     size=1.2) + 
      theme_bw() +
      theme(panel.grid.minor=element_blank(), 
                panel.grid.major.x=element_blank(), 
                #axis.title.x=element_blank(), 
                axis.ticks.x=element_blank(), 
                axis.text.y=element_text(size=11), 
                axis.title.x=element_text(size=14), 
                panel.border=element_rect(colour="dimgrey", size=0.1),
                panel.background=element_rect(fill="white")) +
      xlab("Log(Viability) for each sample")
    
    
    #Assemble all the plots togehter
    # construct the gtable
    wdths = c(0.2, 1.5, 0.2, 1.3*ncol(matValue), 1.5, 0.2)
    hghts = c(0.2, 0.2, 0.0020*nrow(matValue), 0.3, 1, 0.2)
    gt = gtable(widths=unit(wdths, "in"), heights=unit(hghts, "in"))
    
    ## make grobs
    gg1 = ggplotGrob(p1)
    gg2 = ggplotGrob(p2)
    gg3 = ggplotGrob(p3)
    ## fill in the gtable
   
    #HEATMAP
    #5:1 = "PREDICTORS"
    gt = gtable_add_grob(gt, gtable_filter(gg1, "panel"), 3, 4) # add heatmap
    gt = gtable_add_grob(gt, gtable_filter(gg1, "panel"), 3, 4) #add legend
    gt = gtable_add_grob(gt, gtable_filter(gg1, "title"), 1, 4) #add title to plot
    gt = gtable_add_grob(gt, gtable_filter(gg1, "axis-l"), 3, 5) # variable names
    gt = gtable_add_grob(gt, gtable_filter(gg1, "xlab-b"), 2, 4) # axis title
    
    #BARPLOT
    gt = gtable_add_grob(gt, gtable_filter(gg2, "panel"), 3, 2) # add barplot
    gt = gtable_add_grob(gt, gtable_filter(gg2, "axis-b"), 4, 2) # y axis for barplot
    gt = gtable_add_grob(gt, gtable_filter(gg2, "xlab-b"), 2, 2) # y lab for barplot
    
    #SCATTER PLOT
    gt = gtable_add_grob(gt, gtable_filter(gg3, "panel"), 5, 4) # add scatterplot
    gt = gtable_add_grob(gt, gtable_filter(gg3, "xlab-b"), 6, 4) # x label for scatter plot
    gt = gtable_add_grob(gt, gtable_filter(gg3, "axis-l"), 5, 3) #  axis for scatter plot
    
   
    
    #plot
    #grid.draw(gt)
    plotList[[seaName]] <- gt
  }
  return(plotList)
}
```


```{r processData05, echo = FALSE}
#REMOVE
#Subset data to Cytokine only treatments, no drugs 
#Remake dataframe, then can load with this premade in future

df <- dplyr::filter(df, Drug == "DMSO", Cytokine != "No Cytokine") %>% 
  dplyr::mutate(Cytokine = as.character(Cytokine)) %>%
  dplyr::mutate(Cytlabel = ifelse(Cytokine == "IL-2", "IL2",
                            ifelse(Cytokine == "IL-4", "IL4",
                                   ifelse(Cytokine == "IL-6", "IL6", 
                                          ifelse(Cytokine == "IL-10", "IL10",
                                                 ifelse(Cytokine == "IL-10", "IL10",
                                                        ifelse(Cytokine == "IL-21", "IL21",
                                                               ifelse(Cytokine == "sCD40L+IL-4","sCD40L + IL4",
                                                                      ifelse(Cytokine == "IL-15", "IL15",
                                                                             ifelse(Cytokine == "Interferon gamma","Interferon \u03B3",
                                                                                    ifelse(Cytokine == "SDF-1a","SDF-1\u03B1",
                                                                                           ifelse(Cytokine == "IL-1b","IL-1\u03B2",
                                                                                                  ifelse(Cytokine == "TGF-b1","TGF\u03B2", Cytokine))))))))))))) %>%
  
  dplyr::mutate(Cytokine = factor(Cytokine))


df_complete <- df_complete %>%
  dplyr::mutate(Cytokine = as.character(Cytokine)) %>%
  dplyr::mutate(Cytlabel = ifelse(Cytokine == "IL-2", "IL2",
                            ifelse(Cytokine == "IL-4", "IL4",
                                   ifelse(Cytokine == "IL-6", "IL6", 
                                          ifelse(Cytokine == "IL-10", "IL10",
                                                 ifelse(Cytokine == "IL-10", "IL10",
                                                        ifelse(Cytokine == "IL-21", "IL21",
                                                               ifelse(Cytokine == "sCD40L+IL-4","sCD40L + IL4",
                                                                      ifelse(Cytokine == "IL-15", "IL15",
                                                                             ifelse(Cytokine == "Interferon gamma","Interferon \u03B3",
                                                                                    ifelse(Cytokine == "SDF-1a","SDF-1\u03B1",
                                                                                           ifelse(Cytokine == "IL-1b","IL-1\u03B2",
                                                                                                  ifelse(Cytokine == "TGF-b1","TGF\u03B2", Cytokine))))))))))))) %>%
  
  dplyr::mutate(Cytokine = factor(Cytokine))


patMeta$treatment <- as.factor(patMeta$treatment)


```

## Systematic analysis of the effect of genetic features on responses to stimuli:
### Univariate analysis identifies IGHV status and tirsomy 12 as key modulators 
(Figure \@ref(fig:stimuliGeneAssosciations)).

(ref:stimuliGeneAssosciations) 

```{r stimuliGeneAssosciations, fig.cap='(ref:stimuliGeneAssosciations)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '100%', dev = 'cairo_pdf'}

##################### List of mutations with >2 postive cases #############################
selected_mutations <- 
  patMeta %>% 
  mutate(Ras_Raf = as.factor(ifelse(BRAF == 1 | KRAS == 1| NRAS == 1, 1, 0))) %>% 
  dplyr::select(-BRAF, -KRAS, -NRAS) %>% 
  dplyr::select( -gender, 
                 -diagnosis, 
                 ) %>%
  pivot_longer(-PatientID, 
               names_to = "Genetic_Alteration", 
               values_to = "alteration_value") %>% 
  dplyr::filter(alteration_value %in% c(1, "U")) %>% 
  dplyr::group_by(Genetic_Alteration) %>% 
  dplyr::count() %>% 
  dplyr::filter(n>2) %>% 
  dplyr::select(Genetic_Alteration) %>% 
  unlist()

##################### t tests and p-value adjustment ###############################  
p_values <-
  ## Select columns from screening data 
  dplyr::select(df, PatientID, Log, Cytokine) %>% 
  
  ## Join Screening data with metadata
  left_join(patMeta, by = "PatientID") %>%
  
  ## Add Ras/Raf column, remove single columns
  mutate(Ras_Raf=as.factor(ifelse(BRAF == 1 | KRAS == 1 | NRAS == 1, 1, 0))) %>% 
  dplyr::select(-BRAF, -KRAS, -NRAS) %>% 
## remove unused columns from metadata
  dplyr::select( -gender, 
                 -diagnosis, 
                 -treatment) %>% 
  
  ## transform data to long format
  pivot_longer(cols=c(-PatientID, -Log, -Cytokine), 
               names_to = "Genetic_alt", 
               values_to = "alt_value") %>% 
  
  ## filter to selected mutations (see above)
  dplyr::filter(Genetic_alt %in% selected_mutations) %>% 
  
  
  ## group by Cytokine and Genetic alteration  
  dplyr::group_by(Cytokine, Genetic_alt) %>% 
  
  ## Perform t.test on every combination of Cytokine and genetic alteration
  do(tidy(t.test(Log ~ alt_value, data = ., var.equal = T))) %>% 
  
  ## ungroup before adjusting p-value
  ungroup() %>% 
  
  ## adjust p-values to multiple testing using BH method 
  mutate(adj.p.value=p.adjust(p.value, method = "BH"))
  
################################## Order of cytokines by descending significance ############################
Cytokine_order <-
  p_values %>% 
  dplyr::group_by(Cytokine) %>% 
  dplyr::arrange(adj.p.value) %>% 
  dplyr::filter(row_number() == 1) %>% 
  ungroup() %>% 
  dplyr::arrange(adj.p.value) %>% 
  dplyr::select(Cytokine) %>% 
  unlist()

Genetic_alt_order <-
  p_values %>% 
  dplyr::group_by(Genetic_alt) %>% 
  dplyr::arrange(adj.p.value) %>% 
  dplyr::filter(row_number() == 1) %>% 
  ungroup() %>% 
  dplyr::arrange(adj.p.value) %>% 
  dplyr::select(Genetic_alt) %>% 
  unlist()

############################# Define FDR cutoff #################################
fdr = 0.1
  
############################################ Plot ########################################
p_values %<>% 
  mutate(Cytokine=factor(Cytokine, levels = Cytokine_order)) %>% 
  mutate(Genetic_alt=factor(Genetic_alt, levels = Genetic_alt_order))
  
  ggplot(dplyr::filter(p_values, adj.p.value>fdr), 
         aes(x = Cytokine, y = -log10(adj.p.value))) +
  geom_point(color = "lightgrey", size = 3) +
  geom_beeswarm(data = dplyr::filter(p_values, adj.p.value <= fdr), 
                aes( color=Genetic_alt), size = 5, cex = 1.7) +
  ##FDR line  
  geom_hline(yintercept = -log10(fdr),linetype = "dashed", size=0.3)+
  
  ##Main Theme
  t1 +
  
  theme(axis.text.x = element_text(angle = 45, 
                                   size = 18, 
                                   face = "bold",
                                   hjust = 1, 
                                   vjust = 1),
        axis.title.x=element_blank()) +
  ##Legend Theme
  theme(legend.position="bottom", 
        legend.title = element_text(face='bold', hjust = 0, size=18), 
        legend.key = element_blank(),  
        legend.text = element_text(size=18)) +
  guides(colour = guide_legend(nrow=3, title = "Mutations"), 
         shape = guide_legend(ncol = 1)) +
  scale_color_manual(name = "Mutations", values = colors, labels=c("IGHV.status"="IGHV status", 
                                                                   "del9p"="del(9p)", 
                                                                   "trisomy12"="trisomy 12", 
                                                                   "gain17q"="gain(17q)", 
                                                                   "gain2p"="gain(2p)",
                                                                   "gain19p"="gain(19p)",
                                                                   "gain19q"="gain(19q)",
                                                                   "del7q"="del(7q)",
                                                                   "del9q"="del(9q)", 
                                                                   "del4p"= "del(4p)", 
                                                                   "SPEN"="SPEN", 
                                                                   "del11q"="del(11q)", 
                                                                   "del1q" = "del(1q)")) +
  scale_y_continuous(expression("BH-adjusted  "* italic(p)*"-value"), 
                     breaks=seq(0,10,5),
                     labels=math_format(expr=10^.x)(-seq(0,10,5))) +
  
  scale_x_discrete(labels = c("TGF-b1"="TGF-\u03B21", "sCD40L+IL-4"="sCD40L + IL4", "IL-1b"="IL1\u03B2", "IL-4"="IL4", "IL-6"="IL6","IL-15"="IL15","IL-10"="IL10", "IL-21"="IL21","IL-2"="IL2", "Interferon gamma"= "Interferon \u03B3", "SDF-1a"="SDF-1\u03B1"))



```


### Multivariate analysis demosntrates multiple layers of biology involved
(Figure \@ref(fig:stimuliGeneAssosciationsMulti)).

(ref:stimuliGeneAssosciationsMulti) 

```{r stimuliGeneAssosciationsMulti, fig.cap='(ref:stimuliGeneAssosciationsMulti)', message = FALSE,  echo = FALSE, fig.height=3, fig.width=8.5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

#Generate feature Matrix
#select features from patient meta file
geneMatrix <- 
  dplyr::select(patMeta,
                -c(gender:treatment)) %>%
  
  #adjust IGHV.status levels  U and M to numeric 1 and 0 
  mutate(IGHV = ifelse(is.na(IGHV.status), NA,
                       ifelse(IGHV.status == "M", 1, 0)), 
         #adjust Methylation_Cluster levels  LP, IP, HP to 0, 0.5, 1
         Methylation = ifelse(is.na(Methylation_Cluster), NA,
                              ifelse(Methylation_Cluster == "LP", 0,
                                     ifelse(Methylation_Cluster == "IP", 0.5, 1))),
         #remove old columns
         IGHV.status=NULL, Methylation_Cluster=NULL ) %>%

  
  #convert factors to numeric
  mutate_if(is.factor, as.character) %>%
  mutate_at(vars(-PatientID), as.numeric) %>%
  
  #convert to matrix format, with patient IDs as rownames
  data.frame() %>% 
  column_to_rownames("PatientID") %>% 
  as.matrix()
#Tidy matrix for use in glmnet function
#Remove genes with higher than 20% missing values
geneMatrix <- geneMatrix[,colSums(is.na(geneMatrix))/nrow(geneMatrix) <= 0.2]
#Filter for patients with complete data
geneMatrix.complete <- geneMatrix[complete.cases(geneMatrix),]
#Combine KRAS, NRAS and BRAF mutations into a single column
#set up empty matrix
Ras_Raf <- matrix(NA, 
                  nrow = nrow(geneMatrix.complete), 
                  ncol = 1)
colnames(Ras_Raf) <- "RAS/RAF"
#add RAS/RAF column to matrix
geneMatrix.complete <- cbind(geneMatrix.complete, Ras_Raf)
#Annotate RAS_RAF where where any of KRAS, NRAS or BRAF are mutated
geneMatrix.complete[,"RAS/RAF"] <- ifelse(geneMatrix.complete[,"KRAS"]==1,1,
		                                        ifelse(geneMatrix.complete[,"BRAF"]==1,1,
	                	                          ifelse(geneMatrix.complete[,"NRAS"]==1, 1, 0)))
#remove KRAS, NRAS and BRAF columns
geneMatrix.complete <- 
  geneMatrix.complete[, colnames(geneMatrix.complete) != "KRAS"]
geneMatrix.complete <- 
  geneMatrix.complete[, colnames(geneMatrix.complete) != "BRAF"]
geneMatrix.complete <- 
  geneMatrix.complete[, colnames(geneMatrix.complete) != "NRAS"]


# Set up viability response matrix
viabMatrix <- 
  #select patients with complete meta data 
  dplyr::filter(df,
                PatientID %in% row.names(geneMatrix.complete)) %>%
  #select cytokine, patient and log(viability values only)
  dplyr::select(Cytokine, 
                Log, 
                PatientID) %>% 
  #reshape data
  spread(key = PatientID, value = Log) %>% 
  data.frame() %>% 
  #make Cytokine the row names
  remove_rownames() %>%
  column_to_rownames("Cytokine")
#make sample order same as in geneMatrix
viabMatrix <- viabMatrix[,rownames(geneMatrix.complete)]


#Run lasso regression
#set object to hold model outputs
dataResult <- list()
#fit model for each stimulus
for (i in rownames(viabMatrix)){  
  
    #prepare input and response matrices
    y <- unlist(viabMatrix[i,]) # viability for each patient with given condition
    X <- geneMatrix.complete #genetic features for each patient
    #fit the model
    cvglmfit <- runGlm05(X, y, method="lasso", repeats=30, folds=3)
    
    #collect the results for each stimulus in one object
    dataResult[[i]] <- cvglmfit
}


#Get predictor profiles for stimuli of interest

heatMaps_cyt <- lassoPlot05(dataResult[c("IL-4","CpG ODN", "TGF-b1")] , 
                          geneMatrix.complete, #use gene matrix for heatmap 
                          viabMatrix, #use viab matrix for scatter plot
                          freqCut = 0.75, #coefficients should be selected in <75% of bootstrapped model files
                          coefCut = 0.00) #no minimum value for coefficients 


#Assemble legend for heatmaps
#G# = gene mutations, I = IGHV, M = Methylation Cluster
legendFor = c("G", "I", "M")

#assign colours for I, G and M
coldef<-list()
coldef["I"] <- list(c(offwhite,"#707372")) #U-CLL and M-CLL
coldef["M"] <- list(c(offwhite, "#707372", "#A8A99E")) #IP, HP, LP
coldef["G"] <- list(c(offwhite, "#373A36")) #WT and Mutated
legends = makelegends(legendFor=c("G","I","M"),coldef)

#assemble
Fig3B_1 <- wrap_elements(grid.arrange( grobs = heatMaps_cyt[1], ncol =1))
Fig3B_2 <- wrap_elements(grid.arrange( grobs = heatMaps_cyt[2], ncol =1))
Fig3B_3 <- wrap_elements(grid.arrange( grobs = heatMaps_cyt[3], ncol =1))
Fig3B_leg <- legends[["plot"]] %>% wrap_elements()


blankplot <- 
  ggplot() +
  geom_blank() +
  theme(panel.background = element_blank())


wrap_elements(Fig3B_1 + Fig3B_2 + Fig3B_3 +
                wrap_elements(blankplot + Fig3B_leg + 
                                plot_layout(ncol=2, widths  = c(0.15,0.9))) +
                plot_layout(ncol=1, heights = c(1.7,1.7,1.7,0.5)))



```

### individual cytokine – gene interactions of note
-Role of IGHV status and BCR signalling in Cytokine response
(Figure \@ref(fig:IHGV_BCR)).

(ref:IHGV_BCR) 

```{r IHGV_BCR, fig.cap='(ref:IHGV_BCR)', message = FALSE,  echo = FALSE, fig.height=3, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}


#add genetic data to viability dataframe
df_patmeta <- left_join(df, patMeta, by = "PatientID")

gg = 
  lapply(c("sCD40L", "IL-1\u03B2","TGF\u03B2"), function(x){

df_patmeta %>% 
  dplyr::filter(Cytlabel==x,
                !is.na(IGHV.status)) %>%

         
  ggplot(aes(x=IGHV.status,y=Log,color=(IGHV.status)))+
  geom_hline(yintercept = 0)+
  geom_boxplot()+
  geom_beeswarm(cex=1.5) +
  guides(color="none", shape="none")+
  stat_compare_means(method = "t.test",
                     label.x.npc = "center", 
                     size=5)+
  xlab("IGHV status") +
  ylab("Log(Viability)") +
  ggtitle(x) +
  scale_color_manual(values=c(colors[1], colors[2])) + 

  t2

})

wrap_elements(gg[[1]]) + wrap_elements(gg[[2]]) + wrap_elements(gg[[3]]) 


```

-TLR response stratified by IGHV status (and trisomy 12) and discussion of interactions between BCR and TLR pathways
(Figure \@ref(fig:TLR_IHGV_tri12)).

(ref:TLR_IHGV_tri12) 

```{r TLR_IHGV_tri12, fig.cap='(ref:TLR_IHGV_tri12)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}


#filter for Resiquimod-only treatment, only show patients who are annoyated for IGHV and Trisomy12
df_patmeta %>% 
  dplyr::filter(Cytokine=="Resiquimod",
                !is.na(trisomy12),
                !is.na(IGHV.status)) %>%
         
  ggplot(aes(x=interaction(IGHV.status, trisomy12),
                    y=Log,
                    color=(trisomy12)))+
  geom_hline(yintercept = 0)+
  geom_boxplot()+
  geom_beeswarm(cex=1.5) +
  guides(color="none", shape="none")+
  scale_x_discrete(labels=c("M.0"="IGHV-M\n WT",
                            "U.0"="IGHV-U\n WT",
                            "M.1"="IGHV-M\n trisomy 12",
                            "U.1"="IGHV-U\n trisomy 12"))+
  stat_compare_means(method = "t.test",
                     label.x.npc = "center", 
                     comparisons = list( c(1,2), c(3,4), c(1,3)),
                     size=7)+
  xlab("") +
  ylab("Log(Viability)") +
  ggtitle("Resiquimod") +
  scale_color_manual(values=c(colors[1], colors[2])) + 
  coord_cartesian(ylim = c(-1.5, 3.1), clip="off") +
  t1+
  theme(axis.text.x = element_text(angle = 45, vjust =1))


```

-Modulators of TLR signalling incl. effect of del11q and ATM mutations on TLR response
(Figure \@ref(fig:TLR_genes)).

(ref:TLR_genes) 

Potentially show effect of IGHV, then ivnestgiation of TLR combined boave and belwo, then look intro tri 12

These plots are not working right now - can work on them if i haev time 

```{r TLR_genes, fig.cap='(ref:TLR_genes)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

hh = 
  lapply(c("ATM", "del11q", "SF3B1", "del13q", "del17p", "TP53"), function(x){

df_patmeta %>% 
  dplyr::filter(Cytokine=="Resiquimod") %>%
      filter(!is.na(x)) %>%
         
  ggplot(aes(x=x,y=Log,color=(x)))+
  geom_hline(yintercept = 0)+
  geom_boxplot()+
  geom_beeswarm(cex=1.5) +
  guides(color="none", shape="none")+
  stat_compare_means(method = "t.test",
                     label.x.npc = "center", 
                     size=5)+
  xlab(x) +
  ylab("Log(Viability)") +
  ggtitle("Resiquimod") +
  scale_color_manual(values=c(colors[1], colors[2])) + 

  t2

})

wrap_elements(hh[[1]]) + wrap_elements(hh[[2]]) + wrap_elements(hh[[3]]) 

```

-samples with KRAS mutations benefit less from IL4 stimulation

(Figure \@ref(fig:IL4_KRAS)).

(ref:IL4_KRAS) 

```{r IL4_KRAS, fig.cap='(ref:IL4_KRAS)', message = FALSE,  echo = FALSE, fig.height=3, fig.width = 3, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

df_patmeta %>% 
   dplyr::filter(Cytokine=="IL-4",
                !is.na(KRAS)) %>% 
         
  ggplot(aes(x=KRAS,y=Log, color=(KRAS)))+
  geom_hline(yintercept = 0)+
  geom_boxplot()+
  geom_beeswarm(cex=1.5) +
  guides(color="none", shape="none")+
  stat_compare_means(method = "t.test",
                     label.x.npc = "center", 
                     size=5)+
  xlab("KRAS") +
  ylab("Log(Viability)") +
  ggtitle("IL4") +
  scale_color_manual(values=c(colors[1], colors[2])) + 

  t2


```

## Profiling the effects of trisomy 12 on cytokine response, and how this may be possible
### Trisomy 12 modulates responses to IL4, TGFbeta and TLR stimuli
(Figure \@ref(fig:tri12_cytResponse)).

(ref:tri12_cytResponse) 

```{r tri12_cytResponse, fig.cap='(ref:tri12_cytResponse)', message = FALSE,  echo = FALSE, fig.height=3, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

#add genetic data to viability dataframe
df_patmeta <- left_join(df, patMeta, by = "PatientID")

ii = 
  lapply(c("IL4", "sCD40L + IL4", "TGF\u03B2"), function(x){

df_patmeta %>% 
  dplyr::filter(Cytlabel==x,
                !is.na(trisomy12)) %>%

         
  ggplot(aes(x=trisomy12,y=Log,color=(trisomy12)))+
  geom_hline(yintercept = 0)+
  geom_boxplot()+
  geom_beeswarm(cex=1.5) +
  guides(color="none", shape="none")+
  stat_compare_means(method = "t.test",
                     label.x.npc = "center", 
                     size=5)+
  xlab("trisomy 12") +
  ylab("Log(Viability)") +
  ggtitle(x) +
  scale_color_manual(values=c(colors[1], colors[2])) + 

  t2

})

wrap_elements(ii[[1]]) + wrap_elements(ii[[2]]) + wrap_elements(ii[[3]]) 

```

### Gene dosage effects
(Figure \@ref(fig:tri12_geneDosage).

(ref:tri12_geneDosage) 

```{r tri12_geneDosage, fig.cap='(ref:tri12_geneDosage)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}


```

### Investigation of non-trisomy12 samples that are phenocopies of trisomy12 CLL+ (based on their responses to stimuli) 
(Figure \@ref(fig:tri12Phenocopies).

(ref:tri12Phenocopies) 

```{r tri12Phenocopies, fig.cap='(ref:tri12Phenocopies)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

```

### Investigation of TF activity in trisomy12+ CLL and finding that SPIB and PU1 TFs are upregulated in trisomy12+ CLL
tri 12 annotation 

running diffTF, add a diagram to explain?

visualisation of results 


(Figure \@ref(fig:tri12annotation).

(ref:tri12annotation) 

```{r tri12annotation, fig.cap='(ref:tri12annotation)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}




```


(Figure \@ref(fig:tri12diffTF).

(ref:tri12diffTF) 

Also refer to : (Figure \@ref(fig:diffTFsmall_volPlot)

```{r tri12diffTF, fig.cap='(ref:tri12diffTF)', message = FALSE,  echo = FALSE,fig.height=6, fig.width=15, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}


#Get all TFs that show differential activity
plotTab.diffTF <- dplyr::filter(diffTF_large, pvalueAdj < 0.05)
#change to name of TF quoted in text
plotTab.diffTF$TF <- gsub("SPI1", "PU1", plotTab.diffTF$TF)
#order by weighted mean difference
idx <- order(plotTab.diffTF[["weighted_meanDifference"]], decreasing = TRUE)
plotTab.diffTF$TF <- factor(plotTab.diffTF$TF,levels=rev(unique(plotTab.diffTF$TF[idx])))
#plot figure

  ggplot(plotTab.diffTF,
         aes(x=weighted_meanDifference,
             y=TF,
             fill= ifelse(weighted_meanDifference>0, 
                          palreds[5], palblues[1]))) +
  geom_bar(stat = "identity", width = 0.75) +
  xlim(c(-0.2, 0.175)) +
  
  #colours and theme
  scale_fill_manual(values = c(palblues[1], palreds[8]), guide = "none") +
  t1 + 
  theme(axis.text.x = element_text(angle = 45, vjust =1))+
  
  #Labels
  ylab("") + 
  xlab("Change in TF activity") +
  ggtitle("Diffential TF activity in trisomy 12 CLL") + 
  
  #Add annotations
  #arrow 1
  annotate("segment", x = -0.15, xend = -0.15, y = 8.4, yend = 0.75, 
           colour = "#5e5e5e", size=3, alpha=1, arrow=arrow()) + 
  #arrow 2
  annotate("segment", x = -0.15, xend = -0.15, y = 8.6, yend = 17, 
           colour = "#5e5e5e", size=3, alpha=1, arrow=arrow()) + 
 
   #2 arrow labels
  annotate("text", x = c(-0.185,-0.185), y = c(4.8,12.5), 
          label = c("Lower activity in trisomy 12 CLL", 
                    "Higher activity in trisomy 12 CLL") , 
          color="black", 
          size=5 , fontface="bold") +
  #flip plot
  coord_flip() 
  
  
  

  

```

## Profiling the downstream effects of SPIB and PU1:
### Over-representation of BCR, TLR, interleukin and TGFbeta response genes amongst SPIB and PU1 targets in ChIPseq data
(Figure \@ref(fig:SpiBChipSeq).

(ref:SpiBChipSeq) 

```{r SpiBChipSeq, fig.cap='(ref:SpiBChipSeq)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

```

### Visualisation of SPIB and PU1 binding sites from ChIPseq data in BCR, TLR, interleukin and TGFbeta signalling genes
(Figure \@ref(fig:SpiBBindingSites).

(ref:SpiBBindingSites) 

```{r SpiBBindingSites, fig.cap='(ref:SpiBBindingSites)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

```

### Investigation of DE genes within IL4 and TGFbeta pathways in SPIB/PU1 DKO mice
(Figure \@ref(fig:SpiBDEgenes).

(ref:SpiBDEgenes) 

```{r SpiBDEgenes, fig.cap='(ref:SpiBDEgenes)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

```

### SPIb and PU1 linked genes 
Key result: generation of GRN to identify SPIB and PU1 regulated genes, and investigation of associated log2FCs of these genes in trisomy 12 vs non-trisomy12 samples (including CD79A and B) 

(Figure \@ref(fig:SpiBGRN).

(ref:SpiBGRN) 

```{r SpiBGRN, fig.cap='(ref:SpiBGRN)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

```


### SPIB/PU1 DKOs in trisomy 12+ cell lines 
(Figure \@ref(fig:SpiB_shRNAKD).

(ref:SpiB_shRNAKD) 

_The data here was generated by Tina Bericoviv and was included in Bruch & Giles et al. 2021. It is included in this thesis for completeness._ 

```{r SpiB_shRNAKD, fig.cap='(ref:SpiB_shRNAKD)', message = FALSE,  echo = FALSE, fig.height=3, fig.width = 5, fig.align="center", out.width = '100%', dev = 'cairo_pdf'}


  SPIB_KD_Cellcounts %>% 

  dplyr::filter(treatment%in%c("scr_shRNA", "PU.1-KD","SpiB-KD", "Double-KD")) %>% 
  mutate(treatment=factor(treatment, levels = c("scr_shRNA", "PU.1-KD", "SpiB-KD", "Double-KD"))) %>% 
  
  
  
  ggplot(aes(x=Day, y=NormalizedCellCount, color=treatment, group=interaction( treatment)))+
  stat_summary(fun = mean,geom='line',aes(color=treatment), size=1) +
  stat_summary(fun=mean,geom='point', size=2) +
  facet_wrap(~Cellline, scales="fixed")+
  scale_y_log10()+
  labs(y="Relative Cellcount")+
 t2+
  theme(legend.position = "bottom", legend.key = element_blank(), legend.title = element_text(face='bold', size=fontsize+4),  legend.text = element_text(size=fontsize+2), strip.background = element_blank(), strip.text = element_text(size=18, face="bold"))+
      scale_color_manual(name = "Condition", values = c("darkgrey", colors[c(7,8,3)]), labels=c("scr_shRNA"="Control","PU.1-KD"="PU.1-Knockdown","SpiB-KD"="Spi-B-Knockdown","Double-KD"="Spi-B + PU.1 Double Knockdown" ))


```


## Conclusions 

so everything is here 
focus on the content you have and finihsh - add additioanl if time 