# Genetic modulators of responses to microenvironmental stimulation
Profiling the effects of the panel of stimuli revealed the potency of the IL4 and TLR pathways in modulating CLL viability, and the heterogenous nature of responses to stimulation. IL4 uniformly increased viability across the patient samples, but in certain subgroups the effect was even stronger. Stimulation of TLR 7/8/9 increased viability in some samples, whilst it reduced viability in others. Each of the four clusters showed distinct response profiles, and this related to clinical disease progression. 

It was next natural to ask to what extent this heterogeneity of response relates to the molecular profiles of the tumours. We combined our screening dataset with multi-omics profiles of the patient samples. Patient molecular profiling was available in the PACE repository[REFERENCE], consisting of whole-exome sequencing, DNA-methylation, RNA-sequencing and copy number variant data. In addition, for a small subset of patients we also generated  ATAC sequencing and Mass Spectrometry data. 

Collectively, these data enable us to probe genetic and epigenetic modulators of microenvironmental signalling, in a heterogeneous cohort that encompasses the clinical and molecular diversity of CLL. I began by performing a comprehensive survey of how the genetic features realte to microenvironmental response, and next investigated individual findings of this analysis. 


Feedback from Wolfgang: 
-include gene dosage and tir 12 classfier - its ok if these aren't final in themselves
-include before the SPIb aowrk - say this is where we ended up 
-don't try and cram in more work this is fine

More generally: 
-keep talking to him about content as i go 
-focus on clsoing dowb what i haev now, add more if time 

```{r, echo = FALSE}

knitr::opts_chunk$set(
   echo = FALSE, 
   message = FALSE, 
   warning = FALSE,
   fig.align="center"
)

```

```{r setup05, echo = FALSE}
#libraries

library(clusterProfiler)
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
library(ChIPseeker)
library(genomation)
library(DESeq2)
library(ggbeeswarm)
library(ggpubr)
library(ggplot2)
library(magrittr)
library(gtable)
library(glmnet)
library(patchwork)
library(scales)
library(gridExtra)
library(ggrepel)
library(broom)
library(plyr)
library(dplyr)
library(tidyverse)
library(msigdbr)
library(png)
library(cowplot)
library(Hmisc)
library(MultiAssayExperiment)
library(biomartr)
library(biomaRt)
library(patchwork)


```

```{r loadData05, eval = TRUE}

#data("df", "patMeta", "LDT", "survT", "dds_smp")
#df_complete <- df

#Data
#df: tibble containing all screening viability data
load("data/df.RData")
df_complete <- df

#patMeta: tibble containing all patient genetic data
load("data/patMeta.RData")


#diffTF_large: tibble containing diffTF weighted mean difference values for 636 TFs, for trisomy 12 versus non-trisomy 12 CLL ATACseq data
load( "data/diffTF_large.RData")


#diffTF_small: tibble containing diffTF weighted mean difference values for 636 TFs, for trisomy 12 versus non-trisomy 12 CLL ATACseq data
load( "data/diffTF_small.RData")

#ChIPseq data for SPIB and PU1 binding in OCILY3 cell line (downloaded from GEO)
#get info on dataset of interest
chipSeq <- getGEOInfo(genome="hg19", simplify=FALSE) %>% dplyr::filter(series_id=="GSE56857")
#Define gsm id for SPIB and PU1 ChIPseq dataset in OCILY3 cell line (DLBCL)
gsm_spib = "GSM1370276"
gsm_pu1 = "GSM1370275"                                                         
#download bed file if not already there                                          
downloadGSMbedFiles(gsm_spib, destDir="data/")
downloadGSMbedFiles(gsm_pu1, destDir="data/")

#read the data
#ChIPseq data downloaded from GEO Accession Number: GSE56857
peak.spib <-  genomation ::readBed("data/GSM1370276_OCILY3_SPIB_GEM_events.bed.gz", track.line = "auto")
peak.pu1 <-  genomation ::readBed( "data/GSM1370275_OCILY3_PU1_GEM_events.bed.gz", track.line = "auto")


#SPIB_KD_Cellcounts: a tibble containing Normalized Cell Counts for Cell Lines infected with shRNA against SPIB and PU1 from Tina Becirovic
load( "data/SPIB_KD_Cellcounts.RData")


#Load proteomics data from Sophie Herbst
load("data/multiomics_MAE.RData")


#Full RNA dataset: Deseq2 object
load("data/dds_smp_full.RData")

#Matching samples only RNA dataset: Deseq2 object
load("data/dds_smp.RData")

```

```{r themes_functions05, echo = FALSE, eval = TRUE}

### ggplot themes
fontsize = 11

## theme for ggplots
t1 <- 
  theme(                              
  plot.background = element_blank(), 
  panel.grid.major = element_line(),
  panel.grid.major.x = element_line(linetype = "dotted", colour = "grey"),
  panel.grid.minor = element_blank(), 
  panel.border = element_blank(), 
  panel.background = element_blank(),
  axis.line = element_line(size=.4),
  axis.line.x = element_line(),
  axis.line.y = element_line(),
  axis.text.x  = element_text(angle=90, size=11, hjust = 1, vjust = 0.4),
  axis.text.y = element_text(size = 11),
  axis.ticks.length = unit(0.3,"cm"),
  axis.title.x = element_text(face="bold", size=13), 
  axis.title.y = element_text(face="bold", size=13),
  plot.title = element_text(face="bold", size=14, hjust = 0.5),
  strip.text = element_text(size = fontsize)
)

t2 <- t1+
  theme( axis.text.x  = element_text(angle=0, size=11, hjust = 0.5, vjust = 1))

## theme for legends
t.leg <-  theme(legend.title = element_text(face='bold', 
                                            hjust = 1, size=11),
                legend.key = element_blank(),
                legend.text = element_text(size=11),
                legend.background = element_rect(color = "black"))



### Set colour palettes

#For Categorical: 
colors <- c("#A1BE1F", #green
            "#F4C61F", #yellow
            "#734595", #purple
            "#D41645", #red
            "#3B6FB6", #blue
            "#B65417", #orange
            "#E2E868", #light green
            "#CBA3D8", #light purple
            "#E58F9E", #light purple
            "#8BB8E8", #light blue
            "#F49E17", #light orange
            "#303030", #black
            "#A8A99E", #grey
            "#007B53") #dark green



#For Divergent: 
Divergent <- c("#003DA5", "#2055B0", "#406EBC", "#6086C7", "#809ED2", "#9FB6DD", "#BFCFE9", "#DFE7F4", "white", "white", "white","#F4E0E7", "#E9C2CF", "#DEA3B6", "#D3849E", "#C76586", "#BC476E", "#B12855", "#A6093D")

#for negatives only: 
palblues <- c("#003DA5", "#2055B0", "#406EBC", "#6086C7", "#809ED2", "#9FB6DD", "#BFCFE9", "#DFE7F4")

#for positives only:
palreds <- c("#F4E0E7", "#E9C2CF", "#DEA3B6", "#D3849E", "#C76586", "#BC476E", "#B12855", "#A6093D")

#For mutations: 
Mutant <- c("#b5b5b5","#373A36")
Sex <- c("#707372","#D0D0CE")
IGHV <- c("#373A36","#D0D0CE")
Methylation_cluster <- c("#373A36","#A8A99E","#D0D0CE")

#For drugs: 
drugpal <- c("#734595", "#CBA3D8") #purples

#For cytokines: 
cytpal <- c("#F49E17", "#EFC06E") #yellows


#neutral
offwhite <- "#f8f8ff"
lightergrey <- "#D0D0CE"
darkergrey <- "#707372"


na_color="#f0f0f0"


###FUNCTIONS
#select lasso or ridge 
runGlm05 <- function(X, y, method = "lasso", repeats=30, folds = 3) {
  
  #set up objects
  modelList <- list()
  lambdaList <- c()
  varExplain <- c()
  
  #set up a matrix for values of coefficients, with a row for each feature, and a column for each repeat
  coefMat <- matrix(NA, ncol(X), repeats)
  
  #make row names = genetic features
  rownames(coefMat) <- colnames(X)
  #set alpha according to selected method
  if (method == "lasso"){
    alpha = 1
  } else if (method == "ridge") {
    alpha = 0
  }
  
  #Run cv.glmnet for chosen number of repeats 
  for (i in seq(repeats)) {
    
    #if there are more than two features, fit a glm
    if (ncol(X) > 2) {
      
      res <- cv.glmnet(X,y, type.measure = "mse", family="gaussian", 
                       nfolds = folds, alpha = alpha, standardize = FALSE)
      
      #add lambda min from this repeat to the list of lambdas
      lambdaList <- c(lambdaList, res$lambda.min)
      
      #put the res object (with lambdas) into the list of models
      modelList[[i]] <- res
      
      #extract the coefficients for each feature, for lambda.min
      coefModel <- coef(res, s = "lambda.min")[-1] #remove intercept row
      
      #put these coefficients into  column of coefMatrix corresponding to the repeat
      coefMat[,i] <- coefModel
      
      #calculate variance explained
      y.pred <- predict(res, s = "lambda.min", newx = X)
      varExp <- cor(as.vector(y),as.vector(y.pred))^2
      varExplain[i] <- ifelse(is.na(varExp), 0, varExp) 
      
     
      
    } else {
      #if there are only two features, fit a linear model
      fitlm<-lm(y~., data.frame(X))
      varExp <- summary(fitlm)$r.squared
      varExplain <- c(varExplain, varExp)
      
    }
  }
  #gather all lists
  list(modelList = modelList, lambdaList = lambdaList, varExplain = varExplain, coefMat = coefMat)
}


makelegends <- function (legendFor, colors) {
    x = NULL
    y = NULL
    colors = colors[names(colors) %in% legendFor]
    nleg = length(colors)
    
    #set up gtable
    wdths = c(0.4,2,2,2,1.5)
    hghts = c(2)
    
    gtl = gtable(widths=unit(wdths, "in"), heights=unit(hghts, "in"))
    n = 2
    
    #legend for Methylation Cluster
    if ("M" %in% names(colors)) {
        Mgg = ggplot(data = data.frame(x = 1, 
                                       y = factor(c("LP", "IP", "HP"), 
                                                  levels = c("LP", "IP", "HP"))), 
                     aes(x = x, y = y, fill = y)) + 
              geom_tile() + 
              scale_fill_manual(name = "Methylation cluster", 
                                values = setNames(colors[["M"]], 
                                                  nm = c("LP", "IP","HP"))) + 
              theme(legend.title = element_text(size = 12), 
                    legend.text = element_text(size = 12))
        
        gtl = gtable_add_grob(gtl, gtable_filter(ggplotGrob(Mgg), "guide-box"), 1, n)
        n = n + 1
    }
    
    #Legend for IGHV status
    if ("I" %in% names(colors)) {
        Igg = ggplot(data = data.frame(x = 1, y = factor(c("Unmutated", 
            "Mutated"), levels = c("Unmutated", "Mutated"))), 
            aes(x = x, y = y, fill = y)) + geom_tile() + scale_fill_manual(name = "IGHV", 
            values = setNames(colors[["I"]], nm = c("Unmutated", "Mutated"))) + 
            theme(legend.title = element_text(size = 12), 
                  legend.text = element_text(size = 12))
        gtl = gtable_add_grob(gtl, gtable_filter(ggplotGrob(Igg), 
            "guide-box"), 1, n)
        n = n + 1
    }
    
    #legend for gene mutations
    if ("G" %in% names(colors)) {
        Ggg = ggplot(data = data.frame(x = 1, y = factor(c("Wild Type", 
            "Mutated"), levels = c("Wild Type", "Mutated"))), 
            aes(x = x, y = y, fill = y)) + geom_tile() + scale_fill_manual(name = "Gene", 
            values = setNames(colors[["G"]], nm = c("Wild Type", "Mutated"))) + 
            theme(legend.title = element_text(size = 12), 
                  legend.text = element_text(size = 12))
        gtl = gtable_add_grob(gtl, gtable_filter(ggplotGrob(Ggg), 
            "guide-box"), 1, n)
        n = n + 1
    }
    
    return(list(plot = gtl, width = sum(wdths), height = sum(hghts)))
}

lassoPlot05 <- function(lassoOut, geneMatrix, viabMatrix, freqCut = 1, coefCut = 0.01) {
  
  plotList <- list()
  
  for (seaName in names(lassoOut)) { 
    ###FOR THE BAR PLOT
    #extract coefMat for each stimulus
    barValue <- rowMeans(lassoOut[[seaName]]$coefMat)
    #extract proportion of repeats for which each coefficient is significant
    freqValue <- rowMeans(abs(sign(lassoOut[[seaName]]$coefMat)))
    #filter out coefficients that don't meet freqCut and coefCut thresholds
    barValue <- barValue[abs(barValue) >= coefCut & freqValue >= freqCut] 
    #arrange the bar values in numerical order
    barValue <- barValue[order(barValue)]
    #if there are no sig coefficients, don't plot
    if(length(barValue) == 0) {
      plotList[[seaName]] <- NA
      next
    }
    ###FOR THE HEATMAP AND SCATTER PLOT
    #get feature matrix and response matrix to plot
    allData <- geneMatrix
    viabValue <- unlist(viabMatrix[seaName,])
    
    #get feature matrix for sig coefficients only
    tabValue <- allData[, names(barValue),drop=FALSE]
    ord <- order(viabValue)
    viabValue <- viabValue[ord]
    tabValue <- tabValue[ord, ,drop=FALSE]
    sampleIDs <- rownames(tabValue)
    tabValue <- as_tibble(tabValue)
    tabValue$Sample <- sampleIDs
    
    
    #annotate mutations by mutation, methylation or IGHV
    matValue <- gather(tabValue, key = "Var",value = "Value", -Sample)
    matValue$Type <- "mut"
    
    #for Methylation Cluster
    matValue$Type[grep("Methylation",matValue$Var)] <- "meth"
    
    #for IGHV status
    matValue$Type[grep("IGHV",matValue$Var)] <- "ighv"
    
    #change the scale of the value so that IGHV, Methylation and Mutation do not overlap
    matValue[matValue$Type == "mut",]$Value = matValue[matValue$Type == "mut",]$Value + 10
    matValue[matValue$Type == "meth",]$Value = matValue[matValue$Type == "meth",]$Value + 20
    matValue[matValue$Type == "ighv",]$Value = matValue[matValue$Type == "ighv",]$Value + 30
    
    #change continuous to categorical
    matValue$Value <- factor(matValue$Value,levels = sort(unique(matValue$Value)))
    
    #arrange order of heatmap
    matValue$Var <- factor(matValue$Var, levels = names(barValue))
    matValue$Sample <- factor(matValue$Sample, levels = names(viabValue))
    
    #change labels if mutation is a Doehner mutation 
    matValue$Var <- revalue(matValue$Var, c("del11q" = "del(11q)", "del13q" = "del(13q)", "del17p" = "del(17p)", "trisomy12" = "trisomy 12")) 
    
   
     #plot the heatmap 
      #title
    if(seaName=="TGF-b1"){
      thetitle <- "TGF-\u03B21"     
      } else {
        if(seaName=='IL-4'){ 
          thetitle <- "IL4"  
          } else {
            thetitle <- seaName
          }
        }
    
      p1 <- ggplot(matValue, aes(x=Sample, y=Var)) + 
            geom_tile(aes(fill=Value), color = "white") + #ghost white
            theme_bw()+
            scale_y_discrete(expand=c(0,0)) + 
            theme(axis.title.y = element_text( size=14),
                  axis.title.x = element_text( size=14),
                  axis.text.x=element_text(hjust=0, size=11),
                  axis.text.y=element_text(hjust=0.1, size=11),
                  axis.ticks=element_blank(),
                  panel.border=element_rect(colour="gainsboro"),  
                  plot.title=element_text(face="bold", size = 18, margin = margin(t = -5, b = 1)), 
                  panel.background=element_blank(),
                  panel.grid.major=element_blank(), 
                  panel.grid.minor=element_blank()) + 
            xlab("Mutation status for each patient") + 
            ylab("") + 
            scale_fill_manual(name="Mutated", 
                              values=c(`10`= offwhite,  #WT
                                       `11`="#373A36", #Mutant
                                       `20`= offwhite, #LP
                                       `20.5`= "#707372", #IP
                                       `21` = "#A8A99E", #HP
                                       `30` = offwhite, #IGHV-U
                                       `31` = "#707372"), #IGHV-M
                                       guide="none") + 
            ggtitle(thetitle)
        
         
    #Plot the bar plot on the left of the heatmap 
    barDF = data.frame(barValue, nm=factor(names(barValue),levels=names(barValue)))
    
    p2 <- ggplot(data=barDF, aes(x=nm, y=barValue)) + 
      geom_bar(stat="identity", 
               fill=ifelse(barValue<0,
                           palblues[6],palreds[8]), 
               colour="black", 
               size=0.3) + 
      scale_x_discrete(expand=c(0,0.5)) + 
      scale_y_continuous(expand=c(0,0)) + 
      coord_flip(ylim=c(-0.3,0.35)) + #changed from min(barValue) and max(barValue)
      theme(panel.grid.major=element_blank(), 
            panel.background=element_blank(), 
            axis.ticks.y = element_blank(),
            panel.grid.minor = element_blank(), 
            axis.text=element_text(size=11, angle = 45, hjust = 1, vjust = 1),
                axis.title = element_text(size=14), 
            panel.border=element_blank()) +
      ylab("Size of predictor") + 
      geom_vline(xintercept=c(0.5), 
                 color="black", 
                 size=0.6)
    
    #Plot the scatter plot under the heatmap
    scatterDF = data.frame(X=factor(names(viabValue), 
                                    levels=names(viabValue)), 
                           Y=unlist(viabValue))
    
    p3 <- 
      ggplot(scatterDF, aes(x=X, y=Y)) + 
      geom_point(shape=21, 
                     fill="dimgrey", 
                     colour="#707372", #dark grey
                     size=1.2) + 
      theme_bw() +
      theme(panel.grid.minor=element_blank(), 
                panel.grid.major.x=element_blank(), 
                #axis.title.x=element_blank(), 
                axis.ticks.x=element_blank(), 
                axis.text.y=element_text(size=11), 
                axis.title.x=element_text(size=14), 
                panel.border=element_rect(colour="dimgrey", size=0.1),
                panel.background=element_rect(fill="white")) +
      xlab("Log(Viability) for each sample")
    
    
    #Assemble all the plots togehter
    # construct the gtable
    wdths = c(0.2, 1.5, 0.2, 1.3*ncol(matValue), 1.5, 0.2)
    hghts = c(0.2, 0.2, 0.0020*nrow(matValue), 0.3, 1, 0.2)
    gt = gtable(widths=unit(wdths, "in"), heights=unit(hghts, "in"))
    
    ## make grobs
    gg1 = ggplotGrob(p1)
    gg2 = ggplotGrob(p2)
    gg3 = ggplotGrob(p3)
    ## fill in the gtable
   
    #HEATMAP
    #5:1 = "PREDICTORS"
    gt = gtable_add_grob(gt, gtable_filter(gg1, "panel"), 3, 4) # add heatmap
    gt = gtable_add_grob(gt, gtable_filter(gg1, "panel"), 3, 4) #add legend
    gt = gtable_add_grob(gt, gtable_filter(gg1, "title"), 1, 4) #add title to plot
    gt = gtable_add_grob(gt, gtable_filter(gg1, "axis-l"), 3, 5) # variable names
    gt = gtable_add_grob(gt, gtable_filter(gg1, "xlab-b"), 2, 4) # axis title
    
    #BARPLOT
    gt = gtable_add_grob(gt, gtable_filter(gg2, "panel"), 3, 2) # add barplot
    gt = gtable_add_grob(gt, gtable_filter(gg2, "axis-b"), 4, 2) # y axis for barplot
    gt = gtable_add_grob(gt, gtable_filter(gg2, "xlab-b"), 2, 2) # y lab for barplot
    
    #SCATTER PLOT
    gt = gtable_add_grob(gt, gtable_filter(gg3, "panel"), 5, 4) # add scatterplot
    gt = gtable_add_grob(gt, gtable_filter(gg3, "xlab-b"), 6, 4) # x label for scatter plot
    gt = gtable_add_grob(gt, gtable_filter(gg3, "axis-l"), 5, 3) #  axis for scatter plot
    
   
    
    #plot
    #grid.draw(gt)
    plotList[[seaName]] <- gt
  }
  return(plotList)
  
  
  
}



########### Scaling and Centering of Viability Matrix ################

# medianCenter_MadScale: 
#function to  scale with MAD, center to merdian
medianCenter_MadScale05 <- function(x) {
  s <- median(x)
  #s=0
  (x - s) / mad(x, center = s)
}

# scaleCytResp function:  to apply medianCenter_MadScale row wise to viability matrix
scaleCytResp05  <- function(x) t(apply(x, 1, medianCenter_MadScale05)) 

#ROC cruve

#function to plot ROC curves
plotROC05 <- function(glmModel, X, y, lambda = "lambda.1se") {
  lambdaChose <- glmModel[[lambda]]
  glmPred <- prediction(predict(glmModel, type = "response", newx = X, s=lambdaChose), y)
  glmPerform <- performance(glmPred,"tpr","fpr")
  aucVal <- performance(glmPred, measure = "auc")@y.values[[1]]
  xname <- glmPerform@x.name
  yname <- glmPerform@y.name
  plotTab <- tibble(x= glmPerform@x.values[[1]],
                    y = glmPerform@y.values[[1]])
  p <- ggplot(plotTab, aes(x=x, y =y )) + geom_line(color = "red") +
    xlab(xname) + ylab(yname) + theme(panel.grid = element_blank())
  
  if (!is.null(aucVal)) {
    p <- p + annotate("text", x= 0.75, y = 0.25, label = sprintf("AUC: %1.2f", aucVal))
  }
  list(plot=p, auc = aucVal)
}

#indicate how to split test set and training set
testPartition05 <- function(y, ratio) {
  #balanced sampling of test set
  ord <- seq_along(y)
  testIdx <- lapply(unique(y),function(n) {
    subY <- ord[y == n]
    sample(subY, size = as.integer(length(subY)  * ratio)) 
  }) %>% do.call(c,.) %>% sort()
  return(testIdx)
}

#Function for multi-variant binomial regression
runGlm.bin <- function(X, y, method = "lasso", repeats=20, folds = 3, testRatio = NULL, lambda = "lambda.1se") {
  modelList <- list()
  lambdaList <- c()
  aucCV <- c()
  aucTest <- c()
  rocTest <- list()
  coefMat  <- matrix(NA, ncol(X), repeats)
  rownames(coefMat) <- colnames(X)
  
  if (method == "lasso"){
    alpha = 1
  } else if (method == "ridge") {
    alpha = 0
  }
  
  for (i in seq(repeats)) {
    if (!is.null(testRatio)) {
      testIdx <- testPartition(y, testRatio)
      X.test <- X[testIdx,]
      X.train <- X[-testIdx,]
      y.test <- y[testIdx]
      y.train <- y[-testIdx]
    } else {
      X.train <- X
      y.train <- y
    }
    
    #need to add a set seed for this function, but ensure that it changes with each loop otherwise all 10 repeats are the same 
    vecFold <- mltools::folds(y.train, nfolds = folds, stratified = TRUE, seed = i)
    
    #train model
    res <- cv.glmnet(X.train,y.train, type.measure = "auc",
                      foldid = vecFold, alpha = alpha, standardize = FALSE,
                     intercept = TRUE, family = "binomial")
    lambdaList <- c(lambdaList, res[[lambda]])
    #If you arent splitting the data into a training and testing dataset. you can use the AUC for the CV
    aucCV <- c(aucCV, res$cvm[res$lambda == res[[lambda]]])
    modelList[[i]] <- res
    coefMat[,i] <- coef(res, s = lambda)[-1]
    
    #test model if testRatio is speficied- can use this AUC to identify the best model 
    if(!is.null(testRatio)) {
      rocRes <- plotROC(res, X.test, y.test, lambda)
      aucTest <- c(aucTest, rocRes$auc)
      rocTest[[i]] <- rocRes$plot
    }
  }
  list(modelList = modelList, lambdaList = lambdaList, aucCV = aucCV, coefMat = coefMat,
       aucTest = aucTest, rocTest = rocTest)
}

#function to visualise predictors of Trisomy 12 status
lassoPlot_bin <- function(lassoOut, xIn, yIn, freqCut = 1, coefCut = 0.01, symbolMap = NULL, modelIndex = NULL, textWidth =0.8, rowHeight = 0.0005) {
  
    #for the barplot on the left of the heatmap
    if (is.null(modelIndex)) {
      #average all models
      barValue <- rowMeans(lassoOut$coefMat)
      freqValue <- rowMeans(lassoOut$coefMat !=0)
      barValue <- barValue[abs(barValue) > coefCut & freqValue >= freqCut] # a certain threshold
      barValue <- barValue[order(barValue)]
      if(length(barValue) == 0) {
        return(NA)
      }
    } else {
      #using a specificed model
      barValue <- lassoOut$coefMat[,modelIndex]
      barValue <- barValue[abs(barValue) > coefCut] # a certain threshold
      barValue <- barValue[order(barValue)]
      if(length(barValue) == 0) {
        return(NA)
      }
    }
    
    #for the heatmap and scatter plot below the heatmap
    mapData <- xIn
    scatterData <- yIn
    
    ord <- order(scatterData)
    scatterData <- scatterData[ord]
    mapData <- mapData[ord, names(barValue) ,drop=FALSE]
    idOrd <- rownames(mapData) #to record the order of the rows
    mapData <- mapData %>% as_tibble() %>% mutate(row = idOrd) %>%
      gather(key = "Var", value = "Value", -row) %>%
      mutate(row = factor(row, levels = idOrd),
             Var = factor(Var, levels = names(barValue)))
    #add tri12 status
    tri12meta <- dplyr::select(patMeta, PatientID, trisomy12) 
    tri12meta$PatientID <- as.factor(tri12meta$PatientID)
    colnames(mapData) <- c("PatientID", "Var", "Value")
    mapData_tri12 <- left_join(mapData, tri12meta, by =  "PatientID")
    
    #update labeling 
    mapData_tri12$Var <- revalue(mapData_tri12$Var, c("Resiquimod" = "Resiquimod", "TGF-b1" = "TGF-\u03B21", "sCD40L+IL-4" = "sCD40L + IL4")) 
    
    #plot the heatmap
    p1 <- ggplot(mapData_tri12, aes(x=PatientID, y=Var)) + geom_tile(aes(fill=Value),color = "gray") + 
      theme_bw()  + 
      theme(axis.text.x=element_blank(), axis.ticks=element_blank(), 
            panel.border=element_rect(colour="gainsboro"),  
            plot.title=element_text(size=18), 
            panel.background=element_blank(), 
            panel.grid.major=element_blank(), 
            panel.grid.minor=element_blank()) + 
      xlab("Samples") + ylab("Log(Viability)") + 
      ggtitle("")+
      scale_fill_gradient2(low = palblues[1],high = palreds[8], mid = "white",guide="none")+ 
      facet_grid(~trisomy12, space = "free", scales = "free") + theme(strip.background =element_rect(fill="white"))
    
    #Plot the bar plot on the left of the heatmap
    if (is.null(symbolMap)) {
      barDF = data.frame(barValue=barValue, 
                         nm=factor(names(barValue),levels=names(barValue)))
    } else {
      #a id to symbol dataframe is provided
      barDF = data.frame(barValue = barValue, 
                         nm = symbolMap[names(barValue),1])
      barDF$nm <- factor(barDF$nm, levels = unique(barDF$nm))
    }
    
    p2 <- ggplot(data=barDF, aes(x=nm, y=barValue)) + 
      geom_bar(stat="identity", fill=darkergrey, colour="black", position = "identity", width=.66, size=0.2) + 
      theme_bw() + geom_hline(yintercept=0, size=0.3) + scale_x_discrete(expand=c(0,0.5)) + 
      scale_y_continuous(expand=c(0,0)) + 
      coord_flip(ylim=c(min(barValue),max(barValue))) + 
      theme(panel.grid.major=element_blank(), 
            panel.background=element_blank(), axis.ticks.y = element_blank(),
            panel.grid.minor = element_blank(), axis.text.y=element_text(size=8, hjust=0),
            axis.text.x = element_text(size=10),
            panel.border=element_blank()) +
      xlab("") + 
      ylab("Size of predictor") + 
      geom_vline(xintercept=c(0.5), color="black", size=0.6)
    

    #Scale bar for continuous variable

    Vgg = ggplot(mapData, aes(x=PatientID, y=Var, col = Value)) + 
      geom_point() + 
      scale_color_gradient2(low = palblues[1],high = palreds[8], mid = "white", name = "z-score") + 
      theme(legend.title=element_text(size=10), legend.text=element_text(size=10))
    
    #Assemble all the plots together

    # construct the gtable
    
    wdths = c(1.5, 0.2, 1.3*ncol(mapData), textWidth,0.8)
    hghts = c(0.2, rowHeight*nrow(mapData), 0.2, 0.5)
    gt = gtable(widths=unit(wdths, "in"), heights=unit(hghts, "in"))
    
    ## make grobs
    gg1 = ggplotGrob(p1)
    gg2 = ggplotGrob(p2)
    gg4 = ggplotGrob(Vgg)
    
    ## fill in the gtable
    gt = gtable_add_grob(gt, gtable_filter(gg2, "panel"), 2, 1) # add barplot
    gt = gtable_add_grob(gt, gtable_filter(gg2, "xlab-b"), 1, 1) # add  y axis label to barplot 
    gt = gtable_add_grob(gt, gtable_filter(gg1, "panel"), 2, 3) # add heatmap
    gt = gtable_add_grob(gt, gtable_filter(gg1, "strip-t"), 1, 3) # add facetstip to plot
   
    gt = gtable_add_grob(gt, gtable_filter(gg2, "axis-b"), 3, 1) # y axis for barplot
    gt = gtable_add_grob(gt, gtable_filter(gg2, "axis-l"), 2, 4) # variable names
    gt = gtable_add_grob(gt, gtable_filter(gg4, "guide-box"), 2, 5) # scale bar for continous variables

    
    #plot
    #grid.draw(gt)
    gt
}

```


```{r processData05, echo = FALSE}
#REMOVE
#Subset data to Cytokine only treatments, no drugs 
#Remake dataframe, then can load with this premade in future

df <- dplyr::filter(df, Drug == "DMSO", Cytokine != "No Cytokine") %>% 
  dplyr::mutate(Cytokine = as.character(Cytokine)) %>%
  dplyr::mutate(Cytlabel = ifelse(Cytokine == "IL-2", "IL2",
                            ifelse(Cytokine == "IL-4", "IL4",
                                   ifelse(Cytokine == "IL-6", "IL6", 
                                          ifelse(Cytokine == "IL-10", "IL10",
                                                 ifelse(Cytokine == "IL-10", "IL10",
                                                        ifelse(Cytokine == "IL-21", "IL21",
                                                               ifelse(Cytokine == "sCD40L+IL-4","sCD40L + IL4",
                                                                      ifelse(Cytokine == "IL-15", "IL15",
                                                                             ifelse(Cytokine == "Interferon gamma","Interferon \u03B3",
                                                                                    ifelse(Cytokine == "SDF-1a","SDF-1\u03B1",
                                                                                           ifelse(Cytokine == "IL-1b","IL-1\u03B2",
                                                                                                  ifelse(Cytokine == "TGF-b1","TGF\u03B2", Cytokine))))))))))))) %>%
  
  dplyr::mutate(Cytokine = factor(Cytokine))


df_complete <- df_complete %>%
  dplyr::mutate(Cytokine = as.character(Cytokine)) %>%
  dplyr::mutate(Cytlabel = ifelse(Cytokine == "IL-2", "IL2",
                            ifelse(Cytokine == "IL-4", "IL4",
                                   ifelse(Cytokine == "IL-6", "IL6", 
                                          ifelse(Cytokine == "IL-10", "IL10",
                                                 ifelse(Cytokine == "IL-10", "IL10",
                                                        ifelse(Cytokine == "IL-21", "IL21",
                                                               ifelse(Cytokine == "sCD40L+IL-4","sCD40L + IL4",
                                                                      ifelse(Cytokine == "IL-15", "IL15",
                                                                             ifelse(Cytokine == "Interferon gamma","Interferon \u03B3",
                                                                                    ifelse(Cytokine == "SDF-1a","SDF-1\u03B1",
                                                                                           ifelse(Cytokine == "IL-1b","IL-1\u03B2",
                                                                                                  ifelse(Cytokine == "TGF-b1","TGF\u03B2", Cytokine))))))))))))) %>%
  
  dplyr::mutate(Cytokine = factor(Cytokine))

#patMeta
patMeta$treatment <- as.factor(patMeta$treatment)

#Lists of Cytokines
thecytokines <- unique(df$Cytokine) %>% setdiff("No Cytokine")


```

## Systematic analysis of the effect of genetic features on responses to stimuli:
### Univariate analysis identifies IGHV status and trisomy 12 as key modulators of microenvironmental response
To begin, I ran a univariate analysis to compare viability values post-stimulation for patient samples with and without each genetic feature. In total 63 genetic features were surveyed, including IGHV status, somatic gene mutations and structural variants where there were at least three patient samples in each group (Figure \@ref(fig:stimuliGeneAssosciations)).. 

This analysis revealed the extent to which genetic features modulate microenvironmental response: for 10/17 stimuli, at least one genetic feature determined  response and for 6/17 stimuli, 2 or more genetic features significantly altered the response (Student's  t-tests,  FDR  =  10%).The most common features to modulate response were IGHV status and trisomy 12. Del(11q) also affected reponses to several stimuli. Notably, the other Doehner mutations del(13q) and del(17p) had no impact of the responses to external signals. 

(ref:stimuliGeneAssosciations) Plot showing BH-adjusted p values from Student's t-tests (two-sided, with equal variance), for all tested gene-stimulus associations. Tests performed for IGHV status and somatic mutations and copy number aberrations with ≥3 patient samples in each group (n = 63). Each circle represents a gene-stimulus association. Associations that meet 10% FDR cut off are indicated in colour, where the colour denotes the genetic feature. 

```{r stimuliGeneAssosciations, fig.cap='(ref:stimuliGeneAssosciations)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '70%', dev = 'cairo_pdf'}

##################### List of mutations with >2 postive cases #############################
selected_mutations <- 
  patMeta %>% 
  mutate(Ras_Raf = as.factor(ifelse(BRAF == 1 | KRAS == 1| NRAS == 1, 1, 0))) %>% 
  dplyr::select(-BRAF, -KRAS, -NRAS) %>% 
  dplyr::select( -gender, 
                 -diagnosis, 
                 ) %>%
  pivot_longer(-PatientID, 
               names_to = "Genetic_Alteration", 
               values_to = "alteration_value") %>% 
  dplyr::filter(alteration_value %in% c(1, "U")) %>% 
  dplyr::group_by(Genetic_Alteration) %>% 
  dplyr::count() %>% 
  dplyr::filter(n>2) %>% 
  dplyr::select(Genetic_Alteration) %>% 
  unlist()

##################### t tests and p-value adjustment ###############################  
p_values <-
  ## Select columns from screening data 
  dplyr::select(df, PatientID, Log, Cytokine) %>% 
  
  ## Join Screening data with metadata
  left_join(patMeta, by = "PatientID") %>%
  
  ## Add Ras/Raf column, remove single columns
  mutate(Ras_Raf=as.factor(ifelse(BRAF == 1 | KRAS == 1 | NRAS == 1, 1, 0))) %>% 
  dplyr::select(-BRAF, -KRAS, -NRAS) %>% 
## remove unused columns from metadata
  dplyr::select( -gender, 
                 -diagnosis, 
                 -treatment) %>% 
  
  ## transform data to long format
  pivot_longer(cols=c(-PatientID, -Log, -Cytokine), 
               names_to = "Genetic_alt", 
               values_to = "alt_value") %>% 
  
  ## filter to selected mutations (see above)
  dplyr::filter(Genetic_alt %in% selected_mutations) %>% 
  
  
  ## group by Cytokine and Genetic alteration  
  dplyr::group_by(Cytokine, Genetic_alt) %>% 
  
  ## Perform t.test on every combination of Cytokine and genetic alteration
  do(tidy(t.test(Log ~ alt_value, data = ., var.equal = T))) %>% 
  
  ## ungroup before adjusting p-value
  ungroup() %>% 
  
  ## adjust p-values to multiple testing using BH method 
  mutate(adj.p.value=p.adjust(p.value, method = "BH"))
  
################################## Order of cytokines by descending significance ############################
Cytokine_order <-
  p_values %>% 
  dplyr::group_by(Cytokine) %>% 
  dplyr::arrange(adj.p.value) %>% 
  dplyr::filter(row_number() == 1) %>% 
  ungroup() %>% 
  dplyr::arrange(adj.p.value) %>% 
  dplyr::select(Cytokine) %>% 
  unlist()

Genetic_alt_order <-
  p_values %>% 
  dplyr::group_by(Genetic_alt) %>% 
  dplyr::arrange(adj.p.value) %>% 
  dplyr::filter(row_number() == 1) %>% 
  ungroup() %>% 
  dplyr::arrange(adj.p.value) %>% 
  dplyr::select(Genetic_alt) %>% 
  unlist()

############################# Define FDR cutoff #################################
fdr = 0.1
  
############################################ Plot ########################################
p_values %<>% 
  mutate(Cytokine=factor(Cytokine, levels = Cytokine_order)) %>% 
  mutate(Genetic_alt=factor(Genetic_alt, levels = Genetic_alt_order))
  
  ggplot(dplyr::filter(p_values, adj.p.value>fdr), 
         aes(x = Cytokine, y = -log10(adj.p.value))) +
  geom_point(color = "lightgrey", size = 3) +
  geom_beeswarm(data = dplyr::filter(p_values, adj.p.value <= fdr), 
                aes( color=Genetic_alt), size = 3, cex = 1.7) +
  ##FDR line  
  geom_hline(yintercept = -log10(fdr),linetype = "dashed", size=0.3)+
  
  ##Main Theme
  t1 +
  
  theme(axis.text.x = element_text(angle = 45,
                                   hjust = 1, 
                                   vjust = 1),
        axis.title.x=element_blank()) +
  ##Legend Theme
  theme(legend.position=c(0.65,0.65), 
        legend.title = element_text(face='bold', hjust = 0, size=13), 
        legend.key = element_blank(),  
        legend.text = element_text(size=11)) +
  guides(colour = guide_legend(nrow=7, title = "Mutations"), 
         shape = guide_legend(ncol = 1)) +
  scale_color_manual(name = "Mutations", values = colors, labels=c("IGHV.status"="IGHV status", 
                                                                   "del9p"="del(9p)", 
                                                                   "trisomy12"="trisomy 12", 
                                                                   "gain17q"="gain(17q)", 
                                                                   "gain2p"="gain(2p)",
                                                                   "gain19p"="gain(19p)",
                                                                   "gain19q"="gain(19q)",
                                                                   "del7q"="del(7q)",
                                                                   "del9q"="del(9q)", 
                                                                   "del4p"= "del(4p)", 
                                                                   "SPEN"="SPEN", 
                                                                   "del11q"="del(11q)", 
                                                                   "del1q" = "del(1q)")) +
  scale_y_continuous(expression("BH-adjusted  "* italic(p)*"-value"), 
                     breaks=seq(0, 10, 5),
                     labels=math_format(expr=10^.x)(-seq(0,10,5))) +
  
  scale_x_discrete(labels = c("TGF-b1"="TGF-\u03B21", "sCD40L+IL-4"="sCD40L + IL4", "IL-1b"="IL1\u03B2", "IL-4"="IL4", "IL-6"="IL6","IL-15"="IL15","IL-10"="IL10", "IL-21"="IL21","IL-2"="IL2", "Interferon gamma"= "Interferon \u03B3", "SDF-1a"="SDF-1\u03B1"))



```


### Multivariate analysis of gene - stimulus assosciations reveals multiple layers of biology involved
It was possible that there may be interplay between genetic factors in determining responses to external signals. To address this, I applied multivariate modelling to integrate the influence of genetic features, IGHV status and DNA methylation on the size of response. I used a Gaussian linear model with L1-penalty (i.e., lasso regression),  to derive a predictor for each stimulus, comprised of these covariates. 

As input to the model, the response matrix was composed of the log transformed viability values for each stimulus. To generate the feautre amtrix (137 samples versus 39 features), I excluded genetic features for which >20\% of there values were missing, and  patient samples with incomplete annotation. As predictors, I included genetic mutations and CNVs (p= 39), IGHV status (coded as 0-1) and Methylation CLuster (coded as 0, 0.5, 1). I ran lasso regression, as implemented `R` package [@glmnet][REFERENCE], using 3-fold cross-validation with misclassifcation error as loss. The resulting predictors are the mean  of those coefficients that were selected in at least 75% of 30 bootstrapped repeats. 

Using the output of the regression, I generated predictor profiles for each stimulus. For 5 / 17 stimuli, there was at least one genetic predictor that met the cut-offs (a selection are shown in figure \@ref(fig:stimuliGeneAssosciationsMulti)).

(ref:stimuliGeneAssosciationsMulti) 

```{r stimuliGeneAssosciationsMulti, fig.cap='(ref:stimuliGeneAssosciationsMulti)', message = FALSE,  echo = FALSE, fig.height=9, fig.width=8.5, fig.align="center", out.width = '70%', dev = 'cairo_pdf'}

#Generate feature Matrix
#select features from patient meta file
geneMatrix <- 
  dplyr::select(patMeta,
                -c(gender:treatment)) %>%
  
  #adjust IGHV.status levels  U and M to numeric 1 and 0 
  mutate(IGHV = ifelse(is.na(IGHV.status), NA,
                       ifelse(IGHV.status == "M", 1, 0)), 
         #adjust Methylation_Cluster levels  LP, IP, HP to 0, 0.5, 1
         Methylation = ifelse(is.na(Methylation_Cluster), NA,
                              ifelse(Methylation_Cluster == "LP", 0,
                                     ifelse(Methylation_Cluster == "IP", 0.5, 1))),
         #remove old columns
         IGHV.status=NULL, Methylation_Cluster=NULL ) %>%

  
  #convert factors to numeric
  mutate_if(is.factor, as.character) %>%
  mutate_at(vars(-PatientID), as.numeric) %>%
  
  #convert to matrix format, with patient IDs as rownames
  data.frame() %>% 
  column_to_rownames("PatientID") %>% 
  as.matrix()
#Tidy matrix for use in glmnet function
#Remove genes with higher than 20% missing values
geneMatrix <- geneMatrix[,colSums(is.na(geneMatrix))/nrow(geneMatrix) <= 0.2]
#Filter for patients with complete data
geneMatrix.complete <- geneMatrix[complete.cases(geneMatrix),]
#Combine KRAS, NRAS and BRAF mutations into a single column
#set up empty matrix
Ras_Raf <- matrix(NA, 
                  nrow = nrow(geneMatrix.complete), 
                  ncol = 1)
colnames(Ras_Raf) <- "RAS/RAF"
#add RAS/RAF column to matrix
geneMatrix.complete <- cbind(geneMatrix.complete, Ras_Raf)
#Annotate RAS_RAF where where any of KRAS, NRAS or BRAF are mutated
geneMatrix.complete[,"RAS/RAF"] <- ifelse(geneMatrix.complete[,"KRAS"]==1,1,
		                                        ifelse(geneMatrix.complete[,"BRAF"]==1,1,
	                	                          ifelse(geneMatrix.complete[,"NRAS"]==1, 1, 0)))
#remove KRAS, NRAS and BRAF columns
geneMatrix.complete <- 
  geneMatrix.complete[, colnames(geneMatrix.complete) != "KRAS"]
geneMatrix.complete <- 
  geneMatrix.complete[, colnames(geneMatrix.complete) != "BRAF"]
geneMatrix.complete <- 
  geneMatrix.complete[, colnames(geneMatrix.complete) != "NRAS"]


# Set up viability response matrix
viabMatrix <- 
  #select patients with complete meta data 
  dplyr::filter(df,
                PatientID %in% row.names(geneMatrix.complete)) %>%
  #select cytokine, patient and log(viability values only)
  dplyr::select(Cytokine, 
                Log, 
                PatientID) %>% 
  #reshape data
  spread(key = PatientID, value = Log) %>% 
  data.frame() %>% 
  #make Cytokine the row names
  remove_rownames() %>%
  column_to_rownames("Cytokine")
#make sample order same as in geneMatrix
viabMatrix <- viabMatrix[,rownames(geneMatrix.complete)]


#Run lasso regression
#set object to hold model outputs
dataResult <- list()
#fit model for each stimulus
for (i in rownames(viabMatrix)){  
  
    #prepare input and response matrices
    y <- unlist(viabMatrix[i,]) # viability for each patient with given condition
    X <- geneMatrix.complete #genetic features for each patient
    #fit the model
    cvglmfit <- runGlm05(X, y, method="lasso", repeats=30, folds=3)
    
    #collect the results for each stimulus in one object
    dataResult[[i]] <- cvglmfit
}


#Get predictor profiles for stimuli of interest

heatMaps_cyt <- lassoPlot05(dataResult[c("IL-4","CpG ODN", "TGF-b1")] , 
                          geneMatrix.complete, #use gene matrix for heatmap 
                          viabMatrix, #use viab matrix for scatter plot
                          freqCut = 0.75, #coefficients should be selected in <75% of bootstrapped model files
                          coefCut = 0.00) #no minimum value for coefficients 


#Assemble legend for heatmaps
#G# = gene mutations, I = IGHV, M = Methylation Cluster
legendFor = c("G", "I", "M")

#assign colours for I, G and M
coldef<-list()
coldef["I"] <- list(c(offwhite,"#707372")) #U-CLL and M-CLL
coldef["M"] <- list(c(offwhite, "#707372", "#A8A99E")) #IP, HP, LP
coldef["G"] <- list(c(offwhite, "#373A36")) #WT and Mutated
legends = makelegends(legendFor=c("G","I","M"),coldef)

#assemble
Fig3B_1 <- invisible(wrap_elements(arrangeGrob( grobs = heatMaps_cyt[1], ncol =1)))
Fig3B_2 <- invisible(wrap_elements(arrangeGrob( grobs = heatMaps_cyt[2], ncol =1)))
Fig3B_3 <- invisible(wrap_elements(arrangeGrob( grobs = heatMaps_cyt[3], ncol =1)))
Fig3B_leg <- legends[["plot"]] %>% wrap_elements()



blankplot <- 
  ggplot() +
  geom_blank() +
  theme(panel.background = element_blank())

wrap_elements(Fig3B_1 + Fig3B_2 + Fig3B_3 +
                wrap_elements(blankplot + Fig3B_leg + 
                                plot_layout(ncol=2, widths  = c(0.15,0.9))) +
                plot_layout(ncol=1, heights = c(1.7,1.7,1.7,0.5)))



```
Trisomy 12 and IGHV status were again the most common features. 

The number of predictors for the response to TLR stimuliation by CpG ODN indicates the multiple layers of biology involved. TLR response was highly heterogenous and a determinant of cluster memebrship that also related to clinical disease pgroess. For this reason, I decided to ivnestae these in more detail. 

### The imapct of RNA 
Also RNA plot 
### Individual cytokine – gene interactions of note
Role of IGHV status and BCR signalling in Cytokine response
(Figure \@ref(fig:IHGVBCR)).

(ref:IHGVBCR) 


```{r IHGVBCR, fig.cap='(ref:IHGVBCR)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 8, fig.align="center", out.width = '100%', dev = 'cairo_pdf'}


#add genetic data to viability dataframe
df_patmeta <- left_join(df, patMeta, by = "PatientID")

gg = 
  lapply(c("sCD40L", "IL-1\u03B2","TGF\u03B2"), function(x){

df_patmeta %>% 
  dplyr::filter(Cytlabel==x,
                !is.na(IGHV.status)) %>%

         
  ggplot(aes(x=IGHV.status,y=Log,color=(IGHV.status)))+
  geom_hline(yintercept = 0)+
  geom_boxplot()+
  geom_beeswarm(cex=1.5) +
  guides(color="none", shape="none")+
  stat_compare_means(method = "t.test",
                     label.x.npc = "center", 
                     size=3)+
  xlab("IGHV status") +
  ylab("Log(Viability)") +
  ggtitle(x) +
  scale_color_manual(values=c(colors[1], colors[2])) + 

  t2

})

wrap_elements(gg[[1]]) + wrap_elements(gg[[2]]) + wrap_elements(gg[[3]]) 


```

TLR response stratified by IGHV status (and trisomy 12) and discussion of interactions between BCR and TLR pathways
(Figure \@ref(fig:TLRIHGVtri12)).

(ref:TLRIHGVtri12) 


```{r TLRIHGVtri12, fig.cap='(ref:TLRIHGVtri12)', message = FALSE,  echo = FALSE, fig.height = 5, fig.width = 8,  fig.align="center", out.width = '100%', dev = 'cairo_pdf'}


#filter for Resiquimod-only treatment, only show patients who are annoyated for IGHV and Trisomy12
df_patmeta %>% 
  dplyr::filter(Cytokine=="Resiquimod",
                !is.na(trisomy12),
                !is.na(IGHV.status)) %>%
         
  ggplot(aes(x=interaction(IGHV.status, trisomy12),
                    y=Log,
                    color=(trisomy12)))+
  geom_hline(yintercept = 0)+
  geom_boxplot()+
  geom_beeswarm(cex=1.5) +
  guides(color="none", shape="none")+
  scale_x_discrete(labels=c("M.0"="IGHV-M\n WT",
                            "U.0"="IGHV-U\n WT",
                            "M.1"="IGHV-M\n trisomy 12",
                            "U.1"="IGHV-U\n trisomy 12"))+
  stat_compare_means(method = "t.test",
                     label.x.npc = "center", 
                     comparisons = list( c(1,2), c(3,4), c(1,3)),
                     size=3)+
  xlab("") +
  ylab("Log(Viability)") +
  ggtitle("Resiquimod") +
  scale_color_manual(values=c(colors[1], colors[2])) + 
  coord_cartesian(ylim = c(-1.5, 3.1), clip="off") +
  t1+
  theme(axis.text.x = element_text(angle = 45, vjust =1))


```

-Modulators of TLR signalling incl. effect of del11q and ATM mutations on TLR response
(Figure \@ref(fig:TLRGenes)).

(ref:TLRGenes) 

Potentially show effect of IGHV, then ivnestgiation of TLR combined boave and belwo, then look intro tri 12

These plots are not working right now - can work on them if i haev time 

```{r TLRGenes, fig.cap='(ref:TLRGenes)', message = FALSE,  echo = FALSE, fig.height = 5, fig.width = 10,  fig.align="center", out.width = '100%', dev = 'cairo_pdf'}


ATM <- 
df_patmeta %>% 
  dplyr::filter(Cytokine=="Resiquimod") %>%
      filter(!is.na(ATM)) %>%
         
  ggplot(aes(x=ATM,y=Log,color=(ATM)))+
  geom_hline(yintercept = 0)+
  geom_boxplot()+
  geom_beeswarm(cex=1.5) +
  guides(color="none", shape="none")+
  stat_compare_means(method = "t.test",
                     label.x.npc = "center", 
                     size=3)+
  xlab("ATM") +
  ylab("Log(Viability)") +
  ggtitle("") +
  scale_color_manual(values=c(colors[1], colors[2])) + 

  t2

del11q <- 
df_patmeta %>% 
  dplyr::filter(Cytokine=="Resiquimod") %>%
      filter(!is.na(del11q)) %>%
         
  ggplot(aes(x=del11q,y=Log,color=(del11q)))+
  geom_hline(yintercept = 0)+
  geom_boxplot()+
  geom_beeswarm(cex=1.5) +
  guides(color="none", shape="none")+
  stat_compare_means(method = "t.test",
                     label.x.npc = "center", 
                     size=3)+
  xlab("del(11q)") +
  ylab("Log(Viability)") +
  ggtitle("") +
  scale_color_manual(values=c(colors[1], colors[2])) + 

  t2

SF3B1 <- 
df_patmeta %>% 
  dplyr::filter(Cytokine=="Resiquimod") %>%
      filter(!is.na(SF3B1)) %>%
         
  ggplot(aes(x=SF3B1,y=Log,color=(SF3B1)))+
  geom_hline(yintercept = 0)+
  geom_boxplot()+
  geom_beeswarm(cex=1.5) +
  guides(color="none", shape="none")+
  stat_compare_means(method = "t.test",
                     label.x.npc = "center", 
                     size=3)+
  xlab("SF3B1") +
  ylab("Log(Viability)") +
  ggtitle("") +
  scale_color_manual(values=c(colors[1], colors[2])) + 

  t2

trisomy12 <- 
df_patmeta %>% 
  dplyr::filter(Cytokine=="Resiquimod") %>%
      filter(!is.na(trisomy12)) %>%
         
  ggplot(aes(x=trisomy12,y=Log,color=(trisomy12)))+
  geom_hline(yintercept = 0)+
  geom_boxplot()+
  geom_beeswarm(cex=1.5) +
  guides(color="none", shape="none")+
  stat_compare_means(method = "t.test",
                     label.x.npc = "center", 
                     size=3)+
  xlab("trisomy 12") +
  ylab("Log(Viability)") +
  ggtitle("Resiquimod") +
  scale_color_manual(values=c(colors[1], colors[2])) + 

  t2

del17p <- 
df_patmeta %>% 
  dplyr::filter(Cytokine=="Resiquimod") %>%
      filter(!is.na(del17p)) %>%
         
  ggplot(aes(x = del17p,y = Log,color = (del17p)))+
  geom_hline(yintercept = 0)+
  geom_boxplot()+
  geom_beeswarm(cex=1.5) +
  guides(color="none", shape="none")+
  stat_compare_means(method = "t.test",
                     label.x.npc = "center", 
                     size=3)+
  xlab("del(17p)") +
  ylab("Log(Viability)") +
  ggtitle("") +
  scale_color_manual(values=c(colors[1], colors[2])) + 

  t2

TP53 <- 
df_patmeta %>% 
  dplyr::filter(Cytokine=="Resiquimod") %>%
      filter(!is.na(TP53)) %>%
         
  ggplot(aes(x=TP53,y=Log,color=(TP53)))+
  geom_hline(yintercept = 0)+
  geom_boxplot()+
  geom_beeswarm(cex=1.5) +
  guides(color="none", shape="none")+
  stat_compare_means(method = "t.test",
                     label.x.npc = "center", 
                     size=3)+
  xlab("TP53") +
  ylab("Log(Viability)") +
  ggtitle("") +
  scale_color_manual(values=c(colors[1], colors[2])) + 

  t2



(del11q + trisomy12 + del17p)/
(ATM + SF3B1 + TP53)  

```

-samples with KRAS mutations benefit less from IL4 stimulation

(Figure \@ref(fig:IL4KRAS)).

(ref:IL4KRAS) 

```{r IL4KRAS, fig.cap='(ref:IL4KRAS)', message = FALSE,  echo = FALSE, fig.height = 5, fig.width = 8,  fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

df_patmeta %>% 
   dplyr::filter(Cytokine=="IL-4",
                !is.na(KRAS)) %>% 
         
  ggplot(aes(x=KRAS,y=Log, color=(KRAS)))+
  geom_hline(yintercept = 0)+
  geom_boxplot()+
  geom_beeswarm(cex=1.5) +
  guides(color="none", shape="none")+
  stat_compare_means(method = "t.test",
                     label.x.npc = "center", 
                     size=3)+
  xlab("KRAS") +
  ylab("Log(Viability)") +
  ggtitle("IL4") +
  scale_color_manual(values=c(colors[1], colors[2])) + 

  t2


```

## Profiling the effects of trisomy 12 on cytokine response, and how this may be possible
### Trisomy 12 modulates responses to IL4, TGFbeta and TLR stimuli


(Figure \@ref(fig:tri12cytResponse)).

(ref:tri12cytResponse) 

```{r tri12cytResponse, fig.cap='(ref:tri12cytResponse)', message = FALSE,  echo = FALSE, fig.height = 5, fig.width = 8,  fig.align="center", out.width = '100%', dev = 'cairo_pdf'}

#add genetic data to viability dataframe
df_patmeta <- left_join(df, patMeta, by = "PatientID")

ii = 
  lapply(c("IL4", "sCD40L + IL4", "TGF\u03B2"), function(x){

df_patmeta %>% 
  dplyr::filter(Cytlabel==x,
                !is.na(trisomy12)) %>%

         
  ggplot(aes(x=trisomy12,y=Log,color=(trisomy12)))+
  geom_hline(yintercept = 0)+
  geom_boxplot()+
  geom_beeswarm(cex=1.5) +
  guides(color="none", shape="none")+
  stat_compare_means(method = "t.test",
                     label.x.npc = "center", 
                     size=3)+
  xlab("trisomy 12") +
  ylab("Log(Viability)") +
  ggtitle(x) +
  scale_color_manual(values=c(colors[1], colors[2])) + 

  t2

})

wrap_elements(ii[[1]]) + wrap_elements(ii[[2]]) + wrap_elements(ii[[3]]) 

```

### Gene dosage effects
(Figure \@ref(fig:SMAD3geneDosage).

(ref:SMAD3geneDosage) 

_The Proteomics dataset was kindly shared by Sophie Herbst, from her paper..._ 
```{r prepgeneDosage}

#get table to convert between IDS
ensembl = useEnsembl(biomart="ensembl", dataset="hsapiens_gene_ensembl", version="90")
transAnno <- getBM(attributes=c('ensembl_gene_id','hgnc_symbol','chromosome_name', 'transcript_biotype'), mart = ensembl)
transAnno <- dplyr::filter(transAnno, hgnc_symbol != "")
id2gene <- dplyr::select(transAnno, ensembl_gene_id, hgnc_symbol)

```

```{r SMAD3geneDosage, fig.cap='(ref:SMAD3geneDosage)', message = FALSE,  echo = FALSE, fig.height = 5, fig.width = 8,  fig.align="center", out.width = '100%', dev = 'cairo_pdf'}

data <- plotCounts(dds_smp, gene="ENSG00000166949", #SMAD3
                   intgroup=c("trisomy12"), 
                   returnData=TRUE) %>%
      filter(trisomy12 %in% c("0", "1"))


#Make plot
SMAD3rna <- 
  ggplot(data, aes(x = trisomy12, y = count, color=trisomy12)) +
  geom_boxplot() +
  geom_beeswarm() + 
  ggtitle("SMAD3") +
  ylab("Counts") +
  xlab("Trisomy 12 Status") +
  t2 +
  scale_color_manual(values = c(colors[1], colors[2])) +
  theme(aspect.ratio=1,  legend.position = "none") +
  stat_compare_means(method = "t.test", comparisons = list(c("1", "0")), size=3)
 
    
df_tbl_prot <- 
  wideFormat(multiomics_MAE["SMAD3",,"proteomics"] ) %>% as_tibble()

colnames(df_tbl_prot)[2] <- "protein"

mae_mini <- multiomics_MAE[,,c("SNPs", "chrom_abber", "health_record_bin")]

df_tbl_SNP <- wideFormat(mae_mini["trisomy12",,]) %>% as_tibble()
colnames(df_tbl_SNP)[2] <- "alteration"
df_tbl_SNP <- df_tbl_SNP %>% mutate(alteration=factor(alteration))
df_tbl <- left_join(df_tbl_prot, df_tbl_SNP, by="primary" )

#Make plot
SMAD3pro = 
  df_tbl %>% 
  filter(!is.na(alteration), !is.na(protein)) %>% 
  
  ggplot(aes(alteration, protein, colour = alteration, group=alteration)) +
  scale_y_log10() +
  geom_boxplot() + 
  geom_point() +
  ggtitle("SMAD3") +
  ylab("Protein abundance") +
  xlab("trisomy 12") + 
  theme_bw() +  
  scale_colour_manual(values = c(colors[1],colors[2])) + 
  t2 + 
  theme(aspect.ratio=1,  legend.position = "none") +
  stat_compare_means(method = "t.test", comparisons = list(c("1", "0")), size=3)
  
     
SMAD3rna + SMAD3pro

```

(Figure \@ref(fig:STATgeneDosage)).

(ref:STATgeneDosage) 
```{r STATgeneDosage, fig.cap='(ref:STATgeneDosage)', message = FALSE,  echo = FALSE, fig.height = 5, fig.width = 8,  fig.align="center", out.width = '100%', dev = 'cairo_pdf'}


stat_rna <- 
lapply(c("ENSG00000166888", "ENSG00000170581"), function(x){
    #Prep data for plot
    data <- plotCounts(dds_smp, gene=x, intgroup=c("trisomy12"), returnData=TRUE) %>%
      filter(trisomy12 %in% c("0", "1"))
    
    #get hgnc symbol from ENS id
    gene_name <-
    as.data.frame(rowData(dds_smp)) %>% 
    rownames_to_column("ensembl_gene_id") %>%
    filter(ensembl_gene_id == x) %>%
    dplyr::select(symbol)
    if(gene_name== "SPI1"){genename <- "PU.1"}
    
    #Make plot
    ggplot(data, aes(x = trisomy12, y = count, color=trisomy12)) +
      geom_boxplot() +
      geom_beeswarm() + 
      ggtitle(gene_name) +
      ylab("Counts") +
      xlab("") +
      
      t2 +
      scale_color_manual(values = c(colors[1], colors[2])) +
      theme(aspect.ratio=1,  legend.position = "none") +
      
      stat_compare_means(method = "t.test", comparisons = list(c("1", "0")), size=3)
  
})


stat_pro = lapply(c("STAT6", "STAT2"), function(x){
     #Prepare data for plot
     df_tbl_prot <- wideFormat(multiomics_MAE[x,,"proteomics"] ) %>% as_tibble()
     colnames(df_tbl_prot)[2] <- "protein"
     
     mae_mini <- multiomics_MAE[,,c("SNPs", "chrom_abber", "health_record_bin")]
     df_tbl_SNP <- wideFormat(mae_mini["trisomy12",,]) %>% as_tibble()
     colnames(df_tbl_SNP)[2] <- "alteration"
     df_tbl_SNP <- df_tbl_SNP %>% mutate(alteration=factor(alteration))
     
     df_tbl <- left_join(df_tbl_prot, df_tbl_SNP, by="primary" )
     
     #Make plot
      gg = df_tbl %>% filter(!is.na(alteration), !is.na(protein)) %>% ggplot(aes(alteration, protein, colour = alteration, group=alteration)) +
       geom_boxplot() + 
       geom_point() +
        
       ggtitle("")+
       ylab("Protein abundance") +
       xlab("trisomy 12") + 
        
       theme_bw()+  
       scale_colour_manual(values = c(colors[1],colors[2])) + 
       t2 +  
       theme(aspect.ratio=1, legend.position = "none") +
       stat_compare_means(method = "t.test", comparisons = list(c("1", "0")), size=3)
  
})

(stat_rna[[1]] + stat_rna[[2]])/
  (stat_pro[[1]] + stat_pro[[2]]) 

```

(Figure \@ref(fig:IRAK4geneDosage)).

(ref:IRAK4geneDosage) 

```{r IRAK4geneDosage, fig.cap='(ref:IRAK4geneDosage)', message = FALSE,  echo = FALSE, fig.height = 5, fig.width = 8,  fig.align="center", out.width = '100%', dev = 'cairo_pdf'}

#IRAK4

irak4_rna <- 
lapply(c("ENSG00000198001","ENSG00000172936"),  function(x){
    #Prep data for plot
    data <- plotCounts(dds_smp, gene=x, intgroup=c("trisomy12"), returnData=TRUE) %>%
      filter(trisomy12 %in% c("0", "1"))
    
    #get hgnc symbol from ENS id
    gene_name <-
    as.data.frame(rowData(dds_smp)) %>% 
    rownames_to_column("ensembl_gene_id") %>%
    filter(ensembl_gene_id == x) %>%
    dplyr::select(symbol)
    if(gene_name== "SPI1"){genename <- "PU.1"}
    
    #Make plot
    ggplot(data, aes(x = trisomy12, y = count, color=trisomy12)) +
      geom_boxplot() +
      geom_beeswarm() + 
      ggtitle(gene_name) +
      ylab("Counts") +
      xlab("") +
      
      t2 +
      scale_color_manual(values = c(colors[1], colors[2])) +
      theme(aspect.ratio=1,  legend.position = "none") +
      
      stat_compare_means(method = "t.test", comparisons = list(c("1", "0")), size=3)
  
})


irak4_pro = lapply(c("IRAK4", "MYD88"), function(x){
     #Prepare data for plot
     df_tbl_prot <- wideFormat(multiomics_MAE[x,,"proteomics"] ) %>% as_tibble()
     colnames(df_tbl_prot)[2] <- "protein"
     
     mae_mini <- multiomics_MAE[,,c("SNPs", "chrom_abber", "health_record_bin")]
     df_tbl_SNP <- wideFormat(mae_mini["trisomy12",,]) %>% as_tibble()
     colnames(df_tbl_SNP)[2] <- "alteration"
     df_tbl_SNP <- df_tbl_SNP %>% mutate(alteration=factor(alteration))
     
     df_tbl <- left_join(df_tbl_prot, df_tbl_SNP, by="primary" )
     
     #Make plot
      gg = df_tbl %>% filter(!is.na(alteration), !is.na(protein)) %>% ggplot(aes(alteration, protein, colour = alteration, group=alteration)) +
       geom_boxplot() + 
       geom_point() +
        
       ggtitle("")+
       ylab("Protein abundance") +
       xlab("trisomy 12") + 
        
       theme_bw()+  
       scale_colour_manual(values = c(colors[1],colors[2])) + 
       t2 +  
       theme(aspect.ratio=1, legend.position = "none") +
       stat_compare_means(method = "t.test", comparisons = list(c("1", "0")), size=3) 
  
})

(irak4_rna[[1]] + irak4_rna[[2]])/
  (irak4_pro[[1]] + irak4_pro[[2]])

```

### Investigation of non-trisomy12 samples that are phenocopies of trisomy12 CLL+ (based on their responses to stimuli) 
(Figure \@ref(fig:tri12Classfier).

(ref:tri12Classfier) 

```{R phenocopyPrework}
########### Viability matrix ################
#make viability matrix for cytokine treatments for patients
cyt_viab <- dplyr::select(df, 
                PatientID, 
                Log, 
                Cytokine) %>% spread(Cytokine, Log) %>% as.data.frame()

#make PatID the row names
rownames(cyt_viab) <- unlist(cyt_viab[,1]) # the first row will be the header
cyt_viab <- dplyr::select(cyt_viab,-PatientID) 

cyt_viab <- t(cyt_viab)

#run scaleCytResp on viability matrix
x <- scaleCytResp05(cyt_viab)
x <-t(x)


## Response matrix 
#Get Tri 12 status
y <- dplyr::select(patMeta, PatientID, trisomy12) %>% dplyr::filter(!is.na(trisomy12)) %>% column_to_rownames("PatientID")

y$trisomy12 <- as.numeric(as.character(y$trisomy12))

y <- t(y)

#Remove NA patients from viability matrix
x <- x[colnames(y),]


# Run logistic regression
X <- x

#set whether to spilt data into test and training set
testRatio = FALSE

#set parameters for running model
repeats <- 50
folds <- 3

if(testRatio){
  
  glmRes <- runGlm.bin(X,as.integer(y), repeats = repeats, folds =folds, testRatio = 0.3)
  
}else{
  
  glmRes <- runGlm.bin(X, as.integer(y), repeats = repeats, folds = folds, testRatio = NULL)}



## ROC of best model
if(testRatio){
iBest <- which.max(glmRes$aucCV)
glmRes$rocTest[[iBest]]
}else{
  iBest <- which.max(glmRes$aucCV)
}

```



```{r tri12Classfier, fig.cap='(ref:tri12Classfier)', message = FALSE,  echo = FALSE, fig.height=3, fig.width = 8, fig.align="center", out.width = '100%', dev = 'cairo_pdf'}

X.scale <- X

p <- lassoPlot_bin(glmRes, X.scale, y,freqCut = 0, coefCut = 0.01, modelIndex = iBest, rowHeight = 0.001)

grid.arrange(p)

```

(Figure \@ref(fig:tri12PhenocopiesCytResponse).

(ref:tri12PhenocopiesCytResponse) 

```{r tri12PhenocopiesCytResponse, fig.cap='(ref:tri12PhenocopiesCytResponse)', message = FALSE,  echo = FALSE, fig.height=3, fig.width = 3, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}


plotTab <- filter(df, Cytokine%in% c("Resiquimod", "sCD40L+IL-4", "TGF-b1"))
plotTab %>%
  ggplot(aes(x=Cytokine,
             y=Log)) + 
  geom_boxplot(fill=NA)+
  geom_beeswarm(cex=0.3, alpha=0.4, color= lightergrey)+
  t2 +
  ylab("Log(Viability") + xlab("")+
  geom_point(data=dplyr::filter(plotTab, PatientID =="Pat_033"), aes(x=Cytokine, y=Log), colour=palblues[1], size=2) + 
    geom_point(data=dplyr::filter(plotTab, PatientID =="Pat_114"), aes(x=Cytokine, y=Log), colour=palreds[8], size=2)+ scale_x_discrete(labels = c("Resiquimod" = "Resiquimod", "sCD40L+IL-4" = "sCD40L \n+ IL4", "TGF-b1" = "TGF-\u03B21"))


```
(Figure \@ref(fig:tri12PhenocopiesRNA).

(ref:tri12PhenocopiesRNA) 

```{r tri12PhenocopiesRNA, fig.cap='(ref:tri12PhenocopiesRNA)', message = FALSE,  echo = FALSE, fig.height=6, fig.width = 5, fig.align="center", out.width = '80%', dev = 'cairo_pdf'}
genes <- c("ENSG00000269404", #SPIB
           "ENSG00000166888", #STAT6
           "ENSG00000198001", #IRAK4
           "ENSG00000166949" #SMAD3
           )
RNAplots <- 
lapply(genes, function(x){
  data <- plotCounts(dds_smp_full, gene=x, intgroup=c("trisomy12"), returnData=TRUE) %>%
      filter(trisomy12 %in% c("0", "1")) %>% rownames_to_column("PatientID")
    
    #get hgnc symbol from ENS id
    gene_name <-
    as.data.frame(rowData(dds_smp_full)) %>% 
    rownames_to_column("ensembl_gene_id") %>%
    filter(ensembl_gene_id == x) %>%
    dplyr::select(symbol)
    if(gene_name== "SPI1"){genename <- "PU.1"}
    
    #Make plot
    ggplot(data, aes(x = trisomy12, y = count, color = trisomy12)) +
      geom_boxplot() +
      geom_beeswarm() + 
      ggtitle(gene_name) +
      ylab("Counts") +
      xlab("trisomy 12") +
      
      t2 +
      scale_color_manual(values = c(colors[1], colors[2])) +
      theme(aspect.ratio=1,  text = element_text(size = 15), legend.position = "none") +
      
      stat_compare_means(method = "t.test", comparisons = list(c("1", "0")), size = 3) + 
      
      geom_point(data=dplyr::filter(data, PatientID =="Pat_114"), aes(x=trisomy12, y=count), colour=palreds[8], size=2) +
      geom_point(data=dplyr::filter(data, PatientID =="Pat_033"), aes(x=trisomy12, y=count), colour=palblues[1], size=2)
      
})

(RNAplots[[1]] + RNAplots[[2]])/
(RNAplots[[3]] + RNAplots[[4]]) 


```


Pat033 has duplicated regions: 
-p13.31 - 42 copies
-q24.13 - 10 copies
-q24.33 - 21 copies 
See what Junyan says about this and potentially add a figure 

(Figure \@ref(fig:tri12PhenocopiesCNV).

(ref:tri12PhenocopiesCNV) 

```{r tri12PhenocopiesCNV, fig.cap='(ref:tri12PhenocopiesCNV)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}



```

### Investigation of TF activity in trisomy12+ CLL and finding that SPIB and PU1 TFs are upregulated in trisomy12+ CLL
tri 12 annotation 

running diffTF, add a diagram to explain?

(Figure \@ref(fig:diffTFexplainer).

(ref:diffTFexplainer) 

```{r diffTFexplainer, fig.cap='(ref:diffTFexplainer)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

knitr::include_graphics("figures/diffTF_method.eps")

```

visualisation of results 


(Figure \@ref(fig:tri12annotation).

(ref:tri12annotation) 

```{r tri12annotation, fig.cap='(ref:tri12annotation)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}




```


(Figure \@ref(fig:tri12diffTF).

(ref:tri12diffTF) 

Also refer to : (Figure \@ref(fig:diffTFsmallvolPlot)

```{r tri12diffTF, fig.cap='(ref:tri12diffTF)', message = FALSE,  echo = FALSE,fig.height=6, fig.width=8, fig.align="center", out.width = '80%', dev = 'cairo_pdf'}


#Get all TFs that show differential activity
plotTab.diffTF <- dplyr::filter(diffTF_large, pvalueAdj < 0.05)
#change to name of TF quoted in text
plotTab.diffTF$TF <- gsub("SPI1", "PU1", plotTab.diffTF$TF)
#order by weighted mean difference
idx <- order(plotTab.diffTF[["weighted_meanDifference"]], decreasing = TRUE)
plotTab.diffTF$TF <- factor(plotTab.diffTF$TF,levels=rev(unique(plotTab.diffTF$TF[idx])))
#plot figure

  ggplot(plotTab.diffTF,
         aes(x=weighted_meanDifference,
             y=TF,
             fill= ifelse(weighted_meanDifference>0, 
                          palreds[5], palblues[1]))) +
  geom_bar(stat = "identity", width = 0.75) +
  xlim(c(-0.2, 0.175)) +
  
  #colours and theme
  scale_fill_manual(values = c(palblues[1], palreds[8]), guide = "none") +
  t1 + 
  theme(axis.text.x = element_text(angle = 45, vjust =1))+
  
  #Labels
  ylab("") + 
  xlab("Change in TF activity") +
  ggtitle("Diffential TF activity in trisomy 12 CLL") + 
  
  #Add annotations
  #arrow 1
  annotate("segment", x = -0.15, xend = -0.15, y = 8.4, yend = 0.75, 
           colour = "#5e5e5e", size=3, alpha=1, arrow=arrow()) + 
  #arrow 2
  annotate("segment", x = -0.15, xend = -0.15, y = 8.6, yend = 17, 
           colour = "#5e5e5e", size=3, alpha=1, arrow=arrow()) + 
 
   #2 arrow labels
  annotate("text", x = c(-0.185,-0.185), y = c(4.8,12.5), 
          label = c("Lower activity in trisomy 12 CLL", 
                    "Higher activity in trisomy 12 CLL") , 
          color="black", 
          size=5 , fontface="bold") +
  #flip plot
  coord_flip() 
  
  
  

```

## Profiling the downstream effects of SPIB and PU1:
### Over-representation of BCR, TLR, interleukin and TGFbeta response genes amongst SPIB and PU1 targets in ChIPseq data
(Figure \@ref(fig:SpiBChipSeq).

(ref:SpiBChipSeq) 

```{r SpiBChipSeq, fig.cap='(ref:SpiBChipSeq)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

```

### Visualisation of SPIB and PU1 binding sites from ChIPseq data in BCR, TLR, interleukin and TGFbeta signalling genes
(Figure \@ref(fig:SpiBBindingSites).

(ref:SpiBBindingSites) 

```{r SpiBBindingSites, fig.cap='(ref:SpiBBindingSites)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

```

### Investigation of DE genes within IL4 and TGFbeta pathways in SPIB/PU1 DKO mice
(Figure \@ref(fig:SpiBDEgenes).

(ref:SpiBDEgenes) 

```{r SpiBDEgenes, fig.cap='(ref:SpiBDEgenes)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

```

### SPIb and PU1 linked genes 
Key result: generation of GRN to identify SPIB and PU1 regulated genes, and investigation of associated log2FCs of these genes in trisomy 12 vs non-trisomy12 samples (including CD79A and B) 

(Figure \@ref(fig:SpiBGRN).

(ref:SpiBGRN) 

```{r SpiBGRN, fig.cap='(ref:SpiBGRN)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

```


### SPIB/PU1 DKOs in trisomy 12+ cell lines 
(Figure \@ref(fig:SpiBshRNAKD).

(ref:SpiBshRNAKD) 

_The data here was generated by Tina Bericoviv and was included in Bruch & Giles et al. 2021. It is included in this thesis for completeness._ 

```{r SpiBshRNAKD, fig.cap='(ref:SpiBshRNAKD)', message = FALSE,  echo = FALSE, fig.height=3, fig.width = 7, fig.align="center", out.width = '100%', dev = 'cairo_pdf'}


  SPIB_KD_Cellcounts %>% 

  dplyr::filter(treatment%in%c("scr_shRNA", "PU.1-KD","SpiB-KD", "Double-KD")) %>% 
  mutate(treatment=factor(treatment, levels = c("scr_shRNA", "PU.1-KD", "SpiB-KD", "Double-KD"))) %>% 
  
  
  
  ggplot(aes(x=Day, y=NormalizedCellCount, color=treatment, group=interaction( treatment)))+
  stat_summary(fun = mean,geom='line',aes(color=treatment), size=1) +
  stat_summary(fun=mean,geom='point', size=2) +
  facet_wrap(~Cellline, scales="fixed")+
  scale_y_log10()+
  labs(y="Relative Cellcount")+
 t2+
  theme(legend.position = "bottom", legend.key = element_blank(), legend.title = element_text(face='bold', size=fontsize+4),  legend.text = element_text(size=fontsize+2), strip.background = element_blank(), strip.text = element_text(size=18, face="bold"))+
      scale_color_manual(name = "Condition", values = c("darkgrey", colors[c(7,8,3)]), labels=c("scr_shRNA"="Control","PU.1-KD"="PU.1-Knockdown","SpiB-KD"="Spi-B-Knockdown","Double-KD"="Spi-B + PU.1 Double Knockdown" ))


```


## Conclusions 

so everything is here 
focus on the content you have and finihsh - add additioanl if time 