# Molecular and microenvironmental modulators of drug response in CLL

Intro text 

```{r, echo = FALSE}

knitr::opts_chunk$set(
   echo = FALSE, 
   message = FALSE, 
   warning = FALSE,
   fig.align="center"
)

```

```{r setup06, echo = FALSE}

#libraries
library(patchwork)
library(ggplot2)
library(data.table)
library(magrittr)
library(pheatmap)
library(tidyr)
library(dplyr)
library(plyr)
library(ggpubr)
library(ggbeeswarm)
library(gtable)
library(glmnet)
library(RColorBrewer)
library(gridExtra)
library(cowplot)
library(tidyverse)

```

```{r loadData06, eval = TRUE}

#data("df", "patMeta", "LDT", "survT", "dds_smp")
#df_complete <- df

#Data
#df: tibble containing all screening viability data
load("data/df.RData")
df_complete <- df

#patMeta: tibble containing all patient genetic data
load("data/patMeta.RData")


```

```{r themes_functions06, echo = FALSE, eval = TRUE}

### ggplot themes
fontsize = 11

## theme for ggplots
t1 <- 
  theme(                              
  plot.background = element_blank(), 
  panel.grid.major = element_line(),
  panel.grid.major.x = element_line(linetype = "dotted", colour = "grey"),
  panel.grid.minor = element_blank(), 
  panel.border = element_blank(), 
  panel.background = element_blank(),
  axis.line = element_line(size=.4),
  axis.line.x = element_line(),
  axis.line.y = element_line(),
  axis.text.x  = element_text(angle=90, size=11, hjust = 1, vjust = 0.4),
  axis.text.y = element_text(size = 11),
  axis.ticks.length = unit(0.3,"cm"),
  axis.title.x = element_text(face="bold", size=13), 
  axis.title.y = element_text(face="bold", size=13),
  plot.title = element_text(face="bold", size=14, hjust = 0.5),
  strip.text = element_text(size = fontsize)
)

t2 <- t1+
  theme( axis.text.x  = element_text(angle=0, size=11, hjust = 0.5, vjust = 1))

## theme for legends
t.leg <-  theme(legend.title = element_text(face='bold', 
                                            hjust = 1, size=11),
                legend.key = element_blank(),
                legend.text = element_text(size=11),
                legend.background = element_rect(color = "black"))



### Set colour palettes

#For Categorical: 
colors <- c("#A1BE1F", #green
            "#F4C61F", #yellow
            "#734595", #purple
            "#D41645", #red
            "#3B6FB6", #blue
            "#B65417", #orange
            "#E2E868", #light green
            "#CBA3D8", #light purple
            "#E58F9E", #light purple
            "#8BB8E8", #light blue
            "#F49E17", #light orange
            "#303030", #black
            "#A8A99E", #grey
            "#007B53") #dark green



#For Divergent: 
Divergent <- c("#003DA5", "#2055B0", "#406EBC", "#6086C7", "#809ED2", "#9FB6DD", "#BFCFE9", "#DFE7F4", "white", "white", "white","#F4E0E7", "#E9C2CF", "#DEA3B6", "#D3849E", "#C76586", "#BC476E", "#B12855", "#A6093D")

#for negatives only: 
palblues <- c("#003DA5", "#2055B0", "#406EBC", "#6086C7", "#809ED2", "#9FB6DD", "#BFCFE9", "#DFE7F4")

#for positives only:
palreds <- c("#F4E0E7", "#E9C2CF", "#DEA3B6", "#D3849E", "#C76586", "#BC476E", "#B12855", "#A6093D")

#For mutations: 
Mutant <- c("#b5b5b5","#373A36")
Sex <- c("#707372","#D0D0CE")
IGHV <- c("#373A36","#D0D0CE")
Methylation_cluster <- c("#373A36","#A8A99E","#D0D0CE")

#For drugs: 
drugpal <- c("#734595", "#CBA3D8") #purples

#For cytokines: 
cytpal <- c("#F49E17", "#EFC06E") #yellows


#neutral
offwhite <- "#f8f8ff"
lightergrey <- "#D0D0CE"
darkergrey <- "#707372"


#Function for multi-variant regression
runGlm06 <- function(X, y, method = "lasso", repeats = 20, folds = 3) {
  #set up objects to store results
  modelList <- list()
  lambdaList <- c()
  varExplain <- c()
  coefMat <- matrix(NA, ncol(X), repeats)
  rownames(coefMat) <- colnames(X)

  #set alpha
  if (method == "lasso"){
    alpha = 1
  } else if (method == "ridge") {
    alpha = 0
  }
  
  #for the set number of repeats, run the regression
  for (i in seq(repeats)) {
    if (ncol(X) > 2) {
      #run cross validated generalised linear model with given parameters
      res <- cv.glmnet(X,y, type.measure = "mse", family="gaussian", 
                       nfolds = folds, alpha = alpha, standardize = FALSE)
      
      #store lamdas and min lambda value
      lambdaList <- c(lambdaList, res$lambda.min)
      
      #store result of cv.glmnet
      modelList[[i]] <- res
      
      #get coefficents with min lambda value 
      coefModel <- coef(res, s = "lambda.min")[-1] #remove intercept row
      
      #store coefficients for this repeat
      coefMat[,i] <- coefModel
      
      #calculate variance explained
      if(sum(coefModel !=0)){
      y.pred <- predict(res, s = "lambda.min", newx = X)
      #if there are no predictors, all y.pred will be the same so its not possible to calculate variance explained because the SD is 0
      varExp <- cor(as.vector(y),as.vector(y.pred))^2
      }else{ varExp <- NA}
      varExplain[i] <- ifelse(is.na(varExp), 0, varExp) 
      
     
      
    } else {
      fitlm<-lm(y~., data.frame(X))
      varExp <- summary(fitlm)$r.squared
      varExplain <- c(varExplain, varExp)
      
    }

  }
  #store all results 
  list(modelList = modelList, lambdaList = lambdaList, varExplain = varExplain, coefMat = coefMat)
}


lassoPlot06 <- function(lassoOut, geneMatrix, betaMatrix, freqCut = 1, coefCut = 0.01) {
  #object to hold all plots
  plotList <- list()
  
  #for each drug - stimuli combination, run the following:
  for (seaName in names(lassoOut)) {
    ###FOR THE BAR PLOT
    #extract mean coefficients for each drug - stimuli combination
    barValue <- rowMeans(lassoOut[[seaName]]$coefMat)
    #extract proportion of repeats for which each coefficient is significant
    freqValue <- rowMeans(abs(sign(lassoOut[[seaName]]$coefMat)))
    #filter out coefficients that don't meet freqCut and coefCut thresholds
    barValue <- barValue[abs(barValue) >= coefCut & freqValue >= freqCut] 
    #arrange the bar values in numerical order
    barValue <- barValue[order(barValue)]
    #if there are no sig coefficients, don't plot
    if(length(barValue) == 0) {
      plotList[[seaName]] <- NA
      next
    }
    
   
    ###FOR THE HEATMAP AND SCATTER PLOT
    #get feature matrix and response matrix to plot
    allData <- geneMatrix
    betaValue <- unlist(betaMatrix[seaName,])
    
    #get feature matrix for features with significant coefficients only
    tabValue <- allData[, names(barValue),drop=FALSE]
    ord <- order(betaValue)
    betaValue <- betaValue[ord]
    tabValue <- tabValue[ord, ,drop=FALSE]
    sampleIDs <- rownames(tabValue)
    tabValue <- as_tibble(tabValue)
    tabValue$Sample <- sampleIDs
    
    #annotate features as mutations, methylation cluster or IGHV, and apply different scaling     so that different colours can be used in plotting 
    
    #for mutations:
    matValue <- gather(tabValue, key = "Var",value = "Value", -Sample)
    matValue$Type <- "mut"
    
    #for methylation cluster
    matValue$Type[grep("Methylation",matValue$Var)] <- "meth"
    
    #for IGHV status
    matValue$Type[grep("IGHV.status",matValue$Var)] <- "ighv"
    
    #change the scale of the value so that IGHV, Methylation and Mutation do not overlap
    matValue[matValue$Type == "mut",]$Value = matValue[matValue$Type == "mut",]$Value + 10
    matValue[matValue$Type == "meth",]$Value = matValue[matValue$Type == "meth",]$Value + 20
    matValue[matValue$Type == "ighv",]$Value = matValue[matValue$Type == "ighv",]$Value + 30
    
    #change continous to catagorical
    matValue$Value <- factor(matValue$Value,levels = sort(unique(matValue$Value)))
    
    #arrange order of feature rows and columns in heatmap
    #heatmap rows should align with order of genetic coefficients
    matValue$Var <- factor(matValue$Var, levels = names(barValue))
    
    #sample columns should be ascending order according to value of coefficient of each patient, so that heatmap aligns with scatter plot below
    matValue$Sample <- factor(matValue$Sample, levels = names(betaValue))
    
    #update labels, keep factor levels
    matValue$Var <- revalue(matValue$Var, c("IGHV.status" = "IGHV status", "del11q" = "del(11q)", "del13q" = "del(13q)", "del17p" = "del(17p)", "trisomy12" = "trisomy 12")) 
    
    #MAKE PLOTS
    #plot the heatmap of genetic feature values
    p1 <- ggplot(matValue, aes(x=Sample, y=Var)) + 
      geom_tile(aes(fill=Value), color = "white") + #ghost white
      theme_bw()+
      scale_y_discrete(expand=c(0,0)) + 
      theme(axis.title.x = element_text( size=fontsize+4),
            axis.text.y=element_text(hjust=0, size=18, face="bold"), 
            axis.ticks=element_blank(),
            panel.border=element_rect(colour="gainsboro"),  
            plot.title=element_text(face="bold", size = 18, margin = margin(t = -5, b = 1)), 
            panel.background=element_blank(),
            panel.grid.major=element_blank(), 
            panel.grid.minor=element_blank()) + 
      xlab("Mutation status for each patient") + 
      ylab("") + 
      scale_fill_manual(name="Mutated", 
                              values=c(`10`= offwhite,  #WT
                                       `11`="#373A36", #Mutant
                                       `20`= offwhite, #LP
                                       `20.5`= "#707372", #IP
                                       `21` = "#A8A99E", #HP
                                       `30` = offwhite, #IGHV-U
                                       `31` = "#707372"), #IGHV-M
                                       guide="none") + 
            ggtitle(seaName)
    
    
    #Plot the bar plot on the left of the heatmap 
    barDF = data.frame(barValue, nm=factor(names(barValue),levels=names(barValue)))
    
    p2 <- ggplot(data=barDF, aes(x=nm, y=barValue)) + 
      geom_bar(stat="identity", 
               fill=ifelse(barValue<0,
                           palblues[6],palreds[8]), 
               colour="black", 
               size=0.3) +
      scale_x_discrete(expand=c(0,0.5))+ 
      scale_y_continuous(expand=c(0,0), n.breaks = 4)+ 
      coord_flip(ylim=c(-0.3,0.35))+ 
      theme(panel.grid.major=element_blank(), 
            panel.background=element_blank(), 
            axis.ticks.y = element_blank(),
            panel.grid.minor = element_blank(), 
            axis.text=element_text(size=fontsize, angle = 0, hjust = 0),
                axis.title = element_text(size=fontsize+4), 
            panel.border=element_blank()) +
      ylab("Size of coefficient") + 
      geom_vline(xintercept=c(0.5), 
                 color="black", 
                 size=0.6)
    
    #Plot the scatter plot of patient coefficient values under the heatmap
    scatterDF = data.frame(X=factor(names(betaValue), 
                                    levels=names(betaValue)), 
                           Y=unlist(betaValue))
    
    p3 <- ggplot(scatterDF, aes(x=X, y=Y)) + 
          geom_point(shape=21, 
                     fill="dimgrey", 
                     colour="#707372", #dark grey
                     size=1.2) + 
          theme_bw() +
          theme(panel.grid.minor=element_blank(), 
                panel.grid.major.x=element_blank(), 
                axis.ticks.x=element_blank(), 
                axis.text.y=element_text(size=fontsize),
                axis.title = element_text(size=fontsize+4),
                panel.border=element_rect(colour="dimgrey", size=0.1),
                panel.background=element_rect(fill="white")) +
  xlab(expression(paste("Patient-specific ", beta["int"])))
    
    
    #Assemble all the plots together

    # construct the gtable
    wdths = c(0.2, 1.5, 0.4, 1.3*ncol(matValue), 1.7, 0.2)
    hghts = c(0.3, 0.3, 0.0020*nrow(matValue), 0.2, 0.8, 0.3)*1.5
    gt = gtable(widths=unit(wdths, "in"), heights=unit(hghts, "in"))
    
    ## make grobs
    gg1 = ggplotGrob(p1)
    gg2 = ggplotGrob(p2)
    gg3 = ggplotGrob(p3)

    ## fill in the gtable
   
    #HEATMAP
    #5:1 = "PREDICTORS"
    gt = gtable_add_grob(gt, gtable_filter(gg1, "panel"), 3, 4) # add heatmap
    gt = gtable_add_grob(gt, gtable_filter(gg1, "panel"), 3, 4) #add legend
    gt = gtable_add_grob(gt, gtable_filter(gg1, "title"), 1, 4) #add title to plot
    gt = gtable_add_grob(gt, gtable_filter(gg1, "axis-l"), 3, 5) # variable names
    gt = gtable_add_grob(gt, gtable_filter(gg1, "xlab-b"), 2, 4) # axis title
    
    #BARPLOT
    gt = gtable_add_grob(gt, gtable_filter(gg2, "panel"), 3, 2) # add barplot
    gt = gtable_add_grob(gt, gtable_filter(gg2, "axis-b"), 4, 2) # y axis for barplot
    gt = gtable_add_grob(gt, gtable_filter(gg2, "xlab-b"), 2, 2) # y lab for barplot

    
    #SCATTER PLOT
    gt = gtable_add_grob(gt, gtable_filter(gg3, "panel"), 5, 4) # add scatterplot
    gt = gtable_add_grob(gt, gtable_filter(gg3, "xlab-b"), 6, 4) # x label for scatter plot
    gt = gtable_add_grob(gt, gtable_filter(gg3, "axis-l"), 5, 3) #  axis for scatter plot
    
   

    
    #plot
    plotList[[seaName]] <- gt
  }
  return(plotList)
}

makelegends <- function (legendFor, colors) 
{
    x = NULL
    y = NULL
    colors = colors[names(colors) %in% legendFor]
    nleg = length(colors)
    
    #edit these widths to change alignment with lasso plots in patchwork code
    wdths = c(0.4,2,2,2,1.5)
    hghts = c(2)
    
    gtl = gtable(widths=unit(wdths, "in"), heights=unit(hghts, "in"))
    n = 2
    if ("M" %in% names(colors)) {
        Mgg = ggplot(data = data.frame(x = 1, 
                                       y = factor(c("LP", "IP", "HP"), 
                                                  levels = c("LP", "IP", "HP"))), 
                     aes(x = x, y = y, fill = y)) + 
              geom_tile() + 
              scale_fill_manual(name = "Methylation cluster", 
                                values = setNames(colors[["M"]], 
                                                  nm = c("LP", "IP","HP"))) + 
              theme(legend.title = element_text(size = 12), 
                    legend.text = element_text(size = 12))
        
        gtl = gtable_add_grob(gtl, gtable_filter(ggplotGrob(Mgg), "guide-box"), 1, n)
        n = n + 1
    }
    
    if ("I" %in% names(colors)) {
        Igg = ggplot(data = data.frame(x = 1, y = factor(c("Unmutated", 
            "Mutated"), levels = c("Unmutated", "Mutated"))), 
            aes(x = x, y = y, fill = y)) + geom_tile() + scale_fill_manual(name = "IGHV", 
            values = setNames(colors[["I"]], nm = c("Unmutated", "Mutated"))) + 
            theme(legend.title = element_text(size = 12), 
                  legend.text = element_text(size = 12))
        gtl = gtable_add_grob(gtl, gtable_filter(ggplotGrob(Igg), 
            "guide-box"), 1, n)
        n = n + 1
    }
    
    if ("G" %in% names(colors)) {
        Ggg = ggplot(data = data.frame(x = 1, y = factor(c("Wild Type", 
            "Mutated"), levels = c("Wild Type", "Mutated"))), 
            aes(x = x, y = y, fill = y)) + geom_tile() + scale_fill_manual(name = "Gene", 
            values = setNames(colors[["G"]], nm = c("Wild Type", "Mutated"))) + 
            theme(legend.title = element_text(size = 12), 
                  legend.text = element_text(size = 12))
        gtl = gtable_add_grob(gtl, gtable_filter(ggplotGrob(Ggg), 
            "guide-box"), 1, n)
        n = n + 1
    }
    
    return(list(plot = gtl, width = sum(wdths), height = sum(hghts)))
}


```


Process data
```{r processData06}
# List of Cytokines, Drugs and treatments 
  ##Drugs
  thedrugs <- unique(df$Drug) 

  ##Cytokines
  thecytokines <- unique(df$Cytokine) 

  #Treatments
  #drugs
  drugtreatments <- list()
  drugtreatments <- 
    lapply(thedrugs, function(x){
      paste("treatment_drug", x, sep='')
      })
  
  
  #cytokines
  cytokinetreatments <- list()
  cytokinetreatments <- 
    lapply(thecytokines, function(y){
      paste("treatment_cytokine", y, sep='')
      })

df_complete <- df 

# Tidy up data frame to use in linear model
df <- 
  dplyr::filter(df,
                #use high concentrations only
                Drug_Concentration %in% c("High","None")) %>% 
  #select columns for linear model
  dplyr::select(PatientID, Drug, Cytokine, Log) %>%  
        
  #rename columns 
  plyr::rename(c("Drug"="treatment_drug", "Cytokine"="treatment_cytokine", "Log"="Viability"))


```


## Mapping drug - stimulus interactions
### Linear modelling quantifies interactions between drugs and stimuli

Principle

_Equation \@ref(eq:drugCytInt) quantifies how the viability with any combination can be predicted:_
    \begin{equation}
            log(V) = \beta_{drug}X_{drug} + \beta_{stimulus}X_{stimulus} + \beta_{int}X_{drug}X_{stimulus} + \epsilon
                                       (\#eq:drugCytInt)
    \end{equation}
    
_where $V$ is the viability with a given treatment,_ $\beta_{drug}$, $\beta_{stimulus}$ _and_ $\beta_{int}$ _are coefficients for the drug, stimulus and combinatorial terms and_ $X_{drug}$ _and_ $X_{stimulus}$  _are indicator variables (0 or 1) for the presence or absence of a drug/stimulus._ $\epsilon$ _is the vector of model residuals._
    
Different types of interaction     
(Figure \@ref(fig:drugStimulusDummyPLots)). 

(ref:drugStimulusDummyPLots)  _Figure from Bruch & Giles et al. 2021._ Method \@ref(id)

```{r drugStimulusDummyPLots, fig.cap='(ref:drugStimulusDummyPLots)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}


dummydata <-
  as.data.frame(list(c(1,2,3,4),
                     c(0,0,0,0),
                     c(-0.6,-0.6,0,-0.1), 
                     c(0.6,0.6,0.1,-0.1), 
                     c(0.5,-0.5,0.8,-0.8)), 
                col.names=c("plot",
                            "coord1", 
                            "coord2", 
                            "coord3", 
                            "coord4")) %>% 
  as_tibble() %>% 
  
  mutate(expected = coord2 + coord3, measured = coord4) %>% 
  
  pivot_longer(cols = coord1:measured, names_to = c("plot"), values_to="y.value", names_repair="unique" ) %>% 
  
  dplyr::rename(Plot = plot...1, x.value = plot...2) %>% 
  
  mutate(x.value = factor(x.value, levels = c("coord1", "coord2", "coord3", "coord4", "expected", "measured"))) %>% 
  
  #add annotation of interaction type
  mutate(Plot = factor(Plot, 
                       labels = c("stimulus inhibits drug", 
                                  "drug inhibits stimulus", 
                                  "synergistic pro-survival effect", 
                                  "synergistic toxicity")))


gg_dummy =
  #for each interaction type, make a dummy plot
  lapply(unique(dummydata$Plot), function(i){
    
    tmp <- dplyr::filter(dummydata, Plot==i )
    
    #get value for interaction
    interaction = 
      unlist(dplyr::select(dplyr::filter(tmp, x.value=="measured"), y.value)) - 
      unlist(dplyr::select(dplyr::filter(tmp,x.value=="expected"), y.value))
    
    #plot dummy plot
    gg_dummy <- 
      ggplot(dplyr::filter(tmp, grepl('coord', x.value)), 
             aes(x=x.value, y=y.value, group=Plot))+
  
  geom_hline(yintercept = 0, linetype=2, color="black")+
  geom_point(colour=ifelse(interaction > 0, "#A6093D","#003DA5"), size=1)+
  geom_line(colour=ifelse(interaction > 0, "#A6093D","#003DA5"), size=1)+
  
  geom_segment(data=dplyr::filter(tmp, grepl('expected', x.value)), aes(x = 3.5, y = y.value, xend = 4.5, yend = y.value), color="blue", size=2) +
  geom_segment(data=dplyr::filter(tmp, grepl('measured', x.value)), aes(x = 3.5, y = y.value, xend = 4.5, yend = y.value), color="black", size=2)+
  
  theme_bw() +
  t1 +
  xlab("") + 
  ylab("Log(Viability)") +
  coord_cartesian(ylim = c(-1, 1)) +
  scale_x_discrete(labels=c( "coord1"="DMSO", "coord2"="Drug","coord3"="Stimulus","coord4"="Drug +\nStimulus")) +
    theme(axis.ticks.x= element_blank(),axis.ticks.y = element_blank(),
          plot.tag=element_text(size = 20)) +
      
  scale_y_continuous(breaks = c(-1,0, 1))

})


#construct plot

wrap_elements(
  
  gg_dummy[[1]] + 
  labs(title=expression(paste("Positive ", beta["int"] ))) +
  ylab("Antagonism")+

  gg_dummy[[2]] +
  labs(title=expression(paste("Negative ", beta["int"] )))+

  gg_dummy[[3]] +
  ylab("Synergism") +
    
  gg_dummy[[4]]+
  
  plot_layout(nrow=2) +
  plot_annotation(tag_levels = "I")
)





```
```{r drugStimulusDummyPLots, fig.cap='(ref:drugStimulusDummyPLots)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}


dummydata <-
  as.data.frame(list(c(1,2,3,4),
                     c(0,0,0,0),
                     c(-0.6,-0.6,0,-0.1), 
                     c(0.6,0.6,0.1,-0.1), 
                     c(0.5,-0.5,0.8,-0.8)), 
                col.names=c("plot",
                            "coord1", 
                            "coord2", 
                            "coord3", 
                            "coord4")) %>% 
  as_tibble() %>% 
  
  mutate(expected = coord2 + coord3, measured = coord4) %>% 
  
  pivot_longer(cols = coord1:measured, names_to = c("plot"), values_to="y.value", names_repair="unique" ) %>% 
  
  dplyr::rename(Plot = plot...1, x.value = plot...2) %>% 
  
  mutate(x.value = factor(x.value, levels = c("coord1", "coord2", "coord3", "coord4", "expected", "measured"))) %>% 
  
  #add annotation of interaction type
  mutate(Plot = factor(Plot, 
                       labels = c("stimulus inhibits drug", 
                                  "drug inhibits stimulus", 
                                  "synergistic pro-survival effect", 
                                  "synergistic toxicity"))) %>%
  mutate(type = ifelse(Plot %in% c("stimulus inhibits drug", "drug inhibits stimulus"), "antagonism", "synergism")) %>%
  mutate(direction = ifelse(Plot %in% c("stimulus inhibits drug", "synergistic toxicity"), "negative", "positive"))

#
labs_type <- c("Antagonism", "Synergism")
names(labs_type) <- c("antagonism", "synergism")

labs_size <- c("Postive Bint", "Negative Bint")
names(labs_size) <- c("positive", "negative")


    
    
    #plot dummy plot
ggplot(dplyr::filter(dummydata, grepl('coord', x.value)), 
       aes(x=x.value, y=y.value, group = Plot))+
  geom_hline(yintercept = 0, linetype=2, color="black")+
  geom_point( size=1)+
  geom_line( size=1)+
  facet_grid(type~direction, labeller = labeller(type = labs_type, direction = labs_size))+
  geom_segment(data=dplyr::filter(dummydata, grepl('expected', x.value)), aes(x = 3.5, y = y.value, xend = 4.5, yend = y.value), color="blue", size=2) +
  geom_segment(data=dplyr::filter(dummydata, grepl('measured', x.value)), aes(x = 3.5, y = y.value, xend = 4.5, yend = y.value), color="black", size=2)+
  theme_bw() +
  t1 +
  xlab("") + 
  ylab("Log(Viability)") +
  coord_cartesian(ylim = c(-1, 1)) +
  scale_x_discrete(labels=c( "coord1"="DMSO", "coord2"="Drug","coord3"="Stimulus","coord4"="Drug +\nStimulus")) +
    theme(axis.ticks.x= element_blank(),axis.ticks.y = element_blank(),
          plot.tag=element_text(size = 20)) +
      
  scale_y_continuous(breaks = c(-1,0, 1))



#put colour of interaction back in: 
gg_dummy =
  #for each interaction type, make a dummy plot
  lapply(unique(dummydata$Plot), function(i){
    
    tmp <- dplyr::filter(dummydata, Plot==i )
    
    #get value for interaction
    interaction = 
      unlist(dplyr::select(dplyr::filter(tmp, x.value=="measured"), y.value)) - 
      unlist(dplyr::select(dplyr::filter(tmp,x.value=="expected"), y.value))
    
    
  #sort after ^  

#colour=ifelse(interaction > 0, "#A6093D","#003DA5"),
#colour=ifelse(interaction > 0, "#A6093D","#003DA5"),
#construct plot




```
Wauntify how many fit into each category    
(Figure \@ref(fig:barPlotInteractions)). 

(ref:barPlotInteractions)  _Figure from Bruch & Giles et al. 2021._ Method \@ref(id)

```{r barPlotInteractions, fig.cap='(ref:barPlotInteractions)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

```


### Linear modelling identifies drug - stimulus interactions of potential clinical importance  {#ID}
SHow all actual interactions
(Figure \@ref(fig:drugStimulusIntMapt)). 

(ref:drugStimulusIntMapt)  _Figure from Bruch & Giles et al. 2021._ Method \@ref(id)

```{r drugStimulusIntMapt, fig.cap='(ref:drugStimulusIntMapt)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

```

Discuss specific examples 
(find two replacements for ibr and ifn gamma)
Ibrutinib + IL4 Ibrutinib + IFNgamma, Ralimetinib + IFNgamma, Ibrutinib + Resiquimod  - show in grid graphic 
(Figure \@ref(fig:drugStimulusIntEx)). 

(ref:drugStimulusIntEx)  _Figure from Bruch & Giles et al. 2021._ Method \@ref(id)

```{r drugStimulusIntEx, fig.cap='(ref:drugStimulusIntEx)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

```


### IL4 and Interferon gamma induce resistance to Ibrutinib
(Figure \@ref(fig:IbrRes)). 

(ref:IbrRes)  _Figure from Bruch & Giles et al. 2021._ Method \@ref(id)

```{r IbrRes, fig.cap='(ref:IbrRes)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

```

## Genetic modulators of drug response 
### Univariate analysis identifies genetic modualtors of drug response

(Figure \@ref(fig:drugGeneIntMap)). 

(ref:drugGeneIntMap)  _Figure from Bruch & Giles et al. 2021._ Method \@ref(id)

```{r drugGeneIntMap, fig.cap='(ref:drugGeneIntMap)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

```

### Multivariate analysis identifies genetic modualtors of drug response

(Figure \@ref(fig:drugGeneIntMulti)). 

(ref:drugGeneIntMulti)  _Figure from Bruch & Giles et al. 2021._ Method \@ref(id)

```{r drugGeneIntMulti, fig.cap='(ref:drugGeneIntMulti)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

```


### Specific examples 
(Figure \@ref(fig:drugGeneIntEx)). 

(ref:drugGeneIntEx)  _Figure from Bruch & Giles et al. 2021._ Method \@ref(id)

```{r drugGeneIntEx, fig.cap='(ref:drugGeneIntEx)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

```
in a grid: 
IBET-762 and trisomy 12
Nutlin-3a and trisomy 12 and gene dosage effect of MDM2 in trisomy12 CLL
Correlation of IBET-762 and Nutlin-3a
Ralimetinib and trisomy 12

## The modulatory effect of mutations on drug - stimuli interactions: 
### Fitting patient-specific linear models identfies drug - stimulus - gene interactions 
(Figure \@ref(fig:drugCytGeneIntMap)). 

(ref:drugCytGeneIntMap)  _Figure from Bruch & Giles et al. 2021._ Method \@ref(id)

```{r drugCytGeneIntMap, fig.cap='(ref:drugCytGeneIntMap)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

#Process data 

#only high concentrations of drugs
df <- 
  dplyr::filter(df_complete, Drug_Concentration %in% c("None","High")) %>% 
  dplyr::select(PatientID, Drug, Cytokine, DCK, Log)

#rename columns
df <- 
  plyr::rename(df, 
               c("Drug"="treatment_drug", 
                 "Cytokine"="treatment_cytokine", 
                 "Log"="Viability"))
    
#lists for drug and cytokine treatments
thecytokines <- unique(df$treatment_cytokine) %>% setdiff("No Cytokine")
thedrugs <- unique(df$treatment_drug) %>% setdiff("DMSO")

```

### Patient - specific drug - stimulus interactions may have clinical significance
(Figure \@ref(fig:drugCytGeneIntEx)). 

(ref:drugCytGeneIntEx)  _Figure from Bruch & Giles et al. 2021._ Method \@ref(id)

```{r drugCytGeneIntEx, fig.cap='(ref:drugCytGeneIntEx)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

```

Responses to chemotherapeutic drugs in the context of TLR stimulation are modulated by del11q and ATM mutations,
Trisomy12 enhances pro-survival synergy between Ralimetinib + IFNgamma
Ralimetinib treatment increases the pro-survival effect of IL4, and this is further enhanced in NOTCH1 and Ras/Raf mutants
Fludarabine and TLR
Ibrutinib and IL4
Discuss examples in drug- cyt -gene examples script 

## Using stimuli responses in combination with genetic profiles to predict in vivo drug response 
Can stimuli + genes predict drug response - this would be useful in the clinic. Especially if its just TLR/IL4 response, are specific signals useful?

(Figure \@ref(fig:drugCytGeneIntClinic)). 

(ref:drugCytGeneIntClinic) Method \@ref(id)

```{r drugCytGeneIntClinic, fig.cap='(ref:drugCytGeneIntClinic)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

```

## Summary 

## Discussion 

## Contirbutions Statement

