# Molecular and microenvironmental modulators of drug response in CLL
So far, we have observed that the panel of stimuli have a range of ects on Cll vibaliyt, some higlhy potent. Form this analysis, IL4 and TLR pathways emerge as kye modualtors of CLL biology, in some genetic backgrounds hguley incresaing CLL viability. We have also observed hos these signals can be modualted by genetic features, underlying the fact that cancer biology is integratively influenec by cll-intrinsc and cell-extrisnic feautres. Trisomy 12 and IGHV status emerge as key modualtors that impact on resposnes to exernal signals. 

The microenvironment is known to reduce treatment efficacy within the protective niche [REFERENCE Ahn IE, Farooqui MZH, Tian X, et al. Depth and durability of response to ibrutinib in CLL: 5-year follow-up of a phase 2 study. Blood. 2018;131(21):2357â€“2366.
Abstract/FREE Full TextGoogle Scholar]. Guided by this obesrvation, 

In the second part of this chapter, we intergrate all three axes of the dataset, to ivnesrgatied how drug - microenivronment interactions can be further modulated by genetic features. 

Outline method 

INtor sentence 




Tasks for chapter
Sort figures [] 
Write all text 
Reread and check story and narrative
Reference
Credit and contirbutiosn statement 
Dump in disucssion and intro as I go 
Do a intro, discussion and methods dump 

Intro text 

```{r, echo = FALSE}

knitr::opts_chunk$set(
   echo = FALSE, 
   message = FALSE, 
   warning = FALSE,
   fig.align="center"
)

```

```{r setup06, echo = FALSE}

#libraries
library(patchwork)
library(ggplot2)
library(data.table)
library(magrittr)
library(pheatmap)
library(tidyr)
library(dplyr)
library(plyr)
library(ggpubr)
library(ggbeeswarm)
library(gtable)
library(glmnet)
library(RColorBrewer)
library(gridExtra)
library(cowplot)
library(tidyverse)

```

```{r loadData06, eval = TRUE}

#data("df", "patMeta", "LDT", "survT", "dds_smp")
#df_complete <- df

#Data
#df: tibble containing all screening viability data
load("data/df.RData")
df_complete <- df

#patMeta: tibble containing all patient genetic data
load("data/patMeta.RData")


#data for drug multivariate models 
#lasso coefficients
load("data/coefficients_lasso.RData") #from shiny app data
#gene Matrix
load("data/geneMatrix.RData") #from shiny app data
#viability Matrix
load("data/viabMatrix.RData") #from shiny app data


#data for drug - cyt - gene int analysis
#coefficients from lasso - linear modelling
load("data/coeff_mat_drug_cyt_gene_int.RData") #from figure 6 data script 

#predictor profiles 
load("data/Lasso_Plots_Fig6.RData") #from supplemental data




```

```{r themes_functions06, echo = FALSE, eval = TRUE}

### ggplot themes
fontsize = 11

## theme for ggplots
t1 <- 
  theme(                              
  plot.background = element_blank(), 
  panel.grid.major = element_line(),
  panel.grid.major.x = element_line(linetype = "dotted", colour = "grey"),
  panel.grid.minor = element_blank(), 
  panel.border = element_blank(), 
  panel.background = element_blank(),
  axis.line = element_line(size=.4),
  axis.line.x = element_line(),
  axis.line.y = element_line(),
  axis.text.x  = element_text(angle=90, size=11, hjust = 1, vjust = 0.4),
  axis.text.y = element_text(size = 11),
  axis.ticks.length = unit(0.3,"cm"),
  axis.title.x = element_text(face="bold", size=13), 
  axis.title.y = element_text(face="bold", size=13),
  plot.title = element_text(face="bold", size=14, hjust = 0.5),
  strip.text = element_text(size = fontsize)
)

t2 <- t1+
  theme( axis.text.x  = element_text(angle=0, size=11, hjust = 0.5, vjust = 1))

## theme for legends
t.leg <-  theme(legend.title = element_text(face='bold', 
                                            hjust = 1, size=11),
                legend.key = element_blank(),
                legend.text = element_text(size=11),
                legend.background = element_rect(color = "black"))



### Set colour palettes

#For Categorical: 
colors <- c("#A1BE1F", #green
            "#F4C61F", #yellow
            "#734595", #purple
            "#D41645", #red
            "#3B6FB6", #blue
            "#B65417", #orange
            "#E2E868", #light green
            "#CBA3D8", #light purple
            "#E58F9E", #light purple
            "#8BB8E8", #light blue
            "#F49E17", #light orange
            "#303030", #black
            "#A8A99E", #grey
            "#007B53") #dark green



#For Divergent: 
Divergent <- c("#003DA5", "#2055B0", "#406EBC", "#6086C7", "#809ED2", "#9FB6DD", "#BFCFE9", "#DFE7F4", "white", "white", "white","#F4E0E7", "#E9C2CF", "#DEA3B6", "#D3849E", "#C76586", "#BC476E", "#B12855", "#A6093D")

#for negatives only: 
palblues <- c("#003DA5", "#2055B0", "#406EBC", "#6086C7", "#809ED2", "#9FB6DD", "#BFCFE9", "#DFE7F4")

#for positives only:
palreds <- c("#F4E0E7", "#E9C2CF", "#DEA3B6", "#D3849E", "#C76586", "#BC476E", "#B12855", "#A6093D")

#For mutations: 
Mutant <- c("#b5b5b5","#373A36")
Sex <- c("#707372","#D0D0CE")
IGHV <- c("#373A36","#D0D0CE")
Methylation_cluster <- c("#373A36","#A8A99E","#D0D0CE")

#For drugs: 
drugpal <- c("#734595", "#CBA3D8") #purples

#For cytokines: 
cytpal <- c("#F49E17", "#EFC06E") #yellows


#neutral
offwhite <- "#f8f8ff"
lightergrey <- "#D0D0CE"
darkergrey <- "#707372"


#Function for multi-variant regression
runGlm06 <- function(X, y, method = "lasso", repeats = 20, folds = 3) {
  #set up objects to store results
  modelList <- list()
  lambdaList <- c()
  varExplain <- c()
  coefMat <- matrix(NA, ncol(X), repeats)
  rownames(coefMat) <- colnames(X)

  #set alpha
  if (method == "lasso"){
    alpha = 1
  } else if (method == "ridge") {
    alpha = 0
  }
  
  #for the set number of repeats, run the regression
  for (i in seq(repeats)) {
    if (ncol(X) > 2) {
      #run cross validated generalised linear model with given parameters
      res <- cv.glmnet(X,y, type.measure = "mse", family="gaussian", 
                       nfolds = folds, alpha = alpha, standardize = FALSE)
      
      #store lamdas and min lambda value
      lambdaList <- c(lambdaList, res$lambda.min)
      
      #store result of cv.glmnet
      modelList[[i]] <- res
      
      #get coefficents with min lambda value 
      coefModel <- coef(res, s = "lambda.min")[-1] #remove intercept row
      
      #store coefficients for this repeat
      coefMat[,i] <- coefModel
      
      #calculate variance explained
      if(sum(coefModel !=0)){
      y.pred <- predict(res, s = "lambda.min", newx = X)
      #if there are no predictors, all y.pred will be the same so its not possible to calculate variance explained because the SD is 0
      varExp <- cor(as.vector(y),as.vector(y.pred))^2
      }else{ varExp <- NA}
      varExplain[i] <- ifelse(is.na(varExp), 0, varExp) 
      
     
      
    } else {
      fitlm<-lm(y~., data.frame(X))
      varExp <- summary(fitlm)$r.squared
      varExplain <- c(varExplain, varExp)
      
    }

  }
  #store all results 
  list(modelList = modelList, lambdaList = lambdaList, varExplain = varExplain, coefMat = coefMat)
}


lassoPlot06 <- function(lassoOut, geneMatrix, betaMatrix, freqCut = 1, coefCut = 0.01) {
  #object to hold all plots
  plotList <- list()
  
  #for each drug - stimuli combination, run the following:
  for (seaName in names(lassoOut)) {
    ###FOR THE BAR PLOT
    #extract mean coefficients for each drug - stimuli combination
    barValue <- rowMeans(lassoOut[[seaName]]$coefMat)
    #extract proportion of repeats for which each coefficient is significant
    freqValue <- rowMeans(abs(sign(lassoOut[[seaName]]$coefMat)))
    #filter out coefficients that don't meet freqCut and coefCut thresholds
    barValue <- barValue[abs(barValue) >= coefCut & freqValue >= freqCut] 
    #arrange the bar values in numerical order
    barValue <- barValue[order(barValue)]
    #if there are no sig coefficients, don't plot
    if(length(barValue) == 0) {
      plotList[[seaName]] <- NA
      next
    }
    
   
    ###FOR THE HEATMAP AND SCATTER PLOT
    #get feature matrix and response matrix to plot
    allData <- geneMatrix
    betaValue <- unlist(betaMatrix[seaName,])
    
    #get feature matrix for features with significant coefficients only
    tabValue <- allData[, names(barValue),drop=FALSE]
    ord <- order(betaValue)
    betaValue <- betaValue[ord]
    tabValue <- tabValue[ord, ,drop=FALSE]
    sampleIDs <- rownames(tabValue)
    tabValue <- as_tibble(tabValue)
    tabValue$Sample <- sampleIDs
    
    #annotate features as mutations, methylation cluster or IGHV, and apply different scaling     so that different colours can be used in plotting 
    
    #for mutations:
    matValue <- gather(tabValue, key = "Var",value = "Value", -Sample)
    matValue$Type <- "mut"
    
    #for methylation cluster
    matValue$Type[grep("Methylation",matValue$Var)] <- "meth"
    
    #for IGHV status
    matValue$Type[grep("IGHV.status",matValue$Var)] <- "ighv"
    
    #change the scale of the value so that IGHV, Methylation and Mutation do not overlap
    matValue[matValue$Type == "mut",]$Value = matValue[matValue$Type == "mut",]$Value + 10
    matValue[matValue$Type == "meth",]$Value = matValue[matValue$Type == "meth",]$Value + 20
    matValue[matValue$Type == "ighv",]$Value = matValue[matValue$Type == "ighv",]$Value + 30
    
    #change continous to catagorical
    matValue$Value <- factor(matValue$Value,levels = sort(unique(matValue$Value)))
    
    #arrange order of feature rows and columns in heatmap
    #heatmap rows should align with order of genetic coefficients
    matValue$Var <- factor(matValue$Var, levels = names(barValue))
    
    #sample columns should be ascending order according to value of coefficient of each patient, so that heatmap aligns with scatter plot below
    matValue$Sample <- factor(matValue$Sample, levels = names(betaValue))
    
    #update labels, keep factor levels
    matValue$Var <- revalue(matValue$Var, c("IGHV.status" = "IGHV status", "del11q" = "del(11q)", "del13q" = "del(13q)", "del17p" = "del(17p)", "trisomy12" = "trisomy 12")) 
    
    #MAKE PLOTS
    #plot the heatmap of genetic feature values
    p1 <- ggplot(matValue, aes(x=Sample, y=Var)) + 
      geom_tile(aes(fill=Value), color = "white") + #ghost white
      theme_bw()+
      scale_y_discrete(expand=c(0,0)) + 
      theme(axis.title.x = element_text( size=fontsize+4),
            axis.text.y=element_text(hjust=0, size=18, face="bold"), 
            axis.ticks=element_blank(),
            panel.border=element_rect(colour="gainsboro"),  
            plot.title=element_text(face="bold", size = 18, margin = margin(t = -5, b = 1)), 
            panel.background=element_blank(),
            panel.grid.major=element_blank(), 
            panel.grid.minor=element_blank()) + 
      xlab("Mutation status for each patient") + 
      ylab("") + 
      scale_fill_manual(name="Mutated", 
                              values=c(`10`= offwhite,  #WT
                                       `11`="#373A36", #Mutant
                                       `20`= offwhite, #LP
                                       `20.5`= "#707372", #IP
                                       `21` = "#A8A99E", #HP
                                       `30` = offwhite, #IGHV-U
                                       `31` = "#707372"), #IGHV-M
                                       guide="none") + 
            ggtitle(seaName)
    
    
    #Plot the bar plot on the left of the heatmap 
    barDF = data.frame(barValue, nm=factor(names(barValue),levels=names(barValue)))
    
    p2 <- ggplot(data=barDF, aes(x=nm, y=barValue)) + 
      geom_bar(stat="identity", 
               fill=ifelse(barValue<0,
                           palblues[6],palreds[8]), 
               colour="black", 
               size=0.3) +
      scale_x_discrete(expand=c(0,0.5))+ 
      scale_y_continuous(expand=c(0,0), n.breaks = 4)+ 
      coord_flip(ylim=c(-0.3,0.35))+ 
      theme(panel.grid.major=element_blank(), 
            panel.background=element_blank(), 
            axis.ticks.y = element_blank(),
            panel.grid.minor = element_blank(), 
            axis.text=element_text(size=fontsize, angle = 0, hjust = 0),
                axis.title = element_text(size=fontsize+4), 
            panel.border=element_blank()) +
      ylab("Size of coefficient") + 
      geom_vline(xintercept=c(0.5), 
                 color="black", 
                 size=0.6)
    
    #Plot the scatter plot of patient coefficient values under the heatmap
    scatterDF = data.frame(X=factor(names(betaValue), 
                                    levels=names(betaValue)), 
                           Y=unlist(betaValue))
    
    p3 <- ggplot(scatterDF, aes(x=X, y=Y)) + 
          geom_point(shape=21, 
                     fill="dimgrey", 
                     colour="#707372", #dark grey
                     size=1.2) + 
          theme_bw() +
          theme(panel.grid.minor=element_blank(), 
                panel.grid.major.x=element_blank(), 
                axis.ticks.x=element_blank(), 
                axis.text.y=element_text(size=fontsize),
                axis.title = element_text(size=fontsize+4),
                panel.border=element_rect(colour="dimgrey", size=0.1),
                panel.background=element_rect(fill="white")) +
  xlab(expression(paste("Patient-specific ", beta["int"])))
    
    
    #Assemble all the plots together

    # construct the gtable
    wdths = c(0.2, 1.5, 0.4, 1.3*ncol(matValue), 1.7, 0.2)
    hghts = c(0.3, 0.3, 0.0020*nrow(matValue), 0.2, 0.8, 0.3)*1.5
    gt = gtable(widths=unit(wdths, "in"), heights=unit(hghts, "in"))
    
    ## make grobs
    gg1 = ggplotGrob(p1)
    gg2 = ggplotGrob(p2)
    gg3 = ggplotGrob(p3)

    ## fill in the gtable
   
    #HEATMAP
    #5:1 = "PREDICTORS"
    gt = gtable_add_grob(gt, gtable_filter(gg1, "panel"), 3, 4) # add heatmap
    gt = gtable_add_grob(gt, gtable_filter(gg1, "panel"), 3, 4) #add legend
    gt = gtable_add_grob(gt, gtable_filter(gg1, "title"), 1, 4) #add title to plot
    gt = gtable_add_grob(gt, gtable_filter(gg1, "axis-l"), 3, 5) # variable names
    gt = gtable_add_grob(gt, gtable_filter(gg1, "xlab-b"), 2, 4) # axis title
    
    #BARPLOT
    gt = gtable_add_grob(gt, gtable_filter(gg2, "panel"), 3, 2) # add barplot
    gt = gtable_add_grob(gt, gtable_filter(gg2, "axis-b"), 4, 2) # y axis for barplot
    gt = gtable_add_grob(gt, gtable_filter(gg2, "xlab-b"), 2, 2) # y lab for barplot

    
    #SCATTER PLOT
    gt = gtable_add_grob(gt, gtable_filter(gg3, "panel"), 5, 4) # add scatterplot
    gt = gtable_add_grob(gt, gtable_filter(gg3, "xlab-b"), 6, 4) # x label for scatter plot
    gt = gtable_add_grob(gt, gtable_filter(gg3, "axis-l"), 5, 3) #  axis for scatter plot
    
   

    
    #plot
    plotList[[seaName]] <- gt
  }
  return(plotList)
}

makelegends <- function (legendFor, colors) 
{
    x = NULL
    y = NULL
    colors = colors[names(colors) %in% legendFor]
    nleg = length(colors)
    
    #edit these widths to change alignment with lasso plots in patchwork code
    wdths = c(0.4,2,2,2,1.5)
    hghts = c(2)
    
    gtl = gtable(widths=unit(wdths, "in"), heights=unit(hghts, "in"))
    n = 2
    if ("M" %in% names(colors)) {
        Mgg = ggplot(data = data.frame(x = 1, 
                                       y = factor(c("LP", "IP", "HP"), 
                                                  levels = c("LP", "IP", "HP"))), 
                     aes(x = x, y = y, fill = y)) + 
              geom_tile() + 
              scale_fill_manual(name = "Methylation cluster", 
                                values = setNames(colors[["M"]], 
                                                  nm = c("LP", "IP","HP"))) + 
              theme(legend.title = element_text(size = 12), 
                    legend.text = element_text(size = 12))
        
        gtl = gtable_add_grob(gtl, gtable_filter(ggplotGrob(Mgg), "guide-box"), 1, n)
        n = n + 1
    }
    
    if ("I" %in% names(colors)) {
        Igg = ggplot(data = data.frame(x = 1, y = factor(c("Unmutated", 
            "Mutated"), levels = c("Unmutated", "Mutated"))), 
            aes(x = x, y = y, fill = y)) + geom_tile() + scale_fill_manual(name = "IGHV", 
            values = setNames(colors[["I"]], nm = c("Unmutated", "Mutated"))) + 
            theme(legend.title = element_text(size = 12), 
                  legend.text = element_text(size = 12))
        gtl = gtable_add_grob(gtl, gtable_filter(ggplotGrob(Igg), 
            "guide-box"), 1, n)
        n = n + 1
    }
    
    if ("G" %in% names(colors)) {
        Ggg = ggplot(data = data.frame(x = 1, y = factor(c("Wild Type", 
            "Mutated"), levels = c("Wild Type", "Mutated"))), 
            aes(x = x, y = y, fill = y)) + geom_tile() + scale_fill_manual(name = "Gene", 
            values = setNames(colors[["G"]], nm = c("Wild Type", "Mutated"))) + 
            theme(legend.title = element_text(size = 12), 
                  legend.text = element_text(size = 12))
        gtl = gtable_add_grob(gtl, gtable_filter(ggplotGrob(Ggg), 
            "guide-box"), 1, n)
        n = n + 1
    }
    
    return(list(plot = gtl, width = sum(wdths), height = sum(hghts)))
}


```


Process data
```{r processData06}
# List of Cytokines, Drugs and treatments 
  ##Drugs
  thedrugs <- unique(df$Drug) 

  ##Cytokines
  thecytokines <- unique(df$Cytokine) 

  #Treatments
  #drugs
  drugtreatments <- list()
  drugtreatments <- 
    lapply(thedrugs, function(x){
      paste("treatment_drug", x, sep='')
      })
  
  
  #cytokines
  cytokinetreatments <- list()
  cytokinetreatments <- 
    lapply(thecytokines, function(y){
      paste("treatment_cytokine", y, sep='')
      })

df_complete <- df 

# Tidy up data frame to use in linear model
df <- 
  dplyr::filter(df,
                #use high concentrations only
                Drug_Concentration %in% c("High","None")) %>% 
  #select columns for linear model
  dplyr::select(PatientID, Drug, Cytokine, Log) %>%  
        
  #rename columns 
  plyr::rename(c("Drug"="treatment_drug", "Cytokine"="treatment_cytokine", "Log"="Viability"))


#patMeta
patMeta$treatment <- as.factor(patMeta$treatment)
```

```{r preWork06, message = FALSE}

# define the base level per treatment_type as the "no"-treatment
df$treatment_drug <- as.factor(df$treatment_drug)
df$treatment_cytokine <- as.factor(df$treatment_cytokine)

df$treatment_drug %<>% relevel("DMSO")
df$treatment_cytokine %<>% relevel("No Cytokine")


# inspect design matrix of interaction model
# model.matrix(~treatment_drug + treatment_cytokine + treatment_drug:treatment_cytokine, df)

# fit model with interaction
fit <- lm(Viability ~ treatment_drug * treatment_cytokine, df)



#extract p values

pvalues <- summary(fit)$coefficients[,4]

#order as a dataframe
pvaldf <- as.data.frame(as.matrix(pvalues)) %>% 
          setDT(keep.rownames = TRUE)
          
#rename columns
colnames(pvaldf) <- c("treatment", "pvalue")

#filter out single agent treatments
singletreatments <- c(drugtreatments, cytokinetreatments, "(Intercept)")

pvaldf <- dplyr::filter(pvaldf, !treatment %in% singletreatments)


pvaldf <- 
  pvaldf %>% 
  #remove "treatment" prefix
  mutate_at(vars(treatment), funs(as.character(gsub("treatment_drug", "", .)))) %>% 
  mutate_at(vars(treatment), funs(as.character(gsub("treatment_cytokine", "", .))))
  
  #split drug:cytokine into two columns
pvaldf <-   
  data.frame(pvaldf, do.call(rbind, strsplit(pvaldf$treatment, split = ":", fixed = TRUE)))

#renove treatment column
pvaldf <- pvaldf[,c("pvalue","X1", "X2")]

#rename columns
colnames(pvaldf) <- c("pvalue","drug", "Cytokine")



```

## Mapping drug - stimulus interactions
### Linear modelling quantifies interactions between drugs and stimuli

Principle

Equation \@ref(eq:drugCytInt) quantifies how the viability with any combination can be predicted:
    \begin{equation}
            log(V) = \beta_{drug}X_{drug} + \beta_{stimulus}X_{stimulus} + \beta_{int}X_{drug}X_{stimulus} + \epsilon
                                       (\#eq:drugCytInt)
    \end{equation}
    
_where $V$ is the viability with a given treatment,_ $\beta_{drug}$, $\beta_{stimulus}$ _and_ $\beta_{int}$ _are coefficients for the drug, stimulus and combinatorial terms and_ $X_{drug}$ _and_ $X_{stimulus}$  _are indicator variables (0 or 1) for the presence or absence of a drug/stimulus._ $\epsilon$ _is the vector of model residuals._
    
Different types of interaction     
(Figure \@ref(fig:drugStimulusDummyPLots)). 

(ref:drugStimulusDummyPLots) Graphical representation of the four drug - stimulus interaction categories. Categories are defined according to (i) the nature of the interaction (synergistic or antagonistic), and (ii) whether $\beta_{int}$ is positive or negative i.e. the viability with the combinatorial treatment is higher or lower than would be expected based on additive effects alone. x-axis shows treatment type, y-axis shows log transformed viability values. Red and blue lines depict representative treatment response patterns for given interaction type. To demonstrate the effect of each each interaction on the expected viability, blue and black horizontal lines represent the expected viability for combinatorial treatment based on additive effects alone (blue) and the actual viability accounting for additive effects and the interaction (black).  _Figure and caption from Bruch & Giles et al. 2021._ See Methods section \@ref(drug-stimulus-linear-model-method).


```{r drugStimulusDummyPLots, fig.cap='(ref:drugStimulusDummyPLots)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}


dummydata <-
  as.data.frame(list(c(1,2,3,4),
                     c(0,0,0,0),
                     c(-0.6,-0.6,0,-0.1), 
                     c(0.6,0.6,0.1,-0.1), 
                     c(0.5,-0.5,0.8,-0.8)), 
                col.names=c("plot",
                            "coord1", 
                            "coord2", 
                            "coord3", 
                            "coord4")) %>% 
  as_tibble() %>% 
  
  mutate(expected = coord2 + coord3, measured = coord4) %>% 
  
  pivot_longer(cols = coord1:measured, names_to = c("plot"), values_to="y.value", names_repair="unique" ) %>% 
  
  dplyr::rename(Plot = plot...1, x.value = plot...2) %>% 
  
  mutate(x.value = factor(x.value, levels = c("coord1", "coord2", "coord3", "coord4", "expected", "measured"))) %>% 
  
  #add annotation of interaction type
  mutate(Plot = factor(Plot, 
                       labels = c("I", 
                                  "II", 
                                  "III", 
                                  "IV"))) %>%
  mutate(type = ifelse(Plot %in% c("I", "II"), "antagonism", "synergism")) %>%
  mutate(direction = ifelse(Plot %in% c("I", "III"), "positive", "negative"))

my_labeller <- as_labeller(c("positive"="Positive~\u03B2[int]", "negative"="Negative~\u03B2[int]", "antagonism"="Antagonism", "synergism" = "Synergism"),
                           default = label_parsed)

dummydata$direction <- factor(dummydata$direction, levels = c("positive","negative"))
dummydata$type <- factor(dummydata$type, levels = c("antagonism","synergism"))
                              

#plot dummy plot
ggplot(dplyr::filter(dummydata, grepl('coord', x.value)), 
       aes(x=x.value, y=y.value, group = Plot, color = direction))+
  geom_hline(yintercept = 0, linetype=2, color="black")+
  geom_point(size=1)+
  geom_line(size=1)+
  facet_grid(type~direction, labeller = my_labeller )+
  geom_segment(data=dplyr::filter(dummydata, grepl('expected', x.value)), aes(x = 3.5, y = y.value, xend = 4.5, yend = y.value), color="blue", size=2) +
  geom_segment(data=dplyr::filter(dummydata, grepl('measured', x.value)), aes(x = 3.5, y = y.value, xend = 4.5, yend = y.value), color="black", size=2)+
  theme_bw() +
  t1 +
  xlab("") + 
  ylab("Log(Viability)") +
  coord_cartesian(ylim = c(-1, 1)) +
  scale_x_discrete(labels = c( "coord1" = "DMSO", "coord2" = "Drug","coord3" = "Stimulus","coord4" = "Drug +\nStimulus")) +
    theme(axis.ticks.x = element_blank(),axis.ticks.y = element_blank(),
          plot.tag=element_text(size = 20)) +
  scale_y_continuous(breaks = c(-1,0, 1)) +
  scale_color_manual(values = c("#A6093D","#003DA5"))  + theme(legend.position = "none") +
  geom_text(x = 1, y = 0.75, aes(label=Plot))



```

Quantify how many fit into each category    
(Figure \@ref(fig:barPlotInteractions)). 

(ref:barPlotInteractions) Histogram showing the number of drug-stimulus interactions within each category, where p value for $\beta_{int}$ is <0.05.  _Figure and caption from Bruch & Giles et al. 2021._ See Methods section \@ref(drug-stimulus-linear-model-method).

```{r barPlotInteractions, fig.cap='(ref:barPlotInteractions)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}


fit_as_df <- 
  fit$coefficients %>% 
  as.matrix() %>% 
  as.data.frame() %>%
  setDT(keep.rownames = TRUE) %>% 
  as_tibble() %>% 
  dplyr::rename(Condition = rn, comb_coefficient = V1) %>% 
  dplyr::filter(grepl("drug", Condition)&grepl("cytokine", Condition)) %>% 
  mutate(Condition = gsub("treatment_drug", "", Condition)) %>% 
  mutate(Condition = gsub("treatment_cytokine", "", Condition)) %>% 
  separate(Condition, c("Drug", "Cytokine"), sep=":")

drug_fit_as_df <- 
  fit$coefficients %>% 
  as.matrix() %>%
  as.data.frame() %>%
  setDT(keep.rownames = TRUE) %>% 
  as_tibble() %>% 
  dplyr::rename(Drug = rn, drug_coefficent=V1) %>% 
  dplyr::filter(grepl("drug", Drug)&!grepl("cytokine", Drug)) %>% 
  mutate(Drug = gsub("treatment_drug", "", Drug))


cyt_fit_as_df <-
  fit$coefficients %>% 
  as.matrix() %>% 
  as.data.frame() %>%
  setDT(keep.rownames = TRUE) %>% 
  as_tibble() %>% 
  dplyr::rename(Cytokine=rn, cyt_coefficent=V1) %>% 
  dplyr::filter(!grepl("drug", Cytokine)&grepl("cytokine", Cytokine)) %>% 
  mutate(Cytokine = gsub("treatment_cytokine", "", Cytokine))


fit_combinations <-
  left_join(fit_as_df, drug_fit_as_df, by = "Drug") %>% 
  left_join(cyt_fit_as_df, by = "Cytokine") %>% 
  mutate(predicted_coefficient=cyt_coefficent+drug_coefficent) %>% 
  mutate(obs_coef=predicted_coefficient+comb_coefficient) %>%
  left_join(mutate(pvaldf,drug=as.character(drug),Cytokine=as.character(Cytokine)), by=c("Drug"="drug", "Cytokine"="Cytokine")) %>% 
  dplyr::filter(pvalue<=0.05) %>%
  mutate(Category_Symbol=case_when(
    comb_coefficient>=0&(obs_coef<cyt_coefficent|obs_coef<drug_coefficent)~ "I",  
    comb_coefficient>=0&(obs_coef>cyt_coefficent&obs_coef>drug_coefficent)~ "III",
    comb_coefficient<0&(obs_coef>cyt_coefficent|obs_coef>drug_coefficent)~ "II",  
    comb_coefficient<0&(obs_coef<cyt_coefficent&obs_coef<drug_coefficent)~ "IV"
    )) %>% 
  mutate(Category=case_when(
    Category_Symbol=="I"~ "Positive Antagonistic",  
    Category_Symbol=="III"~ "Positive Synergistic",
    Category_Symbol=="II"~ "Negative Antagonistic",  
    Category_Symbol=="IV"~ "Negative Synergistic"
    )) %>% 
  mutate( Category=factor(Category, levels=c("Positive Antagonistic","Negative Antagonistic","Positive Synergistic","Negative Synergistic")))
  

fit_combinations %>% 
  dplyr::group_by(Category) %>% 
  tally() %>% 
  
ggplot(aes(x=Category, y=n, fill=Category))+
  geom_bar(width = 1, stat = "identity")+
  t1+
  guides(fill="none")+
  xlab("")+ ylab("Number of drug - stimulus pairs")+
  scale_fill_manual(values = c("#A6093D","#003DA5","#A6093D","#003DA5"))



```


### Linear modelling identifies drug - stimulus interactions of potential clinical importance  {#ID}
SHow all actual interactions
(Figure \@ref(fig:drugStimulusIntMapt)). 

(ref:drugStimulusIntMapt) Heatmap of all $\beta_{int}$ values for all drug - stimulus combinations where p for $\beta_{int}$ <0.05, annotated with interaction type. Scale indicates size and sign of $\beta_{int}$. Rows and columns clustered according to hierarchical clustering.  _Figure and caption from Bruch & Giles et al. 2021._ See Methods section \@ref(drug-stimulus-linear-model-method). 

```{r drugStimulusIntMapt, fig.cap='(ref:drugStimulusIntMapt)', message = FALSE,  echo = FALSE, fig.width=12, fig.height = 8, fig.align="center", out.width = '100%', dev = 'cairo_pdf'}

#extract beta interaction values
betavalues <- summary(fit)$coefficients[,1]

#create matrix of beta values 
betadf <- as.data.frame(as.matrix(betavalues)) %>%
          setDT(keep.rownames = TRUE)

#rename columns
colnames(betadf) <- c("treatment", "betavalue")

#remove beta values for drugs and cytokines alone
betadf <- dplyr::filter(betadf, !treatment %in% singletreatments)

#remove treatment prefix
betadf <- betadf %>% 
  mutate_at(vars(treatment), funs(as.character(gsub("treatment_drug", "", .)))) %>% 
  mutate_at(vars(treatment), funs(as.character(gsub("treatment_cytokine", "", .))))

#split drug:cytokine by colon
betadf <- data.frame(betadf, do.call(rbind, strsplit(betadf$treatment, split = ":", fixed = TRUE)))

#select columns of interest
betadf <- betadf[,c("betavalue","X1", "X2")]

#rename columns 
colnames(betadf) <- c("betavalue","drug", "Cytokine")

#set significance
a <- 0.05

#make a matrix of beta values, where  beta = 0 if corresponding p val is > than significance threshold 
pvalmat <- xtabs(pvalue~drug+Cytokine, data=pvaldf)
bvalmat <- xtabs(betavalue~drug+Cytokine, data=betadf)

#check in same order
bvalmat <- bvalmat[,colnames(pvalmat)] 
bvalmat <- bvalmat[rownames(pvalmat),]

bvalmat[pvalmat >= a] = 0

#transform matrix 
bvalmat <- t(bvalmat)

tree <- 
  pheatmap(bvalmat, silent = TRUE)


cyt_order <- rownames(bvalmat[tree$tree_row[["order"]],])

drug_order <- colnames(bvalmat[,tree$tree_col[["order"]]])



  #append p values
  left_join(betadf, pvaldf, by = c("drug", "Cytokine")) %>% 
  #make beta value 0 if p value is < sig
  mutate(betavalue = ifelse(pvalue<a, betavalue, 0)) %>% 
  #put drugs in order of clustered heatmap
  mutate(drug=factor(drug, levels = drug_order)) %>% 
  
  #put stimuli in order of clustered heatmap
  mutate(Cytokine=factor(Cytokine, levels = rev(cyt_order))) %>% 
  
  #make heatmap with ggplot
  ggplot(aes(x=drug, y=Cytokine))+
  
  geom_tile(aes(fill=betavalue),color = "grey")+
  #set colour scale
  scale_fill_gradientn(colors=c("#003DA5",  "white",  "#A6093D"), limits=c(-0.6,0.6))+
  #set theme
  t2+
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5),
        axis.title.x = element_blank(),
        axis.title.y = element_blank(),
        axis.ticks = element_blank(),
        axis.ticks.x =element_blank(),
        panel.background = element_blank(), 
        axis.line.x = element_blank(),
        axis.line.y = element_blank(),
        legend.key.height=unit(2.5, "cm"))+
  labs(fill = expression(paste(beta["int"], "-value")))+
  #add category
  geom_text(data = fit_combinations, aes(x=Drug, label=Category_Symbol) )+
  scale_y_discrete(labels=c("TGF-b1"="TGF-\u03B21", "sCD40L+IL-4"="sCD40L + IL4", "IL-1b"="IL1\u03B2", "IL-4"="IL4", "IL-6"="IL6","IL-15"="IL15","IL-10"="IL10", "IL-21"="IL21","IL-2"="IL2", "Interferon gamma"= "Interferon \u03B3", "SDF-1a"="SDF-1\u03B1"))
  


```

Discuss specific examples 
(find two replacements for ibr and ifn gamma)
Ibrutinib + IL4 Ibrutinib + IFNgamma, Ralimetinib + IFNgamma, Ibrutinib + Resiquimod  - show in grid graphic 
big up interferon gamma 

(Figure \@ref(fig:drugStimulusIntEx)). 

(ref:drugStimulusIntEx)  Examples of drug-stimulus interactions, for each category. Plots show log transformed viability values with each treatment, for all samples. Matching samples are linked across treatments. Black and blue horizontal lines indicate the predicted viability from the linear model for each single and combinatorial treatment. In combinatorial treatment, both the expected viability based on the additive effect of drug and stimulus (blue), and the viability with interaction (black) are shown, to indicate the impact of the interaction. (A+B) Ibrutinib, a clinically-used BTK inhibitor, is blocked by IL4 and IFN$\gamma$. (C) The JAK inhibitor pyridone-6 inhibits the pro-survival effect of sCD40L + IL4 stimulation. (D) The p38 inhibitor ralimetinib and IFN$\gamma$ show a synergistic pro-survival effect not seen in either single treatment. (E) TLR agonists, including CpG ODN (shown) increase sensitivity to BTK inhibition by ibrutinib, despite increasing viability in isolation. (F) Soluble anti-IgM sensitises CLL samples to HSP90 inhibition by luminespib. _Figure and caption from Bruch & Giles et al. 2021._ See Methods section \@ref(drug-stimulus-linear-model-method). 

```{r drugStimulusIntEx, fig.cap='(ref:drugStimulusIntEx)', message = FALSE,  echo = FALSE, fig.height=10, fig.width = 12, fig.align="center", out.width = '80%', dev = 'cairo_pdf'}


#get lists of drugs and cytokines without baseline treatments
thedrugs.only <- thedrugs %>% setdiff("DMSO") %>% sort()
thecytokines.only <- thecytokines %>% setdiff("No Cytokine")%>% sort()

# get a list of interactions for which p value of beta interaction is significant
sign_conditions <-
  pvaldf %>% 
  dplyr::filter(pvalue<=0.05) %>% 
  mutate(condition = paste0("treatment_drug:", 
                            drug, 
                            ":treatment_cytokine:",
                            Cytokine))


gg <- vector(mode="list", length=length(sign_conditions$condition))
names(gg) <- sign_conditions$condition



plotList <- 
lapply(names(gg), function(i){
  
  #preparations for the interaction plot
  drug <- dplyr::filter(sign_conditions, condition ==i)$drug
  drugTreat <- paste("treatment_drug", drug, sep='')
      
  cyt <- dplyr::filter(sign_conditions, condition ==i)$Cytokine
  cytokineTreat <-paste("treatment_cytokine", cyt, sep='')
      
      
  Category <- fit_combinations[which(fit_combinations$Drug==drug &
                               fit_combinations$Cytokine==cyt),]$Category
  
  #extract coefficients to plot

  ## baseline effect
  baseline <- fit$coefficients["(Intercept)"]

  ## single treatment effect for type_1 = "a"
  single_1 <- fit$coefficients[drugTreat]

    ## single treatment effect for type_2 = "b"
    single_2 <- fit$coefficients[cytokineTreat]

    # interaction Viability
    interaction <- fit$coefficients[paste(drugTreat, cytokineTreat, sep=':')]

      
    ## observed single treatment values
    observed_single_1 <- baseline + single_1 
    observed_single_2 <- baseline + single_2

    ## observed double treatment value
    observed_double <- baseline + single_1 + single_2 + interaction

    ## predicted double treatment value (based only on single effects)
    predicted_double <- baseline + single_1 + single_2


    #prepare labels and aesthetics for plotting
    # y_label <- case_when(cyt=="Interferon gamma"~"Interferon\ngamma",
    #                      cyt=="soluble anti-IgM"~"soluble\nanti-IgM",
    #                      TRUE~cyt)

    xlabels <- c("DMSO", drug, 
                 case_when(cyt=="IL-4"~"IL4",
                           cyt=="Interferon gamma"~"Interferon \u03B3",
                           cyt=="sCD40L+IL-4"~"sCD40L \n+ IL4",
                           cyt=="soluble anti-IgM"~"soluble\nanti-IgM",
                           TRUE~cyt),
                paste(drug, " +\n",
                       case_when(cyt=="IL-4"~"IL4",
                           cyt=="Interferon gamma"~"Interferon \u03B3",
                           cyt=="sCD40L+IL-4"~"sCD40L + IL4",
                           TRUE~cyt),
                sep=""))
      
    segment_size <- 2
    
    
    plotTab <-
      df %>%
      
      dplyr::filter(treatment_drug %in% c("DMSO" , drug)) %>%

      dplyr::filter(treatment_cytokine %in% c("No Cytokine", cyt)) %>%

      dplyr::mutate(treatment_combination = 
                  #if baseline treatment
                  ifelse(treatment_drug=="DMSO" & 
                         treatment_cytokine=="No Cytokine",
                         "DMSO",
                         #if drug only treatment
                         ifelse(treatment_drug == drug & 
                                treatment_cytokine =="No Cytokine",
                                paste0("Drug=", drug),
                                #if cytokine only treatment
                                ifelse(treatment_drug == "DMSO" & 
                                       treatment_cytokine == cyt,
                                       paste0("Cytokine=", cyt),
                                       #otherwise combinatorial treatment
                                       paste0("Drug=", drug, "\nCytokine=", cyt))))) %>%
  #make treatment_combination as a factor
  mutate(treatment_combination = factor(treatment_combination, 
                                        levels=c("DMSO", 
                                                 paste0("Drug=", drug),
                                                 paste0("Cytokine=", cyt),
                                                 paste0("Drug=", drug, "\nCytokine=", cyt)
                                                 )))
  #make plot
        
        

  ggplot(plotTab, aes(treatment_combination, Viability, group=PatientID)) +
  
  geom_hline(yintercept = 0, linetype=2, color="black")+

  lemon::geom_pointline( size=1,   
                         colour=ifelse(interaction > 0,
                                       "#A6093D","#003DA5"), alpha=0.3) +
  #add predicted viability with control treatment
  geom_segment( size=segment_size,
                aes(x=1-.2, xend=1+.2, y=baseline, yend=baseline),
                colour = "black") +
  #add predicted viability with drug only treatment
  geom_segment( size=segment_size,
                aes(x=2-.2, xend=2+.2, y=observed_single_1,yend=observed_single_1),
                colour="black") +
  #add predicted viability with cytokine only treatment
  geom_segment( size=segment_size,
                aes(x=3-.2, xend=3+.2, y=observed_single_2, yend=observed_single_2),
                colour="black") +
  #add predicted viability with both treatments
  geom_segment( size=segment_size,
                aes(x=4-.2, xend=4+.2, y=observed_double, yend=observed_double),
                colour="black") +
  #add predicted viability with both treatments, without interaction
  geom_segment( size=segment_size,
                aes(x=4-.2, xend=4+.2, y=predicted_double, 
                    yend=predicted_double), color="blue") + 
 
   #add category and combination as title
    
    
    
  ggtitle(paste( Category,": ", drug, " & ", 
                 case_when(cyt=="IL-4"~"IL4",
                           cyt=="Interferon gamma"~"Interferon \u03B3",
                           cyt=="sCD40L+IL-4"~"sCD40L + IL4",
                           TRUE~cyt), sep="")) + 
  
  #add theme
  t2 + theme(axis.text.x = element_text(size=fontsize+5)) +
    theme(plot.tag=element_text(size = 30))+
  scale_x_discrete(labels = xlabels) +
  ylab("Log (Viability)") + 
  xlab("")


  } ) 

names(plotList) <- names(gg)

design1 <-"
  12
  34
  56"
  
plotList$`treatment_drug:Ibrutinib:treatment_cytokine:IL-4` +
  
plotList$`treatment_drug:Ibrutinib:treatment_cytokine:Interferon gamma` + 
  
plotList$`treatment_drug:Pyridone-6:treatment_cytokine:sCD40L+IL-4` +
  
plotList$`treatment_drug:Ralimetinib:treatment_cytokine:Interferon gamma`+
  
plotList$`treatment_drug:Ibrutinib:treatment_cytokine:CpG ODN` +
  
plotList$`treatment_drug:Luminespib:treatment_cytokine:soluble anti-IgM` +  plot_annotation(tag_levels=c("A"))  + plot_layout(design=design1) 
  
```


### IL4 induces resistance to BCR inhibitors and chemotherapeutics {#IL4resistance}
(Figure \@ref(fig:IL4Res)). 

(ref:IL4Res)  _Figure from Bruch & Giles et al. 2021._ Method \@ref(id)

```{r IL4Res, fig.cap='(ref:IL4Res)', message = FALSE,  echo = FALSE, fig.height=6.6, fig.width = 12, fig.align="center", out.width = '80%', dev = 'cairo_pdf'}


(wrap_plots(plotList$`treatment_drug:Idelalisib:treatment_cytokine:IL-4`) + wrap_plots(plotList$`treatment_drug:PRT062607:treatment_cytokine:IL-4`)) /
  
(wrap_plots(plotList$`treatment_drug:Everolimus:treatment_cytokine:IL-4`) +
  
wrap_plots(plotList$`treatment_drug:Selumetinib:treatment_cytokine:IL-4`))

```

## Genetic modulators of drug response 
### Univariate analysis identifies genetic modualtors of drug response

(Figure \@ref(fig:drugGeneIntMap)). 

(ref:drugGeneIntMap)  _Figure from Bruch & Giles et al. 2021._ Method \@ref(id)

```{r drugGeneIntMap, fig.cap='(ref:drugGeneIntMap)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 7, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

df_univariate <- df_complete %>% filter(Drug_Concentration=="High")

##################### List of mutations with >2 positive cases #############################
selected_mutations <- 
  patMeta %>% 
  mutate(Ras_Raf = as.factor(ifelse(BRAF == 1 | KRAS == 1| NRAS == 1, 1, 0))) %>% 
  dplyr::select(-BRAF, -KRAS, -NRAS) %>% 
  dplyr::select( -gender, 
                 -diagnosis, 
                 ) %>%
  pivot_longer(-PatientID, 
               names_to = "Genetic_Alteration", 
               values_to = "alteration_value") %>% 
  dplyr::filter(alteration_value %in% c(1, "U")) %>% 
  dplyr::group_by(Genetic_Alteration) %>% 
  dplyr::count() %>% 
  dplyr::filter(n>2) %>% 
  dplyr::select(Genetic_Alteration) %>% 
  unlist()

##################### t tests and p-value adjustment ###############################  
p_values <-
  filter(df_univariate, Cytokine == "No Cytokine", Drug != "DMSO") %>%
  ## Select columns from screening data 
  dplyr::select(PatientID, Log, Drug) %>% 
  
  ## Join Screening data with metadata
  left_join(patMeta, by = "PatientID") %>%
  
  ## Add Ras/Raf column, remove single columns
  mutate(Ras_Raf=as.factor(ifelse(BRAF == 1 | KRAS == 1 | NRAS == 1, 1, 0))) %>% 
  dplyr::select(-BRAF, -KRAS, -NRAS) %>% 
## remove unused columns from metadata
  dplyr::select( -gender, 
                 -diagnosis, 
                 -treatment) %>% 
  
  ## transform data to long format
  pivot_longer(cols=c(-PatientID, -Log, -Drug), 
               names_to = "Genetic_alt", 
               values_to = "alt_value") %>% 
  
  ## filter to selected mutations (see above)
  dplyr::filter(Genetic_alt %in% selected_mutations) %>% 
  
  
  ## group by Drug and Genetic alteration  
  dplyr::group_by(Drug, Genetic_alt) %>% 
  
  ## Perform t.test on every combination of Cytokine and genetic alteration
  do(tidy(t.test(Log ~ alt_value, data = ., var.equal = T))) %>% 
  
  ## ungroup before adjusting p-value
  ungroup() %>% 
  
  ## adjust p-values to multiple testing using BH method 
  mutate(adj.p.value=p.adjust(p.value, method = "BH"))
  
################################## Order of cytokines by descending significance ############################
Drug_order <-
  p_values %>% 
  dplyr::group_by(Drug) %>% 
  dplyr::arrange(adj.p.value) %>% 
  dplyr::filter(row_number() == 1) %>% 
  ungroup() %>% 
  dplyr::arrange(adj.p.value) %>% 
  dplyr::select(Drug) %>% 
  unlist()

Genetic_alt_order <-
  p_values %>% 
  dplyr::group_by(Genetic_alt) %>% 
  dplyr::arrange(adj.p.value) %>% 
  dplyr::filter(row_number() == 1) %>% 
  ungroup() %>% 
  dplyr::arrange(adj.p.value) %>% 
  dplyr::select(Genetic_alt) %>% 
  unlist()

############################# Define FDR cutoff #################################
fdr = 0.1
  
############################################ Plot ########################################
p_values %<>% 
  mutate(Drug=factor(Drug, levels = Drug_order)) %>% 
  mutate(Genetic_alt=factor(Genetic_alt, levels = Genetic_alt_order))
  
  ggplot(dplyr::filter(p_values, adj.p.value>fdr), 
         aes(x = Drug, y = -log10(adj.p.value))) +
  geom_point(color = "lightgrey", size = 3) +
  geom_beeswarm(data = dplyr::filter(p_values, adj.p.value <= fdr), 
                aes( color=Genetic_alt), size = 3, cex = 1.7) +
  ##FDR line  
  geom_hline(yintercept = -log10(fdr),linetype = "dashed", size=0.3)+
  
  ##Main Theme
  t1 +
  
  theme(axis.text.x = element_text(angle = 45,
                                   hjust = 1, 
                                   vjust = 1),
        axis.title.x=element_blank()) +
  ##Legend Theme
  theme(legend.position=c(0.65,0.65), 
        legend.title = element_text(face='bold', hjust = 0, size=13), 
        legend.key = element_blank(),  
        legend.text = element_text(size=11)) +
  guides(colour = guide_legend(nrow=7, title = "Mutations"), 
         shape = guide_legend(ncol = 1)) +
  scale_color_manual(name = "Mutations", values = c(colors, "#373A36", "#A8A99E", "#54585A", colors), labels=c("IGHV.status"="IGHV status", 
                                                                   "del9p"="del(9p)", 
                                                                   "trisomy12"="trisomy 12", 
                                                                   "gain17q"="gain(17q)", 
                                                                   "gain2p"="gain(2p)",
                                                                   "gain19p"="gain(19p)",
                                                                   "gain19q"="gain(19q)",
                                                                   "del7q"="del(7q)",
                                                                   "del9q"="del(9q)", 
                                                                   "del4p"= "del(4p)", 
                                                                   "SPEN"="SPEN", 
                                                                   "del11q"="del(11q)", 
                                                                   "del1q" = "del(1q)")) +
  scale_y_continuous(expression("BH-adjusted  "* italic(p)*"-value"), 
                     breaks=seq(0, 10, 5),
                     labels=math_format(expr=10^.x)(-seq(0,10,5))) 

```

### Multivariate analysis identifies genetic modualtors of drug response

(Figure \@ref(fig:drugGeneIntMulti)). 

(ref:drugGeneIntMulti)  _Figure from Bruch & Giles et al. 2021._ Method \@ref(id)

```{r drugGeneIntMulti, fig.cap='(ref:drugGeneIntMulti)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 10, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}



seaName <- paste("I-BET 762", "No Cytokine", sep = ":")
        barValue <- rowMeans(coefficients_lasso[[seaName]])
        freqValue <- rowMeans(abs(sign(coefficients_lasso[[seaName]])))
        barValue <- barValue[abs(barValue) >= 0 & freqValue >= 0.5]
        barValue <- barValue[order(barValue)]
        
        
      
        #for the heatmap and scatter plot below the heatmap
        allData <- geneMatrix
        viabValue <- unlist(viabMatrix[seaName,])
        
        tabValue <- allData[, names(barValue),drop=FALSE]
        ord <- order(viabValue)
        viabValue <- viabValue[ord]
        tabValue <- tabValue[ord, ,drop=FALSE]
        sampleIDs <- rownames(tabValue)
        tabValue <- as.tibble(tabValue)
        
        
        tabValue$Sample <- sampleIDs
        
        #Mark different rows for different scaling in heatmap
        #annotate mutations by mutation, methylation or IGHV
        matValue <- gather(tabValue, key = "Var",value = "Value", -Sample)
        matValue$Type <- "mut"
        
        #for Methylation Cluster
        matValue$Type[grep("Methylation",matValue$Var)] <- "meth"
        
        #for IGHV status
        matValue$Type[grep("IGHV",matValue$Var)] <- "ighv"
        
        #change the scale of the value so that IGHV, Methylation and Mutation do not overlap
        matValue[matValue$Type == "mut",]$Value = matValue[matValue$Type == "mut",]$Value + 10
        matValue[matValue$Type == "meth",]$Value = matValue[matValue$Type == "meth",]$Value + 20
        matValue[matValue$Type == "ighv",]$Value = matValue[matValue$Type == "ighv",]$Value + 30
        
        #change continuous to categorical
        matValue$Value <- factor(matValue$Value,levels = sort(unique(matValue$Value)))
        
        
        #change order of heatmap
        matValue$Var <- factor(matValue$Var, levels = names(barValue))
        matValue$Sample <- factor(matValue$Sample, levels = names(viabValue))
        
        #plot the heatmap NB mutated= 11 = black
        p1 <- ggplot(matValue, aes(x=Sample, y=Var)) + 
          geom_tile(aes(fill=Value), color = "white") + #ghost white
          theme_bw()+
          scale_y_discrete(expand=c(0,0)) + 
          theme(axis.title.y = element_text( size=16),
                axis.text.y=element_text(hjust=0, size=14), 
                axis.ticks=element_blank(),
                panel.border=element_rect(colour="gainsboro"),  
                plot.title=element_text(face="bold", size = 18, margin = margin(t = -5, b = 1)), 
                panel.background=element_blank(),
                panel.grid.major=element_blank(), 
                panel.grid.minor=element_blank()) + 
          xlab("Mutation status for each patient") + 
          ylab("") + 
          scale_fill_manual(name="Mutated", 
                            values=c(`10`= offwhite,  #WT
                                     `11`="#373A36", #Mutant
                                     `20`= offwhite, #LP
                                     `20.5`= "#707372", #IP
                                     `21` = "#A8A99E", #HP
                                     `30` = offwhite, #IGHV-U
                                     `31` = "#707372"), #IGHV-M
                            guide=FALSE) + 
          
          ggtitle("IBET-762")
        
        
        
        
        #Plot the bar plot on the left of the heatmap 
        barDF = data.frame(barValue, nm=factor(names(barValue),levels=names(barValue)))
        
        p2 <- ggplot(data=barDF, aes(x=nm, y=barValue)) + 
          geom_bar(stat="identity", 
                   fill=ifelse(barValue<0,
                               palblues[6],palreds[8]), 
                   colour="black", 
                   size=0.3) + 
          scale_x_discrete(expand=c(0,0.5)) + 
          scale_y_continuous(expand=c(0,0)) + 
          coord_flip() + #changed from min(barValue) and max(barValue)
          theme(panel.grid.major=element_blank(), 
                panel.background=element_blank(), 
                axis.ticks.y = element_blank(),
                axis.ticks.x = element_blank(),
                panel.grid.minor = element_blank(), 
                axis.text=element_text(size=11, angle = 45, hjust = 1.5), 
                panel.border=element_blank()) +
          ylab("Size of predictor") + 
          geom_vline(xintercept=c(0.5), 
                     color="black", 
                     size=0.6)
        
        #Plot the scatter plot under the heatmap
        scatterDF = data.frame(X=factor(names(viabValue), 
                                        levels=names(viabValue)), 
                               Y=unlist(viabValue))
        
        p3 <- ggplot(scatterDF, aes(x=X, y=Y)) + 
          geom_point(shape=21, 
                     fill="dimgrey", 
                     colour=colors[1], 
                     size=1.2) + 
          theme_bw() +
          theme(panel.grid.minor=element_blank(), 
                panel.grid.major.x=element_blank(), 
                axis.ticks.x=element_blank(), 
                axis.text.y=element_text(size=12), 
                panel.border=element_rect(colour="dimgrey", size=0.1),
                panel.background=element_rect(fill="white")) +
          xlab("Log(Viability) for each sample, with selected treatment")
        
        
        #Assemble all the plots togehter
        
        # construct the gtable
        wdths = c(0.2, 1.5, 0.2, 0.9*ncol(matValue), 2, 0.2)
        hghts = c(0.2, 0.2, 0.0020*nrow(matValue), 0.16, 1, 0.2, 1.5)
        gt = gtable(widths=unit(wdths, "in"), heights=unit(hghts, "in"))
        
        ## make grobs
        gg1 = ggplotGrob(p1)
        gg2 = ggplotGrob(p2)
        gg3 = ggplotGrob(p3)
        
        
        #make legend grob
        legendFor = c("G", "I", "M")
        
        coldef<-list()
        coldef["I"] <- list(c(offwhite,"#707372")) #U-CLL and M-CLL
        coldef["M"] <- list(c(offwhite, "#707372", "#A8A99E")) #IP, HP, LP
        coldef["G"] <- list(c(offwhite, "#373A36")) #WT and Mutated

      
        
        legends = makelegends(legendFor=c("G","I", "M"),coldef)
        gg4 <- legends[["plot"]]
        
        ## fill in the gtable
        
        #HEATMAP
        #5:1 = "PREDICTORS"
        gt = gtable_add_grob(gt, gtable_filter(gg1, "panel"), 3, 4) # add heatmap
        gt = gtable_add_grob(gt, gtable_filter(gg1, "panel"), 3, 4) #add legend
        gt = gtable_add_grob(gt, gtable_filter(gg1, "title"), 1, 4) #add title to plot
        gt = gtable_add_grob(gt, gtable_filter(gg1, "axis-l"), 3, 5) # variable names
        gt = gtable_add_grob(gt, gtable_filter(gg1, "xlab-b"), 2, 4) # axis title
        
        #BARPLOT
        gt = gtable_add_grob(gt, gtable_filter(gg2, "panel"), 3, 2) # add barplot
        gt = gtable_add_grob(gt, gtable_filter(gg2, "axis-b"), 4, 2) # y axis for barplot
        gt = gtable_add_grob(gt, gtable_filter(gg2, "xlab-b"), 2, 2) # y lab for barplot
        
        
        #SCATTER PLOT
        gt = gtable_add_grob(gt, gtable_filter(gg3, "panel"), 5, 4) # add scatterplot
        gt = gtable_add_grob(gt, gtable_filter(gg3, "xlab-b"), 6, 4) # x label for scatter plot
        gt = gtable_add_grob(gt, gtable_filter(gg3, "axis-l"), 5, 3) #  axis for scatter plot
        
        #LEGEND
        gt = gtable_add_grob(gt, gg4, 7, 4) #  legend
        
        #plot
        grid.newpage()
        grid.draw(gt)
        


```


### Specific examples 
(Figure \@ref(fig:drugGeneIntEx)). 

(ref:drugGeneIntEx)  _Figure from Bruch & Giles et al. 2021._ Method \@ref(id)

```{r drugGeneIntEx, fig.cap='(ref:drugGeneIntEx)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 10, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

df_patmeta <- left_join(df_univariate, patMeta, by = "PatientID")

ibet_tri12 <- 
df_patmeta %>% 
  dplyr::filter(Drug=="I-BET 762", Drug_Concentration == "High", Cytokine == "No Cytokine") %>%
      filter(!is.na(trisomy12)) %>%
         
  ggplot(aes(x=trisomy12,y=Log,color=(trisomy12)))+
  geom_hline(yintercept = 0)+
  geom_boxplot()+
  geom_beeswarm(cex=1.5) +
  guides(color="none", shape="none")+
  stat_compare_means(method = "t.test",
                     label.x.npc = "center", 
                     size=3)+
  xlab("trisomy 12") +
  ylab("Log(Viability)") +
  ggtitle("IBET-762") +
  scale_color_manual(values=c(colors[1], colors[2])) + 

  t2




nutlin_tri12 <- 
df_patmeta %>% 
  dplyr::filter(Drug=="Nutlin-3a", Drug_Concentration == "High", Cytokine == "No Cytokine") %>%
      filter(!is.na(trisomy12)) %>%
         
  ggplot(aes(x=trisomy12,y=Log,color=(trisomy12)))+
  geom_hline(yintercept = 0)+
  geom_boxplot()+
  geom_beeswarm(cex=1.5) +
  guides(color="none", shape="none")+
  stat_compare_means(method = "t.test",
                     label.x.npc = "center", 
                     size=3)+
  xlab("trisomy 12") +
  ylab("Log(Viability)") +
  ggtitle("Nutlin-3a") +
  scale_color_manual(values=c(colors[1], colors[2])) + 

  t2

ralimetinib_tri12 <- 
df_patmeta %>% 
  dplyr::filter(Drug=="Ralimetinib", Drug_Concentration == "High", Cytokine == "No Cytokine") %>%
      filter(!is.na(trisomy12)) %>%
         
  ggplot(aes(x=trisomy12,y=Log,color=(trisomy12)))+
  geom_hline(yintercept = 0)+
  geom_boxplot()+
  geom_beeswarm(cex=1.5) +
  guides(color="none", shape="none")+
  stat_compare_means(method = "t.test",
                     label.x.npc = "center", 
                     size=3)+
  xlab("trisomy 12") +
  ylab("Log(Viability)") +
  ggtitle("Ralimetinib") +
  scale_color_manual(values=c(colors[1], colors[2])) + 

  t2

ibet_tri12 + nutlin_tri12 + ralimetinib_tri12
```
in a grid: 
IBET-762 and trisomy 12
Nutlin-3a and trisomy 12 and gene dosage effect of MDM2 in trisomy12 CLL
Correlation of IBET-762 and Nutlin-3a
Ralimetinib and trisomy 12

## The modulatory effect of mutations on drug - stimuli interactions: 
### Fitting patient-specific linear models identfies drug - stimulus - gene interactions 
(Figure \@ref(fig:drugCytGeneIntMap)).

Also Appendix figure \@ref(fig:drugcytGeneIntAll) 

(ref:drugCytGeneIntMap)  _Figure from Bruch & Giles et al. 2021._ Method \@ref(id) Have to make sure this figure takes a whol page but doesnt float to the bottom

```{r drugCytGeneIntMap, fig.cap='(ref:drugCytGeneIntMap)', message = FALSE,  echo = FALSE, fig.height=18, fig.width = 9, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

#cluster coeff.mat
fit <-
  coeff.mat %>% 
  dist() %>% 
  hclust()
  
order_comb <- rownames(coeff.mat)[fit$order]

#Put matrix into long format for ggplot
coeff.long.mat <- coeff.mat %>% 
  as.data.frame() %>% 
  tibble::rownames_to_column("int") %>% 
  tidyr::gather(key = "gene", value = "coeff", -int)

#get top 8 coefficients 
order_alt <- 
  coeff.long.mat %>% 
  dplyr::filter(coeff!=0) %>% 
  dplyr::group_by(gene) %>% 
  dplyr::count(sort=T) %>% 
  dplyr::select(gene) %>% 
  unlist() %>%
  .[1:8] 

coeff.long.mat.ordered <-  
  coeff.long.mat %>%
  dplyr::filter(gene %in% order_alt, coeff!=0) %>% 
  dplyr::mutate(gene = factor(gene, levels = order_alt), int = factor(int, levels = rev(order_comb)))  %>% 
  mutate(direction = ifelse(coeff>0, "Positive", "Negative")) %>% 
  separate(int, sep=" \\+ ", into=c("Drug", "Stimulus"), remove = FALSE)

Order_Stim <- 
  coeff.long.mat.ordered %>% 
  dplyr::filter(coeff!=0) %>% 
  dplyr::group_by(Stimulus) %>% 
  dplyr::count(sort=T) %>% 
  dplyr::select(Stimulus) %>% 
  unlist()

Order_Drug <- 
  coeff.long.mat.ordered %>% 
  dplyr::filter(coeff!=0) %>% 
  dplyr::group_by(Drug) %>% 
  dplyr::count(sort=T) %>% 
  dplyr::select(Drug) %>% 
  unlist() %>% 
  rev()

coeff.long.mat.ordered <-
  coeff.long.mat.ordered %>%
  mutate(Stimulus=factor(Stimulus, levels = Order_Stim), 
         Drug=factor(Drug, levels = Order_Drug))
  

cyt_labels = c("TGF-b1"="TGF-\u03B21", "sCD40L+IL-4"="sCD40L + IL4", "IL-1b"="IL1\u03B2", "IL-4"="IL4", "IL-6"="IL6","IL-15"="IL15","IL-10"="IL10", "IL-21"="IL21","IL-2"="IL2", "Interferon gamma"= "Interferon \u03B3", "SDF-1a"="SDF-1\u03B1", "CpG ODN"="CpG ODN", "Resiquimod"="Resiquimod", "sCD40L"="sCD40L", "HS-5 CM"="HS-5 CM", "soluble anti-IgM"="soluble anti-IgM", "BAFF"="BAFF")

gene_labels = c("IGHV.status" = "IGHV status", "del11q" = "del(11q)", "del13q" = "del(13q)", "del17p" = "del(17p)", "trisomy12" = "trisomy 12", "TP53" = "TP53", "SF3B1" = "SF3B1", "ATM" = "ATM")

coeff.long.mat.ordered %>% 
  ggplot(aes(y=Drug, x=gene)) +
  geom_tile(aes(fill=coeff),color = "white") +
  scale_fill_gradientn(colors=c(rep(palblues[1:4],each=2),
                                "white", 
                                rep(palreds[5:8], each=2)),  
                       limits=c(-0.8,.8)) +
  t1 +
  theme(axis.text.x = element_text(angle = 35, hjust = 1, vjust = 1),
        axis.title.x = element_blank(),
        axis.title.y = element_blank(),
        axis.ticks = element_blank(),
        axis.ticks.x =element_blank(),
        panel.background = element_rect(color = "black", fill=NA), 
        axis.line.x = element_blank(),
        axis.line.y = element_blank(),
        legend.key.height=unit(0.5, "cm"),
        legend.key.width=unit(2, "cm"),
        legend.title = element_text(face='bold', hjust = 0, size=fontsize+2),
        legend.position = "top",
        legend.key = element_blank(),
        legend.text = element_text(size=fontsize),
        legend.background = element_rect(color = NA),
        strip.text.y.left = element_text(size=18, angle = 0, face="bold"),
        strip.background = element_blank())+
  
  
  labs(fill = expression(beta["int"]))+
  scale_y_discrete(position = "right")+
  scale_x_discrete(labels = gene_labels) +
  facet_grid(Stimulus~.,  scales = "free_y",  space="free_y", switch = "both", labeller =labeller( Stimulus=cyt_labels))
  

```
### Patient - specific drug - stimulus interactions may have clinical significance

(Figure \@ref(fig:FluCpGExA)). 

(ref:FluCpGExA)  _Figure from Bruch & Giles et al. 2021._ Method \@ref(id)

(Figure \@ref(fig:FluCpGExB)). 

(ref:FluCpGExB)  _Figure from Bruch & Giles et al. 2021._ Method \@ref(id)


```{r FluCpGExA, fig.cap='(ref:FluCpGExA)', message = FALSE,  echo = FALSE, fig.height=6, fig.width = 10, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}


# Plot predictor profile for Ibrutinib + IL4

grid.arrange(grobs = Lasso_Plots_Fig6["Fludarabine + CpG ODN"], ncol = 1)

```

```{r FluCpGExB, fig.cap='(ref:FluCpGExB)', message = FALSE,  echo = FALSE, fig.height=6, fig.width = 6, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}


#set facet labels 
tri12.labs <- c("0" = "Non-\ntrisomy 12", "1" = "trisomy 12")
ighv.labs <- c("U" = "IGHV-U", "M" =  "IGHV-M")

#join viability and genetic data tables
  left_join(df_complete, patMeta, by = "PatientID") %>% 
  #filter for drug:stimulus combinations of interest, make sure no NAs in genetic data
  dplyr::filter(Drug_Concentration %in% c("High", "None"),
    DCK%in%c("DMSO:CpG ODN","Fludarabine:CpG ODN","Fludarabine:No Cytokine"),
                !is.na(trisomy12),
                !is.na(IGHV.status)) %>%
    mutate(DCK=factor(DCK, levels=c("DMSO:CpG ODN","Fludarabine:No Cytokine","Fludarabine:CpG ODN"))) %>% 
  
  #plot treatment combination against viability 
  ggplot(aes(x = DCK,y = Log,color= DCK))+
  geom_hline(yintercept = 0)+
  geom_boxplot()+
  geom_beeswarm(cex=1.5) +
  guides(color="none", shape="none")+
  #add p values
  stat_compare_means(method = "t.test",
                     label.y.npc = 0.8, 
                     paired = TRUE, 
                     comparisons = list(c(1,3), c(2,3)),
                     step.increase=0.2, 
                     size=3) +
  xlab("") +
  ylab("Log(Viability)") +
  #facet by trisomy 12 and IGHV status
  facet_grid(vars(trisomy12), 
             vars(IGHV.status),
             labeller = labeller(trisomy12 = tri12.labs, IGHV.status = ighv.labs))+
  
  scale_x_discrete(labels=c("DMSO:CpG ODN"="CpG ODN",
                            "Fludarabine:No Cytokine"="Fludarabine",
                            "Fludarabine:CpG ODN"="Fludarabine \n+ CpG ODN "))+  
    
  scale_color_manual(values=c(colors[4], colors[5],colors[3])) + 
  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))+
  t2 +
  theme(strip.background =element_rect(fill=NA),
        strip.text = element_text(size=fontsize+4, face="bold"),
        strip.text.y = element_text(angle = 0),
        axis.text.x = element_text(size=fontsize+4, angle = 35, hjust = 1, vjust = 1))
 


```

(Figure \@ref(fig:ibrIl4ExA)).
(ref:ibrIl4ExA)
(Figure \@ref(fig:ibrIl4ExB)).
(ref:fig:ibrIl4ExB)

```{r ibrIl4ExA, fig.cap='(ref:ibrIl4ExA)', message = FALSE,  echo = FALSE, fig.height=6, fig.width = 10, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

# Plot predictor profile for Ibrutinib + IL4

grid.arrange(grobs = Lasso_Plots_Fig6["Ibrutinib + IL-4"], ncol = 1)
```


```{r fig:ibrIl4ExB, fig.cap='(ref:fig:ibrIl4ExB)', message = FALSE,  echo = FALSE, fig.height=6, fig.width = 6, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

  left_join(df_complete, patMeta, by = "PatientID")%>%
  #filter for drug:stimulus combinations of interest, make sure no NAs in genetic data
  dplyr::filter(DCK%in%c("DMSO:IL-4","Ibrutinib:IL-4","Ibrutinib:No Cytokine"),
                Drug_Concentration %in% c("High", "None"),
                !is.na(trisomy12),
                !is.na(IGHV.status)) %>%
     mutate(DCK = factor(DCK, levels=c("DMSO:IL-4",
                                      "Ibrutinib:No Cytokine",
                                      "Ibrutinib:IL-4"))) %>%
  
  #plot treatment combination against viability 
  ggplot(aes(x = DCK,y = Log,color= DCK))+
  geom_hline(yintercept = 0)+
  geom_boxplot()+
  geom_beeswarm(cex=1.5) +
  guides(color="none", shape="none")+
  #add p values
  stat_compare_means(method = "t.test",
                     label.y.npc = 0.8, 
                     paired = TRUE, 
                     comparisons = list(c(1,3), c(2,3)),
                     step.increase=0.2, 
                     size=3) +
  xlab("") +
  ylab("Log(Viability)") +
  #facet by trisomy 12 and IGHV status
  facet_grid(vars(trisomy12), 
             vars(IGHV.status),
             labeller = labeller(trisomy12 = tri12.labs, IGHV.status = ighv.labs))+
  
  scale_x_discrete(labels=c("DMSO:IL-4"="IL4",
                            "Ibrutinib:No Cytokine"="Ibrutinib",
                            "Ibrutinib:IL-4"="Ibrutinib \n+ IL4"))+  
    
  scale_color_manual(values=c(colors[4], colors[5], colors[3])) + 
  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))+
  t2 +
  theme(strip.background =element_rect(fill=NA),
        strip.text = element_text(size=fontsize+4, face="bold"),
        strip.text.y = element_text(angle = 0),
        axis.text.x = element_text(size=fontsize+4, angle = 35, hjust = 1, vjust = 1))
 


```


(Figure \@ref(fig:nutlinCpgEx)).
(ref:nutlinCpgEx)

```{r nutlinCpgEx, fig.cap='(ref:nutlinCpgEx)', message = FALSE,  echo = FALSE, fig.height=6, fig.width = 10, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

# Plot predictor profile for Ibrutinib + IL4

grid.arrange(grobs = Lasso_Plots_Fig6["Nutlin-3a + CpG ODN"], ncol = 1)
```

(Figure \@ref(fig:ralIl4ExA)).
(ref:ralIl4ExA)

```{r ralIl4ExA, fig.cap='(ref:ralIl4ExA)', message = FALSE,  echo = FALSE, fig.height=6, fig.width = 10, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

# Plot predictor profile for Ibrutinib + IL4

grid.arrange(grobs = Lasso_Plots_Fig6["Ralimetinib + IL-4"], ncol = 1)
```


(Figure \@ref(fig:ralIl4ExB)).
(ref:ralIl4ExB)

```{r ralIl4ExB, fig.cap='(ref:ralIl4ExB)', message = FALSE,  echo = FALSE, fig.height=6, fig.width = 10, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

left_join(df_complete, patMeta, by = "PatientID")%>%
  #filter for drug:stimulus combinations of interest, make sure no NAs in genetic data
  dplyr::filter(DCK%in%c("DMSO:IL-4","Ralimetinib:IL-4"),
                Drug_Concentration %in% c("High", "None"),
                !is.na(trisomy12)) %>%
     mutate(DCK = factor(DCK, levels=c("DMSO:IL-4",
                                      
                                      "Ralimetinib:IL-4"))) %>%
  
  #plot treatment combination against viability 
  ggplot(aes(x = DCK,y = Log,color= DCK))+
  geom_hline(yintercept = 0)+
  geom_boxplot()+
  geom_beeswarm(cex=1.5) +
  guides(color="none", shape="none")+
  #add p values
  stat_compare_means(method = "t.test",
                     label.y.npc = 0.8, 
                     paired = TRUE, 
                     step.increase=0.2, 
                     size=3) +
  xlab("") +
  ylab("Log(Viability)") +
  #facet by trisomy 12 and IGHV status
  facet_wrap(vars(trisomy12),
             labeller = labeller(trisomy12 = tri12.labs))+
  
  scale_x_discrete(labels=c("DMSO:IL-4"="IL4",
                           
                            "Ralimetinib:IL-4"="Ralimetinib \n+ IL4"))+  
    
  scale_color_manual(values=c(colors[4], colors[5], colors[3])) + 
  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))+
  t2 +
  theme(strip.background =element_rect(fill=NA),
        strip.text = element_text(size=fontsize+4, face="bold"),
        strip.text.y = element_text(angle = 0),
        axis.text.x = element_text(size=fontsize+4, angle = 35, hjust = 1, vjust = 1)) 

```

Responses to chemotherapeutic drugs in the context of TLR stimulation are modulated by del11q and ATM mutations,
Trisomy12 enhances pro-survival synergy between Ralimetinib + IFNgamma
Ralimetinib treatment increases the pro-survival effect of IL4, and this is further enhanced in NOTCH1 and Ras/Raf mutants
Fludarabine and TLR
Ibrutinib and IL4
Discuss examples in drug- cyt -gene examples script 

## Using stimuli responses in combination with genetic profiles to predict in vivo drug response 
Can stimuli + genes predict drug response - this would be useful in the clinic. Especially if its just TLR/IL4 response, are specific signals useful?

(Figure \@ref(fig:drugCytGeneIntClinic)). 

(ref:drugCytGeneIntClinic) Method \@ref(id)

```{r drugCytGeneIntClinic, fig.cap='(ref:drugCytGeneIntClinic)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

```

## Summary 

## Discussion 

## Contirbutions Statement

