# Discussion
> "All models are wrong, but some are useful."
>
> `r tufte::quote_footer('--- George E. P. Box')`


For many years, the approach to managing CLL has been dominated by an understanding of the disease genetics. The collective work of many labs has identified the characteristic molecular features of CLL [@Dohner2000; @Ferreira2014; @Landau2015; @Puente2015; @Rendeiro2016; @Beekman2018;  @Zenz2019; @Mallm2019; @Rendeiro2020; @HerbstThesis; @MeierAbt2021] and these have informed the development of successful treatment strategies [@Kipps2017]. However, despite the onset of new targeted therapies CLL remains largely incurable [@Bosch2019] and relapse and the occurrence of MRD represent major issues [@Hayden2012; @Bosch2019]. Moreover, the molecular players identified so far only partially account for the level of disease heterogeneity seen amongst CLL patients. 


More recently, the role of the lymph node and bone marrow compartments have become appreciated [@tenHacken2016]. Indeed, several seminal studies have underlined the importance of the tumour microenvironment in CLL pathogenesis. Firstly, CLL cells require external signals to survive *ex vivo*, indicating that malignant B cells survival is likely to rely on external signals as much as on cell-intrinsic features of the tumour [@Collins1989; @Burger2000; @Kurtova2009; @Deaglio2009; @Purroy2015]. Secondly, the lymph node and bone barrow compartments are the main site of proliferation, in which the aggressive fraction of tumour cells are located [@Messmer2005; @Herishanu2011; @Herndon2017]. Thirdly, the lymph node compartment has been linked to drug resistance and incomplete response to therapy and enlarged lymph nodes are associated with MRD [@Moreton2005; @Ahn2018]. 

Unravelling the complex cross-talk between malignant B cells and the tumour microenvironment is likely to support several clinical goals. Many patients harbour MRD after therapy and it is probable that targeting the protective niche in combination with current therapies will lead to deeper remissions. *in vitro* studies of drugs in the absence of microenvironmental stimulation have proven invaluable in the development of treatments that have improved patient outcomes. However, more successful treatment strategies will be underpinned by studying drug action within the context of the microenvironment.  Greater predictive power is required to successfully stratify patients and to move towards more personalised approaches: its therefore possible that prognostic indices should account for the protective niche in addition to tumour cell-intrinsic markers, given its well-established role in treatment resistance and MRD. 

Achieving these goals starts with basic biology. This will require a systematic understanding of the critical pathways in CLLâ€“microenvironment crosstalk. In particular, this provides the opportunity to identify unifying pathogenic mechanisms that operate across heterogeneous disease subtypes.  Moreover, many studies have characterised the individual omic layers of the disease [@Ferreira2014; @Landau2015; @Puente2015; @Rendeiro2016; @Beekman2018; @Zenz2019; @Mallm2019; @Rendeiro2020; @HerbstThesis; @MeierAbt2021]. Integrating these layers, and how they interact with microenvironmental signals will provide a more complex understanding of disease biology. Furthermore, linking cell-intrinsic and cell-extrinsic features to response will provide greater insight into drug resistance mechanisms and guide treatment strategies 

Bridging the gaps between what we currently know, and where we need to be, will require the concerted efforts of many studies toward this direction. This thesis aims to provide some understanding to bridge this gap. 

## *ex vivo* profiling of drug and stimulus responses of CLL primary patient samples
In this thesis, I conceived that a systematic investigation of drugs, molecular features and soluble factors within the CLL microenvironment could provide the building blocks for a more holistic understanding of interactions between tumour genetics, the microenvironment and drugs in more complex *in vivo* situations. I made use of a reductionist model of microenvironmental signalling based on an *ex vivo* perturbation assay to query the effect of 17 microenvironmental stimuli and 12 drugs on primary CLL samples. This dataset was combined with patient multi-omic profiles. Collectively, this work maps the effects of soluble factors in the microenvironment, both in isolation and in combination with drugs, and links these to underlying molecular properties in 192 primary CLL samples (Bruch and Giles et al. 2021). I discuss the key findings below. 

This reductionist approach had both advantages and drawbacks.  Cross-talk between malignant cells and the tumour microenvironment is mediated by soluble factors, along with cell-cell contacts, the extracellular matrix and secreted vesicles (see Section \@ref(intro-microenvironmental-pathways)) [@Guarini2008; @Oppezzo2013; @Crompot2017]. Thus, whilst the functional assay employed here encompasses a cross-section of this network, it does not account for the complete landscape of microenvironmental communication.  Nevertheless, the use of soluble factors enabled the identification of direct causal links between signalling pathways and tumour responses. Moreover, this study complements similar work investigating cell-cell contacts in CLL drug response [@HerbstThesis], and collectively these studies aim to piece together the role that both soluble factors and cell-cell contacts play in drug response.

Additionally, the use of primary CLL samples rather than cell lines, encompassed the disease heterogeneity of CLL. Significantly,  this study represents a much larger scale approach than previous work in order to capture this genetic diversity. For example, the effects of IL4 on CLL cells has been studied in primary samples previously  [@Dancescu1992; @Bhattacharya2015; @AguilarHernandez2016] but only in very small cohorts. 

### Sensitivity to microenvironmental stimulation 
The *ex vivo* functional assay evaluated the impact of 12 soluble factors on CLL viability.  This approach was informed by @Carey2017, who performed a similar screen of 94 soluble factors in AML. Both studies demonstrate the value of systematic *in vitro* functional screening to probe the impact of cell-extrinsic factors on primary samples. In particular, studies of this kind can support the identification of unifying pathogenic mechanisms in genetically heterogenous haematological malignancies. 

We observed that most stimuli increased tumour cell survival. A few signals had a profound impact, most notably IL4 which increased tumour cell viability across all genetic backgrounds. Stimulation of TLR7/8/9 had an equally significant impact on tumour cell viability and increased or decreased viability depending on the molecular features of the tumour cells (Bruch and Giles et al. 2021). A few signals decreased survival of the tumour cells, including TGF$\beta$, and these pathways may represent potential targets to exploit in a clinical setting. 

### Microenvironmental response profiling identified a group of IGHV-M CLLs with faster disease progression

In addition to highlighting key survival pathways, microenvironmental response profiling delineated four CLL subgroups. These subgroups were linked to distinct molecular profiles and showed differential disease progression, independently of known prognostic markers. The subgroup containing mostly IGHV-M CLLs that responded poorly to the panel of stimuli demonstrated faster disease progression (Bruch and Giles et al. 2021).

These subgroups highlight the importance of microenvironmental signaling in CLL pathogenesis and raise the possibility that *in vitro* responses to microenvironmental signals might reflect distinct *in vivo* leukemic potential and thus may have prognostic value. This is not unprecedented; for example @Scielzo2011 proposed that *in vitro* responses to sCD40L distinguished two functional CLL subsets, that demonstrate different clinical courses. 

Stimuli response profiling as a clinical tool may be unconventional. However, CLL is a highly heterogenous disease, and our current understanding of molecular features only goes so far in explaining and predicting disease outcomes and relapse [@Miller2019; @Bosch2019]. These observations suggest that prognostic indices may be improved through accounting for role of the protective niche. The finding that IL4 signalling is measurable and elevated within CLL lymph node biopsy samples, and relates to disease progression further underlines this. Its possible that probing signalling activity as part of lymph node biopsies may have prognostic value.

## Genetic modulators of microenvironmental signalling 
### Systematic analysis of the effect of genetic features on responses to stimuli 
Patient subgroups with distinct molecular features responded differently to the panel of stimuli, pointing to the fact that genetic mutations alter responses to signalling pathways (Bruch and Giles et al. 2021). 

I applied multivariate modelling to quantify the impact of molecular features on microenvironmental pathways. For example, multiple features modulated viability in response to TLR7/8/9 stimulation, including IGHV status, trisomy 12, del(11q), del(13q) and *ATM*. This reflects the complexity of the network of molecular features and pathways that drive pathogenicity in CLL (Bruch and Giles et al. 2021). 

del(11q) and *ATM* mutations are thought to impact on the function of the DDR. Thus, these results suggest possible cross-talk between the TLR and DDR pathways, which is disrupted by mutations in del(11q) and *ATM*. This would support previous observations suggesting a role for *ATM* in TLR signalling [@Neves2017].


### The role of trisomy 12 as a modulator of microenvironmental response
Amongst these gene-microenvironment interactions, trisomy 12 emerged as a key modulator of response to microenvironmental signals (Bruch and Giles et al. 2021). This observation raises the possibility that trisomy 12 CLLs may be more reliant on microenvironmental signals than other subtypes. This would agree with clinical observations that trisomy 12 CLLs show a shorter redistribution lymphocytosis during ibrutinib therapy [@Thompson2018], such that when tumour cells are released from the lymph node they rapidly undergo apoptosis outside of the protective niche. 

I investigated which molecular features of trisomy 12 CLL may lead to enhanced responses to microenvironmental signals. A number of studies have implicated over expression of genes in trisomy 12 pathogenicity, particularly those on chromosome 12 [@Dickinson2006; @Porpaczy2009; @Abruzzo2018], and thus I began by investigating expression of genes within microenvironmental signalling pathways. 
*STAT6*, *IRAK4* and *SMAD3* were all overexpressed in trisomy 12. This may enable trisomy 12 CLLs to respond more strongly to external stimulation. Previous work has also pointed to a critical role for IRAK4 in trisomy 12 CLL. @Reid2021 identified IRAK4 as a key drug target through a drug screen of inhibitors of trisomy 12-related genes. IRAK4 inhibition also selectively targeted trisomy 12 CLLs in patient-derived xenografted mouse models. Thus, IRAK4, and the TLR pathway may represent an important target in trisomy 12 CLL. 

Additionally, I investigated the epigenetic signature of CLL and identified TFs that showed higher binding site accessibility in trisomy 12 CLL (Bruch and Giles et al. 2021). This analysis demonstrated that Spi-B and PU.1 showed higher inferred activity in two independent datasets. Using ChIPseq, I linked these TFs to control of BCR, TLR and TGF$\beta$ signalling genes, suggesting that both these TFs may contribute to enhanced activity of these pathways (Bruch and Giles et al. 2021). These findings build on work by @Beekman2018, who indicated that trisomy 12 CLL is linked to a distinct epigenetic signature. They observed that NFAT, FOX and TCF/LEF TF family binding sites were enriched amongst the open chromatin regions that were unique to trisomy 12 samples. Despite the contrasting results, both studies suggest that the distinctive epigenetic profile and TF activities observed in trisomy 12 CLL may represent a targetable feature. More work will be required to establish how to exploit this in the clinic. 

In this thesis, I noted that IBET-762 inhibited the TF signature of trisomy 12 CLL and showed greater efficacy in trisomy 12 samples (Bruch and Giles et al. 2021). Though speculative, its possible that BET inhibition represents a strategy through which to target enhanced microenvironmental response in trisomy 12 CLL. This is supported by other work on BET inhibitors in haematological malignancies indicating that these therapies may inhibit cytokine signalling.  For example, the BET inhibitor OTX015 has been shown to inhibit growth of cell lines derived from mature B-cell lymphoid tumors via inhibition of NF$\kappa$B, TLR, and JAK-STAT [@Boi2015], whilst the BET inhibitor JQ1 reduced expression of NF$\kappa$B target genes in Mantle Cell Lymphoma [@Sun2015]. 

## Genetic and microenvironmental modulators of drug response
### The impact of stimuli on drug response
In addition to drug-gene interactions, a key feature of this study was the ability to identify drug-stimulus interactions. This refers to cases where the efficacy of a drug was modulated by the activity a microenvironmental pathway, or vice versa. I identified such cases using a linear model, indicated in Equation \@ref(eq:drugCytInt) (Bruch and Giles et al. 2021).

The linear modelling approach represents the first attempt to map drug-microenvironment interactions in CLL on a large scale. Whilst it cannot be assumed that the analysis captured all interactions existing in the dataset, the linear model nevertheless represents a valuable tool and my drug-stimulus interaction analysis recapitulated known phenomena and identified many novel cases. These included the effect of IL4 stimulation on BCR inhibition [@AguilarHernandez2016], discussed in Section \@ref(discussion-IL4-ibrutinib) and the impact of IFN$\gamma$ on BCR inhibitor and chemotherapeutic efficacy (Bruch and Giles et al. 2021). 

The drug-microenvironment interaction map is also not comprehensive in that it only accounts for the pathways included within the screen. However, the approach has guided other interaction analyses, such as that published in @HerbstThesis, who used a similar linear model to identify drug-microenvironment interactions between drugs and BMSC cell lines. Collectively, these studies are piecing together a comprehensive picture of the importance of the microenvironment in drug response, to help guide better therapeutic strategies that account for the role of the protective nice. 

Intriguingly, the comprehensive approach enabled the observation that IL4 and IFN$\gamma$ both interacted with a common set of drugs, suggesting that they may operate via a similar mechanism. This was supported by the observation that inhibition of p38 MAPK with ralimetinib increased the protective effect of both stimuli. Previous work suggests that both IL4 and IFN$\gamma$ both induce nitric oxide synthase (NOS2) in CLL cells to promote survival [@Levesque2003]. Taken together, its possible that both IL4 and IFN$/gamma$ increase viability via NOS2. Further work will be needed to confirm whether this could be targeted in the clinic. 

### The modulatory effect of mutations on drug-stimulus interactions

Together with the impact of cell-extrinsic factors on drug response, the network of cell-intrinsic molecular components adds an additional layer of complexity (Bruch and Giles et al. 2021). I captured the impact of genetic features on drug-stimulus interactions using multivariate modelling. The size of each drug-stimulus interaction within each patient was measured by fitting Equation \@ref(eq:drugCytInt) (Bruch and Giles et al. 2021) in a patient specific manner. Then the impact of genetic features on the size of the interaction for each drug-stimulus combination was quantified through multivariate modelling, with lasso penalisation.  

This chosen method had both benefits and caveats. The approach effectively screened a large number of drug-stimulus-gene combinations to generate a smaller list of interactions that could be interpreted individually. It is possible that certain hits were artefacts of the modelling approach, which meant that interpreting each individual hit within a biological context was important.  Additionally, the use of lasso penalisation generated sparse models with only a few genetic predictors per interaction which supported the interpretation. However, its likely that some genetic features may have been eliminated from the model outputs, due to the fact they correlated with other genetic features that were assigned coefficients. In these cases, it was not possible to know which mutation was causing the observed effect and again careful interpretation was required. 

Nevertheless, this drug-stimulus-gene interaction map successfully highlighted many interesting cases. Strikingly, TLR stimulation induced drug resistance or increased drug efficacy with chemotherapeutic drugs depending on the genetic background. For example, TLR stimulation increased fludarabine efficacy in IGHV-M non-trisomy 12 CLL samples, but induced resistance in all other cases (Bruch and Giles et al. 2021). TLR signalling is also known to be highly active within CLL-infiltrated lymph nodes (Bruch and Giles et al. 2021). Collectively, these observations may partially explain why fludarabine therapy often achieves lasting remission in IGHV-M but not in IGHV-U CLL cases [@Chai2017]. These observations also underline how greater understanding of the protective niche will bridge some of the gaps in our understanding of disease heterogeneity in CLL. 

## The interaction between IL4 and ibrutinib {#discussion-IL4-ibrutinib}
Throughout this thesis, IL4 has continually emerged as a central pro-survival signal, at least within this *in vitro* context. Not only did IL4 stimulation increase viability in the absence of drugs, it also induced resistance to all BCR inhibitors and chemotherapeutics included in the screen. IL4 also represented a unifying mechanism of CLL survival as it increased survival and induced drug resistance across diverse genetic backgrounds. Notably, the impact of IL4 was elevated in trisomy 12 CLL, and slightly reduced in CLL samples with mutation is Ras/Raf/MEK/ERK (Bruch and Giles et al. 2021). 

Given these findings, it was natural to investigate whether IL4 signalling is relevant within the *in vivo* context. This thesis demonstrated increased IL4 activity within CLL-infiltrated lymph nodes compared to non-neoplastic samples and showed that higher IL4 activity was associated with faster disease progression (Bruch and Giles et al. 2021). This builds on work by @AguilarHernandez2016, who identified enrichment of IL4 signature genes within the lymph node compared with blood and bone marrow samples from the same CLL patients. Combined with the observation that enlarged lymph nodes are linked to resistance to BCR inhibition [@Ahn2018], its possible that *in vivo* activity of IL4 within the lymph node could induce resistance to treatment with chemotherapeutics and BCR inhibitors. Future studies should now focus on single cell studies of lymph node biopsies. It will be of particular interest to define the impact of microenvironmental signals on drug response directly within the *in vivo* context, with the aim of generating effective strategies to eliminate MRD within the protective niche (Bruch and Giles et al. 2021).

It will also be important to confirm the source of IL4, as this is still debated. CD4+ T cells may be involved [@Patten2008; @Monserrat2014;@AguilarHernandez2016].  Recent work within out lab suggests that follicular helper T cells provide the source of IL4 [@Roider2020]. Levels of these T cells are higher in CLL patient blood samples and cells thought to be follicular helper T cells have also been observed within lymph node samples [@Ahearne2013]. 

Deeper remissions during ibrutinib and idelalisib therapy are essential to improving patient outcomes and targeting IL4 signalling *in vivo* could support this goal. This thesis identified bromodomain inhibition via IBET-762 to be an effective strategy to inhibit IL4 activity and support BCR inhibitor efficacy within *in vitro* CLL primary samples.  Bromodomain inhibition reduced JAK-STAT signalling in IL4 stimulated CLL cells, in agreement with other studies showing bromodomain suppresses transcriptional responses to JAK-STAT signalling [@Chan2015] in monocytes. 

Moreover, bromodomain inhibition in combination with other therapies may also be useful as a broad spectrum microenvironmental inhibitor: BET inhibitors have been shown to target cytokine signalling via NF$\kappa$B, TLR and JAK-STAT in mature B-cell lymphoid tumour cell lines [@Boi2015]. 

Our results indicate that BET inhibition increases ibrutinib efficacy in the context of IL4, and may be useful as a combinatorial therapy. This agrees with other work showing that bromodomain inhibition reduces expression of BTK in Mantle Cell Lymphoma cells [@Sun2015], indicating that BET inhibition may complement BCR inhibition by operating downstream of current BTK and SYK targeted drugs. BET inhibition has also been shown to synergise with ibrutinib in DLBCL mouse models [@Ceribelli2014]. Beyond ibrutinib and idelalisib, bromodomain inhibition may be a useful tool in combination with other drug types. For example, BRD inhibition shows additional activity in combination with venetoclax [@Carra2019] in CLL cell lines and in *ex vivo* CLL samples. 

Collectively, these studies make the case for evaluating bromodomain inhibitors in combination with current targeted therapies, although more *in vivo* work will be needed here. Current phase I clinical studies have focused on directly targeting JAK-STAT signalling in combination with BCR inhibitors and have shown promise [@Hamlin2019; @Spaner2019]. Bromodomain inhibition may provide an additional tool, inhibiting both cytokine signalling and BCR signalling to increase the efficacy of BCR and BCL2-targeted drugs. 

## Open Science 
Few studies have so far investigated cell-intrinsic and cell-extrinsic features systematically in cancer, and this study represents an important milestone in this direction. Thus an important goal was to ensure that this work is freely available, through the publication of open-access software. In this thesis, and published alongside Bruch and Giles et al. (2021), I present a resource for the study of drug response in the context of molecular and microenvironmental modulators. This resource has already been used by other researchers e.g. @Lu2021 and aims to inform targeted mechanistic investigations and support efforts to identify combination therapies (Giles and Bruch et al. 2021). 

## Summary 
This thesis presents the analysis of an *ex vivo* perturbation assay to query the effect of 17 microenvironmental stimuli and 12 drugs on primary CLL samples, combined with patient multi-omic profiles. The aim was to generate a systematic investigation of drugs, molecular features and soluble factors within the CLL microenvironment and to provide the building blocks for a more holistic understanding of interactions between tumour genetics, the microenvironment and drugs in more complex *in vivo* situations (Bruch and Giles et al. 2021).  