# (APPENDIX) Appendix - Figures {-}

```{r, echo = FALSE}

knitr::opts_chunk$set(
   echo = FALSE, 
   message = FALSE, 
   warning = FALSE,
   fig.align="center"
)
```

```{r loadDataAppendix}

#drug-cyt-gene interactions
load("data/coeff.long.mat_Fig6.RData") # from data for figure 6 in supplement
load("data/patMeta.RData")
load("data/cytokines.RData")
load("data/drugs.RData")
load("data/df.RData")
load("data/cyt_and_receptors.RData")
load("data/diffTF_small.RData")
load("data/Lasso_Plots_Fig6.RData")


library(ggplot2)
library(tidyverse)
library(dplyr)
library(kableExtra)
library(BloodCancerMultiOmics2017)


bwScale = c("0"="white","1"="black","N.A."="grey90")
lfsize = 16 # legend font size

```

```{r processDataAppendix}

#add label column to df
df <- df %>%
  dplyr::mutate(Cytokine = as.character(Cytokine)) %>%
  dplyr::mutate(Cytlabel = ifelse(Cytokine == "IL-2", "IL2",
                            ifelse(Cytokine == "IL-4", "IL4",
                                   ifelse(Cytokine == "IL-6", "IL6",
                                          ifelse(Cytokine == "IL-10", "IL10",

                                                        ifelse(Cytokine == "IL-21", "IL21",
                                                               ifelse(Cytokine == "sCD40L+IL-4","sCD40L + IL4",
                                                                      ifelse(Cytokine == "IL-15", "IL15",
                                                                             ifelse(Cytokine == "Interferon gamma","Interferon \u03B3",
                                                                                    ifelse(Cytokine == "SDF-1a","SDF-1\u03B1",
                                                                                           ifelse(Cytokine == "IL-1b","IL1\u03B2",
                                                                                                  ifelse(Cytokine == "TGF-b1","TGF\u03B21", Cytokine)))))))))))) %>%

  dplyr::mutate(Cytokine = factor(Cytokine))

#generate table of Cytokines and assosciated labels
Cytokine_labels <- df %>% 
  dplyr::filter(PatientID == "Pat_001", Drug == "DMSO") %>% 
  dplyr::select( Cytokine, Cytlabel)

colnames(Cytokine_labels) <- c("name", "Cytlabel")


```

```{r echo = FALSE}

#load themes 
### ggplot themes

fontsize = 11

## theme for ggplots
t1 <- 
  theme(                              
  plot.background = element_blank(), 
  panel.grid.major = element_line(),
  panel.grid.major.x = element_line(linetype = "dotted", colour = "grey"),
  panel.grid.minor = element_blank(), 
  panel.border = element_blank(), 
  panel.background = element_blank(),
  axis.line = element_line(size=.4),
  axis.line.x = element_line(),
  axis.line.y = element_line(),
  axis.text.x  = element_text(angle=90, size=11, hjust = 1, vjust = 0.4),
  axis.text.y = element_text(size = 11),
 # axis.ticks.x = element_line(linetype = "dotted"),
  axis.ticks.length = unit(0.3,"cm"),
  axis.title.x = element_text(face="bold", size=13), 
  axis.title.y = element_text(face="bold", size=13),
  plot.title = element_text(face="bold", size=14, hjust = 0.5),
  strip.text = element_text(size = fontsize)
)

t2 <- t1 +
  theme( axis.text.x  = element_text(angle=0, size=11, hjust = 0.5, vjust = 1))

## theme for legends
t.leg <-  theme(legend.title = element_text(face='bold', 
                                            hjust = 1, size=11),
                legend.key = element_blank(),
                legend.text = element_text(size=11),
                legend.background = element_rect(color = "black"))



### Set colour palettes

#For Categorical: 
colors <- c("#A1BE1F", #green
            "#F4C61F", #yellow
            "#734595", #purple
            "#D41645", #red
            "#3B6FB6", #blue
            "#B65417", #orange
            "#E2E868", #light green
            "#CBA3D8", #light purple
            "#E58F9E", #light purple
            "#8BB8E8", #light blue
            "#F49E17", #light orange
            "#303030", #black
            "#A8A99E", #grey
            "#007B53") #dark green



#For Divergent: 
Divergent <- c("#003DA5", "#2055B0", "#406EBC", "#6086C7", "#809ED2", "#9FB6DD", "#BFCFE9", "#DFE7F4", "white", "white", "white","#F4E0E7", "#E9C2CF", "#DEA3B6", "#D3849E", "#C76586", "#BC476E", "#B12855", "#A6093D")

#for negatives only: 
palblues <- c("#003DA5", "#2055B0", "#406EBC", "#6086C7", "#809ED2", "#9FB6DD", "#BFCFE9", "#DFE7F4")

#for positives only:
palreds <- c("#F4E0E7", "#E9C2CF", "#DEA3B6", "#D3849E", "#C76586", "#BC476E", "#B12855", "#A6093D")

#For mutations: 
Mutant <- c("#b5b5b5","#373A36")
Sex <- c("#707372","#D0D0CE")
IGHV <- c("#373A36","#D0D0CE")
Methylation_cluster <- c("#373A36","#A8A99E","#D0D0CE")

#For drugs: 
drugpal <- c("#734595", "#CBA3D8") #purples

#For cytokines: 
cytpal <- c("#F49E17", "#EFC06E") #yellows


#neutral
offwhite <- "#f8f8ff"
lightergrey <- "#D0D0CE"
darkergrey <- "#707372"


na_color="#f0f0f0"


```



## Figures {-}

(ref:drugPlot) Layout of the 15 drugs in two concentrations on each screening plate. _Plot produced by Peter-Martin Bruch._

```{r drugPlot, fig.cap='(ref:drugPlot)', echo = FALSE, fig.height=5, fig.width = 7, fig.align="center", out.width = '70%', dev = 'cairo_pdf'}

knitr::include_graphics("figures/drugPlate.eps")

```


(ref:stimulusPlot1) Layout of 9 / 18 stimuli on the first screening plate. A.A., S.P., Beads, B.S.A. and C.A. refer to additional agents tested in the screen which were not included in downstream analysis.  _Plot produced by Peter-Martin Bruch._

```{r stimulusPlot1, fig.cap='(ref:stimulusPlot1)', echo = FALSE, fig.height=5, fig.width = 7, fig.align="center", out.width = '70%', dev = 'cairo_pdf'}

knitr::include_graphics("figures/stimulusPlate1.eps")


```

(ref:stimulusPlot2) Layout of the remaining 9 / 18 stimuli on the second screening plate. A.A., S.P., Beads, B.S.A. and C.A. refer to additional agents tested in the screen which were not included in downstream analysis.  _Plot produced by Peter-Martin Bruch._

```{r stimulusPlot2, fig.cap='(ref:stimulusPlot2)', echo = FALSE, fig.height=5, fig.width = 7, fig.align="center", out.width = '70%', dev = 'cairo_pdf'}

knitr::include_graphics("figures/stimulusPlate2.eps")

```

(ref:diffTFsmallvolPlot) "Analysis of ATACseq dataset of two trisomy 12 and two non-trisomy 12 untreated CLL PMBC samples. The volcano plot depicts change TF binding site accessibility (x axis) versus BH-adjusted p-values (y axis), comparing trisomy 12  and non-trisomy 12 samples. The `diffTF` [@Berest2019] software was run in analytical mode to calculate TF activity, measured as  weighted mean difference. TFs are labelled if adjusted p-value < 0.01 and absolute weighted mean difference > 0.15."  _diffTF pipeline was run by Ivan Berest, figure and caption published in Bruch and Giles et al. 2021._

```{r diffTFsmallvolPlot, fig.cap='(ref:diffTFsmallvolPlot)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

plotTab <- diffTF_small %>%  filter(!is.na(pvalueAdj))
#make naming consistent with text

plotTab$TF <- gsub("SPI1", "PU1", plotTab$TF)


  ggplot(data = plotTab,
         aes(x = weighted_meanDifference, y = -log10(pvalueAdj))) +
  geom_point(aes(alpha=0.4, colour = ifelse(-log10(pvalueAdj) > 2,"black","grey")),size=3) +
  geom_label_repel(aes(label=ifelse(-log10(pvalueAdj) > 4&abs(weighted_meanDifference)>0.15,
                                    as.character(TF),''),
                       colour=ifelse(weighted_meanDifference>0,
                                     palreds[1],
                                     palblues[1])),
                   size=7, max.overlaps=10)+
  scale_colour_manual(values=c(palblues[3], palreds[8], "black", "grey")) +
  t2 +
  ggtitle("Trisomy 12 versus non-trisomy 12 CLL") +
  xlab("Change in TF binding site accessibility") +
  guides(color="none", alpha="none") +
  scale_y_continuous(expression("-log(adjusted  "* italic(p)*"-value)"), breaks=seq(0,150,50))

rm(plotTab)


#check number of upregulated TFs
# nrow(filter(diffTF_small, weighted_meanDifference>0, pvalueAdj <0.05))


```


\newpage

(ref:drugcytGeneIntAll)  "Heatmap depicting overview of genetic predictors of drug - stimulus interactions (each row represents the coefficients from fitting a single multivariate model). Stimuli are shown on left, and corresponding drugs on right. Drugs, stimuli and genetic alterations are alphabetically sorted. Coloured fields indicate that the $\beta_{int}$ for given drug and stimulus is modulated by corresponding genetic feature. Positive coefficients are shown in red, indicating $\beta_{int}$ is more positive for given drug and stimulus combination if the feature is present." _Figure and caption generated with Peter-Martin Bruch for the manuscript Bruch and Giles et al. 2021._


```{r drugcytGeneIntAll, fig.cap='(ref:drugcytGeneIntAll)', message = FALSE,  echo = FALSE, fig.height=19.5, fig.width = 16, fig.align="center", out.width = '90%', dev = 'cairo_pdf'} 

colnames(Cytokine_labels) <- c("Stimulus", "label")
coeff.long.mat_Fig6 %>%
  tidyr::separate(int, sep=" \\+ ", into=c("Drug", "Stimulus"), remove =FALSE)%>% 
  left_join(Cytokine_labels, by = "Stimulus") %>%
  dplyr::filter(coeff!=0) %>%
  ggplot(aes(y=Drug, x=gene))+
  geom_tile(aes(fill=coeff),color = "white")+
  scale_fill_gradientn(colors=c(rep(palblues[1:4],each=2),"white", rep(palreds[5:8], each=2)),  limits=c(-0.8,.8))+
  t1+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1,size = 22),
        axis.text.y = element_text(size = 18),
        axis.title.x = element_blank(),
        axis.title.y = element_blank(),
        axis.ticks = element_blank(),
        axis.ticks.x = element_blank(),
        panel.background = element_rect(color = "black", fill=NA),
        axis.line.x = element_blank(),
        axis.line.y = element_blank(),
        legend.key.height = unit(0.5, "cm"),
        legend.key.width = unit(2, "cm"),
        legend.title = element_text(face='bold', hjust = 0, size=fontsize+4),
        legend.position = "bottom",
        legend.key = element_blank(),
        
        legend.text = element_text(size=fontsize+2),
        legend.background = element_rect(color = NA),
        strip.text.y.left = element_text(angle = 0, face="bold", size = 22),
        strip.background = element_blank())+
  labs(fill = expression(paste(beta, "-int")))+
  scale_y_discrete(position = "right",limits=rev)+
  scale_x_discrete(labels=c("KRAS,\nNRAS,\nBRAF"="RAS/RAF",
                            "del11q" = "del(11q)",
                            "del17p" = "del(17p)",
                            "del13q" = "del(13q)",
                            "del6q" = "del(6q)",
                            "gain8q" = "gain(8q)",
                            "IGHV.status" = "IGHV status",
                            "trisomy12" = "trisomy 12 "))+
   facet_grid(label~., scales = "free_y",  space="free_y", switch = "both")
```

\newpage


(ref:NutlinPredictors) Predictor profile plot depicting genetic features that predict *in vitro* response to Nutlin-3a. Bar plot on left indicates size and sign of coefficients for the named predictors. Positive coefficients indicate higher viability after treatment, if the feature is present. Scatter plot and heatmap indicate how each genetic feature relates to patient sample viabilities: Scatter plot indicates log(viability) values, in order of magnitude, for each individual sample. Heatmap shows patient mutation status for each of genetic predictors for corresponding sample in scatter plot.

```{r NutlinPredictors, fig.cap='(ref:NutlinPredictors)', echo = FALSE, fig.height=5, fig.width = 7, fig.align="center", out.width = '70%', dev = 'cairo_pdf'}

knitr::include_graphics("figures/Nutlin-3a.eps")

```


(ref:ifngamma) Beeswarm-boxplots of log-transformed control-normalised viability values for IFN$\gamma$ treated samples, with and without Ralimetinib.

```{r ifngamma, fig.cap='(ref:ifngamma)', message = FALSE,  echo = FALSE, fig.align="center",  dev = 'cairo_pdf', out.width='60%' }


left_join(df, patMeta, by = "PatientID") %>% 
  dplyr::filter(drugC %in% c("DMSO:None", "Ralimetinib:High"), 
                Cytokine =="Interferon gamma") %>%
         
  ggplot(aes(x = DCK, y = Log, color = DCK))+
  geom_hline(yintercept = 0)+
  geom_boxplot()+
  geom_beeswarm(cex = 1.5) +
  guides(color="none", shape = "none") +
  stat_compare_means(
                     method = "t.test",
                     paired = TRUE,
                     label.x.npc = "center", 
                     size = 3) +
  xlab("Treatment") +
  ylab("Log(Viability)") +
  ggtitle("") +
  scale_color_manual(values=c(colors[1], colors[2])) + 
  scale_x_discrete(labels=c("DMSO:Interferon gamma"="Interferon \u03B3",
                            "Ralimetinib:Interferon gamma"="Ralimetinib\n + Interferon \u03B3"))+

  t2

```

(ref:ibetatac) "Differential  TF  binding  site  accessibility  (y  axis)  in  trisomy  12  vs  non-trisomy  12  CLL PBMC samples  (purple)  and  for  IBET-762  vs  DMSO  treated  CLL  samples  (green).  Direction  of 
differential accessibility values are shown for two independent datasets comparing trisomy 12 
vs non-trisomy 12 CLL and IBET-762 vs control-treated CLL, for all TFs with adjusted p value 
<0.05 in the trisomy 12 comparison. Absolute change in TF accessibility can not be compared 
between the two experiments. See Methods section \@ref(treated-ATACseq-method) and \@ref(ibet-ATACseq-method)."  _Figure and caption generated with Peter-Martin Bruch for the manuscript Bruch and Giles et al. 2021._

```{r ibetatac, fig.cap='(ref:ibetatac)', message = FALSE,  echo = FALSE, fig.align="center",  dev = 'cairo_pdf', out.width='100%' }

knitr::include_graphics("figures/IBET-trisomy12-ATAC.eps")


```


```{r removeapp1}

#clear data
rm(list = ls())

```