# (APPENDIX) Appendix {-}

```{r, echo = FALSE}

knitr::opts_chunk$set(
   echo = FALSE, 
   message = FALSE, 
   warning = FALSE,
   fig.align="center"
)
```

```{r loadDataAppendix01}

#drug-cyt-gene interactions
load("data/coeff.long.mat_Fig6.RData") # from data for figure 6 in supplement
load("data/patMeta.RData")
load("data/cytokines.RData")
load("data/drugs.RData")
load("data/df.RData")
df_complete <-df
load("data/cyt_and_receptors.RData")
load("data/diffTF_small.RData")
load("data/Lasso_Plots_Fig6.RData")
load("data/dds_smp.RData")


library(ggplot2)
library(tidyverse)
library(dplyr)
library(kableExtra)
library(BloodCancerMultiOmics2017)
library(DESeq2)
library(msigdbr)
library(clusterProfiler)
library(fgsea)


bwScale = c("0"="white","1"="black","N.A."="grey90")
lfsize = 16 # legend font size

```

```{r processDataAppendix01}

#add label column to df
df <- df %>%
  dplyr::mutate(Cytokine = as.character(Cytokine)) %>%
  dplyr::mutate(Cytlabel = ifelse(Cytokine == "IL-2", "IL2",
                            ifelse(Cytokine == "IL-4", "IL4",
                                   ifelse(Cytokine == "IL-6", "IL6",
                                          ifelse(Cytokine == "IL-10", "IL10",

                                                        ifelse(Cytokine == "IL-21", "IL21",
                                                               ifelse(Cytokine == "sCD40L+IL-4","sCD40L + IL4",
                                                                      ifelse(Cytokine == "IL-15", "IL15",
                                                                             ifelse(Cytokine == "Interferon gamma","Interferon \u03B3",
                                                                                    ifelse(Cytokine == "SDF-1a","SDF-1\u03B1",
                                                                                           ifelse(Cytokine == "IL-1b","IL1\u03B2",
                                                                                                  ifelse(Cytokine == "TGF-b1","TGF\u03B21", Cytokine)))))))))))) %>%

  dplyr::mutate(Cytokine = factor(Cytokine))

#generate table of Cytokines and assosciated labels
Cytokine_labels <- df %>% 
  dplyr::filter(PatientID == "Pat_001", Drug == "DMSO") %>% 
  dplyr::select( Cytokine, Cytlabel)

colnames(Cytokine_labels) <- c("name", "Cytlabel")


```

```{r echo = FALSE}

#load themes 
### ggplot themes

fontsize = 11

## theme for ggplots
t1 <- 
  theme(                              
  plot.background = element_blank(), 
  panel.grid.major = element_line(),
  panel.grid.major.x = element_line(linetype = "dotted", colour = "grey"),
  panel.grid.minor = element_blank(), 
  panel.border = element_blank(), 
  panel.background = element_blank(),
  axis.line = element_line(size=.4),
  axis.line.x = element_line(),
  axis.line.y = element_line(),
  axis.text.x  = element_text(angle=90, size=11, hjust = 1, vjust = 0.4),
  axis.text.y = element_text(size = 11),
 # axis.ticks.x = element_line(linetype = "dotted"),
  axis.ticks.length = unit(0.3,"cm"),
  axis.title.x = element_text(face="bold", size=13), 
  axis.title.y = element_text(face="bold", size=13),
  plot.title = element_text(face="bold", size=14, hjust = 0.5),
  strip.text = element_text(size = fontsize)
)

t2 <- t1 +
  theme( axis.text.x  = element_text(angle=0, size=11, hjust = 0.5, vjust = 1))

## theme for legends
t.leg <-  theme(legend.title = element_text(face='bold', 
                                            hjust = 1, size=11),
                legend.key = element_blank(),
                legend.text = element_text(size=11),
                legend.background = element_rect(color = "black"))



### Set colour palettes

#For Categorical: 
colors <- c("#A1BE1F", #green
            "#F4C61F", #yellow
            "#734595", #purple
            "#D41645", #red
            "#3B6FB6", #blue
            "#B65417", #orange
            "#E2E868", #light green
            "#CBA3D8", #light purple
            "#E58F9E", #light purple
            "#8BB8E8", #light blue
            "#F49E17", #light orange
            "#303030", #black
            "#A8A99E", #grey
            "#007B53") #dark green



#For Divergent: 
Divergent <- c("#003DA5", "#2055B0", "#406EBC", "#6086C7", "#809ED2", "#9FB6DD", "#BFCFE9", "#DFE7F4", "white", "white", "white","#F4E0E7", "#E9C2CF", "#DEA3B6", "#D3849E", "#C76586", "#BC476E", "#B12855", "#A6093D")

#for negatives only: 
palblues <- c("#003DA5", "#2055B0", "#406EBC", "#6086C7", "#809ED2", "#9FB6DD", "#BFCFE9", "#DFE7F4")

#for positives only:
palreds <- c("#F4E0E7", "#E9C2CF", "#DEA3B6", "#D3849E", "#C76586", "#BC476E", "#B12855", "#A6093D")

#For mutations: 
Mutant <- c("#b5b5b5","#373A36")
Sex <- c("#707372","#D0D0CE")
IGHV <- c("#373A36","#D0D0CE")
Methylation_cluster <- c("#373A36","#A8A99E","#D0D0CE")

#For drugs: 
drugpal <- c("#734595", "#CBA3D8") #purples

#For cytokines: 
cytpal <- c("#F49E17", "#EFC06E") #yellows


#neutral
offwhite <- "#f8f8ff"
lightergrey <- "#D0D0CE"
darkergrey <- "#707372"


na_color="#f0f0f0"


```

```{r FunctionsAppendix}
enrichmentPlot <- 
  function (pathway, stats, gseaParam = 1, ticksSize = 0.2)
  {   rnk <- rank(-stats)
  ord <- order(rnk)
  statsAdj <- stats[ord]
  statsAdj <- sign(statsAdj) * (abs(statsAdj)^gseaParam)
  statsAdj <- statsAdj/max(abs(statsAdj))
  pathway <- unname(as.vector(na.omit(match(pathway, names(statsAdj)))))
  pathway <- sort(pathway)
  gseaRes <- calcGseaStat(statsAdj, selectedStats = pathway, returnAllExtremes = TRUE)
  bottoms <- gseaRes$bottoms
  tops <- gseaRes$tops
  n <- length(statsAdj)
  xs <- as.vector(rbind(pathway - 1, pathway))
  ys <- as.vector(rbind(bottoms, tops))
  toPlot <- data.frame(x = c(0, xs, n + 1), y = c(0, ys, 0))
  diff <- (max(tops) - min(bottoms))/8
  x = y = NULL
  g <- ggplot(toPlot, aes(x = x, y = y)) +
    geom_point(color = colors[1], size = 0.1) +
    geom_hline(yintercept = max(tops), colour = palreds[7], linetype = "dashed") +
    geom_hline(yintercept = min(bottoms), colour = palreds[7], linetype = "dashed") +
    geom_hline(yintercept = 0, colour = "black") +
    geom_line(color = colors[1]) +
    geom_segment(data = data.frame(x = pathway), mapping = aes(x = x, y = -diff/2, xend = x, yend = diff/2), 
                 size = ticksSize) +
    t2 +
    labs(x = "Rank", y = "Enrich. score")
  g}



```

# Appendix - Figures {-}

(ref:drugPlot) Layout of the 15 drugs in two concentrations on each screening plate. _Plot produced by Peter-Martin Bruch._

```{r drugPlot, fig.cap='(ref:drugPlot)', echo = FALSE, fig.height=5, fig.width = 7, fig.align="center", out.width = '70%', dev = 'cairo_pdf'}

knitr::include_graphics("figures/drugPlate.eps")

```


(ref:stimulusPlot1) Layout of 9 / 18 stimuli on the first screening plate. A.A., S.P., Beads, B.S.A. and C.A. refer to additional agents tested in the screen which were not included in downstream analysis.  _Plot produced by Peter-Martin Bruch._

```{r stimulusPlot1, fig.cap='(ref:stimulusPlot1)', echo = FALSE, fig.height=5, fig.width = 7, fig.align="center", out.width = '70%', dev = 'cairo_pdf'}

knitr::include_graphics("figures/stimulusPlate1.eps")


```

(ref:stimulusPlot2) Layout of the remaining 9 / 18 stimuli on the second screening plate. A.A., S.P., Beads, B.S.A. and C.A. refer to additional agents tested in the screen which were not included in downstream analysis.  _Plot produced by Peter-Martin Bruch._

```{r stimulusPlot2, fig.cap='(ref:stimulusPlot2)', echo = FALSE, fig.height=5, fig.width = 7, fig.align="center", out.width = '70%', dev = 'cairo_pdf'}

knitr::include_graphics("figures/stimulusPlate2.eps")

```


(ref:consensusClusteringB) Assignment of patient samples (columns), to each cluster, for k = 1 - 7 (rows) to demonstrate stability of cluster membership. Cluster colour for k = 4 match those in heatmap in \@ref(fig:stimuliHeatmap). See Methods Section \@ref(stimulus-heatmap-method).

```{r consensusClusteringB, fig.cap='(ref:consensusClusteringB)', fig.align="center", out.width = '45%', echo = FALSE, eval = TRUE}

knitr::include_graphics("figures/consensus010.eps")

```



(ref:GSEApathways) Enrichment plot of selected pathways. Gene set enrichment analysis (GSEA) was performed with the Hallmark gene sets from the GSEA Molecular Signatures Database. Wald statistic was used to rank the genes.  The green curve corresponds to the Enrichment Score curve, which is the running sum of the weighted enrichment score obtained from GSEA software. See Methods Section \@ref(cluster-rna-method). _Figure and caption adapted from Bruch and Giles et al. (2021)._

```{r rnaPrepApp}
cytSeq <- dds_smp

#Filter out IGHV genes
cytSeq <- cytSeq[!grepl("IG_", rowData(cytSeq)$biotype),] 

#split into 3,4
cytSeq.34 <- cytSeq[,colData(cytSeq)[,"Cluster"] %in% c(3,4)]

#remove patients where IGHV is unknown
cytSeq.34 <- cytSeq.34[,colData(cytSeq.34)[,"IGHV.status"] %in% c("U","M")]

#set order of factors
cytSeq.34$Cluster <- factor(cytSeq.34$Cluster, 
                            levels = c("3","4"))

cytSeq.34$IGHV.status <- factor(cytSeq.34$IGHV.status, 
                                levels = c("U","M"))

design(cytSeq.34) <- ~ IGHV.status + Cluster

```

```{r RunDeseqApp}


cytSeq.34 <- DESeq(cytSeq.34)

```


```{r  deseqResultsApp}

res.ds <- 
  results(cytSeq.34, contrast = c("Cluster", 3, 4), tidy = TRUE) %>%
  dplyr::rename(Symbol = "row") %>% 
  dplyr::arrange(pvalue) 

```

```{r}

#get ensembl ids to Entrez dataframe
ens2entrez <- 
  rowData(cytSeq.34)[c("entrezgene", "symbol")] %>% 
  as.data.frame() %>% 
  rownames_to_column("ENSEMBL") %>%  
  tibble::as_tibble()


#Join 
res.ds <- left_join(res.ds, ens2entrez, by=c("Symbol"="ENSEMBL"))


```


```{r getRanksApp}

d <- 
  dplyr::select(res.ds, entrezgene, stat) %>%
  na.omit() %>% 
  dplyr::group_by(entrezgene) %>% 
  dplyr::summarize(stat=mean(stat)) 

## feature 1: numeric vector
geneList <- d$stat

## feature 2: named vector
names(geneList) <- as.character(d$entrezgene)

## feature 3: decreasing order
geneList <- sort(geneList, decreasing = TRUE)


```

```{r getTERM2GENEApp}


hm2gene <- 
  msigdbr(species = "Homo sapiens", category = "H") %>% 
  dplyr::select(gs_name, entrez_gene)

#tidy up terms
hm2gene$gs_name <- gsub("HALLMARK_", "",hm2gene$gs_name)
hm2gene$gs_name <- gsub("_", " ",hm2gene$gs_name)
hm2gene$gs_name <-gsub("MTORC1 SIGNALING"	,"MTORC1 Signaling",hm2gene$gs_name)
hm2gene$gs_name <-gsub("MYC TARGETS V1",	"MYC Targets V1"	,hm2gene$gs_name)
hm2gene$gs_name <-gsub("OXIDATIVE PHOSPHORYLATION",	"Oxidative Phosphorylation",hm2gene$gs_name)
hm2gene$gs_name <-gsub("TNFA SIGNALING VIA NFKB",	"TNFa Signalling via NFKB"	,hm2gene$gs_name)
hm2gene$gs_name <-gsub("UNFOLDED PROTEIN RESPONSE",	"Unfolded Protein Response"	,hm2gene$gs_name)	
hm2gene$gs_name <-gsub("UV RESPONSE UP",	"UV Response Up",hm2gene$gs_name)
hm2gene$gs_name <-gsub("INTERFERON GAMMA RESPONSE",	"Interferon Gamma Response"	,hm2gene$gs_name)
hm2gene$gs_name <-gsub("G2M CHECKPOINT",	"G2M Checkpoint",hm2gene$gs_name)	
hm2gene$gs_name <-gsub("E2F TARGETS", "E2F Targets"	,hm2gene$gs_name)
hm2gene$gs_name <-gsub("P53 PATHWAY",	"P53 Pathway",hm2gene$gs_name)


 
```


```{r GSEApathways, fig.cap='(ref:GSEApathways)', out.wdith = "60%",   echo = FALSE}

TNFaEnrichment <-
  enrichmentPlot(filter(hm2gene, gs_name =="TNFa Signalling via NFKB")$entrez_gene, geneList) +
  labs(title="TNFA signalling via NFKB")


#myc targets
mycEnrichment <-
  enrichmentPlot(filter(hm2gene, gs_name =="MYC Targets V1")$entrez_gene, geneList) +
  labs(title="MYC targets V1")


#oxidative phosphorylation
oxphosEnrichment <-
  enrichmentPlot(filter(hm2gene, gs_name =="Oxidative Phosphorylation")$entrez_gene, geneList) +
  labs(title="Oxidative Phosphorylation")


#p53 pathway
p53Enrichment <-
  enrichmentPlot(filter(hm2gene, gs_name =="P53 Pathway")$entrez_gene, geneList) +
  labs(title="P53 Pathway")



GSEApathways <-  
  (wrap_elements(TNFaEnrichment) + 
  wrap_elements(mycEnrichment) )/
  (wrap_elements(oxphosEnrichment)  +
  wrap_elements(p53Enrichment) ) +
  plot_annotation(tag_levels = "A")
  
GSEApathways

```

(ref:diffTFsmallvolPlot) "Analysis of ATACseq dataset of two trisomy 12 and two non-trisomy 12 untreated CLL PMBC samples. The volcano plot depicts change TF binding site accessibility (x axis) versus BH-adjusted p-values (y axis), comparing trisomy 12  and non-trisomy 12 samples. The `diffTF` [@Berest2019] software was run in analytical mode to calculate TF activity, measured as  weighted mean difference. TFs are labelled if adjusted p-value < 0.01 and absolute weighted mean difference > 0.15."  _diffTF pipeline was run by Ivan Berest, figure and caption published in Bruch and Giles et al. 2021._

```{r diffTFsmallvolPlot, fig.cap='(ref:diffTFsmallvolPlot)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

plotTab <- diffTF_small %>%  filter(!is.na(pvalueAdj))
#make naming consistent with text

plotTab$TF <- gsub("SPI1", "PU1", plotTab$TF)


  ggplot(data = plotTab,
         aes(x = weighted_meanDifference, y = -log10(pvalueAdj))) +
  geom_point(aes(alpha=0.4, colour = ifelse(-log10(pvalueAdj) > 2,"black","grey")),size=3) +
  geom_label_repel(aes(label=ifelse(-log10(pvalueAdj) > 4&abs(weighted_meanDifference)>0.15,
                                    as.character(TF),''),
                       colour=ifelse(weighted_meanDifference>0,
                                     palreds[1],
                                     palblues[1])),
                   size=7, max.overlaps=10)+
  scale_colour_manual(values=c(palblues[3], palreds[8], "black", "grey")) +
  t2 +
  ggtitle("Trisomy 12 versus non-trisomy 12 CLL") +
  xlab("Change in TF binding site accessibility") +
  guides(color="none", alpha="none") +
  scale_y_continuous(expression("-log(adjusted  "* italic(p)*"-value)"), breaks=seq(0,150,50))

rm(plotTab)


#check number of upregulated TFs
# nrow(filter(diffTF_small, weighted_meanDifference>0, pvalueAdj <0.05))


```

(ref:diffTFexplainer) diffTF workflow from @Berest2019. diffTF accepts a list of TFs along with the genomic locations of their bindings sites. For each TF, the software computes the distribution of fold changes between the trisomy 12 and WT samples, using the ATACseq peaks at each TF binding site in each condition. The software compares this distribution to a set of background fold changes produced from loci that show the same GC content but not the TF binding site motif. Each TF is thus assigned assigned a weighted mean difference value, which quantifies the change in binding site accessibility, and a p value. _Figure from original published in @Berest2019, and adapted with permission._

```{r diffTFexplainer, fig.cap='(ref:diffTFexplainer)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '50%', dev = 'cairo_pdf'}

knitr::include_graphics("figures/diffTF_method.eps")

```


\newpage



(ref:drugcytGeneIntAll)  "Heatmap depicting overview of genetic predictors of drug - stimulus interactions (each row represents the coefficients from fitting a single multivariate model). Stimuli are shown on left, and corresponding drugs on right. Drugs, stimuli and genetic alterations are alphabetically sorted. Coloured fields indicate that the $\beta_{int}$ for given drug and stimulus is modulated by corresponding genetic feature. Positive coefficients are shown in red, indicating $\beta_{int}$ is more positive for given drug and stimulus combination if the feature is present." _Figure and caption generated with Peter-Martin Bruch for the manuscript Bruch and Giles et al. 2021._


```{r drugcytGeneIntAll, fig.cap='(ref:drugcytGeneIntAll)', message = FALSE,  echo = FALSE, fig.height=19.5, fig.width = 16, fig.align="center", out.width = '90%', dev = 'cairo_pdf'} 

colnames(Cytokine_labels) <- c("Stimulus", "label")
coeff.long.mat_Fig6 %>%
  tidyr::separate(int, sep=" \\+ ", into=c("Drug", "Stimulus"), remove =FALSE)%>% 
  left_join(Cytokine_labels, by = "Stimulus") %>%
  dplyr::filter(coeff!=0) %>%
  ggplot(aes(y=Drug, x=gene))+
  geom_tile(aes(fill=coeff),color = "white")+
  scale_fill_gradientn(colors=c(rep(palblues[1:4],each=2),"white", rep(palreds[5:8], each=2)),  limits=c(-0.8,.8))+
  t1+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1,size = 22),
        axis.text.y = element_text(size = 18),
        axis.title.x = element_blank(),
        axis.title.y = element_blank(),
        axis.ticks = element_blank(),
        axis.ticks.x = element_blank(),
        panel.background = element_rect(color = "black", fill=NA),
        axis.line.x = element_blank(),
        axis.line.y = element_blank(),
        legend.key.height = unit(0.5, "cm"),
        legend.key.width = unit(2, "cm"),
        legend.title = element_text(face='bold', hjust = 0, size=fontsize+4),
        legend.position = "bottom",
        legend.key = element_blank(),
        
        legend.text = element_text(size=fontsize+2),
        legend.background = element_rect(color = NA),
        strip.text.y.left = element_text(angle = 0, face="bold", size = 22),
        strip.background = element_blank())+
  labs(fill = expression(paste(beta, "-int")))+
  scale_y_discrete(position = "right",limits=rev)+
  scale_x_discrete(labels=c("KRAS,\nNRAS,\nBRAF"="RAS/RAF",
                            "del11q" = "del(11q)",
                            "del17p" = "del(17p)",
                            "del13q" = "del(13q)",
                            "del6q" = "del(6q)",
                            "gain8q" = "gain(8q)",
                            "IGHV.status" = "IGHV status",
                            "trisomy12" = "trisomy 12 "))+
   facet_grid(label~., scales = "free_y",  space="free_y", switch = "both")
```

\newpage


(ref:NutlinPredictors) Predictor profile plot depicting genetic features that predict *in vitro* response to Nutlin-3a. Bar plot on left indicates size and sign of coefficients for the named predictors. Positive coefficients indicate higher viability after treatment, if the feature is present. Scatter plot and heatmap indicate how each genetic feature relates to patient sample viabilities: Scatter plot indicates log(viability) values, in order of magnitude, for each individual sample. Heatmap shows patient mutation status for each of genetic predictors for corresponding sample in scatter plot.

```{r NutlinPredictors, fig.cap='(ref:NutlinPredictors)', echo = FALSE, fig.height=5, fig.width = 7, fig.align="center", out.width = '70%', dev = 'cairo_pdf'}

knitr::include_graphics("figures/Nutlin-3a.eps")

```


(ref:ifngamma) Beeswarm-boxplots of log-transformed control-normalised viability values for IFN$\gamma$ treated samples, with and without Ralimetinib.

```{r ifngamma, fig.cap='(ref:ifngamma)', message = FALSE,  echo = FALSE, fig.align="center",  dev = 'cairo_pdf', out.width='60%' }


left_join(df, patMeta, by = "PatientID") %>% 
  dplyr::filter(drugC %in% c("DMSO:None", "Ralimetinib:High"), 
                Cytokine =="Interferon gamma") %>%
         
  ggplot(aes(x = DCK, y = Log, color = DCK))+
  geom_hline(yintercept = 0)+
  geom_boxplot()+
  geom_beeswarm(cex = 1.5) +
  guides(color="none", shape = "none") +
  stat_compare_means(
                     method = "t.test",
                     paired = TRUE,
                     label.x.npc = "center", 
                     size = 3) +
  xlab("Treatment") +
  ylab("Log(Viability)") +
  ggtitle("") +
  scale_color_manual(values=c(colors[1], colors[2])) + 
  scale_x_discrete(labels=c("DMSO:Interferon gamma"="Interferon \u03B3",
                            "Ralimetinib:Interferon gamma"="Ralimetinib\n + Interferon \u03B3"))+

  t2

```

(ref:ibetatac) "Differential  TF  binding  site  accessibility  (y  axis)  in  trisomy  12  vs  non-trisomy  12  CLL PBMC samples  (purple)  and  for  IBET-762  vs  DMSO  treated  CLL  samples  (green).  Direction  of 
differential accessibility values are shown for two independent datasets comparing trisomy 12 
vs non-trisomy 12 CLL and IBET-762 vs control-treated CLL, for all TFs with adjusted p value 
<0.05 in the trisomy 12 comparison. Absolute change in TF accessibility can not be compared 
between the two experiments. See Methods Section \@ref(treated-ATACseq-method) and \@ref(ibet-ATACseq-method)."  _Figure and caption generated with Peter-Martin Bruch for the manuscript Bruch and Giles et al. 2021._

```{r ibetatac, fig.cap='(ref:ibetatac)', message = FALSE,  echo = FALSE, fig.align="center",  dev = 'cairo_pdf', out.width='100%' }

knitr::include_graphics("figures/IBET-trisomy12-ATAC.eps")


```



(ref:TLRnutlin) Beeswarm-boxplots showing control-normalised log-transformed viability values, after treatment with nutlin-3a (MDM2), and CpG ODN (TLR9) and nutlin-3a (MDM2), stratified by del(11q), IGHV and trisomy 12 status. p-values from Studentâ€™s t-tests.

```{r TLRnutlin, fig.cap='(ref:TLRnutlin)', message = FALSE,  echo = FALSE, fig.height = 8, fig.width = 6,  fig.align="center", out.width = '100%', dev = 'cairo_pdf'}

TLR_Nut_plotTab <-  
left_join(df_complete, patMeta, by = "PatientID") %>%
  dplyr::filter(Cytokine %in% c("CpG ODN", "No Cytokine"),
                Drug== "Nutlin-3a",
                Drug_Concentration== "High", 
                !is.na(del11q),
                !is.na(IGHV.status), 
                !is.na(trisomy12)) %>% 
  mutate(del11q = ifelse(del11q == "1", "Mutated", "Unmutated")) %>% 
  mutate(IGHV.status = ifelse(IGHV.status == "M", "Mutated", "Unmutated")) %>% 
  mutate(trisomy12 = ifelse(trisomy12 == "1", "Mutated", "Unmutated")) %>% 
  dplyr::select(DCK, Log, trisomy12, del11q, IGHV.status) 

TLR_Nut_plotTab$IGHV.status <- as.factor(TLR_Nut_plotTab$IGHV.status)

TLR_Nut_plotTab%<>% pivot_longer(c(trisomy12, IGHV.status, del11q), names_to = "mutation", values_to= "status")

#adjust factor levels
TLR_Nut_plotTab$DCK <- factor(TLR_Nut_plotTab$DCK, levels = c("Nutlin-3a:No Cytokine", "Nutlin-3a:CpG ODN"))

#adjut labels
# New facet label names for supp variable
DCK.labs <- c("Nutlin-3a", "Nutlin-3a + CpG ODN")
names(DCK.labs) <- c("Nutlin-3a:No Cytokine", "Nutlin-3a:CpG ODN")

mutation.labs <- c("del(11q)", "IGHV status", "trisomy 12")
names(mutation.labs) <- c("del11q", "IGHV.status", "trisomy12")


  ggplot(TLR_Nut_plotTab, aes(x=status,y=Log,color=status))+
  geom_hline(yintercept = 0)+
  geom_boxplot()+
  geom_beeswarm(cex=1.5) +
  guides(color="none", shape="none")+
  stat_compare_means(method = "t.test",
                     label.x.npc = "left", 
                     size=3)+
    
  xlab("") +
  ylab("Log(Viability)") +
  ggtitle("") +
  scale_color_manual(values=c(colors[1], colors[2])) + 
  facet_grid(mutation ~ DCK, labeller = labeller(DCK = DCK.labs, mutation = mutation.labs))+
  t2


```


# Appendix - Tables {-}

```{r patientTable, echo = FALSE}

patMeta %>%
  dplyr::select(PatientID, gender, treatment, IGHV.status, `Methylation_Cluster`, del13q, del11q, trisomy12, del17p) %>%
  dplyr::rename(`Patient ID` = PatientID, `Sex` = gender, `Treated before` = treatment, `IGHV status` = IGHV.status, `Methylation Cluster`=`Methylation_Cluster`, `Del(13q)`=del13q, `Del(11q)`=del11q, `Trisomy 12`=trisomy12, `Del(17p)`=del17p) %>%
  kable(caption = "Summary of the patient samples and selected genetic features included in this study. Table published in Bruch and Giles et al. 2021.", longtable = TRUE) %>% 
  row_spec(0, bold = TRUE) %>%
  kable_styling(latex_options = c("hold_position", "repeat_header", "striped"),   stripe_color = colors[7],
                html_font = "Helvetica", font_size = 7)

```


```{r drugTableAppendix,  echo = FALSE}

rownames(drugs) <- NULL

drugs %>%
  dplyr::select(-ID, -approved_012020,-devel_012020, -pathway) %>%
  dplyr::rename(Drug=Name,
                `Main targets`=main_targets,
                `Drug Group`=group,
                `Target category`=target_category,
                `Conc. 1`=conc1,
                `Conc. 2`=conc2, 
                Company = distributor, 
                `Cat. No.`=cat_no) %>%
  kable(caption = "Characteristics of the drugs included in the screen. Table published in Bruch and Giles et al. 2021. Table generated with Peter-Martin Bruch and published in Bruch and Giles et al. 2021. ") %>%
  row_spec(0,bold=TRUE) %>%
kable_styling(latex_options = c("striped", "scale_down"), 
              stripe_color = colors[7],
                html_font = "Helvetica", 
              font_size = 7)



```

```{r stimulusTableAppendix}

#generate table of Cytokines and assosciated labels
colnames(Cytokine_labels) <- c("name", "Cytlabel")

Cytokine_labels_table <- Cytokine_labels %>%
  dplyr::mutate(name = as.character(name)) %>%
  dplyr::mutate(Cytlabel = ifelse(name == "IL-2", "IL2",
                            ifelse(name == "IL-4", "IL4",
                                   ifelse(name == "IL-6", "IL6",
                                          ifelse(name == "IL-10", "IL10",

                                                        ifelse(name == "IL-21", "IL21",
                                                               ifelse(name == "sCD40L+IL-4","sCD40L + IL4",
                                                                      ifelse(name == "IL-15", "IL15",
                                                                             ifelse(name == "Interferon gamma","Interferon gamma",
                                                                                    ifelse(name == "SDF-1a","SDF-1a",
                                                                                           ifelse(name == "IL-1b","IL1b",
                                                                                                  ifelse(name == "TGF-b1","TGFB1", name)))))))))))) %>%

  dplyr::mutate(name = factor(name))


#tidy names for table
cytokines$stimulus[grepl("IL-4 human recombinant animal component free",cytokines$stimulus)]<-"IL4 human recombinant animal component free"
cytokines$stimulus[grepl("IL-2 human recombinant animal component free",cytokines$stimulus)]<-"IL2 human recombinant animal component free"
cytokines$stimulus[grepl("Human IL-1beta",cytokines$stimulus)]<-"Human IL-1 beta"
cytokines$stimulus[grepl("Human IL-15",cytokines$stimulus)]<-"Human IL15"
cytokines$stimulus[grepl("Human IFN-gamma",cytokines$stimulus)]<-"Human Interferon gamma"
cytokines$stimulus[grepl("IL-10 human Animal component free",cytokines$stimulus)]<-"IL10 human Animal component free"
cytokines$stimulus[grepl("Human TGF-beta1",cytokines$stimulus)] <- "Human TGFbeta"
cytokines$stimulus[grepl("Human IL-6",cytokines$stimulus)]<-"Human IL6"
cytokines$stimulus[grepl("alpha",cytokines$stimulus)] <- "Human SDF1 alpha (CXCL12)"
cytokines$stimulus[grepl("HS-5 konditioniertes Medium",cytokines$stimulus)] <- "HS-5 conditioned medium"
cytokines %>%
  left_join(Cytokine_labels_table, by = "name") %>%
  dplyr::select(-name, -additional_pathways, -source) %>%
  dplyr::rename(Stimulus=stimulus, Name = Cytlabel, Supplier=company, Concentration=conc, `Catalogue Number`=cat_no, `Lot Number`=lot_no, Pathway=pathway) %>%
  dplyr::select(Stimulus, Name,  everything()) %>%
  kable(caption = "Characteristics of the stimuli included in the screen.  Table produced with Peter-Martin Bruch and published in Bruch and Giles et al. 2021.") %>%
  row_spec(0,bold=TRUE) %>%
  kable_styling(latex_options = c("striped", "scale_down"), 
              stripe_color = colors[7],
                html_font = "Helvetica", 
                font_size = 7)




```



```{r}

# select CLL samples
patM = patMeta$PatientID
patMeta <- as.data.frame(patMeta)
rownames(patMeta) <- patM
ighv = factor(setNames(patMeta[patM,"IGHV.status"], nm=patM), levels=c("U","M"))
mut1 = c("del17p", "del11q", "trisomy12", "del13q")
mut2 = c("TP53", "ATM", "SF3B1", "NOTCH1", "MYD88")
mc = patMeta[,c(8:113)] %>% as.matrix()
mc <- matrix(as.numeric(mc),  
             ncol = ncol(mc))
colnames(mc) = colnames(patMeta[,c(8:113)] )
rownames(mc) <- rownames(patMeta[,c(8:113)] )

## SELECTION OF MUTATIONS
# # include mutations with at least incidence of 4
mut2plot = names(which(sort(colSums(mc, na.rm=TRUE), decreasing=TRUE)>3))

# divide mutations into gene mut and cnv
mut2plotSV = mut2plot[grep("[[:lower:]]", mut2plot)]
mut2plotSP = mut2plot[grep("[[:upper:]]", mut2plot)]


# rearrange the top ones to match the order in mut1 and mut2
mut2plotSV = c(mut1, mut2plotSV[!mut2plotSV %in% mut1])
mut2plotSP = c(mut2, mut2plotSP[!mut2plotSP %in% mut2])
factors = data.frame(patMeta[patM, c(mut2plotSV, mut2plotSP)],
                     check.names=FALSE)

# change it to factors
for(i in 1:ncol(factors)) {
  factors[,i] = factor(factors[,i], levels=c(1,0))
}
ord = order(factors[,1], factors[,2], factors[,3], factors[,4], factors[,5],
            factors[,6], factors[,7], factors[,8], factors[,9], factors[,10],
            factors[,11], factors[,12], factors[,13], factors[,14],
            factors[,15], factors[,16], factors[,17], factors[,18],
            factors[,19], factors[,20], factors[,21], factors[,22],
            factors[,23], factors[,24], factors[,25], factors[,26],
            factors[,27], factors[,28], factors[,29], factors[,30],
            factors[,31], factors[,32])
factorsord = factors[ord,]
patM = patM[ord]



```




```{r geneTable, out.width = "70%", fig.cap='(ref:geneTable)', echo = FALSE, eval = TRUE}

plotDF = meltWholeDF(factorsord)
plotDF$Mut =
  ifelse(sapply(plotDF$X,
                function(x) grep(x, list(mut2plotSV, mut2plotSP)))==1,"SV","SP")
plotDF$Status = "N.A."
plotDF$Status[plotDF$Measure==1 & plotDF$Mut=="SV"] = "1a"
plotDF$Status[plotDF$Measure==1 & plotDF$Mut=="SP"] = "1b"
plotDF$Status[plotDF$Measure==0] = "0"
plotDF$Status = factor(plotDF$Status, levels=c("1a","1b","0","N.A."))
plotDF$Y = factor(plotDF$Y, levels=patM)
plotDF$X = factor(plotDF$X, levels=rev(colnames(factorsord)))

mutPL = ggplotGrob(
  ggplot(data=plotDF, aes(x=Y, y=X, fill=Status)) + geom_tile() +
    scale_fill_manual(
      values=c("0"="white","1a"=colors[1],"1b"=colors[2],"N.A."="grey90"),
      name="Mutation", labels=c("WT","CNV","Gene mutation","NA")) +
    ylab("") + xlab("") +
    geom_vline(xintercept=seq(0.5,length(patM)+1,5), colour="grey60") +
    geom_hline(yintercept=seq(0.5,ncol(factorsord)+1,1), colour="grey60") +
    scale_y_discrete(expand=c(0,0)) + scale_x_discrete(expand=c(0,0)) +
    theme(axis.ticks=element_blank(), axis.text.x=element_blank(),
          axis.text.y=element_text(
            size=60, face=ifelse(levels(plotDF$X) %in% mut2plotSV,
                                 "plain","plain")),
          axis.text=element_text(margin=unit(0.5,"cm"), colour="black"),
          legend.key = element_rect(colour = "black"),
          legend.text=element_text(size=lfsize),
          legend.title=element_text(size=lfsize)))
res = table(plotDF[,c("X","Measure")])


knitr::kable(res[order(res[,2], decreasing=TRUE),],
             caption = "Number of cases of each gene mutation within the cohort used in this study. ") %>%
    kable_styling(latex_options = "striped", 
                stripe_color = colors[7],
                html_font = "Helvetica", 
                font_size = 7) 


```


```{r removeapp1}

#clear data
rm(list = ls())

```