# Appendix {-}
```{r loadDataAppendix}

#drug-cyt-gene interactions
load("data/coeff.long.mat_Fig6.RData") # from data for figure 6 in supplement
load("data/patMeta.RData")
load("data/cytokines.RData")
load("data/drugs.RData")
load("data/df.RData")


library(ggplot2)
library(tidyverse)
library(dplyr)
library(kableExtra)
library(BloodCancerMultiOmics2017)


bwScale = c("0"="white","1"="black","N.A."="grey90")
lfsize = 16 # legend font size

```

```{r processDataAppendix}

#add label column to df
df <- df %>%
  dplyr::mutate(Cytokine = as.character(Cytokine)) %>%
  dplyr::mutate(Cytlabel = ifelse(Cytokine == "IL-2", "IL2",
                            ifelse(Cytokine == "IL-4", "IL4",
                                   ifelse(Cytokine == "IL-6", "IL6",
                                          ifelse(Cytokine == "IL-10", "IL10",

                                                        ifelse(Cytokine == "IL-21", "IL21",
                                                               ifelse(Cytokine == "sCD40L+IL-4","sCD40L + IL4",
                                                                      ifelse(Cytokine == "IL-15", "IL15",
                                                                             ifelse(Cytokine == "Interferon gamma","Interferon \u03B3",
                                                                                    ifelse(Cytokine == "SDF-1a","SDF-1\u03B1",
                                                                                           ifelse(Cytokine == "IL-1b","IL1\u03B2",
                                                                                                  ifelse(Cytokine == "TGF-b1","TGF\u03B21", Cytokine)))))))))))) %>%

  dplyr::mutate(Cytokine = factor(Cytokine))

#generate table of Cytokines and assosciated labels
Cytokine_labels <- df %>% 
  dplyr::filter(PatientID == "Pat_001", Drug == "DMSO") %>% 
  dplyr::select( Cytokine, Cytlabel)

colnames(Cytokine_labels) <- c("name", "Cytlabel")


```

## Figures {-}

(ref:drugPlot) Layout of the 15 drugs on each screening plate. 


```{r drugPlot, fig.cap='(ref:drugPlot)', echo = FALSE, fig.height=5, fig.width = 7, fig.align="center", out.width = '100%', dev = 'cairo_pdf'}

knitr::include_graphics("figures/drugPlate.eps")

```


(ref:stimulusPlot) Layout of the 18 stimuli on each screening plate. A.A., S.P., Beads, B.S.A. and C.A. refer to additional agents tested in the screen which were not included in downstream analysis.

```{r stimulusPlot, fig.cap='(ref:stimulusPlot)', echo = FALSE, fig.show="hold", fig.height=5, fig.width = 7, fig.align="center", out.width = '50%', dev = 'cairo_pdf'}

knitr::include_graphics("figures/stimulusPlate1.eps")
knitr::include_graphics("figures/stimulusPlate2.eps")

```

(ref:diffTFsmallvolPlot) Analysis of ATACseq dataset of two trisomy 12 and two non-trisomy 12 untreated CLL PMBC samples. The volcano plot depicts change in TF activity (x axis) versus BH-adjusted p-values (y axis) for  636  TFs, comparing trisomy 12  and non-trisomy 12 samples. The `diffTF` [@Berest2019] software was run in analytical mode to calculate TF activity, measured as  weighted mean difference. TFs are labeled if adjusted p-value < 0.01 and absolute weighted mean difference > 0.15.


```{r diffTFsmallvolPlot, fig.cap='(ref:diffTFsmallvolPlot)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}
knitr::include_graphics("figures/graph.eps")
```

(ref:drugcytGeneIntAll) 


```{r drugcytGeneIntAll, fig.cap='(ref:drugcytGeneIntAll)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}
colnames(Cytokine_labels) <- c("Stimulus", "label")
coeff.long.mat_Fig6 %>%
  tidyr::separate(int, sep=" \\+ ", into=c("Drug", "Stimulus"), remove =FALSE)%>% 
  left_join(Cytokine_labels, by = "Stimulus") %>%
  dplyr::filter(coeff!=0) %>%
  ggplot(aes(y=Drug, x=gene))+
  geom_tile(aes(fill=coeff),color = "white")+
  scale_fill_gradientn(colors=c(rep(palblues[1:4],each=2),"white", rep(palreds[5:8], each=2)),  limits=c(-0.8,.8))+
  t1+
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5),
        axis.title.x = element_blank(),
        axis.title.y = element_blank(),
        axis.ticks = element_blank(),
        axis.ticks.x = element_blank(),
        panel.background = element_rect(color = "black", fill=NA),
        axis.line.x = element_blank(),
        axis.line.y = element_blank(),
        legend.key.height = unit(0.5, "cm"),
        legend.key.width = unit(2, "cm"),
        legend.title = element_text(face='bold', hjust = 0, size=fontsize+2),
        legend.position = "bottom",
        legend.key = element_blank(),
        legend.text = element_text(size=fontsize),
        legend.background = element_rect(color = NA),
        strip.text.y.left = element_text(angle = 0, face="bold", size = 22),
        strip.background = element_blank())+
  labs(fill = expression(paste(beta, "-int")))+
  scale_y_discrete(position = "right",limits=rev)+
  scale_x_discrete(labels=c("KRAS,\nNRAS,\nBRAF"="RAS/RAF"))+
   facet_grid(label~., scales = "free_y",  space="free_y", switch = "both")
```


(ref:NutlinPredictors) ADD NUTLIN-3A Predictor profile


```{r NutlinPredictors, fig.cap='(ref:NutlinPredictors)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 5, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

ggplot(mtcars)

```



## Tables {-}

```{r patientTable, echo = FALSE}

patMeta %>%
  dplyr::select(PatientID, gender, treatment, IGHV.status, `Methylation_Cluster`, del13q, del11q, trisomy12, del17p) %>%
  dplyr::rename(`Patient ID` = PatientID, `Sex` = gender, `Treated before` = treatment, `IGHV status` = IGHV.status, `Methylation Cluster`=`Methylation_Cluster`, `Del13q`=del13q, `Del11q`=del11q, `Trisomy 12`=trisomy12, `Del17p`=del17p) %>%
  kable(caption = "Summary of the patient samples and selected genetic features included in this study. For complete genetic profiles see [online vignette](https://github.com/Huber-group-EMBL/CLLCytokineScreen2021).", longtable = TRUE) %>% 
  row_spec(0, bold = TRUE) %>%
  kable_styling(latex_options = c("hold_position", "repeat_header", "striped"),   stripe_color = colors[7],
                html_font = "Helvetica", font_size = 7)

```


```{r drugTableAppendix,  echo = FALSE}

rownames(drugs) <- NULL

drugs %>%
  dplyr::select(-ID, -approved_012020,-devel_012020, -pathway) %>%
  dplyr::rename(Drug=Name,
                `Main targets`=main_targets,
                `Drug Group`=group,
                `Target category`=target_category,
                `Conc. 1`=conc1,
                `Conc. 2`=conc2, 
                Company = distributor, 
                `Cat. No.`=cat_no) %>%
  kable(caption = "Characteristics of drugs included in the screen.") %>%
  row_spec(0,bold=TRUE) %>%
kable_styling(latex_options = c("striped", "scale_down"), 
              stripe_color = colors[7],
                html_font = "Helvetica", 
              font_size = 7)



```

```{r stimulusTableAppendix,  echo = FALSE, eval = FALSE}

#tidy names for table
cytokines$stimulus[grepl("IL-4 human recombinant animal component free",cytokines$stimulus)]<-"IL4 human recombinant animal component free"
cytokines$stimulus[grepl("IL-2 human recombinant animal component free",cytokines$stimulus)]<-"IL2 human recombinant animal component free"
cytokines$stimulus[grepl("Human IL-1beta",cytokines$stimulus)]<-"Human IL-1$\\beta$"
cytokines$stimulus[grepl("Human IL-15",cytokines$stimulus)]<-"Human IL15"
cytokines$stimulus[grepl("Human IFN-gamma",cytokines$stimulus)]<-"Human Interferon $\\gamma$"
cytokines$stimulus[grepl("IL-10 human Animal component free",cytokines$stimulus)]<-"IL10 human Animal component free"
cytokines$stimulus[grepl("Human TGF-beta1",cytokines$stimulus)]<-"Human TGF$\\beta$"
cytokines$stimulus[grepl("Human IL-6",cytokines$stimulus)]<-"Human IL6"
cytokines$stimulus[grepl("alpha",cytokines$stimulus)]<-"Human SDF-1$\\alpha$ (CXCL12)"
cytokines$stimulus[grepl("HS-5 konditioniertes Medium",cytokines$stimulus)]<-"HS-5 conditioned medium"

newCytokines <-  
cytokines %>%
  left_join(Cytokine_labels, by = "name") %>%
  dplyr::select(-name, -additional_pathways) %>%
  dplyr::rename(Stimulus=stimulus, 
                Name = Cytlabel, 
                Concentration=conc, 
                Pathway=pathway,
                Source=source, 
                Company =company, 
                `Cat. No.`=cat_no, 
                `Lot No.`=lot_no) %>%
  dplyr::select(Stimulus, Name,  everything())


#update label for latex 

newCytokines$Name[grepl("TGF\u03B21",cytokines$Name)] <- "TGF$\\beta$"
newCytokines$Name[grepl("IL1\u03B2",cytokines$Name)] <- "IL1$\\beta$"
newCytokines$Name[grepl("Interferon \u03B3",cytokines$Name)] <- "Interferon $\\gamma$"
newCytokines$Name[grepl("SDF-1\u03B1",cytokines$Name)] <- "SDF-1$\\alpha$"

newCytokines %>% 
  kable(caption = "Characteristics of the stimuli included in the screen.") %>%
  row_spec(0,bold=TRUE) %>%
  kable_styling(latex_options = c("striped", "scale_down"), 
                stripe_color = colors[7],
                html_font = "Helvetica", 
                font_size = 7)

```



```{r receptorPairs, tab.cap='(ref:receptorPairs)', echo = FALSE}

cyt_and_receptors %>%  
    kable(caption = "Table depicting stimulus and receptor pairs used in stimulus response - stimulus receptor expression analysis.") %>%
    kable_styling(latex_options = "striped", 
                stripe_color = colors[7],
                html_font = "Helvetica", 
                font_size = 7) 

```


```{r}

# select CLL samples
patM = patMeta$PatientID
patMeta <- as.data.frame(patMeta)
rownames(patMeta) <- patM
ighv = factor(setNames(patMeta[patM,"IGHV.status"], nm=patM), levels=c("U","M"))
mut1 = c("del17p", "del11q", "trisomy12", "del13q")
mut2 = c("TP53", "ATM", "SF3B1", "NOTCH1", "MYD88")
mc = patMeta[,c(8:113)] %>% as.matrix()
mc <- matrix(as.numeric(mc),  
             ncol = ncol(mc))
colnames(mc) = colnames(patMeta[,c(8:113)] )
rownames(mc) <- rownames(patMeta[,c(8:113)] )

## SELECTION OF MUTATIONS
# # include mutations with at least incidence of 4
mut2plot = names(which(sort(colSums(mc, na.rm=TRUE), decreasing=TRUE)>3))

# divide mutations into gene mut and cnv
mut2plotSV = mut2plot[grep("[[:lower:]]", mut2plot)]
mut2plotSP = mut2plot[grep("[[:upper:]]", mut2plot)]


# rearrange the top ones to match the order in mut1 and mut2
mut2plotSV = c(mut1, mut2plotSV[!mut2plotSV %in% mut1])
mut2plotSP = c(mut2, mut2plotSP[!mut2plotSP %in% mut2])
factors = data.frame(patMeta[patM, c(mut2plotSV, mut2plotSP)],
                     check.names=FALSE)

# change it to factors
for(i in 1:ncol(factors)) {
  factors[,i] = factor(factors[,i], levels=c(1,0))
}
ord = order(factors[,1], factors[,2], factors[,3], factors[,4], factors[,5],
            factors[,6], factors[,7], factors[,8], factors[,9], factors[,10],
            factors[,11], factors[,12], factors[,13], factors[,14],
            factors[,15], factors[,16], factors[,17], factors[,18],
            factors[,19], factors[,20], factors[,21], factors[,22],
            factors[,23], factors[,24], factors[,25], factors[,26],
            factors[,27], factors[,28], factors[,29], factors[,30],
            factors[,31], factors[,32])
factorsord = factors[ord,]
patM = patM[ord]



```




```{r geneTable, out.width = "70%", fig.cap='(ref:geneTable)', echo = FALSE, eval = TRUE}

plotDF = meltWholeDF(factorsord)
plotDF$Mut =
  ifelse(sapply(plotDF$X,
                function(x) grep(x, list(mut2plotSV, mut2plotSP)))==1,"SV","SP")
plotDF$Status = "N.A."
plotDF$Status[plotDF$Measure==1 & plotDF$Mut=="SV"] = "1a"
plotDF$Status[plotDF$Measure==1 & plotDF$Mut=="SP"] = "1b"
plotDF$Status[plotDF$Measure==0] = "0"
plotDF$Status = factor(plotDF$Status, levels=c("1a","1b","0","N.A."))
plotDF$Y = factor(plotDF$Y, levels=patM)
plotDF$X = factor(plotDF$X, levels=rev(colnames(factorsord)))

mutPL = ggplotGrob(
  ggplot(data=plotDF, aes(x=Y, y=X, fill=Status)) + geom_tile() +
    scale_fill_manual(
      values=c("0"="white","1a"=colors[1],"1b"=colors[2],"N.A."="grey90"),
      name="Mutation", labels=c("WT","CNV","Gene mutation","NA")) +
    ylab("") + xlab("") +
    geom_vline(xintercept=seq(0.5,length(patM)+1,5), colour="grey60") +
    geom_hline(yintercept=seq(0.5,ncol(factorsord)+1,1), colour="grey60") +
    scale_y_discrete(expand=c(0,0)) + scale_x_discrete(expand=c(0,0)) +
    theme(axis.ticks=element_blank(), axis.text.x=element_blank(),
          axis.text.y=element_text(
            size=60, face=ifelse(levels(plotDF$X) %in% mut2plotSV,
                                 "plain","plain")),
          axis.text=element_text(margin=unit(0.5,"cm"), colour="black"),
          legend.key = element_rect(colour = "black"),
          legend.text=element_text(size=lfsize),
          legend.title=element_text(size=lfsize)))
res = table(plotDF[,c("X","Measure")])


knitr::kable(res[order(res[,2], decreasing=TRUE),],
             caption = "Number of cases of each gene mutation within the cohort used in this study") %>%
    kable_styling(latex_options = "striped", 
                stripe_color = colors[7],
                html_font = "Helvetica", 
                font_size = 7) 


```