# Molecular and microenvironmental modulators of drug response in CLL {#chapter6}
Chapters \@ref(chapter4) and \@ref(chapter5) provide a systematic exploration of the effect of microenvironmental stimulation on _ex vivo_ CLL biology, and how these pathways can be modulated by genetic features. Collectively, these results underline the fact that cancer biology is determined by an integrated network of cell-intrinsic molecular aberrations and cell-extrinsic signals generated by cell-cell contacts and soluble factors. In this chapter, I present my investigations into the impact of each of these on **drug response**.

The effect of genetic features on _in vitro_ drug response in CLL has been well-characterised, including as part of recent work produced by our lab [@JCIpaper]. Many studies have also demonstrated the impact of individual microenvironmental signals on drug response _ex vivo_, including a number of studies demonstrating that soluble factors can induce resistance to clinical drugs [@Fonte2013; @AguilarHernandez2016; @Jayappa2017; @McWilliams2019]. 

Evidence of microenvironmental signals inducing resistance to therapies  _in vivo_ is less prevalent, although there is widespread consensus that the microenvironment, in particular the lymph node, plays an important role in patient outcomes.  A number of studies have shown enlarged lymph nodes are associated with MRD [@Moreton2005], in particular, incomplete response to ibrutinib is associated with persistently enlarged lymph nodes [@Ahn2018].  For more information see Introduction section \@ref(intro-microenvironment-drug-response).

Guided by these observations, this chapter explores the impact of genetic features, and microenvironmental stimulation on drug response, both individually and in combination. In section \@ref(mapping-interactions), the impact of the panel of stimuli on drug response is quantified by identifying drug - stimulus interactions. In section \@ref(mapping-genetic-modulators), the impact of genetic features on drug response is explored. In section \@ref(drug-stimulus-gene-interactions),  multivariate modelling is applied to investigate how drug - microenvironment interactions can be further modulated by genetic features. Finally, in section \@ref(il4-ibrutinib), I outline my work to investigate the interaction between IL4 and ibrutinib, identified in the above analysis. 

The analyses described here benefit from linear regression and generalised linear modelling approaches, described in more detail in section \@ref(intro-multivariate-modelling). 

Some findings and figures outlined in this chapter have been published in Bruch and Giles et al. 2021, and this is clearly stated where this is the case. 


```{r, echo = FALSE}

knitr::opts_chunk$set(
   echo = FALSE, 
   message = FALSE, 
   warning = FALSE,
   fig.align="center"
)

```

```{r setup06, echo = FALSE}

#libraries
library(patchwork)
library(ggplot2)
library(data.table)
library(magrittr)
library(pheatmap)
library(tidyr)
library(dplyr)
library(plyr)
library(ggpubr)
library(ggbeeswarm)
library(gtable)
library(glmnet)
library(RColorBrewer)
library(gridExtra)
library(cowplot)
library(tidyverse)
library(maxstat)
library(survminer)
library(survival)
library(rlang)
library(biobroom)
library(scales)

```

```{r loadData06, eval = TRUE}

#data("df", "patMeta", "LDT", "survT", "dds_smp")
#df_complete <- df

#Data
#df: tibble containing all screening viability data
load("data/df.RData")
df_complete <- df

#patMeta: tibble containing all patient genetic data
load("data/patMeta.RData")


#data for drug multivariate models 
#lasso coefficients
load("data/coefficients_lasso.RData") #from shiny app data
#gene Matrix
load("data/geneMatrix.RData") #from shiny app data
#viability Matrix
load("data/viabMatrix.RData") #from shiny app data


#data for drug - cyt - gene int analysis
#coefficients from lasso - linear modelling
load("data/coeff_mat_drug_cyt_gene_int.RData") #from figure 6 data script 

#predictor profiles 
load("data/Lasso_Plots_Fig6.RData") #from supplemental data


##ihc_patient_data: 596 x 7 tibble containing IHC staining intensities for pIRAK4, pSTAT6 and STAT6 staining in CLL and healthy tissue 
load( "data/ihc_patient_data.RData")

#ihc_surv_data: 200 x 15 tibble containing IHC staining intensities for pIRAK4, pSTAT6 and STAT6 staining in CLL and healthy tissue, along with meta data (including clinical outcomes) on each patient sample
load( "data/ihc_surv_data.RData")


#IbrStat6iComb.RData: tibble containing  Ibrutinib, IL4 and AS1517499 treated CLL PBMCs
load("data/IbrStat6iComb.RData")


```

```{r themes_functions06, echo = FALSE, eval = TRUE}

### ggplot themes
fontsize = 11

## theme for ggplots
t1 <- 
  theme(                              
  plot.background = element_blank(), 
  panel.grid.major = element_line(),
  panel.grid.major.x = element_line(linetype = "dotted", colour = "grey"),
  panel.grid.minor = element_blank(), 
  panel.border = element_blank(), 
  panel.background = element_blank(),
  axis.line = element_line(size=.4),
  axis.line.x = element_line(),
  axis.line.y = element_line(),
  axis.text.x  = element_text(angle=90, size=11, hjust = 1, vjust = 0.4),
  axis.text.y = element_text(size = 11),
  axis.ticks.length = unit(0.3,"cm"),
  axis.title.x = element_text(face="bold", size=13), 
  axis.title.y = element_text(face="bold", size=13),
  plot.title = element_text(face="bold", size=14, hjust = 0.5),
  strip.text = element_text(size = fontsize)
)

t2 <- t1+
  theme( axis.text.x  = element_text(angle=0, size=11, hjust = 0.5, vjust = 1))

## theme for legends
t.leg <-  theme(legend.title = element_text(face='bold', 
                                            hjust = 1, size=11),
                legend.key = element_blank(),
                legend.text = element_text(size=11),
                legend.background = element_rect(color = "black"))



### Set colour palettes

#For Categorical: 
colors <- c("#A1BE1F", #green
            "#F4C61F", #yellow
            "#734595", #purple
            "#D41645", #red
            "#3B6FB6", #blue
            "#B65417", #orange
            "#E2E868", #light green
            "#CBA3D8", #light purple
            "#E58F9E", #light purple
            "#8BB8E8", #light blue
            "#F49E17", #light orange
            "#303030", #black
            "#A8A99E", #grey
            "#007B53") #dark green



#For Divergent: 
Divergent <- c("#003DA5", "#2055B0", "#406EBC", "#6086C7", "#809ED2", "#9FB6DD", "#BFCFE9", "#DFE7F4", "white", "white", "white","#F4E0E7", "#E9C2CF", "#DEA3B6", "#D3849E", "#C76586", "#BC476E", "#B12855", "#A6093D")

#for negatives only: 
palblues <- c("#003DA5", "#2055B0", "#406EBC", "#6086C7", "#809ED2", "#9FB6DD", "#BFCFE9", "#DFE7F4")

#for positives only:
palreds <- c("#F4E0E7", "#E9C2CF", "#DEA3B6", "#D3849E", "#C76586", "#BC476E", "#B12855", "#A6093D")

#For mutations: 
Mutant <- c("#b5b5b5","#373A36")
Sex <- c("#707372","#D0D0CE")
IGHV <- c("#373A36","#D0D0CE")
Methylation_cluster <- c("#373A36","#A8A99E","#D0D0CE")

#For drugs: 
drugpal <- c("#734595", "#CBA3D8") #purples

#For cytokines: 
cytpal <- c("#F49E17", "#EFC06E") #yellows


#neutral
offwhite <- "#f8f8ff"
lightergrey <- "#D0D0CE"
darkergrey <- "#707372"


#Function for multi-variant regression
runGlm06 <- function(X, y, method = "lasso", repeats = 20, folds = 3) {
  #set up objects to store results
  modelList <- list()
  lambdaList <- c()
  varExplain <- c()
  coefMat <- matrix(NA, ncol(X), repeats)
  rownames(coefMat) <- colnames(X)

  #set alpha
  if (method == "lasso"){
    alpha = 1
  } else if (method == "ridge") {
    alpha = 0
  }
  
  #for the set number of repeats, run the regression
  for (i in seq(repeats)) {
    if (ncol(X) > 2) {
      #run cross validated generalised linear model with given parameters
      res <- cv.glmnet(X,y, type.measure = "mse", family="gaussian", 
                       nfolds = folds, alpha = alpha, standardize = FALSE)
      
      #store lamdas and min lambda value
      lambdaList <- c(lambdaList, res$lambda.min)
      
      #store result of cv.glmnet
      modelList[[i]] <- res
      
      #get coefficents with min lambda value 
      coefModel <- coef(res, s = "lambda.min")[-1] #remove intercept row
      
      #store coefficients for this repeat
      coefMat[,i] <- coefModel
      
      #calculate variance explained
      if(sum(coefModel !=0)){
      y.pred <- predict(res, s = "lambda.min", newx = X)
      #if there are no predictors, all y.pred will be the same so its not possible to calculate variance explained because the SD is 0
      varExp <- cor(as.vector(y),as.vector(y.pred))^2
      }else{ varExp <- NA}
      varExplain[i] <- ifelse(is.na(varExp), 0, varExp) 
      
     
      
    } else {
      fitlm<-lm(y~., data.frame(X))
      varExp <- summary(fitlm)$r.squared
      varExplain <- c(varExplain, varExp)
      
    }

  }
  #store all results 
  list(modelList = modelList, lambdaList = lambdaList, varExplain = varExplain, coefMat = coefMat)
}


lassoPlot06 <- function(lassoOut, geneMatrix, betaMatrix, freqCut = 1, coefCut = 0.01) {
  #object to hold all plots
  plotList <- list()
  
  #for each drug - stimuli combination, run the following:
  for (seaName in names(lassoOut)) {
    ###FOR THE BAR PLOT
    #extract mean coefficients for each drug - stimuli combination
    barValue <- rowMeans(lassoOut[[seaName]]$coefMat)
    #extract proportion of repeats for which each coefficient is significant
    freqValue <- rowMeans(abs(sign(lassoOut[[seaName]]$coefMat)))
    #filter out coefficients that don't meet freqCut and coefCut thresholds
    barValue <- barValue[abs(barValue) >= coefCut & freqValue >= freqCut] 
    #arrange the bar values in numerical order
    barValue <- barValue[order(barValue)]
    #if there are no sig coefficients, don't plot
    if(length(barValue) == 0) {
      plotList[[seaName]] <- NA
      next
    }
    
   
    ###FOR THE HEATMAP AND SCATTER PLOT
    #get feature matrix and response matrix to plot
    allData <- geneMatrix
    betaValue <- unlist(betaMatrix[seaName,])
    
    #get feature matrix for features with significant coefficients only
    tabValue <- allData[, names(barValue),drop=FALSE]
    ord <- order(betaValue)
    betaValue <- betaValue[ord]
    tabValue <- tabValue[ord, ,drop=FALSE]
    sampleIDs <- rownames(tabValue)
    tabValue <- as_tibble(tabValue)
    tabValue$Sample <- sampleIDs
    
    #annotate features as mutations, methylation cluster or IGHV, and apply different scaling     so that different colours can be used in plotting 
    
    #for mutations:
    matValue <- gather(tabValue, key = "Var",value = "Value", -Sample)
    matValue$Type <- "mut"
    
    #for methylation cluster
    matValue$Type[grep("Methylation",matValue$Var)] <- "meth"
    
    #for IGHV status
    matValue$Type[grep("IGHV.status",matValue$Var)] <- "ighv"
    
    #change the scale of the value so that IGHV, Methylation and Mutation do not overlap
    matValue[matValue$Type == "mut",]$Value = matValue[matValue$Type == "mut",]$Value + 10
    matValue[matValue$Type == "meth",]$Value = matValue[matValue$Type == "meth",]$Value + 20
    matValue[matValue$Type == "ighv",]$Value = matValue[matValue$Type == "ighv",]$Value + 30
    
    #change continous to catagorical
    matValue$Value <- factor(matValue$Value,levels = sort(unique(matValue$Value)))
    
    #arrange order of feature rows and columns in heatmap
    #heatmap rows should align with order of genetic coefficients
    matValue$Var <- factor(matValue$Var, levels = names(barValue))
    
    #sample columns should be ascending order according to value of coefficient of each patient, so that heatmap aligns with scatter plot below
    matValue$Sample <- factor(matValue$Sample, levels = names(betaValue))
    
    #update labels, keep factor levels
    matValue$Var <- revalue(matValue$Var, c("IGHV.status" = "IGHV status", "del11q" = "del(11q)", "del13q" = "del(13q)", "del17p" = "del(17p)", "trisomy12" = "trisomy 12")) 
    
    #MAKE PLOTS
    #plot the heatmap of genetic feature values
    p1 <- ggplot(matValue, aes(x=Sample, y=Var)) + 
      geom_tile(aes(fill=Value), color = "white") + #ghost white
      theme_bw()+
      scale_y_discrete(expand=c(0,0)) + 
      theme(axis.title.x = element_text( size=fontsize+4),
            axis.text.y=element_text(hjust=0, size=18, face="bold"), 
            axis.ticks=element_blank(),
            panel.border=element_rect(colour="gainsboro"),  
            plot.title=element_text(face="bold", size = 18, margin = margin(t = -5, b = 1)), 
            panel.background=element_blank(),
            panel.grid.major=element_blank(), 
            panel.grid.minor=element_blank()) + 
      xlab("Mutation status for each patient") + 
      ylab("") + 
      scale_fill_manual(name="Mutated", 
                              values=c(`10`= offwhite,  #WT
                                       `11`="#373A36", #Mutant
                                       `20`= offwhite, #LP
                                       `20.5`= "#707372", #IP
                                       `21` = "#A8A99E", #HP
                                       `30` = offwhite, #IGHV-U
                                       `31` = "#707372"), #IGHV-M
                                       guide="none") + 
            ggtitle(seaName)
    
    
    #Plot the bar plot on the left of the heatmap 
    barDF = data.frame(barValue, nm=factor(names(barValue),levels=names(barValue)))
    
    p2 <- ggplot(data=barDF, aes(x=nm, y=barValue)) + 
      geom_bar(stat="identity", 
               fill=ifelse(barValue<0,
                           palblues[6],palreds[8]), 
               colour="black", 
               size=0.3) +
      scale_x_discrete(expand=c(0,0.5))+ 
      scale_y_continuous(expand=c(0,0), n.breaks = 4)+ 
      coord_flip(ylim=c(-0.3,0.35))+ 
      theme(panel.grid.major=element_blank(), 
            panel.background=element_blank(), 
            axis.ticks.y = element_blank(),
            panel.grid.minor = element_blank(), 
            axis.text=element_text(size=fontsize, angle = 0, hjust = 0),
                axis.title = element_text(size=fontsize+4), 
            panel.border=element_blank()) +
      ylab("Size of coefficient") + 
      geom_vline(xintercept=c(0.5), 
                 color="black", 
                 size=0.6)
    
    #Plot the scatter plot of patient coefficient values under the heatmap
    scatterDF = data.frame(X=factor(names(betaValue), 
                                    levels=names(betaValue)), 
                           Y=unlist(betaValue))
    
    p3 <- ggplot(scatterDF, aes(x=X, y=Y)) + 
          geom_point(shape=21, 
                     fill="dimgrey", 
                     colour="#707372", #dark grey
                     size=1.2) + 
          theme_bw() +
          theme(panel.grid.minor=element_blank(), 
                panel.grid.major.x=element_blank(), 
                axis.ticks.x=element_blank(), 
                axis.text.y=element_text(size=fontsize),
                axis.title = element_text(size=fontsize+4),
                panel.border=element_rect(colour="dimgrey", size=0.1),
                panel.background=element_rect(fill="white")) +
  xlab(expression(paste("Patient-specific ", beta["int"])))
    
    
    #Assemble all the plots together

    # construct the gtable
    wdths = c(0.2, 1.5, 0.4, 1.3*ncol(matValue), 1.7, 0.2)
    hghts = c(0.3, 0.3, 0.0020*nrow(matValue), 0.2, 0.8, 0.3)*1.5
    gt = gtable(widths=unit(wdths, "in"), heights=unit(hghts, "in"))
    
    ## make grobs
    gg1 = ggplotGrob(p1)
    gg2 = ggplotGrob(p2)
    gg3 = ggplotGrob(p3)

    ## fill in the gtable
   
    #HEATMAP
    #5:1 = "PREDICTORS"
    gt = gtable_add_grob(gt, gtable_filter(gg1, "panel"), 3, 4) # add heatmap
    gt = gtable_add_grob(gt, gtable_filter(gg1, "panel"), 3, 4) #add legend
    gt = gtable_add_grob(gt, gtable_filter(gg1, "title"), 1, 4) #add title to plot
    gt = gtable_add_grob(gt, gtable_filter(gg1, "axis-l"), 3, 5) # variable names
    gt = gtable_add_grob(gt, gtable_filter(gg1, "xlab-b"), 2, 4) # axis title
    
    #BARPLOT
    gt = gtable_add_grob(gt, gtable_filter(gg2, "panel"), 3, 2) # add barplot
    gt = gtable_add_grob(gt, gtable_filter(gg2, "axis-b"), 4, 2) # y axis for barplot
    gt = gtable_add_grob(gt, gtable_filter(gg2, "xlab-b"), 2, 2) # y lab for barplot

    
    #SCATTER PLOT
    gt = gtable_add_grob(gt, gtable_filter(gg3, "panel"), 5, 4) # add scatterplot
    gt = gtable_add_grob(gt, gtable_filter(gg3, "xlab-b"), 6, 4) # x label for scatter plot
    gt = gtable_add_grob(gt, gtable_filter(gg3, "axis-l"), 5, 3) #  axis for scatter plot
    
   

    
    #plot
    plotList[[seaName]] <- gt
  }
  return(plotList)
}

makelegends <- function (legendFor, colors) 
{
    x = NULL
    y = NULL
    colors = colors[names(colors) %in% legendFor]
    nleg = length(colors)
    
    #edit these widths to change alignment with lasso plots in patchwork code
    wdths = c(0.4,2,2,2,1.5)
    hghts = c(2)
    
    gtl = gtable(widths=unit(wdths, "in"), heights=unit(hghts, "in"))
    n = 2
    if ("M" %in% names(colors)) {
        Mgg = ggplot(data = data.frame(x = 1, 
                                       y = factor(c("LP", "IP", "HP"), 
                                                  levels = c("LP", "IP", "HP"))), 
                     aes(x = x, y = y, fill = y)) + 
              geom_tile() + 
              scale_fill_manual(name = "Methylation cluster", 
                                values = setNames(colors[["M"]], 
                                                  nm = c("LP", "IP","HP"))) + 
              theme(legend.title = element_text(size = 12), 
                    legend.text = element_text(size = 12))
        
        gtl = gtable_add_grob(gtl, gtable_filter(ggplotGrob(Mgg), "guide-box"), 1, n)
        n = n + 1
    }
    
    if ("I" %in% names(colors)) {
        Igg = ggplot(data = data.frame(x = 1, y = factor(c("Unmutated", 
            "Mutated"), levels = c("Unmutated", "Mutated"))), 
            aes(x = x, y = y, fill = y)) + geom_tile() + scale_fill_manual(name = "IGHV", 
            values = setNames(colors[["I"]], nm = c("Unmutated", "Mutated"))) + 
            theme(legend.title = element_text(size = 12), 
                  legend.text = element_text(size = 12))
        gtl = gtable_add_grob(gtl, gtable_filter(ggplotGrob(Igg), 
            "guide-box"), 1, n)
        n = n + 1
    }
    
    if ("G" %in% names(colors)) {
        Ggg = ggplot(data = data.frame(x = 1, y = factor(c("Wild Type", 
            "Mutated"), levels = c("Wild Type", "Mutated"))), 
            aes(x = x, y = y, fill = y)) + geom_tile() + scale_fill_manual(name = "Gene", 
            values = setNames(colors[["G"]], nm = c("Wild Type", "Mutated"))) + 
            theme(legend.title = element_text(size = 12), 
                  legend.text = element_text(size = 12))
        gtl = gtable_add_grob(gtl, gtable_filter(ggplotGrob(Ggg), 
            "guide-box"), 1, n)
        n = n + 1
    }
    
    return(list(plot = gtl, width = sum(wdths), height = sum(hghts)))
}


```

```{r processData06}
# List of Cytokines, Drugs and treatments 
  ##Drugs
  thedrugs <- unique(df$Drug) 

  ##Cytokines
  thecytokines <- unique(df$Cytokine) 

  #Treatments
  #drugs
  drugtreatments <- list()
  drugtreatments <- 
    lapply(thedrugs, function(x){
      paste("treatment_drug", x, sep='')
      })
  
  
  #cytokines
  cytokinetreatments <- list()
  cytokinetreatments <- 
    lapply(thecytokines, function(y){
      paste("treatment_cytokine", y, sep='')
      })

df_complete <- df 

# Tidy up data frame to use in linear model
df <- 
  dplyr::filter(df,
                #use high concentrations only
                Drug_Concentration %in% c("High","None")) %>% 
  #select columns for linear model
  dplyr::select(PatientID, Drug, Cytokine, Log) %>%  
        
  #rename columns 
  plyr::rename(c("Drug"="treatment_drug", "Cytokine"="treatment_cytokine", "Log"="Viability"))


#patMeta
patMeta$treatment <- as.factor(patMeta$treatment)


#convert stain type to factor
ihc_patient_data$Stain <- factor(ihc_patient_data$Stain, levels = c("pSTAT6", "STAT6" , "pIRAK4"))

```

```{r preWork06, message = FALSE}

# define the base level per treatment_type as the "no"-treatment
df$treatment_drug <- as.factor(df$treatment_drug)
df$treatment_cytokine <- as.factor(df$treatment_cytokine)

df$treatment_drug %<>% relevel("DMSO")
df$treatment_cytokine %<>% relevel("No Cytokine")


# inspect design matrix of interaction model
# model.matrix(~treatment_drug + treatment_cytokine + treatment_drug:treatment_cytokine, df)

# fit model with interaction
fit <- lm(Viability ~ treatment_drug * treatment_cytokine, df)



#extract p values

pvalues <- summary(fit)$coefficients[,4]

#order as a dataframe
pvaldf <- as.data.frame(as.matrix(pvalues)) %>% 
          setDT(keep.rownames = TRUE)
          
#rename columns
colnames(pvaldf) <- c("treatment", "pvalue")

#filter out single agent treatments
singletreatments <- c(drugtreatments, cytokinetreatments, "(Intercept)")

pvaldf <- dplyr::filter(pvaldf, !treatment %in% singletreatments)


pvaldf <- 
  pvaldf %>% 
  #remove "treatment" prefix
  mutate_at(vars(treatment), funs(as.character(gsub("treatment_drug", "", .)))) %>% 
  mutate_at(vars(treatment), funs(as.character(gsub("treatment_cytokine", "", .))))
  
  #split drug:cytokine into two columns
pvaldf <-   
  data.frame(pvaldf, do.call(rbind, strsplit(pvaldf$treatment, split = ":", fixed = TRUE)))

#renove treatment column
pvaldf <- pvaldf[,c("pvalue","X1", "X2")]

#rename columns
colnames(pvaldf) <- c("pvalue","drug", "Cytokine")



```

## The impact of microenvironmental stimuli on drug response {#mapping-interactions}
### Linear modelling maps interactions between drugs and stimuli {#drug-stimulus-linear-model}
To begin the analysis, I first aimed to screen for cases where stimulus pathways specifically interacted with drug target pathways (Bruch and Giles et al. 2021).  I evaluated computational methods for quantifying this, and decided to apply linear modelling based on the following principle:

When a stimulus is individually applied to a CLL sample, the stimulus activates signalling cascades that modulate CLL viability and impact the rate of spontaneous apoptosis. When a stimulus is co-applied with a drug, the stimulus will continue to impact upon baseline viability, at the same time as the drug inhibits baseline viability. Assuming the drug and stimulus do not interact, the viability of the tumour cells with the combinatorial treatment will equate to the additive impact of both the compounds. 

In the case that there is some interaction between the stimulus pathway and drug target pathway, the resulting viability will not simply be additive. The difference between the additive, or expected viability, and the true measured viability, can be quantified by an **interaction factor**. Linear modelling aims to quantify this  **interaction factor**. [IS THIS CLEAR?]

Based on this principle, I fitted linear models to the viability data for each drug - stimulus combination, outlined in Equation \@ref(eq:drugCytInt) (Bruch and Giles et al. 2021). I used the `lm` function implemented in the `R` package `stats`[@R-base]. 

Equation \@ref(eq:drugCytInt) quantifies how the viability with any combination can be predicted:
    \begin{equation}
            log(V) = \beta_{drug}X_{drug} + \beta_{stimulus}X_{stimulus} + \beta_{int}X_{drug}X_{stimulus} + \epsilon
                                       (\#eq:drugCytInt)
    \end{equation}
    
_where $V$ is the predicted viability with a given treatment,_ $\beta_{drug}$, $\beta_{stimulus}$ _and_ $\beta_{int}$ _are coefficients for the drug, stimulus and combinatorial terms and_ $X_{drug}$ _and_ $X_{stimulus}$  _are indicator variables (0 or 1) for the presence or absence of a drug/stimulus._ $\epsilon$ _is the vector of model residuals. See also Methods section \@ref(drug-stimulus-linear-model-method). Equation from Bruch and Giles et al. 2021._

Here $\beta_{int}$  quantifies how the combined treatment effect differs from the sum of the individual treatments.  This approach identified 45 drug-stimulus combinations (out of 204), where $\beta_{int}$ had p < 0.05, highlighting the extent to which drug action can be modulated by cell-extrinsic signals _ex vivo_ (Bruch and Giles et al. 2021). 

### Drug - stimulus interactions can be categorised by their mode of action {#drug-stimulus-categories}
The interactions identified through linear modelling demonstrated different modes of action. I was most interested to identify stimuli which induce drug resistance, and drugs that may inhibit the pro-survival effects of a stimulus, which represent candidate drugs for targeting the microenvironment in the clinic. To establish which interactions may be of interest within a clinical context, the drug - stimulus interactions were classified into four categories (Bruch and Giles et al. 2021).  

The categories were firstly defined based on whether the interaction was antagonistic or synergistic, i.e. the stimulus / drug acted to oppose or reinforce the activity of the other. These were further divided  based on the sign of $\beta_{int}$ i.e. whether the combinatorial viability was higher (positive) or lower (negative) than would be expected based on additive effects (Figure \@ref(fig:drugStimulusDummyPLots)A). 

(ref:drugStimulusDummyPLots) (A) Graphical line plots representing typical response patterns for each of the four drug - stimulus interaction categories (types I - IV). x-axis shows treatment, y-axis shows log transformed viability values, facet labels indicate interaction type. Blue and black horizontal lines demonstrate effect of interaction on viability with combinatorial treatment. Blue line shows predicted viability based on additive effects alone and black line shows viability accounting for additive effects and interaction. (B) Histogram showing number of drug-stimulus interactions within each category, where p value for $\beta_{int}$ is <0.05.  See Methods section \@ref(drug-stimulus-linear-model-method) and \@ref(drug-stimulus-categories-method).  _Figure generated with Peter-Martin Bruch for the manuscript Bruch & Giles et al. 2021, and caption adapted from manuscript._


```{r drugStimulusDummyPLots, fig.cap='(ref:drugStimulusDummyPLots)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 8, fig.align="center", out.width = '100%', dev = 'cairo_pdf'}


dummydata <-
  as.data.frame(list(c(1,2,3,4),
                     c(0,0,0,0),
                     c(-0.6,-0.6,0,-0.1), 
                     c(0.6,0.6,0.1,-0.1), 
                     c(0.5,-0.5,0.8,-0.8)), 
                col.names=c("plot",
                            "coord1", 
                            "coord2", 
                            "coord3", 
                            "coord4")) %>% 
  as_tibble() %>% 
  
  mutate(expected = coord2 + coord3, measured = coord4) %>% 
  
  pivot_longer(cols = coord1:measured, names_to = c("plot"), values_to="y.value", names_repair="unique" ) %>% 
  
  dplyr::rename(Plot = plot...1, x.value = plot...2) %>% 
  
  mutate(x.value = factor(x.value, levels = c("coord1", "coord2", "coord3", "coord4", "expected", "measured"))) %>% 
  
  #add annotation of interaction type
  mutate(Plot = factor(Plot, 
                       labels = c("I", 
                                  "II", 
                                  "III", 
                                  "IV"))) %>%
  mutate(type = ifelse(Plot %in% c("I", "II"), "antagonism", "synergism")) %>%
  mutate(direction = ifelse(Plot %in% c("I", "III"), "positive", "negative"))

my_labeller <- as_labeller(c("positive"="Positive~\u03B2[int]", "negative"="Negative~\u03B2[int]", "antagonism"="Antagonism", "synergism" = "Synergism"),
                           default = label_parsed)

dummydata$direction <- factor(dummydata$direction, levels = c("positive","negative"))
dummydata$type <- factor(dummydata$type, levels = c("antagonism","synergism"))
                              

#plot dummy plot
dummyPlot<- 
  ggplot(dplyr::filter(dummydata, grepl('coord', x.value)), 
       aes(x=x.value, y=y.value, group = Plot, color = direction))+
  geom_hline(yintercept = 0, linetype=2, color="black")+
  geom_point(size=1)+
  geom_line(size=1)+
  facet_grid(type~direction, labeller = my_labeller )+
  geom_segment(data=dplyr::filter(dummydata, grepl('expected', x.value)), aes(x = 3.5, y = y.value, xend = 4.5, yend = y.value), color="blue", size=2) +
  geom_segment(data=dplyr::filter(dummydata, grepl('measured', x.value)), aes(x = 3.5, y = y.value, xend = 4.5, yend = y.value), color="black", size=2)+
  theme_bw() +
  t1 +
  xlab("") + 
  ylab("Log(Viability)") +
  coord_cartesian(ylim = c(-1, 1)) +
  scale_x_discrete(labels = c( "coord1" = "DMSO", "coord2" = "Drug","coord3" = "Stimulus","coord4" = "Drug +\nStimulus")) +
    theme(axis.ticks.x = element_blank(),axis.ticks.y = element_blank(),
          plot.tag=element_text(size = 20), 
          axis.text.x = element_text(angle = 45,
                                   hjust = 1, 
                                   vjust = 1)) +
  scale_y_continuous(breaks = c(-1,0, 1)) +
  scale_color_manual(values = c("#A6093D","#003DA5"))  + theme(legend.position = "none") +
  geom_text(x = 1, y = 0.75, aes(label=Plot))


fit_as_df <- 
  fit$coefficients %>% 
  as.matrix() %>% 
  as.data.frame() %>%
  setDT(keep.rownames = TRUE) %>% 
  as_tibble() %>% 
  dplyr::rename(Condition = rn, comb_coefficient = V1) %>% 
  dplyr::filter(grepl("drug", Condition)&grepl("cytokine", Condition)) %>% 
  mutate(Condition = gsub("treatment_drug", "", Condition)) %>% 
  mutate(Condition = gsub("treatment_cytokine", "", Condition)) %>% 
  separate(Condition, c("Drug", "Cytokine"), sep=":")

drug_fit_as_df <- 
  fit$coefficients %>% 
  as.matrix() %>%
  as.data.frame() %>%
  setDT(keep.rownames = TRUE) %>% 
  as_tibble() %>% 
  dplyr::rename(Drug = rn, drug_coefficent=V1) %>% 
  dplyr::filter(grepl("drug", Drug)&!grepl("cytokine", Drug)) %>% 
  mutate(Drug = gsub("treatment_drug", "", Drug))


cyt_fit_as_df <-
  fit$coefficients %>% 
  as.matrix() %>% 
  as.data.frame() %>%
  setDT(keep.rownames = TRUE) %>% 
  as_tibble() %>% 
  dplyr::rename(Cytokine=rn, cyt_coefficent=V1) %>% 
  dplyr::filter(!grepl("drug", Cytokine)&grepl("cytokine", Cytokine)) %>% 
  mutate(Cytokine = gsub("treatment_cytokine", "", Cytokine))


fit_combinations <-
  left_join(fit_as_df, drug_fit_as_df, by = "Drug") %>% 
  left_join(cyt_fit_as_df, by = "Cytokine") %>% 
  mutate(predicted_coefficient=cyt_coefficent+drug_coefficent) %>% 
  mutate(obs_coef=predicted_coefficient+comb_coefficient) %>%
  left_join(mutate(pvaldf,drug=as.character(drug),Cytokine=as.character(Cytokine)), by=c("Drug"="drug", "Cytokine"="Cytokine")) %>% 
  dplyr::filter(pvalue<=0.05) %>%
  mutate(Category_Symbol=case_when(
    comb_coefficient>=0&(obs_coef<cyt_coefficent|obs_coef<drug_coefficent)~ "I",  
    comb_coefficient>=0&(obs_coef>cyt_coefficent&obs_coef>drug_coefficent)~ "III",
    comb_coefficient<0&(obs_coef>cyt_coefficent|obs_coef>drug_coefficent)~ "II",  
    comb_coefficient<0&(obs_coef<cyt_coefficent&obs_coef<drug_coefficent)~ "IV"
    )) %>% 
  mutate(Category=case_when(
    Category_Symbol=="I"~ "Positive \nAntagonistic",  
    Category_Symbol=="III"~ "Positive \nSynergistic",
    Category_Symbol=="II"~ "Negative \nAntagonistic",  
    Category_Symbol=="IV"~ "Negative \nSynergistic"
    )) %>% 
  mutate( Category=factor(Category, levels=c("Positive \nAntagonistic","Negative \nAntagonistic","Positive \nSynergistic","Negative \nSynergistic")))
  
barPlot <- 
fit_combinations %>% 
  dplyr::group_by(Category) %>% 
  tally() %>% 
 
  
 
ggplot(aes(x=Category, y=n, fill=Category))+
  geom_bar(width = 1, stat = "identity")+
  t1+
  guides(fill="none")+
  xlab("")+ ylab("Number of drug - stimulus pairs")+
  scale_fill_manual(values = c("#A6093D","#003DA5","#A6093D","#003DA5")) + 
  theme(axis.text.x = element_text(angle = 45,
                                   hjust = 1, 
                                   vjust = 1))

dummyPlot + barPlot + plot_annotation(tag_levels = c("A"))

```

These categories encompassed the following interaction types. Type I defined positive antagonisms in which stimuli reversed drug action, leading to decreased drug efficacy and increased CLL viability.  This group highlights drug - stimulus interactions that could be relevant to treatment resistance pathways _in vivo_. Type II negative antagonisms highlighted cases where the drug inhibited the stimulus, reducing the effect of the stimulus on baseline viability, and decreasing CLL viability. These interactions include drugs that represent candidates to target microenvironmental signalling pathways _in vivo_. Type III and type IV interactions cover synergies, in which the drug and stimulus increase the effect of the other. Type III positive synergies led to increased CLL viability with the combinatorial treatment, whilst type IV negative synergies led to increased drug efficacy and lower viability. In the latter case, these may highlight cases where drug efficacy is dependent on external pathways, and indicate where co-culture approaches may be more appropriate in laboratory studies of _ex vivo_ drug action. Figure \@ref(fig:drugStimulusDummyPLots)B quantifies the number of interactions in each category. 


### Investigating specific drug-stimulus interactions indicates that INF$\gamma$ induces resistance to ibrutinib _in vitro_  {#drug-stimulus-interaction-examples}
The combined approach of fitting linear models and subsequently classifying these interactions represents an attempt to generate a comprehensive map of drug - stimulus interactions in CLL (Bruch and Giles et al. 2021, Figure \@ref(fig:drugStimulusIntMapt)). The aim is that this resource may facilitate further work into these drug - stimulus interactions, such as the work outlined in section \@ref(il4-ibrutinib).

(ref:drugStimulusIntMapt) "Heatmap of $\beta_{int}$ values for all drug - stimulus combinations where p for $\beta_{int}$ <0.05, annotated with interaction type (I - IV). Scale indicates size and sign of $\beta_{int}$. Rows and columns clustered according to hierarchical clustering."  See Methods section \@ref(drug-stimulus-linear-model-method) and \@ref(drug-stimulus-categories-method). _Figure generated with Peter-Martin Bruch for the manuscript Bruch & Giles et al. 2021, and caption taken from manuscript._

```{r drugStimulusIntMapt, fig.cap='(ref:drugStimulusIntMapt)', message = FALSE,  echo = FALSE, fig.width = 7, fig.height = 7, fig.align="center", out.width = '80%', dev = 'cairo_pdf'}

#extract beta interaction values
betavalues <- summary(fit)$coefficients[,1]

#create matrix of beta values 
betadf <- as.data.frame(as.matrix(betavalues)) %>%
          setDT(keep.rownames = TRUE)

#rename columns
colnames(betadf) <- c("treatment", "betavalue")

#remove beta values for drugs and cytokines alone
betadf <- dplyr::filter(betadf, !treatment %in% singletreatments)

#remove treatment prefix
betadf <- betadf %>% 
  mutate_at(vars(treatment), funs(as.character(gsub("treatment_drug", "", .)))) %>% 
  mutate_at(vars(treatment), funs(as.character(gsub("treatment_cytokine", "", .))))

#split drug:cytokine by colon
betadf <- data.frame(betadf, do.call(rbind, strsplit(betadf$treatment, split = ":", fixed = TRUE)))

#select columns of interest
betadf <- betadf[,c("betavalue","X1", "X2")]

#rename columns 
colnames(betadf) <- c("betavalue","drug", "Cytokine")

#set significance
a <- 0.05

#make a matrix of beta values, where  beta = 0 if corresponding p val is > than significance threshold 
pvalmat <- xtabs(pvalue~drug+Cytokine, data=pvaldf)
bvalmat <- xtabs(betavalue~drug+Cytokine, data=betadf)

#check in same order
bvalmat <- bvalmat[,colnames(pvalmat)] 
bvalmat <- bvalmat[rownames(pvalmat),]

bvalmat[pvalmat >= a] = 0

#transform matrix 
bvalmat <- t(bvalmat)

tree <- 
  pheatmap(bvalmat, silent = TRUE)


cyt_order <- rownames(bvalmat[tree$tree_row[["order"]],])

drug_order <- colnames(bvalmat[,tree$tree_col[["order"]]])



  #append p values
  left_join(betadf, pvaldf, by = c("drug", "Cytokine")) %>% 
  #make beta value 0 if p value is < sig
  mutate(betavalue = ifelse(pvalue<a, betavalue, 0)) %>% 
  #put drugs in order of clustered heatmap
  mutate(drug=factor(drug, levels = drug_order)) %>% 
  
  #put stimuli in order of clustered heatmap
  mutate(Cytokine=factor(Cytokine, levels = rev(cyt_order))) %>% 
  
  #make heatmap with ggplot
  ggplot(aes(x=drug, y=Cytokine))+
  
  geom_tile(aes(fill=betavalue),color = "grey")+
  #set colour scale
  scale_fill_gradientn(colors=c("#003DA5",  "white",  "#A6093D"), limits=c(-0.6,0.6))+
  #set theme
  t2+
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5),
        axis.title.x = element_blank(),
        axis.title.y = element_blank(),
        axis.ticks = element_blank(),
        axis.ticks.x =element_blank(),
        panel.background = element_blank(), 
        axis.line.x = element_blank(),
        axis.line.y = element_blank(),
        legend.key.height=unit(2.5, "cm"))+
  labs(fill = expression(paste(beta["int"], "-value")))+
  #add category
  geom_text(data = fit_combinations, aes(x=Drug, label=Category_Symbol) )+
  scale_y_discrete(labels=c("TGF-b1"="TGF-\u03B21", 
                            "sCD40L+IL-4"="sCD40L + IL4", 
                            "IL-1b"="IL1\u03B2", 
                            "IL-4"="IL4", 
                            "IL-6"="IL6",
                            "IL-15"="IL15",
                            "IL-10"="IL10", 
                            "IL-21"="IL21",
                            "IL-2"="IL2", 
                            "Interferon gamma"= "Interferon \u03B3",
                            "SDF-1a"="SDF-1\u03B1"))
  


```

Having generated an overview of drug - stimulus interactions, I next examined each interaction within each category with the aim of identifying the most clinically interesting examples (Figure \@ref(fig:drugStimulusIntEx), (Bruch and Giles et al. 2021)). 

**Positive antagonistic** Interactions categorised as positive antagonistic were the most common i.e. the stimulus reversed drug action. This group included known resistance mechanisms such as the inactivation of ibrutinib by IL4 stimulation (Figure \@ref(fig:drugStimulusIntEx)A)[@AguilarHernandez2016], indicating that the modelling and classification approach  successfully recapitulated established drug - stimulus interactions (Bruch and Giles et al. 2021).  

Additionally, this analysis demonstrated the broader significance of IL4-induced resistance, beyond what previous work has shown. IL4 showed the highest number of positive antagonistic interactions amongst all of the stimuli and inhibited a range of drugs, including BCR inhibitors and chemotherapeutics. This is investigated in more detail in section \@ref(IL4resistance) below.

In addition to IL4,  IFN$\gamma$ showed the second largest number of positive antagonistic interactions. IFN$\gamma$ stimulation induced resistance to BCR inhibition by ibrutinib and idelalisib, and to the chemotherapeutic Nutlin-3a (MDM2) (Figure \@ref(fig:drugStimulusIntEx)B, Bruch and Giles et al. 2021), and may represent a novel targetable resistance mechanism (Bruch and Giles et al. 2021). 

IL4 and IFN$\gamma$ induced resistance to common pathways and thus I stipulated that both stimuli may share the same mechanism of action. This was supported by the observations in section \@ref(multivariate-gene-stimulus-assosciations), in which I demonstrated that IL4 showed a smaller increase in viability in cells that had mutations in Ras-Raf-MEK-ERK, and that inhibition of p38 MAPK activity with ralimetinib increased the pro-survival effect of IL4 stimulation. IFN$\gamma$ demonstrated similar behaviour whereby p38 MAPK inhibition increased the pro-survival effect of stimulating this pathway (Appendix Figure \@ref(fig:ifngamma)).  Collectively these observations suggest that IL4 and IFN$\gamma$ may operate via a common pro-survival mechanism, which is negatively regulated by activity of MAPK signalling. [CHECK]

(ref:drugStimulusIntEx)  Line plots showing examples of drug-stimulus interactions. x-axis indicates treatment, y-axis shows log transformed viability values, with matching samples linked across treatments. Black horizontal lines represent  predicted viability for each treatment, using coefficients from linear model fit. For combinatorial treatment, blue line indicates predicted viability based on additive effects alone, and black line indicates predicted viability accounting for additive effects plus interaction. See Methods section \@ref(drug-stimulus-linear-model-method).  _Figure generated with Peter-Martin Bruch for the manuscript Bruch & Giles et al. 2021, and caption taken from manuscript._

```{r drugStimulusIntEx, fig.cap='(ref:drugStimulusIntEx)', message = FALSE,  echo = FALSE, fig.height=10, fig.width = 13, fig.align="center", out.width = '90%', dev = 'cairo_pdf'}

#update labelling
fit_combinations <-
  fit_combinations%>% 
  mutate(Category=case_when(
    Category_Symbol=="I"~ "Positive Antagonistic\n",  
    Category_Symbol=="III"~ "Positive Synergistic\n",
    Category_Symbol=="II"~ "Negative Antagonistic\n",  
    Category_Symbol=="IV"~ "Negative Synergistic\n"
    )) %>% 

  mutate(Category=factor(Category, levels=c("Positive Antagonistic\n","Negative Antagonistic\n","Positive Synergistic\n","Negative Synergistic\n")))

#get lists of drugs and cytokines without baseline treatments
thedrugs.only <- thedrugs %>% setdiff("DMSO") %>% sort()
thecytokines.only <- thecytokines %>% setdiff("No Cytokine")%>% sort()

# get a list of interactions for which p value of beta interaction is significant
sign_conditions <-
  pvaldf %>% 
  dplyr::filter(pvalue<=0.05) %>% 
  mutate(condition = paste0("treatment_drug:", 
                            drug, 
                            ":treatment_cytokine:",
                            Cytokine))

gg <- vector(mode = "list", length = length(sign_conditions$condition))
names(gg) <- sign_conditions$condition


plotList <- 
lapply(names(gg), function(i){
  
  #preparations for the interaction plot
  drug <- dplyr::filter(sign_conditions, condition == i)$drug
  drugTreat <- paste("treatment_drug", drug, sep = '')
      
  cyt <- dplyr::filter(sign_conditions, condition == i)$Cytokine
  cytokineTreat <- paste("treatment_cytokine", cyt, sep = '')
      
      
  Category <- fit_combinations[which(fit_combinations$Drug==drug &
                               fit_combinations$Cytokine==cyt),]$Category
  
  #extract coefficients to plot

  ## baseline effect
  baseline <- fit$coefficients["(Intercept)"]

  ## single treatment effect for type_1 = "a"
  single_1 <- fit$coefficients[drugTreat]

    ## single treatment effect for type_2 = "b"
    single_2 <- fit$coefficients[cytokineTreat]

    # interaction Viability
    interaction <- fit$coefficients[paste(drugTreat, cytokineTreat, sep=':')]

      
    ## observed single treatment values
    observed_single_1 <- baseline + single_1 
    observed_single_2 <- baseline + single_2

    ## observed double treatment value
    observed_double <- baseline + single_1 + single_2 + interaction

    ## predicted double treatment value (based only on single effects)
    predicted_double <- baseline + single_1 + single_2


    #prepare labels and aesthetics for plotting
    # y_label <- case_when(cyt=="Interferon gamma"~"Interferon\ngamma",
    #                      cyt=="soluble anti-IgM"~"soluble\nanti-IgM",
    #                      TRUE~cyt)

    xlabels <- c("DMSO", drug, 
                 case_when(cyt=="IL-4"~"IL4",
                           cyt=="Interferon gamma"~"Interferon \u03B3",
                           cyt=="sCD40L+IL-4"~"sCD40L \n+ IL4",
                           cyt=="soluble anti-IgM"~"soluble\nanti-IgM",
                           TRUE~cyt),
                paste(drug, " +\n",
                       case_when(cyt=="IL-4"~"IL4",
                           cyt=="Interferon gamma"~"Interferon \u03B3",
                           cyt=="sCD40L+IL-4"~"sCD40L + IL4",
                           TRUE~cyt),
                sep=""))
      
    segment_size <- 2
    
    
    plotTab <-
      df %>%
      
      dplyr::filter(treatment_drug %in% c("DMSO" , drug)) %>%

      dplyr::filter(treatment_cytokine %in% c("No Cytokine", cyt)) %>%

      dplyr::mutate(treatment_combination = 
                  #if baseline treatment
                  ifelse(treatment_drug=="DMSO" & 
                         treatment_cytokine=="No Cytokine",
                         "DMSO",
                         #if drug only treatment
                         ifelse(treatment_drug == drug & 
                                treatment_cytokine =="No Cytokine",
                                paste0("Drug=", drug),
                                #if cytokine only treatment
                                ifelse(treatment_drug == "DMSO" & 
                                       treatment_cytokine == cyt,
                                       paste0("Cytokine=", cyt),
                                       #otherwise combinatorial treatment
                                       paste0("Drug=", drug, "\nCytokine=", cyt))))) %>%
  #make treatment_combination as a factor
  mutate(treatment_combination = factor(treatment_combination, 
                                        levels=c("DMSO", 
                                                 paste0("Drug=", drug),
                                                 paste0("Cytokine=", cyt),
                                                 paste0("Drug=", drug, "\nCytokine=", cyt)
                                                 )))
  #make plot
        
        

  ggplot(plotTab, aes(treatment_combination, Viability, group=PatientID)) +
  
  geom_hline(yintercept = 0, linetype=2, color="black")+

  lemon::geom_pointline( size=1,   
                         colour=ifelse(interaction > 0,
                                       "#A6093D","#003DA5"), alpha=0.3) +
  #add predicted viability with control treatment
  geom_segment( size=segment_size,
                aes(x=1-.2, xend=1+.2, y=baseline, yend=baseline),
                colour = "black") +
  #add predicted viability with drug only treatment
  geom_segment( size=segment_size,
                aes(x=2-.2, xend=2+.2, y=observed_single_1,yend=observed_single_1),
                colour="black") +
  #add predicted viability with cytokine only treatment
  geom_segment( size=segment_size,
                aes(x=3-.2, xend=3+.2, y=observed_single_2, yend=observed_single_2),
                colour="black") +
  #add predicted viability with both treatments
  geom_segment( size=segment_size,
                aes(x=4-.2, xend=4+.2, y=observed_double, yend=observed_double),
                colour="black") +
  #add predicted viability with both treatments, without interaction
  geom_segment( size=segment_size,
                aes(x=4-.2, xend=4+.2, y=predicted_double, 
                    yend=predicted_double), color="blue") + 
 
   #add category and combination as title
    
    
    
  ggtitle(paste( Category, drug, " & ", 
                 case_when(cyt=="IL-4"~"IL4",
                           cyt=="Interferon gamma"~"Interferon \u03B3",
                           cyt=="sCD40L+IL-4"~"sCD40L + IL4",
                           TRUE~cyt), sep="")) + 
  
  #add theme
  t2 + theme(axis.text.x = element_text(size=fontsize+5)) +
    theme(plot.tag=element_text(size = 30))+
  scale_x_discrete(labels = xlabels) +
  ylab("Log (Viability)") + 
  xlab("")


  } ) 

names(plotList) <- names(gg)

design1 <-"
  12
  34
  56"
  
plotList$`treatment_drug:Ibrutinib:treatment_cytokine:IL-4` +
  
plotList$`treatment_drug:Ibrutinib:treatment_cytokine:Interferon gamma` + 
  
plotList$`treatment_drug:Pyridone-6:treatment_cytokine:sCD40L+IL-4` +
  
plotList$`treatment_drug:Ralimetinib:treatment_cytokine:Interferon gamma`+
  
plotList$`treatment_drug:Ibrutinib:treatment_cytokine:CpG ODN` +
  
plotList$`treatment_drug:Luminespib:treatment_cytokine:soluble anti-IgM` +  plot_annotation(tag_levels=c("A"))  + plot_layout(design=design1) 
  
```


**Negative antagonisms** There were six cases of negative antagonisms, whereby drug action inhibited the pro-survival effect of the stimulus. Such cases could potentially be used to target micronenivronmental signals _in vivo_. As IL4 appeared to be the most potent conferrer of drug resistance, it was of greatest interest to identify drugs that may inhibit this pathway. For example, the Pan-JAK inhibitor pyridone-6 inhibited the increase in viability with sCD40L + IL4 stimulation  (Figure \@ref(fig:drugStimulusIntEx)C, Bruch and Giles et al. 2021).

**Positive synergisms** The model identified a single positive synergism, in which IFN$\gamma$ treatment in combination with ralimetinib, a p38 MAPK inhibitor, stimulated a large increase in viability that was not observed in each of the single treatments (Figure \@ref(fig:drugStimulusIntEx)D, [@Giles2021]). As discussed above, this finding suggests a potential inhibitory effect of p38 MAPK activity on signalling via IFN$\gamma$ (Bruch and Giles et al. 2021).

**Negative synergisms** 16 drug - stimulus combinations demonstrated negative synergistic interactions. This implied that the efficacy of the drug was somehow dependent upon, or increased by, activation of the stimulus pathway. For example, luminespib, a HSP90 inhibitor, showed higher efficacy with a number of stimuli, including soluble anti-IgM, indicating that _ex vivo_ measurements of luminespib action are likely to be affected by the presence or absence of microenvironmental signals (Figure \@ref(fig:drugStimulusIntEx)F, Bruch and Giles et al. 2021). 

Collectively, these results represent a comprehensive and systematic study of the influence of the microenvironment on drug efficacy in CLL. This work highlights key resistance pathways, strategies for targeting microenvironmental resistance, and underlines important pathways in _ex vivo_ studies of drug efficacy (Bruch and Giles et al. 2021). 

This work  aims to serve as a resource, to provide a basis for further follow-up studies. All drug - stimulus combinations and interactions can be explored on the [online shiny app](https://www.imbi.uni-heidelberg.de/dietrichlab/CLL_Microenvironment/), and the whole analysis can be replicated, or adapted, from the [online code repository](https://github.com/Huber-group-EMBL/CLLCytokineScreen2021), published alongside Bruch and Giles et al. 2021. In this thesis, I have selected IL4-induced resistance to BCR inhibition and ibrutinib to investgiate in more detail (see below, section \@ref(il4-ibrutinib).

## Genetic modulators of drug response {#mapping-genetic-modulators}
### Univariate analysis identifies trisomy 12 as a key modulator of drug response {#univariate-gene-drug-associations}
I next aimed to screen for cases where molecular features modulated drug responses. Previous work in our lab [@JCIpaper] has generated the largest survey of molecular determinants of response in CLL to date.  Here I repeat a similar analysis, as it was valuable to establish the effects of mutations independently of stimuli on drug response within my dataset before investigating the impact of mutations on drug - stimulus interactions. 

I performed a comprehensive survey of genetic determinants of drug response, to establish the effects of mutations independently of stimuli on drug response. I used the genetic data, including all mutations, copy number variants and IGHV status for which there were greater than three cases in the cohort (n = 54), to perform t-tests to identify molecular features that impact on drug response (Figure \@ref(fig:drugGeneIntMap)). 

This analysis recapitulated the findings of previous work [@JCIpaper]. For example IGHV-U CLL samples responded more strongly to BCR inhibition by ibrutinib (BTK) and idelalisib (Pi3K), and TP53 and del(17p) mutations conferred resistance to nutlin-3a (MDM2) and fludarabine (Purine analogue). 

(ref:drugGeneIntMap) Plot showing BH-adjusted p values from Student's t-tests (two-sided, with equal variance), for all tested drug-gene associations. Tests performed for IGHV status, somatic mutations and copy number aberrations (n = 54). Each coloured circle represents a gene-stimulus association meeting 10% FDR cut off. See Methods section \@ref(univariate-gene-stimulus-associations-method) _Figure from Bruch and Giles et al. 2021._ 

```{r drugGeneIntMap, fig.cap='(ref:drugGeneIntMap)', message = FALSE,  echo = FALSE, fig.height=6, fig.width = 7, fig.align="center", out.width = '70%', dev = 'cairo_pdf'}

df_univariate <- df_complete %>% filter(Drug_Concentration=="High")

##################### List of mutations with >2 positive cases #############################
selected_mutations <- 
  patMeta %>% 
  mutate(Ras_Raf = as.factor(ifelse(BRAF == 1 | KRAS == 1| NRAS == 1, 1, 0))) %>% 
  dplyr::select(-BRAF, -KRAS, -NRAS) %>% 
  dplyr::select( -gender, 
                 -diagnosis, 
                 ) %>%
  pivot_longer(-PatientID, 
               names_to = "Genetic_Alteration", 
               values_to = "alteration_value") %>% 
  dplyr::filter(alteration_value %in% c(1, "U")) %>% 
  dplyr::group_by(Genetic_Alteration) %>% 
  dplyr::count() %>% 
  dplyr::filter(n>2) %>% 
  dplyr::select(Genetic_Alteration) %>% 
  unlist()

##################### t tests and p-value adjustment ###############################  
p_values <-
  filter(df_univariate, Cytokine == "No Cytokine", Drug != "DMSO") %>%
  ## Select columns from screening data 
  dplyr::select(PatientID, Log, Drug) %>% 
  
  ## Join Screening data with metadata
  left_join(patMeta, by = "PatientID") %>%
  
  ## Add Ras/Raf column, remove single columns
  mutate(Ras_Raf=as.factor(ifelse(BRAF == 1 | KRAS == 1 | NRAS == 1, 1, 0))) %>% 
  dplyr::select(-BRAF, -KRAS, -NRAS) %>% 
## remove unused columns from metadata
  dplyr::select( -gender, 
                 -diagnosis, 
                 -treatment) %>% 
  
  ## transform data to long format
  pivot_longer(cols=c(-PatientID, -Log, -Drug), 
               names_to = "Genetic_alt", 
               values_to = "alt_value") %>% 
  
  ## filter to selected mutations (see above)
  dplyr::filter(Genetic_alt %in% selected_mutations) %>% 
  
  
  ## group by Drug and Genetic alteration  
  dplyr::group_by(Drug, Genetic_alt) %>% 
  
  ## Perform t.test on every combination of Cytokine and genetic alteration
  do(tidy(t.test(Log ~ alt_value, data = ., var.equal = T))) %>% 
  
  ## ungroup before adjusting p-value
  ungroup() %>% 
  
  ## adjust p-values to multiple testing using BH method 
  mutate(adj.p.value=p.adjust(p.value, method = "BH"))
  
################################## Order of cytokines by descending significance ############################
Drug_order <-
  p_values %>% 
  dplyr::group_by(Drug) %>% 
  dplyr::arrange(adj.p.value) %>% 
  dplyr::filter(row_number() == 1) %>% 
  ungroup() %>% 
  dplyr::arrange(adj.p.value) %>% 
  dplyr::select(Drug) %>% 
  unlist()

Genetic_alt_order <-
  p_values %>% 
  dplyr::group_by(Genetic_alt) %>% 
  dplyr::arrange(adj.p.value) %>% 
  dplyr::filter(row_number() == 1) %>% 
  ungroup() %>% 
  dplyr::arrange(adj.p.value) %>% 
  dplyr::select(Genetic_alt) %>% 
  unlist()

############################# Define FDR cutoff #################################
fdr = 0.1
  
############################################ Plot ########################################
p_values %<>% 
  mutate(Drug=factor(Drug, levels = Drug_order)) %>% 
  mutate(Genetic_alt=factor(Genetic_alt, levels = Genetic_alt_order))
  
 

 ggplot(dplyr::filter(p_values, adj.p.value>fdr), 
         aes(x = Drug, y = -log10(adj.p.value))) +
  geom_point(color = "lightgrey", size = 3) +
  geom_beeswarm(data = dplyr::filter(p_values, adj.p.value <= fdr), 
                aes( color=Genetic_alt), size = 3, cex = 1.7) +
  ##FDR line  
  geom_hline(yintercept = -log10(fdr),linetype = "dashed", size=0.3)+
  
  ##Main Theme
  t1 +
  
  theme(axis.text.x = element_text(angle = 45,
                                   hjust = 1, 
                                   vjust = 1),
        axis.title.x=element_blank()) +
  
   ##Legend Theme
  theme(legend.position=c(0.6,0.75), 
        legend.title = element_text(face='bold', hjust = 0, size=13), 
        legend.key = element_blank(),  
        legend.text = element_text(size=11)) +
   guides(colour = guide_legend(nrow=7, title = "Mutations"), 
         shape = guide_legend(ncol = 1)) +
   scale_color_manual(name = "Mutations", 
                      values = c(colors, 
                                 "#D0DEBB",
                                 "#20b2aa",
                                 "#9300ff",
                                 "#955c20",
                                 "#e9c26e",
                                 "#a5dd86"),
                      labels=c("IGHV.status"="IGHV status",
                              "del9p"="del(9p)", 
                              "trisomy12"="trisomy 12", 
                              "gain17q"="gain(17q)", 
                              "gain2p"="gain(2p)",
                              "gain19p"="gain(19p)",
                              "gain19q"="gain(19q)",
                              "del7q"="del(7q)",
                              "del9q"="del(9q)", 
                              "del4p"= "del(4p)", 
                              "SPEN"="SPEN", 
                              "del11q"="del(11q)", 
                              "del1q" = "del(1q)",
                              "Ras_Raf" = "NRAS/KRAS/BRAF")) +
   scale_y_continuous(expression("BH-adjusted  "* italic(p)*"-value"), 
                      breaks=seq(0, 10, 5),
                      labels=math_format(expr=10^.x)(-seq(0,10,5))) 

```

In addition, this analysis highlighted the broad impact of trisomy 12 on drug response _ex vivo_ (Figure \@ref(fig:drugGeneIntEx)). While trisomy 12 CLL demonstrates higher proliferative capacity, tumours with trisomy 12 are more treatable due to higher sensitivity to chemotherapeutics and ibrutinib. This was reflected in the _in vitro_ data in this study; the presence of trisomy 12 increased sensitivity to 8 / 12 drugs. 

These included increased sensitivity to p38 MAPK inhibition by ralimetinib, agreeing with previous work demonstrating higher trisomy 12 sensitivity with a range of MEK and ERK inhibitors and suggesting an essential role for MEK/ERK signalling in trisomy 12 CLL [@JCIpaper]. 

I also noted that trisomy 12 CLL showed increased sensitivity to IBET-762, a bromodomain inhibitor that has been shown to suppress transcriptional responses to cytokine signalling via JAK-STAT [@Chan2015]. The results in section \@ref(trisomy12-modulator) indicated that trisomy 12 enhances responses to some cytokines. I was interested to see whether bromodomain inhibition may provide a strategy to target this feature of trisomy 12 samples. I decided to investigate the impact of IBET-762 treatment on the trisomy 12 transcription factor signature determined in Figure \@ref(fig:tri12diffTF).  I made use of an additional ATACseq dataset consisting of four CLL PBMC samples each treated  with  IBET-762  and  DMSO  as control. I visualised inferred TF activity for trisomy 12 signature TFs, calculated using `diffTF` [@Berest2019]. All nine TFs showing  higher  accessibility  in  trisomy  12 CLL (Figure \@ref(fig:tri12diffTF)) exhibited  decreased  accessibility  upon  treatment with IBET-762  (Appendix Figure \@ref(ibetatac)). This result suggests that  bromodomain inhibition may represent a potential target in CLL patients with trisomy 12 (Bruch and Giles et al. 2021). [Does it make sense to include this here, or is it a distraction from the main story?]


(ref:drugGeneIntEx) Beeswarm-boxplots showing control-normalised log transformed viability values after treatment with (A) Ralimetinib (p38 MAPK) and (B) IBET-762 (BRD2/3/4), stratified by trisomy 12.

```{r drugGeneIntEx, fig.cap='(ref:drugGeneIntEx)', message = FALSE,  echo = FALSE, fig.height=5, fig.width = 7.5, fig.align="center", out.width = '70%', dev = 'cairo_pdf'}

df_patmeta <- left_join(df_univariate, patMeta, by = "PatientID")

ibet_tri12 <- 
df_patmeta %>% 
  dplyr::filter(Drug=="I-BET 762", Drug_Concentration == "High", Cytokine == "No Cytokine") %>%
      filter(!is.na(trisomy12)) %>%
         
  ggplot(aes(x=trisomy12,y=Log,color=(trisomy12)))+
  geom_hline(yintercept = 0)+
  geom_boxplot()+
  geom_beeswarm(cex=1.5) +
  guides(color="none", shape="none")+
  stat_compare_means(method = "t.test",
                     label.x.npc = "center", 
                     size=3)+
  xlab("trisomy 12") +
  ylab("Log(Viability)") +
  ggtitle("IBET-762") +
  scale_color_manual(values=c(colors[1], colors[2])) + 

  t2


ralimetinib_tri12 <- 
df_patmeta %>% 
  dplyr::filter(Drug=="Ralimetinib", Drug_Concentration == "High", Cytokine == "No Cytokine") %>%
      filter(!is.na(trisomy12)) %>%
         
  ggplot(aes(x=trisomy12,y=Log,color=(trisomy12)))+
  geom_hline(yintercept = 0)+
  geom_boxplot()+
  geom_beeswarm(cex=1.5) +
  guides(color="none", shape="none")+
  stat_compare_means(method = "t.test",
                     label.x.npc = "center", 
                     size=3)+
  xlab("trisomy 12") +
  ylab("Log(Viability)") +
  ggtitle("Ralimetinib") +
  scale_color_manual(values=c(colors[1], colors[2])) + 

  t2

ralimetinib_tri12 + ibet_tri12
```


## The modulatory effect of mutations on drug - stimuli interactions {#drug-stimulus-gene-interactions} 
### Patient - specific linear modelling identifies drug - stimulus - gene interactions 
Sections \@ref(mapping-interactions) & \@ref(mapping-genetic-modulators) explore the effects of microenvironmental signals and molecular features on drug response independently. I next aimed to investigate their collective effect on _in vitro_ drug efficacy in CLL, by quantifying the extent to which genetic driver mutations modulated the interactions between stimuli and drugs (Bruch and Giles et al. 2021).

I began by conceptualising methods to quantify the collective effect of the drugs, stimuli and genetic features on CLL viability and biology.  I decided to adapt the linear model in Equation \@ref(eq:drugCytInt), by fitting this model in a patient sample - specific manner (Bruch and Giles et al. 2021):  

\begin{equation}
\begin{aligned}
log(V) ={} & beta_{drug}X_{drug} + \beta_{stimulus}X_{stimulus} + \beta_{patient}X_{patient} \\
           & \beta_{drug-stimulus}X_{drug}X_{stimulus} \\+ \beta_{drug-patient}X_{drug}X_{patient} + \beta_{stimulus-patient}X_{stimulus}X_{patient} \\
           & \beta_{int}X_{drug}X_{stimulus}X_{patient} + \epsilon
           (\#eq:drugCytGeneInt)
\end{aligned}
\end{equation}
    
_where $V$ is the predicted viability of a patient sample with a given treatment,_ $\beta_{drug}$,  $\beta_{stimulus}$, $\beta_{patient}$, $\beta_{drug-stimulus}$, $\beta_{drug-patient}$, $\beta_{stimulus-patient}$ _and_ $\beta_{int}$ _are regression coefficients for the drug, stimulus, patient sample and combinatorial terms and_ $X_{drug}$, $X_{stimulus}$ _and_ $X_{patient}$ _are indicator variables (0 or 1) for the presence or absence of a drug/stimulus/patient sample._ $\epsilon$ _is the vector of model residuals. See also Methods section \@ref(drug-stimulus-gene-interactions-method). Equation from Bruch and Giles et al. 2021._

With these patient sample-specific $\beta_{int}$ terms, it was possible to search for associations between the size of $\beta_{int}$ and genetic features. The aim was to screen for molecular features that increased or decreased the size of a drug - stimulus interactions, using multivariate regression with L1 (lasso) regularisation.

To assemble the inputs for the model, first the response matrix was composed of the sample - specific $\beta_{int}$ values for each drug-stimulus combination. To generate the feature matrix (137 samples versus 40 features), I excluded genetic features for which >20\% of the values were missing, and  patient samples with incomplete annotation. As predictors, I included genetic mutations and CNVs (p = 39) and IGHV status (coded as 0-1). I ran lasso regression, as implemented in the `R` package `glmnet`[@R-glmnet], using three-fold cross-validation with misclassification error as loss. This approach identified genetic predictors of the size of drug - stimulus interactions,  where the predictors represent the mean coefficient values that were selected in at least 90% of 30 bootstrapped repeats. 

(ref:drugCytGeneIntMap) Heatmap summarising genetic predictors of drug - stimulus interactions. Each row depicts a single drug - stimulus combination, and each coloured tile indicates that $\beta_{int}$ for given drug and stimulus combination is modulated by corresponding genetic feature. Thus each row represents the output of a single model fit (as in Figure \@ref(fig:FluCpGExA)). Colour of tile indicates size and sign of coefficient assigned to genetic feature, where a positive coefficient corresponds to a more positive $\beta_{int}$ if the feature is present. Only top 8 most commonly selected genetic features are shown. Drug - stimulus combinations with no genetic predictors of $\beta_{int}$ amongst top 8 shown are omitted for clarity. See Methods section \@ref(drug-stimulus-gene-interactions-method). _Figure from Bruch and Giles et al. 2021. Peter-Martin Bruch contributed to figure design._

```{r drugCytGeneIntMap, fig.cap='(ref:drugCytGeneIntMap)', message = FALSE,  echo = FALSE, fig.height=18, fig.width = 9, fig.align="center", out.width = '50%', dev = 'cairo_pdf', fig.pos = "hp"}

#cluster coeff.mat
fit <-
  coeff.mat %>% 
  dist() %>% 
  hclust()
  
order_comb <- rownames(coeff.mat)[fit$order]

#Put matrix into long format for ggplot
coeff.long.mat <- coeff.mat %>% 
  as.data.frame() %>% 
  tibble::rownames_to_column("int") %>% 
  tidyr::gather(key = "gene", value = "coeff", -int)

#get top 8 coefficients 
order_alt <- 
  coeff.long.mat %>% 
  dplyr::filter(coeff!=0) %>% 
  dplyr::group_by(gene) %>% 
  dplyr::count(sort=T) %>% 
  dplyr::select(gene) %>% 
  unlist() %>%
  .[1:8] 

coeff.long.mat.ordered <-  
  coeff.long.mat %>%
  dplyr::filter(gene %in% order_alt, coeff!=0) %>% 
  dplyr::mutate(gene = factor(gene, levels = order_alt), int = factor(int, levels = rev(order_comb)))  %>% 
  mutate(direction = ifelse(coeff>0, "Positive", "Negative")) %>% 
  separate(int, sep=" \\+ ", into=c("Drug", "Stimulus"), remove = FALSE)

Order_Stim <- 
  coeff.long.mat.ordered %>% 
  dplyr::filter(coeff!=0) %>% 
  dplyr::group_by(Stimulus) %>% 
  dplyr::count(sort=T) %>% 
  dplyr::select(Stimulus) %>% 
  unlist()

Order_Drug <- 
  coeff.long.mat.ordered %>% 
  dplyr::filter(coeff!=0) %>% 
  dplyr::group_by(Drug) %>% 
  dplyr::count(sort=T) %>% 
  dplyr::select(Drug) %>% 
  unlist() %>% 
  rev()

coeff.long.mat.ordered <-
  coeff.long.mat.ordered %>%
  mutate(Stimulus=factor(Stimulus, levels = Order_Stim), 
         Drug=factor(Drug, levels = Order_Drug))
  

cyt_labels = c("TGF-b1"="TGF-\u03B21", "sCD40L+IL-4"="sCD40L + IL4", "IL-1b"="IL1\u03B2", "IL-4"="IL4", "IL-6"="IL6","IL-15"="IL15","IL-10"="IL10", "IL-21"="IL21","IL-2"="IL2", "Interferon gamma"= "Interferon \u03B3", "SDF-1a"="SDF-1\u03B1", "CpG ODN"="CpG ODN", "Resiquimod"="Resiquimod", "sCD40L"="sCD40L", "HS-5 CM"="HS-5 CM", "soluble anti-IgM"="soluble anti-IgM", "BAFF"="BAFF")

gene_labels = c("IGHV.status" = "IGHV status", "del11q" = "del(11q)", "del13q" = "del(13q)", "del17p" = "del(17p)", "trisomy12" = "trisomy 12", "TP53" = "TP53", "SF3B1" = "SF3B1", "ATM" = "ATM")

coeff.long.mat.ordered %>% 
  ggplot(aes(y=Drug, x=gene)) +
  geom_tile(aes(fill=coeff),color = "white") +
  scale_fill_gradientn(colors=c(rep(palblues[1:4],each=2),
                                "white", 
                                rep(palreds[5:8], each=2)),  
                       limits=c(-0.8,.8)) +
  t1 +
  theme(axis.text.x = element_text(angle = 35, hjust = 1, vjust = 1, size = 18),
        axis.text.y = element_text(size = 18),
        axis.title.x = element_blank(),
        axis.title.y = element_blank(),
        axis.ticks = element_blank(),
        axis.ticks.x =element_blank(),
        panel.background = element_rect(color = "black", fill=NA), 
        axis.line.x = element_blank(),
        axis.line.y = element_blank(),
        legend.key.height=unit(0.5, "cm"),
        legend.key.width=unit(2, "cm"),
        legend.title = element_text(face='bold', hjust = 0, size=fontsize+2),
        legend.position = "top",
        legend.key = element_blank(),
        legend.text = element_text(size=fontsize),
        legend.background = element_rect(color = NA),
        strip.text.y.left = element_text(size=18, angle = 0, face="bold"),
        strip.background = element_blank())+
  
  
  labs(fill = expression(beta["int"]))+
  scale_y_discrete(position = "right")+
  scale_x_discrete(labels = gene_labels) +
  facet_grid(Stimulus~.,  scales = "free_y",  space="free_y", switch = "both", labeller =labeller( Stimulus=cyt_labels))
  

```

This analysis revealed that 60/204 drug - stimulus interactions were modulated by at least one genetic feature (Figure \@ref(fig:drugCytGeneIntMap),  Appendix Figure \@ref(fig:drugcytGeneIntAll), Bruch and Giles et al. 2021). A positive coefficient here indicates that the presence of the genetic feature is associated with more positive $\beta_{int}$, in other words, the viability with the drug and stimulus is higher than expected in the presence of the genetic feature. 

Applying this broad scale screening approach established some wider trends. Firstly, trisomy 12 and IGHV status impacted the largest number of drug - stimulus interactions, indicating that their impact on stimulus and drug response individually also extends to drug-stimulus interactions. Secondly, out of all the stimuli, IL4 and sCD40L + IL4-based interactions were modulated by the largest number of features. IL4 alone increased viability uniformly across genetic backgrounds; its interaction with drugs may be more influenced by genetic features. Notably, the presence of trisomy 12 acted to increase IL4-induced drug resistance, in almost all drugs for which $\beta_{int}$>0.05. 

### Patient - specific drug - stimulus interactions of clinical interest {#drug-stimulus-gene-predictor-profile}
The multivariate modelling approach outlined above was a valuable screening tool. Using the results of this screening approach, I next examined each individual hit to establish which drug - stimulus - gene interactions were the most biologically interesting. This analysis highlighted several interactions that may have clinical importance and may warrant further investigation (Bruch and Giles et al. 2021).

The first of these concerned the interaction between fludarabine (purine analogue) and CpG ODN (TLR9). The value of $\beta_{int}$ for fludarabine and CpG ODN was modulated by six genetic factors; most strongly by IGHV status, del(11q) and trisomy 12 (Figure \@ref(fig:FluCpGExA), Bruch and Giles et al. 2021).

(ref:FluCpGExA) "Genetic features that modulate the size of $\beta_{int}$ between fludarabine (purine analogue) and CpG ODN (TLR9). Scatter plot, heatmap and bar plots indicate inputs and outputs of equation \@ref(eq:drugCytGeneInt). Bar plots indicate size and sign of coefficients assigned to genetic features named on right. Scatter plot depicts $\beta{int}$ values for each patient sample i.e.response matrix.  Heatmap tiles indicate mutation status for the selected genetic features (i.e. feature matrix) corresponding to same sample in scatter plot below, to show how size of $\beta{int}$ varies with each feature." See Methods section \@ref(drug-stimulus-gene-interactions-method). _Figure and caption from Bruch and Giles et al. 2021._


```{r FluCpGExA, fig.cap='(ref:FluCpGExA)', message = FALSE,  echo = FALSE, fig.height = 6, fig.width = 11, fig.align="center", out.width = '70%', dev = 'cairo_pdf'}


# Plot predictor profile for Ibrutinib + IL4

grid.arrange(grobs = Lasso_Plots_Fig6["Fludarabine + CpG ODN"], ncol = 1)

```

I visualised sample responses to CpG ODN and fludarabine stratified by two of these features: IGHV status and trisomy 12 (Figure \@ref(fig:FluCpGExB), Bruch and Giles et al. 2021). In IGHV-M non-trisomy 12 samples, TLR stimulation increased fludarabine efficacy. In samples that were either IGHV-U or trisomy 12, the reverse was true and TLR stimulation induced resistance to fludarabine (Bruch and Giles et al. 2021). 

(ref:FluCpGExB)  "Beeswarm-boxplots of log-transformed viability values for fludarabine (purine analogue) and CpG ODN (TLR9) single and combinatorial treatments, faceted by IGHV status and trisomy 12 status. P-values from paired Students t-tests." _Figure and caption from Bruch and Giles et al. 2021._

```{r FluCpGExB, fig.cap='(ref:FluCpGExB)', message = FALSE,  echo = FALSE, fig.height=6, fig.width = 6, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}


#set facet labels 
tri12.labs <- c("0" = "Non-\ntrisomy 12", "1" = "trisomy 12")
ighv.labs <- c("U" = "IGHV-U", "M" =  "IGHV-M")

#join viability and genetic data tables
  left_join(df_complete, patMeta, by = "PatientID") %>% 
  #filter for drug:stimulus combinations of interest, make sure no NAs in genetic data
  dplyr::filter(Drug_Concentration %in% c("High", "None"),
    DCK%in%c("DMSO:CpG ODN","Fludarabine:CpG ODN","Fludarabine:No Cytokine"),
                !is.na(trisomy12),
                !is.na(IGHV.status)) %>%
    mutate(DCK=factor(DCK, levels=c("DMSO:CpG ODN","Fludarabine:No Cytokine","Fludarabine:CpG ODN"))) %>% 
  
  #plot treatment combination against viability 
  ggplot(aes(x = DCK,y = Log,color= DCK))+
  geom_hline(yintercept = 0)+
  geom_boxplot()+
  geom_beeswarm(cex=1.5) +
  guides(color="none", shape="none")+
  #add p values
  stat_compare_means(method = "t.test",
                     label.y.npc = 0.8, 
                     paired = TRUE, 
                     comparisons = list(c(1,3), c(2,3)),
                     step.increase=0.2, 
                     size=3) +
  xlab("") +
  ylab("Log(Viability)") +
  #facet by trisomy 12 and IGHV status
  facet_grid(vars(trisomy12), 
             vars(IGHV.status),
             labeller = labeller(trisomy12 = tri12.labs, IGHV.status = ighv.labs))+
  
  scale_x_discrete(labels=c("DMSO:CpG ODN"="CpG ODN",
                            "Fludarabine:No Cytokine"="Fludarabine",
                            "Fludarabine:CpG ODN"="Fludarabine \n+ CpG ODN "))+  
    
  scale_color_manual(values=c(colors[4], colors[5],colors[3])) + 
  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))+
  t2 +
  theme(strip.background =element_rect(fill=NA),
        strip.text = element_text(size=fontsize+4, face="bold"),
        strip.text.y = element_text(angle = 0),
        axis.text.x = element_text(size=fontsize+4, angle = 35, hjust = 1, vjust = 1))
 


```

I observed a similar effect with other chemotherapeutic drugs. Nutlin-3a is an MDM2 inhibitor, which in our dataset showed higher efficacy in IGHV-U, trisomy12, and del(11q) CLLs (Figure \@ref(fig:drugGeneIntMap), Figure \@ref(fig:drugGeneIntEx), Appendix Figure \@ref(fig:NutlinPredictors)). TLR stimulation reduced nutlin-3a toxicity in these same genetic backgrounds, i.e  nutlin-3a efficacy in the context of CpG ODN stimulation was lower in IGHV-U, trisomy 12 and del(11q) samples (Figure \@ref(fig:nutlinCpgEx), Figure \@ref(fig:TLRCpGIGHVdel11q)).  This observation underlines the need to observe _in vitro_ drug activity  in the context of microenvironmental signals [CHECK].

(ref:nutlinCpgEx) Predictor profile depicting genetic features that modulate the size of $\beta{int}$ between nutlin-3a (MDM2) and CpG ODN (TLR9).  Plot generated as in Figure \@ref(fig:FluCpGExA). See Methods section \@ref(drug-stimulus-gene-interactions-method).


```{r nutlinCpgEx, fig.cap='(ref:nutlinCpgEx)', message = FALSE,  echo = FALSE, fig.height=6, fig.width = 10, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

# Plot predictor profile for Nutlin + CpG ODN

grid.arrange(grobs = Lasso_Plots_Fig6["Nutlin-3a + CpG ODN"], ncol = 1)

```


(ref:TLRCpGIGHVdel11q) Beeswarm-boxplot showing control-normalised log transformed viability values, after treatment with CpG ODN (TLR9) and nutlin-3a (MDM2), stratified by del(11q) and IGHV status. WT indicates sample is not annotated as del(11q). p-values from Students t-tests.

```{r TLRCpGIGHVdel11q, fig.cap='(ref:TLRCpGIGHVdel11q)', message = FALSE,  echo = FALSE, fig.height = 4, fig.width = 6,  fig.align="center", out.width = '80%', dev = 'cairo_pdf'}

 left_join(df_complete, patMeta, by = "PatientID") %>%
  dplyr::filter(Cytokine=="CpG ODN",
                Drug== "Nutlin-3a",
                Drug_Concentration== "High", 
                !is.na(del11q),
                !is.na(IGHV.status)) %>%
         
  ggplot(aes(x=interaction(del11q, IGHV.status),
                    y=Log,
                    color=(del11q)))+
  geom_hline(yintercept = 0)+
  geom_boxplot()+
  geom_beeswarm(cex=1.5) +
  guides(color="none", shape="none")+
  scale_x_discrete(labels=c("0.M"="IGHV-M\n WT",
                            "0.U"="IGHV-U\n WT",
                            "1.M"="IGHV-M\n del(11q)",
                            "1.U"="IGHV-U\n del(11q)"))+
  stat_compare_means(method = "t.test",
                     label.x.npc = "center", 
                     comparisons = list( c(1,2), c(3,4), c(1,3)),
                     size=3)+
  xlab("") +
  ylab("Log(Viability)") +
  ggtitle("Nutlin-3a + CpG ODN") +
  scale_color_manual(values=c(colors[1], colors[2])) + 

  t1+
  theme(axis.text.x = element_text(angle = 45, vjust =1))


```

The modelling approach also identified interactions where the differential size of $\beta_{int}$ was driven by the single treatment effect, rather than the combination. For example, if a drug was more efficacious in a certain genetic background, but a stimulus nonetheless eradicated drug toxicity, the size of the interaction was larger in this genetic background. This is because the increase in expected viability in these samples (quantified by $\beta_{int}$) would be even higher than in other genetic backgrounds. 

This was the case with ibrutinib + IL4 for instance (Bruch and Giles et al. 2021). Ibrutinib showed higher efficacy in trisomy 12 and IGHV-U samples and IL4 induced complete resistance to ibrutinib independently of genetic background. Thus in trisomy 12 and IGHV-U samples treated with ibrutinib, the increase in viability in the context of IL4 stimulation was larger than in non-trisomy 12 and IGHV-M samples, and thus these features were assigned positive coefficients. 

This result highlights that IL4-induced resistance may be a broad spectrum resistance mechanism, even in tumours with molecular features that are associated with higher ibrutinib efficacy. IL4 signalling may represent a drug resistance mechanism common to many CLLs, in a disease known for its molecular and genetic heterogeneity, though more *in vivo* evidence is required here (Figure \@ref(fig:ibrIl4ExA), Figure \@ref(fig:ibrIl4ExB), (Bruch and Giles et al. 2021).

(ref:ibrIl4ExA) Predictor profile depicting genetic features that modulate the size of $\beta_{int}$ between ibrutinib (BTK) and IL4. Plot generated as in Figure \@ref(fig:FluCpGExA). See also Methods section \@ref(drug-stimulus-gene-interactions-method). _Figure from Bruch and Giles et al. 2021._

(ref:ibrIl4ExB) "Beeswarm-boxplots of log-transformed viability values for ibrutinib (BTK) and IL4 single and combinatorial treatments, faceted by IGHV status and trisomy 12 status. P-values from paired Students t-tests." _Figure and caption from Bruch and Giles et al. 2021._

```{r ibrIl4ExA, fig.cap='(ref:ibrIl4ExA)', message = FALSE,  echo = FALSE, fig.height=6, fig.width = 10, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

# Plot predictor profile for Ibrutinib + IL4

grid.arrange(grobs = Lasso_Plots_Fig6["Ibrutinib + IL-4"], ncol = 1)
```


```{r ibrIl4ExB, fig.cap='(ref:ibrIl4ExB)', message = FALSE,  echo = FALSE, fig.height=6, fig.width = 6, fig.align="center", out.width = '60%', dev = 'cairo_pdf'}

  left_join(df_complete, patMeta, by = "PatientID")%>%
  #filter for drug:stimulus combinations of interest, make sure no NAs in genetic data
  dplyr::filter(DCK%in%c("DMSO:IL-4","Ibrutinib:IL-4","Ibrutinib:No Cytokine"),
                Drug_Concentration %in% c("High", "None"),
                !is.na(trisomy12),
                !is.na(IGHV.status)) %>%
     mutate(DCK = factor(DCK, levels=c("DMSO:IL-4",
                                      "Ibrutinib:No Cytokine",
                                      "Ibrutinib:IL-4"))) %>%
  
  #plot treatment combination against viability 
  ggplot(aes(x = DCK,y = Log,color= DCK))+
  geom_hline(yintercept = 0)+
  geom_boxplot()+
  geom_beeswarm(cex=1.5) +
  guides(color="none", shape="none")+
  #add p values
  stat_compare_means(method = "t.test",
                     label.y.npc = 0.8, 
                     paired = TRUE, 
                     comparisons = list(c(1,3), c(2,3)),
                     step.increase=0.2, 
                     size=3) +
  xlab("") +
  ylab("Log(Viability)") +
  #facet by trisomy 12 and IGHV status
  facet_grid(vars(trisomy12), 
             vars(IGHV.status),
             labeller = labeller(trisomy12 = tri12.labs, IGHV.status = ighv.labs))+
  
  scale_x_discrete(labels=c("DMSO:IL-4"="IL4",
                            "Ibrutinib:No Cytokine"="Ibrutinib",
                            "Ibrutinib:IL-4"="Ibrutinib \n+ IL4"))+  
    
  scale_color_manual(values=c(colors[4], colors[5], colors[3])) + 
  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))+
  t2 +
  theme(strip.background =element_rect(fill=NA),
        strip.text = element_text(size=fontsize+4, face="bold"),
        strip.text.y = element_text(angle = 0),
        axis.text.x = element_text(size=fontsize+4, angle = 35, hjust = 1, vjust = 1))
 


```

## The impact of IL4 on drug response  {#il4-ibrutinib}
### IL4 induces resistance to BCR inhibitors and chemotherapeutics {#IL4resistance}
(refer to htis further up as well, elaborate here)

(Figure \@ref(fig:IL4Res)). 

(ref:IL4Res) Examples of IL4-induced resistance. Plots show log transformed viability values with each treatment, for all samples. Matching samples are linked across treatments. Black and blue horizontal lines indicate the predicted viability from the linear model for each single and combinatorial treatment. In combinatorial treatment, both the expected viability based on the additive effect of drug and stimulus (blue), and the viability with interaction (black) are shown, to indicate the impact of the interaction. Plots show IL4 + (A) idelalisib (Pi3K), (B) PRT062607 (SYK) (C) everolimus (mTOR), and (D) selumetinib  (MEK). See Methods section \@ref(drug-stimulus-linear-model-method). _Figure from Bruch & Giles et al. 2021._

```{r IL4Res, fig.cap='(ref:IL4Res)', message = FALSE,  echo = FALSE, fig.height=6.6, fig.width = 12, fig.align="center", out.width = '80%', dev = 'cairo_pdf'}


(wrap_plots(plotList$`treatment_drug:Idelalisib:treatment_cytokine:IL-4`) + wrap_plots(plotList$`treatment_drug:PRT062607:treatment_cytokine:IL-4`)) /
  
(wrap_plots(plotList$`treatment_drug:Everolimus:treatment_cytokine:IL-4`) +
  
wrap_plots(plotList$`treatment_drug:Selumetinib:treatment_cytokine:IL-4`)) + plot_annotation(tag_levels = c("A"))

```

This is accross genetic backgrounds  (refer to above)

### Levels of _in vivo_ IL4 signalling correlate with outcome {#il4-ihc}
(Figure \@ref(fig:IL4ihc)). 

(ref:IL4ihc) See Methods sections  \@ref(il4-ihc-exp) and \@ref(il4-ihc-method).  _Figure from Bruch & Giles et al. 2021._

```{r message = FALSE,  echo = FALSE, results = FALSE}

STAT6_beeswarm <-
  ihc_patient_data %>%
  #get staining intensities for pSTAT6 only
  dplyr::filter(Stain %in% c("pSTAT6")) %>%
  #plot staining intensity, stratified by CLL / Healthy LN
  ggplot(aes(x = Tissue, y = Intensity)) +
  geom_boxplot() +
  geom_beeswarm(aes(color=Tissue), alpha=1, cex=2) +
  #add p values
  stat_compare_means(method = "t.test", comparisons=list(c(1,2)), size=6) +
  scale_color_manual(values = c(palreds[8], palblues[2])) +
  xlab("") +
  ylab("Mean pSTAT6 Staining Intensity") +
  guides(color = "none") +
  scale_x_discrete(labels = c( "CLL"="CLL-infiltrated \nlymph nodes", "LK"="Non-neoplastic \nlymph nodes")) +
  coord_cartesian(clip = "off") +
  #add preset theme 2
  t2



##read image
CLL_LN <- cowplot::ggdraw() +
  cowplot::draw_image("/Volumes/huber/users/giles/thesis/index/data/CLL1_I3_pSTAT6.png", scale = 1)


##read image
healthy_LN <- cowplot::ggdraw() + 
  cowplot::draw_image("/Volumes/huber/users/giles/thesis/index/data/LK2_B4_pSTAT6.png", scale = 1)


#Define stains for which want to visualize TTT
stains <- c("pSTAT6")

#Get optimal cut offs 
stats <- lapply(stains, function(stn){
 
  survival <- dplyr::select(ihc_surv_data, PatientID, Tissue, Diagnosis, Sex, TTT, treatedAfter, all_of(stn))
  colnames(survival) <- c("PatientID", "Tissue", "Diagnosis", "Sex", "TTT", "treatedAfter", "target")
  
  #Run test to obtain cut off threshold for high pSTAT6 versus low pSTAT6
  maxtest <- maxstat.test(Surv(TTT, treatedAfter)~ target, 
                          data = survival,
                          smethod = "LogRank",
                          alpha = NULL)
  
  cutpoint <- maxtest$estimate
  
 })
  
  names(stats) <- c("pSTAT6")
  
                                          
  #Annotate by cutoff point
  survdf <- mutate(ihc_surv_data, 
                   phosphoSTAT6 = ifelse(pSTAT6 < stats$pSTAT6, "low", "high"))
  
                                      
  #fit survival models
  f_pstat6 <- survfit(Surv(TTT, treatedAfter) ~ phosphoSTAT6, data= survdf)
  
  fits <- list(pSTAT6 = f_pstat6)
  
  #Make plot
  gg = ggsurvplot_list(fits, 
                       survdf,  
                       pval=TRUE, 
                       palette=c(palreds[8],palreds[3]),  
                       risk.table = TRUE, legend.title = "", 
                       ggtheme = t2, 
                       legend.labs =list(c("High", "Low"),c("High", "Low")),  
                       xlab="Time in years", ylab="\nTime to next treatment (probability)", title= "", legend = "bottom")
  
#get plot for pSTAT6 stain
pSTAT6_KM <-
  wrap_elements(gg$pSTAT6$plot + 
                  theme(axis.title.x = element_blank()) +
                  gg$pSTAT6$table + 
                  theme(plot.title = element_blank()) +
                  plot_layout(ncol = 1, heights = c(85, 15)))
  
```

```{r, IL4ihc, fig.cap='(ref:IL4ihc)', message = FALSE,  echo = FALSE, results = FALSE,  fig.height=7, fig.width = 16, fig.align="center", out.width = '80%', dev = 'cairo_pdf'}

design1 <-"
  ABD
  ACD
  "
tp <- theme(plot.tag = element_text(size = 30, vjust = 1, face="plain"))

ihc <-
  wrap_elements(STAT6_beeswarm) + tp +
  CLL_LN + tp +
  healthy_LN + tp +
  pSTAT6_KM + tp +
  
  plot_layout(design = design1, widths = c(1,0.7,1), heights =c(0.8, 1))+
  plot_annotation(tag_levels = "A", theme = theme(title=element_text(size = 20)))
  
ihc
```


##IBET-762 rescues ibrutinib toxicity in the context of IL4 stimulation 


(Figure \@ref(fig:ibetil4)). 

(ref:ibetil4)  

```{r ibetil4, fig.cap='(ref:ibetil4)', message = FALSE,  echo = FALSE, fig.height=6.6, fig.width = 12, fig.align="center", out.width = '80%', dev = 'cairo_pdf', eval = FALSE}

#this plto doesnt exist so have to make from scratch 
wrap_plots(plotList$`treatment_drug:I-BET 762:treatment_cytokine:IL-4`)

```

### Multiomics profiling of IL4, ibrutinib and IBET-762 treated CLL samples {#ch7-overview}
To investigate IL4 induced resistance,  I used the following dataset (Figure \@ref(fig:ibribetil4)). 


(ref:ibribetil4) Overview of the approach for the validation experiments. Four CLL PBMC samples were incubated with ibrutinib, IL4 and IBET-762, both individually and in all combinations. Samples were taken for ATAC sequencing and RNA sequencing after 6 hours and for mass spectrometry after 24 hours. Sequencing data was processed for downstream analysis, outline in Chapter \@ref(chapter7). 

```{r ibribetil4, out.width = "100%", fig.cap='(ref:ibribetil4)', echo = FALSE}

knitr::include_graphics("figures/ibribetil4.eps")

```

### STAT6 is necessary for IL4 induced resistance {#treated-atacseq}
{#treated-rnaseq}

IL4 and IL4 + ibrutinib ATAC and RNA sew

STAT6 inhibition 

(Figure \@ref(fig:stat6inhib)). 


(ref:stat6inhib)


```{r stat6inhib, out.width = "100%", fig.cap='(ref:stat6inhib)', echo = FALSE}

IbrStat6iComb %>%
  mutate(trisomy12=as.character(trisomy12)) %>% 
  
  ggplot( aes(x = interaction(Drug_2, Drug_1, Cytokine), y = Log))+
  geom_boxplot(outlier.shape = NA)+
  geom_beeswarm(aes(color = IGHV.status, shape = trisomy12), cex = 2) +
  geom_hline(yintercept = 0) +
  t2 +
  theme(legend.key = element_blank()) +
  ylab("\n\n\nViability") +
  ggtitle(paste0("")) +
  xlab("") +
  ylab("Log(Viability)") +
  scale_color_manual(values = c("M"=palblues[1], "U"=palreds[8]), na.value="#7a7c80") +
  scale_x_discrete(labels=c("DMSO.DMSO.IL-4" = "IL4",
                            "Ibrutinib.DMSO.IL-4" = "IL4\nIbrutinib",
                            "DMSO.STAT6i.IL-4" = "IL4\nSTAT6i",
                            "Ibrutinib.STAT6i.IL-4" = "IL4\nIbrutinib\nSTAT6i"))+
  labs(shape = "trisomy 12", color = "IGHV status")

```

(fix heading links)
### ANythign esle STAT6 targets etc



### IBET-762 inhibits STAT6 targets, without affecting STAT6 activity 
ATAC - doesnt inihibt STAT6 acitivty
RNA - does inhibit STAT6 targets
anything more sepcific 
Futrue work 

## Summary 
This work aims to integrate the effects of mutations and the microenvironment in drug response in CLL, at a large scale. The results demonstrate the value of using multi-omic data (consisting of both observational data on patient samples and perturbation data generated through screening) to generate more complex biological insights that can guide further clinical studies. 

More specifically, these results identify a number of novel drug-resistance pathways, including the impact of IFN$\gamma$ on ibrutinib. Trisomy 12 is identified as an important modulator of drug response. Additionally, these results show that certain resistance pathways are context-dependent. In particular, the ability of TLR stimulation to induce resistance to chemotherapeutics is modulated by IGHV and trisomy 12 status. 

Overall these findings illustrate how the presence of known genetic alterations determines how drugs and external stimuli interact with each other.

```{r remove06}

#clear data
rm(list = ls())

```
