# Abstract {-}
Chronic Lymphocytic Leukaemia is a malignancy of mature B lymphocytes, characterised by the accumulation of malignant cells within the blood, bone marrow and lymph nodes. The disease follows a highly heterogeneous course driven by multiple factors, including molecular features, signalling via the B cell receptor (BCR) and interactions with non-neoplastic cells within the lymphoid tissues (known as the tumour microenvironment). Many individual drivers of CLL pathogenesis have been well-characterised, and the disease represents a success story in linking advances in molecular understanding to improvements in clinical approaches. Building on this, multiomics promises to further our understanding of disease biology and offers the ability to study CLL at multiple levels, via both molecular data and perturbation data from chemical or genetic screening. Current work investigating the role of the microenvironment in CLL is limited to smaller-scale studies and there is a clinical need to generate a systematic understanding of cell-intrinsic and cell-extrinsic causes of disease progression and drug resistance.  

In this thesis, I investigate the integrative impact of these features on CLL biology. I employ a dataset consisting of perturbation data of CLL primary samples treated with drugs and microenvironmental stimuli, along with molecular profiles of the same samples. The perturbation assay was performed using 12 drugs co-applied with 17 individual microenvironmental stimuli on CLL peripheral blood mononucleated cells (PBMCs, n = 192). The molecular profiles included whole-exome sequencing, DNA-methylation, RNA-sequencing and copy number variant data of matching tumour samples. The dataset is complemented with *in vivo* data on patient clinical outcomes, and immunohistochemistry (IHC) of patient lymph nodes stained for pSTAT6, pIRAK4 and STAT6. I apply statistical inference and regression modelling to integrate multiple data types, with the aim of finding new drivers and mechanisms of drug resistance in CLL. 
 
The work revealed several key findings. Firstly, CLL samples demonstrate heterogeneous responses to the panel of microenvironmental stimuli and IL4 and TLR signalling have the strongest impact. The response profiles  delineate four patient subgroups that have differential disease progression dynamics and molecular profiles, suggesting the dynamics of the microenvironment is important to disease outcomes. Secondly,  a systematic survey of genetic determinants of microenvironmental response identifies trisomy 12 as a modulator of microenvironmental stimulation. The data suggest that the activity of the transcription factors Spi-B and PU.1 may mediate this effect. Thirdly, I generate a map of interactions between microenvironmental signalling pathways and drugs, and identify novel drug-resistance pathways including the effect of IFN$\gamma$ on ibrutinib toxicity. I demonstrate how these drug - microenvironment interactions can be further modulated by molecular features, and identify context-dependent drug resistance mechanism that occur in specific genetic backgrounds. For example, TLR activity induced resistance to fludarabine in IGHV-U and trisomy 12 CLLs, but not in IGHV-M. Finally, the *in vivo* relevance of these findings is established within CLL-infiltrated lymph nodes, which show higher levels of IL4 and TLR signalling compared to non-neoplastic samples (p<0.001). Elevated IL4 signalling in CLL-infiltrated lymph nodes correlates with poorer outcomes (p=0.038). 

This data set represents an important resource for exploring the impact of these features. The presented dataset and associated analysis have been published as part of an [online git repository](https://github.com/Huber-group-EMBL/CLLCytokineScreen2021) and interactive [shiny app](https://www.imbi.uni-heidelberg.de/dietrichlab/CLL_Microenvironment/) to provide open access to the broader community. Collectively, these results demonstrate the integrative impact of molecular features and microenvironmental stimuli on drug response and propose potential novel pathogenic mechanisms and causes of drug resistance, along with strategies for targeting these in future work. 